{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Today's session\n",
    "\n",
    "In today's session, we will get to know some of the genomics tools that were introduced in the theoretical lectures. In particular, we will see how to work with sequences, align them, and so on. To do this, we will be using `Biopyton`, a comprehensive Python module that allows you to do these and many other things. For a complete tour of `Biopython`, take a look at the [complete tutorial](http://biopython.org/DIST/docs/tutorial/Tutorial.html).\n",
    "\n",
    "In the final exercise for this class you will be working on the following case study:\n",
    "\n",
    ">A few weeks ago, a patient was admitted to the hospital where you work. The patient was coughing, and had difficulty breathing, as well as a fever. Your colleagues Oscar and Maria, from the genomics platform at the hospital, have been able to isolate a viral genome from the patient; this virus is thought to be responsible for the disease. You are expected to identify what the virus is and, thus, provide a diagnosis. You are also expected to compare the virus' genome with the genome of similar viruses to try to understand where the disease came from and how is it different from known diseases. \n",
    "\n",
    "# What is Biopython?\n",
    "\n",
    "The Biopython Project is an international association of developers of freely available Python tools for computational molecular biology.\n",
    "\n",
    "![Biopython](Media/biopython_logo.png)\n",
    "\n",
    "\n",
    "The main Biopython releases have lots of functionality, including:\n",
    "\n",
    "* The ability to parse bioinformatics files into Python utilizable data structures, including support for the following formats:\n",
    "    - Blast output – both from standalone and WWW Blast\n",
    "    - Clustalw\n",
    "    - FASTA\n",
    "    - GenBank\n",
    "    - PubMed and Medline\n",
    "    - ExPASy files, like Enzyme and Prosite\n",
    "    - SCOP, including ‘dom’ and ‘lin’ files\n",
    "    - UniGene\n",
    "    - SwissProt \n",
    "* Files in the supported formats can be iterated over record by record or indexed and accessed via a Dictionary interface.\n",
    "* Code to deal with popular on-line bioinformatics destinations such as:\n",
    "    - NCBI – Blast, Entrez and PubMed services\n",
    "    - ExPASy – Swiss-Prot and Prosite entries, as well as Prosite searches \n",
    "* Interfaces to common bioinformatics programs such as:\n",
    "    - Standalone Blast from NCBI\n",
    "    - Clustalw alignment program\n",
    "    - EMBOSS command line tools \n",
    "* A standard sequence class that deals with sequences, ids on sequences, and sequence features.\n",
    "* Tools for performing common operations on sequences, such as translation, transcription and weight calculations.\n",
    "* Code for dealing with alignments, including a standard way to create and deal with substitution matrices.\n",
    "* Code making it easy to split up parallelizable tasks into separate processes.\n",
    "* GUI-based programs to do basic sequence manipulations, translations, BLASTing, etc.\n",
    "* Extensive documentation and help with using the modules, including the tutorial, on-line wiki documentation, the web site, and the mailing list.\n",
    "* Integration with BioSQL, a sequence database schema also supported by the BioPerl and BioJava projects."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Biopython sequences\n",
    "\n",
    "We will start with a quick introduction to the `Biopython` class for dealing with sequences. Most of the time, when we think of sequences what we have in my mind is a string of letters like 'AGTACACTGGT'. In `Biopython` you can create a `Seq` instance from such a sequence as follows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AGTACACTGGT')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "my_seq = Seq(\"AGTACACTGGT\")\n",
    "display(my_seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sequence alphabet\n",
    "\n",
    "The sequence above seems to represent a fragment of DNA (that is, a sequence of bases), but it could also be a protein, since A, G, T, and C could also represent aminoacids. To specify the type of sequence we are dealing with, the `Seq` class has an `alphabet` attribute. Let's see how it works:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AGTACACTGGT', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "IUPACUnambiguousDNA()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from Bio.Alphabet import IUPAC\n",
    "\n",
    "my_seq = Seq(\"AGTACACTGGT\", alphabet=IUPAC.unambiguous_dna)\n",
    "display(my_seq)\n",
    "display(my_seq.alphabet)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transcription and  translation\n",
    "\n",
    "So, *now* we know that we are dealing with a DNA sequence. And, in fact, we can do things with this sequence that we could not do with a sequence of aminoacids. For example, assuming that the `coding_dna` sequence corresponds to the 5'-to-3′ strand of the DNA, we can obtain the corresponding 5'-to-3' complementary strand:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('CTATCGGGCACCCTTTCAGCGGCCCATTACAATGGCCAT', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "coding_dna = Seq(\"ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\", IUPAC.unambiguous_dna)\n",
    "display(coding_dna.reverse_complement())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Or, we can **transcribe** this DNA sequence into an RNA sequence:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "messenger_rna = coding_dna.transcribe()\n",
    "display(messenger_rna)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Translation is a bit tricker, because codons are translated differently in different organisms and even different cellular compartments. Here is what happens if we translate and RNA sequence without any parameters:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('MAIVMGR*KGAR*', HasStopCodon(IUPACProtein(), '*'))"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "messenger_rna.translate()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You should notice in the above protein sequence that, in addition to the end stop character, there is an internal stop as well. This was a deliberate choice of example, as it gives an excuse to talk about some optional arguments, including different translation tables (genetic codes).\n",
    "\n",
    "The translation tables available in Biopython are based on those from [the NCBI](https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi). By default, translation will use the standard genetic code (NCBI table id 1). But suppose we are dealing with a mitochondrial sequence---we need to tell the translation function to use the relevant genetic code instead:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('MAIVMGRWKGAR*', HasStopCodon(IUPACProtein(), '*'))"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "messenger_rna.translate(table=\"Vertebrate Mitochondrial\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Otherwise, sequences are similar to strings\n",
    "\n",
    "Other than the situations described above, most of the time `Seq` instances can be dealt with just like strings. For example, we can extract the elements of a sequence using indices, or slice a sequence as we would slice a string: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'G'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "Seq('GATGGGCC', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of the sequence:       32\n",
      "Number of As in the sequence: 10\n"
     ]
    }
   ],
   "source": [
    "# Seq instances work, mostly, as strings\n",
    "my_seq = Seq(\"GATCGATGGGCCTATATAGGATCGAAAATCGC\", IUPAC.unambiguous_dna)\n",
    "display(my_seq[4])\n",
    "display(my_seq[4:12])\n",
    "print('Length of the sequence:      ', len(my_seq))\n",
    "print('Number of As in the sequence:', my_seq.count('A'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## More on the `Seq` class\n",
    "\n",
    "This is a good opportunity to learn a new Python trick. To get more information about a given class, you can use the `help()` function. For example, to learn more about the `Seq`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class Seq in module Bio.Seq:\n",
      "\n",
      "class Seq(builtins.object)\n",
      " |  Seq(data, alphabet=Alphabet())\n",
      " |  \n",
      " |  Read-only sequence object (essentially a string with an alphabet).\n",
      " |  \n",
      " |  Like normal python strings, our basic sequence object is immutable.\n",
      " |  This prevents you from doing my_seq[5] = \"A\" for example, but does allow\n",
      " |  Seq objects to be used as dictionary keys.\n",
      " |  \n",
      " |  The Seq object provides a number of string like methods (such as count,\n",
      " |  find, split and strip), which are alphabet aware where appropriate.\n",
      " |  \n",
      " |  In addition to the string like sequence, the Seq object has an alphabet\n",
      " |  property. This is an instance of an Alphabet class from Bio.Alphabet,\n",
      " |  for example generic DNA, or IUPAC DNA. This describes the type of molecule\n",
      " |  (e.g. RNA, DNA, protein) and may also indicate the expected symbols\n",
      " |  (letters).\n",
      " |  \n",
      " |  The Seq object also provides some biological methods, such as complement,\n",
      " |  reverse_complement, transcribe, back_transcribe and translate (which are\n",
      " |  not applicable to sequences with a protein alphabet).\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __add__(self, other)\n",
      " |      Add another sequence or string to this sequence.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> Seq(\"MELKI\", generic_protein) + \"LV\"\n",
      " |      Seq('MELKILV', ProteinAlphabet())\n",
      " |      \n",
      " |      When adding two Seq (like) objects, the alphabets are important.\n",
      " |      Consider this example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet.IUPAC import unambiguous_dna, ambiguous_dna\n",
      " |      >>> unamb_dna_seq = Seq(\"ACGT\", unambiguous_dna)\n",
      " |      >>> ambig_dna_seq = Seq(\"ACRGT\", ambiguous_dna)\n",
      " |      >>> unamb_dna_seq\n",
      " |      Seq('ACGT', IUPACUnambiguousDNA())\n",
      " |      >>> ambig_dna_seq\n",
      " |      Seq('ACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      If we add the ambiguous and unambiguous IUPAC DNA alphabets, we get\n",
      " |      the more general ambiguous IUPAC DNA alphabet:\n",
      " |      \n",
      " |      >>> unamb_dna_seq + ambig_dna_seq\n",
      " |      Seq('ACGTACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      However, if the default generic alphabet is included, the result is\n",
      " |      a generic alphabet:\n",
      " |      \n",
      " |      >>> Seq(\"\") + ambig_dna_seq\n",
      " |      Seq('ACRGT')\n",
      " |      \n",
      " |      You can't add RNA and DNA sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"ACGU\", generic_rna)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      \n",
      " |      You can't add nucleotide and protein sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_protein\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"MELKI\", generic_protein)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __contains__(self, char)\n",
      " |      Implement the 'in' keyword, like a python string.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna, generic_protein\n",
      " |      >>> my_dna = Seq(\"ATATGAAATTTGAAAA\", generic_dna)\n",
      " |      >>> \"AAA\" in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\") in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\", generic_dna) in my_dna\n",
      " |      True\n",
      " |      \n",
      " |      Like other Seq methods, this will raise a type error if another Seq\n",
      " |      (or Seq like) object with an incompatible alphabet is used:\n",
      " |      \n",
      " |      >>> Seq(\"AAA\", generic_rna) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      >>> Seq(\"AAA\", generic_protein) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __eq__(self, other)\n",
      " |      Compare the sequence to another sequence or a string (README).\n",
      " |      \n",
      " |      Historically comparing Seq objects has done Python object comparison.\n",
      " |      After considerable discussion (keeping in mind constraints of the\n",
      " |      Python language, hashes and dictionary support), Biopython now uses\n",
      " |      simple string comparison (with a warning about the change).\n",
      " |      \n",
      " |      Note that incompatible alphabets (e.g. DNA to RNA) will trigger a\n",
      " |      warning.\n",
      " |      \n",
      " |      During this transition period, please just do explicit comparisons:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> seq1 = Seq(\"ACGT\")\n",
      " |      >>> seq2 = Seq(\"ACGT\")\n",
      " |      >>> id(seq1) == id(seq2)\n",
      " |      False\n",
      " |      >>> str(seq1) == str(seq2)\n",
      " |      True\n",
      " |      \n",
      " |      The new behaviour is to use string-like equality:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq1 == seq2\n",
      " |      True\n",
      " |      >>> seq1 == \"ACGT\"\n",
      " |      True\n",
      " |      >>> seq1 == Seq(\"ACGT\", generic_dna)\n",
      " |      True\n",
      " |  \n",
      " |  __ge__(self, other)\n",
      " |      Implement the greater-than or equal operand.\n",
      " |  \n",
      " |  __getitem__(self, index)\n",
      " |      Return a subsequence of single letter, use my_seq[index].\n",
      " |      \n",
      " |      >>> my_seq = Seq('ACTCGACGTCG')\n",
      " |      >>> my_seq[5]\n",
      " |      'A'\n",
      " |  \n",
      " |  __gt__(self, other)\n",
      " |      Implement the greater-than operand.\n",
      " |  \n",
      " |  __hash__(self)\n",
      " |      Hash for comparison.\n",
      " |      \n",
      " |      See the __cmp__ documentation - this has changed from past\n",
      " |      versions of Biopython!\n",
      " |  \n",
      " |  __imul__(self, other)\n",
      " |      Multiply Seq in-place.\n",
      " |      \n",
      " |      Note although Seq is immutable, the in-place method is\n",
      " |      included to match the behaviour for regular Python strings.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq = Seq('ATG', generic_dna)\n",
      " |      >>> seq *= 2\n",
      " |      >>> seq\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __init__(self, data, alphabet=Alphabet())\n",
      " |      Create a Seq object.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - seq - Sequence, required (string)\n",
      " |       - alphabet - Optional argument, an Alphabet object from\n",
      " |         Bio.Alphabet\n",
      " |      \n",
      " |      You will typically use Bio.SeqIO to read in sequences from files as\n",
      " |      SeqRecord objects, whose sequence will be exposed as a Seq object via\n",
      " |      the seq property.\n",
      " |      \n",
      " |      However, will often want to create your own Seq objects directly:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF', IUPACProtein())\n",
      " |      >>> print(my_seq)\n",
      " |      MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\n",
      " |      >>> my_seq.alphabet\n",
      " |      IUPACProtein()\n",
      " |  \n",
      " |  __le__(self, other)\n",
      " |      Implement the less-than or equal operand.\n",
      " |  \n",
      " |  __len__(self)\n",
      " |      Return the length of the sequence, use len(my_seq).\n",
      " |  \n",
      " |  __lt__(self, other)\n",
      " |      Implement the less-than operand.\n",
      " |  \n",
      " |  __mul__(self, other)\n",
      " |      Multiply Seq by integer.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> Seq('ATG') * 2\n",
      " |      Seq('ATGATG')\n",
      " |      >>> Seq('ATG', generic_dna) * 2\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __ne__(self, other)\n",
      " |      Implement the not-equal operand.\n",
      " |  \n",
      " |  __radd__(self, other)\n",
      " |      Add a sequence on the left.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> \"LV\" + Seq(\"MELKI\", generic_protein)\n",
      " |      Seq('LVMELKI', ProteinAlphabet())\n",
      " |      \n",
      " |      Adding two Seq (like) objects is handled via the __add__ method.\n",
      " |  \n",
      " |  __repr__(self)\n",
      " |      Return (truncated) representation of the sequence for debugging.\n",
      " |  \n",
      " |  __rmul__(self, other)\n",
      " |      Multiply integer by Seq.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> 2 * Seq('ATG')\n",
      " |      Seq('ATGATG')\n",
      " |      >>> 2 * Seq('ATG', generic_dna)\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __str__(self)\n",
      " |      Return the full sequence as a python string, use str(my_seq).\n",
      " |      \n",
      " |      Note that Biopython 1.44 and earlier would give a truncated\n",
      " |      version of repr(my_seq) for str(my_seq).  If you are writing code\n",
      " |      which need to be backwards compatible with really old Biopython,\n",
      " |      you should continue to use my_seq.tostring() as follows::\n",
      " |      \n",
      " |          try:\n",
      " |              # The old way, removed in Biopython 1.73\n",
      " |              as_string = seq_obj.tostring()\n",
      " |          except AttributeError:\n",
      " |              # The new way, needs Biopython 1.45 or later.\n",
      " |              # Don't use this on Biopython 1.44 or older as truncates\n",
      " |              as_string = str(seq_obj)\n",
      " |  \n",
      " |  back_transcribe(self)\n",
      " |      Return the DNA sequence from an RNA sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> messenger_rna = Seq(\"AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG\",\n",
      " |      ...                     IUPAC.unambiguous_rna)\n",
      " |      >>> messenger_rna\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      >>> messenger_rna.back_transcribe()\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      Trying to back-transcribe a protein or DNA sequence raises an\n",
      " |      exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.back_transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be back transcribed!\n",
      " |  \n",
      " |  complement(self)\n",
      " |      Return the complement sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAG\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGCTATC', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-GD\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-GD', DNAAlphabet())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGctaT-CH', DNAAlphabet())\n",
      " |      \n",
      " |      Note in the above example, ambiguous character D denotes\n",
      " |      G, A or T so its complement is H (for C, T or A).\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception.\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  count(self, sub, start=0, end=9223372036854775807)\n",
      " |      Return a non-overlapping count, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name,\n",
      " |      which does a non-overlapping count!\n",
      " |      \n",
      " |      For an overlapping search use the newer count_overlap() method.\n",
      " |      \n",
      " |      Returns an integer, the number of occurrences of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      Optional arguments start and end are interpreted as in slice\n",
      " |      notation.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"AAAATGA\")\n",
      " |      >>> print(my_seq.count(\"A\"))\n",
      " |      5\n",
      " |      >>> print(my_seq.count(\"ATG\"))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(Seq(\"AT\")))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(\"AT\", 2, -1))\n",
      " |      1\n",
      " |      \n",
      " |      HOWEVER, please note because python strings and Seq objects (and\n",
      " |      MutableSeq objects) do a non-overlapping search, this may not give\n",
      " |      the answer you expect:\n",
      " |      \n",
      " |      >>> \"AAAA\".count(\"AA\")\n",
      " |      2\n",
      " |      >>> print(Seq(\"AAAA\").count(\"AA\"))\n",
      " |      2\n",
      " |      \n",
      " |      An overlapping search, as implemented in .count_overlap(),\n",
      " |      would give the answer as three!\n",
      " |  \n",
      " |  count_overlap(self, sub, start=0, end=9223372036854775807)\n",
      " |      Return an overlapping count.\n",
      " |      \n",
      " |      For a non-overlapping search use the count() method.\n",
      " |      \n",
      " |      Returns an integer, the number of occurrences of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      Optional arguments start and end are interpreted as in slice\n",
      " |      notation.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> print(Seq(\"AAAA\").count_overlap(\"AA\"))\n",
      " |      3\n",
      " |      >>> print(Seq(\"ATATATATA\").count_overlap(\"ATA\"))\n",
      " |      4\n",
      " |      >>> print(Seq(\"ATATATATA\").count_overlap(\"ATA\", 3, -1))\n",
      " |      1\n",
      " |      \n",
      " |      Where substrings do not overlap, should behave the same as\n",
      " |      the count() method:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"AAAATGA\")\n",
      " |      >>> print(my_seq.count_overlap(\"A\"))\n",
      " |      5\n",
      " |      >>> my_seq.count_overlap(\"A\") == my_seq.count(\"A\")\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(\"ATG\"))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(\"ATG\") == my_seq.count(\"ATG\")\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(Seq(\"AT\")))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(Seq(\"AT\")) == my_seq.count(Seq(\"AT\"))\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(\"AT\", 2, -1))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(\"AT\", 2, -1) == my_seq.count(\"AT\", 2, -1)\n",
      " |      True\n",
      " |      \n",
      " |      HOWEVER, do not use this method for such cases because the\n",
      " |      count() method is much for efficient.\n",
      " |  \n",
      " |  encode(self, encoding='utf-8', errors='strict')\n",
      " |      Return an encoded version of the sequence as a bytes object.\n",
      " |      \n",
      " |      The Seq object aims to match the interfact of a Python string.\n",
      " |      (This means on Python 3 it acts like a unicode string.)\n",
      " |      \n",
      " |      This is essentially to save you doing str(my_seq).encode() when\n",
      " |      you need a bytes string, for example for computing a hash:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> Seq(\"ACGT\").encode(\"ascii\")\n",
      " |      b'ACGT'\n",
      " |  \n",
      " |  endswith(self, suffix, start=0, end=9223372036854775807)\n",
      " |      Return True if the Seq ends with the given suffix, False otherwise.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence ends with the specified suffix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      suffix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.endswith(\"UUG\")\n",
      " |      True\n",
      " |      >>> my_rna.endswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.endswith(\"AUG\", 0, 18)\n",
      " |      True\n",
      " |      >>> my_rna.endswith((\"UCC\", \"UCA\", \"UUG\"))\n",
      " |      True\n",
      " |  \n",
      " |  find(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the first occurrence of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the first typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.find(\"AUG\")\n",
      " |      3\n",
      " |  \n",
      " |  index(self, sub, start=0, end=9223372036854775807)\n",
      " |      Like find() but raise ValueError when the substring is not found.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.find(\"T\")\n",
      " |      -1\n",
      " |      >>> my_rna.index(\"T\")\n",
      " |      Traceback (most recent call last):\n",
      " |                 ...\n",
      " |      ValueError: substring not found...\n",
      " |  \n",
      " |  join(self, other)\n",
      " |      Return a merge of the sequences in other, spaced by the sequence from self.\n",
      " |      \n",
      " |      Accepts all Seq objects and Strings as objects to be concatenated with the spacer\n",
      " |      \n",
      " |      >>> concatenated = Seq('NNNNN').join([Seq(\"AAA\"), Seq(\"TTT\"), Seq(\"PPP\")])\n",
      " |      >>> concatenated\n",
      " |      Seq('AAANNNNNTTTNNNNNPPP')\n",
      " |      \n",
      " |      Throws error if other is not an iterable and if objects inside of the iterable\n",
      " |      are not Seq or String objects\n",
      " |  \n",
      " |  lower(self)\n",
      " |      Return a lower case copy of the sequence.\n",
      " |      \n",
      " |      This will adjust the alphabet if required. Note that the IUPAC\n",
      " |      alphabets are upper case only, and thus a generic alphabet must be\n",
      " |      substituted.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import Gapped, generic_dna\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAG*AAAAAA\",\n",
      " |      ...          Gapped(IUPAC.unambiguous_dna, \"*\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG*AAAAAA', Gapped(IUPACUnambiguousDNA(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('cggtacgcttatgtcacgtag*aaaaaa', Gapped(DNAAlphabet(), '*'))\n",
      " |      \n",
      " |      See also the upper method.\n",
      " |  \n",
      " |  lstrip(self, chars=None)\n",
      " |      Return a new Seq object with leading (left) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.lstrip(\"-\"))\n",
      " |      \n",
      " |      See also the strip and rstrip methods.\n",
      " |  \n",
      " |  reverse_complement(self)\n",
      " |      Return the reverse complement sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAGNR\", IUPAC.ambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAGNR', IUPACAmbiguousDNA())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('YNCTATCGGGGG', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      Note in the above example, since R = G or A, its complement\n",
      " |      is Y (which denotes C or T).\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-G\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-G', DNAAlphabet())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('C-TatcGGGGG', DNAAlphabet())\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.reverse_complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  rfind(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find from right method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the last (right most) occurrence of\n",
      " |      substring argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the last typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.rfind(\"AUG\")\n",
      " |      15\n",
      " |  \n",
      " |  rindex(self, sub, start=0, end=9223372036854775807)\n",
      " |      Like rfind() but raise ValueError when the substring is not found.\n",
      " |  \n",
      " |  rsplit(self, sep=None, maxsplit=-1)\n",
      " |      Do a right split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done COUNTING FROM THE RIGHT.\n",
      " |      If maxsplit is omitted, all splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g. print(my_seq.rsplit(\"*\",1))\n",
      " |      \n",
      " |      See also the split method.\n",
      " |  \n",
      " |  rstrip(self, chars=None)\n",
      " |      Return a new Seq object with trailing (right) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. Removing a nucleotide sequence's polyadenylation (poly-A tail):\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAGAAAAAA\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      >>> my_seq.rstrip(\"A\")\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      See also the strip and lstrip methods.\n",
      " |  \n",
      " |  split(self, sep=None, maxsplit=-1)\n",
      " |      Split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done.  If maxsplit is omitted, all\n",
      " |      splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_aa = my_rna.translate()\n",
      " |      >>> my_aa\n",
      " |      Seq('VMAIVMGR*KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> for pep in my_aa.split(\"*\"):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> for pep in my_aa.split(\"*\", 1):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      \n",
      " |      See also the rsplit method:\n",
      " |      \n",
      " |      >>> for pep in my_aa.rsplit(\"*\", 1):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR*KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |  \n",
      " |  startswith(self, prefix, start=0, end=9223372036854775807)\n",
      " |      Return True if the Seq starts with the given prefix, False otherwise.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence starts with the specified prefix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      prefix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.startswith(\"GUC\")\n",
      " |      True\n",
      " |      >>> my_rna.startswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.startswith(\"AUG\", 3)\n",
      " |      True\n",
      " |      >>> my_rna.startswith((\"UCC\", \"UCA\", \"UCG\"), 1)\n",
      " |      True\n",
      " |  \n",
      " |  strip(self, chars=None)\n",
      " |      Return a new Seq object with leading and trailing ends stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.strip(\"-\"))\n",
      " |      \n",
      " |      See also the lstrip and rstrip methods.\n",
      " |  \n",
      " |  tomutable(self)\n",
      " |      Return the full sequence as a MutableSeq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAAL\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      >>> my_seq.tomutable()\n",
      " |      MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      \n",
      " |      Note that the alphabet is preserved.\n",
      " |  \n",
      " |  transcribe(self)\n",
      " |      Return the RNA sequence from a DNA sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> coding_dna = Seq(\"ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\",\n",
      " |      ...                  IUPAC.unambiguous_dna)\n",
      " |      >>> coding_dna\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> coding_dna.transcribe()\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      \n",
      " |      Trying to transcribe a protein or RNA sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be transcribed!\n",
      " |  \n",
      " |  translate(self, table='Standard', stop_symbol='*', to_stop=False, cds=False, gap=None)\n",
      " |      Turn a nucleotide sequence into a protein sequence by creating a new Seq object.\n",
      " |      \n",
      " |      This method will translate DNA or RNA sequences, and those with a\n",
      " |      nucleotide or generic alphabet.  Trying to translate a protein\n",
      " |      sequence raises an exception.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - table - Which codon table to use?  This can be either a name\n",
      " |         (string), an NCBI identifier (integer), or a CodonTable\n",
      " |         object (useful for non-standard genetic codes).  This\n",
      " |         defaults to the \"Standard\" table.\n",
      " |       - stop_symbol - Single character string, what to use for\n",
      " |         terminators.  This defaults to the asterisk, \"*\".\n",
      " |       - to_stop - Boolean, defaults to False meaning do a full\n",
      " |         translation continuing on past any stop codons (translated as the\n",
      " |         specified stop_symbol).  If True, translation is terminated at\n",
      " |         the first in frame stop codon (and the stop_symbol is not\n",
      " |         appended to the returned protein sequence).\n",
      " |       - cds - Boolean, indicates this is a complete CDS.  If True,\n",
      " |         this checks the sequence starts with a valid alternative start\n",
      " |         codon (which will be translated as methionine, M), that the\n",
      " |         sequence length is a multiple of three, and that there is a\n",
      " |         single in frame stop codon at the end (this will be excluded\n",
      " |         from the protein sequence, regardless of the to_stop option).\n",
      " |         If these tests fail, an exception is raised.\n",
      " |       - gap - Single character string to denote symbol used for gaps.\n",
      " |         It will try to guess the gap character from the alphabet.\n",
      " |      \n",
      " |      e.g. Using the standard table:\n",
      " |      \n",
      " |      >>> coding_dna = Seq(\"GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\")\n",
      " |      >>> coding_dna.translate()\n",
      " |      Seq('VAIVMGR*KGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(stop_symbol=\"@\")\n",
      " |      Seq('VAIVMGR@KGAR@', HasStopCodon(ExtendedIUPACProtein(), '@'))\n",
      " |      >>> coding_dna.translate(to_stop=True)\n",
      " |      Seq('VAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      Now using NCBI table 2, where TGA is not a stop codon:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2)\n",
      " |      Seq('VAIVMGRWKGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(table=2, to_stop=True)\n",
      " |      Seq('VAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      In fact, GTG is an alternative start codon under NCBI table 2, meaning\n",
      " |      this sequence could be a complete CDS:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2, cds=True)\n",
      " |      Seq('MAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      It isn't a valid CDS under NCBI table 1, due to both the start codon\n",
      " |      and also the in frame stop codons:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=1, cds=True)\n",
      " |      Traceback (most recent call last):\n",
      " |          ...\n",
      " |      Bio.Data.CodonTable.TranslationError: First codon 'GTG' is not a start codon\n",
      " |      \n",
      " |      If the sequence has no in-frame stop codon, then the to_stop argument\n",
      " |      has no effect:\n",
      " |      \n",
      " |      >>> coding_dna2 = Seq(\"TTGGCCATTGTAATGGGCCGC\")\n",
      " |      >>> coding_dna2.translate()\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      >>> coding_dna2.translate(to_stop=True)\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      When translating gapped sequences, the gap character is inferred from\n",
      " |      the alphabet:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import Gapped\n",
      " |      >>> coding_dna3 = Seq(\"GTG---GCCATT\", Gapped(IUPAC.unambiguous_dna))\n",
      " |      >>> coding_dna3.translate()\n",
      " |      Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))\n",
      " |      \n",
      " |      It is possible to pass the gap character when the alphabet is missing:\n",
      " |      \n",
      " |      >>> coding_dna4 = Seq(\"GTG---GCCATT\")\n",
      " |      >>> coding_dna4.translate(gap='-')\n",
      " |      Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))\n",
      " |      \n",
      " |      NOTE - Ambiguous codons like \"TAN\" or \"NNN\" could be an amino acid\n",
      " |      or a stop codon.  These are translated as \"X\".  Any invalid codon\n",
      " |      (e.g. \"TA?\" or \"T-A\") will throw a TranslationError.\n",
      " |      \n",
      " |      NOTE - This does NOT behave like the python string's translate\n",
      " |      method.  For that use str(my_seq).translate(...) instead.\n",
      " |  \n",
      " |  ungap(self, gap=None)\n",
      " |      Return a copy of the sequence without the gap character(s).\n",
      " |      \n",
      " |      The gap character can be specified in two ways - either as an explicit\n",
      " |      argument, or via the sequence's alphabet. For example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"-ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('-ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap(\"-\")\n",
      " |      Seq('ATATGAAATTTGAAAA', DNAAlphabet())\n",
      " |      \n",
      " |      If the gap character is not given as an argument, it will be taken from\n",
      " |      the sequence's alphabet (if defined). Notice that the returned\n",
      " |      sequence's alphabet is adjusted since it no longer requires a gapped\n",
      " |      alphabet:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped, HasStopCodon\n",
      " |      >>> my_pro = Seq(\"MVVLE=AD*\", HasStopCodon(Gapped(IUPAC.protein, \"=\")))\n",
      " |      >>> my_pro\n",
      " |      Seq('MVVLE=AD*', HasStopCodon(Gapped(IUPACProtein(), '='), '*'))\n",
      " |      >>> my_pro.ungap()\n",
      " |      Seq('MVVLEAD*', HasStopCodon(IUPACProtein(), '*'))\n",
      " |      \n",
      " |      Or, with a simpler gapped DNA example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped\n",
      " |      >>> my_seq = Seq(\"CGGGTAG=AAAAAA\", Gapped(IUPAC.unambiguous_dna, \"=\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap()\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      As long as it is consistent with the alphabet, although it is\n",
      " |      redundant, you can still supply the gap character as an argument to\n",
      " |      this method:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"=\")\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      However, if the gap character given as the argument disagrees with that\n",
      " |      declared in the alphabet, an exception is raised:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"-\")\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap '-' does not match '=' from alphabet\n",
      " |      \n",
      " |      Finally, if a gap character is not supplied, and the alphabet does not\n",
      " |      define one, an exception is raised:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap character not given and not defined in alphabet\n",
      " |  \n",
      " |  upper(self)\n",
      " |      Return an upper case copy of the sequence.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import HasStopCodon, generic_protein\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"VHLTPeeK*\", HasStopCodon(generic_protein))\n",
      " |      >>> my_seq\n",
      " |      Seq('VHLTPeeK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('vhltpeek*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.upper()\n",
      " |      Seq('VHLTPEEK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      \n",
      " |      This will adjust the alphabet if required. See also the lower method.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# You don't need to read the whole output...\n",
    "help(Seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reading and writing annotated sequence files\n",
    "\n",
    "Typically, we do not want to create `Seq` instances by entering manually the sequence of letters (bases or aminoacids). Rather, we usually want to read a bunch of annotated sequences from a file or from some remote database. By annotated sequence here we mean a sequence to which additional information has been added (for example, about the organism, the positions of coding sequences within the sequence, etc.). Here we will learn how to read (and write) annotated sequences from (and to) files. First, we need to import the input/output functions in `Biopython`:  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GeneBank\n",
    "\n",
    "Now, we are going to read the file `Files/ls_orchid.gb`, which contains several annotated sequences in GeneBank format. But before we read it using `Biopython`..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "### Exercise\n",
    "\n",
    "Open the file `Files/ls_orchid.gb` using a text editor such as Notepad or Word. Familiarize yourself with the format of the file. Copy the first complete annotated record in the file and paste it in the cell below."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Write your answer here.*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, let's do something similar but using `Biopython`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The file comprises 94 records\n"
     ]
    }
   ],
   "source": [
    "# Read the GeneBank file and create a list with the records\n",
    "records = list(SeqIO.parse('Files/ls_orchid.gb', 'genbank'))\n",
    "\n",
    "print('The file comprises', len(records), 'records')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let us now select the first record and print it:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: Z78533.1\n",
      "Name: Z78533\n",
      "Description: C.irapeanum 5.8S rRNA gene and ITS1 and ITS2 DNA\n",
      "Number of features: 5\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=PLN\n",
      "/date=30-NOV-2006\n",
      "/accessions=['Z78533']\n",
      "/sequence_version=1\n",
      "/gi=2765658\n",
      "/keywords=['5.8S ribosomal RNA', '5.8S rRNA gene', 'internal transcribed spacer', 'ITS1', 'ITS2']\n",
      "/source=Cypripedium irapeanum\n",
      "/organism=Cypripedium irapeanum\n",
      "/taxonomy=['Eukaryota', 'Viridiplantae', 'Streptophyta', 'Embryophyta', 'Tracheophyta', 'Spermatophyta', 'Magnoliophyta', 'Liliopsida', 'Asparagales', 'Orchidaceae', 'Cypripedioideae', 'Cypripedium']\n",
      "/references=[Reference(title='Phylogenetics of the slipper orchids (Cypripedioideae: Orchidaceae): nuclear rDNA ITS sequences', ...), Reference(title='Direct Submission', ...)]\n",
      "Seq('CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGATGAGACCGTGG...CGC', IUPACAmbiguousDNA())\n"
     ]
    }
   ],
   "source": [
    "print(records[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `SeqRecord` class that holds annotated sequences has the following information as attributes:\n",
    "\n",
    "**.seq**\n",
    "– The sequence itself, typically a Seq object.\n",
    "\n",
    "**.id**\n",
    "– The primary ID used to identify the sequence – a string. In most cases this is something like an accession number.\n",
    "\n",
    "**.name**\n",
    "– A “common” name/id for the sequence – a string. In some cases this will be the same as the accession number, but it could also be a clone name. I think of this as being analogous to the LOCUS id in a GenBank record.\n",
    "\n",
    "**.description**\n",
    "– A human readable description or expressive name for the sequence – a string.\n",
    "\n",
    "**.letter_annotations**\n",
    "– Holds per-letter-annotations using a (restricted) dictionary of additional information about the letters in the sequence. The keys are the name of the information, and the information is contained in the value as a Python sequence (i.e. a list, tuple or string) with the same length as the sequence itself. This is often used for quality scores or secondary structure information.\n",
    "\n",
    "**.annotations**\n",
    "– A dictionary of additional information about the sequence. The keys are the name of the information, and the information is contained in the value. This allows the addition of more “unstructured” information to the sequence.\n",
    "\n",
    "**.features**\n",
    "– A list of SeqFeature objects with more structured information about the features on a sequence (e.g. position of genes on a genome, or domains on a protein sequence).\n",
    "\n",
    "**.dbxrefs** - A list of database cross-references as strings.\n",
    "\n",
    "For example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SEQUENCE: CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGATGAGACCGTGGAATAAACGATCGAGTGAATCCGGAGGACCGGTGTACTCAGCTCACCGGGGGCATTGCTCCCGTGGTGACCCTGATTTGTTGTTGGGCCGCCTCGGGAGCGTCCATGGCGGGTTTGAACCTCTAGCCCGGCGCAGTTTGGGCGCCAAGCCATATGAAAGCATCACCGGCGAATGGCATTGTCTTCCCCAAAACCCGGAGCGGCGGCGTGCTGTCGCGTGCCCAATGAATTTTGATGACTCTCGCAAACGGGAATCTTGGCTCTTTGCATCGGATGGAAGGACGCAGCGAAATGCGATAAGTGGTGTGAATTGCAAGATCCCGTGAACCATCGAGTCTTTTGAACGCAAGTTGCGCCCGAGGCCATCAGGCTAAGGGCACGCCTGCTTGGGCGTCGCGCTTCGTCTCTCTCCTGCCAATGCTTGCCCGGCATACAGCCAGGCCGGCGTGGTGCGGATGTGAAAGATTGGCCCCTTGTGCCTAGGTGCGGCGGGTCCAAGAGCTGGTGTTTTGATGGCCCGGAACCCGGCAAGAGGTGGACGGATGCTGGCAGCAGCTGCCGTGCGAATCCCCCATGTTGTCGTGCTTGTCGGACAGGCAGGAGAACCCTTCCGAACCCCAATGGAGGGCGGTTGACCGCCATTCGGATGTGACCCCAGGTCAGGCGGGGGCACCCGCTGAGTTTACGC \n",
      "\n",
      "SEQUENCE LENGTH: 740 \n",
      "\n",
      "FEATURES:\n",
      "type: source\n",
      "location: [0:740](+)\n",
      "qualifiers:\n",
      "    Key: db_xref, Value: ['taxon:49711']\n",
      "    Key: mol_type, Value: ['genomic DNA']\n",
      "    Key: organism, Value: ['Cypripedium irapeanum']\n",
      "\n",
      "type: misc_feature\n",
      "location: [0:380](+)\n",
      "qualifiers:\n",
      "    Key: note, Value: ['internal transcribed spacer 1']\n",
      "\n",
      "type: gene\n",
      "location: [380:550](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['5.8S rRNA']\n",
      "\n",
      "type: rRNA\n",
      "location: [380:550](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['5.8S rRNA']\n",
      "    Key: product, Value: ['5.8S ribosomal RNA']\n",
      "\n",
      "type: misc_feature\n",
      "location: [550:740](+)\n",
      "qualifiers:\n",
      "    Key: note, Value: ['internal transcribed spacer 2']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Print some of the features of the first annotated record in the file\n",
    "print('SEQUENCE:', records[0].seq, '\\n')\n",
    "print('SEQUENCE LENGTH:', len(records[0].seq), '\\n')\n",
    "print('FEATURES:')\n",
    "for feature in records[0].features:\n",
    "    print(feature)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "### Exercise\n",
    "\n",
    "Select the record with the longest sequence in the file, and print its features. Then, explain in words what each feature corresponds to."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write your code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Explain succinctly each feature*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "\n",
    "## FASTA\n",
    "\n",
    "GeneBank annotated records are quite co"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sjsfnsjns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"roger.guimera@urv.cat\"  # Always tell NCBI who you are\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id='MN908947.3', rettype=\"gb\", retmode=\"text\")\n",
    "patient_rec = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: MN908947.3\n",
      "Name: MN908947\n",
      "Description: Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
      "Number of features: 23\n",
      "/taxonomy=['Viruses', 'Riboviria', 'Nidovirales', 'Cornidovirineae', 'Coronaviridae', 'Orthocoronavirinae', 'Betacoronavirus', 'Sarbecovirus']\n",
      "/topology=linear\n",
      "/keywords=['']\n",
      "/date=11-FEB-2020\n",
      "/sequence_version=3\n",
      "/comment=On Jan 17, 2020 this sequence version replaced MN908947.2.\n",
      "/references=[Reference(title='A new coronavirus associated with human respiratory disease in China', ...), Reference(title='Direct Submission', ...)]\n",
      "/data_file_division=VRL\n",
      "/source=Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n",
      "/organism=Severe acute respiratory syndrome coronavirus 2\n",
      "/structured_comment=OrderedDict([('Assembly-Data', OrderedDict([('Assembly Method', 'Megahit v. V1.1.3'), ('Sequencing Technology', 'Illumina')]))])\n",
      "/accessions=['MN908947']\n",
      "/molecule_type=ss-RNA\n",
      "Seq('ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGT...AAA', IUPACAmbiguousDNA())\n"
     ]
    }
   ],
   "source": [
    "print(patient_rec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGATGCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTCGTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTAGGCGACGAGCTTGGCACTGATCCTTATGAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTGTTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGGCCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTCATGCACTTTGTCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTGCTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCCTTAAATTCCATAATCAAGACTATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTTATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCACCAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCAGACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACTACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAATTTATTGTCCAGCATGTCACAATTCAGAAGTAGGACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCGCACTATTGCCTTTGGAGGCTGTGTGTTCTCTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCACGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACAACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGATTATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAAAAAGGTGCCTGGAATATTGGTGAACAGAAATCAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCGTGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCCGCTATAACAATACTAGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTGGCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTTAAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAATTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGTAAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCCTACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAAGCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAAGGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAAATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACCACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAATTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAGAAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATTTGGTGCCACTTCTGCTGCTCTTCAACCTGAAGAAGAGCAAGAAGAAGATTGGTTAGATGATGATAGTCAACAAACTGTTGGTCAACAAGACGGCAGTGAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTCAACCTCAATTAGAGATGGAACTTACACCAGTTGTTCAGACTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACAGTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTACTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAAGTGGGTGGTAGTTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGTGAAGACATTCAACTTCTTAAGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTATTATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAATGTCTACTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGTGAAAAGCAAGTTGAACAAAAGATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTAAACCTTCAGTTGAACAGAGAAAACAAGATGATAAGAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAACTCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCAGATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAATGCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAATGGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTATTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGCAGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAATATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAACAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTTGGCTATGTAACACATGGCTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTTTCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAGAACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACAACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCACCTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTAAGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACAACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGTAAAACATTTTATGTTTTACCTAATGATGACACTCTACGTGTTGAGGCTTTTGAGTACTACCACACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAATGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAAATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTAACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAATGAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGTGGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAATTTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTCACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGTGAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAGACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAGTTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAGTTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAACCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATTTTTCCCTGACTTAAATGGTGATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAACCTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTGTCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTGAAGTCAGAGGACGCGCAGGGAATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAATAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTTACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTTGCTACTCATGGTTTAGCTGCTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTACAACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTATTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAAAGCATCTATGCCGACTACTATAGCAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAACTGATAAATATTATAATTTGGTTTTTACTATTAAGTGTTTGCCTAGGTTCTTTAATCTACTCAACCGCTGCTTTAGGTGTTTTAATGTCTAATTTAGGCATGCCTTCTTACTGTACTGGTTACAGAGAAGGCTATTTGAACTCTACTAATGTCACTATTGCAACCTACTGTACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTAGACACCTATCCTTCTTTAGAAACTATACAAATTACCATTTCATCTTTTAAATGGGATTTAACTGCTTTTGGCTTAGTTGCAGAGTGGTTTTTGGCATATATTCTTTTCACTAGGTTTTTCTATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCGATTTCAGCTATGGTTAGAATGTACATCTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGACGGTTGTAATTCATCAACTTGTATGATGTGTTACAAACGTAATAGAGCAACAAGAGTCGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACACAATTGGAATTGTGTTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGATGAAGTTGCGAGAGACTTGTCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCTTACATCGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAAGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAACCTGAGAGCTAATAACACTAAAGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCAAAATCAGCGTCTGTTTACTACAGTCAGCTTATGTGTCAACCTATACTGTTACTAGATCAGGCATTAGTGTCTGATGTTGGTGATAGTGCGGAAGTTGCAGTTAAAATGTTTGATGCTTACGTTAATACGTTTTCATCAACTTTTAACGTACCAATGGAAAAACTCAAAACACTAGTTGCAACTGCAGAAGCTGAACTTGCAAAGAATGTGTCCTTAGACAATGTCTTATCTACTTTTATTTCAGCAGCTCGGCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACATCAATCTGACATAGAAGTTACTGGCGATAGTTGTAATAACTATATGCTCACCTATAACAAAGTTGAAAACATGACACCCCGTGACCTTGGTGCTTGTATTGACTGTAGTGCGCGTCATATTAATGCGCAGGTAGCAAAAAGTCACAACATTGCTTTGATATGGAACGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGTAGTGCTGCTAAAAAGAATAACTTACCTTTTAAGTTGACATGTGCAACTACTAGACAAGTTGTTAATGTTGTAACAACAAAGATAGCACTTAAGGGTGGTAAAATTGTTAATAATTGGTTGAAGCAGTTAATTAAAGTTACACTTGTGTTCCTTTTTGTTGCTGCTATTTTCTATTTAATAACACCTGTTCATGTCATGTCTAAACATACTGACTTTTCAAGTGAAATCATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCATCTACAGATACTTGTTTTGCTAACAAACATGCTGATTTTGACACATGGTTTAGCCAGCGTGGTGGTAGTTATACTAATGACAAAGCTTGCCCATTGATTGCTGCAGTCATAACAAGAGAAGTGGGTTTTGTCGTGCCTGGTTTGCCTGGCACGATATTACGCACAACTAATGGTGACTTTTTGCATTTCTTACCTAGAGTTTTTAGTGCAGTTGGTAACATCTGTTACACACCATCAAAACTTATAGAGTACACTGACTTTGCAACATCAGCTTGTGTTTTGGCTGCTGAATGTACAATTTTTAAAGATGCTTCTGGTAAGCCAGTACCATATTGTTATGATACCAATGTACTAGAAGGTTCTGTTGCTTATGAAAGTTTACGCCCTGACACACGTTATGTGCTCATGGATGGCTCTATTATTCAATTTCCTAACACCTACCTTGAAGGTTCTGTTAGAGTGGTAACAACTTTTGATTCTGAGTACTGTAGGCACGGCACTTGTGAAAGATCAGAAGCTGGTGTTTGTGTATCTACTAGTGGTAGATGGGTACTTAACAATGATTATTACAGATCTTTACCAGGAGTTTTCTGTGGTGTAGATGCTGTAAATTTACTTACTAATATGTTTACACCACTAATTCAACCTATTGGTGCTTTGGACATATCAGCATCTATAGTAGCTGGTGGTATTGTAGCTATCGTAGTAACATGCCTTGCCTACTATTTTATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAATACTTTACTATTCCTTATGTCATTCACTGTACTCTGTTTAACACCAGTTTACTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTTTTAGCACATATTCAGTGGATGGTTATGTTCACACCTTTAGTACCTTTCTGGATAACAATTGCTTATATCATTTGTATTTCCACAAAGCATTTCTATTGGTTCTTTAGTAATTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACTTTTGAAGAAGCTGCGCTGTGCACCTTTTTGTTAAATAAAGAAATGTATCTAAAGTTGCGTAGTGATGTGCTATTACCTCTTACGCAATATAATAGATACTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGAGCAATGGATACAACTAGCTACAGAGAAGCTGCTTGTTGTCATCTCGCAAAGGCTCTCAATGACTTCAGTAACTCAGGTTCTGATGTTCTTTACCAACCACCACAAACCTCTATCACCTCAGCTGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAGGGTTGTATGGTACAAGTAACTTGTGGTACAACTACACTTAACGGTCTTTGGCTTGATGACGTAGTTTACTGTCCAAGACATGTGATCTGCACCTCTGAAGACATGCTTAACCCTAATTATGAAGATTTACTCATTCGTAAGTCTAATCATAATTTCTTGGTACAGGCTGGTAATGTTCAACTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTTAAGCTTAAGGTTGATACAGCCAATCCTAAGACACCTAAGTATAAGTTTGTTCGCATTCAACCAGGACAGACTTTTTCAGTGTTAGCTTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCTATGAGGCCCAATTTCACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAACATAGATTATGACTGTGTCTCTTTTTGTTACATGCACCATATGGAATTACCAACTGGAGTTCATGCTGGCACAGACTTAGAAGGTAACTTTTATGGACCTTTTGTTGACAGGCAAACAGCACAAGCAGCTGGTACGGACACAACTATTACAGTTAATGTTTTAGCTTGGTTGTACGCTGCTGTTATAAATGGAGACAGGTGGTTTCTCAATCGATTTACCACAACTCTTAATGACTTTAACCTTGTGGCTATGAAGTACAATTATGAACCTCTAACACAAGACCATGTTGACATACTAGGACCTCTTTCTGCTCAAACTGGAATTGCCGTTTTAGATATGTGTGCTTCATTAAAAGAATTACTGCAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTTGATGTTGTTAGACAATGCTCAGGTGTTACTTTCCAAAGTGCAGTGAAAAGAACAATCAAGGGTACACACCACTGGTTGTTACTCACAATTTTGACTTCACTTTTAGTTTTAGTCCAGAGTACTCAATGGTCTTTGTTCTTTTTTTTGTATGAAAATGCCTTTTTACCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCAATGATGTTTGTCAAACATAAGCATGCATTTCTCTGTTTGTTTTTGTTACCTTCTCTTGCCACTGTAGCTTATTTTAATATGGTCTATATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATGGTTGATACTAGTTTGTCTGGTTTTAAGCTAAAAGACTGTGTTATGTATGCATCAGCTGTAGTGTTACTAATCCTTATGACAGCAAGAACTGTGTATGATGATGGTGCTAGGAGAGTGTGGACACTTATGAATGTCTTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGATCAAGCCATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGCCCTATTTTCTTCATAACTGGTAATACACTTCAGTGTATAATGCTAGTTTATTGTTTCTTAGGCTATTTTTGTACTTGTTACTTTGGCCTCTTTTGTTTACTCAACCGCTACTTTAGACTGACTCTTGGTGTTTATGATTACTTAGTTTCTACACAGGAGTTTAGATATATGAATTCACAGGGACTACTCCCACCCAAGAATAGCATAGATGCCTTCAAACTCAACATTAAATTGTTGGGTGTTGGTGGCAAACCTTGTATCAAAGTAGCCACTGTACAGTCTAAAATGTCAGATGTAAAGTGCACATCAGTAGTCTTACTCTCAGTTTTGCAACAACTCAGAGTAGAATCATCATCTAAATTGTGGGCTCAATGTGTCCAGTTACACAATGACATTCTCTTAGCTAAAGATACTACTGAAGCCTTTGAAAAAATGGTTTCACTACTTTCTGTTTTGCTTTCCATGCAGGGTGCTGTAGACATAAACAAGCTTTGTGAAGAAATGCTGGACAACAGGGCAACCTTACAAGCTATAGCCTCAGAGTTTAGTTCCCTTCCATCATATGCAGCTTTTGCTACTGCTCAAGAAGCTTATGAGCAGGCTGTTGCTAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTGAAGAAGTCTTTGAATGTGGCTAAATCTGAATTTGACCGTGATGCAGCCATGCAACGTAAGTTGGAAAAGATGGCTGATCAAGCTATGACCCAAATGTATAAACAGGCTAGATCTGAGGACAAGAGGGCAAAAGTTACTAGTGCTATGCAGACAATGCTTTTCACTATGCTTAGAAAGTTGGATAATGATGCACTCAACAACATTATCAACAATGCAAGAGATGGTTGTGTTCCCTTGAACATAATACCTCTTACAACAGCAGCCAAACTAATGGTTGTCATACCAGACTATAACACATATAAAAATACGTGTGATGGTACAACATTTACTTATGCATCAGCATTGTGGGAAATCCAACAGGTTGTAGATGCAGATAGTAAAATTGTTCAACTTAGTGAAATTAGTATGGACAATTCACCTAATTTAGCATGGCCTCTTATTGTAACAGCTTTAAGGGCCAATTCTGCTGTCAAATTACAGAATAATGAGCTTAGTCCTGTTGCACTACGACAGATGTCTTGTGCTGCCGGTACTACACAAACTGCTTGCACTGATGACAATGCGTTAGCTTACTACAACACAACAAAGGGAGGTAGGTTTGTACTTGCACTGTTATCCGATTTACAGGATTTGAAATGGGCTAGATTCCCTAAGAGTGATGGAACTGGTACTATCTATACAGAACTGGAACCACCTTGTAGGTTTGTTACAGACACACCTAAAGGTCCTAAAGTGAAGTATTTATACTTTATTAAAGGATTAAACAACCTAAATAGAGGTATGGTACTTGGTAGTTTAGCTGCCACAGTACGTCTACAAGCTGGTAATGCAACAGAAGTGCCTGCCAATTCAACTGTATTATCTTTCTGTGCTTTTGCTGTAGATGCTGCTAAAGCTTACAAAGATTATCTAGCTAGTGGGGGACAACCAATCACTAATTGTGTTAAGATGTTGTGTACACACACTGGTACTGGTCAGGCAATAACAGTTACACCGGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCATCGTGTTGTCTGTACTGCCGTTGCCACATAGATCATCCAAATCCTAAAGGATTTTGTGACTTAAAAGGTAAGTATGTACAAATACCTACAACTTGTGCTAATGACCCTGTGGGTTTTACACTTAAAAACACAGTCTGTACCGTCTGCGGTATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGCTTCAGTCAGCTGATGCACAATCGTTTTTAAACGGGTTTGCGGTGTAAGTGCAGCCCGTCTTACACCGTGCGGCACAGGCACTAGTACTGATGTCGTATACAGGGCTTTTGACATCTACAATGATAAAGTAGCTGGTTTTGCTAAATTCCTAAAAACTAATTGTTGTCGCTTCCAAGAAAAGGACGAAGATGACAATTTAATTGATTCTTACTTTGTAGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAAGGATTGTCCAGCTGTTGCTAAACATGACTTCTTTAAGTTTAGAATAGACGGTGACATGGTACCACATATATCACGTCAACGTCTTACTAAATACACAATGGCAGACCTCGTCTATGCTTTAAGGCATTTTGATGAAGGTAATTGTGACACATTAAAAGAAATACTTGTCACATACAATTGTTGTGATGATGATTATTTCAATAAAAAGGACTGGTATGATTTTGTAGAAAACCCAGATATATTACGCGTATACGCCAACTTAGGTGAACGTGTACGCCAAGCTTTGTTAAAAACAGTACAATTCTGTGATGCCATGCGAAATGCTGGTATTGTTGGTGTACTGACATTAGATAATCAAGATCTCAATGGTAACTGGTATGATTTCGGTGATTTCATACAAACCACGCCAGGTAGTGGAGTTCCTGTTGTAGATTCTTATTATTCATTGTTAATGCCTATATTAACCTTGACCAGGGCTTTAACTGCAGAGTCACATGTTGACACTGACTTAACAAAGCCTTACATTAAGTGGGATTTGTTAAAATATGACTTCACGGAAGAGAGGTTAAAACTCTTTGACCGTTATTTTAAATATTGGGATCAGACATACCACCCAAATTGTGTTAACTGTTTGGATGACAGATGCATTCTGCATTGTGCAAACTTTAATGTTTTATTCTCTACAGTGTTCCCACCTACAAGTTTTGGACCACTAGTGAGAAAAATATTTGTTGATGGTGTTCCATTTGTAGTTTCAACTGGATACCACTTCAGAGAGCTAGGTGTTGTACATAATCAGGATGTAAACTTACATAGCTCTAGACTTAGTTTTAAGGAATTACTTGTGTATGCTGCTGACCCTGCTATGCACGCTGCTTCTGGTAATCTATTACTAGATAAACGCACTACGTGCTTTTCAGTAGCTGCACTTACTAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAACAAAGACTTCTATGACTTTGCTGTGTCTAAGGGTTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGTAATGCTGCTATCAGCGATTATGACTACTATCGTTATAATCTACCAACAATGTGTGATATCAGACAACTACTATTTGTAGTTGAAGTTGTTGATAAGTACTTTGATTGTTACGATGGTGGCTGTATTAATGCTAACCAAGTCATCGTCAACAACCTAGACAAATCAGCTGGTTTTCCATTTAATAAATGGGGTAAGGCTAGACTTTATTATGATTCAATGAGTTATGAGGATCAAGATGCACTTTTCGCATATACAAAACGTAATGTCATCCCTACTATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACCGTAGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAAAAATTATTGAAATCAATAGCCGCCACTAGAGGAGCTACTGTAGTAATTGGAACAAGCAAATTCTATGGTGGTTGGCACAACATGTTAAAAACTGTTTATAGTGATGTAGAAAACCCTCACCTTATGGGTTGGGATTATCCTAAATGTGATAGAGCCATGCCTAACATGCTTAGAATTATGGCCTCACTTGTTCTTGCTCGCAAACATACAACGTGTTGTAGCTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCTCAAGTATTGAGTGAAATGGTCATGTGTGGCGGTTCACTATATGTTAAACCAGGTGGAACCTCATCAGGAGATGCCACAACTGCTTATGCTAATAGTGTTTTTAACATTTGTCAAGCTGTCACGGCCAATGTTAATGCACTTTTATCTACTGATGGTAACAAAATTGCCGATAAGTATGTCCGCAATTTACAACACAGACTTTATGAGTGTCTCTATAGAAATAGAGATGTTGACACAGACTTTGTGAATGAGTTTTACGCATATTTGCGTAAACATTTCTCAATGATGATACTCTCTGACGATGCTGTTGTGTGTTTCAATAGCACTTATGCATCTCAAGGTCTAGTGGCTAGCATAAAGAACTTTAAGTCAGTTCTTTATTATCAAAACAATGTTTTTATGTCTGAAGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGCTCTCAACATACAATGCTAGTTAAACAGGGTGATGATTATGTGTACCTTCCTTACCCAGATCCATCAAGAATCCTAGGGGCCGGCTGTTTTGTAGATGATATCGTAAAAACAGATGGTACACTTATGATTGAACGGTTCGTGTCTTTAGCTATAGATGCTTACCCACTTACTAAACATCCTAATCAGGAGTATGCTGATGTCTTTCATTTGTACTTACAATACATAAGAAAGCTACATGATGAGTTAACAGGACACATGTTAGACATGTATTCTGTTATGCTTACTAATGATAACACTTCAAGGTATTGGGAACCTGAGTTTTATGAGGCTATGTACACACCGCATACAGTCTTACAGGCTGTTGGGGCTTGTGTTCTTTGCAATTCACAGACTTCATTAAGATGTGGTGCTTGCATACGTAGACCATTCTTATGTTGTAAATGCTGTTACGACCATGTCATATCAACATCACATAAATTAGTCTTGTCTGTTAATCCGTATGTTTGCAATGCTCCAGGTTGTGATGTCACAGATGTGACTCAACTTTACTTAGGAGGTATGAGCTATTATTGTAAATCACATAAACCACCCATTAGTTTTCCATTGTGTGCTAATGGACAAGTTTTTGGTTTATATAAAAATACATGTGTTGGTAGCGATAATGTTACTGACTTTAATGCAATTGCAACATGTGACTGGACAAATGCTGGTGATTACATTTTAGCTAACACCTGTACTGAAAGACTCAAGCTTTTTGCAGCAGAAACGCTCAAAGCTACTGAGGAGACATTTAAACTGTCTTATGGTATTGCTACTGTACGTGAAGTGCTGTCTGACAGAGAATTACATCTTTCATGGGAAGTTGGTAAACCTAGACCACCACTTAACCGAAATTATGTCTTTACTGGTTATCGTGTAACTAAAAACAGTAAAGTACAAATAGGAGAGTACACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTTTACCGAGGTACAACAACTTACAAATTAAATGTTGGTGATTATTTTGTGCTGACATCACATACAGTAATGCCATTAAGTGCACCTACACTAGTGCCACAAGAGCACTATGTTAGAATTACTGGCTTATACCCAACACTCAATATCTCAGATGAGTTTTCTAGCAATGTTGCAAATTATCAAAAGGTTGGTATGCAAAAGTATTCTACACTCCAGGGACCACCTGGTACTGGTAAGAGTCATTTTGCTATTGGCCTAGCTCTCTACTACCCTTCTGCTCGCATAGTGTATACAGCTTGCTCTCATGCCGCTGTTGATGCACTATGTGAGAAGGCATTAAAATATTTGCCTATAGATAAATGTAGTAGAATTATACCTGCACGTGCTCGTGTAGAGTGTTTTGATAAATTCAAAGTGAATTCAACATTAGAACAGTATGTCTTTTGTACTGTAAATGCATTGCCTGAGACGACAGCAGATATAGTTGTCTTTGATGAAATTTCAATGGCCACAAATTATGATTTGAGTGTTGTCAATGCCAGATTACGTGCTAAGCACTATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTGCTGAAATTGTTGACACTGTGAGTGCTTTGGTTTATGATAATAAGCTTAAAGCACATAAAGACAAATCAGCTCAATGCTTTAAAATGTTTTATAAGGGTGTTATCACGCATGATGTTTCATCTGCAATTAACAGGCCACAAATAGGCGTGGTAAGAGAATTCCTTACACGTAACCCTGCTTGGAGAAAAGCTGTCTTTATTTCACCTTATAATTCACAGAATGCTGTAGCCTCAAAGATTTTGGGACTACCAACTCAAACTGTTGATTCATCACAGGGCTCAGAATATGACTATGTCATATTCACTCAAACCACTGAAACAGCTCACTCTTGTAATGTAAACAGATTTAATGTTGCTATTACCAGAGCAAAAGTAGGCATACTTTGCATAATGTCTGATAGAGACCTTTATGACAAGTTGCAATTTACAAGTCTTGAAATTCCACGTAGGAATGTGGCAACTTTACAAGCTGAAAATGTAACAGGACTCTTTAAAGATTGTAGTAAGGTAATCACTGGGTTACATCCTACACAGGCACCTACACACCTCAGTGTTGACACTAAATTCAAAACTGAAGGTTTATGTGTTGACATACCTGGCATACCTAAGGACATGACCTATAGAAGACTCATCTCTATGATGGGTTTTAAAATGAATTATCAAGTTAATGGTTACCCTAACATGTTTATCACCCGCGAAGAAGCTATAAGACATGTACGTGCATGGATTGGCTTCGATGTCGAGGGGTGTCATGCTACTAGAGAAGCTGTTGGTACCAATTTACCTTTACAGCTAGGTTTTTCTACAGGTGTTAACCTAGTTGCTGTACCTACAGGTTATGTTGATACACCTAATAATACAGATTTTTCCAGAGTTAGTGCTAAACCACCGCCTGGAGATCAATTTAAACACCTCATACCACTTATGTACAAAGGACTTCCTTGGAATGTAGTGCGTATAAAGATTGTACAAATGTTAAGTGACACACTTAAAAATCTCTCTGACAGAGTCGTATTTGTCTTATGGGCACATGGCTTTGAGTTGACATCTATGAAGTATTTTGTGAAAATAGGACCTGAGCGCACCTGTTGTCTATGTGATAGACGTGCCACATGCTTTTCCACTGCTTCAGACACTTATGCCTGTTGGCATCATTCTATTGGATTTGATTACGTCTATAATCCGTTTATGATTGATGTTCAACAATGGGGTTTTACAGGTAACCTACAAAGCAACCATGATCTGTATTGTCAAGTCCATGGTAATGCACATGTAGCTAGTTGTGATGCAATCATGACTAGGTGTCTAGCTGTCCACGAGTGCTTTGTTAAGCGTGTTGACTGGACTATTGAATATCCTATAATTGGTGATGAACTGAAGATTAATGCGGCTTGTAGAAAGGTTCAACACATGGTTGTTAAAGCTGCATTATTAGCAGACAAATTCCCAGTTCTTCACGACATTGGTAACCCTAAAGCTATTAAGTGTGTACCTCAAGCTGATGTAGAATGGAAGTTCTATGATGCACAGCCTTGTAGTGACAAAGCTTATAAAATAGAAGAATTATTCTATTCTTATGCCACACATTCTGACAAATTCACAGATGGTGTATGCCTATTTTGGAATTGCAATGTCGATAGATATCCTGCTAATTCCATTGTTTGTAGATTTGACACTAGAGTGCTATCTAACCTTAACTTGCCTGGTTGTGATGGTGGCAGTTTGTATGTAAATAAACATGCATTCCACACACCAGCTTTTGATAAAAGTGCTTTTGTTAATTTAAAACAATTACCATTTTTCTATTACTCTGACAGTCCATGTGAGTCTCATGGAAAACAAGTAGTGTCAGATATAGATTATGTACCACTAAAGTCTGCTACGTGTATAACACGTTGCAATTTAGGTGGTGCTGTCTGTAGACATCATGCTAATGAGTACAGATTGTATCTCGATGCTTATAACATGATGATCTCAGCTGGCTTTAGCTTGTGGGTTTACAAACAATTTGATACTTATAACCTCTGGAACACTTTTACAAGACTTCAGAGTTTAGAAAATGTGGCTTTTAATGTTGTAAATAAGGGACACTTTGATGGACAACAGGGTGAAGTACCAGTTTCTATCATTAATAACACTGTTTACACAAAAGTTGATGGTGTTGATGTAGAATTGTTTGAAAATAAAACAACATTACCTGTTAATGTAGCATTTGAGCTTTGGGCTAAGCGCAACATTAAACCAGTACCAGAGGTGAAAATACTCAATAATTTGGGTGTGGACATTGCTGCTAATACTGTGATCTGGGACTACAAAAGAGATGCTCCAGCACATATATCTACTATTGGTGTTTGTTCTATGACTGACATAGCCAAGAAACCAACTGAAACGATTTGTGCACCACTCACTGTCTTTTTTGATGGTAGAGTTGATGGTCAAGTAGACTTATTTAGAAATGCCCGTAATGGTGTTCTTATTACAGAAGGTAGTGTTAAAGGTTTACAACCATCTGTAGGTCCCAAACAAGCTAGTCTTAATGGAGTCACATTAATTGGAGAAGCCGTAAAAACACAGTTCAATTATTATAAGAAAGTTGATGGTGTTGTCCAACAATTACCTGAAACTTACTTTACTCAGAGTAGAAATTTACAAGAATTTAAACCCAGGAGTCAAATGGAAATTGATTTCTTAGAATTAGCTATGGATGAATTCATTGAACGGTATAAATTAGAAGGCTATGCCTTCGAACATATCGTTTATGGAGATTTTAGTCATAGTCAGTTAGGTGGTTTACATCTACTGATTGGACTAGCTAAACGTTTTAAGGAATCACCTTTTGAATTAGAAGATTTTATTCCTATGGACAGTACAGTTAAAAACTATTTCATAACAGATGCGCAAACAGGTTCATCTAAGTGTGTGTGTTCTGTTATTGATTTATTACTTGATGATTTTGTTGAAATAATAAAATCCCAAGATTTATCTGTAGTTTCTAAGGTTGTCAAAGTGACTATTGACTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGCCATGTAGAAACATTTTACCCAAAATTACAATCTAGTCAAGCGTGGCAACCGGGTGTTGCTATGCCTAATCTTTACAAAATGCAAAGAATGCTATTAGAAAAGTGTGACCTTCAAAATTATGGTGATAGTGCAACATTACCTAAAGGCATAATGATGAATGTCGCAAAATATACTCAACTGTGTCAATATTTAAACACATTAACATTAGCTGTACCCTATAATATGAGAGTTATACATTTTGGTGCTGGTTCTGATAAAGGAGTTGCACCAGGTACAGCTGTTTTAAGACAGTGGTTGCCTACGGGTACGCTGCTTGTCGATTCAGATCTTAATGACTTTGTCTCTGATGCAGATTCAACTTTGATTGGTGATTGTGCAACTGTACATACAGCTAATAAATGGGATCTCATTATTAGTGATATGTACGACCCTAAGACTAAAAATGTTACAAAAGAAAATGACTCTAAAGAGGGTTTTTTCACTTACATTTGTGGGTTTATACAACAAAAGCTAGCTCTTGGAGGTTCCGTGGCTATAAAGATAACAGAACATTCTTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCCTTTGTTACTAATGTGAATGCGTCATCATCTGAAGCATTTTTAATTGGATGTAATTATCTTGGCAAACCACGCGAACAAATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAATACAAATCCAATTCAGTTGTCTTCCTATTCTTTATTTGACATGAGTAAATTTCCCCTTAAATTAAGGGGTACTGCTGTTATGTCTTTAAAAGAAGGTCAAATCAATGATATGATTTTATCTCTTCTTAGTAAAGGTAGACTTATAATTAGAGAAAACAACAGAGTTGTTATTTCTAGTGATGTTCTTGTTAACAACTAAACGAACAATGTTTGTTTTTCTTGTTTTATTGCCACTAGTCTCTAGTCAGTGTGTTAATCTTACAACCAGAACTCAATTACCCCCTGCATACACTAATTCTTTCACACGTGGTGTTTATTACCCTGACAAAGTTTTCAGATCCTCAGTTTTACATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCAATGGTACTAAGAGGTTTGATAACCCTGTCCTACCATTTAATGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACTTTAGATTCGAAGACCCAGTCCCTACTTATTGTTAATAACGCTACTAATGTTGTTATTAAAGTCTGTGAATTTCAATTTTGTAATGATCCATTTTTGGGTGTTTATTACCACAAAAACAACAAAAGTTGGATGGAAAGTGAGTTCAGAGTTTATTCTAGTGCGAATAATTGCACTTTTGAATATGTCTCTCAGCCTTTTCTTATGGACCTTGAAGGAAAACAGGGTAATTTCAAAAATCTTAGGGAATTTGTGTTTAAGAATATTGATGGTTATTTTAAAATATATTCTAAGCACACGCCTATTAATTTAGTGCGTGATCTCCCTCAGGGTTTTTCGGCTTTAGAACCATTGGTAGATTTGCCAATAGGTATTAACATCACTAGGTTTCAAACTTTACTTGCTTTACATAGAAGTTATTTGACTCCTGGTGATTCTTCTTCAGGTTGGACAGCTGGTGCTGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGATGCTGTAGACTGTGCACTTGACCCTCTCTCAGAAACAAAGTGTACGTTGAAATCCTTCACTGTAGAAAAAGGAATCTATCAAACTTCTAACTTTAGAGTCCAACCAACAGAATCTATTGTTAGATTTCCTAATATTACAAACTTGTGCCCTTTTGGTGAAGTTTTTAACGCCACCAGATTTGCATCTGTTTATGCTTGGAACAGGAAGAGAATCAGCAACTGTGTTGCTGATTATTCTGTCCTATATAATTCCGCATCATTTTCCACTTTTAAGTGTTATGGAGTGTCTCCTACTAAATTAAATGATCTCTGCTTTACTAATGTCTATGCAGATTCATTTGTAATTAGAGGTGATGAAGTCAGACAAATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAAATTACCAGATGATTTTACAGGCTGCGTTATAGCTTGGAATTCTAACAATCTTGATTCTAAGGTTGGTGGTAATTATAATTACCTGTATAGATTGTTTAGGAAGTCTAATCTCAAACCTTTTGAGAGAGATATTTCAACTGAAATCTATCAGGCCGGTAGCACACCTTGTAATGGTGTTGAAGGTTTTAATTGTTACTTTCCTTTACAATCATATGGTTTCCAACCCACTAATGGTGTTGGTTACCAACCATACAGAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTGGTTAAAAACAAATGTGTCAATTTCAACTTCAATGGTTTAACAGGCACAGGTGTTCTTACTGAGTCTAACAAAAAGTTTCTGCCTTTCCAACAATTTGGCAGAGACATTGCTGACACTACTGATGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTAACCAGGTTGCTGTTCTTTATCAGGATGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGTTTAATAGGGGCTGAACATGTCAACAACTCATATGAGTGTGACATACCCATTGGTGCAGGTATATGCGCTAGTTATCAGACTCAGACTAATTCTCCTCGGCGGGCACGTAGTGTAGCTAGTCAATCCATCATTGCCTACACTATGTCACTTGGTGCAGAAAATTCAGTTGCTTACTCTAATAACTCTATTGCCATACCCACAAATTTTACTATTAGTGTTACCACAGAAATTCTACCAGTGTCTATGACCAAGACATCAGTAGATTGTACAATGTACATTTGTGGTGATTCAACTGAATGCAGCAATCTTTTGTTGCAATATGGCAGTTTTTGTACACAATTAAACCGTGCTTTAACTGGAATAGCTGTTGAACAAGACAAAAACACCCAAGAAGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCTACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGTGATTGCCTTGGTGATATTGCTGCTAGAGACCTCATTTGTGCACAAAAGTTTAACGGCCTTACTGTTTTGCCACCTTTGCTCACAGATGAAATGATTGCTCAATACACTTCTGCACTGTTAGCGGGTACAATCACTTCTGGTTGGACCTTTGGTGCAGGTGCTGCATTACAAATACCATTTGCTATGCAAATGGCTTATAGGTTTAATGGTATTGGAGTTACACAGAATGTTCTCTATGAGAACCAAAAATTGATTGCCAACCAATTTAATAGTGCTATTGGCAAAATTCAAGACTCACTTTCTTCCACAGCAAGTGCACTTGGAAAACTTCAAGATGTGGTCAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGCTCCAATTTTGGTGCAATTTCAAGTGTTTTAAATGATATCCTTTCACGTCTTGACAAAGTTGAGGCTGAAGTGCAAATTGATAGGTTGATCACAGGCAGACTTCAAAGTTTGCAGACATATGTGACTCAACAATTAATTAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCTACTAAAATGTCAGAGTGTGTACTTGGACAATCAAAAAGAGTTGATTTTTGTGGAAAGGGCTATCATCTTATGTCCTTCCCTCAGTCAGCACCTCATGGTGTAGTCTTCTTGCATGTGACTTATGTCCCTGCACAAGAAAAGAACTTCACAACTGCTCCTGCCATTTGTCATGATGGAAAAGCACACTTTCCTCGTGAAGGTGTCTTTGTTTCAAATGGCACACACTGGTTTGTAACACAAAGGAATTTTTATGAACCACAAATCATTACTACAGACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAATTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGATTTAGGTGACATCTCTGGCATTAATGCTTCAGTTGTAAACATTCAAAAAGAAATTGACCGCCTCAATGAGGTTGCCAAGAATTTAAATGAATCTCTCATCGATCTCCAAGAACTTGGAAAGTATGAGCAGTATATAAAATGGCCATGGTACATTTGGCTAGGTTTTATAGCTGGCTTGATTGCCATAGTAATGGTGACAATTATGCTTTGCTGTATGACCAGTTGCTGTAGTTGTCTCAAGGGCTGTTGTTCTTGTGGATCCTGCTGCAAATTTGATGAAGACGACTCTGAGCCAGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATCTTCACAATTGGAACTGTAACTTTGAAGCAAGGTGAAATCAAGGATGCTACTCCTTCAGATTTTGTTCGCGCTACTGCAACGATACCGATACAAGCCTCACTCCCTTTCGGATGGCTTATTGTTGGCGTTGCACTTCTTGCTGTTTTTCAGAGCGCTTCCAAAATCATAACCCTCAAAAAGAGATGGCAACTAGCACTCTCCAAGGGTGTTCACTTTGTTTGCAACTTGCTGTTGTTGTTTGTAACAGTTTACTCACACCTTTTGCTCGTTGCTGCTGGCCTTGAAGCCCCTTTTCTCTATCTTTATGCTTTAGTCTACTTCTTGCAGAGTATAAACTTTGTAAGAATAATAATGAGGCTTTGGCTTTGCTGGAAATGCCGTTCCAAAAACCCATTACTTTATGATGCCAACTATTTTCTTTGCTGGCATACTAATTGTTACGACTATTGTATACCTTACAATAGTGTAACTTCTTCAATTGTCATTACTTCAGGTGATGGCACAACAAGTCCTATTTCTGAACATGACTACCAGATTGGTGGTTATACTGAAAAATGGGAATCTGGAGTAAAAGACTGTGTTGTATTACACAGTTACTTCACTTCAGACTATTACCAGCTGTACTCAACTCAATTGAGTACAGACACTGGTGTTGAACATGTTACCTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTCCAAATTCACACAATCGACGGTTCATCCGGAGTTGTTAATCCAGTAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTCGTGGTATTCTTGCTAGTTACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTCTGGTCTAAACGAACTAAATATTATATTAGTTTTTCTGTTTGGAACTTTAATTTTAGCCATGGCAGATTCCAACGGTACTATTACCGTTGAAGAGCTTAAAAAGCTCCTTGAACAATGGAACCTAGTAATAGGTTTCCTATTCCTTACATGGATTTGTCTTCTACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTATATAATTAAGTTAATTTTCCTCTGGCTGTTATGGCCAGTAACTTTAGCTTGTTTTGTGCTTGCTGCTGTTTACAGAATAAATTGGATCACCGGTGGAATTGCTATCGCAATGGCTTGTCTTGTAGGCTTGATGTGGCTCAGCTACTTCATTGCTTCTTTCAGACTGTTTGCGCGTACGCGTTCCATGTGGTCATTCAATCCAGAAACTAACATTCTTCTCAACGTGCCACTCCATGGCACTATTCTGACCAGACCGCTTCTAGAAAGTGAACTCGTAATCGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCTGGACACCATCTAGGACGCTGTGACATCAAGGACCTGCCTAAAGAAATCACTGTTGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGCAACTATAAATTAAACACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTACTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTACATCATAAACCTCATAATTAAAAATTTATCTAAGTCACTAACTGAGAATAAATATTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACGAACATGAAAATTATTCTTTTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAGTGTGTTAGAGGTACAACAGTACTTTTAAAAGAACCTTGCTCTTCTGGAACATACGAGGGCAATTCACCATTTCATCCTCTAGCTGATAACAAATTTGCACTGACTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGCGTAAAACACGTCTATCAGTTACGTGCCAGATCAGTTTCACCTAAACTGTTCATCAGACAAGAGGAAGTTCAAGAACTTTACTCTCCAATTTTTCTTATTGTTGCGGCAATAGTGTTTATAACACTTTGCTTCACACTCAAAAGAAAGACAGAATGATTGAACTTTCATTAATTGACTTCTATTTGTGCTTTTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTTATTATCTTTTGGTTCTCACTTGAACTGCAAGATCATAATGAAACTTGTCACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAATCATCACAACTGTAGCTGCATTTCACCAAGAATGTAGTTTACAGTCATGTACTCAACATCAACCATATGTAGTTGATGACCCGTGTCCTATTCACTTCTATTCTAAATGGTATATTAGAGTAGGAGCTAGAAAATCAGCACCTTTAATTGAATTGTGCGTGGATGAGGCTGGTTCTAAATCACCCATTCAGTACATCGATATCGGTAATTATACAGTTTCCTGTTTACCTTTTACAATTAATTGCCAGGAACCTAAATTGGGTAGTCTTGTAGTGCGTTGTTCGTTCTATGAAGACTTTTTAGAGTATCATGACGTTCGTGTTGTTTTAGATTTCATCTAAACGAACAAACTAAAATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAAACTCATGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATCTCACATAGCAATCTTTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACATTTTCACCGAGGCCACGCGGAGTACGATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAATGACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n"
     ]
    }
   ],
   "source": [
    "print(patient_rec.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(29903), strand=1), type='source'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(265), strand=1), type=\"5'UTR\"),\n",
       " SeqFeature(FeatureLocation(ExactPosition(265), ExactPosition(21555), strand=1), type='gene'),\n",
       " SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(265), ExactPosition(13468), strand=1), FeatureLocation(ExactPosition(13467), ExactPosition(21555), strand=1)], 'join'), type='CDS', location_operator='join'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(21562), ExactPosition(25384), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(21562), ExactPosition(25384), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(25392), ExactPosition(26220), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(25392), ExactPosition(26220), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26244), ExactPosition(26472), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26244), ExactPosition(26472), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26522), ExactPosition(27191), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26522), ExactPosition(27191), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27201), ExactPosition(27387), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27201), ExactPosition(27387), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27393), ExactPosition(27759), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27393), ExactPosition(27759), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27893), ExactPosition(28259), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27893), ExactPosition(28259), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(28273), ExactPosition(29533), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(28273), ExactPosition(29533), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29557), ExactPosition(29674), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29557), ExactPosition(29674), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29674), ExactPosition(29903), strand=1), type=\"3'UTR\")]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_rec.features\n",
    "# CDS=coding sequence"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sgsgsgsg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIWWW"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function qblast in module Bio.Blast.NCBIWWW:\n",
      "\n",
      "qblast(program, database, sequence, url_base='https://blast.ncbi.nlm.nih.gov/Blast.cgi', auto_format=None, composition_based_statistics=None, db_genetic_code=None, endpoints=None, entrez_query='(none)', expect=10.0, filter=None, gapcosts=None, genetic_code=None, hitlist_size=50, i_thresh=None, layout=None, lcase_mask=None, matrix_name=None, nucl_penalty=None, nucl_reward=None, other_advanced=None, perc_ident=None, phi_pattern=None, query_file=None, query_believe_defline=None, query_from=None, query_to=None, searchsp_eff=None, service=None, threshold=None, ungapped_alignment=None, word_size=None, short_query=None, alignments=500, alignment_view=None, descriptions=500, entrez_links_new_window=None, expect_low=None, expect_high=None, format_entrez_query=None, format_object=None, format_type='XML', ncbi_gi=None, results_file=None, show_overview=None, megablast=None, template_type=None, template_length=None)\n",
      "    BLAST search using NCBI's QBLAST server or a cloud service provider.\n",
      "    \n",
      "    Supports all parameters of the old qblast API for Put and Get.\n",
      "    \n",
      "    Please note that NCBI uses the new Common URL API for BLAST searches\n",
      "    on the internet (http://ncbi.github.io/blast-cloud/dev/api.html). Thus,\n",
      "    some of the parameters used by this function are not (or are no longer)\n",
      "    officially supported by NCBI. Although they are still functioning, this\n",
      "    may change in the future.\n",
      "    \n",
      "    The Common URL API (http://ncbi.github.io/blast-cloud/dev/api.html) allows\n",
      "    doing BLAST searches on cloud servers. To use this feature, please set\n",
      "    ``url_base='http://host.my.cloud.service.provider.com/cgi-bin/blast.cgi'``\n",
      "    and ``format_object='Alignment'``. For more details, please see\n",
      "    https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=CloudBlast\n",
      "    \n",
      "    Some useful parameters:\n",
      "    \n",
      "     - program        blastn, blastp, blastx, tblastn, or tblastx (lower case)\n",
      "     - database       Which database to search against (e.g. \"nr\").\n",
      "     - sequence       The sequence to search.\n",
      "     - ncbi_gi        TRUE/FALSE whether to give 'gi' identifier.\n",
      "     - descriptions   Number of descriptions to show.  Def 500.\n",
      "     - alignments     Number of alignments to show.  Def 500.\n",
      "     - expect         An expect value cutoff.  Def 10.0.\n",
      "     - matrix_name    Specify an alt. matrix (PAM30, PAM70, BLOSUM80, BLOSUM45).\n",
      "     - filter         \"none\" turns off filtering.  Default no filtering\n",
      "     - format_type    \"HTML\", \"Text\", \"ASN.1\", or \"XML\".  Def. \"XML\".\n",
      "     - entrez_query   Entrez query to limit Blast search\n",
      "     - hitlist_size   Number of hits to return. Default 50\n",
      "     - megablast      TRUE/FALSE whether to use MEga BLAST algorithm (blastn only)\n",
      "     - short_query    TRUE/FALSE whether to adjust the search parameters for a\n",
      "                      short query sequence. Note that this will override\n",
      "                      manually set parameters like word size and e value. Turns\n",
      "                      off when sequence length is > 30 residues. Default: None.\n",
      "     - service        plain, psi, phi, rpsblast, megablast (lower case)\n",
      "    \n",
      "    This function does no checking of the validity of the parameters\n",
      "    and passes the values to the server as is.  More help is available at:\n",
      "    https://ncbi.github.io/blast-cloud/dev/api.html\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(NCBIWWW.qblast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 92 ms, sys: 44 ms, total: 136 ms\n",
      "Wall time: 10min 14s\n"
     ]
    }
   ],
   "source": [
    "%time result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", record_dict['MN908947.3'].seq)\n",
    "with open(\"Files/my_blast.xml\", \"w\") as out_handle:\n",
    "    out_handle.write(result_handle.read())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIXML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = open(\"Files/my_blast.xml\")\n",
    "blast_records = list(NCBIXML.parse(result_handle))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![UML diagram of the BlastRecord](Media/BlastRecord.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1798174254|ref|NC_045512.2| Wuhan seafood market pneumonia virus isolate Wuhan-Hu-1, complete genome >gi|1798172431|gb|MN908947.3| Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29903\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293633|gb|MT019531.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-03/2019, complete genome\n",
      "length    : 29899\n",
      "e value   : 0.0\n",
      "identities: 29898\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293611|gb|MT019529.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-01/2019, complete genome\n",
      "length    : 29899\n",
      "e value   : 0.0\n",
      "identities: 29896\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633808|gb|MN996528.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV04, complete genome\n",
      "length    : 29891\n",
      "e value   : 0.0\n",
      "identities: 29891\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808633715|gb|MT049951.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/Yunnan-01/human/2020/CHN, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29898\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293644|gb|MT019532.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-04/2019, complete genome\n",
      "length    : 29890\n",
      "e value   : 0.0\n",
      "identities: 29890\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800455117|gb|MN988668.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU01, complete genome >gi|1800455123|gb|MN988669.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU02, complete genome\n",
      "length    : 29881\n",
      "e value   : 0.0\n",
      "identities: 29881\n",
      "TTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860439|gb|MT039890.1| Severe acute respiratory syndrome coronavirus 2 isolate SNU01, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29894\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293655|gb|MT019533.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-05/2020, complete genome\n",
      "length    : 29883\n",
      "e value   : 0.0\n",
      "identities: 29882\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800242661|gb|MN975262.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-005b_2020, complete genome\n",
      "length    : 29891\n",
      "e value   : 0.0\n",
      "identities: 29886\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802471970|gb|MN994468.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA2/2020, complete genome\n",
      "length    : 29883\n",
      "e value   : 0.0\n",
      "identities: 29881\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1806553209|gb|MT027064.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA5/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29880\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1806553187|gb|MT027062.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA3/2020, complete genome >gi|1806553198|gb|MT027063.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA4/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800408777|gb|MN985325.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1/2020, complete genome >gi|1805599854|gb|MT020880.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-A12/2020, complete genome >gi|1805599865|gb|MT020881.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-F6/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293622|gb|MT019530.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-02/2019, complete genome\n",
      "length    : 29889\n",
      "e value   : 0.0\n",
      "identities: 29883\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1803016604|gb|MT007544.1| Severe acute respiratory syndrome coronavirus 2 isolate Australia/VIC01/2020, complete genome\n",
      "length    : 29893\n",
      "e value   : 0.0\n",
      "identities: 29890\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802498786|gb|MN997409.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-AZ1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29878\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860452|gb|MT039887.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WI1/2020, complete genome\n",
      "length    : 29879\n",
      "e value   : 0.0\n",
      "identities: 29878\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860463|gb|MT039888.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-MA1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29877\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800489756|gb|MN988713.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294619|gb|MT093571.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/01/human/2020/SWE, complete genome\n",
      "length    : 29886\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816337|dbj|LC522974.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-501/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816315|dbj|LC522972.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/KY/V-029/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153870|emb|LR757996.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29868\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "CAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153869|emb|LR757995.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29872\n",
      "e value   : 0.0\n",
      "identities: 29869\n",
      "TTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816348|dbj|LC522975.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-521/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29873\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816326|dbj|LC522973.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-012/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29873\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1809484504|gb|MT066176.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU02/2020/TWN, complete genome\n",
      "length    : 29870\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1809484493|gb|MT066175.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU01/2020/TWN, complete genome\n",
      "length    : 29870\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808500211|gb|MT044257.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL2/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29875\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802471959|gb|MN994467.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29875\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294674|gb|MT093631.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/WH-09/human/2020/CHN, complete genome\n",
      "length    : 29911\n",
      "e value   : 0.0\n",
      "identities: 29863\n",
      "TTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294674|gb|MT093631.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/WH-09/human/2020/CHN, complete genome\n",
      "length    : 29911\n",
      "e value   : 6.60229e-13\n",
      "identities: 51\n",
      "AGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTAT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTAT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153872|emb|LR757998.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29866\n",
      "e value   : 0.0\n",
      "identities: 29864\n",
      "AACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633830|gb|MN996530.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV06, complete genome\n",
      "length    : 29854\n",
      "e value   : 0.0\n",
      "identities: 29854\n",
      "CCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633841|gb|MN996531.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV07, complete genome\n",
      "length    : 29857\n",
      "e value   : 0.0\n",
      "identities: 29855\n",
      "ACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633819|gb|MN996529.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV05, complete genome\n",
      "length    : 29852\n",
      "e value   : 0.0\n",
      "identities: 29850\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860410|gb|MT039873.1| Severe acute respiratory syndrome coronavirus 2 isolate HZ-1, complete genome\n",
      "length    : 29833\n",
      "e value   : 0.0\n",
      "identities: 29833\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800242639|gb|MN938384.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-002a_2020, complete genome\n",
      "length    : 29838\n",
      "e value   : 0.0\n",
      "identities: 29835\n",
      "CAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808500222|gb|MT044258.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA6/2020, complete genome\n",
      "length    : 29858\n",
      "e value   : 0.0\n",
      "identities: 29858\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633797|gb|MN996527.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV02, complete genome\n",
      "length    : 29825\n",
      "e value   : 0.0\n",
      "identities: 29823\n",
      "AACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCAT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCAT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1810678290|gb|MT072688.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS0CoV-2/61-TW/human/2020/ NPL, complete genome\n",
      "length    : 29811\n",
      "e value   : 0.0\n",
      "identities: 29810\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633852|gb|MN996532.1| Bat coronavirus RaTG13, complete genome\n",
      "length    : 29855\n",
      "e value   : 0.0\n",
      "identities: 28717\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "||| ||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CTTTCCAGGTAACAAACCAACGAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805599841|gb|MT020781.1| Severe acute respiratory syndrome coronavirus 2 isolate nCoV-FIN-29-Jan-2020, partial genome\n",
      "length    : 29847\n",
      "e value   : 0.0\n",
      "identities: 21581\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805599841|gb|MT020781.1| Severe acute respiratory syndrome coronavirus 2 isolate nCoV-FIN-29-Jan-2020, partial genome\n",
      "length    : 29847\n",
      "e value   : 0.0\n",
      "identities: 8103\n",
      "ATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125417|gb|MG772933.1| Bat SARS-like coronavirus isolate bat-SL-CoVZC45, complete genome\n",
      "length    : 29802\n",
      "e value   : 0.0\n",
      "identities: 20145\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| |||||||||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125417|gb|MG772933.1| Bat SARS-like coronavirus isolate bat-SL-CoVZC45, complete genome\n",
      "length    : 29802\n",
      "e value   : 0.0\n",
      "identities: 6063\n",
      "TACCAACCATACA-GAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAA...\n",
      "|||||| | |||| | || || ||  | || ||||| |||||| |||||||||| || ||||||||||| |||  ...\n",
      "TACCAAGC-TACAAGGGTTGTTGTTTTGTCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACCAAAACT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541441|gb|KY417146.1| Bat SARS-like coronavirus isolate Rs4231, complete genome\n",
      "length    : 29782\n",
      "e value   : 0.0\n",
      "identities: 21517\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAACGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541441|gb|KY417146.1| Bat SARS-like coronavirus isolate Rs4231, complete genome\n",
      "length    : 29782\n",
      "e value   : 0.0\n",
      "identities: 2393\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 17067\n",
      "ACTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTG...\n",
      "|||| ||||   |||| ||| |  |||||||||||| | ||||||||||| | |||||||||||| |||||||| ...\n",
      "ACTAGTGAACAAAATAATTTCACAGGTTATTTAAAATTAACTGACAATGTTTTCATTAAAAATGCTGACATTGTA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 6048\n",
      "TCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTGGTTAAAAACAAATGT...\n",
      "|| ||||| |||||| |||||||||| || ||||||||||| |||   || ||  |  | || |||||  | || ...\n",
      "TCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACCAAAATTATCCACACAACTAGTAAAAAATCAGTGC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 2983\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| |||||||||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 19382\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAATGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 2114\n",
      "TAAACGAACATGAAAATTATTCTTTTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAG...\n",
      "||||||||||||||||||||||| ||| || || ||||  |||| ||  |||| ||||| |||||||| || |||...\n",
      "TAAACGAACATGAAAATTATTCTCTTCCTGACATTGATTGCACTTGCATCTTGCGAGCTATATCACTATCAGGAG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 2404\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 14872\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAACGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 5645\n",
      "AGAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTG...\n",
      "|| || || || ||||| ||||| ||| || ||||||| || || |||||||||||||||   || ||     | ...\n",
      "AGGGTTGTTGTGCTTTCATTTGAGCTTTTAAATGCACCTGCTACAGTTTGTGGACCTAAACTATCCACAGGACTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 2389\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 1.19056e-66\n",
      "identities: 375\n",
      "TGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGA...\n",
      "||| ||||||||||| ||| ||||  ||  ||  || ||| |  | |  | |||||||||||| || || || ||...\n",
      "TGCTGCTTATTATGTTGGTAATCTAAAATATACCACCTTTATGCTTAGTTTTAATGAAAATGGGACTATCACTGA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 2.80736e-11\n",
      "identities: 78\n",
      "TTTGATAACCCTGTCCTACCATTTAATGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGC...\n",
      "|||||||||||| |  ||   |||  |||||| |||||||| ||  | ||||| ||||||||  |||| ||||||...\n",
      "TTTGATAACCCTATTTTAGACTTTGGTGATGGCGTTTATTTCGCCGCTACTGAAAAGTCTAATGTAATCAGAGGC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 14838\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| ||| ||||||||||||||||| |||||   ||||||   ||  ||||||||  | || ||  |||  |  ...\n",
      "TTTACTGGCTATTTAAAACTTACTGATAATGTTGCCATTAAGTGTGTTGACATTGTTAAGGAGGCACAAAGCGCC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 6107\n",
      "TGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGA...\n",
      "||| || |||||||| || |||||| | ||||| || ||| | ||| ||| ||||   ||||| || |||    |...\n",
      "TGCTGCATATTATGTAGGCTATCTTAAGCCTAGAACATTTATGTTACAATTTAATAGCAATGGCACAATTGTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 2304\n",
      "TCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAA---TT...\n",
      "|||||||||||||||||||||||||||||||||| ||||||||||||| ||||||||  |||||||||||   ||...\n",
      "TCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCGCTCGGCTGCATGCCTAGTGCACCTACGCAGTATAAATATT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 14075\n",
      "CTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTAC...\n",
      "||| || || | |||||| || || || |||||  |  ||   || ||||  || ||   ||| ||| |||| ||...\n",
      "CTTTAAAAAGGGTGCTCCTTACATTGTTGGTGACATCATTAGCGAAGGTGAATTGACCTGTGTTGTTTTACCAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 3948\n",
      "GCTGGTGCTGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACC...\n",
      "||||  ||| | ||||| | ||||||  ||||  |||||   ||  | ||  |  | | | | || ||||| |||...\n",
      "GCTGACGCTTCTGCTTACTTTGTGGGCCATCTAAAACCTCTCACCATGCTTGTTGACTTTGACGAGAATGGCACC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 1359\n",
      "TTCGTGTTGTTTTAGATTTC----ATCTAAACGAACAAACTAAAATGTCTGATAATGGACCC-CAAAATCAGCGA...\n",
      "|| || |||| ||||| ||     ||||||||||||||  ||||||| ||||||||||||   |||| || ||| ...\n",
      "TTTGTTTTGTGTTAGAATTACAAGATCTAAACGAACAA--TAAAATGACTGATAATGGACAATCAAACTC-GCGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 1409\n",
      "TTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGT...\n",
      "|||||||||||||| ||||||||| ||||||||||  |||||||| ||||||||||  | ||||    |||||||...\n",
      "TTTAAAATCTGTGTAGCTGTCACTTGGCTGCATGCCCAGTGCACTTACGCAGTATATCTTATAAACTTTTACTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 1.11507e-60\n",
      "identities: 335\n",
      "CCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATG...\n",
      "|| |||||| |||| || || ||| |  | ||| | || || |  ||  | ||||||||  |  | |||  | | ...\n",
      "CCTGTTTGTGTTAATGGTCTAATGCTCTTAGAATTGAAGGAAAAGGAGCAATACTGTGCATTGTCTCCTTCTCTA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 6.18363e-07\n",
      "identities: 117\n",
      "TGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACTTTAGATTC...\n",
      "||||||||||||||||| |   || ||||| |||||  |  |  |||| |||||||||||| | || |||||   ...\n",
      "TGATGGTGTTTATTTTGGTCTAACCGAGAAATCTAATGTTTTTCGAGGTTGGATTTTTGGTTCGACATTAGACAA...\n"
     ]
    }
   ],
   "source": [
    "E_VALUE_THRESH = 0.01\n",
    "print(len(blast_records[0].alignments))\n",
    "for alignment in blast_records[0].alignments:\n",
    "    for hsp in alignment.hsps:  # HSPs = high-scoring pairs\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            print(\"\\n****Alignment****\")\n",
    "            print(\"sequence  :\", alignment.title)\n",
    "            print(\"length    :\", alignment.length)\n",
    "            print(\"e value   :\", hsp.expect)\n",
    "            print(\"identities:\", hsp.identities)\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.match[0:75] + \"...\")\n",
    "            print(hsp.sbjct[0:75] + \"...\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"roger.guimera@urv.cat\"  # Always tell NCBI who you are\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"MG772933.1\", rettype=\"gb\", retmode=\"text\")\n",
    "record = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('ATATTAGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGT...AAA', IUPACAmbiguousDNA())"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq1 = record.seq\n",
    "seq2 = patient_rec.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Align"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligner = Align.PairwiseAligner()\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: 0.000000\n",
      "  target_internal_open_gap_score: 0.000000\n",
      "  target_internal_extend_gap_score: 0.000000\n",
      "  target_left_open_gap_score: 0.000000\n",
      "  target_left_extend_gap_score: 0.000000\n",
      "  target_right_open_gap_score: 0.000000\n",
      "  target_right_extend_gap_score: 0.000000\n",
      "  query_internal_open_gap_score: 0.000000\n",
      "  query_internal_extend_gap_score: 0.000000\n",
      "  query_left_open_gap_score: 0.000000\n",
      "  query_left_extend_gap_score: 0.000000\n",
      "  query_right_open_gap_score: 0.000000\n",
      "  query_right_extend_gap_score: 0.000000\n",
      "  mode: global\n",
      "\n",
      "Needleman-Wunsch\n"
     ]
    }
   ],
   "source": [
    "print(aligner)\n",
    "print(aligner.algorithm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rguimera/.local/lib/python3.5/site-packages/Bio/Align/substitution_matrices/__init__.py:21: BiopythonExperimentalWarning: Bio.Align.substitution_matrices is an experimental module which may still undergo significant changes. In particular, the location of this module may change, and the Array class defined in this module may be moved to other existing or new modules in Biopython.\n",
      "  BiopythonExperimentalWarning)\n"
     ]
    }
   ],
   "source": [
    "from Bio.Align import substitution_matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BENNER22 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3B.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.5 -1.2 -0.2 -0.3 -3.1  0.8 -1.6 -0.4 -1.0 -1.7 -0.8  0.0  0.8 -0.9 -1.2  1.3  1.4  0.4 -5.5 -3.5\n",
      "C -1.2 12.6 -3.7 -4.3 -0.1 -1.7 -1.5 -2.4 -3.3 -2.6 -2.5 -1.9 -3.1 -3.3 -1.6  0.3 -1.1 -1.7  0.5  0.6\n",
      "D -0.2 -3.7  4.8  3.9 -5.4  0.7  0.3 -4.0  0.2 -4.9 -3.9  2.4 -1.8  0.6 -1.0  0.1 -0.7 -3.0 -6.4 -3.0\n",
      "E -0.3 -4.3  3.9  4.6 -5.7  0.5 -0.2 -3.6  1.0 -4.4 -3.4  1.2 -1.7  1.7 -0.1 -0.5 -0.9 -2.7 -6.3 -4.0\n",
      "F -3.1 -0.1 -5.4 -5.7  7.7 -5.8  0.3  0.5 -5.1  2.2  0.7 -3.5 -3.4 -3.6 -4.3 -2.2 -2.6 -0.1  0.5  5.9\n",
      "G  0.8 -1.7  0.7  0.5 -5.8  6.2 -2.0 -3.8 -1.0 -4.9 -3.8  0.4 -1.8 -1.4 -0.7  0.6 -0.7 -2.5 -4.5 -4.8\n",
      "H -1.6 -1.5  0.3 -0.2  0.3 -2.0  6.1 -3.2  0.8 -2.1 -2.4  1.4 -0.4  2.4  1.5 -0.5 -1.1 -3.0 -2.7  3.7\n",
      "I -0.4 -2.4 -4.0 -3.6  0.5 -3.8 -3.2  4.2 -3.0  2.7  3.1 -2.7 -2.3 -2.7 -3.2 -1.4  0.3  3.6 -4.4 -2.2\n",
      "K -1.0 -3.3  0.2  1.0 -5.1 -1.0  0.8 -3.0  4.4 -3.3 -2.0  1.0 -1.6  2.2  3.9 -0.4 -0.4 -2.7 -3.7 -3.6\n",
      "L -1.7 -2.6 -4.9 -4.4  2.2 -4.9 -2.1  2.7 -3.3  4.6  3.2 -3.5 -1.3 -2.0 -2.9 -2.1 -1.0  2.0 -1.8 -0.7\n",
      "M -0.8 -2.5 -3.9 -3.4  0.7 -3.8 -2.4  3.1 -2.0  3.2  4.9 -2.6 -2.0 -1.7 -2.1 -1.5  0.1  2.5 -2.8 -1.8\n",
      "N  0.0 -1.9  2.4  1.2 -3.5  0.4  1.4 -2.7  1.0 -3.5 -2.6  3.3 -1.1  0.5  0.4  1.1  0.5 -2.3 -5.2 -1.2\n",
      "P  0.8 -3.1 -1.8 -1.7 -3.4 -1.8 -0.4 -2.3 -1.6 -1.3 -2.0 -1.1  7.0 -0.1 -1.2  1.1  0.4 -1.7 -5.8 -3.5\n",
      "Q -0.9 -3.3  0.6  1.7 -3.6 -1.4  2.4 -2.7  2.2 -2.0 -1.7  0.5 -0.1  4.2  2.2 -0.6 -0.7 -2.4 -3.3 -1.9\n",
      "R -1.2 -1.6 -1.0 -0.1 -4.3 -0.7  1.5 -3.2  3.9 -2.9 -2.1  0.4 -1.2  2.2  5.0 -0.5 -0.7 -2.9 -1.1 -2.7\n",
      "S  1.3  0.3  0.1 -0.5 -2.2  0.6 -0.5 -1.4 -0.4 -2.1 -1.5  1.1  1.1 -0.6 -0.5  2.0  1.5 -0.9 -3.9 -1.9\n",
      "T  1.4 -1.1 -0.7 -0.9 -2.6 -0.7 -1.1  0.3 -0.4 -1.0  0.1  0.5  0.4 -0.7 -0.7  1.5  2.5  0.4 -4.5 -3.0\n",
      "V  0.4 -1.7 -3.0 -2.7 -0.1 -2.5 -3.0  3.6 -2.7  2.0  2.5 -2.3 -1.7 -2.4 -2.9 -0.9  0.4  3.7 -4.5 -2.6\n",
      "W -5.5  0.5 -6.4 -6.3  0.5 -4.5 -2.7 -4.4 -3.7 -1.8 -2.8 -5.2 -5.8 -3.3 -1.1 -3.9 -4.5 -4.5 15.7  1.5\n",
      "Y -3.5  0.6 -3.0 -4.0  5.9 -4.8  3.7 -2.2 -3.6 -0.7 -1.8 -1.2 -3.5 -1.9 -2.7 -1.9 -3.0 -2.6  1.5  9.0\n",
      "\n",
      "BENNER6 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3A.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.5 -1.7 -0.6 -0.7 -3.2  0.8 -2.1  0.1 -1.9 -1.3 -0.2  0.0  1.1 -1.7 -1.7  1.4  1.7  0.7 -4.3 -4.0\n",
      "C -1.7 12.1 -3.7 -4.7 -0.1 -1.3 -1.2 -3.6 -2.8 -3.8 -3.7 -1.6 -2.7 -3.2 -0.4  0.9 -1.5 -3.1  1.6  2.6\n",
      "D -0.6 -3.7  5.2  4.4 -5.7  0.8  0.1 -4.2 -0.2 -5.3 -4.3  2.5 -2.8  0.6 -1.5 -0.4 -1.2 -3.3 -6.3 -2.3\n",
      "E -0.7 -4.7  4.4  5.2 -6.7  0.5 -0.2 -4.1  0.9 -5.0 -4.1  1.1 -2.6  2.1 -0.4 -1.2 -1.6 -3.0 -5.6 -4.1\n",
      "F -3.2 -0.1 -5.7 -6.7  8.3 -5.7  0.1  0.0 -6.3  2.4 -0.1 -3.5 -3.2 -4.4 -4.9 -1.8 -2.4 -0.5 -1.6  5.6\n",
      "G  0.8 -1.3  0.8  0.5 -5.7  5.8 -2.1 -3.4 -1.4 -4.6 -3.7 -0.1 -1.7 -1.6 -0.1  0.8 -0.5 -2.3 -1.7 -4.9\n",
      "H -2.1 -1.2  0.1 -0.2  0.1 -2.1  6.1 -3.7  0.9 -2.2 -3.4  1.4 -0.4  3.2  1.8 -0.9 -1.7 -3.8 -2.8  4.4\n",
      "I  0.1 -3.6 -4.2 -4.1  0.0 -3.4 -3.7  4.4 -3.8  2.4  4.0 -2.5 -2.0 -3.8 -3.8 -1.2  0.7  3.9 -5.0 -3.3\n",
      "K -1.9 -2.8 -0.2  0.9 -6.3 -1.4  0.9 -3.8  5.6 -4.1 -2.9  1.0 -2.3  2.5  4.3 -1.2 -1.1 -3.8 -1.4 -4.0\n",
      "L -1.3 -3.8 -5.3 -5.0  2.4 -4.6 -2.2  2.4 -4.1  4.8 -2.9 -3.4 -0.2 -2.4 -3.2 -1.5 -0.4  1.9 -3.0 -1.6\n",
      "M -0.2 -3.7 -4.3 -4.1 -0.1 -3.7 -3.4  4.0 -2.9 -2.9  4.8 -2.5 -1.8 -3.1 -3.0 -1.3  0.6  3.3 -4.4 -3.6\n",
      "N  0.0 -1.6  2.5  1.1 -3.5 -0.1  1.4 -2.5  1.0 -3.4 -2.5  3.6 -1.1  0.1 -0.1  1.2  0.5 -2.4 -4.4 -0.9\n",
      "P  1.1 -2.7 -2.8 -2.6 -3.2 -1.7 -0.4 -2.0 -2.3 -0.2 -1.8 -1.1  6.5  0.1 -1.3  1.4  0.6 -1.6 -4.8 -3.8\n",
      "Q -1.7 -3.2  0.6  2.1 -4.4 -1.6  3.2 -3.8  2.5 -2.4 -3.1  0.1  0.1  5.3  2.5 -1.4 -1.7 -3.5 -2.6 -1.4\n",
      "R -1.7 -0.4 -1.5 -0.4 -4.9 -0.1  1.8 -3.8  4.3 -3.2 -3.0 -0.1 -1.3  2.5  5.1 -0.9 -1.3 -3.7  2.0 -2.6\n",
      "S  1.4  0.9 -0.4 -1.2 -1.8  0.8 -0.9 -1.2 -1.2 -1.5 -1.3  1.2  1.4 -1.4 -0.9  2.1  1.5 -0.9 -2.9 -1.8\n",
      "T  1.7 -1.5 -1.2 -1.6 -2.4 -0.5 -1.7  0.7 -1.1 -0.4  0.6  0.5  0.6 -1.7 -1.3  1.5  2.4  0.6 -2.6 -3.4\n",
      "V  0.7 -3.1 -3.3 -3.0 -0.5 -2.3 -3.8  3.9 -3.8  1.9  3.3 -2.4 -1.6 -3.5 -3.7 -0.9  0.6  4.0 -4.8 -3.8\n",
      "W -4.3  1.6 -6.3 -5.6 -1.6 -1.7 -2.8 -5.0 -1.4 -3.0 -4.4 -4.4 -4.8 -2.6  2.0 -2.9 -2.6 -4.8 14.7 -0.3\n",
      "Y -4.0  2.6 -2.3 -4.1  5.6 -4.9  4.4 -3.3 -4.0 -1.6 -3.6 -0.9 -3.8 -1.4 -2.6 -1.8 -3.4 -3.8 -0.3  9.5\n",
      "\n",
      "BENNER74 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3C.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.4  0.3 -0.3 -0.1 -2.6  0.6 -1.0 -0.8 -0.4 -1.4 -0.8 -0.2  0.4 -0.3 -0.8  1.1  0.7  0.1 -4.1 -2.6\n",
      "C  0.3 11.8 -3.2 -3.2 -0.7 -2.0 -1.3 -1.2 -2.9 -1.6 -1.2 -1.8 -3.1 -2.6 -2.2  0.1 -0.6 -0.2 -0.9 -0.4\n",
      "D -0.3 -3.2  4.8  2.9 -4.7  0.2  0.4 -3.9  0.4 -4.2 -3.2  2.2 -1.0  0.8 -0.5  0.4 -0.2 -2.9 -5.5 -2.8\n",
      "E -0.1 -3.2  2.9  3.7 -4.3 -0.5  0.2 -2.9  1.2 -3.1 -2.2  1.0 -0.7  1.7  0.3  0.1 -0.2 -2.1 -4.7 -3.0\n",
      "F -2.6 -0.7 -4.7 -4.3  7.2 -5.4  0.0  0.9 -3.6  2.1  1.3 -3.2 -3.8 -2.8 -3.5 -2.6 -2.2  0.1  3.0  5.3\n",
      "G  0.6 -2.0  0.2 -0.5 -5.4  6.6 -1.6 -4.3 -1.1 -4.6 -3.5  0.4 -1.7 -1.1 -1.0  0.4 -1.0 -3.1 -4.1 -4.3\n",
      "H -1.0 -1.3  0.4  0.2  0.0 -1.6  6.1 -2.3  0.6 -1.9 -1.5  1.2 -1.0  1.4  1.0 -0.3 -0.5 -2.1 -1.0  2.5\n",
      "I -0.8 -1.2 -3.9 -2.9  0.9 -4.3 -2.3  4.0 -2.3  2.8  2.6 -2.8 -2.6 -2.0 -2.6 -1.8 -0.3  3.2 -2.3 -1.0\n",
      "K -0.4 -2.9  0.4  1.2 -3.6 -1.1  0.6 -2.3  3.4 -2.4 -1.5  0.9 -0.8  1.7  2.9  0.0  0.1 -1.9 -3.6 -2.4\n",
      "L -1.4 -1.6 -4.2 -3.1  2.1 -4.6 -1.9  2.8 -2.4  4.2  2.9 -3.1 -2.2 -1.7 -2.4 -2.2 -1.1  1.9 -0.9 -0.1\n",
      "M -0.8 -1.2 -3.2 -2.2  1.3 -3.5 -1.5  2.6 -1.5  2.9  4.5 -2.2 -2.4 -1.0 -1.8 -1.4 -0.4  1.8 -1.3 -0.5\n",
      "N -0.2 -1.8  2.2  1.0 -3.2  0.4  1.2 -2.8  0.9 -3.1 -2.2  3.6 -1.0  0.7  0.3  0.9  0.4 -2.2 -4.0 -1.4\n",
      "P  0.4 -3.1 -1.0 -0.7 -3.8 -1.7 -1.0 -2.6 -0.8 -2.2 -2.4 -1.0  7.5 -0.2 -0.1  0.5  0.1 -1.9 -5.2 -3.4\n",
      "Q -0.3 -2.6  0.8  1.7 -2.8 -1.1  1.4 -2.0  1.7 -1.7 -1.0  0.7 -0.2  3.0  1.6  0.1 -0.1 -1.7 -2.8 -1.8\n",
      "R -0.8 -2.2 -0.5  0.3 -3.5 -1.0  1.0 -2.6  2.9 -2.4 -1.8  0.3 -0.1  1.6  4.8 -0.2 -0.3 -2.2 -1.6 -2.0\n",
      "S  1.1  0.1  0.4  0.1 -2.6  0.4 -0.3 -1.8  0.0 -2.2 -1.4  0.9  0.5  0.1 -0.2  2.1  1.4 -1.0 -3.4 -1.9\n",
      "T  0.7 -0.6 -0.2 -0.2 -2.2 -1.0 -0.5 -0.3  0.1 -1.1 -0.4  0.4  0.1 -0.1 -0.3  1.4  2.5  0.2 -3.7 -2.1\n",
      "V  0.1 -0.2 -2.9 -2.1  0.1 -3.1 -2.1  3.2 -1.9  1.9  1.8 -2.2 -1.9 -1.7 -2.2 -1.0  0.2  3.4 -2.9 -1.4\n",
      "W -4.1 -0.9 -5.5 -4.7  3.0 -4.1 -1.0 -2.3 -3.6 -0.9 -1.3 -4.0 -5.2 -2.8 -1.6 -3.4 -3.7 -2.9 14.7  3.6\n",
      "Y -2.6 -0.4 -2.8 -3.0  5.3 -4.3  2.5 -1.0 -2.4 -0.1 -0.5 -1.4 -3.4 -1.8 -2.0 -1.9 -2.1 -1.4  3.6  8.1\n",
      "\n",
      "BLOSUM45 #  Matrix made by matblas from blosum45.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 45\n",
      "#  Entropy =   0.3795, Expected =  -0.2789\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -1.0 -2.0 -1.0 -1.0 -1.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  0.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -5.0\n",
      "R -2.0  7.0  0.0 -1.0 -3.0  1.0  0.0 -2.0  0.0 -3.0 -2.0  3.0 -1.0 -2.0 -2.0 -1.0 -1.0 -2.0 -1.0 -2.0 -1.0  0.0 -1.0 -5.0\n",
      "N -1.0  0.0  6.0  2.0 -2.0  0.0  0.0  0.0  1.0 -2.0 -3.0  0.0 -2.0 -2.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  4.0  0.0 -1.0 -5.0\n",
      "D -2.0 -1.0  2.0  7.0 -3.0  0.0  2.0 -1.0  0.0 -4.0 -3.0  0.0 -3.0 -4.0 -1.0  0.0 -1.0 -4.0 -2.0 -3.0  5.0  1.0 -1.0 -5.0\n",
      "C -1.0 -3.0 -2.0 -3.0 12.0 -3.0 -3.0 -3.0 -3.0 -3.0 -2.0 -3.0 -2.0 -2.0 -4.0 -1.0 -1.0 -5.0 -3.0 -1.0 -2.0 -3.0 -2.0 -5.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  6.0  2.0 -2.0  1.0 -2.0 -2.0  1.0  0.0 -4.0 -1.0  0.0 -1.0 -2.0 -1.0 -3.0  0.0  4.0 -1.0 -5.0\n",
      "E -1.0  0.0  0.0  2.0 -3.0  2.0  6.0 -2.0  0.0 -3.0 -2.0  1.0 -2.0 -3.0  0.0  0.0 -1.0 -3.0 -2.0 -3.0  1.0  4.0 -1.0 -5.0\n",
      "G  0.0 -2.0  0.0 -1.0 -3.0 -2.0 -2.0  7.0 -2.0 -4.0 -3.0 -2.0 -2.0 -3.0 -2.0  0.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -1.0 -5.0\n",
      "H -2.0  0.0  1.0  0.0 -3.0  1.0  0.0 -2.0 10.0 -3.0 -2.0 -1.0  0.0 -2.0 -2.0 -1.0 -2.0 -3.0  2.0 -3.0  0.0  0.0 -1.0 -5.0\n",
      "I -1.0 -3.0 -2.0 -4.0 -3.0 -2.0 -3.0 -4.0 -3.0  5.0  2.0 -3.0  2.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0  3.0 -3.0 -3.0 -1.0 -5.0\n",
      "L -1.0 -2.0 -3.0 -3.0 -2.0 -2.0 -2.0 -3.0 -2.0  2.0  5.0 -3.0  2.0  1.0 -3.0 -3.0 -1.0 -2.0  0.0  1.0 -3.0 -2.0 -1.0 -5.0\n",
      "K -1.0  3.0  0.0  0.0 -3.0  1.0  1.0 -2.0 -1.0 -3.0 -3.0  5.0 -1.0 -3.0 -1.0 -1.0 -1.0 -2.0 -1.0 -2.0  0.0  1.0 -1.0 -5.0\n",
      "M -1.0 -1.0 -2.0 -3.0 -2.0  0.0 -2.0 -2.0  0.0  2.0  2.0 -1.0  6.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0  1.0 -2.0 -1.0 -1.0 -5.0\n",
      "F -2.0 -2.0 -2.0 -4.0 -2.0 -4.0 -3.0 -3.0 -2.0  0.0  1.0 -3.0  0.0  8.0 -3.0 -2.0 -1.0  1.0  3.0  0.0 -3.0 -3.0 -1.0 -5.0\n",
      "P -1.0 -2.0 -2.0 -1.0 -4.0 -1.0  0.0 -2.0 -2.0 -2.0 -3.0 -1.0 -2.0 -3.0  9.0 -1.0 -1.0 -3.0 -3.0 -3.0 -2.0 -1.0 -1.0 -5.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0  0.0  0.0 -1.0 -2.0 -3.0 -1.0 -2.0 -2.0 -1.0  4.0  2.0 -4.0 -2.0 -1.0  0.0  0.0  0.0 -5.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  2.0  5.0 -3.0 -1.0  0.0  0.0 -1.0  0.0 -5.0\n",
      "W -2.0 -2.0 -4.0 -4.0 -5.0 -2.0 -3.0 -2.0 -3.0 -2.0 -2.0 -2.0 -2.0  1.0 -3.0 -4.0 -3.0 15.0  3.0 -3.0 -4.0 -2.0 -2.0 -5.0\n",
      "Y -2.0 -1.0 -2.0 -2.0 -3.0 -1.0 -2.0 -3.0  2.0  0.0  0.0 -1.0  0.0  3.0 -3.0 -2.0 -1.0  3.0  8.0 -1.0 -2.0 -2.0 -1.0 -5.0\n",
      "V  0.0 -2.0 -3.0 -3.0 -1.0 -3.0 -3.0 -3.0 -3.0  3.0  1.0 -2.0  1.0  0.0 -3.0 -1.0  0.0 -3.0 -1.0  5.0 -3.0 -3.0 -1.0 -5.0\n",
      "B -1.0 -1.0  4.0  5.0 -2.0  0.0  1.0 -1.0  0.0 -3.0 -3.0  0.0 -2.0 -3.0 -2.0  0.0  0.0 -4.0 -2.0 -3.0  4.0  2.0 -1.0 -5.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  4.0  4.0 -2.0  0.0 -3.0 -2.0  1.0 -1.0 -3.0 -1.0  0.0 -1.0 -2.0 -2.0 -3.0  2.0  4.0 -1.0 -5.0\n",
      "X  0.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -5.0\n",
      "* -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0  1.0\n",
      "\n",
      "BLOSUM50 #  Matrix made by matblas from blosum50.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 50\n",
      "#  Entropy =   0.4808, Expected =  -0.3573\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -1.0 -2.0 -1.0 -1.0 -1.0  0.0 -2.0 -1.0 -2.0 -1.0 -1.0 -3.0 -1.0  1.0  0.0 -3.0 -2.0  0.0 -2.0 -1.0 -1.0 -5.0\n",
      "R -2.0  7.0 -1.0 -2.0 -4.0  1.0  0.0 -3.0  0.0 -4.0 -3.0  3.0 -2.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -3.0 -1.0  0.0 -1.0 -5.0\n",
      "N -1.0 -1.0  7.0  2.0 -2.0  0.0  0.0  0.0  1.0 -3.0 -4.0  0.0 -2.0 -4.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  4.0  0.0 -1.0 -5.0\n",
      "D -2.0 -2.0  2.0  8.0 -4.0  0.0  2.0 -1.0 -1.0 -4.0 -4.0 -1.0 -4.0 -5.0 -1.0  0.0 -1.0 -5.0 -3.0 -4.0  5.0  1.0 -1.0 -5.0\n",
      "C -1.0 -4.0 -2.0 -4.0 13.0 -3.0 -3.0 -3.0 -3.0 -2.0 -2.0 -3.0 -2.0 -2.0 -4.0 -1.0 -1.0 -5.0 -3.0 -1.0 -3.0 -3.0 -2.0 -5.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  7.0  2.0 -2.0  1.0 -3.0 -2.0  2.0  0.0 -4.0 -1.0  0.0 -1.0 -1.0 -1.0 -3.0  0.0  4.0 -1.0 -5.0\n",
      "E -1.0  0.0  0.0  2.0 -3.0  2.0  6.0 -3.0  0.0 -4.0 -3.0  1.0 -2.0 -3.0 -1.0 -1.0 -1.0 -3.0 -2.0 -3.0  1.0  5.0 -1.0 -5.0\n",
      "G  0.0 -3.0  0.0 -1.0 -3.0 -2.0 -3.0  8.0 -2.0 -4.0 -4.0 -2.0 -3.0 -4.0 -2.0  0.0 -2.0 -3.0 -3.0 -4.0 -1.0 -2.0 -2.0 -5.0\n",
      "H -2.0  0.0  1.0 -1.0 -3.0  1.0  0.0 -2.0 10.0 -4.0 -3.0  0.0 -1.0 -1.0 -2.0 -1.0 -2.0 -3.0  2.0 -4.0  0.0  0.0 -1.0 -5.0\n",
      "I -1.0 -4.0 -3.0 -4.0 -2.0 -3.0 -4.0 -4.0 -4.0  5.0  2.0 -3.0  2.0  0.0 -3.0 -3.0 -1.0 -3.0 -1.0  4.0 -4.0 -3.0 -1.0 -5.0\n",
      "L -2.0 -3.0 -4.0 -4.0 -2.0 -2.0 -3.0 -4.0 -3.0  2.0  5.0 -3.0  3.0  1.0 -4.0 -3.0 -1.0 -2.0 -1.0  1.0 -4.0 -3.0 -1.0 -5.0\n",
      "K -1.0  3.0  0.0 -1.0 -3.0  2.0  1.0 -2.0  0.0 -3.0 -3.0  6.0 -2.0 -4.0 -1.0  0.0 -1.0 -3.0 -2.0 -3.0  0.0  1.0 -1.0 -5.0\n",
      "M -1.0 -2.0 -2.0 -4.0 -2.0  0.0 -2.0 -3.0 -1.0  2.0  3.0 -2.0  7.0  0.0 -3.0 -2.0 -1.0 -1.0  0.0  1.0 -3.0 -1.0 -1.0 -5.0\n",
      "F -3.0 -3.0 -4.0 -5.0 -2.0 -4.0 -3.0 -4.0 -1.0  0.0  1.0 -4.0  0.0  8.0 -4.0 -3.0 -2.0  1.0  4.0 -1.0 -4.0 -4.0 -2.0 -5.0\n",
      "P -1.0 -3.0 -2.0 -1.0 -4.0 -1.0 -1.0 -2.0 -2.0 -3.0 -4.0 -1.0 -3.0 -4.0 10.0 -1.0 -1.0 -4.0 -3.0 -3.0 -2.0 -1.0 -2.0 -5.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0 -1.0  0.0 -1.0 -3.0 -3.0  0.0 -2.0 -3.0 -1.0  5.0  2.0 -4.0 -2.0 -2.0  0.0  0.0 -1.0 -5.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  2.0  5.0 -3.0 -2.0  0.0  0.0 -1.0  0.0 -5.0\n",
      "W -3.0 -3.0 -4.0 -5.0 -5.0 -1.0 -3.0 -3.0 -3.0 -3.0 -2.0 -3.0 -1.0  1.0 -4.0 -4.0 -3.0 15.0  2.0 -3.0 -5.0 -2.0 -3.0 -5.0\n",
      "Y -2.0 -1.0 -2.0 -3.0 -3.0 -1.0 -2.0 -3.0  2.0 -1.0 -1.0 -2.0  0.0  4.0 -3.0 -2.0 -2.0  2.0  8.0 -1.0 -3.0 -2.0 -1.0 -5.0\n",
      "V  0.0 -3.0 -3.0 -4.0 -1.0 -3.0 -3.0 -4.0 -4.0  4.0  1.0 -3.0  1.0 -1.0 -3.0 -2.0  0.0 -3.0 -1.0  5.0 -4.0 -3.0 -1.0 -5.0\n",
      "B -2.0 -1.0  4.0  5.0 -3.0  0.0  1.0 -1.0  0.0 -4.0 -4.0  0.0 -3.0 -4.0 -2.0  0.0  0.0 -5.0 -3.0 -4.0  5.0  2.0 -1.0 -5.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  4.0  5.0 -2.0  0.0 -3.0 -3.0  1.0 -1.0 -4.0 -1.0  0.0 -1.0 -2.0 -2.0 -3.0  2.0  5.0 -1.0 -5.0\n",
      "X -1.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0  0.0 -3.0 -1.0 -1.0 -1.0 -1.0 -1.0 -5.0\n",
      "* -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0  1.0\n",
      "\n",
      "BLOSUM62 #  Matrix made by matblas from blosum62.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/2 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 62\n",
      "#  Entropy =   0.6979, Expected =  -0.5209\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  4.0 -1.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  0.0 -3.0 -2.0  0.0 -2.0 -1.0  0.0 -4.0\n",
      "R -1.0  5.0  0.0 -2.0 -3.0  1.0  0.0 -2.0  0.0 -3.0 -2.0  2.0 -1.0 -3.0 -2.0 -1.0 -1.0 -3.0 -2.0 -3.0 -1.0  0.0 -1.0 -4.0\n",
      "N -2.0  0.0  6.0  1.0 -3.0  0.0  0.0  0.0  1.0 -3.0 -3.0  0.0 -2.0 -3.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  3.0  0.0 -1.0 -4.0\n",
      "D -2.0 -2.0  1.0  6.0 -3.0  0.0  2.0 -1.0 -1.0 -3.0 -4.0 -1.0 -3.0 -3.0 -1.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0\n",
      "C  0.0 -3.0 -3.0 -3.0  9.0 -3.0 -4.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -2.0 -3.0 -1.0 -1.0 -2.0 -2.0 -1.0 -3.0 -3.0 -2.0 -4.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  5.0  2.0 -2.0  0.0 -3.0 -2.0  1.0  0.0 -3.0 -1.0  0.0 -1.0 -2.0 -1.0 -2.0  0.0  3.0 -1.0 -4.0\n",
      "E -1.0  0.0  0.0  2.0 -4.0  2.0  5.0 -2.0  0.0 -3.0 -3.0  1.0 -2.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0\n",
      "G  0.0 -2.0  0.0 -1.0 -3.0 -2.0 -2.0  6.0 -2.0 -4.0 -4.0 -2.0 -3.0 -3.0 -2.0  0.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -1.0 -4.0\n",
      "H -2.0  0.0  1.0 -1.0 -3.0  0.0  0.0 -2.0  8.0 -3.0 -3.0 -1.0 -2.0 -1.0 -2.0 -1.0 -2.0 -2.0  2.0 -3.0  0.0  0.0 -1.0 -4.0\n",
      "I -1.0 -3.0 -3.0 -3.0 -1.0 -3.0 -3.0 -4.0 -3.0  4.0  2.0 -3.0  1.0  0.0 -3.0 -2.0 -1.0 -3.0 -1.0  3.0 -3.0 -3.0 -1.0 -4.0\n",
      "L -1.0 -2.0 -3.0 -4.0 -1.0 -2.0 -3.0 -4.0 -3.0  2.0  4.0 -2.0  2.0  0.0 -3.0 -2.0 -1.0 -2.0 -1.0  1.0 -4.0 -3.0 -1.0 -4.0\n",
      "K -1.0  2.0  0.0 -1.0 -3.0  1.0  1.0 -2.0 -1.0 -3.0 -2.0  5.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  0.0  1.0 -1.0 -4.0\n",
      "M -1.0 -1.0 -2.0 -3.0 -1.0  0.0 -2.0 -3.0 -2.0  1.0  2.0 -1.0  5.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0  1.0 -3.0 -1.0 -1.0 -4.0\n",
      "F -2.0 -3.0 -3.0 -3.0 -2.0 -3.0 -3.0 -3.0 -1.0  0.0  0.0 -3.0  0.0  6.0 -4.0 -2.0 -2.0  1.0  3.0 -1.0 -3.0 -3.0 -1.0 -4.0\n",
      "P -1.0 -2.0 -2.0 -1.0 -3.0 -1.0 -1.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -4.0  7.0 -1.0 -1.0 -4.0 -3.0 -2.0 -2.0 -1.0 -2.0 -4.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0  0.0  0.0 -1.0 -2.0 -2.0  0.0 -1.0 -2.0 -1.0  4.0  1.0 -3.0 -2.0 -2.0  0.0  0.0  0.0 -4.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  5.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -4.0\n",
      "W -3.0 -3.0 -4.0 -4.0 -2.0 -2.0 -3.0 -2.0 -2.0 -3.0 -2.0 -3.0 -1.0  1.0 -4.0 -3.0 -2.0 11.0  2.0 -3.0 -4.0 -3.0 -2.0 -4.0\n",
      "Y -2.0 -2.0 -2.0 -3.0 -2.0 -1.0 -2.0 -3.0  2.0 -1.0 -1.0 -2.0 -1.0  3.0 -3.0 -2.0 -2.0  2.0  7.0 -1.0 -3.0 -2.0 -1.0 -4.0\n",
      "V  0.0 -3.0 -3.0 -3.0 -1.0 -2.0 -2.0 -3.0 -3.0  3.0  1.0 -2.0  1.0 -1.0 -2.0 -2.0  0.0 -3.0 -1.0  4.0 -3.0 -2.0 -1.0 -4.0\n",
      "B -2.0 -1.0  3.0  4.0 -3.0  0.0  1.0 -1.0  0.0 -3.0 -4.0  0.0 -3.0 -3.0 -2.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  3.0  4.0 -2.0  0.0 -3.0 -3.0  1.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0\n",
      "X  0.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0  0.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -4.0\n",
      "* -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0  1.0\n",
      "\n",
      "BLOSUM80 #  Matrix made by matblas from blosum80_3.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 80\n",
      "#  Entropy =   0.9868, Expected =  -0.7442\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  7.0 -3.0 -3.0 -3.0 -1.0 -2.0 -2.0  0.0 -3.0 -3.0 -3.0 -1.0 -2.0 -4.0 -1.0  2.0  0.0 -5.0 -4.0 -1.0 -3.0 -2.0 -1.0 -8.0\n",
      "R -3.0  9.0 -1.0 -3.0 -6.0  1.0 -1.0 -4.0  0.0 -5.0 -4.0  3.0 -3.0 -5.0 -3.0 -2.0 -2.0 -5.0 -4.0 -4.0 -2.0  0.0 -2.0 -8.0\n",
      "N -3.0 -1.0  9.0  2.0 -5.0  0.0 -1.0 -1.0  1.0 -6.0 -6.0  0.0 -4.0 -6.0 -4.0  1.0  0.0 -7.0 -4.0 -5.0  5.0 -1.0 -2.0 -8.0\n",
      "D -3.0 -3.0  2.0 10.0 -7.0 -1.0  2.0 -3.0 -2.0 -7.0 -7.0 -2.0 -6.0 -6.0 -3.0 -1.0 -2.0 -8.0 -6.0 -6.0  6.0  1.0 -3.0 -8.0\n",
      "C -1.0 -6.0 -5.0 -7.0 13.0 -5.0 -7.0 -6.0 -7.0 -2.0 -3.0 -6.0 -3.0 -4.0 -6.0 -2.0 -2.0 -5.0 -5.0 -2.0 -6.0 -7.0 -4.0 -8.0\n",
      "Q -2.0  1.0  0.0 -1.0 -5.0  9.0  3.0 -4.0  1.0 -5.0 -4.0  2.0 -1.0 -5.0 -3.0 -1.0 -1.0 -4.0 -3.0 -4.0 -1.0  5.0 -2.0 -8.0\n",
      "E -2.0 -1.0 -1.0  2.0 -7.0  3.0  8.0 -4.0  0.0 -6.0 -6.0  1.0 -4.0 -6.0 -2.0 -1.0 -2.0 -6.0 -5.0 -4.0  1.0  6.0 -2.0 -8.0\n",
      "G  0.0 -4.0 -1.0 -3.0 -6.0 -4.0 -4.0  9.0 -4.0 -7.0 -7.0 -3.0 -5.0 -6.0 -5.0 -1.0 -3.0 -6.0 -6.0 -6.0 -2.0 -4.0 -3.0 -8.0\n",
      "H -3.0  0.0  1.0 -2.0 -7.0  1.0  0.0 -4.0 12.0 -6.0 -5.0 -1.0 -4.0 -2.0 -4.0 -2.0 -3.0 -4.0  3.0 -5.0 -1.0  0.0 -2.0 -8.0\n",
      "I -3.0 -5.0 -6.0 -7.0 -2.0 -5.0 -6.0 -7.0 -6.0  7.0  2.0 -5.0  2.0 -1.0 -5.0 -4.0 -2.0 -5.0 -3.0  4.0 -6.0 -6.0 -2.0 -8.0\n",
      "L -3.0 -4.0 -6.0 -7.0 -3.0 -4.0 -6.0 -7.0 -5.0  2.0  6.0 -4.0  3.0  0.0 -5.0 -4.0 -3.0 -4.0 -2.0  1.0 -7.0 -5.0 -2.0 -8.0\n",
      "K -1.0  3.0  0.0 -2.0 -6.0  2.0  1.0 -3.0 -1.0 -5.0 -4.0  8.0 -3.0 -5.0 -2.0 -1.0 -1.0 -6.0 -4.0 -4.0 -1.0  1.0 -2.0 -8.0\n",
      "M -2.0 -3.0 -4.0 -6.0 -3.0 -1.0 -4.0 -5.0 -4.0  2.0  3.0 -3.0  9.0  0.0 -4.0 -3.0 -1.0 -3.0 -3.0  1.0 -5.0 -3.0 -2.0 -8.0\n",
      "F -4.0 -5.0 -6.0 -6.0 -4.0 -5.0 -6.0 -6.0 -2.0 -1.0  0.0 -5.0  0.0 10.0 -6.0 -4.0 -4.0  0.0  4.0 -2.0 -6.0 -6.0 -3.0 -8.0\n",
      "P -1.0 -3.0 -4.0 -3.0 -6.0 -3.0 -2.0 -5.0 -4.0 -5.0 -5.0 -2.0 -4.0 -6.0 12.0 -2.0 -3.0 -7.0 -6.0 -4.0 -4.0 -2.0 -3.0 -8.0\n",
      "S  2.0 -2.0  1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -2.0 -4.0 -4.0 -1.0 -3.0 -4.0 -2.0  7.0  2.0 -6.0 -3.0 -3.0  0.0 -1.0 -1.0 -8.0\n",
      "T  0.0 -2.0  0.0 -2.0 -2.0 -1.0 -2.0 -3.0 -3.0 -2.0 -3.0 -1.0 -1.0 -4.0 -3.0  2.0  8.0 -5.0 -3.0  0.0 -1.0 -2.0 -1.0 -8.0\n",
      "W -5.0 -5.0 -7.0 -8.0 -5.0 -4.0 -6.0 -6.0 -4.0 -5.0 -4.0 -6.0 -3.0  0.0 -7.0 -6.0 -5.0 16.0  3.0 -5.0 -8.0 -5.0 -5.0 -8.0\n",
      "Y -4.0 -4.0 -4.0 -6.0 -5.0 -3.0 -5.0 -6.0  3.0 -3.0 -2.0 -4.0 -3.0  4.0 -6.0 -3.0 -3.0  3.0 11.0 -3.0 -5.0 -4.0 -3.0 -8.0\n",
      "V -1.0 -4.0 -5.0 -6.0 -2.0 -4.0 -4.0 -6.0 -5.0  4.0  1.0 -4.0  1.0 -2.0 -4.0 -3.0  0.0 -5.0 -3.0  7.0 -6.0 -4.0 -2.0 -8.0\n",
      "B -3.0 -2.0  5.0  6.0 -6.0 -1.0  1.0 -2.0 -1.0 -6.0 -7.0 -1.0 -5.0 -6.0 -4.0  0.0 -1.0 -8.0 -5.0 -6.0  6.0  0.0 -3.0 -8.0\n",
      "Z -2.0  0.0 -1.0  1.0 -7.0  5.0  6.0 -4.0  0.0 -6.0 -5.0  1.0 -3.0 -6.0 -2.0 -1.0 -2.0 -5.0 -4.0 -4.0  0.0  6.0 -1.0 -8.0\n",
      "X -1.0 -2.0 -2.0 -3.0 -4.0 -2.0 -2.0 -3.0 -2.0 -2.0 -2.0 -2.0 -2.0 -3.0 -3.0 -1.0 -1.0 -5.0 -3.0 -2.0 -3.0 -1.0 -2.0 -8.0\n",
      "* -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0  1.0\n",
      "\n",
      "BLOSUM90 #  Matrix made by matblas from blosum90.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/2 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 90\n",
      "#  Entropy =   1.1806, Expected =  -0.8887\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -2.0 -3.0 -1.0 -1.0 -1.0  0.0 -2.0 -2.0 -2.0 -1.0 -2.0 -3.0 -1.0  1.0  0.0 -4.0 -3.0 -1.0 -2.0 -1.0 -1.0 -6.0\n",
      "R -2.0  6.0 -1.0 -3.0 -5.0  1.0 -1.0 -3.0  0.0 -4.0 -3.0  2.0 -2.0 -4.0 -3.0 -1.0 -2.0 -4.0 -3.0 -3.0 -2.0  0.0 -2.0 -6.0\n",
      "N -2.0 -1.0  7.0  1.0 -4.0  0.0 -1.0 -1.0  0.0 -4.0 -4.0  0.0 -3.0 -4.0 -3.0  0.0  0.0 -5.0 -3.0 -4.0  4.0 -1.0 -2.0 -6.0\n",
      "D -3.0 -3.0  1.0  7.0 -5.0 -1.0  1.0 -2.0 -2.0 -5.0 -5.0 -1.0 -4.0 -5.0 -3.0 -1.0 -2.0 -6.0 -4.0 -5.0  4.0  0.0 -2.0 -6.0\n",
      "C -1.0 -5.0 -4.0 -5.0  9.0 -4.0 -6.0 -4.0 -5.0 -2.0 -2.0 -4.0 -2.0 -3.0 -4.0 -2.0 -2.0 -4.0 -4.0 -2.0 -4.0 -5.0 -3.0 -6.0\n",
      "Q -1.0  1.0  0.0 -1.0 -4.0  7.0  2.0 -3.0  1.0 -4.0 -3.0  1.0  0.0 -4.0 -2.0 -1.0 -1.0 -3.0 -3.0 -3.0 -1.0  4.0 -1.0 -6.0\n",
      "E -1.0 -1.0 -1.0  1.0 -6.0  2.0  6.0 -3.0 -1.0 -4.0 -4.0  0.0 -3.0 -5.0 -2.0 -1.0 -1.0 -5.0 -4.0 -3.0  0.0  4.0 -2.0 -6.0\n",
      "G  0.0 -3.0 -1.0 -2.0 -4.0 -3.0 -3.0  6.0 -3.0 -5.0 -5.0 -2.0 -4.0 -5.0 -3.0 -1.0 -3.0 -4.0 -5.0 -5.0 -2.0 -3.0 -2.0 -6.0\n",
      "H -2.0  0.0  0.0 -2.0 -5.0  1.0 -1.0 -3.0  8.0 -4.0 -4.0 -1.0 -3.0 -2.0 -3.0 -2.0 -2.0 -3.0  1.0 -4.0 -1.0  0.0 -2.0 -6.0\n",
      "I -2.0 -4.0 -4.0 -5.0 -2.0 -4.0 -4.0 -5.0 -4.0  5.0  1.0 -4.0  1.0 -1.0 -4.0 -3.0 -1.0 -4.0 -2.0  3.0 -5.0 -4.0 -2.0 -6.0\n",
      "L -2.0 -3.0 -4.0 -5.0 -2.0 -3.0 -4.0 -5.0 -4.0  1.0  5.0 -3.0  2.0  0.0 -4.0 -3.0 -2.0 -3.0 -2.0  0.0 -5.0 -4.0 -2.0 -6.0\n",
      "K -1.0  2.0  0.0 -1.0 -4.0  1.0  0.0 -2.0 -1.0 -4.0 -3.0  6.0 -2.0 -4.0 -2.0 -1.0 -1.0 -5.0 -3.0 -3.0 -1.0  1.0 -1.0 -6.0\n",
      "M -2.0 -2.0 -3.0 -4.0 -2.0  0.0 -3.0 -4.0 -3.0  1.0  2.0 -2.0  7.0 -1.0 -3.0 -2.0 -1.0 -2.0 -2.0  0.0 -4.0 -2.0 -1.0 -6.0\n",
      "F -3.0 -4.0 -4.0 -5.0 -3.0 -4.0 -5.0 -5.0 -2.0 -1.0  0.0 -4.0 -1.0  7.0 -4.0 -3.0 -3.0  0.0  3.0 -2.0 -4.0 -4.0 -2.0 -6.0\n",
      "P -1.0 -3.0 -3.0 -3.0 -4.0 -2.0 -2.0 -3.0 -3.0 -4.0 -4.0 -2.0 -3.0 -4.0  8.0 -2.0 -2.0 -5.0 -4.0 -3.0 -3.0 -2.0 -2.0 -6.0\n",
      "S  1.0 -1.0  0.0 -1.0 -2.0 -1.0 -1.0 -1.0 -2.0 -3.0 -3.0 -1.0 -2.0 -3.0 -2.0  5.0  1.0 -4.0 -3.0 -2.0  0.0 -1.0 -1.0 -6.0\n",
      "T  0.0 -2.0  0.0 -2.0 -2.0 -1.0 -1.0 -3.0 -2.0 -1.0 -2.0 -1.0 -1.0 -3.0 -2.0  1.0  6.0 -4.0 -2.0 -1.0 -1.0 -1.0 -1.0 -6.0\n",
      "W -4.0 -4.0 -5.0 -6.0 -4.0 -3.0 -5.0 -4.0 -3.0 -4.0 -3.0 -5.0 -2.0  0.0 -5.0 -4.0 -4.0 11.0  2.0 -3.0 -6.0 -4.0 -3.0 -6.0\n",
      "Y -3.0 -3.0 -3.0 -4.0 -4.0 -3.0 -4.0 -5.0  1.0 -2.0 -2.0 -3.0 -2.0  3.0 -4.0 -3.0 -2.0  2.0  8.0 -3.0 -4.0 -3.0 -2.0 -6.0\n",
      "V -1.0 -3.0 -4.0 -5.0 -2.0 -3.0 -3.0 -5.0 -4.0  3.0  0.0 -3.0  0.0 -2.0 -3.0 -2.0 -1.0 -3.0 -3.0  5.0 -4.0 -3.0 -2.0 -6.0\n",
      "B -2.0 -2.0  4.0  4.0 -4.0 -1.0  0.0 -2.0 -1.0 -5.0 -5.0 -1.0 -4.0 -4.0 -3.0  0.0 -1.0 -6.0 -4.0 -4.0  4.0  0.0 -2.0 -6.0\n",
      "Z -1.0  0.0 -1.0  0.0 -5.0  4.0  4.0 -3.0  0.0 -4.0 -4.0  1.0 -2.0 -4.0 -2.0 -1.0 -1.0 -4.0 -3.0 -3.0  0.0  4.0 -1.0 -6.0\n",
      "X -1.0 -2.0 -2.0 -2.0 -3.0 -1.0 -2.0 -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -3.0 -2.0 -2.0 -2.0 -1.0 -2.0 -6.0\n",
      "* -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0  1.0\n",
      "\n",
      "DAYHOFF #  M.O. Dayhoff, R.M. Schwartz, and B.C. Orcutt:\n",
      "#  \"A Model of Evolutionary Change in Proteins.\"\n",
      "#  Margaret O. Dayhoff: Atlas of Protein Sequence and Structure,\n",
      "#  Volume 5, Supplement 3, 1978, pages 345-352.\n",
      "#  The National Biomedical Research Foundation, 1979.\n",
      "#  Figure 84, page 352.\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  0.2 -0.2  0.0  0.0 -0.4  0.1 -0.1 -0.1 -0.1 -0.2 -0.1  0.0  0.1  0.0 -0.2  0.1  0.1  0.0 -0.6 -0.3\n",
      "C -0.2  1.2 -0.5 -0.5 -0.4 -0.3 -0.3 -0.2 -0.5 -0.6 -0.5 -0.4 -0.3 -0.5 -0.4  0.0 -0.2 -0.2 -0.8  0.0\n",
      "D  0.0 -0.5  0.4  0.3 -0.6  0.1  0.1 -0.2  0.0 -0.4 -0.3  0.2 -0.1  0.2 -0.1  0.0  0.0 -0.2 -0.7 -0.4\n",
      "E  0.0 -0.5  0.3  0.4 -0.5  0.0  0.1 -0.2  0.0 -0.3 -0.2  0.1 -0.1  0.2 -0.1  0.0  0.0 -0.2 -0.7 -0.4\n",
      "F -0.4 -0.4 -0.6 -0.5  0.9 -0.5 -0.2  0.1 -0.5  0.2  0.0 -0.4 -0.5 -0.5 -0.4 -0.3 -0.3 -0.1  0.0  0.7\n",
      "G  0.1 -0.3  0.1  0.0 -0.5  0.5 -0.2 -0.3 -0.2 -0.4 -0.3  0.0 -0.1 -0.1 -0.3  0.1  0.0 -0.1 -0.7 -0.5\n",
      "H -0.1 -0.3  0.1  0.1 -0.2 -0.2  0.6 -0.2  0.0 -0.2 -0.2  0.2  0.0  0.3  0.2 -0.1 -0.1 -0.2 -0.3  0.0\n",
      "I -0.1 -0.2 -0.2 -0.2  0.1 -0.3 -0.2  0.5 -0.2  0.2  0.2 -0.2 -0.2 -0.2 -0.2 -0.1  0.0  0.4 -0.5 -0.1\n",
      "K -0.1 -0.5  0.0  0.0 -0.5 -0.2  0.0 -0.2  0.5 -0.3  0.0  0.1 -0.1  0.1  0.3  0.0  0.0 -0.2 -0.3 -0.4\n",
      "L -0.2 -0.6 -0.4 -0.3  0.2 -0.4 -0.2  0.2 -0.3  0.6  0.4 -0.3 -0.3 -0.2 -0.3 -0.3 -0.2  0.2 -0.2 -0.1\n",
      "M -0.1 -0.5 -0.3 -0.2  0.0 -0.3 -0.2  0.2  0.0  0.4  0.6 -0.2 -0.2 -0.1  0.0 -0.2 -0.1  0.2 -0.4 -0.2\n",
      "N  0.0 -0.4  0.2  0.1 -0.4  0.0  0.2 -0.2  0.1 -0.3 -0.2  0.2 -0.1  0.1  0.0  0.1  0.0 -0.2 -0.4 -0.2\n",
      "P  0.1 -0.3 -0.1 -0.1 -0.5 -0.1  0.0 -0.2 -0.1 -0.3 -0.2 -0.1  0.6  0.0  0.0  0.1  0.0 -0.1 -0.6 -0.5\n",
      "Q  0.0 -0.5  0.2  0.2 -0.5 -0.1  0.3 -0.2  0.1 -0.2 -0.1  0.1  0.0  0.4  0.1 -0.1 -0.1 -0.2 -0.5 -0.4\n",
      "R -0.2 -0.4 -0.1 -0.1 -0.4 -0.3  0.2 -0.2  0.3 -0.3  0.0  0.0  0.0  0.1  0.6  0.0 -0.1 -0.2  0.2 -0.4\n",
      "S  0.1  0.0  0.0  0.0 -0.3  0.1 -0.1 -0.1  0.0 -0.3 -0.2  0.1  0.1 -0.1  0.0  0.2  0.1 -0.1 -0.2 -0.3\n",
      "T  0.1 -0.2  0.0  0.0 -0.3  0.0 -0.1  0.0  0.0 -0.2 -0.1  0.0  0.0 -0.1 -0.1  0.1  0.3  0.0 -0.5 -0.3\n",
      "V  0.0 -0.2 -0.2 -0.2 -0.1 -0.1 -0.2  0.4 -0.2  0.2  0.2 -0.2 -0.1 -0.2 -0.2 -0.1  0.0  0.4 -0.6 -0.2\n",
      "W -0.6 -0.8 -0.7 -0.7  0.0 -0.7 -0.3 -0.5 -0.3 -0.2 -0.4 -0.4 -0.6 -0.5  0.2 -0.2 -0.5 -0.6  1.7  0.0\n",
      "Y -0.3  0.0 -0.4 -0.4  0.7 -0.5  0.0 -0.1 -0.4 -0.1 -0.2 -0.2 -0.5 -0.4 -0.4 -0.3 -0.3 -0.2  0.0  1.0\n",
      "\n",
      "FENG #  D.F. Feng, M.S. Johnson, R.F. Doolittle:\n",
      "#  \"Aligning amino acid sequences: Comparison of commonly used methods.\"\n",
      "#  Journal of Molecular Evolution 21(2): 112-125 (1985).\n",
      "#  Table 1, upper triangle.\n",
      "#  PMID 6100188\n",
      "    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y\n",
      "A 6.0 2.0 4.0 4.0 2.0 5.0 2.0 2.0 3.0 2.0 2.0 3.0 5.0 3.0 2.0 5.0 5.0 5.0 2.0 2.0\n",
      "C 2.0 6.0 1.0 0.0 3.0 3.0 2.0 2.0 0.0 2.0 2.0 2.0 2.0 1.0 2.0 4.0 2.0 2.0 3.0 3.0\n",
      "D 4.0 1.0 6.0 5.0 1.0 4.0 3.0 1.0 3.0 1.0 0.0 5.0 2.0 4.0 2.0 3.0 2.0 3.0 0.0 2.0\n",
      "E 4.0 0.0 5.0 6.0 0.0 4.0 2.0 1.0 4.0 1.0 1.0 3.0 3.0 4.0 2.0 3.0 3.0 4.0 1.0 1.0\n",
      "F 2.0 3.0 1.0 0.0 6.0 1.0 2.0 4.0 0.0 4.0 2.0 1.0 2.0 1.0 1.0 3.0 1.0 4.0 3.0 5.0\n",
      "G 5.0 3.0 4.0 4.0 1.0 6.0 1.0 2.0 2.0 2.0 1.0 3.0 3.0 2.0 3.0 5.0 2.0 4.0 3.0 2.0\n",
      "H 2.0 2.0 3.0 2.0 2.0 1.0 6.0 1.0 3.0 3.0 1.0 4.0 3.0 4.0 4.0 3.0 2.0 1.0 1.0 3.0\n",
      "I 2.0 2.0 1.0 1.0 4.0 2.0 1.0 6.0 2.0 5.0 4.0 2.0 2.0 1.0 2.0 2.0 3.0 5.0 2.0 3.0\n",
      "K 3.0 0.0 3.0 4.0 0.0 2.0 3.0 2.0 6.0 2.0 2.0 4.0 2.0 4.0 5.0 3.0 4.0 3.0 1.0 1.0\n",
      "L 2.0 2.0 1.0 1.0 4.0 2.0 3.0 5.0 2.0 6.0 5.0 1.0 3.0 2.0 2.0 2.0 2.0 5.0 4.0 3.0\n",
      "M 2.0 2.0 0.0 1.0 2.0 1.0 1.0 4.0 2.0 5.0 6.0 1.0 2.0 2.0 2.0 1.0 3.0 4.0 3.0 2.0\n",
      "N 3.0 2.0 5.0 3.0 1.0 3.0 4.0 2.0 4.0 1.0 1.0 6.0 2.0 3.0 2.0 5.0 4.0 2.0 0.0 3.0\n",
      "P 5.0 2.0 2.0 3.0 2.0 3.0 3.0 2.0 2.0 3.0 2.0 2.0 6.0 3.0 3.0 4.0 4.0 3.0 2.0 2.0\n",
      "Q 3.0 1.0 4.0 4.0 1.0 2.0 4.0 1.0 4.0 2.0 2.0 3.0 3.0 6.0 3.0 3.0 3.0 2.0 1.0 2.0\n",
      "R 2.0 2.0 2.0 2.0 1.0 3.0 4.0 2.0 5.0 2.0 2.0 2.0 3.0 3.0 6.0 3.0 3.0 2.0 2.0 1.0\n",
      "S 5.0 4.0 3.0 3.0 3.0 5.0 3.0 2.0 3.0 2.0 1.0 5.0 4.0 3.0 3.0 6.0 5.0 2.0 2.0 3.0\n",
      "T 5.0 2.0 2.0 3.0 1.0 2.0 2.0 3.0 4.0 2.0 3.0 4.0 4.0 3.0 3.0 5.0 6.0 3.0 1.0 2.0\n",
      "V 5.0 2.0 3.0 4.0 4.0 4.0 1.0 5.0 3.0 5.0 4.0 2.0 3.0 2.0 2.0 2.0 3.0 6.0 3.0 3.0\n",
      "W 2.0 3.0 0.0 1.0 3.0 3.0 1.0 2.0 1.0 4.0 3.0 0.0 2.0 1.0 2.0 2.0 1.0 3.0 6.0 3.0\n",
      "Y 2.0 3.0 2.0 1.0 5.0 2.0 3.0 3.0 1.0 3.0 2.0 3.0 2.0 2.0 1.0 3.0 2.0 3.0 3.0 6.0\n",
      "\n",
      "GENETIC #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences.\"\n",
      "#  Figure 5.\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  4.0 -1.9  1.0  1.3 -2.4  1.2 -2.1 -1.8 -1.9 -2.3 -2.0 -1.7  0.8 -2.1 -1.6  0.1  0.9  1.0 -2.2 -2.4\n",
      "C -1.9  5.5 -1.6 -3.0  1.8  1.0 -1.6 -1.9 -3.2 -1.3 -2.7 -1.5 -1.9 -3.1  0.7  1.5 -1.9 -2.2  4.1  2.6\n",
      "D  1.0 -1.6  4.8  3.8 -1.7  1.1  1.7 -2.1  0.3 -2.4 -2.5  1.7 -2.2  0.3 -2.3 -2.1 -2.1  1.0 -2.9  2.3\n",
      "E  1.3 -3.0  3.8  5.7 -2.9  1.4  0.3 -2.3  2.0 -2.5 -1.8  0.3 -2.1  2.0 -2.0 -2.8 -2.1  1.3 -3.2 -0.9\n",
      "F -2.4  1.8 -1.7 -2.9  4.5 -1.9 -1.1  1.3 -2.8  2.2  0.5 -1.3 -1.8 -2.1 -1.5  0.0 -2.1  1.0  0.0  2.0\n",
      "G  1.2  1.0  1.1  1.4 -1.9  4.2 -2.2 -2.5 -2.2 -2.2 -2.3 -2.6 -1.8 -2.1  0.8 -0.6 -2.1  1.1  1.4 -1.8\n",
      "H -2.1 -1.6  1.7  0.3 -1.1 -2.2  4.7 -1.8  0.6 -0.1 -1.8  1.8  0.7  3.6  3.6 -1.6 -1.8 -2.1 -2.1  2.3\n",
      "I -1.8 -1.9 -2.1 -2.3  1.3 -2.5 -1.8  4.1  0.7  1.2  3.3  0.9 -1.6 -1.9 -1.2 -0.5  0.8  1.0 -2.2 -1.6\n",
      "K -1.9 -3.2  0.3  2.0 -2.8 -2.2  0.6  0.7  5.6 -2.0  1.6  3.5 -1.5  2.2 -0.2 -1.5  1.0 -2.1 -3.0 -0.8\n",
      "L -2.3 -1.3 -2.4 -2.5  2.2 -2.2 -0.1  1.2 -2.0  3.4  1.5 -2.2  0.0  0.1 -0.4 -1.2 -1.9  1.1 -0.3 -1.6\n",
      "M -2.0 -2.7 -2.5 -1.8  0.5 -2.3 -1.8  3.3  1.6  1.5  5.4  0.1 -1.4 -1.2 -0.4 -1.3  0.7  1.0 -2.0 -2.9\n",
      "N -1.7 -1.5  1.7  0.3 -1.3 -2.6  1.8  0.9  3.5 -2.2  0.1  4.7 -1.6  0.4 -1.5 -0.3  0.9 -2.2 -3.0  2.5\n",
      "P  0.8 -1.9 -2.2 -2.1 -1.8 -1.8  0.7 -1.6 -1.5  0.0 -1.4 -1.6  3.8  1.0  0.3  0.4  1.1 -2.1 -1.6 -2.3\n",
      "Q -2.1 -3.1  0.3  2.0 -2.1 -2.1  3.6 -1.9  2.2  0.1 -1.2  0.4  1.0  5.5  0.3 -2.3 -1.7 -2.0 -2.3 -0.8\n",
      "R -1.6  0.7 -2.3 -2.0 -1.5  0.8  3.6 -1.2 -0.2 -0.4 -0.4 -1.5  0.3  0.3  2.9  0.3 -0.6 -2.1  1.8 -1.9\n",
      "S  0.1  1.5 -2.1 -2.8  0.0 -0.6 -1.6 -0.5 -1.5 -1.2 -1.3 -0.3  0.4 -2.3  0.3  2.6  1.0 -2.2  0.8  0.3\n",
      "T  0.9 -1.9 -2.1 -2.1 -2.1 -2.1 -1.8  0.8  1.0 -1.9  0.7  0.9  1.1 -1.7 -0.6  1.0  4.0 -2.2 -2.2 -2.1\n",
      "V  1.0 -2.2  1.0  1.3  1.0  1.1 -2.1  1.0 -2.1  1.1  1.0 -2.2 -2.1 -2.0 -2.1 -2.2 -2.2  4.1 -2.1 -2.2\n",
      "W -2.2  4.1 -2.9 -3.2  0.0  1.4 -2.1 -2.2 -3.0 -0.3 -2.0 -3.0 -1.6 -2.3  1.8  0.8 -2.2 -2.1  7.5 -0.5\n",
      "Y -2.4  2.6  2.3 -0.9  2.0 -1.8  2.3 -1.6 -0.8 -1.6 -2.9  2.5 -2.3 -0.8 -1.9  0.3 -2.1 -2.2 -0.5  6.5\n",
      "\n",
      "GONNET1992 #  Gaston H. Gonnet, Mark A. Cohen, Steven A. Benner:\n",
      "#  \"Exhaustive matching of the entire protein sequence database.\"\n",
      "#  Science 256(5062): 1443-1445 (1992).\n",
      "#  Figure 2.\n",
      "#  PMID 1604319\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.4  0.5 -0.3  0.0 -2.3  0.5 -0.8 -0.8 -0.4 -1.2 -0.7 -0.3  0.3 -0.2 -0.6  1.1  0.6  0.1 -3.6 -2.2\n",
      "C  0.5 11.5 -3.2 -3.0 -0.8 -2.0 -1.3 -1.1 -2.8 -1.5 -0.9 -1.8 -3.1 -2.4 -2.2  0.1 -0.5  0.0 -1.0 -0.5\n",
      "D -0.3 -3.2  4.7  2.7 -4.5  0.1  0.4 -3.8  0.5 -4.0 -3.0  2.2 -0.7  0.9 -0.3  0.5  0.0 -2.9 -5.2 -2.8\n",
      "E  0.0 -3.0  2.7  3.6 -3.9 -0.8  0.4 -2.7  1.2 -2.8 -2.0  0.9 -0.5  1.7  0.4  0.2 -0.1 -1.9 -4.3 -2.7\n",
      "F -2.3 -0.8 -4.5 -3.9  7.0 -5.2 -0.1  1.0 -3.3  2.0  1.6 -3.1 -3.8 -2.6 -3.2 -2.8 -2.2  0.1  3.6  5.1\n",
      "G  0.5 -2.0  0.1 -0.8 -5.2  6.6 -1.4 -4.5 -1.1 -4.4 -3.5  0.4 -1.6 -1.0 -1.0  0.4 -1.1 -3.3 -4.0 -4.0\n",
      "H -0.8 -1.3  0.4  0.4 -0.1 -1.4  6.0 -2.2  0.6 -1.9 -1.3  1.2 -1.1  1.2  0.6 -0.2 -0.3 -2.0 -0.8  2.2\n",
      "I -0.8 -1.1 -3.8 -2.7  1.0 -4.5 -2.2  4.0 -2.1  2.8  2.5 -2.8 -2.6 -1.9 -2.4 -1.8 -0.6  3.1 -1.8 -0.7\n",
      "K -0.4 -2.8  0.5  1.2 -3.3 -1.1  0.6 -2.1  3.2 -2.1 -1.4  0.8 -0.6  1.5  2.7  0.1  0.1 -1.7 -3.5 -2.1\n",
      "L -1.2 -1.5 -4.0 -2.8  2.0 -4.4 -1.9  2.8 -2.1  4.0  2.8 -3.0 -2.3 -1.6 -2.2 -2.1 -1.3  1.8 -0.7  0.0\n",
      "M -0.7 -0.9 -3.0 -2.0  1.6 -3.5 -1.3  2.5 -1.4  2.8  4.3 -2.2 -2.4 -1.0 -1.7 -1.4 -0.6  1.6 -1.0 -0.2\n",
      "N -0.3 -1.8  2.2  0.9 -3.1  0.4  1.2 -2.8  0.8 -3.0 -2.2  3.8 -0.9  0.7  0.3  0.9  0.5 -2.2 -3.6 -1.4\n",
      "P  0.3 -3.1 -0.7 -0.5 -3.8 -1.6 -1.1 -2.6 -0.6 -2.3 -2.4 -0.9  7.6 -0.2 -0.9  0.4  0.1 -1.8 -5.0 -3.1\n",
      "Q -0.2 -2.4  0.9  1.7 -2.6 -1.0  1.2 -1.9  1.5 -1.6 -1.0  0.7 -0.2  2.7  1.5  0.2  0.0 -1.5 -2.7 -1.7\n",
      "R -0.6 -2.2 -0.3  0.4 -3.2 -1.0  0.6 -2.4  2.7 -2.2 -1.7  0.3 -0.9  1.5  4.7 -0.2 -0.2 -2.0 -1.6 -1.8\n",
      "S  1.1  0.1  0.5  0.2 -2.8  0.4 -0.2 -1.8  0.1 -2.1 -1.4  0.9  0.4  0.2 -0.2  2.2  1.5 -1.0 -3.3 -1.9\n",
      "T  0.6 -0.5  0.0 -0.1 -2.2 -1.1 -0.3 -0.6  0.1 -1.3 -0.6  0.5  0.1  0.0 -0.2  1.5  2.5  0.0 -3.5 -1.9\n",
      "V  0.1  0.0 -2.9 -1.9  0.1 -3.3 -2.0  3.1 -1.7  1.8  1.6 -2.2 -1.8 -1.5 -2.0 -1.0  0.0  3.4 -2.6 -1.1\n",
      "W -3.6 -1.0 -5.2 -4.3  3.6 -4.0 -0.8 -1.8 -3.5 -0.7 -1.0 -3.6 -5.0 -2.7 -1.6 -3.3 -3.5 -2.6 14.2  4.1\n",
      "Y -2.2 -0.5 -2.8 -2.7  5.1 -4.0  2.2 -0.7 -2.1  0.0 -0.2 -1.4 -3.1 -1.7 -1.8 -1.9 -1.9 -1.1  4.1  7.8\n",
      "\n",
      "JOHNSON #  Mark S. Johnson and John P. Overington:\n",
      "#  \"A structural basis for sequence comparisons. An evaluation of scoring\n",
      "#  methodologies.\"\n",
      "#  Journal of Molecular Biology 233(4): 716-738 (1993).\n",
      "#  Table 3, upper triangle.\n",
      "#  PMID 8411177\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  0.6 -0.3 -0.2 -0.1 -0.3 -0.1 -0.3 -0.2 -0.1 -0.3 -0.1 -0.1 -0.1 -0.1 -0.2  0.0 -0.1 -0.1 -0.6 -0.4\n",
      "C -0.3  1.6 -1.0 -0.7 -0.4 -0.8 -0.8 -0.8 -0.9 -0.9 -0.4 -0.8 -0.9 -0.7 -0.6 -0.8 -0.6 -0.5 -0.9 -0.8\n",
      "D -0.2 -1.0  0.8  0.2 -0.7 -0.2 -0.1 -0.5 -0.1 -0.8 -0.6  0.3 -0.1 -0.1 -0.3 -0.0 -0.2 -0.5 -0.6 -0.4\n",
      "E -0.1 -0.7  0.2  0.9 -0.6 -0.2 -0.2 -0.5  0.1 -0.6 -0.3 -0.1 -0.1  0.2 -0.0 -0.2 -0.1 -0.4 -0.8 -0.4\n",
      "F -0.3 -0.4 -0.7 -0.6  1.0 -0.9 -0.2  0.1 -0.6  0.2 -0.1 -0.4 -0.5 -0.6 -0.6 -0.5 -0.5 -0.1  0.3  0.3\n",
      "G -0.1 -0.8 -0.2 -0.2 -0.9  0.8 -0.3 -0.6 -0.3 -0.7 -0.5 -0.1 -0.2 -0.3 -0.3 -0.1 -0.4 -0.6 -0.6 -0.5\n",
      "H -0.3 -0.8 -0.1 -0.2 -0.2 -0.3  1.3 -0.5  0.0 -0.4 -0.2  0.2 -0.4  0.1  0.0 -0.3 -0.3 -0.4 -0.4 -0.0\n",
      "I -0.2 -0.8 -0.5 -0.5  0.1 -0.6 -0.5  0.8 -0.5  0.3  0.3 -0.5 -0.6 -0.7 -0.5 -0.5 -0.3  0.4 -0.3 -0.2\n",
      "K -0.1 -0.9 -0.1  0.1 -0.6 -0.3  0.0 -0.5  0.8 -0.3 -0.2  0.0 -0.1  0.1  0.3 -0.1 -0.0 -0.4 -0.5 -0.4\n",
      "L -0.3 -0.9 -0.8 -0.6  0.2 -0.7 -0.4  0.3 -0.3  0.7  0.4 -0.5 -0.3 -0.4 -0.4 -0.5 -0.5  0.2 -0.1 -0.2\n",
      "M -0.1 -0.4 -0.6 -0.3 -0.1 -0.5 -0.2  0.3 -0.2  0.4  1.1 -0.4 -1.0 -0.1 -0.4 -0.5 -0.3  0.1 -0.1 -0.1\n",
      "N -0.1 -0.8  0.3 -0.1 -0.4 -0.1  0.2 -0.5  0.0 -0.5 -0.4  0.8 -0.2 -0.1 -0.1  0.1  0.0 -0.6 -0.6 -0.1\n",
      "P -0.1 -0.9 -0.1 -0.1 -0.5 -0.2 -0.4 -0.6 -0.1 -0.3 -1.0 -0.2  1.0 -0.4 -0.4 -0.1 -0.2 -0.5 -0.7 -0.7\n",
      "Q -0.1 -0.7 -0.1  0.2 -0.6 -0.3  0.1 -0.7  0.1 -0.4 -0.1 -0.1 -0.4  0.9  0.2 -0.1 -0.0 -0.4 -0.8 -0.5\n",
      "R -0.2 -0.6 -0.3 -0.0 -0.6 -0.3  0.0 -0.5  0.3 -0.4 -0.4 -0.1 -0.4  0.2  1.0 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "S  0.0 -0.8 -0.0 -0.2 -0.5 -0.1 -0.3 -0.5 -0.1 -0.5 -0.5  0.1 -0.1 -0.1 -0.1  0.6  0.2 -0.4 -0.6 -0.3\n",
      "T -0.1 -0.6 -0.2 -0.1 -0.5 -0.4 -0.3 -0.3 -0.0 -0.5 -0.3  0.0 -0.2 -0.0 -0.1  0.2  0.7 -0.2 -0.9 -0.3\n",
      "V -0.1 -0.5 -0.5 -0.4 -0.1 -0.6 -0.4  0.4 -0.4  0.2  0.1 -0.6 -0.5 -0.4 -0.5 -0.4 -0.2  0.7 -0.5 -0.2\n",
      "W -0.6 -0.9 -0.6 -0.8  0.3 -0.6 -0.4 -0.3 -0.5 -0.1 -0.1 -0.6 -0.7 -0.8 -0.4 -0.6 -0.9 -0.5  1.5  0.2\n",
      "Y -0.4 -0.8 -0.4 -0.4  0.3 -0.5 -0.0 -0.2 -0.4 -0.2 -0.1 -0.1 -0.7 -0.5 -0.2 -0.3 -0.3 -0.2  0.2  1.1\n",
      "\n",
      "JONES #  David T. Jones, William R. Taylor, Janet M. Thornton:\n",
      "#  \"The rapid generation of mutation data matrices from protein sequences.\"\n",
      "#  Computer Applications in the Biosciences: CABIOS 8(3): 275-282 (1992).\n",
      "#  Table I, lower triangle.\n",
      "#  PMID 1633570\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V\n",
      "A  0.2 -0.1  0.0  0.0 -0.1 -0.1 -0.1  0.1 -0.2  0.0 -0.1 -0.1 -0.1 -0.3  0.1  0.1  0.2 -0.4 -0.3  0.1\n",
      "R -0.1  0.5  0.0 -0.1 -0.1  0.2  0.0  0.0  0.2 -0.3 -0.3  0.4 -0.2 -0.4 -0.1 -0.1 -0.1  0.0 -0.2 -0.3\n",
      "N  0.0  0.0  0.3  0.2 -0.1  0.0  0.1  0.0  0.1 -0.2 -0.3  0.1 -0.2 -0.3 -0.1  0.1  0.1 -0.5 -0.1 -0.2\n",
      "D  0.0 -0.1  0.2  0.5 -0.3  0.1  0.4  0.1  0.0 -0.3 -0.4  0.0 -0.3 -0.5 -0.2  0.0 -0.1 -0.5 -0.2 -0.2\n",
      "C -0.1 -0.1 -0.1 -0.3  1.1 -0.3 -0.4 -0.1  0.0 -0.2 -0.3 -0.3 -0.2  0.0 -0.2  0.1 -0.1  0.1  0.2 -0.2\n",
      "Q -0.1  0.2  0.0  0.1 -0.3  0.5  0.2 -0.1  0.2 -0.3 -0.2  0.2 -0.2 -0.4  0.0 -0.1 -0.1 -0.3 -0.2 -0.3\n",
      "E -0.1  0.0  0.1  0.4 -0.4  0.2  0.5  0.0  0.0 -0.3 -0.4  0.1 -0.3 -0.5 -0.2 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "G  0.1  0.0  0.0  0.1 -0.1 -0.1  0.0  0.5 -0.2 -0.3 -0.4 -0.1 -0.3 -0.5 -0.1  0.1 -0.1 -0.2 -0.4 -0.2\n",
      "H -0.2  0.2  0.1  0.0  0.0  0.2  0.0 -0.2  0.6 -0.3 -0.2  0.1 -0.2  0.0  0.0 -0.1 -0.1 -0.3  0.4 -0.3\n",
      "I  0.0 -0.3 -0.2 -0.3 -0.2 -0.3 -0.3 -0.3 -0.3  0.4  0.2 -0.3  0.3  0.0 -0.2 -0.1  0.1 -0.4 -0.2  0.4\n",
      "L -0.1 -0.3 -0.3 -0.4 -0.3 -0.2 -0.4 -0.4 -0.2  0.2  0.5 -0.3  0.3  0.2  0.0 -0.2 -0.1 -0.2 -0.1  0.2\n",
      "K -0.1  0.4  0.1  0.0 -0.3  0.2  0.1 -0.1  0.1 -0.3 -0.3  0.5 -0.2 -0.5 -0.2 -0.1 -0.1 -0.3 -0.3 -0.3\n",
      "M -0.1 -0.2 -0.2 -0.3 -0.2 -0.2 -0.3 -0.3 -0.2  0.3  0.3 -0.2  0.6  0.0 -0.2 -0.1  0.0 -0.3 -0.2  0.2\n",
      "F -0.3 -0.4 -0.3 -0.5  0.0 -0.4 -0.5 -0.5  0.0  0.0  0.2 -0.5  0.0  0.8 -0.3 -0.2 -0.2 -0.1  0.5  0.0\n",
      "P  0.1 -0.1 -0.1 -0.2 -0.2  0.0 -0.2 -0.1  0.0 -0.2  0.0 -0.2 -0.2 -0.3  0.6  0.1  0.1 -0.4 -0.3 -0.1\n",
      "S  0.1 -0.1  0.1  0.0  0.1 -0.1 -0.1  0.1 -0.1 -0.1 -0.2 -0.1 -0.1 -0.2  0.1  0.2  0.1 -0.3 -0.1 -0.1\n",
      "T  0.2 -0.1  0.1 -0.1 -0.1 -0.1 -0.1 -0.1 -0.1  0.1 -0.1 -0.1  0.0 -0.2  0.1  0.1  0.2 -0.4 -0.3  0.0\n",
      "W -0.4  0.0 -0.5 -0.5  0.1 -0.3 -0.5 -0.2 -0.3 -0.4 -0.2 -0.3 -0.3 -0.1 -0.4 -0.3 -0.4  1.5  0.0 -0.3\n",
      "Y -0.3 -0.2 -0.1 -0.2  0.2 -0.2 -0.4 -0.4  0.4 -0.2 -0.1 -0.3 -0.2  0.5 -0.3 -0.1 -0.3  0.0  0.9 -0.3\n",
      "V  0.1 -0.3 -0.2 -0.2 -0.2 -0.3 -0.2 -0.2 -0.3  0.4  0.2 -0.3  0.2  0.0 -0.1 -0.1  0.0 -0.3 -0.3  0.4\n",
      "\n",
      "LEVIN #  Jonathan M. Levin, Barry Robson, Jean Garnier:\n",
      "#  \"An algorithm for secondary structure determination in proteins based on\n",
      "#  sequence similarity.\"\n",
      "#  FEBS Letters 205(2): 303-308 (1986).\n",
      "#  Figure 1.\n",
      "#  PMID 3743779\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.0  0.0  0.0  1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0  0.0  0.0  1.0  0.0  0.0 -1.0 -1.0\n",
      "C  0.0  2.0  0.0  0.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0\n",
      "D  0.0  0.0  2.0  1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  1.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "E  1.0  0.0  1.0  2.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  1.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "F -1.0 -1.0 -1.0 -1.0  2.0 -1.0 -1.0  1.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  1.0\n",
      "G  0.0  0.0  0.0  0.0 -1.0  2.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "H  0.0  0.0  0.0  0.0 -1.0  0.0  2.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0  0.0\n",
      "I  0.0  0.0 -1.0 -1.0  1.0 -1.0 -1.0  2.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  1.0  0.0  0.0\n",
      "K  0.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  2.0 -1.0 -1.0  1.0  0.0  0.0  1.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "L  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  2.0  2.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  1.0  0.0  0.0\n",
      "M  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  2.0  2.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  0.0  0.0\n",
      "N  0.0  0.0  1.0  0.0 -1.0  0.0  0.0 -1.0  1.0 -1.0 -1.0  3.0  0.0  1.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "P -1.0  0.0  0.0 -1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  3.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "Q  0.0  0.0  0.0  1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  1.0  0.0  2.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "R  0.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  1.0 -1.0 -1.0  0.0  0.0  0.0  2.0  0.0  0.0 -1.0  0.0 -1.0\n",
      "S  1.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  2.0  0.0 -1.0 -1.0 -1.0\n",
      "T  0.0  0.0  0.0  0.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  2.0  0.0 -1.0 -1.0\n",
      "V  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  1.0 -1.0  1.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  2.0  0.0  0.0\n",
      "W -1.0 -1.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0  2.0  0.0\n",
      "Y -1.0 -1.0 -1.0 -1.0  1.0 -1.0  0.0  0.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  2.0\n",
      "\n",
      "MCLACHLAN #  A.D. McLachlan:\n",
      "#  \"Tests for comparing related amino-acid sequences. Cytochrome c and\n",
      "#  cytochrome c 551.\"\n",
      "#  Journal of Molecular Biology 61(2): 409-424 (1971).\n",
      "#  Figure 1.\n",
      "#  PMID 5167087\n",
      "    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y\n",
      "A 8.0 1.0 3.0 4.0 1.0 3.0 3.0 2.0 3.0 2.0 3.0 3.0 4.0 3.0 2.0 4.0 3.0 3.0 1.0 1.0\n",
      "C 1.0 9.0 1.0 0.0 0.0 1.0 3.0 1.0 0.0 0.0 3.0 1.0 0.0 0.0 1.0 2.0 2.0 1.0 2.0 1.0\n",
      "D 3.0 1.0 8.0 5.0 1.0 3.0 4.0 1.0 3.0 1.0 2.0 5.0 3.0 4.0 1.0 3.0 3.0 1.0 0.0 1.0\n",
      "E 4.0 0.0 5.0 8.0 0.0 3.0 2.0 1.0 4.0 1.0 1.0 4.0 4.0 5.0 3.0 4.0 4.0 2.0 1.0 2.0\n",
      "F 1.0 0.0 1.0 0.0 9.0 0.0 4.0 3.0 0.0 5.0 5.0 0.0 1.0 0.0 1.0 2.0 1.0 3.0 6.0 6.0\n",
      "G 3.0 1.0 3.0 3.0 0.0 8.0 2.0 1.0 3.0 1.0 1.0 3.0 3.0 2.0 3.0 3.0 2.0 2.0 1.0 0.0\n",
      "H 3.0 3.0 4.0 2.0 4.0 2.0 8.0 2.0 4.0 2.0 3.0 4.0 3.0 4.0 5.0 3.0 4.0 2.0 3.0 4.0\n",
      "I 2.0 1.0 1.0 1.0 3.0 1.0 2.0 8.0 1.0 5.0 5.0 1.0 1.0 0.0 1.0 2.0 3.0 5.0 3.0 3.0\n",
      "K 3.0 0.0 3.0 4.0 0.0 3.0 4.0 1.0 8.0 2.0 1.0 4.0 3.0 4.0 5.0 3.0 3.0 2.0 1.0 1.0\n",
      "L 2.0 0.0 1.0 1.0 5.0 1.0 2.0 5.0 2.0 8.0 6.0 1.0 1.0 3.0 2.0 2.0 3.0 5.0 3.0 3.0\n",
      "M 3.0 3.0 2.0 1.0 5.0 1.0 3.0 5.0 1.0 6.0 8.0 2.0 1.0 3.0 1.0 2.0 3.0 4.0 1.0 2.0\n",
      "N 3.0 1.0 5.0 4.0 0.0 3.0 4.0 1.0 4.0 1.0 2.0 8.0 1.0 4.0 3.0 5.0 3.0 1.0 0.0 2.0\n",
      "P 4.0 0.0 3.0 4.0 1.0 3.0 3.0 1.0 3.0 1.0 1.0 1.0 8.0 3.0 3.0 3.0 3.0 2.0 0.0 0.0\n",
      "Q 3.0 0.0 4.0 5.0 0.0 2.0 4.0 0.0 4.0 3.0 3.0 4.0 3.0 8.0 5.0 4.0 3.0 2.0 2.0 1.0\n",
      "R 2.0 1.0 1.0 3.0 1.0 3.0 5.0 1.0 5.0 2.0 1.0 3.0 3.0 5.0 8.0 4.0 3.0 2.0 3.0 2.0\n",
      "S 4.0 2.0 3.0 4.0 2.0 3.0 3.0 2.0 3.0 2.0 2.0 5.0 3.0 4.0 4.0 8.0 5.0 2.0 3.0 3.0\n",
      "T 3.0 2.0 3.0 4.0 1.0 2.0 4.0 3.0 3.0 3.0 3.0 3.0 3.0 3.0 3.0 5.0 8.0 3.0 2.0 1.0\n",
      "V 3.0 1.0 1.0 2.0 3.0 2.0 2.0 5.0 2.0 5.0 4.0 1.0 2.0 2.0 2.0 2.0 3.0 8.0 2.0 3.0\n",
      "W 1.0 2.0 0.0 1.0 6.0 1.0 3.0 3.0 1.0 3.0 1.0 0.0 0.0 2.0 3.0 3.0 2.0 2.0 9.0 6.0\n",
      "Y 1.0 1.0 1.0 2.0 6.0 0.0 4.0 3.0 1.0 3.0 2.0 2.0 0.0 1.0 2.0 3.0 1.0 3.0 6.0 9.0\n",
      "\n",
      "MDM78 #  R.M. Schwartz and M.O. Dayhoff:\n",
      "#  \"Matrices for Detecting Distant Relationships.\"\n",
      "#  Margaret O. Dayhoff: Atlas of Protein Sequence and Structure,\n",
      "#  Volume 5, Supplement 3, 1978, pages 353-358.\n",
      "#  The National Biomedical Research Foundation, 1979.\n",
      "#  Figure 85, page 354.\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V\n",
      "A  0.2 -0.1  0.0  0.0 -0.2 -0.0  0.0  0.1 -0.1 -0.1 -0.2 -0.1 -0.1 -0.3  0.1  0.1  0.1 -0.6 -0.3  0.0\n",
      "R -0.1  0.6  0.0 -0.1 -0.4  0.1 -0.1 -0.3  0.2 -0.2 -0.3  0.3 -0.0 -0.5 -0.0 -0.0 -0.1  0.2 -0.4 -0.2\n",
      "N  0.0  0.0  0.2  0.2 -0.4  0.1  0.1  0.0  0.2 -0.2 -0.3  0.1 -0.2 -0.3 -0.1  0.1  0.0 -0.4 -0.2 -0.2\n",
      "D  0.0 -0.1  0.2  0.4 -0.5  0.2  0.3  0.1  0.1 -0.2 -0.4  0.0 -0.3 -0.6 -0.1  0.0 -0.0 -0.7 -0.4 -0.2\n",
      "C -0.2 -0.4 -0.4 -0.5  1.2 -0.5 -0.5 -0.3 -0.3 -0.2 -0.6 -0.5 -0.5 -0.4 -0.3  0.0 -0.2 -0.8  0.0 -0.2\n",
      "Q -0.0  0.1  0.1  0.2 -0.5  0.4  0.2 -0.1  0.3 -0.2 -0.2  0.1 -0.1 -0.5  0.0 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "E  0.0 -0.1  0.1  0.3 -0.5  0.2  0.4  0.0  0.1 -0.2 -0.3 -0.0 -0.2 -0.5 -0.1  0.0 -0.0 -0.7 -0.4 -0.2\n",
      "G  0.1 -0.3  0.0  0.1 -0.3 -0.1  0.0  0.5 -0.2 -0.3 -0.4 -0.2 -0.3 -0.5 -0.1  0.1  0.0 -0.7 -0.5 -0.1\n",
      "H -0.1  0.2  0.2  0.1 -0.3  0.3  0.1 -0.2  0.7 -0.2 -0.2  0.0 -0.2 -0.2 -0.0 -0.1 -0.1 -0.3 -0.0 -0.2\n",
      "I -0.1 -0.2 -0.2 -0.2 -0.2 -0.2 -0.2 -0.3 -0.2  0.5  0.2 -0.2  0.2  0.1 -0.2 -0.1  0.0 -0.5 -0.1  0.4\n",
      "L -0.2 -0.3 -0.3 -0.4 -0.6 -0.2 -0.3 -0.4 -0.2  0.2  0.6 -0.3  0.4  0.2 -0.2 -0.3 -0.2 -0.2 -0.1  0.2\n",
      "K -0.1  0.3  0.1  0.0 -0.5  0.1 -0.0 -0.2  0.0 -0.2 -0.3  0.5  0.0 -0.5 -0.1 -0.0  0.0 -0.3 -0.4 -0.2\n",
      "M -0.1 -0.0 -0.2 -0.3 -0.5 -0.1 -0.2 -0.3 -0.2  0.2  0.4  0.0  0.6  0.0 -0.2 -0.2 -0.1 -0.4 -0.2  0.2\n",
      "F -0.3 -0.5 -0.3 -0.6 -0.4 -0.5 -0.5 -0.5 -0.2  0.1  0.2 -0.5  0.0  0.9 -0.5 -0.3 -0.3  0.0  0.7 -0.1\n",
      "P  0.1 -0.0 -0.1 -0.1 -0.3  0.0 -0.1 -0.1 -0.0 -0.2 -0.2 -0.1 -0.2 -0.5  0.6  0.1  0.0 -0.6 -0.5 -0.1\n",
      "S  0.1 -0.0  0.1  0.0  0.0 -0.1  0.0  0.1 -0.1 -0.1 -0.3 -0.0 -0.2 -0.3  0.1  0.2  0.1 -0.2 -0.3 -0.1\n",
      "T  0.1 -0.1  0.0 -0.0 -0.2 -0.1 -0.0  0.0 -0.1  0.0 -0.2  0.0 -0.1 -0.3  0.0  0.1  0.3 -0.5 -0.3  0.0\n",
      "W -0.6  0.2 -0.4 -0.7 -0.8 -0.5 -0.7 -0.7 -0.3 -0.5 -0.2 -0.3 -0.4  0.0 -0.6 -0.2 -0.5  1.7 -0.0 -0.6\n",
      "Y -0.3 -0.4 -0.2 -0.4  0.0 -0.4 -0.4 -0.5 -0.0 -0.1 -0.1 -0.4 -0.2  0.7 -0.5 -0.3 -0.3 -0.0  1.0 -0.2\n",
      "V  0.0 -0.2 -0.2 -0.2 -0.2 -0.2 -0.2 -0.1 -0.2  0.4  0.2 -0.2  0.2 -0.1 -0.1 -0.1  0.0 -0.6 -0.2  0.4\n",
      "\n",
      "NUC.4.4 #  \n",
      "#  This matrix was created by Todd Lowe   12/10/92\n",
      "#  \n",
      "#  Uses ambiguous nucleotide codes, probabilities rounded to\n",
      "#  nearest integer\n",
      "#  \n",
      "#  Lowest score = -4, Highest score = 5\n",
      "#  \n",
      "     A    T    G    C    S    W    R    Y    K    M    B    V    H    D    N\n",
      "A  5.0 -4.0 -4.0 -4.0 -4.0  1.0  1.0 -4.0 -4.0  1.0 -4.0 -1.0 -1.0 -1.0 -2.0\n",
      "T -4.0  5.0 -4.0 -4.0 -4.0  1.0 -4.0  1.0  1.0 -4.0 -1.0 -4.0 -1.0 -1.0 -2.0\n",
      "G -4.0 -4.0  5.0 -4.0  1.0 -4.0  1.0 -4.0  1.0 -4.0 -1.0 -1.0 -4.0 -1.0 -2.0\n",
      "C -4.0 -4.0 -4.0  5.0  1.0 -4.0 -4.0  1.0 -4.0  1.0 -1.0 -1.0 -1.0 -4.0 -2.0\n",
      "S -4.0 -4.0  1.0  1.0 -1.0 -4.0 -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -3.0 -3.0 -1.0\n",
      "W  1.0  1.0 -4.0 -4.0 -4.0 -1.0 -2.0 -2.0 -2.0 -2.0 -3.0 -3.0 -1.0 -1.0 -1.0\n",
      "R  1.0 -4.0  1.0 -4.0 -2.0 -2.0 -1.0 -4.0 -2.0 -2.0 -3.0 -1.0 -3.0 -1.0 -1.0\n",
      "Y -4.0  1.0 -4.0  1.0 -2.0 -2.0 -4.0 -1.0 -2.0 -2.0 -1.0 -3.0 -1.0 -3.0 -1.0\n",
      "K -4.0  1.0  1.0 -4.0 -2.0 -2.0 -2.0 -2.0 -1.0 -4.0 -1.0 -3.0 -3.0 -1.0 -1.0\n",
      "M  1.0 -4.0 -4.0  1.0 -2.0 -2.0 -2.0 -2.0 -4.0 -1.0 -3.0 -1.0 -1.0 -3.0 -1.0\n",
      "B -4.0 -1.0 -1.0 -1.0 -1.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -2.0 -2.0 -2.0 -1.0\n",
      "V -1.0 -4.0 -1.0 -1.0 -1.0 -3.0 -1.0 -3.0 -3.0 -1.0 -2.0 -1.0 -2.0 -2.0 -1.0\n",
      "H -1.0 -1.0 -4.0 -1.0 -3.0 -1.0 -3.0 -1.0 -3.0 -1.0 -2.0 -2.0 -1.0 -2.0 -1.0\n",
      "D -1.0 -1.0 -1.0 -4.0 -3.0 -1.0 -1.0 -3.0 -1.0 -3.0 -2.0 -2.0 -2.0 -1.0 -1.0\n",
      "N -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0\n",
      "\n",
      "PAM250 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 250 substitution matrix, scale = ln(2)/3 = 0.231049\n",
      "#  \n",
      "#  Expected score = -0.844, Entropy = 0.354 bits\n",
      "#  \n",
      "#  Lowest score = -8, Highest score = 17\n",
      "#  \n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  2.0 -2.0  0.0  0.0 -2.0  0.0  0.0  1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -3.0  1.0  1.0  1.0 -6.0 -3.0  0.0  0.0  0.0  0.0 -8.0\n",
      "R -2.0  6.0  0.0 -1.0 -4.0  1.0 -1.0 -3.0  2.0 -2.0 -3.0  3.0  0.0 -4.0  0.0  0.0 -1.0  2.0 -4.0 -2.0 -1.0  0.0 -1.0 -8.0\n",
      "N  0.0  0.0  2.0  2.0 -4.0  1.0  1.0  0.0  2.0 -2.0 -3.0  1.0 -2.0 -3.0  0.0  1.0  0.0 -4.0 -2.0 -2.0  2.0  1.0  0.0 -8.0\n",
      "D  0.0 -1.0  2.0  4.0 -5.0  2.0  3.0  1.0  1.0 -2.0 -4.0  0.0 -3.0 -6.0 -1.0  0.0  0.0 -7.0 -4.0 -2.0  3.0  3.0 -1.0 -8.0\n",
      "C -2.0 -4.0 -4.0 -5.0 12.0 -5.0 -5.0 -3.0 -3.0 -2.0 -6.0 -5.0 -5.0 -4.0 -3.0  0.0 -2.0 -8.0  0.0 -2.0 -4.0 -5.0 -3.0 -8.0\n",
      "Q  0.0  1.0  1.0  2.0 -5.0  4.0  2.0 -1.0  3.0 -2.0 -2.0  1.0 -1.0 -5.0  0.0 -1.0 -1.0 -5.0 -4.0 -2.0  1.0  3.0 -1.0 -8.0\n",
      "E  0.0 -1.0  1.0  3.0 -5.0  2.0  4.0  0.0  1.0 -2.0 -3.0  0.0 -2.0 -5.0 -1.0  0.0  0.0 -7.0 -4.0 -2.0  3.0  3.0 -1.0 -8.0\n",
      "G  1.0 -3.0  0.0  1.0 -3.0 -1.0  0.0  5.0 -2.0 -3.0 -4.0 -2.0 -3.0 -5.0  0.0  1.0  0.0 -7.0 -5.0 -1.0  0.0  0.0 -1.0 -8.0\n",
      "H -1.0  2.0  2.0  1.0 -3.0  3.0  1.0 -2.0  6.0 -2.0 -2.0  0.0 -2.0 -2.0  0.0 -1.0 -1.0 -3.0  0.0 -2.0  1.0  2.0 -1.0 -8.0\n",
      "I -1.0 -2.0 -2.0 -2.0 -2.0 -2.0 -2.0 -3.0 -2.0  5.0  2.0 -2.0  2.0  1.0 -2.0 -1.0  0.0 -5.0 -1.0  4.0 -2.0 -2.0 -1.0 -8.0\n",
      "L -2.0 -3.0 -3.0 -4.0 -6.0 -2.0 -3.0 -4.0 -2.0  2.0  6.0 -3.0  4.0  2.0 -3.0 -3.0 -2.0 -2.0 -1.0  2.0 -3.0 -3.0 -1.0 -8.0\n",
      "K -1.0  3.0  1.0  0.0 -5.0  1.0  0.0 -2.0  0.0 -2.0 -3.0  5.0  0.0 -5.0 -1.0  0.0  0.0 -3.0 -4.0 -2.0  1.0  0.0 -1.0 -8.0\n",
      "M -1.0  0.0 -2.0 -3.0 -5.0 -1.0 -2.0 -3.0 -2.0  2.0  4.0  0.0  6.0  0.0 -2.0 -2.0 -1.0 -4.0 -2.0  2.0 -2.0 -2.0 -1.0 -8.0\n",
      "F -3.0 -4.0 -3.0 -6.0 -4.0 -5.0 -5.0 -5.0 -2.0  1.0  2.0 -5.0  0.0  9.0 -5.0 -3.0 -3.0  0.0  7.0 -1.0 -4.0 -5.0 -2.0 -8.0\n",
      "P  1.0  0.0  0.0 -1.0 -3.0  0.0 -1.0  0.0  0.0 -2.0 -3.0 -1.0 -2.0 -5.0  6.0  1.0  0.0 -6.0 -5.0 -1.0 -1.0  0.0 -1.0 -8.0\n",
      "S  1.0  0.0  1.0  0.0  0.0 -1.0  0.0  1.0 -1.0 -1.0 -3.0  0.0 -2.0 -3.0  1.0  2.0  1.0 -2.0 -3.0 -1.0  0.0  0.0  0.0 -8.0\n",
      "T  1.0 -1.0  0.0  0.0 -2.0 -1.0  0.0  0.0 -1.0  0.0 -2.0  0.0 -1.0 -3.0  0.0  1.0  3.0 -5.0 -3.0  0.0  0.0 -1.0  0.0 -8.0\n",
      "W -6.0  2.0 -4.0 -7.0 -8.0 -5.0 -7.0 -7.0 -3.0 -5.0 -2.0 -3.0 -4.0  0.0 -6.0 -2.0 -5.0 17.0  0.0 -6.0 -5.0 -6.0 -4.0 -8.0\n",
      "Y -3.0 -4.0 -2.0 -4.0  0.0 -4.0 -4.0 -5.0  0.0 -1.0 -1.0 -4.0 -2.0  7.0 -5.0 -3.0 -3.0  0.0 10.0 -2.0 -3.0 -4.0 -2.0 -8.0\n",
      "V  0.0 -2.0 -2.0 -2.0 -2.0 -2.0 -2.0 -1.0 -2.0  4.0  2.0 -2.0  2.0 -1.0 -1.0 -1.0  0.0 -6.0 -2.0  4.0 -2.0 -2.0 -1.0 -8.0\n",
      "B  0.0 -1.0  2.0  3.0 -4.0  1.0  3.0  0.0  1.0 -2.0 -3.0  1.0 -2.0 -4.0 -1.0  0.0  0.0 -5.0 -3.0 -2.0  3.0  2.0 -1.0 -8.0\n",
      "Z  0.0  0.0  1.0  3.0 -5.0  3.0  3.0  0.0  2.0 -2.0 -3.0  0.0 -2.0 -5.0  0.0  0.0 -1.0 -6.0 -4.0 -2.0  2.0  3.0 -1.0 -8.0\n",
      "X  0.0 -1.0  0.0 -1.0 -3.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  0.0  0.0 -4.0 -2.0 -1.0 -1.0 -1.0 -1.0 -8.0\n",
      "* -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0  1.0\n",
      "\n",
      "PAM30 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 30 substitution matrix, scale = ln(2)/2 = 0.346574\n",
      "#  \n",
      "#  Expected score = -5.06, Entropy = 2.57 bits\n",
      "#  \n",
      "#  Lowest score = -17, Highest score = 13\n",
      "#  \n",
      "      A     R     N     D     C     Q     E     G     H     I     L     K     M     F     P     S     T     W     Y     V     B     Z     X     *\n",
      "A   6.0  -7.0  -4.0  -3.0  -6.0  -4.0  -2.0  -2.0  -7.0  -5.0  -6.0  -7.0  -5.0  -8.0  -2.0   0.0  -1.0 -13.0  -8.0  -2.0  -3.0  -3.0  -3.0 -17.0\n",
      "R  -7.0   8.0  -6.0 -10.0  -8.0  -2.0  -9.0  -9.0  -2.0  -5.0  -8.0   0.0  -4.0  -9.0  -4.0  -3.0  -6.0  -2.0 -10.0  -8.0  -7.0  -4.0  -6.0 -17.0\n",
      "N  -4.0  -6.0   8.0   2.0 -11.0  -3.0  -2.0  -3.0   0.0  -5.0  -7.0  -1.0  -9.0  -9.0  -6.0   0.0  -2.0  -8.0  -4.0  -8.0   6.0  -3.0  -3.0 -17.0\n",
      "D  -3.0 -10.0   2.0   8.0 -14.0  -2.0   2.0  -3.0  -4.0  -7.0 -12.0  -4.0 -11.0 -15.0  -8.0  -4.0  -5.0 -15.0 -11.0  -8.0   6.0   1.0  -5.0 -17.0\n",
      "C  -6.0  -8.0 -11.0 -14.0  10.0 -14.0 -14.0  -9.0  -7.0  -6.0 -15.0 -14.0 -13.0 -13.0  -8.0  -3.0  -8.0 -15.0  -4.0  -6.0 -12.0 -14.0  -9.0 -17.0\n",
      "Q  -4.0  -2.0  -3.0  -2.0 -14.0   8.0   1.0  -7.0   1.0  -8.0  -5.0  -3.0  -4.0 -13.0  -3.0  -5.0  -5.0 -13.0 -12.0  -7.0  -3.0   6.0  -5.0 -17.0\n",
      "E  -2.0  -9.0  -2.0   2.0 -14.0   1.0   8.0  -4.0  -5.0  -5.0  -9.0  -4.0  -7.0 -14.0  -5.0  -4.0  -6.0 -17.0  -8.0  -6.0   1.0   6.0  -5.0 -17.0\n",
      "G  -2.0  -9.0  -3.0  -3.0  -9.0  -7.0  -4.0   6.0  -9.0 -11.0 -10.0  -7.0  -8.0  -9.0  -6.0  -2.0  -6.0 -15.0 -14.0  -5.0  -3.0  -5.0  -5.0 -17.0\n",
      "H  -7.0  -2.0   0.0  -4.0  -7.0   1.0  -5.0  -9.0   9.0  -9.0  -6.0  -6.0 -10.0  -6.0  -4.0  -6.0  -7.0  -7.0  -3.0  -6.0  -1.0  -1.0  -5.0 -17.0\n",
      "I  -5.0  -5.0  -5.0  -7.0  -6.0  -8.0  -5.0 -11.0  -9.0   8.0  -1.0  -6.0  -1.0  -2.0  -8.0  -7.0  -2.0 -14.0  -6.0   2.0  -6.0  -6.0  -5.0 -17.0\n",
      "L  -6.0  -8.0  -7.0 -12.0 -15.0  -5.0  -9.0 -10.0  -6.0  -1.0   7.0  -8.0   1.0  -3.0  -7.0  -8.0  -7.0  -6.0  -7.0  -2.0  -9.0  -7.0  -6.0 -17.0\n",
      "K  -7.0   0.0  -1.0  -4.0 -14.0  -3.0  -4.0  -7.0  -6.0  -6.0  -8.0   7.0  -2.0 -14.0  -6.0  -4.0  -3.0 -12.0  -9.0  -9.0  -2.0  -4.0  -5.0 -17.0\n",
      "M  -5.0  -4.0  -9.0 -11.0 -13.0  -4.0  -7.0  -8.0 -10.0  -1.0   1.0  -2.0  11.0  -4.0  -8.0  -5.0  -4.0 -13.0 -11.0  -1.0 -10.0  -5.0  -5.0 -17.0\n",
      "F  -8.0  -9.0  -9.0 -15.0 -13.0 -13.0 -14.0  -9.0  -6.0  -2.0  -3.0 -14.0  -4.0   9.0 -10.0  -6.0  -9.0  -4.0   2.0  -8.0 -10.0 -13.0  -8.0 -17.0\n",
      "P  -2.0  -4.0  -6.0  -8.0  -8.0  -3.0  -5.0  -6.0  -4.0  -8.0  -7.0  -6.0  -8.0 -10.0   8.0  -2.0  -4.0 -14.0 -13.0  -6.0  -7.0  -4.0  -5.0 -17.0\n",
      "S   0.0  -3.0   0.0  -4.0  -3.0  -5.0  -4.0  -2.0  -6.0  -7.0  -8.0  -4.0  -5.0  -6.0  -2.0   6.0   0.0  -5.0  -7.0  -6.0  -1.0  -5.0  -3.0 -17.0\n",
      "T  -1.0  -6.0  -2.0  -5.0  -8.0  -5.0  -6.0  -6.0  -7.0  -2.0  -7.0  -3.0  -4.0  -9.0  -4.0   0.0   7.0 -13.0  -6.0  -3.0  -3.0  -6.0  -4.0 -17.0\n",
      "W -13.0  -2.0  -8.0 -15.0 -15.0 -13.0 -17.0 -15.0  -7.0 -14.0  -6.0 -12.0 -13.0  -4.0 -14.0  -5.0 -13.0  13.0  -5.0 -15.0 -10.0 -14.0 -11.0 -17.0\n",
      "Y  -8.0 -10.0  -4.0 -11.0  -4.0 -12.0  -8.0 -14.0  -3.0  -6.0  -7.0  -9.0 -11.0   2.0 -13.0  -7.0  -6.0  -5.0  10.0  -7.0  -6.0  -9.0  -7.0 -17.0\n",
      "V  -2.0  -8.0  -8.0  -8.0  -6.0  -7.0  -6.0  -5.0  -6.0   2.0  -2.0  -9.0  -1.0  -8.0  -6.0  -6.0  -3.0 -15.0  -7.0   7.0  -8.0  -6.0  -5.0 -17.0\n",
      "B  -3.0  -7.0   6.0   6.0 -12.0  -3.0   1.0  -3.0  -1.0  -6.0  -9.0  -2.0 -10.0 -10.0  -7.0  -1.0  -3.0 -10.0  -6.0  -8.0   6.0   0.0  -5.0 -17.0\n",
      "Z  -3.0  -4.0  -3.0   1.0 -14.0   6.0   6.0  -5.0  -1.0  -6.0  -7.0  -4.0  -5.0 -13.0  -4.0  -5.0  -6.0 -14.0  -9.0  -6.0   0.0   6.0  -5.0 -17.0\n",
      "X  -3.0  -6.0  -3.0  -5.0  -9.0  -5.0  -5.0  -5.0  -5.0  -5.0  -6.0  -5.0  -5.0  -8.0  -5.0  -3.0  -4.0 -11.0  -7.0  -5.0  -5.0  -5.0  -5.0 -17.0\n",
      "* -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0   1.0\n",
      "\n",
      "PAM70 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 70 substitution matrix, scale = ln(2)/2 = 0.346574\n",
      "#  \n",
      "#  Expected score = -2.77, Entropy = 1.60 bits\n",
      "#  \n",
      "#  Lowest score = -11, Highest score = 13\n",
      "#  \n",
      "      A     R     N     D     C     Q     E     G     H     I     L     K     M     F     P     S     T     W     Y     V     B     Z     X     *\n",
      "A   5.0  -4.0  -2.0  -1.0  -4.0  -2.0  -1.0   0.0  -4.0  -2.0  -4.0  -4.0  -3.0  -6.0   0.0   1.0   1.0  -9.0  -5.0  -1.0  -1.0  -1.0  -2.0 -11.0\n",
      "R  -4.0   8.0  -3.0  -6.0  -5.0   0.0  -5.0  -6.0   0.0  -3.0  -6.0   2.0  -2.0  -7.0  -2.0  -1.0  -4.0   0.0  -7.0  -5.0  -4.0  -2.0  -3.0 -11.0\n",
      "N  -2.0  -3.0   6.0   3.0  -7.0  -1.0   0.0  -1.0   1.0  -3.0  -5.0   0.0  -5.0  -6.0  -3.0   1.0   0.0  -6.0  -3.0  -5.0   5.0  -1.0  -2.0 -11.0\n",
      "D  -1.0  -6.0   3.0   6.0  -9.0   0.0   3.0  -1.0  -1.0  -5.0  -8.0  -2.0  -7.0 -10.0  -4.0  -1.0  -2.0 -10.0  -7.0  -5.0   5.0   2.0  -3.0 -11.0\n",
      "C  -4.0  -5.0  -7.0  -9.0   9.0  -9.0  -9.0  -6.0  -5.0  -4.0 -10.0  -9.0  -9.0  -8.0  -5.0  -1.0  -5.0 -11.0  -2.0  -4.0  -8.0  -9.0  -6.0 -11.0\n",
      "Q  -2.0   0.0  -1.0   0.0  -9.0   7.0   2.0  -4.0   2.0  -5.0  -3.0  -1.0  -2.0  -9.0  -1.0  -3.0  -3.0  -8.0  -8.0  -4.0  -1.0   5.0  -2.0 -11.0\n",
      "E  -1.0  -5.0   0.0   3.0  -9.0   2.0   6.0  -2.0  -2.0  -4.0  -6.0  -2.0  -4.0  -9.0  -3.0  -2.0  -3.0 -11.0  -6.0  -4.0   2.0   5.0  -3.0 -11.0\n",
      "G   0.0  -6.0  -1.0  -1.0  -6.0  -4.0  -2.0   6.0  -6.0  -6.0  -7.0  -5.0  -6.0  -7.0  -3.0   0.0  -3.0 -10.0  -9.0  -3.0  -1.0  -3.0  -3.0 -11.0\n",
      "H  -4.0   0.0   1.0  -1.0  -5.0   2.0  -2.0  -6.0   8.0  -6.0  -4.0  -3.0  -6.0  -4.0  -2.0  -3.0  -4.0  -5.0  -1.0  -4.0   0.0   1.0  -3.0 -11.0\n",
      "I  -2.0  -3.0  -3.0  -5.0  -4.0  -5.0  -4.0  -6.0  -6.0   7.0   1.0  -4.0   1.0   0.0  -5.0  -4.0  -1.0  -9.0  -4.0   3.0  -4.0  -4.0  -3.0 -11.0\n",
      "L  -4.0  -6.0  -5.0  -8.0 -10.0  -3.0  -6.0  -7.0  -4.0   1.0   6.0  -5.0   2.0  -1.0  -5.0  -6.0  -4.0  -4.0  -4.0   0.0  -6.0  -4.0  -4.0 -11.0\n",
      "K  -4.0   2.0   0.0  -2.0  -9.0  -1.0  -2.0  -5.0  -3.0  -4.0  -5.0   6.0   0.0  -9.0  -4.0  -2.0  -1.0  -7.0  -7.0  -6.0  -1.0  -2.0  -3.0 -11.0\n",
      "M  -3.0  -2.0  -5.0  -7.0  -9.0  -2.0  -4.0  -6.0  -6.0   1.0   2.0   0.0  10.0  -2.0  -5.0  -3.0  -2.0  -8.0  -7.0   0.0  -6.0  -3.0  -3.0 -11.0\n",
      "F  -6.0  -7.0  -6.0 -10.0  -8.0  -9.0  -9.0  -7.0  -4.0   0.0  -1.0  -9.0  -2.0   8.0  -7.0  -4.0  -6.0  -2.0   4.0  -5.0  -7.0  -9.0  -5.0 -11.0\n",
      "P   0.0  -2.0  -3.0  -4.0  -5.0  -1.0  -3.0  -3.0  -2.0  -5.0  -5.0  -4.0  -5.0  -7.0   7.0   0.0  -2.0  -9.0  -9.0  -3.0  -4.0  -2.0  -3.0 -11.0\n",
      "S   1.0  -1.0   1.0  -1.0  -1.0  -3.0  -2.0   0.0  -3.0  -4.0  -6.0  -2.0  -3.0  -4.0   0.0   5.0   2.0  -3.0  -5.0  -3.0   0.0  -2.0  -1.0 -11.0\n",
      "T   1.0  -4.0   0.0  -2.0  -5.0  -3.0  -3.0  -3.0  -4.0  -1.0  -4.0  -1.0  -2.0  -6.0  -2.0   2.0   6.0  -8.0  -4.0  -1.0  -1.0  -3.0  -2.0 -11.0\n",
      "W  -9.0   0.0  -6.0 -10.0 -11.0  -8.0 -11.0 -10.0  -5.0  -9.0  -4.0  -7.0  -8.0  -2.0  -9.0  -3.0  -8.0  13.0  -3.0 -10.0  -7.0 -10.0  -7.0 -11.0\n",
      "Y  -5.0  -7.0  -3.0  -7.0  -2.0  -8.0  -6.0  -9.0  -1.0  -4.0  -4.0  -7.0  -7.0   4.0  -9.0  -5.0  -4.0  -3.0   9.0  -5.0  -4.0  -7.0  -5.0 -11.0\n",
      "V  -1.0  -5.0  -5.0  -5.0  -4.0  -4.0  -4.0  -3.0  -4.0   3.0   0.0  -6.0   0.0  -5.0  -3.0  -3.0  -1.0 -10.0  -5.0   6.0  -5.0  -4.0  -2.0 -11.0\n",
      "B  -1.0  -4.0   5.0   5.0  -8.0  -1.0   2.0  -1.0   0.0  -4.0  -6.0  -1.0  -6.0  -7.0  -4.0   0.0  -1.0  -7.0  -4.0  -5.0   5.0   1.0  -2.0 -11.0\n",
      "Z  -1.0  -2.0  -1.0   2.0  -9.0   5.0   5.0  -3.0   1.0  -4.0  -4.0  -2.0  -3.0  -9.0  -2.0  -2.0  -3.0 -10.0  -7.0  -4.0   1.0   5.0  -3.0 -11.0\n",
      "X  -2.0  -3.0  -2.0  -3.0  -6.0  -2.0  -3.0  -3.0  -3.0  -3.0  -4.0  -3.0  -3.0  -5.0  -3.0  -1.0  -2.0  -7.0  -5.0  -2.0  -2.0  -3.0  -3.0 -11.0\n",
      "* -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0   1.0\n",
      "\n",
      "RAO #  J.K. Mohana Rao:\n",
      "#  \"New scoring matrix for amino acid residue exchanges based on residue\n",
      "#  characteristic physical parameters.\"\n",
      "#  International Journal of Peptide and Protein Research: 29(2): 276-281 (1987).\n",
      "#  Figure 1, lower triangle.\n",
      "#  PMID 3570667\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A 16.0 11.0  9.0 10.0 10.0  8.0 11.0  9.0 10.0 11.0 11.0  9.0  6.0 11.0  8.0 10.0 10.0  9.0 11.0  9.0\n",
      "C 11.0 16.0  8.0  9.0 10.0  8.0 10.0  8.0  9.0 11.0 10.0  9.0  7.0 10.0  8.0 10.0 10.0  8.0 11.0 10.0\n",
      "D  9.0  8.0 16.0 11.0  4.0  9.0  9.0  3.0 11.0  6.0  5.0 11.0  8.0 11.0 10.0 10.0  9.0  3.0  6.0  7.0\n",
      "E 10.0  9.0 11.0 16.0  6.0  6.0 11.0  4.0 11.0  7.0  8.0 10.0  5.0 11.0  9.0  9.0  8.0  4.0  7.0  6.0\n",
      "F 10.0 10.0  4.0  6.0 16.0  7.0  9.0 12.0  6.0 11.0 10.0  6.0  4.0  7.0  5.0  8.0 10.0 11.0 11.0 10.0\n",
      "G  8.0  8.0  9.0  6.0  7.0 16.0  7.0  6.0  7.0  6.0  4.0 10.0 11.0  8.0  7.0 11.0 10.0  6.0  8.0 10.0\n",
      "H 11.0 10.0  9.0 11.0  9.0  7.0 16.0  8.0 11.0 10.0 10.0 10.0  5.0 11.0 10.0 10.0 10.0  9.0 10.0  9.0\n",
      "I  9.0  8.0  3.0  4.0 12.0  6.0  8.0 16.0  4.0 10.0  9.0  5.0  3.0  6.0  4.0  8.0 10.0 12.0 11.0 10.0\n",
      "K 10.0  9.0 11.0 11.0  6.0  7.0 11.0  4.0 16.0  7.0  8.0 11.0  6.0 12.0 11.0 10.0  9.0  5.0  7.0  7.0\n",
      "L 11.0 11.0  6.0  7.0 11.0  6.0 10.0 10.0  7.0 16.0 11.0  7.0  4.0  9.0  6.0  8.0  9.0 10.0 11.0  9.0\n",
      "M 11.0 10.0  5.0  8.0 10.0  4.0 10.0  9.0  8.0 11.0 16.0  6.0  2.0  9.0  6.0  7.0  8.0  9.0 10.0  8.0\n",
      "N  9.0  9.0 11.0 10.0  6.0 10.0 10.0  5.0 11.0  7.0  6.0 16.0  9.0 11.0 10.0 11.0 10.0  5.0  8.0  8.0\n",
      "P  6.0  7.0  8.0  5.0  4.0 11.0  5.0  3.0  6.0  4.0  2.0  9.0 16.0  7.0  6.0 10.0  8.0  3.0  6.0  8.0\n",
      "Q 11.0 10.0 11.0 11.0  7.0  8.0 11.0  6.0 12.0  9.0  9.0 11.0  7.0 16.0 10.0 10.0 10.0  6.0  9.0  8.0\n",
      "R  8.0  8.0 10.0  9.0  5.0  7.0 10.0  4.0 11.0  6.0  6.0 10.0  6.0 10.0 16.0  9.0  9.0  5.0  7.0  7.0\n",
      "S 10.0 10.0 10.0  9.0  8.0 11.0 10.0  8.0 10.0  8.0  7.0 11.0 10.0 10.0  9.0 16.0 11.0  8.0 10.0 11.0\n",
      "T 10.0 10.0  9.0  8.0 10.0 10.0 10.0 10.0  9.0  9.0  8.0 10.0  8.0 10.0  9.0 11.0 16.0 10.0 11.0 11.0\n",
      "V  9.0  8.0  3.0  4.0 11.0  6.0  9.0 12.0  5.0 10.0  9.0  5.0  3.0  6.0  5.0  8.0 10.0 16.0 11.0 10.0\n",
      "W 11.0 11.0  6.0  7.0 11.0  8.0 10.0 11.0  7.0 11.0 10.0  8.0  6.0  9.0  7.0 10.0 11.0 11.0 16.0 11.0\n",
      "Y  9.0 10.0  7.0  6.0 10.0 10.0  9.0 10.0  7.0  9.0  8.0  8.0  8.0  8.0  7.0 11.0 11.0 10.0 11.0 16.0\n",
      "\n",
      "RISLER #  J.L. Risler, M.O. Delorme, H. Delacroix, A. Henaut:\n",
      "#  \"Amino acid substitutions in structurally related proteins. A pattern\n",
      "#  recognition approach. Determination of a new and efficient scoring matrix.\"\n",
      "#  Journal of Molecular Biology 204(4): 1019-1029 (1988).\n",
      "#  Figure 5.\n",
      "#  PMID 3221397\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.2 -1.5  0.2  1.7  0.6  0.6 -0.6  1.7  1.4  1.3  1.0  1.3 -0.2  1.8  1.5  2.0  1.9  2.0 -0.9  0.2\n",
      "C -1.5  2.2 -1.7 -1.5 -1.6 -1.7 -1.8 -1.6 -1.6 -1.5 -1.6 -1.6 -1.8 -1.4 -1.5 -1.3 -1.4 -1.4 -1.8 -1.1\n",
      "D  0.2 -1.7  2.2  1.0 -0.3 -0.4 -1.3  0.0  0.1 -0.2 -0.5  0.8 -1.2  0.6 -0.1  0.7  0.0  0.0 -1.4 -0.4\n",
      "E  1.7 -1.5  1.0  2.2  0.6  0.3 -0.6  1.5  1.4  0.9  0.6  1.4 -0.1  2.1  1.9  1.8  1.6  1.6 -1.0  0.2\n",
      "F  0.6 -1.6 -0.3  0.6  2.2 -0.4 -1.1  1.0  0.1  1.0 -0.2  0.4 -1.1  0.7  0.4  0.5  0.3  0.8 -0.9  2.0\n",
      "G  0.6 -1.7 -0.4  0.3 -0.4  2.2 -1.2  0.0 -0.1 -0.2 -0.4  0.2 -1.2  0.2  0.1  0.7  0.2  0.1 -1.3 -0.2\n",
      "H -0.6 -1.8 -1.3 -0.6 -1.1 -1.2  2.2 -0.8 -1.0 -0.9 -1.2 -0.3 -1.6 -0.5 -0.4 -0.4 -0.9 -0.7 -1.7 -0.8\n",
      "I  1.7 -1.6  0.0  1.5  1.0  0.0 -0.8  2.2  1.0  2.1  0.9  0.9 -0.6  1.4  1.4  1.6  1.6  2.2 -0.7  0.4\n",
      "K  1.4 -1.6  0.1  1.4  0.1 -0.1 -1.0  1.0  2.2  0.7  0.4  1.0 -0.7  1.7  2.1  1.4  1.2  1.2 -1.1  0.5\n",
      "L  1.3 -1.5 -0.2  0.9  1.0 -0.2 -0.9  2.1  0.7  2.2  1.8  0.8 -0.8  1.1  1.2  1.3  1.2  2.0 -0.8  0.5\n",
      "M  1.0 -1.6 -0.5  0.6 -0.2 -0.4 -1.2  0.9  0.4  1.8  2.2  0.0 -1.2  1.2  1.1  0.6  0.8  0.8 -1.3 -0.2\n",
      "N  1.3 -1.6  0.8  1.4  0.4  0.2 -0.3  0.9  1.0  0.8  0.0  2.2 -1.0  1.6  1.2  1.9  1.1  1.1 -1.1 -0.1\n",
      "P -0.2 -1.8 -1.2 -0.1 -1.1 -1.2 -1.6 -0.6 -0.7 -0.8 -1.2 -1.0  2.2 -0.6 -0.3 -0.3 -0.5 -0.6 -1.6 -1.2\n",
      "Q  1.8 -1.4  0.6  2.1  0.7  0.2 -0.5  1.4  1.7  1.1  1.2  1.6 -0.6  2.2  2.0  1.8  1.7  1.5 -1.0  0.5\n",
      "R  1.5 -1.5 -0.1  1.9  0.4  0.1 -0.4  1.4  2.1  1.2  1.1  1.2 -0.3  2.0  2.2  2.0  1.9  1.5 -0.8  0.8\n",
      "S  2.0 -1.3  0.7  1.8  0.5  0.7 -0.4  1.6  1.4  1.3  0.6  1.9 -0.3  1.8  2.0  2.2  2.1  1.8 -0.8  0.4\n",
      "T  1.9 -1.4  0.0  1.6  0.3  0.2 -0.9  1.6  1.2  1.2  0.8  1.1 -0.5  1.7  1.9  2.1  2.2  1.6 -1.0  0.3\n",
      "V  2.0 -1.4  0.0  1.6  0.8  0.1 -0.7  2.2  1.2  2.0  0.8  1.1 -0.6  1.5  1.5  1.8  1.6  2.2 -0.7  0.3\n",
      "W -0.9 -1.8 -1.4 -1.0 -0.9 -1.3 -1.7 -0.7 -1.1 -0.8 -1.3 -1.1 -1.6 -1.0 -0.8 -0.8 -1.0 -0.7  2.2 -0.6\n",
      "Y  0.2 -1.1 -0.4  0.2  2.0 -0.2 -0.8  0.4  0.5  0.5 -0.2 -0.1 -1.2  0.5  0.8  0.4  0.3  0.3 -0.6  2.2\n",
      "\n",
      "SCHNEIDER #  Adrian Schneider, Gina M. Cannarozzi, and Gaston H. Gonnet:\n",
      "#  \"Empirical codon substitution matrix.\"\n",
      "#  BMC Bioinformatics 6:134 (2005).\n",
      "#  Additional File 3.\n",
      "#  PMID 15927081\n",
      "      AAA   AAC   AAG   AAT   ACA   ACC   ACG   ACT   AGA   AGC   AGG   AGT   ATA   ATC   ATG   ATT   CAA   CAC   CAG   CAT   CCA   CCC   CCG   CCT   CGA   CGC   CGG   CGT   CTA   CTC   CTG   CTT   GAA   GAC   GAG   GAT   GCA   GCC   GCG   GCT   GGA   GGC   GGG   GGT   GTA   GTC   GTG   GTT   TAA   TAC   TAG   TAT   TCA   TCC   TCG   TCT   TGA   TGC   TGG   TGT   TTA   TTC   TTG   TTT\n",
      "AAA  11.6  -2.7   9.7  -1.7  -2.7  -6.4  -3.9  -5.6   5.1  -5.0   3.6  -4.2  -6.3 -13.0  -7.1 -11.5   0.4  -6.0  -1.9  -5.3  -8.5 -11.2  -8.9 -10.8   2.1   0.0   1.4   0.2 -10.2 -13.5 -13.0 -12.5  -2.6  -8.5  -5.0  -8.1  -6.3  -9.9  -7.5  -9.0  -7.1 -10.2  -8.2  -9.2  -8.2 -12.5 -11.1 -11.4 -50.0 -14.8 -50.0 -13.8  -7.3 -10.1  -8.4  -9.1 -50.0 -13.0 -13.5 -12.4 -10.7 -18.1 -11.8 -17.2\n",
      "AAC  -2.7  13.0  -3.3  10.9  -3.5  -0.4  -3.3  -1.8  -5.4   4.6  -5.5   3.0 -10.2  -7.9  -9.9  -9.6  -5.0   0.5  -5.5  -1.0 -10.3  -8.1  -9.4  -9.6  -8.1  -5.0  -7.3  -6.3 -13.4 -11.3 -14.4 -12.9  -6.3   0.8  -6.4  -1.1  -7.4  -5.0  -6.2  -6.5  -5.6  -1.6  -4.7  -3.0 -10.8  -8.7 -11.9 -10.0 -50.0  -6.2 -50.0  -7.5  -6.3  -4.3  -6.2  -5.4 -50.0  -7.0 -16.3  -8.2 -13.2 -12.3 -13.1 -13.3\n",
      "AAG   9.7  -3.3  11.6  -2.8  -4.5  -6.7  -3.1  -6.9   3.3  -5.5   4.8  -5.1  -8.9 -13.2  -5.7 -12.6  -1.5  -6.1  -0.6  -6.1 -10.0 -11.8  -8.6 -11.9   1.2   0.5   2.2   0.1 -11.8 -14.0 -11.9 -13.4  -4.9  -9.1  -3.4  -8.9  -8.0 -10.0  -7.2 -10.0  -9.1 -10.2  -7.1  -9.9 -10.1 -13.0 -10.6 -12.7 -50.0 -14.9 -50.0 -14.4  -8.9 -10.8  -8.9 -10.4 -50.0 -13.1 -11.8 -13.0 -12.4 -19.4 -11.5 -17.8\n",
      "AAT  -1.7  10.9  -2.8  12.9  -2.7  -2.2  -2.8   0.2  -4.8   2.9  -5.2   5.2  -9.1  -9.5  -9.0  -7.0  -4.0  -1.0  -5.0   1.0  -9.0  -8.8  -8.9  -7.7  -7.0  -6.8  -7.1  -4.3 -12.4 -12.9 -13.9 -10.0  -4.9  -0.7  -5.6   1.4  -6.0  -6.5  -6.2  -4.8  -5.1  -3.2  -5.0  -1.2  -9.8  -9.9 -11.3  -8.2 -50.0  -7.6 -50.0  -5.1  -5.5  -5.8  -5.9  -4.2 -50.0  -8.3 -15.5  -5.9 -11.6 -14.2 -12.0 -11.6\n",
      "ACA  -2.7  -3.5  -4.5  -2.7  11.7   9.0  10.6   9.6  -3.2  -0.7  -5.1  -0.4   0.7  -4.6  -0.4  -3.6  -4.8  -8.9  -6.5  -7.8  -1.4  -4.1  -2.9  -3.3  -8.0  -9.1  -8.0  -8.2  -6.0  -9.3  -7.7  -8.5  -6.4 -10.1  -7.5  -8.6   2.8  -0.7   0.9   0.1  -5.9  -7.3  -6.5  -6.5  -0.4  -3.9  -2.6  -3.1 -50.0 -14.6 -50.0 -12.6   2.9   0.1   1.8   1.0 -50.0  -9.2 -13.6  -7.7  -4.8 -12.2  -6.2 -11.0\n",
      "ACC  -6.4  -0.4  -6.7  -2.2   9.0  12.3   9.8   9.6  -7.1   2.4  -7.2   0.3  -2.9  -1.2  -3.0  -3.3  -7.6  -6.7  -8.1  -7.9  -4.9  -1.9  -4.5  -3.8  -9.5  -7.8  -9.3  -8.9  -9.4  -7.1  -9.0  -9.0  -9.4  -7.5  -9.3  -9.2  -0.7   2.6   0.2  -0.1  -8.0  -4.6  -7.7  -6.1  -3.6  -1.0  -4.1  -3.1 -50.0 -11.6 -50.0 -11.6   0.2   2.3   0.4   0.5 -50.0  -6.5 -15.3  -7.8  -8.4  -9.5  -8.3 -11.1\n",
      "ACG  -3.9  -3.3  -3.1  -2.8  10.6   9.8  12.2   9.8  -4.9   0.1  -3.6  -0.2  -1.7  -4.1   0.9  -3.5  -5.9  -8.9  -5.6  -8.5  -3.9  -3.9  -2.2  -4.5  -7.3  -7.5  -6.4  -8.5  -7.1  -8.2  -6.7  -7.8  -7.9  -9.1  -7.1  -9.4   0.9  -0.2   2.5  -0.3  -6.9  -6.5  -5.5  -6.8  -1.7  -3.4  -1.7  -2.8 -50.0 -13.0 -50.0 -11.5   1.5   0.7   2.7   0.6 -50.0  -8.9 -12.1  -7.4  -6.1 -11.5  -5.4 -10.4\n",
      "ACT  -5.6  -1.8  -6.9   0.2   9.6   9.6   9.8  11.6  -6.6   0.9  -7.2   2.5  -2.3  -3.4  -2.3  -0.6  -6.5  -8.2  -8.0  -6.3  -3.7  -3.5  -3.9  -1.6  -9.9  -8.9  -9.6  -7.0  -8.6  -8.9  -8.8  -6.8  -8.6  -8.2  -8.8  -6.8   0.3   0.1   0.2   2.4  -7.4  -6.2  -6.9  -4.4  -2.9  -2.9  -3.5  -0.9 -50.0 -12.2 -50.0 -10.1   1.2   0.6   1.2   2.4 -50.0  -7.6 -16.1  -5.8  -7.2 -10.8  -7.4  -9.0\n",
      "AGA   5.1  -5.4   3.3  -4.8  -3.2  -7.1  -4.9  -6.6  13.3  -2.8  11.2  -1.9  -5.7 -12.5  -7.2 -11.6  -0.7  -4.8  -3.1  -4.3 -10.0 -11.6  -8.8 -11.7  10.5   7.7   9.1   8.5  -9.8 -12.7 -11.7 -11.8  -6.3 -11.4  -8.8 -11.0  -7.5 -10.3  -8.3  -9.9  -1.9  -6.7  -4.1  -6.4  -7.9 -12.5 -11.4 -11.7 -50.0 -14.0 -50.0 -13.2  -8.5 -10.4  -9.1  -9.6 -50.0  -9.8  -7.7  -8.8 -10.3 -17.9 -11.8 -16.0\n",
      "AGC  -5.0   4.6  -5.5   2.9  -0.7   2.4   0.1   0.9  -2.8  12.8  -2.6  11.0  -8.4  -6.3  -8.3  -7.7  -6.1  -2.9  -6.5  -4.2  -8.5  -5.9  -7.5  -7.7  -5.5  -2.4  -5.5  -4.1 -12.9 -10.3 -12.7 -11.3  -7.4  -3.1  -7.4  -4.9  -4.4  -1.8  -3.5  -3.5  -1.6   3.0  -0.8   0.9  -8.4  -6.3  -9.6  -7.3 -50.0  -9.0 -50.0  -9.7  -2.3  -0.2  -1.2  -1.5 -50.0  -0.7 -13.3  -2.0 -11.8 -11.9 -12.3 -12.7\n",
      "AGG   3.6  -5.5   4.8  -5.2  -5.1  -7.2  -3.6  -7.2  11.2  -2.6  13.4  -2.1  -7.7 -12.5  -5.3 -11.9  -2.2  -4.4  -1.9  -4.8 -10.2 -11.5  -8.1 -12.0   9.3   8.2  10.0   8.0 -10.8 -11.9 -10.7 -12.4  -8.2 -11.8  -6.9 -11.5  -8.4  -9.9  -7.1 -10.6  -4.7  -6.8  -1.2  -7.0  -9.8 -11.6  -9.6 -12.1 -50.0 -14.1 -50.0 -12.7 -10.0 -10.8  -8.8 -10.5 -50.0  -9.2  -4.2  -9.3 -11.0 -18.2 -11.1 -16.1\n",
      "AGT  -4.2   3.0  -5.1   5.2  -0.4   0.3  -0.2   2.5  -1.9  11.0  -2.1  13.2  -7.6  -8.5  -8.0  -5.3  -5.6  -4.5  -6.8  -2.4  -7.8  -7.9  -7.9  -6.2  -5.7  -5.1  -5.4  -2.1 -12.4 -12.3 -13.1 -10.1  -6.6  -4.4  -6.9  -2.6  -3.6  -3.8  -3.7  -2.0  -1.4   0.8  -0.9   3.0  -8.2  -8.1  -9.2  -5.8 -50.0 -10.1 -50.0  -7.4  -1.7  -2.0  -1.5  -0.5 -50.0  -2.1 -12.6  -0.4 -11.6 -13.9 -11.4 -11.1\n",
      "ATA  -6.3 -10.2  -8.9  -9.1   0.7  -2.9  -1.7  -2.3  -5.7  -8.4  -7.7  -7.6  13.2   9.6   3.5   9.7  -8.7 -12.4 -10.7 -10.7  -7.8 -10.2  -8.8  -9.6  -9.5 -11.9 -10.9 -10.5   2.3  -0.4  -0.2  -0.3  -9.7 -15.5 -11.4 -13.7  -3.0  -6.4  -4.1  -5.4  -9.6 -11.9 -10.5 -11.8   6.2   3.3   3.7   3.6 -50.0 -13.6 -50.0 -11.9  -5.6  -8.8  -7.2  -8.9 -50.0 -12.4 -14.1 -11.6   2.8  -6.4   0.5  -5.2\n",
      "ATC -13.0  -7.9 -13.2  -9.5  -4.6  -1.2  -4.1  -3.4 -12.5  -6.3 -12.5  -8.5   9.6  12.7   0.2  10.5 -12.4 -11.6 -13.2 -12.0 -12.4 -10.1 -11.6 -12.0 -15.1 -13.1 -14.3 -13.4  -1.4   1.4  -1.6  -0.5 -14.9 -14.3 -15.2 -16.8  -7.7  -4.7  -6.3  -6.6 -14.6 -10.8 -12.8 -13.0   2.3   6.0   2.5   3.6 -50.0 -11.2 -50.0 -11.9 -10.5  -9.2 -10.5 -10.5 -50.0 -10.3 -16.2 -11.3  -1.7  -3.7  -2.5  -5.4\n",
      "ATG  -7.1  -9.9  -5.7  -9.0  -0.4  -3.0   0.9  -2.3  -7.2  -8.3  -5.3  -8.0   3.5   0.2  14.3   1.0  -7.3 -11.2  -6.9  -9.7  -8.5 -10.2  -7.7  -9.8 -10.0 -10.4  -8.4 -10.2   1.0  -0.6   1.7  -0.5 -11.1 -14.8  -9.7 -14.5  -3.7  -5.8  -3.1  -5.6 -10.4 -11.5  -8.2 -11.1   0.6  -1.6   1.6  -1.1 -50.0 -12.4 -50.0 -11.5  -5.5  -8.0  -4.9  -7.2 -50.0 -12.4 -10.3 -11.6   0.7  -6.8   2.4  -6.0\n",
      "ATT -11.5  -9.6 -12.6  -7.0  -3.6  -3.3  -3.5  -0.6 -11.6  -7.7 -11.9  -5.3   9.7  10.5   1.0  12.6 -12.0 -12.2 -12.5  -9.7 -11.1 -11.2 -11.9  -9.4 -12.9 -14.3 -14.0 -11.9  -1.0  -0.4  -1.6   1.2 -13.9 -15.4 -13.7 -12.6  -6.2  -6.7  -6.3  -4.4 -13.0 -12.2 -13.0 -10.4   2.9   3.8   2.5   5.8 -50.0 -12.5 -50.0 -10.2  -9.6  -9.7  -9.6  -8.3 -50.0 -11.6 -15.4  -9.4  -1.1  -5.4  -1.6  -3.3\n",
      "CAA   0.4  -5.0  -1.5  -4.0  -4.8  -7.6  -5.9  -6.5  -0.7  -6.1  -2.2  -5.6  -8.7 -12.4  -7.3 -12.0  12.8   2.3  10.2   3.0   0.0  -3.4  -0.7  -3.2   2.5  -0.8   0.9   0.2  -3.0  -7.2  -5.7  -6.1  -0.2  -6.6  -1.8  -6.0  -5.3  -8.2  -5.7  -7.6  -6.7  -9.6  -7.5  -9.3  -7.1 -10.5  -9.6  -9.4 -50.0  -8.1 -50.0  -7.2  -4.3  -6.8  -5.7  -6.3 -50.0  -9.5  -9.6  -8.9  -6.2 -12.8  -6.8 -11.8\n",
      "CAC  -6.0   0.5  -6.1  -1.0  -8.9  -6.7  -8.9  -8.2  -4.8  -2.9  -4.4  -4.5 -12.4 -11.6 -11.2 -12.2   2.3  14.6   1.9  12.9  -5.5  -2.6  -4.1  -4.7  -1.2   3.0  -1.1   1.6  -8.0  -4.7  -8.7  -6.0  -8.2  -4.7  -7.8  -6.7 -10.0  -8.3  -8.9  -9.9 -10.8  -7.1 -10.0  -9.2 -11.6 -10.1 -12.3 -11.9 -50.0   2.2 -50.0   0.9  -7.9  -5.9  -7.4  -6.9 -50.0  -4.6 -11.6  -5.6  -9.7  -5.9  -9.9  -7.3\n",
      "CAG  -1.9  -5.5  -0.6  -5.0  -6.5  -8.1  -5.6  -8.0  -3.1  -6.5  -1.9  -6.8 -10.7 -13.2  -6.9 -12.5  10.2   1.9  11.9   2.1  -2.3  -4.1  -0.5  -4.6  -0.1  -0.9   2.4  -0.8  -5.1  -6.9  -4.8  -6.7  -2.3  -7.1  -0.8  -7.2  -7.2  -8.0  -5.6  -8.5  -9.4  -9.7  -7.2  -9.8  -9.2 -11.1  -9.5 -10.7 -50.0  -8.7 -50.0  -8.6  -6.1  -7.5  -5.5  -7.7 -50.0 -10.4  -7.7 -10.0  -7.3 -13.6  -6.4 -13.3\n",
      "CAT  -5.3  -1.0  -6.1   1.0  -7.8  -7.9  -8.5  -6.3  -4.3  -4.2  -4.8  -2.4 -10.7 -12.0  -9.7  -9.7   3.0  12.9   2.1  14.7  -4.9  -3.8  -4.0  -2.4  -1.0   0.8  -1.3   3.4  -7.6  -5.7  -8.1  -3.5  -6.8  -5.8  -7.2  -4.2  -8.9 -10.2  -8.4  -8.2  -9.9  -8.4  -9.7  -6.4 -11.1 -11.1 -11.8 -10.0 -50.0   0.4 -50.0   2.6  -7.2  -7.0  -7.4  -5.1 -50.0  -5.7  -9.8  -3.2  -8.4  -7.4  -8.9  -5.9\n",
      "CCA  -8.5 -10.3 -10.0  -9.0  -1.4  -4.9  -3.9  -3.7 -10.0  -8.5 -10.2  -7.8  -7.8 -12.4  -8.5 -11.1   0.0  -5.5  -2.3  -4.9  12.6  10.0  11.1  10.5  -6.0  -8.3  -6.9  -7.5  -2.2  -7.4  -5.3  -5.9  -8.9 -12.5  -9.4 -11.6  -0.7  -3.9  -2.3  -3.0  -9.1  -9.8  -9.2  -9.5  -5.6  -9.0  -8.0  -8.5 -50.0 -15.3 -50.0 -14.1   2.3  -1.4   0.3  -0.5 -50.0 -13.6 -14.3 -11.5  -5.3 -13.7  -6.1 -12.7\n",
      "CCC -11.2  -8.1 -11.8  -8.8  -4.1  -1.9  -3.9  -3.5 -11.6  -5.9 -11.5  -7.9 -10.2 -10.1 -10.2 -11.2  -3.4  -2.6  -4.1  -3.8  10.0  13.1  10.7  10.6  -8.1  -5.2  -8.0  -6.7  -6.1  -3.7  -7.5  -5.6 -11.1 -10.3 -10.6 -11.9  -3.2  -1.0  -2.2  -2.8 -10.5  -8.1  -9.4  -9.5  -8.3  -6.9  -9.4  -8.8 -50.0 -11.3 -50.0 -12.8  -0.6   2.2  -0.3   0.1 -50.0 -10.1 -17.5 -11.1  -8.6  -9.9  -8.6 -11.9\n",
      "CCG  -8.9  -9.4  -8.6  -8.9  -2.9  -4.5  -2.2  -3.9  -8.8  -7.5  -8.1  -7.9  -8.8 -11.6  -7.7 -11.9  -0.7  -4.1  -0.5  -4.0  11.1  10.7  13.2  10.4  -5.7  -5.8  -3.5  -6.1  -3.6  -6.5  -3.3  -5.5  -9.4 -11.0  -8.7 -11.2  -1.8  -2.7   0.7  -3.0  -9.1  -8.2  -7.3  -9.4  -7.2  -8.2  -7.2  -8.2 -50.0 -13.2 -50.0 -13.3   0.5  -0.9   1.8  -0.6 -50.0 -11.4 -11.1 -10.5  -6.1 -12.9  -5.0 -11.9\n",
      "CCT -10.8  -9.6 -11.9  -7.7  -3.3  -3.8  -4.5  -1.6 -11.7  -7.7 -12.0  -6.2  -9.6 -12.0  -9.8  -9.4  -3.2  -4.7  -4.6  -2.4  10.5  10.6  10.4  12.6  -8.2  -7.9  -9.0  -4.7  -6.1  -6.0  -7.5  -2.9 -10.9 -12.2 -11.2 -10.9  -2.4  -2.9  -2.7  -0.9 -10.4  -9.7 -10.2  -8.0  -8.4  -9.1  -9.3  -6.5 -50.0 -13.3 -50.0 -11.0  -0.0  -0.4  -0.4   2.3 -50.0 -10.9 -17.2  -8.3  -8.3 -12.7  -7.8  -9.3\n",
      "CGA   2.1  -8.1   1.2  -7.0  -8.0  -9.5  -7.3  -9.9  10.5  -5.5   9.3  -5.7  -9.5 -15.1 -10.0 -12.9   2.5  -1.2  -0.1  -1.0  -6.0  -8.1  -5.7  -8.2  13.8  11.3  11.8  12.1  -6.0  -9.1  -9.0  -9.5  -8.9 -13.4 -10.0 -13.1  -9.9 -11.7  -9.8 -12.8  -5.9  -9.4  -6.5  -8.1 -10.7 -14.0 -13.3 -13.9 -50.0 -11.9 -50.0  -9.7  -8.9 -10.8  -9.0 -11.1 -50.0  -6.5  -5.3  -6.3 -10.5 -16.2 -10.4 -15.2\n",
      "CGC   0.0  -5.0   0.5  -6.8  -9.1  -7.8  -7.5  -8.9   7.7  -2.4   8.2  -5.1 -11.9 -13.1 -10.4 -14.3  -0.8   3.0  -0.9   0.8  -8.3  -5.2  -5.8  -7.9  11.3  15.0  11.2  12.8  -9.7  -5.3  -9.2  -7.8 -11.7 -10.6 -10.1 -12.2 -11.7  -8.8  -8.9 -11.8  -9.3  -5.0  -8.0  -8.2 -13.6 -11.4 -12.8 -13.0 -50.0  -7.2 -50.0  -8.4 -11.0  -7.7  -8.7 -10.2 -50.0  -1.5  -7.1  -4.9 -11.2 -11.6 -11.6 -15.0\n",
      "CGG   1.4  -7.3   2.2  -7.1  -8.0  -9.3  -6.4  -9.6   9.1  -5.5  10.0  -5.4 -10.9 -14.3  -8.4 -14.0   0.9  -1.1   2.4  -1.3  -6.9  -8.0  -3.5  -9.0  11.8  11.2  13.4  11.4  -7.3  -8.8  -6.5  -8.5 -10.0 -12.5  -7.8 -12.5  -9.6 -11.0  -7.1 -10.7  -7.9  -8.0  -4.3  -9.4 -11.4 -12.3 -10.9 -12.9 -50.0 -11.8 -50.0 -11.2  -9.4  -9.8  -7.3 -10.5 -50.0  -6.2  -2.2  -6.7  -9.8 -15.2  -8.5 -14.9\n",
      "CGT   0.2  -6.3   0.1  -4.3  -8.2  -8.9  -8.5  -7.0   8.5  -4.1   8.0  -2.1 -10.5 -13.4 -10.2 -11.9   0.2   1.6  -0.8   3.4  -7.5  -6.7  -6.1  -4.7  12.1  12.8  11.4  14.7  -8.7  -7.4  -9.4  -5.3  -9.9 -11.8 -10.5 -10.3  -9.9 -10.1  -9.3  -9.3  -8.1  -7.2  -8.0  -5.2 -12.0 -12.1 -12.6 -10.9 -50.0  -7.9 -50.0  -5.3  -9.3  -8.8  -8.9  -7.0 -50.0  -3.6  -7.1  -1.2 -10.2 -12.7  -9.9 -11.0\n",
      "CTA -10.2 -13.4 -11.8 -12.4  -6.0  -9.4  -7.1  -8.6  -9.8 -12.9 -10.8 -12.4   2.3  -1.4   1.0  -1.0  -3.0  -8.0  -5.1  -7.6  -2.2  -6.1  -3.6  -6.1  -6.0  -9.7  -7.3  -8.7  11.2   7.9   8.9   8.1 -12.1 -17.4 -13.2 -16.8  -6.3  -9.5  -7.5  -8.8 -12.7 -15.2 -13.1 -14.2   0.0  -3.1  -1.8  -3.1 -50.0 -10.7 -50.0  -9.3  -4.6  -8.9  -5.9  -8.3 -50.0 -12.8  -9.6 -10.6   9.5  -3.2   8.2  -2.8\n",
      "CTC -13.5 -11.3 -14.0 -12.9  -9.3  -7.1  -8.2  -8.9 -12.7 -10.3 -11.9 -12.3  -0.4   1.4  -0.6  -0.4  -7.2  -4.7  -6.9  -5.7  -7.4  -3.7  -6.5  -6.0  -9.1  -5.3  -8.8  -7.4   7.9  11.9   7.8   9.3 -15.2 -15.1 -14.6 -17.2  -9.7  -7.3  -8.2  -9.2 -15.0 -12.8 -14.1 -14.9  -2.8   0.2  -2.9  -2.3 -50.0  -7.4 -50.0  -8.7  -9.4  -7.3  -9.4  -9.4 -50.0  -8.0 -11.9  -9.7   6.5   0.5   6.4  -1.8\n",
      "CTG -13.0 -14.4 -11.9 -13.9  -7.7  -9.0  -6.7  -8.8 -11.7 -12.7 -10.7 -13.1  -0.2  -1.6   1.7  -1.6  -5.7  -8.7  -4.8  -8.1  -5.3  -7.5  -3.3  -7.5  -9.0  -9.2  -6.5  -9.4   8.9   7.8  10.1   7.8 -14.3 -17.7 -13.0 -17.8  -8.0  -9.4  -6.2  -9.2 -15.1 -14.9 -12.2 -14.9  -1.8  -3.2  -0.8  -3.2 -50.0 -10.8 -50.0  -9.9  -7.3  -9.6  -6.4  -9.4 -50.0 -12.3  -8.7 -11.3   7.6  -3.3   8.7  -2.9\n",
      "CTT -12.5 -12.9 -13.4 -10.0  -8.5  -9.0  -7.8  -6.8 -11.8 -11.3 -12.4 -10.1  -0.3  -0.5  -0.5   1.2  -6.1  -6.0  -6.7  -3.5  -5.9  -5.6  -5.5  -2.9  -9.5  -7.8  -8.5  -5.3   8.1   9.3   7.8  11.8 -14.4 -15.7 -14.5 -14.6  -8.4  -8.7  -7.9  -7.1 -14.3 -13.8 -13.8 -12.3  -2.4  -2.2  -2.9  -0.2 -50.0  -8.9 -50.0  -7.1  -8.1  -9.1  -8.7  -6.6 -50.0  -9.7 -11.4  -8.0   6.8  -1.8   6.9   0.4\n",
      "GAA  -2.6  -6.3  -4.9  -4.9  -6.4  -9.4  -7.9  -8.6  -6.3  -7.4  -8.2  -6.6  -9.7 -14.9 -11.1 -13.9  -0.2  -8.2  -2.3  -6.8  -8.9 -11.1  -9.4 -10.9  -8.9 -11.7 -10.0  -9.9 -12.1 -15.2 -14.3 -14.4  11.1   2.9   9.2   3.5  -3.0  -7.0  -4.3  -6.2  -2.2  -6.6  -3.7  -5.7  -6.0 -10.4  -8.6  -9.5 -50.0 -15.5 -50.0 -13.9  -7.8 -10.5  -8.7  -9.6 -50.0 -16.5 -17.1 -14.8 -12.1 -18.9 -12.4 -17.4\n",
      "GAC  -8.5   0.8  -9.1  -0.7 -10.1  -7.5  -9.1  -8.2 -11.4  -3.1 -11.8  -4.4 -15.5 -14.3 -14.8 -15.4  -6.6  -4.7  -7.1  -5.8 -12.5 -10.3 -11.0 -12.2 -13.4 -10.6 -12.5 -11.8 -17.4 -15.1 -17.7 -15.7   2.9  12.5   3.1  10.3  -7.7  -5.2  -6.2  -7.3  -5.4  -1.7  -5.4  -3.7 -11.6  -9.1 -13.0 -10.8 -50.0 -10.3 -50.0 -11.2  -9.8  -8.3  -9.1  -9.7 -50.0 -12.7 -20.6 -13.1 -16.0 -16.6 -17.0 -18.4\n",
      "GAG  -5.0  -6.4  -3.4  -5.6  -7.5  -9.3  -7.1  -8.8  -8.8  -7.4  -6.9  -6.9 -11.4 -15.2  -9.7 -13.7  -1.8  -7.8  -0.8  -7.2  -9.4 -10.6  -8.7 -11.2 -10.0 -10.1  -7.8 -10.5 -13.2 -14.6 -13.0 -14.5   9.2   3.1  10.8   3.0  -4.5  -6.7  -2.7  -6.7  -5.0  -6.8  -2.5  -6.7  -7.5 -10.4  -7.4 -10.2 -50.0 -15.5 -50.0 -14.2  -8.9 -10.2  -8.5 -10.0 -50.0 -15.6 -15.3 -15.5 -13.0 -18.8 -12.6 -18.3\n",
      "GAT  -8.1  -1.1  -8.9   1.4  -8.6  -9.2  -9.4  -6.8 -11.0  -4.9 -11.5  -2.6 -13.7 -16.8 -14.5 -12.6  -6.0  -6.7  -7.2  -4.2 -11.6 -11.9 -11.2 -10.9 -13.1 -12.2 -12.5 -10.3 -16.8 -17.2 -17.8 -14.6   3.5  10.3   3.0  12.4  -6.9  -7.5  -6.8  -4.9  -5.4  -4.2  -5.4  -1.3 -10.8 -11.5 -12.4  -8.2 -50.0 -12.5 -50.0  -8.9  -9.2 -10.0  -9.4  -7.9 -50.0 -14.3 -19.9 -11.4 -16.2 -19.2 -15.8 -15.9\n",
      "GCA  -6.3  -7.4  -8.0  -6.0   2.8  -0.7   0.9   0.3  -7.5  -4.4  -8.4  -3.6  -3.0  -7.7  -3.7  -6.2  -5.3 -10.0  -7.2  -8.9  -0.7  -3.2  -1.8  -2.4  -9.9 -11.7  -9.6  -9.9  -6.3  -9.7  -8.0  -8.4  -3.0  -7.7  -4.5  -6.9  11.3   8.2   9.4   9.1  -1.0  -3.3  -1.8  -2.4   1.5  -2.1  -0.4  -1.1 -50.0 -14.5 -50.0 -12.9   2.8  -0.0   1.4   0.7 -50.0  -8.7 -14.1  -7.5  -5.2 -12.5  -6.5 -10.7\n",
      "GCC  -9.9  -5.0 -10.0  -6.5  -0.7   2.6  -0.2   0.1 -10.3  -1.8  -9.9  -3.8  -6.4  -4.7  -5.8  -6.7  -8.2  -8.3  -8.0 -10.2  -3.9  -1.0  -2.7  -2.9 -11.7  -8.8 -11.0 -10.1  -9.5  -7.3  -9.4  -8.7  -7.0  -5.2  -6.7  -7.5   8.2  11.6   8.8   9.1  -3.7  -0.9  -3.1  -3.0  -2.0   1.2  -2.3  -1.3 -50.0 -12.1 -50.0 -12.6   0.1   2.5   0.7   0.6 -50.0  -6.0 -14.8  -7.5  -8.6  -9.4  -8.5 -11.2\n",
      "GCG  -7.5  -6.2  -7.2  -6.2   0.9   0.2   2.5   0.2  -8.3  -3.5  -7.1  -3.7  -4.1  -6.3  -3.1  -6.3  -5.7  -8.9  -5.6  -8.4  -2.3  -2.2   0.7  -2.7  -9.8  -8.9  -7.1  -9.3  -7.5  -8.2  -6.2  -7.9  -4.3  -6.2  -2.7  -6.8   9.4   8.8  12.1   8.8  -2.1  -2.0   0.1  -2.4   0.0  -1.1   1.3  -0.8 -50.0 -11.9 -50.0 -12.6   1.2   0.7   3.5   0.8 -50.0  -7.1 -12.4  -7.1  -7.1 -10.6  -5.0 -10.6\n",
      "GCT  -9.0  -6.5 -10.0  -4.8   0.1  -0.1  -0.3   2.4  -9.9  -3.5 -10.6  -2.0  -5.4  -6.6  -5.6  -4.4  -7.6  -9.9  -8.5  -8.2  -3.0  -2.8  -3.0  -0.9 -12.8 -11.8 -10.7  -9.3  -8.8  -9.2  -9.2  -7.1  -6.2  -7.3  -6.7  -4.9   9.1   9.1   8.8  11.2  -3.1  -3.0  -3.3  -1.0  -1.1  -1.4  -2.0   1.2 -50.0 -12.8 -50.0 -10.9   0.8   0.5   0.7   2.3 -50.0  -7.5 -16.5  -5.9  -7.8 -11.1  -7.8  -9.0\n",
      "GGA  -7.1  -5.6  -9.1  -5.1  -5.9  -8.0  -6.9  -7.4  -1.9  -1.6  -4.7  -1.4  -9.6 -14.6 -10.4 -13.0  -6.7 -10.8  -9.4  -9.9  -9.1 -10.5  -9.1 -10.4  -5.9  -9.3  -7.9  -8.1 -12.7 -15.0 -15.1 -14.3  -2.2  -5.4  -5.0  -5.4  -1.0  -3.7  -2.1  -3.1  12.8   9.6  11.1  10.1  -4.8  -9.4  -8.1  -8.6 -50.0 -18.3 -50.0 -15.4  -5.9  -7.5  -6.6  -7.1 -50.0  -9.5 -11.7  -8.4 -12.5 -17.3 -13.2 -15.4\n",
      "GGC -10.2  -1.6 -10.2  -3.2  -7.3  -4.6  -6.5  -6.2  -6.7   3.0  -6.8   0.8 -11.9 -10.8 -11.5 -12.2  -9.6  -7.1  -9.7  -8.4  -9.8  -8.1  -8.2  -9.7  -9.4  -5.0  -8.0  -7.2 -15.2 -12.8 -14.9 -13.8  -6.6  -1.7  -6.8  -4.2  -3.3  -0.9  -2.0  -3.0   9.6  12.8   9.7  10.5  -8.5  -5.6  -9.3  -7.8 -50.0 -12.7 -50.0 -13.8  -6.8  -5.1  -5.3  -6.5 -50.0  -4.1 -12.8  -5.8 -14.4 -14.1 -13.6 -14.8\n",
      "GGG  -8.2  -4.7  -7.1  -5.0  -6.5  -7.7  -5.5  -6.9  -4.1  -0.8  -1.2  -0.9 -10.5 -12.8  -8.2 -13.0  -7.5 -10.0  -7.2  -9.7  -9.2  -9.4  -7.3 -10.2  -6.5  -8.0  -4.3  -8.0 -13.1 -14.1 -12.2 -13.8  -3.7  -5.4  -2.5  -5.4  -1.8  -3.1   0.1  -3.3  11.1   9.7  12.9   9.8  -5.5  -8.1  -5.4  -7.8 -50.0 -16.4 -50.0 -14.8  -5.9  -7.1  -4.7  -7.2 -50.0  -8.4  -6.4  -8.1 -12.1 -16.2 -10.3 -15.8\n",
      "GGT  -9.2  -3.0  -9.9  -1.2  -6.5  -6.1  -6.8  -4.4  -6.4   0.9  -7.0   3.0 -11.8 -13.0 -11.1 -10.4  -9.3  -9.2  -9.8  -6.4  -9.5  -9.5  -9.4  -8.0  -8.1  -8.2  -9.4  -5.2 -14.2 -14.9 -14.9 -12.3  -5.7  -3.7  -6.7  -1.3  -2.4  -3.0  -2.4  -1.0  10.1  10.5   9.8  13.1  -7.2  -8.0  -8.4  -5.0 -50.0 -14.7 -50.0 -11.4  -6.3  -6.7  -5.8  -5.0 -50.0  -6.1 -13.4  -3.6 -13.0 -16.0 -13.7 -13.3\n",
      "GTA  -8.2 -10.8 -10.1  -9.8  -0.4  -3.6  -1.7  -2.9  -7.9  -8.4  -9.8  -8.2   6.2   2.3   0.6   2.9  -7.1 -11.6  -9.2 -11.1  -5.6  -8.3  -7.2  -8.4 -10.7 -13.6 -11.4 -12.0   0.0  -2.8  -1.8  -2.4  -6.0 -11.6  -7.5 -10.8   1.5  -2.0   0.0  -1.1  -4.8  -8.5  -5.5  -7.2  11.9   8.6  10.0   9.0 -50.0 -14.2 -50.0 -12.8  -4.1  -7.4  -5.7  -6.5 -50.0 -11.0 -14.3  -9.9   1.4  -7.9  -0.7  -6.8\n",
      "GTC -12.5  -8.7 -13.0  -9.9  -3.9  -1.0  -3.4  -2.9 -12.5  -6.3 -11.6  -8.1   3.3   6.0  -1.6   3.8 -10.5 -10.1 -11.1 -11.1  -9.0  -6.9  -8.2  -9.1 -14.0 -11.4 -12.3 -12.1  -3.1   0.2  -3.2  -2.2 -10.4  -9.1 -10.4 -11.5  -2.1   1.2  -1.1  -1.4  -9.4  -5.6  -8.1  -8.0   8.6  12.4   8.6   9.5 -50.0 -10.4 -50.0 -11.6  -7.2  -5.7  -7.5  -7.4 -50.0  -8.6 -15.7  -9.7  -3.0  -3.7  -3.2  -5.9\n",
      "GTG -11.1 -11.9 -10.6 -11.3  -2.6  -4.1  -1.7  -3.5 -11.4  -9.6  -9.6  -9.2   3.7   2.5   1.6   2.5  -9.6 -12.3  -9.5 -11.8  -8.0  -9.4  -7.2  -9.3 -13.3 -12.8 -10.9 -12.6  -1.8  -2.9  -0.8  -2.9  -8.6 -13.0  -7.4 -12.4  -0.4  -2.3   1.3  -2.0  -8.1  -9.3  -5.4  -8.4  10.0   8.6  11.4   8.9 -50.0 -13.7 -50.0 -13.2  -6.1  -8.1  -5.5  -7.5 -50.0 -11.3 -12.7 -10.1  -1.6  -7.8  -0.3  -7.2\n",
      "GTT -11.4 -10.0 -12.7  -8.2  -3.1  -3.1  -2.8  -0.9 -11.7  -7.3 -12.1  -5.8   3.6   3.6  -1.1   5.8  -9.4 -11.9 -10.7 -10.0  -8.5  -8.8  -8.2  -6.5 -13.9 -13.0 -12.9 -10.9  -3.1  -2.3  -3.2  -0.2  -9.5 -10.8 -10.2  -8.2  -1.1  -1.3  -0.8   1.2  -8.6  -7.8  -7.8  -5.0   9.0   9.5   8.9  12.0 -50.0 -13.0 -50.0 -10.3  -6.2  -7.0  -7.1  -5.4 -50.0  -9.3 -14.0  -8.2  -2.4  -6.3  -2.4  -3.9\n",
      "TAA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0  30.6 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAC -14.8  -6.2 -14.9  -7.6 -14.6 -11.6 -13.0 -12.2 -14.0  -9.0 -14.1 -10.1 -13.6 -11.2 -12.4 -12.5  -8.1   2.2  -8.7   0.4 -15.3 -11.3 -13.2 -13.3 -11.9  -7.2 -11.8  -7.9 -10.7  -7.4 -10.8  -8.9 -15.5 -10.3 -15.5 -12.5 -14.5 -12.1 -11.9 -12.8 -18.3 -12.7 -16.4 -14.7 -14.2 -10.4 -13.7 -13.0 -50.0  15.1 -50.0  13.3  -9.9  -5.8  -8.9  -7.6 -50.0  -1.5  -7.8  -3.0  -8.1   3.6  -9.4   2.0\n",
      "TAG -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  30.6 -50.0  35.2 -50.0 -50.0 -50.0 -50.0 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAT -13.8  -7.5 -14.4  -5.1 -12.6 -11.6 -11.5 -10.1 -13.2  -9.7 -12.7  -7.4 -11.9 -11.9 -11.5 -10.2  -7.2   0.9  -8.6   2.6 -14.1 -12.8 -13.3 -11.0  -9.7  -8.4 -11.2  -5.3  -9.3  -8.7  -9.9  -7.1 -13.9 -11.2 -14.2  -8.9 -12.9 -12.6 -12.6 -10.9 -15.4 -13.8 -14.8 -11.4 -12.8 -11.6 -13.2 -10.3 -50.0  13.3 -50.0  15.2  -8.6  -7.0  -8.4  -4.8 -50.0  -3.0  -7.1  -0.1  -7.3   2.3  -8.7   3.9\n",
      "TCA  -7.3  -6.3  -8.9  -5.5   2.9   0.2   1.5   1.2  -8.5  -2.3 -10.0  -1.7  -5.6 -10.5  -5.5  -9.6  -4.3  -7.9  -6.1  -7.2   2.3  -0.6   0.5  -0.0  -8.9 -11.0  -9.4  -9.3  -4.6  -9.4  -7.3  -8.1  -7.8  -9.8  -8.9  -9.2   2.8   0.1   1.2   0.8  -5.9  -6.8  -5.9  -6.3  -4.1  -7.2  -6.1  -6.2 -50.0  -9.9 -50.0  -8.6  12.5   9.4  11.0   9.8 -50.0  -4.8  -8.7  -3.5  -0.0  -8.6  -2.1  -7.0\n",
      "TCC -10.1  -4.3 -10.8  -5.8   0.1   2.3   0.7   0.6 -10.4  -0.2 -10.8  -2.0  -8.8  -9.2  -8.0  -9.7  -6.8  -5.9  -7.5  -7.0  -1.4   2.2  -0.9  -0.4 -10.8  -7.7  -9.8  -8.8  -8.9  -7.3  -9.6  -9.1 -10.5  -8.3 -10.2 -10.0  -0.0   2.5   0.7   0.5  -7.5  -5.1  -7.1  -6.7  -7.4  -5.7  -8.1  -7.0 -50.0  -5.8 -50.0  -7.0   9.4  12.7  10.1  10.0 -50.0  -1.5 -11.3  -3.3  -5.4  -4.2  -5.6  -6.8\n",
      "TCG  -8.4  -6.2  -8.9  -5.9   1.8   0.4   2.7   1.2  -9.1  -1.2  -8.8  -1.5  -7.2 -10.5  -4.9  -9.6  -5.7  -7.4  -5.5  -7.4   0.3  -0.3   1.8  -0.4  -9.0  -8.7  -7.3  -8.9  -5.9  -9.4  -6.4  -8.7  -8.7  -9.1  -8.5  -9.4   1.4   0.7   3.5   0.7  -6.6  -5.3  -4.7  -5.8  -5.7  -7.5  -5.5  -7.1 -50.0  -8.9 -50.0  -8.4  11.0  10.1  13.2  10.2 -50.0  -4.2  -6.1  -3.6  -3.5  -8.4  -1.2  -6.8\n",
      "TCT  -9.1  -5.4 -10.4  -4.2   1.0   0.5   0.6   2.4  -9.6  -1.5 -10.5  -0.5  -8.9 -10.5  -7.2  -8.3  -6.3  -6.9  -7.7  -5.1  -0.5   0.1  -0.6   2.3 -11.1 -10.2 -10.5  -7.0  -8.3  -9.4  -9.4  -6.6  -9.6  -9.7 -10.0  -7.9   0.7   0.6   0.8   2.3  -7.1  -6.5  -7.2  -5.0  -6.5  -7.4  -7.5  -5.4 -50.0  -7.6 -50.0  -4.8   9.8  10.0  10.2  12.1 -50.0  -3.1 -11.4  -0.7  -4.6  -6.6  -4.9  -3.9\n",
      "TGA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TGC -13.0  -7.0 -13.1  -8.3  -9.2  -6.5  -8.9  -7.6  -9.8  -0.7  -9.2  -2.1 -12.4 -10.3 -12.4 -11.6  -9.5  -4.6 -10.4  -5.7 -13.6 -10.1 -11.4 -10.9  -6.5  -1.5  -6.2  -3.6 -12.8  -8.0 -12.3  -9.7 -16.5 -12.7 -15.6 -14.3  -8.7  -6.0  -7.1  -7.5  -9.5  -4.1  -8.4  -6.1 -11.0  -8.6 -11.3  -9.3 -50.0  -1.5 -50.0  -3.0  -4.8  -1.5  -4.2  -3.1 -50.0  16.4  -5.1  14.2 -10.1  -4.3 -10.0  -5.4\n",
      "TGG -13.5 -16.3 -11.8 -15.5 -13.6 -15.3 -12.1 -16.1  -7.7 -13.3  -4.2 -12.6 -14.1 -16.2 -10.3 -15.4  -9.6 -11.6  -7.7  -9.8 -14.3 -17.5 -11.1 -17.2  -5.3  -7.1  -2.2  -7.1  -9.6 -11.9  -8.7 -11.4 -17.1 -20.6 -15.3 -19.9 -14.1 -14.8 -12.4 -16.5 -11.7 -12.8  -6.4 -13.4 -14.3 -15.7 -12.7 -14.0 -50.0  -7.8 -50.0  -7.1  -8.7 -11.3  -6.1 -11.4 -50.0  -5.1  18.6  -4.8  -8.6  -8.1  -4.5  -7.3\n",
      "TGT -12.4  -8.2 -13.0  -5.9  -7.7  -7.8  -7.4  -5.8  -8.8  -2.0  -9.3  -0.4 -11.6 -11.3 -11.6  -9.4  -8.9  -5.6 -10.0  -3.2 -11.5 -11.1 -10.5  -8.3  -6.3  -4.9  -6.7  -1.2 -10.6  -9.7 -11.3  -8.0 -14.8 -13.1 -15.5 -11.4  -7.5  -7.5  -7.1  -5.9  -8.4  -5.8  -8.1  -3.6  -9.9  -9.7 -10.1  -8.2 -50.0  -3.0 -50.0  -0.1  -3.5  -3.3  -3.6  -0.7 -50.0  14.2  -4.8  16.4  -8.9  -5.8  -8.9  -3.4\n",
      "TTA -10.7 -13.2 -12.4 -11.6  -4.8  -8.4  -6.1  -7.2 -10.3 -11.8 -11.0 -11.6   2.8  -1.7   0.7  -1.1  -6.2  -9.7  -7.3  -8.4  -5.3  -8.6  -6.1  -8.3 -10.5 -11.2  -9.8 -10.2   9.5   6.5   7.6   6.8 -12.1 -16.0 -13.0 -16.2  -5.2  -8.6  -7.1  -7.8 -12.5 -14.4 -12.1 -13.0   1.4  -3.0  -1.6  -2.4 -50.0  -8.1 -50.0  -7.3  -0.0  -5.4  -3.5  -4.6 -50.0 -10.1  -8.6  -8.9  13.2  -0.9   9.7  -0.3\n",
      "TTC -18.1 -12.3 -19.4 -14.2 -12.2  -9.5 -11.5 -10.8 -17.9 -11.9 -18.2 -13.9  -6.4  -3.7  -6.8  -5.4 -12.8  -5.9 -13.6  -7.4 -13.7  -9.9 -12.9 -12.7 -16.2 -11.6 -15.2 -12.7  -3.2   0.5  -3.3  -1.8 -18.9 -16.6 -18.8 -19.2 -12.5  -9.4 -10.6 -11.1 -17.3 -14.1 -16.2 -16.0  -7.9  -3.7  -7.8  -6.3 -50.0   3.6 -50.0   2.3  -8.6  -4.2  -8.4  -6.6 -50.0  -4.3  -8.1  -5.8  -0.9  14.2  -1.8  11.6\n",
      "TTG -11.8 -13.1 -11.5 -12.0  -6.2  -8.3  -5.4  -7.4 -11.8 -12.3 -11.1 -11.4   0.5  -2.5   2.4  -1.6  -6.8  -9.9  -6.4  -8.9  -6.1  -8.6  -5.0  -7.8 -10.4 -11.6  -8.5  -9.9   8.2   6.4   8.7   6.9 -12.4 -17.0 -12.6 -15.8  -6.5  -8.5  -5.0  -7.8 -13.2 -13.6 -10.3 -13.7  -0.7  -3.2  -0.3  -2.4 -50.0  -9.4 -50.0  -8.7  -2.1  -5.6  -1.2  -4.9 -50.0 -10.0  -4.5  -8.9   9.7  -1.8  11.3  -0.9\n",
      "TTT -17.2 -13.3 -17.8 -11.6 -11.0 -11.1 -10.4  -9.0 -16.0 -12.7 -16.1 -11.1  -5.2  -5.4  -6.0  -3.3 -11.8  -7.3 -13.3  -5.9 -12.7 -11.9 -11.9  -9.3 -15.2 -15.0 -14.9 -11.0  -2.8  -1.8  -2.9   0.4 -17.4 -18.4 -18.3 -15.9 -10.7 -11.2 -10.6  -9.0 -15.4 -14.8 -15.8 -13.3  -6.8  -5.9  -7.2  -3.9 -50.0   2.0 -50.0   3.9  -7.0  -6.8  -6.8  -3.9 -50.0  -5.4  -7.3  -3.4  -0.3  11.6  -0.9  14.1\n",
      "\n",
      "STR #  Steven Henikoff and Jorja G. Henikoff:\n",
      "#  \"Performance evaluation of amino acid substitution matrices.\"\n",
      "#  Proteins: Structure, Function, and Genetics: 17(1): 49-61 (1993).\n",
      "#  Figure 1, lower triangle.\n",
      "#  PMID 8234244\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  4.0 -2.0 -1.0  0.0 -3.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0 -1.0 -1.0  0.0 -1.0  0.0 -1.0  0.0 -3.0 -3.0\n",
      "C -2.0 11.0 -7.0 -3.0 -2.0 -6.0 -6.0 -4.0 -4.0 -6.0 -5.0 -6.0 -8.0 -3.0 -2.0 -4.0 -5.0 -4.0 -6.0 -6.0\n",
      "D -1.0 -7.0  6.0  2.0 -5.0 -1.0  0.0 -3.0 -1.0 -6.0 -4.0  2.0 -1.0  0.0 -2.0  0.0 -1.0 -4.0 -6.0 -3.0\n",
      "E  0.0 -3.0  2.0  5.0 -4.0 -2.0 -2.0 -3.0  1.0 -4.0 -2.0  0.0 -1.0  2.0  0.0 -1.0  0.0 -2.0 -6.0 -2.0\n",
      "F -3.0 -2.0 -5.0 -4.0  7.0 -6.0 -2.0  1.0 -3.0  2.0  0.0 -3.0 -5.0 -4.0 -4.0 -3.0 -3.0 -1.0  2.0  3.0\n",
      "G  0.0 -6.0 -1.0 -2.0 -6.0  5.0 -3.0 -5.0 -3.0 -5.0 -4.0 -1.0 -2.0 -2.0 -2.0 -1.0 -3.0 -4.0 -4.0 -3.0\n",
      "H -2.0 -6.0  0.0 -2.0 -2.0 -3.0  8.0 -5.0  0.0 -3.0 -2.0  2.0 -3.0  0.0  0.0 -2.0 -2.0 -2.0 -3.0  0.0\n",
      "I -2.0 -4.0 -3.0 -3.0  1.0 -5.0 -5.0  6.0 -3.0  2.0  1.0 -3.0 -4.0 -5.0 -3.0 -3.0 -2.0  2.0 -2.0 -1.0\n",
      "K -1.0 -4.0 -1.0  1.0 -3.0 -3.0  0.0 -3.0  5.0 -2.0 -1.0  0.0 -1.0  1.0  2.0 -1.0  0.0 -3.0 -3.0 -2.0\n",
      "L -2.0 -6.0 -6.0 -4.0  2.0 -5.0 -3.0  2.0 -2.0  5.0  3.0 -3.0 -3.0 -3.0 -3.0 -4.0 -3.0  1.0 -1.0 -2.0\n",
      "M  0.0 -5.0 -4.0 -2.0  0.0 -4.0 -2.0  1.0 -1.0  3.0  8.0 -2.0 -6.0  1.0 -4.0 -4.0 -2.0  0.0 -2.0 -1.0\n",
      "N -1.0 -6.0  2.0  0.0 -3.0 -1.0  2.0 -3.0  0.0 -3.0 -2.0  5.0 -2.0  0.0 -1.0  0.0  0.0 -4.0 -5.0 -1.0\n",
      "P -1.0 -8.0 -1.0 -1.0 -5.0 -2.0 -3.0 -4.0 -1.0 -3.0 -6.0 -2.0  7.0 -2.0 -2.0 -1.0 -1.0 -4.0 -4.0 -6.0\n",
      "Q  0.0 -3.0  0.0  2.0 -4.0 -2.0  0.0 -5.0  1.0 -3.0  1.0  0.0 -2.0  6.0  1.0 -1.0  0.0 -2.0 -5.0 -3.0\n",
      "R -1.0 -2.0 -2.0  0.0 -4.0 -2.0  0.0 -3.0  2.0 -3.0 -4.0 -1.0 -2.0  1.0  7.0  0.0 -1.0 -3.0 -2.0 -1.0\n",
      "S  0.0 -4.0  0.0 -1.0 -3.0 -1.0 -2.0 -3.0 -1.0 -4.0 -4.0  0.0 -1.0 -1.0  0.0  4.0  1.0 -3.0 -5.0 -2.0\n",
      "T -1.0 -5.0 -1.0  0.0 -3.0 -3.0 -2.0 -2.0  0.0 -3.0 -2.0  0.0 -1.0  0.0 -1.0  1.0  5.0 -1.0 -5.0 -2.0\n",
      "V  0.0 -4.0 -4.0 -2.0 -1.0 -4.0 -2.0  2.0 -3.0  1.0  0.0 -4.0 -4.0 -2.0 -3.0 -3.0 -1.0  5.0 -4.0 -1.0\n",
      "W -3.0 -6.0 -6.0 -6.0  2.0 -4.0 -3.0 -2.0 -3.0 -1.0 -2.0 -5.0 -4.0 -5.0 -2.0 -5.0 -5.0 -4.0 10.0  2.0\n",
      "Y -3.0 -6.0 -3.0 -2.0  3.0 -3.0  0.0 -1.0 -2.0 -2.0 -1.0 -1.0 -6.0 -3.0 -1.0 -2.0 -2.0 -1.0  2.0  7.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for m in substitution_matrices.load():\n",
    "    print(m, substitution_matrices.load(m))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#  Adrian Schneider, Gina M. Cannarozzi, and Gaston H. Gonnet:\n",
      "#  \"Empirical codon substitution matrix.\"\n",
      "#  BMC Bioinformatics 6:134 (2005).\n",
      "#  Additional File 3.\n",
      "#  PMID 15927081\n",
      "      AAA   AAC   AAG   AAT   ACA   ACC   ACG   ACT   AGA   AGC   AGG   AGT   ATA   ATC   ATG   ATT   CAA   CAC   CAG   CAT   CCA   CCC   CCG   CCT   CGA   CGC   CGG   CGT   CTA   CTC   CTG   CTT   GAA   GAC   GAG   GAT   GCA   GCC   GCG   GCT   GGA   GGC   GGG   GGT   GTA   GTC   GTG   GTT   TAA   TAC   TAG   TAT   TCA   TCC   TCG   TCT   TGA   TGC   TGG   TGT   TTA   TTC   TTG   TTT\n",
      "AAA  11.6  -2.7   9.7  -1.7  -2.7  -6.4  -3.9  -5.6   5.1  -5.0   3.6  -4.2  -6.3 -13.0  -7.1 -11.5   0.4  -6.0  -1.9  -5.3  -8.5 -11.2  -8.9 -10.8   2.1   0.0   1.4   0.2 -10.2 -13.5 -13.0 -12.5  -2.6  -8.5  -5.0  -8.1  -6.3  -9.9  -7.5  -9.0  -7.1 -10.2  -8.2  -9.2  -8.2 -12.5 -11.1 -11.4 -50.0 -14.8 -50.0 -13.8  -7.3 -10.1  -8.4  -9.1 -50.0 -13.0 -13.5 -12.4 -10.7 -18.1 -11.8 -17.2\n",
      "AAC  -2.7  13.0  -3.3  10.9  -3.5  -0.4  -3.3  -1.8  -5.4   4.6  -5.5   3.0 -10.2  -7.9  -9.9  -9.6  -5.0   0.5  -5.5  -1.0 -10.3  -8.1  -9.4  -9.6  -8.1  -5.0  -7.3  -6.3 -13.4 -11.3 -14.4 -12.9  -6.3   0.8  -6.4  -1.1  -7.4  -5.0  -6.2  -6.5  -5.6  -1.6  -4.7  -3.0 -10.8  -8.7 -11.9 -10.0 -50.0  -6.2 -50.0  -7.5  -6.3  -4.3  -6.2  -5.4 -50.0  -7.0 -16.3  -8.2 -13.2 -12.3 -13.1 -13.3\n",
      "AAG   9.7  -3.3  11.6  -2.8  -4.5  -6.7  -3.1  -6.9   3.3  -5.5   4.8  -5.1  -8.9 -13.2  -5.7 -12.6  -1.5  -6.1  -0.6  -6.1 -10.0 -11.8  -8.6 -11.9   1.2   0.5   2.2   0.1 -11.8 -14.0 -11.9 -13.4  -4.9  -9.1  -3.4  -8.9  -8.0 -10.0  -7.2 -10.0  -9.1 -10.2  -7.1  -9.9 -10.1 -13.0 -10.6 -12.7 -50.0 -14.9 -50.0 -14.4  -8.9 -10.8  -8.9 -10.4 -50.0 -13.1 -11.8 -13.0 -12.4 -19.4 -11.5 -17.8\n",
      "AAT  -1.7  10.9  -2.8  12.9  -2.7  -2.2  -2.8   0.2  -4.8   2.9  -5.2   5.2  -9.1  -9.5  -9.0  -7.0  -4.0  -1.0  -5.0   1.0  -9.0  -8.8  -8.9  -7.7  -7.0  -6.8  -7.1  -4.3 -12.4 -12.9 -13.9 -10.0  -4.9  -0.7  -5.6   1.4  -6.0  -6.5  -6.2  -4.8  -5.1  -3.2  -5.0  -1.2  -9.8  -9.9 -11.3  -8.2 -50.0  -7.6 -50.0  -5.1  -5.5  -5.8  -5.9  -4.2 -50.0  -8.3 -15.5  -5.9 -11.6 -14.2 -12.0 -11.6\n",
      "ACA  -2.7  -3.5  -4.5  -2.7  11.7   9.0  10.6   9.6  -3.2  -0.7  -5.1  -0.4   0.7  -4.6  -0.4  -3.6  -4.8  -8.9  -6.5  -7.8  -1.4  -4.1  -2.9  -3.3  -8.0  -9.1  -8.0  -8.2  -6.0  -9.3  -7.7  -8.5  -6.4 -10.1  -7.5  -8.6   2.8  -0.7   0.9   0.1  -5.9  -7.3  -6.5  -6.5  -0.4  -3.9  -2.6  -3.1 -50.0 -14.6 -50.0 -12.6   2.9   0.1   1.8   1.0 -50.0  -9.2 -13.6  -7.7  -4.8 -12.2  -6.2 -11.0\n",
      "ACC  -6.4  -0.4  -6.7  -2.2   9.0  12.3   9.8   9.6  -7.1   2.4  -7.2   0.3  -2.9  -1.2  -3.0  -3.3  -7.6  -6.7  -8.1  -7.9  -4.9  -1.9  -4.5  -3.8  -9.5  -7.8  -9.3  -8.9  -9.4  -7.1  -9.0  -9.0  -9.4  -7.5  -9.3  -9.2  -0.7   2.6   0.2  -0.1  -8.0  -4.6  -7.7  -6.1  -3.6  -1.0  -4.1  -3.1 -50.0 -11.6 -50.0 -11.6   0.2   2.3   0.4   0.5 -50.0  -6.5 -15.3  -7.8  -8.4  -9.5  -8.3 -11.1\n",
      "ACG  -3.9  -3.3  -3.1  -2.8  10.6   9.8  12.2   9.8  -4.9   0.1  -3.6  -0.2  -1.7  -4.1   0.9  -3.5  -5.9  -8.9  -5.6  -8.5  -3.9  -3.9  -2.2  -4.5  -7.3  -7.5  -6.4  -8.5  -7.1  -8.2  -6.7  -7.8  -7.9  -9.1  -7.1  -9.4   0.9  -0.2   2.5  -0.3  -6.9  -6.5  -5.5  -6.8  -1.7  -3.4  -1.7  -2.8 -50.0 -13.0 -50.0 -11.5   1.5   0.7   2.7   0.6 -50.0  -8.9 -12.1  -7.4  -6.1 -11.5  -5.4 -10.4\n",
      "ACT  -5.6  -1.8  -6.9   0.2   9.6   9.6   9.8  11.6  -6.6   0.9  -7.2   2.5  -2.3  -3.4  -2.3  -0.6  -6.5  -8.2  -8.0  -6.3  -3.7  -3.5  -3.9  -1.6  -9.9  -8.9  -9.6  -7.0  -8.6  -8.9  -8.8  -6.8  -8.6  -8.2  -8.8  -6.8   0.3   0.1   0.2   2.4  -7.4  -6.2  -6.9  -4.4  -2.9  -2.9  -3.5  -0.9 -50.0 -12.2 -50.0 -10.1   1.2   0.6   1.2   2.4 -50.0  -7.6 -16.1  -5.8  -7.2 -10.8  -7.4  -9.0\n",
      "AGA   5.1  -5.4   3.3  -4.8  -3.2  -7.1  -4.9  -6.6  13.3  -2.8  11.2  -1.9  -5.7 -12.5  -7.2 -11.6  -0.7  -4.8  -3.1  -4.3 -10.0 -11.6  -8.8 -11.7  10.5   7.7   9.1   8.5  -9.8 -12.7 -11.7 -11.8  -6.3 -11.4  -8.8 -11.0  -7.5 -10.3  -8.3  -9.9  -1.9  -6.7  -4.1  -6.4  -7.9 -12.5 -11.4 -11.7 -50.0 -14.0 -50.0 -13.2  -8.5 -10.4  -9.1  -9.6 -50.0  -9.8  -7.7  -8.8 -10.3 -17.9 -11.8 -16.0\n",
      "AGC  -5.0   4.6  -5.5   2.9  -0.7   2.4   0.1   0.9  -2.8  12.8  -2.6  11.0  -8.4  -6.3  -8.3  -7.7  -6.1  -2.9  -6.5  -4.2  -8.5  -5.9  -7.5  -7.7  -5.5  -2.4  -5.5  -4.1 -12.9 -10.3 -12.7 -11.3  -7.4  -3.1  -7.4  -4.9  -4.4  -1.8  -3.5  -3.5  -1.6   3.0  -0.8   0.9  -8.4  -6.3  -9.6  -7.3 -50.0  -9.0 -50.0  -9.7  -2.3  -0.2  -1.2  -1.5 -50.0  -0.7 -13.3  -2.0 -11.8 -11.9 -12.3 -12.7\n",
      "AGG   3.6  -5.5   4.8  -5.2  -5.1  -7.2  -3.6  -7.2  11.2  -2.6  13.4  -2.1  -7.7 -12.5  -5.3 -11.9  -2.2  -4.4  -1.9  -4.8 -10.2 -11.5  -8.1 -12.0   9.3   8.2  10.0   8.0 -10.8 -11.9 -10.7 -12.4  -8.2 -11.8  -6.9 -11.5  -8.4  -9.9  -7.1 -10.6  -4.7  -6.8  -1.2  -7.0  -9.8 -11.6  -9.6 -12.1 -50.0 -14.1 -50.0 -12.7 -10.0 -10.8  -8.8 -10.5 -50.0  -9.2  -4.2  -9.3 -11.0 -18.2 -11.1 -16.1\n",
      "AGT  -4.2   3.0  -5.1   5.2  -0.4   0.3  -0.2   2.5  -1.9  11.0  -2.1  13.2  -7.6  -8.5  -8.0  -5.3  -5.6  -4.5  -6.8  -2.4  -7.8  -7.9  -7.9  -6.2  -5.7  -5.1  -5.4  -2.1 -12.4 -12.3 -13.1 -10.1  -6.6  -4.4  -6.9  -2.6  -3.6  -3.8  -3.7  -2.0  -1.4   0.8  -0.9   3.0  -8.2  -8.1  -9.2  -5.8 -50.0 -10.1 -50.0  -7.4  -1.7  -2.0  -1.5  -0.5 -50.0  -2.1 -12.6  -0.4 -11.6 -13.9 -11.4 -11.1\n",
      "ATA  -6.3 -10.2  -8.9  -9.1   0.7  -2.9  -1.7  -2.3  -5.7  -8.4  -7.7  -7.6  13.2   9.6   3.5   9.7  -8.7 -12.4 -10.7 -10.7  -7.8 -10.2  -8.8  -9.6  -9.5 -11.9 -10.9 -10.5   2.3  -0.4  -0.2  -0.3  -9.7 -15.5 -11.4 -13.7  -3.0  -6.4  -4.1  -5.4  -9.6 -11.9 -10.5 -11.8   6.2   3.3   3.7   3.6 -50.0 -13.6 -50.0 -11.9  -5.6  -8.8  -7.2  -8.9 -50.0 -12.4 -14.1 -11.6   2.8  -6.4   0.5  -5.2\n",
      "ATC -13.0  -7.9 -13.2  -9.5  -4.6  -1.2  -4.1  -3.4 -12.5  -6.3 -12.5  -8.5   9.6  12.7   0.2  10.5 -12.4 -11.6 -13.2 -12.0 -12.4 -10.1 -11.6 -12.0 -15.1 -13.1 -14.3 -13.4  -1.4   1.4  -1.6  -0.5 -14.9 -14.3 -15.2 -16.8  -7.7  -4.7  -6.3  -6.6 -14.6 -10.8 -12.8 -13.0   2.3   6.0   2.5   3.6 -50.0 -11.2 -50.0 -11.9 -10.5  -9.2 -10.5 -10.5 -50.0 -10.3 -16.2 -11.3  -1.7  -3.7  -2.5  -5.4\n",
      "ATG  -7.1  -9.9  -5.7  -9.0  -0.4  -3.0   0.9  -2.3  -7.2  -8.3  -5.3  -8.0   3.5   0.2  14.3   1.0  -7.3 -11.2  -6.9  -9.7  -8.5 -10.2  -7.7  -9.8 -10.0 -10.4  -8.4 -10.2   1.0  -0.6   1.7  -0.5 -11.1 -14.8  -9.7 -14.5  -3.7  -5.8  -3.1  -5.6 -10.4 -11.5  -8.2 -11.1   0.6  -1.6   1.6  -1.1 -50.0 -12.4 -50.0 -11.5  -5.5  -8.0  -4.9  -7.2 -50.0 -12.4 -10.3 -11.6   0.7  -6.8   2.4  -6.0\n",
      "ATT -11.5  -9.6 -12.6  -7.0  -3.6  -3.3  -3.5  -0.6 -11.6  -7.7 -11.9  -5.3   9.7  10.5   1.0  12.6 -12.0 -12.2 -12.5  -9.7 -11.1 -11.2 -11.9  -9.4 -12.9 -14.3 -14.0 -11.9  -1.0  -0.4  -1.6   1.2 -13.9 -15.4 -13.7 -12.6  -6.2  -6.7  -6.3  -4.4 -13.0 -12.2 -13.0 -10.4   2.9   3.8   2.5   5.8 -50.0 -12.5 -50.0 -10.2  -9.6  -9.7  -9.6  -8.3 -50.0 -11.6 -15.4  -9.4  -1.1  -5.4  -1.6  -3.3\n",
      "CAA   0.4  -5.0  -1.5  -4.0  -4.8  -7.6  -5.9  -6.5  -0.7  -6.1  -2.2  -5.6  -8.7 -12.4  -7.3 -12.0  12.8   2.3  10.2   3.0   0.0  -3.4  -0.7  -3.2   2.5  -0.8   0.9   0.2  -3.0  -7.2  -5.7  -6.1  -0.2  -6.6  -1.8  -6.0  -5.3  -8.2  -5.7  -7.6  -6.7  -9.6  -7.5  -9.3  -7.1 -10.5  -9.6  -9.4 -50.0  -8.1 -50.0  -7.2  -4.3  -6.8  -5.7  -6.3 -50.0  -9.5  -9.6  -8.9  -6.2 -12.8  -6.8 -11.8\n",
      "CAC  -6.0   0.5  -6.1  -1.0  -8.9  -6.7  -8.9  -8.2  -4.8  -2.9  -4.4  -4.5 -12.4 -11.6 -11.2 -12.2   2.3  14.6   1.9  12.9  -5.5  -2.6  -4.1  -4.7  -1.2   3.0  -1.1   1.6  -8.0  -4.7  -8.7  -6.0  -8.2  -4.7  -7.8  -6.7 -10.0  -8.3  -8.9  -9.9 -10.8  -7.1 -10.0  -9.2 -11.6 -10.1 -12.3 -11.9 -50.0   2.2 -50.0   0.9  -7.9  -5.9  -7.4  -6.9 -50.0  -4.6 -11.6  -5.6  -9.7  -5.9  -9.9  -7.3\n",
      "CAG  -1.9  -5.5  -0.6  -5.0  -6.5  -8.1  -5.6  -8.0  -3.1  -6.5  -1.9  -6.8 -10.7 -13.2  -6.9 -12.5  10.2   1.9  11.9   2.1  -2.3  -4.1  -0.5  -4.6  -0.1  -0.9   2.4  -0.8  -5.1  -6.9  -4.8  -6.7  -2.3  -7.1  -0.8  -7.2  -7.2  -8.0  -5.6  -8.5  -9.4  -9.7  -7.2  -9.8  -9.2 -11.1  -9.5 -10.7 -50.0  -8.7 -50.0  -8.6  -6.1  -7.5  -5.5  -7.7 -50.0 -10.4  -7.7 -10.0  -7.3 -13.6  -6.4 -13.3\n",
      "CAT  -5.3  -1.0  -6.1   1.0  -7.8  -7.9  -8.5  -6.3  -4.3  -4.2  -4.8  -2.4 -10.7 -12.0  -9.7  -9.7   3.0  12.9   2.1  14.7  -4.9  -3.8  -4.0  -2.4  -1.0   0.8  -1.3   3.4  -7.6  -5.7  -8.1  -3.5  -6.8  -5.8  -7.2  -4.2  -8.9 -10.2  -8.4  -8.2  -9.9  -8.4  -9.7  -6.4 -11.1 -11.1 -11.8 -10.0 -50.0   0.4 -50.0   2.6  -7.2  -7.0  -7.4  -5.1 -50.0  -5.7  -9.8  -3.2  -8.4  -7.4  -8.9  -5.9\n",
      "CCA  -8.5 -10.3 -10.0  -9.0  -1.4  -4.9  -3.9  -3.7 -10.0  -8.5 -10.2  -7.8  -7.8 -12.4  -8.5 -11.1   0.0  -5.5  -2.3  -4.9  12.6  10.0  11.1  10.5  -6.0  -8.3  -6.9  -7.5  -2.2  -7.4  -5.3  -5.9  -8.9 -12.5  -9.4 -11.6  -0.7  -3.9  -2.3  -3.0  -9.1  -9.8  -9.2  -9.5  -5.6  -9.0  -8.0  -8.5 -50.0 -15.3 -50.0 -14.1   2.3  -1.4   0.3  -0.5 -50.0 -13.6 -14.3 -11.5  -5.3 -13.7  -6.1 -12.7\n",
      "CCC -11.2  -8.1 -11.8  -8.8  -4.1  -1.9  -3.9  -3.5 -11.6  -5.9 -11.5  -7.9 -10.2 -10.1 -10.2 -11.2  -3.4  -2.6  -4.1  -3.8  10.0  13.1  10.7  10.6  -8.1  -5.2  -8.0  -6.7  -6.1  -3.7  -7.5  -5.6 -11.1 -10.3 -10.6 -11.9  -3.2  -1.0  -2.2  -2.8 -10.5  -8.1  -9.4  -9.5  -8.3  -6.9  -9.4  -8.8 -50.0 -11.3 -50.0 -12.8  -0.6   2.2  -0.3   0.1 -50.0 -10.1 -17.5 -11.1  -8.6  -9.9  -8.6 -11.9\n",
      "CCG  -8.9  -9.4  -8.6  -8.9  -2.9  -4.5  -2.2  -3.9  -8.8  -7.5  -8.1  -7.9  -8.8 -11.6  -7.7 -11.9  -0.7  -4.1  -0.5  -4.0  11.1  10.7  13.2  10.4  -5.7  -5.8  -3.5  -6.1  -3.6  -6.5  -3.3  -5.5  -9.4 -11.0  -8.7 -11.2  -1.8  -2.7   0.7  -3.0  -9.1  -8.2  -7.3  -9.4  -7.2  -8.2  -7.2  -8.2 -50.0 -13.2 -50.0 -13.3   0.5  -0.9   1.8  -0.6 -50.0 -11.4 -11.1 -10.5  -6.1 -12.9  -5.0 -11.9\n",
      "CCT -10.8  -9.6 -11.9  -7.7  -3.3  -3.8  -4.5  -1.6 -11.7  -7.7 -12.0  -6.2  -9.6 -12.0  -9.8  -9.4  -3.2  -4.7  -4.6  -2.4  10.5  10.6  10.4  12.6  -8.2  -7.9  -9.0  -4.7  -6.1  -6.0  -7.5  -2.9 -10.9 -12.2 -11.2 -10.9  -2.4  -2.9  -2.7  -0.9 -10.4  -9.7 -10.2  -8.0  -8.4  -9.1  -9.3  -6.5 -50.0 -13.3 -50.0 -11.0  -0.0  -0.4  -0.4   2.3 -50.0 -10.9 -17.2  -8.3  -8.3 -12.7  -7.8  -9.3\n",
      "CGA   2.1  -8.1   1.2  -7.0  -8.0  -9.5  -7.3  -9.9  10.5  -5.5   9.3  -5.7  -9.5 -15.1 -10.0 -12.9   2.5  -1.2  -0.1  -1.0  -6.0  -8.1  -5.7  -8.2  13.8  11.3  11.8  12.1  -6.0  -9.1  -9.0  -9.5  -8.9 -13.4 -10.0 -13.1  -9.9 -11.7  -9.8 -12.8  -5.9  -9.4  -6.5  -8.1 -10.7 -14.0 -13.3 -13.9 -50.0 -11.9 -50.0  -9.7  -8.9 -10.8  -9.0 -11.1 -50.0  -6.5  -5.3  -6.3 -10.5 -16.2 -10.4 -15.2\n",
      "CGC   0.0  -5.0   0.5  -6.8  -9.1  -7.8  -7.5  -8.9   7.7  -2.4   8.2  -5.1 -11.9 -13.1 -10.4 -14.3  -0.8   3.0  -0.9   0.8  -8.3  -5.2  -5.8  -7.9  11.3  15.0  11.2  12.8  -9.7  -5.3  -9.2  -7.8 -11.7 -10.6 -10.1 -12.2 -11.7  -8.8  -8.9 -11.8  -9.3  -5.0  -8.0  -8.2 -13.6 -11.4 -12.8 -13.0 -50.0  -7.2 -50.0  -8.4 -11.0  -7.7  -8.7 -10.2 -50.0  -1.5  -7.1  -4.9 -11.2 -11.6 -11.6 -15.0\n",
      "CGG   1.4  -7.3   2.2  -7.1  -8.0  -9.3  -6.4  -9.6   9.1  -5.5  10.0  -5.4 -10.9 -14.3  -8.4 -14.0   0.9  -1.1   2.4  -1.3  -6.9  -8.0  -3.5  -9.0  11.8  11.2  13.4  11.4  -7.3  -8.8  -6.5  -8.5 -10.0 -12.5  -7.8 -12.5  -9.6 -11.0  -7.1 -10.7  -7.9  -8.0  -4.3  -9.4 -11.4 -12.3 -10.9 -12.9 -50.0 -11.8 -50.0 -11.2  -9.4  -9.8  -7.3 -10.5 -50.0  -6.2  -2.2  -6.7  -9.8 -15.2  -8.5 -14.9\n",
      "CGT   0.2  -6.3   0.1  -4.3  -8.2  -8.9  -8.5  -7.0   8.5  -4.1   8.0  -2.1 -10.5 -13.4 -10.2 -11.9   0.2   1.6  -0.8   3.4  -7.5  -6.7  -6.1  -4.7  12.1  12.8  11.4  14.7  -8.7  -7.4  -9.4  -5.3  -9.9 -11.8 -10.5 -10.3  -9.9 -10.1  -9.3  -9.3  -8.1  -7.2  -8.0  -5.2 -12.0 -12.1 -12.6 -10.9 -50.0  -7.9 -50.0  -5.3  -9.3  -8.8  -8.9  -7.0 -50.0  -3.6  -7.1  -1.2 -10.2 -12.7  -9.9 -11.0\n",
      "CTA -10.2 -13.4 -11.8 -12.4  -6.0  -9.4  -7.1  -8.6  -9.8 -12.9 -10.8 -12.4   2.3  -1.4   1.0  -1.0  -3.0  -8.0  -5.1  -7.6  -2.2  -6.1  -3.6  -6.1  -6.0  -9.7  -7.3  -8.7  11.2   7.9   8.9   8.1 -12.1 -17.4 -13.2 -16.8  -6.3  -9.5  -7.5  -8.8 -12.7 -15.2 -13.1 -14.2   0.0  -3.1  -1.8  -3.1 -50.0 -10.7 -50.0  -9.3  -4.6  -8.9  -5.9  -8.3 -50.0 -12.8  -9.6 -10.6   9.5  -3.2   8.2  -2.8\n",
      "CTC -13.5 -11.3 -14.0 -12.9  -9.3  -7.1  -8.2  -8.9 -12.7 -10.3 -11.9 -12.3  -0.4   1.4  -0.6  -0.4  -7.2  -4.7  -6.9  -5.7  -7.4  -3.7  -6.5  -6.0  -9.1  -5.3  -8.8  -7.4   7.9  11.9   7.8   9.3 -15.2 -15.1 -14.6 -17.2  -9.7  -7.3  -8.2  -9.2 -15.0 -12.8 -14.1 -14.9  -2.8   0.2  -2.9  -2.3 -50.0  -7.4 -50.0  -8.7  -9.4  -7.3  -9.4  -9.4 -50.0  -8.0 -11.9  -9.7   6.5   0.5   6.4  -1.8\n",
      "CTG -13.0 -14.4 -11.9 -13.9  -7.7  -9.0  -6.7  -8.8 -11.7 -12.7 -10.7 -13.1  -0.2  -1.6   1.7  -1.6  -5.7  -8.7  -4.8  -8.1  -5.3  -7.5  -3.3  -7.5  -9.0  -9.2  -6.5  -9.4   8.9   7.8  10.1   7.8 -14.3 -17.7 -13.0 -17.8  -8.0  -9.4  -6.2  -9.2 -15.1 -14.9 -12.2 -14.9  -1.8  -3.2  -0.8  -3.2 -50.0 -10.8 -50.0  -9.9  -7.3  -9.6  -6.4  -9.4 -50.0 -12.3  -8.7 -11.3   7.6  -3.3   8.7  -2.9\n",
      "CTT -12.5 -12.9 -13.4 -10.0  -8.5  -9.0  -7.8  -6.8 -11.8 -11.3 -12.4 -10.1  -0.3  -0.5  -0.5   1.2  -6.1  -6.0  -6.7  -3.5  -5.9  -5.6  -5.5  -2.9  -9.5  -7.8  -8.5  -5.3   8.1   9.3   7.8  11.8 -14.4 -15.7 -14.5 -14.6  -8.4  -8.7  -7.9  -7.1 -14.3 -13.8 -13.8 -12.3  -2.4  -2.2  -2.9  -0.2 -50.0  -8.9 -50.0  -7.1  -8.1  -9.1  -8.7  -6.6 -50.0  -9.7 -11.4  -8.0   6.8  -1.8   6.9   0.4\n",
      "GAA  -2.6  -6.3  -4.9  -4.9  -6.4  -9.4  -7.9  -8.6  -6.3  -7.4  -8.2  -6.6  -9.7 -14.9 -11.1 -13.9  -0.2  -8.2  -2.3  -6.8  -8.9 -11.1  -9.4 -10.9  -8.9 -11.7 -10.0  -9.9 -12.1 -15.2 -14.3 -14.4  11.1   2.9   9.2   3.5  -3.0  -7.0  -4.3  -6.2  -2.2  -6.6  -3.7  -5.7  -6.0 -10.4  -8.6  -9.5 -50.0 -15.5 -50.0 -13.9  -7.8 -10.5  -8.7  -9.6 -50.0 -16.5 -17.1 -14.8 -12.1 -18.9 -12.4 -17.4\n",
      "GAC  -8.5   0.8  -9.1  -0.7 -10.1  -7.5  -9.1  -8.2 -11.4  -3.1 -11.8  -4.4 -15.5 -14.3 -14.8 -15.4  -6.6  -4.7  -7.1  -5.8 -12.5 -10.3 -11.0 -12.2 -13.4 -10.6 -12.5 -11.8 -17.4 -15.1 -17.7 -15.7   2.9  12.5   3.1  10.3  -7.7  -5.2  -6.2  -7.3  -5.4  -1.7  -5.4  -3.7 -11.6  -9.1 -13.0 -10.8 -50.0 -10.3 -50.0 -11.2  -9.8  -8.3  -9.1  -9.7 -50.0 -12.7 -20.6 -13.1 -16.0 -16.6 -17.0 -18.4\n",
      "GAG  -5.0  -6.4  -3.4  -5.6  -7.5  -9.3  -7.1  -8.8  -8.8  -7.4  -6.9  -6.9 -11.4 -15.2  -9.7 -13.7  -1.8  -7.8  -0.8  -7.2  -9.4 -10.6  -8.7 -11.2 -10.0 -10.1  -7.8 -10.5 -13.2 -14.6 -13.0 -14.5   9.2   3.1  10.8   3.0  -4.5  -6.7  -2.7  -6.7  -5.0  -6.8  -2.5  -6.7  -7.5 -10.4  -7.4 -10.2 -50.0 -15.5 -50.0 -14.2  -8.9 -10.2  -8.5 -10.0 -50.0 -15.6 -15.3 -15.5 -13.0 -18.8 -12.6 -18.3\n",
      "GAT  -8.1  -1.1  -8.9   1.4  -8.6  -9.2  -9.4  -6.8 -11.0  -4.9 -11.5  -2.6 -13.7 -16.8 -14.5 -12.6  -6.0  -6.7  -7.2  -4.2 -11.6 -11.9 -11.2 -10.9 -13.1 -12.2 -12.5 -10.3 -16.8 -17.2 -17.8 -14.6   3.5  10.3   3.0  12.4  -6.9  -7.5  -6.8  -4.9  -5.4  -4.2  -5.4  -1.3 -10.8 -11.5 -12.4  -8.2 -50.0 -12.5 -50.0  -8.9  -9.2 -10.0  -9.4  -7.9 -50.0 -14.3 -19.9 -11.4 -16.2 -19.2 -15.8 -15.9\n",
      "GCA  -6.3  -7.4  -8.0  -6.0   2.8  -0.7   0.9   0.3  -7.5  -4.4  -8.4  -3.6  -3.0  -7.7  -3.7  -6.2  -5.3 -10.0  -7.2  -8.9  -0.7  -3.2  -1.8  -2.4  -9.9 -11.7  -9.6  -9.9  -6.3  -9.7  -8.0  -8.4  -3.0  -7.7  -4.5  -6.9  11.3   8.2   9.4   9.1  -1.0  -3.3  -1.8  -2.4   1.5  -2.1  -0.4  -1.1 -50.0 -14.5 -50.0 -12.9   2.8  -0.0   1.4   0.7 -50.0  -8.7 -14.1  -7.5  -5.2 -12.5  -6.5 -10.7\n",
      "GCC  -9.9  -5.0 -10.0  -6.5  -0.7   2.6  -0.2   0.1 -10.3  -1.8  -9.9  -3.8  -6.4  -4.7  -5.8  -6.7  -8.2  -8.3  -8.0 -10.2  -3.9  -1.0  -2.7  -2.9 -11.7  -8.8 -11.0 -10.1  -9.5  -7.3  -9.4  -8.7  -7.0  -5.2  -6.7  -7.5   8.2  11.6   8.8   9.1  -3.7  -0.9  -3.1  -3.0  -2.0   1.2  -2.3  -1.3 -50.0 -12.1 -50.0 -12.6   0.1   2.5   0.7   0.6 -50.0  -6.0 -14.8  -7.5  -8.6  -9.4  -8.5 -11.2\n",
      "GCG  -7.5  -6.2  -7.2  -6.2   0.9   0.2   2.5   0.2  -8.3  -3.5  -7.1  -3.7  -4.1  -6.3  -3.1  -6.3  -5.7  -8.9  -5.6  -8.4  -2.3  -2.2   0.7  -2.7  -9.8  -8.9  -7.1  -9.3  -7.5  -8.2  -6.2  -7.9  -4.3  -6.2  -2.7  -6.8   9.4   8.8  12.1   8.8  -2.1  -2.0   0.1  -2.4   0.0  -1.1   1.3  -0.8 -50.0 -11.9 -50.0 -12.6   1.2   0.7   3.5   0.8 -50.0  -7.1 -12.4  -7.1  -7.1 -10.6  -5.0 -10.6\n",
      "GCT  -9.0  -6.5 -10.0  -4.8   0.1  -0.1  -0.3   2.4  -9.9  -3.5 -10.6  -2.0  -5.4  -6.6  -5.6  -4.4  -7.6  -9.9  -8.5  -8.2  -3.0  -2.8  -3.0  -0.9 -12.8 -11.8 -10.7  -9.3  -8.8  -9.2  -9.2  -7.1  -6.2  -7.3  -6.7  -4.9   9.1   9.1   8.8  11.2  -3.1  -3.0  -3.3  -1.0  -1.1  -1.4  -2.0   1.2 -50.0 -12.8 -50.0 -10.9   0.8   0.5   0.7   2.3 -50.0  -7.5 -16.5  -5.9  -7.8 -11.1  -7.8  -9.0\n",
      "GGA  -7.1  -5.6  -9.1  -5.1  -5.9  -8.0  -6.9  -7.4  -1.9  -1.6  -4.7  -1.4  -9.6 -14.6 -10.4 -13.0  -6.7 -10.8  -9.4  -9.9  -9.1 -10.5  -9.1 -10.4  -5.9  -9.3  -7.9  -8.1 -12.7 -15.0 -15.1 -14.3  -2.2  -5.4  -5.0  -5.4  -1.0  -3.7  -2.1  -3.1  12.8   9.6  11.1  10.1  -4.8  -9.4  -8.1  -8.6 -50.0 -18.3 -50.0 -15.4  -5.9  -7.5  -6.6  -7.1 -50.0  -9.5 -11.7  -8.4 -12.5 -17.3 -13.2 -15.4\n",
      "GGC -10.2  -1.6 -10.2  -3.2  -7.3  -4.6  -6.5  -6.2  -6.7   3.0  -6.8   0.8 -11.9 -10.8 -11.5 -12.2  -9.6  -7.1  -9.7  -8.4  -9.8  -8.1  -8.2  -9.7  -9.4  -5.0  -8.0  -7.2 -15.2 -12.8 -14.9 -13.8  -6.6  -1.7  -6.8  -4.2  -3.3  -0.9  -2.0  -3.0   9.6  12.8   9.7  10.5  -8.5  -5.6  -9.3  -7.8 -50.0 -12.7 -50.0 -13.8  -6.8  -5.1  -5.3  -6.5 -50.0  -4.1 -12.8  -5.8 -14.4 -14.1 -13.6 -14.8\n",
      "GGG  -8.2  -4.7  -7.1  -5.0  -6.5  -7.7  -5.5  -6.9  -4.1  -0.8  -1.2  -0.9 -10.5 -12.8  -8.2 -13.0  -7.5 -10.0  -7.2  -9.7  -9.2  -9.4  -7.3 -10.2  -6.5  -8.0  -4.3  -8.0 -13.1 -14.1 -12.2 -13.8  -3.7  -5.4  -2.5  -5.4  -1.8  -3.1   0.1  -3.3  11.1   9.7  12.9   9.8  -5.5  -8.1  -5.4  -7.8 -50.0 -16.4 -50.0 -14.8  -5.9  -7.1  -4.7  -7.2 -50.0  -8.4  -6.4  -8.1 -12.1 -16.2 -10.3 -15.8\n",
      "GGT  -9.2  -3.0  -9.9  -1.2  -6.5  -6.1  -6.8  -4.4  -6.4   0.9  -7.0   3.0 -11.8 -13.0 -11.1 -10.4  -9.3  -9.2  -9.8  -6.4  -9.5  -9.5  -9.4  -8.0  -8.1  -8.2  -9.4  -5.2 -14.2 -14.9 -14.9 -12.3  -5.7  -3.7  -6.7  -1.3  -2.4  -3.0  -2.4  -1.0  10.1  10.5   9.8  13.1  -7.2  -8.0  -8.4  -5.0 -50.0 -14.7 -50.0 -11.4  -6.3  -6.7  -5.8  -5.0 -50.0  -6.1 -13.4  -3.6 -13.0 -16.0 -13.7 -13.3\n",
      "GTA  -8.2 -10.8 -10.1  -9.8  -0.4  -3.6  -1.7  -2.9  -7.9  -8.4  -9.8  -8.2   6.2   2.3   0.6   2.9  -7.1 -11.6  -9.2 -11.1  -5.6  -8.3  -7.2  -8.4 -10.7 -13.6 -11.4 -12.0   0.0  -2.8  -1.8  -2.4  -6.0 -11.6  -7.5 -10.8   1.5  -2.0   0.0  -1.1  -4.8  -8.5  -5.5  -7.2  11.9   8.6  10.0   9.0 -50.0 -14.2 -50.0 -12.8  -4.1  -7.4  -5.7  -6.5 -50.0 -11.0 -14.3  -9.9   1.4  -7.9  -0.7  -6.8\n",
      "GTC -12.5  -8.7 -13.0  -9.9  -3.9  -1.0  -3.4  -2.9 -12.5  -6.3 -11.6  -8.1   3.3   6.0  -1.6   3.8 -10.5 -10.1 -11.1 -11.1  -9.0  -6.9  -8.2  -9.1 -14.0 -11.4 -12.3 -12.1  -3.1   0.2  -3.2  -2.2 -10.4  -9.1 -10.4 -11.5  -2.1   1.2  -1.1  -1.4  -9.4  -5.6  -8.1  -8.0   8.6  12.4   8.6   9.5 -50.0 -10.4 -50.0 -11.6  -7.2  -5.7  -7.5  -7.4 -50.0  -8.6 -15.7  -9.7  -3.0  -3.7  -3.2  -5.9\n",
      "GTG -11.1 -11.9 -10.6 -11.3  -2.6  -4.1  -1.7  -3.5 -11.4  -9.6  -9.6  -9.2   3.7   2.5   1.6   2.5  -9.6 -12.3  -9.5 -11.8  -8.0  -9.4  -7.2  -9.3 -13.3 -12.8 -10.9 -12.6  -1.8  -2.9  -0.8  -2.9  -8.6 -13.0  -7.4 -12.4  -0.4  -2.3   1.3  -2.0  -8.1  -9.3  -5.4  -8.4  10.0   8.6  11.4   8.9 -50.0 -13.7 -50.0 -13.2  -6.1  -8.1  -5.5  -7.5 -50.0 -11.3 -12.7 -10.1  -1.6  -7.8  -0.3  -7.2\n",
      "GTT -11.4 -10.0 -12.7  -8.2  -3.1  -3.1  -2.8  -0.9 -11.7  -7.3 -12.1  -5.8   3.6   3.6  -1.1   5.8  -9.4 -11.9 -10.7 -10.0  -8.5  -8.8  -8.2  -6.5 -13.9 -13.0 -12.9 -10.9  -3.1  -2.3  -3.2  -0.2  -9.5 -10.8 -10.2  -8.2  -1.1  -1.3  -0.8   1.2  -8.6  -7.8  -7.8  -5.0   9.0   9.5   8.9  12.0 -50.0 -13.0 -50.0 -10.3  -6.2  -7.0  -7.1  -5.4 -50.0  -9.3 -14.0  -8.2  -2.4  -6.3  -2.4  -3.9\n",
      "TAA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0  30.6 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAC -14.8  -6.2 -14.9  -7.6 -14.6 -11.6 -13.0 -12.2 -14.0  -9.0 -14.1 -10.1 -13.6 -11.2 -12.4 -12.5  -8.1   2.2  -8.7   0.4 -15.3 -11.3 -13.2 -13.3 -11.9  -7.2 -11.8  -7.9 -10.7  -7.4 -10.8  -8.9 -15.5 -10.3 -15.5 -12.5 -14.5 -12.1 -11.9 -12.8 -18.3 -12.7 -16.4 -14.7 -14.2 -10.4 -13.7 -13.0 -50.0  15.1 -50.0  13.3  -9.9  -5.8  -8.9  -7.6 -50.0  -1.5  -7.8  -3.0  -8.1   3.6  -9.4   2.0\n",
      "TAG -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  30.6 -50.0  35.2 -50.0 -50.0 -50.0 -50.0 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAT -13.8  -7.5 -14.4  -5.1 -12.6 -11.6 -11.5 -10.1 -13.2  -9.7 -12.7  -7.4 -11.9 -11.9 -11.5 -10.2  -7.2   0.9  -8.6   2.6 -14.1 -12.8 -13.3 -11.0  -9.7  -8.4 -11.2  -5.3  -9.3  -8.7  -9.9  -7.1 -13.9 -11.2 -14.2  -8.9 -12.9 -12.6 -12.6 -10.9 -15.4 -13.8 -14.8 -11.4 -12.8 -11.6 -13.2 -10.3 -50.0  13.3 -50.0  15.2  -8.6  -7.0  -8.4  -4.8 -50.0  -3.0  -7.1  -0.1  -7.3   2.3  -8.7   3.9\n",
      "TCA  -7.3  -6.3  -8.9  -5.5   2.9   0.2   1.5   1.2  -8.5  -2.3 -10.0  -1.7  -5.6 -10.5  -5.5  -9.6  -4.3  -7.9  -6.1  -7.2   2.3  -0.6   0.5  -0.0  -8.9 -11.0  -9.4  -9.3  -4.6  -9.4  -7.3  -8.1  -7.8  -9.8  -8.9  -9.2   2.8   0.1   1.2   0.8  -5.9  -6.8  -5.9  -6.3  -4.1  -7.2  -6.1  -6.2 -50.0  -9.9 -50.0  -8.6  12.5   9.4  11.0   9.8 -50.0  -4.8  -8.7  -3.5  -0.0  -8.6  -2.1  -7.0\n",
      "TCC -10.1  -4.3 -10.8  -5.8   0.1   2.3   0.7   0.6 -10.4  -0.2 -10.8  -2.0  -8.8  -9.2  -8.0  -9.7  -6.8  -5.9  -7.5  -7.0  -1.4   2.2  -0.9  -0.4 -10.8  -7.7  -9.8  -8.8  -8.9  -7.3  -9.6  -9.1 -10.5  -8.3 -10.2 -10.0  -0.0   2.5   0.7   0.5  -7.5  -5.1  -7.1  -6.7  -7.4  -5.7  -8.1  -7.0 -50.0  -5.8 -50.0  -7.0   9.4  12.7  10.1  10.0 -50.0  -1.5 -11.3  -3.3  -5.4  -4.2  -5.6  -6.8\n",
      "TCG  -8.4  -6.2  -8.9  -5.9   1.8   0.4   2.7   1.2  -9.1  -1.2  -8.8  -1.5  -7.2 -10.5  -4.9  -9.6  -5.7  -7.4  -5.5  -7.4   0.3  -0.3   1.8  -0.4  -9.0  -8.7  -7.3  -8.9  -5.9  -9.4  -6.4  -8.7  -8.7  -9.1  -8.5  -9.4   1.4   0.7   3.5   0.7  -6.6  -5.3  -4.7  -5.8  -5.7  -7.5  -5.5  -7.1 -50.0  -8.9 -50.0  -8.4  11.0  10.1  13.2  10.2 -50.0  -4.2  -6.1  -3.6  -3.5  -8.4  -1.2  -6.8\n",
      "TCT  -9.1  -5.4 -10.4  -4.2   1.0   0.5   0.6   2.4  -9.6  -1.5 -10.5  -0.5  -8.9 -10.5  -7.2  -8.3  -6.3  -6.9  -7.7  -5.1  -0.5   0.1  -0.6   2.3 -11.1 -10.2 -10.5  -7.0  -8.3  -9.4  -9.4  -6.6  -9.6  -9.7 -10.0  -7.9   0.7   0.6   0.8   2.3  -7.1  -6.5  -7.2  -5.0  -6.5  -7.4  -7.5  -5.4 -50.0  -7.6 -50.0  -4.8   9.8  10.0  10.2  12.1 -50.0  -3.1 -11.4  -0.7  -4.6  -6.6  -4.9  -3.9\n",
      "TGA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TGC -13.0  -7.0 -13.1  -8.3  -9.2  -6.5  -8.9  -7.6  -9.8  -0.7  -9.2  -2.1 -12.4 -10.3 -12.4 -11.6  -9.5  -4.6 -10.4  -5.7 -13.6 -10.1 -11.4 -10.9  -6.5  -1.5  -6.2  -3.6 -12.8  -8.0 -12.3  -9.7 -16.5 -12.7 -15.6 -14.3  -8.7  -6.0  -7.1  -7.5  -9.5  -4.1  -8.4  -6.1 -11.0  -8.6 -11.3  -9.3 -50.0  -1.5 -50.0  -3.0  -4.8  -1.5  -4.2  -3.1 -50.0  16.4  -5.1  14.2 -10.1  -4.3 -10.0  -5.4\n",
      "TGG -13.5 -16.3 -11.8 -15.5 -13.6 -15.3 -12.1 -16.1  -7.7 -13.3  -4.2 -12.6 -14.1 -16.2 -10.3 -15.4  -9.6 -11.6  -7.7  -9.8 -14.3 -17.5 -11.1 -17.2  -5.3  -7.1  -2.2  -7.1  -9.6 -11.9  -8.7 -11.4 -17.1 -20.6 -15.3 -19.9 -14.1 -14.8 -12.4 -16.5 -11.7 -12.8  -6.4 -13.4 -14.3 -15.7 -12.7 -14.0 -50.0  -7.8 -50.0  -7.1  -8.7 -11.3  -6.1 -11.4 -50.0  -5.1  18.6  -4.8  -8.6  -8.1  -4.5  -7.3\n",
      "TGT -12.4  -8.2 -13.0  -5.9  -7.7  -7.8  -7.4  -5.8  -8.8  -2.0  -9.3  -0.4 -11.6 -11.3 -11.6  -9.4  -8.9  -5.6 -10.0  -3.2 -11.5 -11.1 -10.5  -8.3  -6.3  -4.9  -6.7  -1.2 -10.6  -9.7 -11.3  -8.0 -14.8 -13.1 -15.5 -11.4  -7.5  -7.5  -7.1  -5.9  -8.4  -5.8  -8.1  -3.6  -9.9  -9.7 -10.1  -8.2 -50.0  -3.0 -50.0  -0.1  -3.5  -3.3  -3.6  -0.7 -50.0  14.2  -4.8  16.4  -8.9  -5.8  -8.9  -3.4\n",
      "TTA -10.7 -13.2 -12.4 -11.6  -4.8  -8.4  -6.1  -7.2 -10.3 -11.8 -11.0 -11.6   2.8  -1.7   0.7  -1.1  -6.2  -9.7  -7.3  -8.4  -5.3  -8.6  -6.1  -8.3 -10.5 -11.2  -9.8 -10.2   9.5   6.5   7.6   6.8 -12.1 -16.0 -13.0 -16.2  -5.2  -8.6  -7.1  -7.8 -12.5 -14.4 -12.1 -13.0   1.4  -3.0  -1.6  -2.4 -50.0  -8.1 -50.0  -7.3  -0.0  -5.4  -3.5  -4.6 -50.0 -10.1  -8.6  -8.9  13.2  -0.9   9.7  -0.3\n",
      "TTC -18.1 -12.3 -19.4 -14.2 -12.2  -9.5 -11.5 -10.8 -17.9 -11.9 -18.2 -13.9  -6.4  -3.7  -6.8  -5.4 -12.8  -5.9 -13.6  -7.4 -13.7  -9.9 -12.9 -12.7 -16.2 -11.6 -15.2 -12.7  -3.2   0.5  -3.3  -1.8 -18.9 -16.6 -18.8 -19.2 -12.5  -9.4 -10.6 -11.1 -17.3 -14.1 -16.2 -16.0  -7.9  -3.7  -7.8  -6.3 -50.0   3.6 -50.0   2.3  -8.6  -4.2  -8.4  -6.6 -50.0  -4.3  -8.1  -5.8  -0.9  14.2  -1.8  11.6\n",
      "TTG -11.8 -13.1 -11.5 -12.0  -6.2  -8.3  -5.4  -7.4 -11.8 -12.3 -11.1 -11.4   0.5  -2.5   2.4  -1.6  -6.8  -9.9  -6.4  -8.9  -6.1  -8.6  -5.0  -7.8 -10.4 -11.6  -8.5  -9.9   8.2   6.4   8.7   6.9 -12.4 -17.0 -12.6 -15.8  -6.5  -8.5  -5.0  -7.8 -13.2 -13.6 -10.3 -13.7  -0.7  -3.2  -0.3  -2.4 -50.0  -9.4 -50.0  -8.7  -2.1  -5.6  -1.2  -4.9 -50.0 -10.0  -4.5  -8.9   9.7  -1.8  11.3  -0.9\n",
      "TTT -17.2 -13.3 -17.8 -11.6 -11.0 -11.1 -10.4  -9.0 -16.0 -12.7 -16.1 -11.1  -5.2  -5.4  -6.0  -3.3 -11.8  -7.3 -13.3  -5.9 -12.7 -11.9 -11.9  -9.3 -15.2 -15.0 -14.9 -11.0  -2.8  -1.8  -2.9   0.4 -17.4 -18.4 -18.3 -15.9 -10.7 -11.2 -10.6  -9.0 -15.4 -14.8 -15.8 -13.3  -6.8  -5.9  -7.2  -3.9 -50.0   2.0 -50.0   3.9  -7.0  -6.8  -6.8  -3.9 -50.0  -5.4  -7.3  -3.4  -0.3  11.6  -0.9  14.1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "matrix = substitution_matrices.load(\"SCHNEIDER\")\n",
    "print(matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligner = Align.PairwiseAligner(matrix=matrix)\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: 0.000000\n",
      "  target_internal_open_gap_score: 0.000000\n",
      "  target_internal_extend_gap_score: 0.000000\n",
      "  target_left_open_gap_score: 0.000000\n",
      "  target_left_extend_gap_score: 0.000000\n",
      "  target_right_open_gap_score: 0.000000\n",
      "  target_right_extend_gap_score: 0.000000\n",
      "  query_internal_open_gap_score: 0.000000\n",
      "  query_internal_extend_gap_score: 0.000000\n",
      "  query_left_open_gap_score: 0.000000\n",
      "  query_left_extend_gap_score: 0.000000\n",
      "  query_right_open_gap_score: 0.000000\n",
      "  query_right_extend_gap_score: 0.000000\n",
      "  mode: global\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(aligner)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "OverflowError",
     "evalue": "number of optimal alignments is larger than 9223372036854775807",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOverflowError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-e0f92bf7d593>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malignments\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/Bio/Align/__init__.py\u001b[0m in \u001b[0;36m__len__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1345\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1346\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1347\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpaths\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1348\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1349\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOverflowError\u001b[0m: number of optimal alignments is larger than 9223372036854775807"
     ]
    }
   ],
   "source": [
    "print(len(alignments))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: -2.000000\n",
      "  target_internal_open_gap_score: -2.500000\n",
      "  target_internal_extend_gap_score: -2.500000\n",
      "  target_left_open_gap_score: -2.500000\n",
      "  target_left_extend_gap_score: -2.500000\n",
      "  target_right_open_gap_score: -2.500000\n",
      "  target_right_extend_gap_score: -2.500000\n",
      "  query_internal_open_gap_score: -2.500000\n",
      "  query_internal_extend_gap_score: -2.500000\n",
      "  query_left_open_gap_score: -2.500000\n",
      "  query_left_extend_gap_score: -2.500000\n",
      "  query_right_open_gap_score: -2.500000\n",
      "  query_right_extend_gap_score: -2.500000\n",
      "  mode: global\n",
      "\n"
     ]
    }
   ],
   "source": [
    "aligner = Align.PairwiseAligner()\n",
    "aligner.match_score = 1.0\n",
    "aligner.mismatch_score = -2.0\n",
    "aligner.gap_score = -2.5\n",
    "print(aligner)\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19106.5\n"
     ]
    }
   ],
   "source": [
    "print(alignments[0].score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATATTA-GGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGACTGTCACTTAGCTGCATGCTTAGTGCACTCACGCAGTTTAATTA-TAATTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGTTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCATACCTTGGTTTCGTCCGGGTGTGACCGAGAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGGTCCA-CTCAGTTTGCCTGTTTTACAGGTTCGTGACGTGCTTGTACGTGGCTTTGGAGACTCCGTAGAGGAAG-CTTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGTGTTTTACCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCTGATGCCCGAACTGCACCTCACGGCCATGTTATGGTCGAATTAGTAGCAGAACTCGATGGCATTCAGTATGGTCGTAGTGGTGAGACACTTGGTGTTCTTGTCCCTCATGTAGGAGAGGTACCAGTTGCTTACCGTAAAGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCTTTTGACTTAGGCGACGAGCTTGGTACTGATCCT-ATTGAAGATTTTCAAGAAAATTGGAACACTAAACATGGCAGTGGTGTTACCCGTGAACTCAAGCGTGAGCTTAACGGAGGTGCATACACTCGCTATGTAGACAACAACTTTTGTGGCCCAGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTGGCTCGTGCTGGTAAGGCTTCTTATGC--TTTGTCTGAACAATTGGATTTTATTGACACTAAGAGAGGTGTGTACTGCTGCCGTGAACACGAGCATGAAATTGCTTGGTACACGGAACGCTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAGTTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCTCT-AAATTCAACAATCAAGACCATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGG-TTTCATGGGTAGAATTCGATCTGTCTATCCTGTTGCTTCACCAAATGAATGCAACCAAATGTGCCTCTCAATTCTCATGAAGTGTGACCATTGTGGTGAAACTTCATGGCAGACGGGTGATTTTGTTAGAGCCACTTGCGAATTCTGTGGTACTGAAAATTTGACTAAAGAAGGTGCCACAACTTGTGGTTACTTACCTCAAAATGCTGTTGT-AAAACTTTATTGTCCAGCATGTCATAATCCAGAAGTAGGACCTGAGCATAGTCTTGCTGAATATCATAATGAGTCTGGTTTGAAAACCGTTCTTCGTAAGGGTGGTCGTACCATTGC-TTATGGGGGCTGTGTGTT-TGCTTATGTTGGTTGCTACAACAAGTGTGCCTATTGGGTTCCACGTGCTAGTGCTAACATAGGCTGTAATCACACAGGTGTTGTTGGAGAAGGTTCTGAAAGTCTAAACGACAACCTTCTTGAAATATTGCAAAAGGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATTGCCATTATTTTGGCATCTTTTTCTGCCTCTATAAGTGCTTTTGTAGAAACTGTAAAAGGTTTGGATTACAAAACATTCAAACAAATTGTTGAATCCTGTGGTAACTTTAAAGTTACGAAGGGAAAAGC-AAAGAAAGGTGCCTGGAACATAGGTGAACA-AAGTTCAATACTGAGTCCGCTCTATGCGTTCCCTTCAGATGCTGCTCATGTTGTACGGTCTATCTTCTCACGCACACTAGAAACTGCTCACCATTCTGTGCATGTCTTACAAAAGGCTGCTATAATTAT-CTTAGATGGAATTTCACAGTATTCATTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGTTACTAACAATCTAGTTGTAATGGCTTACATTACGGGTGGTGTTGTACAAATGACTTCACAGTGGCTAACAAATATCTTTGGCACTGTTTATGAAAAACTTAAACCGGTTCTTGATTGGCTCGAAGAGAAATTCAAAGAAGGTATAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACTTGTGCTTGTGAAATAGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTCGTAAATAAATTTTTGGCTTTGTGTGCTGACTCCATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACACACTCAAAGGGATTGTACAGAAAGTGTGTTAGATCCAGAGAAGAAACTGGCTTACTCTTGCCTCTGAAAGCTCCAAAAGAAATTATTTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGTTTTACAACCATTAGAACAACCTACTAATGAGGCTGTTGAAGCTCCATTGATTGGTACACCAGTCTGTATTAACGGGCTCATGTTGCTCGAAATTAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAATAATACCTTCACACTTAAAGGTGGTGCACCAACAAAAGTCACTTTTGGTGATGACACTGTGATTGAAGTGCAGGGTTACAAGAGTGTAAACATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTAATTACACAGTTGAACTCGGTACAGAGGTAAACGAGTTCGCTTGTGTTGTTGCTGATGCTGTCATAAAAACTTTACAACCAGTATCTGAACTAATCATACCACTGGGCATTGATTTAGACGAGTGGAGTATGGCTACATACTACTTGTTTGATGAGTCCGGTGAATTTAAATTGTCTTCACATATGTACTGTTCTTTCTACCCTCCTGA--AG-ATGAAGGGGAAGATGATTGTGAAGAAGGACAGTGTGAACCATCAACTCAATATGAGTATGGTACTGAGGATGACTACCAAGGTAAACCTTTGGAGTTTGGTGCTACTTCTTTTTCTTCTTCTTCACAGGAAGAAGAACAAGAAGAGGATTGGTTAGAATC-TGATAGTCAGGA--C-G--GC-CAAGA-GACTGCAGTTGAAGAAAAT-A-A-AA-TAC----CGAG---T-GTTGAAGTTCCACCTGTTTT-GCAGGTGGAA-TCAACACCAGTTGTTACTGAAACTAGTGAACAAAATAATTTCACAGGTTATTTAAAA-TTAACTGACAATGTCTTCATTAAAAATGCTGACATTGTAGAAGAAGCTAAAAAGGTAAAGCCTACAGTAGTTGTTAATGCAGCTAATGTTTACCTTAAACATGGAGGAGGTGTTGCTGGAGCTTTAAATAAGGCAACTAACAACGCCATGCAGGTTGAATCTGATAAGTACATAACTACCAATGGGCCACT-AATTGTGGGTGGTGGTTGTGTTTTGAGTGGACATAACCTTGCTAAAAATTGTCTTCATGTTGTTGGCCCTAATGTCAACAGAGGTGAAGACATTCAA-TTGCTTAAAAATGCTTATGAAAATTTCAATCAACATGAGATTT-TACTCGCACCATTATTATCAGCTGGTATTTTTGGTGCTGATCCTGTACATTCTTTAAGAGTTTGTGTAGAAACTGTTCACACAAATGTCTATCT-AGTTGTCTTTGACAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTAGAGATGAAGAGTGAAAAACAAGTAGAACAAAAAGTT-GCTGAAAATCCTAAAGAGGAAGTTAAGCCATTTTTTACTGAAAATAAACCTTCAGTTGAACAAAGACAACAAGCTGAAGAGAAGAAAATCAAAGCCAGTATTGAAGAAGTTACAACTACTCTAGAGGAGACCAAGTTCCTTACAGAAAACTTGTTACTTTATATTGACATCAATGGCAATCTTCACCCAGATTCTGCCACTCTTGTTAAAGATATTGACACCACTTTCTTGAAGAAAGATGTTCCATATATAGTGGGTGATGTTATTAAAGAAGGCGCTTTAACTGCTGTAGTTATACCTACTAAAAAGGCTGGTGGCACTACCGAAATGCTTGCTAAAGCTTTAAGAAAAGTGCCAACAGATAATTATATAACCACCTACCCTGGTCAGGGTTTAAATGGTTACACTGTAGAAGAAGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCTTTTTACATTTTACCATCTATTATCTCTAATGCGAAGCAAGAAATTCTAGGAACTGTTTCTTGGAACTTGCGAGAGATGCTCGCACATGCAGAAGAAACACGTAAGTTAATGCCTGTTTGTATGGAGACTAAAGCTATAGTTTCAACTATACAACGTAAGTACAAAGGCATTAAAATACAGGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACTAGTAAAACTACTGTAGCATCACTTATTAACACACTTAACAATCTAAATGAGACTCTTGTCACAATGCCA-TTAGGATATGTGACGCACGGTCT-AAATTTAGAAGAAGCTGCTCGGTACATGAGGTCTCTCAAAGTACCGGCTACAGTCTCTGTTTCTTCACCAGATGCTGTTACAGCATATAATGGTTATCTTACTTCTTCTTCAAAAACACCTGAAGAACACTTTATTGAAACCGTTTCACTTGCTGGTTCCTATAAAGATTGGTCTTATTCTGGACAGTCTACACAACTAGGCATAGAATTTCTTAAGAGAGGTGATAAGAGTGTATATTACACCAGTAATCCCACTACATTTCACCTAGATGGTGAAACTATCACCTTTGATAATCTTAAGACACTTCTCTCTTTGAGAGAAGTGAGGAATATTAAAGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGATATGTCTATGACATATGGACAACAGTTCGGTCCAATTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCACAATTCACATGAAGGTAAAACATTTTATGTTTTGCCTAATGATGATAC-CTTACGTGCAGAGGCTTTTGAGTACTACCATACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCTTTAAACCACACTAAAAAGTGGAAATACCCACAAGTAAATGGTTTAACTTCTATAAAATGGGCAGATAACAATTGTTATCTTGCTACTGCATTGTTAACAATCCAACAAATAGAGTTGAAATTTAATCCACCAGCTTTACAAGACGCCTATTATAGGGCAAGAGCTGGTGAGGCTGCTAATTTTTGTGCACTTATCCTGGCCTATTGTAATAAGACAGTAGGTGAGTTAGGTGATGTCAGAGAAACAATGAATTATTTGTTTCAACATGCCAATTTAGATTCTTGTAAAAGAGTCTTGAATGTGGTGTGTAAAACTTGTGGACAACAGCAAACAACTCTCAAGGGTGTAGAAGCTGTTATGTATATGGGCACACTTTCTTATGAACAACTTAAGAAGGGTGTGCAGATACCTTGTATGTGTGGTAAACAAGCTACACAATATCTGGTACAACAAGAGTCACCTTTTGTTATGATGTCTGCACCGCCCGCCCAATATGAACTTAAGCATGGTACATTTGTTTGTGCTAGTGAGTATACTGGTAATTACCAGTGTGGTCACTACAAACATATAACTTCTAAAGAAACCTTGTATTGCATAGATGGCGCTTTACTTACAAAGTCCTCTGAGTATAAAGGTTCTATTACAGATGTTTTCTATAAAGAAAACAGTTATACAACAACCATAAAACCAGTTACATACAAGTTGGATGGTGTTGTTTGTACAGAAATTGATCCTAAGTTGGATGGTTATTATAAGAAAGATAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCTTACCCGAATGCAAGCTTTGACAATTTTAAGTTTGTATGTGATAATACCAAATTTGCCGATGATTTAAATCAATTGTCTGGTTATAAGAAACCTGCCTCGAGAGAGCTTAAGGTTACATTCTTTCC-TGACTTAAATGGTGATGTAGTGGCTATTGATTATAAGCACTACACACCTTCTTTTAAGAAAGGAGCTAAATTGCTGCATAAGCCAATTGTTTGGCATGTTAACAATGCAACTAACAAAGCAACGTACAAACCAAATATTTGGTGCATACGTTGTCTTTGGAGTACAAAACCGGTTGAAACATCAAATTCTTTTGATGCACTGGAATT-AGGGGACACACAGGGAATGGATAATCTTGCCTGTGAAGTTCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACATTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCGGCAAGTGATGGTCTAAAAATTACAAAAGAGGTTGGTCATACAGACCTAATGGCTGCTTATGTTGACAATTCAAGTCTTACTATTAAGAAACCTAATGAATTATCCAGAGTATTAGGTTTGAAAACT-TTAGCCACTCATGGCTTGGCTGCTATTAATAGTGTTCCTTGGGACACTATAGCTAATTATGTTAAGCCTTTCCTTAATAAGGTTGTTAGCACAACTACTAACATAGTTACACGGTGTCTAAACCGTGTTTGTACTAATTATATGCCTTATTTGTTTACTTTATTGCTACAATTGTGTACTTTTACTAAAAGTACAAATTCTAGAAT-AAGAGCATCTATGCCAACCACTATAGCAAAGAATACTGTTAAAAGTGTTGGTAAATTTTGTATAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAATTGATAAATATTGTAATTTGGTTTTTATTATTAAGTGTTTGCCTAGGTTCTTTAATCTATTCAACTGCTGCTTTAGGTGTCTTAATGTCTAATTTAGGCATGCCTTCTTATTGTACTGTTTACAGAGAAGGTTATTTGAACTCTACTAATGTCACTACTGCAACCTACTGCACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTGGATACTTACCCATCCTTAGAAACTATACAAATTACCATTTCGTCTTTTAAATGGGATTTAACTGCTTTTGGTCT-AGTTGCAGAGTGGTTTTTGGCATATATTCTTTTTACTAGGTTCTTT-TATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCAATTTCAGCTATGGTTAGAATGTATATTTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGATGGTTGTACTTCATCAACTTGTATGATGTGTTATAAACGTAATAGAGCAACAAGAGTTGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACATAACTGGAATTGTATTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGACGAAGTTGCTAGAGACTTATCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCCTATATTGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAGGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAATCTGAGAGCTAATAACACTAAGGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCTAAATCAGCGTCTGTTTATTATAGTCAGCTTATGTGTCAACCTATACTGTTACTGGACCAGGCATTAGTGTCTGATGTTGGTGACAGTGCAGAAGTTGCAGTTAAGATGTTTGATGCTTATGTTAATATATTTTCATCAACTTTCAATGTTCCAATGGAAAAACTCAAAGCATTAGTTGCGACTGCAGAAGCTGAACTTGCAAAGAATGTGTCTTTAGACAACGTCTTATCTACTTTTATTTCAGCAGCTCGTCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACACCAATCTGACATAGAAGTTACAGGTGACAGTTGTAACAATTACATGCTCACTTATAACAAAGTTGAAAACATGACACCTCGGGATCTTGGTGCTTGTATTGATTGTAGTGCACGTCATATCAATGCACAGGTGGCAAAAAGTCATAACATAGCTTTGATTTGGAATGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGCAGTGCTGCTAAGAAGAATAACTTGCCTTTTAGA-TTGACATGTGCAACCACTAGACAAGTTGTTAATGTTGTTACAACAAAAATAGCACTTAAGGGTGGTAAAATTGTTAACAACTGGTTGAAGCAGCTGATTAAGGTTACACTTGTGTTTCTTTTCATCACTGTTATCTTCTATTTAATAACACCTGTTCATGTCATGTTTAAACACAATGACTTTTCAAGTGAAATTATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCGTCAACAGATACTTGTTTTGCTAACAAACATGCTGACTTTGACTCTTGGTTTAGTCAGCGCGGTGGTAGTTATACTAATGATAAAGCTTGCCCATTGGTAGCAGCTGTTATCACAAGGGAAGTTGGCTTTGTTGTGCCCGGTTTGCCTGGCACAATATTACGCACAATTAATGGTGATTTTTTGCATTTTCTT-CCTAGAGTGTTTAGTGCGGTAGGTAACATTTGTTACACTCCTTCTAAACTTATAGAGTACACTGACTTTGCAACATCGGCATGCGTTTTAGCTGCTGAATGTACCATTTTTAAAGATGCTTCTGGTAAACCAGTACCTTATTGTTATGATACTAATGTACTAGAAGGTTCTGTTGCGTATGAAAGTCTCCGCCCTGACACACGCTATGTGCTCATGGACGGTTCTATAATTCAATTCCCTAACACTTACCTTGAAGGTTCTGTTAGAGTAGTAACAACTTTTGATTCAGAGTATTGTAGACATGGTACTTGCGAAAGATCAGAGGCTGGCATTTGTGTATCTACTAGTGGTAGATGGGTACTTAATAATGATTATTACAGATCCTTGCCAGGAGTTTTTTGTGGTGTAGATGCTGTGAATTTACTTACTAATATGTTCACGCCATTAATTCAACCTATTGGTGCTTTGGACATATCTGCATCTATTGTAGCAGGTGGTGTTGTAGCTATTATAGTAACTTGTCTAGCCTACTACTTCATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAACACTCTACTATTCTTTATGTCATTCACTGTACTCTGTTTAACACCAGTCTATTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTCTTAGCACATATCCAATGGATGGTTATGTTCACACCCTTAGTGCCTTTCTGGATGACAATTGTTTATGTCATTTGCATTTCCACAAAGCATTTTTATTGGTTCTTTAGTAACTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACATTTGAGGAGGCTGCATTATGTACCTTTTTGTTAAATAAAGAAATGTATCTGAAA-TTGCGTAGTGATGTACT-TCTACCTCTTACGCAATACAATAGATATTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGGGCCATGGACACTACCAGTTATAGAGAAGCAGCTTGCTGTCATCTGGCTAAGGCTATAAATGATTTCAGTAATTCAGGTTCTGATGTCCTCTACCAACCACCACAAACTTCAATCACATCAGCGGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAAGGTTGCATGGTACAAGTTACTTGTGGTACCACTACACTTAATGGTCTTTGGCTTGATGATGTAGTTTACTGTCCACGACATGTGATCTGCACTTCTGAAGACATGCTCAATCCTAATTATGAAGATTTACTTATACGTAAATCTAACCATAATTTTTTAGTTCAGGCTGGTAATGTTCAACTTAGAGTTGTTGGACATTCTATGCAAAATTGTGTTCTTAAGCTTAAAGTAGATACAGCTAATCCTAAGACACCTAAGTATAAGTTTGTGCGCATTCAACCCGGACAGACTTTTTCAGTATTAGCCTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCCATGAGACCTAATTTTACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAATATAGACTATGACTGTGTCTCTTTTTGTTATATGCATCATATGGAGTTACCAACGGGAGTTCATGCTGGCACAGACTTAGAAGGTACCTTCTACGGACCTTTTGTTGACAGACAGACAGCACAAGCGGCTGGTACTGACACAACTATTACAGTTAATGTTCTAGCTTGGTTGTATGCAGCTGTTATAAACGGAGATAGATGGTTCCTTAATAGGTTTACCACAACTCTAAACGATTTTAATCTTGTGGCTATGAAGTATAATTATGAACCTCTAACACAAGACCATCTTGACATACTAGGACCTCTTTCAGCTCAAACTGGAATTGCAGTCCTAGATATGTGTGCTTCATTAAAAGAATTATTACAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTCGATGTTGTTAGACAATGTTCAGGTGTCACCTTTC-AAAGTGCAGTGAAAAGGACAATCAAGGGCACGCACCATTGGTTGTTGCTTACAGTTTTGACTTCACTCTTAGTTTTAGTTCAGAGTACTCAATGGTCTTTGTTCTTCTTTGTGTATGAAAATGCCTTTATGCCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCTATGATGTTTGTCAAACATAAGCATGCATTCCTCTGTTTGTTCCTGTTACCTTCTCTTGCTACTGTAGCTTATTTTAATATGGTCTACATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATAGTTGATACTAGTTTGTCTGGTTTCAAGCTAAAGGACTGTGTTATGTATGCATCAGCTGTAGTGTTATTAATCCTCATGACAGCAAGAACCGTATATGATGATGGTGCTAGAAGAGTTTGGACACTTATGAATGTCCTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGACCAAGCTATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGTCCTATCTTCTTTATAACTGGCAATACACTCCAGTGTATAATGCTAGTTTATTGTTTCTTAGGTTATTTCTGTACTTGTTATTTTGGTCTCTTCTGTTTACTCAACCGCTATTTTAGGCTTACTCTTGGTGTTTATGACTATTTAGTCTCTACACAAGAATTTAGGTACATGAACTCTCAGGGGCTCCTGCCACCTAAGAGTAGTATTGACGCTTTCAAGCTTAACATTAAATTGTTGGGCATTGGAGGTAAACCTTGTATTAAGGTTGCTACTGTACAGTCTAAAATGTCTGACGTAAAGTGCACATCAGTAGTGCTT-CTCTCAGTTCTT-CAGCAACTTAGAGTAGAGTCATCTTCTAAATTGTGGGCACAGTGTGTACAACTT-CACAATGATATTCTTCTT-GCCAAGGACACTACTGAAGCTTTTGAGAAGATGGTTTCACTT-TTGTCTGTTTTGCTGTCCATGCAGGGTGCTGTAGACATTAACAAG-TTGTGCGAGGAAATGCTCGACAACCGTGCTACCCTT-CAGGCTATTGCTTCAGAATTCAGTTCT-TTACCTTCATATGCTGCTTATGCCACTGCTCAAGAGGCTTATGAGCAGGCTGTAGCAAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTAAAGAAATCTTTGAATGTGGCTAAATCTGAGTTTGACCGTGATGCTGCCATGCAACGTAAGTTGGAAAAGATGGCGGATCAGGCTATGACCCAAATGTACAAGCAGGCAAGATCTGAGGACAAGAGGGCAAAAGTAACTAGTGCAATGCAAACTATGCTTTTCACTATGCTTAGAAAACTTG-ATAATGATGCACTTAACAACATTATCAACAATGCACGTGATGGTTGTGTACCACTC-AACATCATACCACTCACAACAGCAGCAAAACTCATGGTTGTTGTCCCTGACTATGGAAC-C-TACAAGAATACTTGTGATGGTAACA-CTTTTACATATGCATCAGCACTCTGGGAAATCCAGCAAGTTGTTGATGCAGATAGTAAAATTGTGCAGCTTAGTGAAATCAACATGGACAACTCACCAAATTTGGCTTGGCCTCTTATTGTTACTGCTTTAAGAGCCAATTCAGCTGTCAAACTACAGAACAATGAGCTGAGTCCAGTAGCACTACGACAGATGTCCTGTGCGGCTGGTACTACACAGACAGCTTGTACTGATGACAACGCACTT-GCCTACTATAACA-ATTCTAAGGGAGGTAGGTTTGTGCTTGCATTACTATCAGACC-ACCAGGATCTCAAATGGGCTAGATTCCCTAAGAGTGATGGTACAGGTACTATCTATACAGAATTGGAACCACCTTGCAGGTTTGTTACAGACACACCTAAAGGACCTAAAGTGAAGTATTTGTACTTTATTAAGGGTCTTAA-CAACCTAAATAGAGGTATGGTACTGGGTAGTTTGGCTGCCACAGTACGTCTTCAGGCTGGTAATGCGACAGAAGTGCCTGCCAATTCAACTGTGCTT-TCTTTTTGTGCTTTTGCTGTAGACCCAGCTAAAGCTTACAAAGATTACCTAGCAAGTGGTGGACAACCAATCACGAATTGTGTGAAGATGTTGTGTACACACACTGGTACAGGACAGGCAATCACGGTAACACCAGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCTTCATGCTGTCTGTATTGTAGATGCCACATTGATCATCCAAATCCTAAGGGATTTTGTGACTTGAAAGGTAAGTATGTTCAAATACCTACCACTTGTGTTAATGACCCTGTGGGGTTTACACTCAGAAACACAGTCTGTACCGTCTGCGGAATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGATGCAGTCTGCGGATGCGTCAA-CGTTTTTAAACGGGTTTGCGGTGTAAGTGCGGCCCGTCTTACACCGTGCGGCACAGGCACTAGCACTGATGTCGTTTATAGGGCTTTTGATATTTACAACGAGAAAGTTGCTGGTTTTGC-AAAGTTCCTAAAAACTAATTGCTGTCGCTTCCAAGAAAAGGATGAGGAAGGCAATTTA-TTAGACTCTTA-TTTCGTAGTTAAGAGGCACACAATGTCCAACTACCAACATGAAGAGACTATTTACAACTTGGTTAAAGGGT-GTCCAGCCGTTGCTGTTCATGACTT-TTTCAAGTTTAGAGTAGATGGTGACATGGTACCACATATATCACGTCAACGTCTAACTAAATACACAATGGCTGATTTAGTCTATGCTCTACGTCATTTTGACGAGGGCAATTGTGATACATTAAAGGAAATACTTGTCACATACAAATGTTGTGATGACAACTATTTCAATAAGAAGGATTGGTATGACTTTGTAGAAAATCCTGATGT-TCTACGCGTATACGCGAACCTAGGTGAGCGTGTACGTCAAGCCTTATTGAAAACTGTGCAATTCTGCGATGCTATGCGCGATGCGGGTATCGTAGGTGTACTGACACTAGACAATCAGGATCTGAATGGGAATTGGTACGACTTCGGAGATTTCGTACAGGTAGCAC-C-AGGCT-GCGGAGTCCCTATTGTGGATTCATACTATTCTTTGCTGATGCCTAT-TCTGACACTC-ACAAGGGCTTTGGCTGCTGAGTCCCATATGGACGCTGATCTCG-CGAAGCCACTTA--TTAAGTGGGATTTGCTGAAATATGATTTCACGGAAGAGAGACTATGTCT-TTTCGACCGTTATTTTAAATATTGGGACCAGACGTACCATCCTAATTGTATTAATTGTTTGGATGACAGGTGTATCCTTCATTGTGCAAACTTTAATGTGTTATTTTCTACTGTGTTTCCACCTACGAGTTTTGGACCACTAGTAAGAAAGATATTTGTAGATGGTGTACCTTTTGTTGTTTCAACGGGATACCATTTCCGTGAGCTAGGGGTTGTACATAATCAGGATGTAAACTTACATAACTCGCGTCTCAGTTTTAAGGAACTT--TTAGTGTACGCTGCTGATCCAGCTATGCATGCTGCCTCTGGCAATTTGTTGTTAGACAAACGCACTACATGCTTTTCAGTAGCTGCACTAACAAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAATAAAGACTTTTATGACTTTGCTGTGTCTAAAGGCTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGCAATGCTGCTATCAGTGATTATGACTATTACCGTTATAATCTGCCAACAATGTGTGATATTAGACAACTCCTATTCGTAGTTGAGGTTGTCGATAAATATTTTGATTGTTACGACGGTGGCTGTATCAATGCCAACCAAGTTATCGTTAACAATCTGGACAAATCAGCCGGTTTCCCATTTAATAAGTGGGGTAAGGCTAGACTTTATTATGACTCAATGAGTTATGAGGATCAAGATGCACTGTTCGCATACACTAAGCGTAACGTCATCCCTACAATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACTGTTGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAGAAATTATT-AAAGTCAATAGCCGCCACTAGAGGAGCTACTGTGGTAATTGGAACAAGCAAATTTTATGGTGGCTGGCATAACATGTTAAAAACTGTTTACAGTGATGTAGAAAGTCCTCACCTTATGGGTTGGGACTACCCAAAATGTGACAGAGCCATGCCTAATATGCTTAGAATCATGGCTTCCCTCGTTCTTGCTCGCAAACATAGCA-CTTGTTGTAACTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCACAAGTATTAAGTGAGATGGTCATGTGTGGCGGCTCATTATATGTGAAACCAGGTGGAACATCATCCGGTGATGCCACAACTGCTTATGCTAATAGTGTGTTTAACATCTGTCAAGCAGTAACAGCTAATGTAAATGCACTTCTT-TCAACTGATGGTAATAAGATTGCTGATAAGTATGTCCGCAACCTT-CAACACAGACTTTACGAGTGTCTCTATAGAAATAGAGACGTAGAT-CAGGAATTCGTGGATGAATTTTATGCATATTTGCGTAAGCATTTCTCCATGATGATTCTTTCTGATGATGCCGTCGTATGCTAT-AATAGTAATTACGCGGCACAGGGTCTAGTAGCTAGCATTAAGAACTTTAAAG-CAGTTCTTTATTACCAAAATAATGTATTCATGTCTGAGGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGTTCTCAGCATACCATGCTAGTTAAACAGGGAGATGATTATGTGTACCTGCCTTATCCAGATCCATCCAGAATATTAGGCGCAGGCTGTTTTGTCGATGACATTGTCAAAACAGATGGTACACTTATGATTGAAAGGTTTGTGTCATTAGCAATTGACGCCTATCCACTTAC-AAAGCACCCTAATCAAGAGTATGCTGATGT-TTTCCATTTATACTTACAGTATATTAGGAAAT-TACATGATGAGCTTA-CTGGTCACATGTTGGACATGTACTCTGTAATGCTAACTAATGACAACACCTCGAGGTACTGGGAACCTGAGTTTTATGAGGCAATGTACACACCACACACAGTTTTGCAAGCTGTAGGTGCGTGTGTG-TTATGCAATTCACAGACTTCACTTC-GTTGCGGTGCTTGTATTAGG-AGACCTTTCCTT-TGTTGCAAGTGCTGCTATGATCACGTCATCTCAACATCACATAAATTAGTGTTGTCTGTTAATCCCTATGTTTGCAATGCACCCGGTTGTGATGTCACAGACGTAACACAACTCTATTTGGGAGGTATGAGCTATTACTGCAAGTCACATAAACCACCCATTAGTTTTCCGTTGTGTGCTAATGGTCAGGTTTTTGGTTTGTACAAAAACACATGTGTGGGCAGTGATAACGTAACTGACTTCAATGCAATAGCGACATGCGACTGGACTAATGCTGGCGATTACATACTT-GCCAACACTTGCACAGAGAGACTCAAACTTTTTGCAGCGGAAACGCTCAAAGCTACTGAGGAAACATTCAAACTATCTTATGGTATTGCCACTGTTCGTGAAGTACTGTCAGATAGAGAACTT-CATCTTTCATGGGAGGTAGGAAAACCTAGACCACCA-TTGAATAGAAATTACGTCTTTACTGGTTACCGTGTGACCAAAAATAGTAAAGTACAGATTGGAGAGTATACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTGTACAGAGGTACTACAACTTATAAATTGAATGTTGGTGATTACTTTGTGTTAACATCACACACAGTAATGCCACTAAGTGCACCAACACTAGTGCCACAAGAGCACTATGTGCGAATAACTGGCTTATACCCTACACTTAATATCTCTGATGAGTTTTCTAGCAATGTTGCAAATTACCAAAAGGTCGGCATGCAGAAGTACTCCACACTCCAGGGACCACCTGGTACTGGTAAGAGTCACTTTGCTATTGGACTTGCCCTCTACTACCCATCTGCCCGCATAGTGTATACAGCTTGCTCTCATGCTGCTGTTGATGCGCTATGTGAGAAGGCATTAAAATACTTGCCTATAGATAAGTGTAGTAGAATTATTCCTGCACGTGCGCGTGTAGAGTGTTTTGACAAATTCAAAGTGAATTCAACCTTAGAACAGTATGT-TTTCTGCACTGTCAATGCGCTGCCTGAAACTACTGCTGATATAGTGGTCTTTGATGAAATTTCAATGGCCACTAATTATGATTTGAGTGTTGTCAATGCCAGACTACGTGCAAAACACTACGTTTACATTGGTGATCCGGCTCAATTACCTGCACCACGCACATTGCTAACAAAAGGCACACTTGAACCAGAATATTTCAATTCAGTGTGCAGACTTATGAAAACAATAGGTCCAGACATGTTCCTTGGGACTTGTCGTCGTTGTCCCGCTGAAATTGTTGACACAGTGAGTGCTTTAGTTTATGATAATAAGCTAAAAGCACACAAAGAGAAGTCAGCTCAATGCTTCAAGATGTTTTACAAGGGTGTGATTACGCATGATGTGTCATCCGCAATCAACAGACCACAAATAGGTGTAGTAAGAGAATTTCTTACGCGCAATCCAGCTTGGAGAAAAGCTGTTTTTATTTCACCATATAATTCACAGAATGCAGTGGCATCAAAGATTTTAGGGT-TGCCTACTCAAACTGTTGATTCTTCACAGGGTTCTGAATATGACTATGTCATATTCACACAAACCACTGAGACTGCACACTCTTGTAATGTAAACCGCTTTAATGTGGCCATTACAAGAGCAAAGATT-GGCATT-TTGTGCATAATGTCTGATAGAGACCTTTATGACAAGCTT-CAATTTATGAGTCTAGAAGTACCGCGTCGAAATGTGGCTACTTTACAAGCAGAAAATGTGACTGGACTCTTTAAGGACTGTAGCAAGATCATTACTGGTCTT-CATCCAACACAGGCACCTACGCATCTCAGTGTTGATACTAAATTCAAGACTGAAGGACTT-TGTGTCGACATACCAGGAATACCAAAGGACATGACCTATCGTAGACTCATCTCTATGATGGGCTTTAAAATGAATTACCAAGTTAATGGTTACCCTAATATGTTTATCACCCGTGAAGAAGCTATTCGTCACGTTCGTGCATGGATAGGCTTTGATGTTGAGGGTTGTCATGCGACTAGAGATGCTGTAGGAACAAATCTACCACTCCAGTTAGGGTTTTCAACAGGTGTTAACCTAGTGGCTGTACCAACTGGCTATGTTGA--CAC-TGAGCACAGCACAGAATTTACCAGAGTTAATGCAAAACCTCCTCCAGGTGATCAATTTAAGCATCTTATACCACTTATGTACAAAGG-CTTGCCCTGGAACGTGGTGCGTATTAAGATTGTTCAAATGCTCAGTGATACACTGAAAGGAT-TATCAGACAGAGTTGTGTTTGTCCTT-TGGGCACATGGCTTTGAACTT-ACATCGATGAAGTATTTTGTTAAGATCGGACCAGAAAGAACGTGTTGTCTGTGTGACAAACGCGCGACTTGCTTCTCTACTTCATCTGACACTTATGCCTGTTGGAATCACTCTGTGGGCTTTGACTATGTCTATAACCCGTTTATGATTGATGTCCAGCAGTGGGGTTTTACAGGTAACCTTCAAAGTAACCATGATCAACACTGCCAAGTGCATGGTAATGCCCATGTAGCTAGTTGTGATGCTATCATGACTAGATGTCTTGCAGTCCATGAGTGCTTTGTTAAGCGCGTTGATTGGTCTGTTGAATACCCGATTATTGGAGATGAACTGAAGATTAATGCCGCATGCAGAAAAGTACAGCATATGGTTGTTAAATCTGCATTGCTT-GCTGATAAATTCCCAGTTCTTCATGATATAGGAAACCC-AAAGGCTATTAGA-TGTGTGCCGCA-GTCTGAAGTGGACTGGAAATTCTACGACGCTCAGCCTTGCAGTGACAAAGCTTATAAAATAGAAGAACTCTTCTACTCATATGCCACACAT-CATGACAAGTTCACAGATGGTGTTTGCTTGTTTTGGAACTGTAACGTTGATCGTTACCCGGCTAATGCTATTGTGTGTAGGTTTGATACTAGAGTGCTTTCTAAT-TTAAACCTACCAGGTTGTGATGGTGGTAGTTTGTATGTTAATAAGCATGCGTTCCACACTCCAGCTTTTGATAAGAGTGCATTTAC-ACATTTGAAACAACTGCCTTTCTTT-TATTACTCTGACAGTCCGTGTGAGTCTCATGGTAAACAGGTTGTGTCAGATATTGATTATGTCCCACTAAAGTCTGCTACGTGTATTACACGATGCAACTTAGGTGGTGCCGTTTGTAGACATCATGCAAACGAGTACAGACAGTA-CTTGGATGCATATAATATGATGATTTCTGCTGGATTTAGCCTT-TGGATTTATAAACAATTTGATACTTACAACTTGTGGAACACTTTCACCAAG--TTGCAGAGTTTAGAAAATGTGGCTTATAATGTTATCAACAAGGGACACTTTGATGGACAGA-ATGGTGAAGCACCTGTGTCTATCGTTAATAATGCTGTTTACACTAA-GTTAGATGGTGTTGATGTGGAGATC-TTTGAAAATAAGACAACACTT-CCTGTTAATGTTGCATTTGAGCTTTGGGCTAAACGTAACATTAAACCGGTGCCAGAGATTAAAATACTCAATAATTTGGGTGTTGATATCGCTGCTAATACTGTTATATGGGACTACAAGAGAGAAGCGCCAGCGCATGTTTCTACAATAGGTGTCTGTACAATGACTGACATTGCAAAGAAACCTACTGAGAGTGCTT-GTTCATCACTTACTGTCTTATTTGATGGTAGAGTTGAGGGACAGGTAGACCTT-TTTAGAAACGCCCGTAATGGTGTT-TTAATAACAGAAGGTTCAGTTAAGGGCTTAACA-CCTTC-GAAAGGACCTGCACAGGCTAGTGTCAACGGAGTCACATTAATTGGAGAATCAGTAAAAACACAGTTCAATTACTT-TAAGAAAGTGGATGGCATTATTCAGCAATTGCCAGAAACCTACTTTACTCAAAGCAGAGACTTAGAGGA-TTTCAAGCCCAG-A-TCACAAATGGAAACTGATTTCCTT-GAGCTCGCTATGGATGAATTCATAGAACGATATAAGCTAGATGGCTATGCTTTCGAGCACATCGTTTATGGAGATTTTAGTCATGGACAATTAGGCGGACTT-CATTTATTGATAGGACTGGCCAAAAGGTC-ACAGGACTCAC-TGTTAAAGCTAGAGGATTTTATTCCTATGGATAGCACAGTGAAAAACTACTTCATAACAGATGCGCAAACGGGTTCATCTAAGTGTGTATGCTCTGTTATCGACCTT-TTACTTGATGACTTTGTTGAAATAATAAAGTCACAAGATCTT-TCAGTGGTTTCAAAAGTAGTCAAAGTTACGATTGATTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGGCATGTGGAAACTTTTTACCCAAAATTACAATCTAGTCAAGCATGGCAACCAGGTGTTGCTATGCCTAATCTCTATAAAATGCAGAGAATGTTACTGGAAAAGTGTGATCTTCAAAATTATGGTGATAGTGCTATATTGCCTAAAGGCATAATGATGAATGTCGCAAAGTACACTCAACTGTGTCAGTATTTAAATACACTTA-CTTTAGCTGTGCCCTATAATATGAGAGTTATACATTTTGGCGCGGGCTCTGATAAAGGAGTAGCACCTGGCACAGCTGTTCTTA-GACAGTGGTTGCCAACTGGTACACTACTTGTCGATTCTGAT-TTAAATGATTTTGTTTCTGATGCAGACTCAACATTAATTGGTCATTGTGCAACCGTACATACGGCTAATAAATGGGATCTCATTATTAGCGATATGTATGATCCTAAGACTAAAAATGTTACAAAAGAGAATGATTCCAAAGAAGGATTTTTCACTTACATTTGTGGATTTATACAGCAAAAATTAGCCCTCGGAGGTTCCGTAGCTGTAAAGATAACAGAGCATTCCTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCTTTTGTTACTAATGTTAATGCTTCATCTTCAGAGGC-TTTCTTAATTGGTTGTAATTATCTTGGCAAACCACGTGAGCAGATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAACACAAATCCAATACAATTGTCTTCCTACTCATTATTTGACATGAGTAAGTTTCCTCTTAAATTAAGAGGTACTGCTGTTATGTCATTAAAAGATGGACAAATCAATGATATGATTTTGTCTCTTCTTAGTAAAGGCAGACTTATTGTTAGAGAGAATAATAGAGTTATTATCTCTAGTGATGTTCTTGTTAACAACTAAACGAACA-TGTT-GTTTTTCTTGTTTCTTCAGTTCGCCT-TAGTAAACTCC-CAGTGTGTTAA-CTTGACAGGCAGAACCCCACTCAATCCCA--ATTATACTAATTCTT-CACAAAGAGGTGTTTATTACCCTGACACAA-TTTAT-AGATCAGACACACTTGTGC-T-CAGC-CAGGGTTATTTTCTT-CCATT-TTATTCTAATGTTAGCT-GGT---AT--TACT-CAT-TAACAACCAACAATGCTGCCACAAAGAGGACTGATAATCCTATATTAG-ATTTCAAGGACGGCATATACTTTGCTGCCACTGA-ACACTCAAATATTATCAGGGGCTGGATCTTTGGAACAACTCTT-GA--C-AACACTTCTCAATCTCT-CTTGATAGTTAACAACGCAACGAATGTTATTATCAAGGTTTGT-AATTTTGATTTTTGTTATGATCCCTACCTTAG-TGGTTACTATCATAACAACAA-AACA-T-GGA-GCATCAGAGAATTTGCTGTCTATTCTTCTTA-TGC---TAATTGTACTTTTGAGTATGTTTCGAAATCCTTTATGTT--GAACATTTCTGGTAATG-GTGGTC-TGTTCAACACTCTTAGAGAGTTTGT-TTTCAGAA-ATGTCGATGGGCATTTCAAGATTTACTCAAAGTTTACACC-AGTAAATTTAAAT-CGTGG-CTTGCCTACTGG-TCTCTCAGTGCTTCAG--CCATTGGTTGAATTACCAGTTAGC-ATAAATATTACTAAATT-CAGAACACTCCTCACTATT-CATAGAGG--A---GAC-CCTA-TGCCTAATAA-C-GGCTGGACTGCTTTTTCAGCTGCTTATT-TCGTGGGCTATCTTAAACC-ACGTACCTTTATGCTGAAATATAATGAGAATGGCACCATTACTGATGCTGTTGATTGTGCACTTGACCCTCTTTCGGAGACAAAGTGTACGTTAAAATCTCTT-ACTGTCCAAAAGGGCATCTATCAGACTTCTAACTTCCGAGTGCAACCCACTCAGTCTGTAGTTAGATTTCCTAATATTACCAA-TGTGTGTCCATTTCAC-AAGGTTTTTAATGCCACGAGGTTTCCTTCCGTCTATGCGTGGGAA-AG-AACTAAAATTTCTG-AT-TGCATTGCAGATTACACTGTTTTCTACAATTCAACTTCTTTTTCTACTTTTAAATGTTATGGTGTTTCACCTTCTAAATTGATTGATTTGTGCTTTACGAGTGTGTATGCTGATACATTTCTCATAAGATTCTCA-GAAGTCAGACAGGTGGCACCAGGACAAACTGGTGTCATTGCTGACTATAATTATAAATTACCTGATGATTTTACAGGTTGTGTCATAGCTTGGAACACTGCCAA-----A--C-A-GGATG-TAGGTAATTATT-TCTACAGGTCTC-ATCGTTCTACCAA----A-T-TGAAACCATTTGAAAGAGATCTTTC--CTCAGA-CGAGAATGGT-G-TCCGTACAC-TT--A--G-T----AC--TT--A-T-G--ACTT-CA---ACCCT-A-ATG-T----A-CC-ACT--TGAA-T-----ACCAAGC-TACAAGGGTTGTTGTT-TTGTCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACC-AAAACTA-TCCACACAACTAG-TAAAAAATCAG-TGCGTTAATTTCAACTTTAACGGACTCA-AGGGCACTGGTGT-CTTGACTGATTCTTCCAAGAGGTTTCAGTCATTCCAACAATTTGGTAAAGATGCGT--CTGACTTTATTGATTCAGTACGTGATCCTCAAACACTTGAGATACTTGACATTACACCTTGCTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAACACTTCTTTAG-AGGTGGCTGTTCTTTACCAAGATGTTAACTGCACTGATGTACCAACTACTATACATGCAGACCAACTAACACCTGCTTGGCGTATTTATGCTACTGGCACTAATGTGTTTCAAACTCAAGCAGGCTGTCTTA-TAGGAGCTGAACATGTCAATG-CTTCTTATGAGTGTGACAT-CCCAATTGGTGCTGGTATTTGTGCTAGCTACCATACGGCTT-CTA-TA-T--TA--CG--CA-GTACA--AGCCAGA-AAGCTATTGTGGCTTATACTATGTCCCTTGGTGCTGAGAACTCTA-TCGCTTA-TGCTAACAATTCTATAGCCATACCTACAAATTTTTCAATTAGTGTTACCACTGAAGTTATGCCTGTATCAATGGCTAAAACTTCTGTAGATTGTACTATGTATATCTGTGGTGACTCTA-TAGAGTGTAGCAA-CTTGTTGTTACAATATGGCAGTTTTTGCACACAACTAAATCGTGCTTTAAGTGGGATTGCTATTGAGCAAGACAAGAACACCCAAGAGGTTTTTGCTCAAGTTAAGCAAATCTATAAAACACCACCTATTAAGGATTTTGGTGGTTTTAATTTTTCACAGATACTACCTGACCCATCTAAACCCAGCAAGAGGTCGTTTATTGAAGA-CTTACTCTTCAATAAAGTCACTCTTGCTGATGCCGGTTTTATCAAACAGTACGGTGATTGT-TTGGGTGGTATTTCTGCTAGAGATTTGATTTGTGCTCAAAAGTTCAATGGACTTACTGTCTTACCACCATTGCTCACAGATGAAATGATCGCTGCTTATACAG--CTGCATTAATTAGCGGC-ACTGCCACTGCTGGATGGACCTTTGGTGCTGGTGCTGCTCTT-CAAATACCATTTGCCATGCAAATGGCTTATAGGTTTAATGGAATTGGAGTTACTCAGAATGTTCTCTATGAGAATCAGAAATTAATAGCCAATCAGTTTAATAGTGCTATTGGAAAAATCCAAGAGTC--TTTGACATCTACAGCTAGTGCACTTGGAAAA-TTGCAGGATGTTGTTAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGTTCCAATTTTGGTGCAATTTCAAGCGTGTTGAATGACATTCTTTCACGCCTTGACAAAGTCGAGGCTGAGGTTCAGATTGATAGGTTGATCACAGGTAGACTTCAGAGTTTACAGACGTATGTGACTCAACAATTAATCAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCGACTAAAATGTCCGAGTGTGTACTAGGACAATCTAAAAGAGTTGATTTTTGTGGAAAAGGTTATCACCTAATGTCTTTTCCC-CAGTCAGCGCCTCATGGTGTTGTCTTCTTACATGTGACTTACATTCCTTCGCAAGAAAAGAACTTCACAACAGCTCCTGCCATTTGCCATGAAGGTAAAGCACACTTCCCACGTGAAGGTGT-TTTCGTTTCGAATGGCACACACTGGTTTGTAACACAAAGGAACTTTTATGAACCTA-AAATTATAACCACTGACAATACATTTGTCTCTGGTAACTGTGATGTTGTAATTGGAATTATCAACAACACAGTTTATGATCCTTTACAACCAGAACTT-GATTCATTTAAGGAGGAGTTAGATAAATATTTTAAAAATCATACATCACCTGATATTGATCTT-GGTGATATTTCTGGCATTAATGCTTCTGTTGTCAATATTCAAAAGGAAATTGACCGCCTCAATGAGGTTGCCA-GAAATTTAAATGAATCACTCATTGATCTCCAAGAACTTGGAAAATATGAGCAATATATCAAATGGCCATGGTATGTTTGGCTCGGCTTCATTGCTGGACTC-ATTGCTATAGTCATGGTTACAATCCTGCTTTGTTGCATGACAAGTTGTTGCAGTTGTCTCAAGGGCTGTTGTTCTTGCGGATCTTGCTGTAAATTTGATGAAGACGACTCTGAGCCTGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATTTTCACTCTTGGAACTGTTACTCTTAAA-CAAGGTGAAATCAAAGG-TGCTACTCCTACAAATTCTGTTCGCACTACTGCAACAATACCGATACAAGCCACACTCCCTTTCGGATGGCTTGTTGTTGGCGTTGCAATTCTTGCTGTTTTTCAAAGCGCTTCAAAAATAATTACACTCAAAAAGAGATGGCAGTTAGCCCTCTCTAAAGGTGTTCATTTTGTTTGCAACTTGCT-TCTGCTGTTTTTAACAGTTTATTCTCAC-TTGTTGCTTC-TTGCTGGTGGC-TTGGAAGCCACTTTTCTCTTTCTTTATGCATTAG-CTTA-TTGCTTGCAAACTGTAAATTTTGTGAGAATAATAATGCGA-TTCTGG-TTGTGCTGGAAGTGCCGTTCCAAGAATCCTTTACTCTATGATGCCAACTACTTTCTTTGTTGGCATACTAATTGTTATGACTATTGTATACCATACAATAGTGTAACCTCTTCAATTGTCATCACATGTGGTGATGGTACTACGAA-TCCCATTTCTGAGGACGACTACCAAATTGGTGGTTACACGGAAAAGTGGGAGTCTGGTGTTAAGGACTGTGTTGTATTACATAGTTATTTCACCTCAGATTACTACCAGCTGTACTCAACACAAGTGAGTACAGACACTGGTGTTGAACATGTTACTTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTTCAAATTCACACAATCGACGGTACATCTGGAGTTGTTAATCCAGCAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTTGTGGTATTCTTGCTAGTCACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTTTGGTCTAAACGAACTAAATATTATATTAGTCTTTCTGTTTGGAACTTTAATTTTAGCCATGTCAGGTGACAACGGTACCATTACCGTTGAAGAGCTTAAAAAGCTC-TTAGAACAATGGAACCTAGTAATAGGATTCTTGTTTCTTACATGGATTTGT-TTGTTACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTACATAATTAAGTTAATTTTCCTCTGGCTGCTT-TGGCCAGTAACTTTAGCTTGCTTTGTGCTTGCTGCTGTTTACAGAATAAACTGGATCACTGGTGGAATTGCCATTGCAATGGCCTGTCTTGTAGGCTTGATGTGGCTTAGCTACTTCATTGCTTCTTTCAGGCTGTTTGCTCGTACGCGTTCCATGTGGTCATTTAACCCAGAAACTAACATTCTTTTGAACGTGCCTCTTCATGGCACAATTCTGACCAGGCCGCTTCTAGAGAGTGAACTCGTAATTGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCAGGACACCATCTGGGACGCTGTGACATCAAGGACCTGCCCAAAGAAATCACTGTAGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGTAATTACAAATTAAATACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTATTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTATATCATAAATCTCATAATTAAAAATCTATCTAAGCCTCCAACTGAGAATAACTGTTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACTAACATGAAAATTATT-TTCTTCTTGGTACTGATAACACTTGTTACTGGCGAGCTTTACCACTACCAAGAGTGTATAAAAGGTACAACTGTACTTTTAAAAGAACCTTGCTCTTCAGGAACATATGAAGGCAATTCACCATTTCATCCTCTAGCTGATAATAAATTTGCACTGGCTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGTGTTAGACACACCTTTCAGTTACGTGCGAGATCAGTTTCACCCAAACTGTTTACCAGACAAGAGGAAGTTCAAGAA-TTATACTCACCTGTTTTCCTTATCGTTGCAGCTATAGTGTTCATAATACTTTGCTTCACATTCAAAAGAAAAATAGAATGAGTGAATTTTCATTAATTGACTTCTATTTGTGCTTCTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTCATTATCTTTTGGTTCTCACTAGAACTGCAAGATCATAATGAAACTT-TCCACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAATTCTT-ACAACAGTAGCTGCATTCCATCAGGAATGTAGTTTACAGTCATGTGCTCAGCATCAACCCTATGTAGTTGATGACCCTTGTCCAATTCACTTCTACTC-ACGATGGTATATCAGAGTGGGAGCTAGAAAATCAGCACCTTTGATTGAATTGTGTGTTGATGAGG-TAGGCTCTAAGTCACCCATTCAATACATTGACATTGGTAATTACACAGTTTCCTGTTCT-CCTTTTACAATTAATTGCCAGGAACCTAAATTAGGTAGTCTCGTAGTACGGTGTTCGTAT-TATGAAGACTTTCTAGAGTACCATGACATTCGTGTTGTCTTAGATTTCATCTAAACGAACTAACTAAAATGTCTGATAATGGACCCCAAAACCAACGTAGTGCACCCCGCATTACATTTGGTGGACCCTCAGATTCAAGTGACAATAGCAAAAACGGAGAGCGCAATGGTGCACGACCTAAACAACGTCGACCCCAAGGCTTACCCAATAATACTGCATCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAAACCTTACGTTCCCTCGAGGGCAAGGTGTTCCAATCAACACCAATAGCTCTAAAGATGACCAAATTGGCTACTACCGTAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGAGCTCAGCCCCAGATGGTATTTTTACTATCTAGGAACTGGACCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGAAGGCATCATATGGGTTGCAACTGAGGGAGCCTTAAACACACCGAAAGACCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCTTCTTCACGCTCCTCATCACGTAGTCGCAACAGTTCAAGAAACTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGACACTGCTCTTGCTTTGCTGCTGCTAGATAGGTTGAACCAGCTTGAGAACAAAGTATCTGGCAAAGGCCAACAACAACAGGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCATCTAAAAAGCCTCGCCAAAAACGTACTGCTACAAAA-CAGTACAACGTCACTCAAGCATTTGGG-AGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAAGAATTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCTCCAAGTGCCTCTGCATTCTTTGGAATGTCACGCATTGGCATGGAAGTCACACCTTCGGGAACATGGCTGACTTATCAT-GGAGCCATTAAATTGGATGACAAAGATCCACAATT-CAAAGAT-AACGTCATACTGCTGAATAAGCACATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAAAAGGCTGATGAA-CTTCAGGCTTTACCGCAGAGACAGAAGAAACAACAAACTGTGACCCTTCTTCCTGCTGCAGATTTGGATGAATTCTCCAAACAGTTGCAACAATCCATGAGTGGTACTGATTCAACCCAGGCTTAAACTCGTGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATTTCACATAGCAATCTTTAATCAATGTGTAACATTGGGGAGGACTTGAAAGAGCCACCACGTTTTCACCGAGGCCACGCGGAGTACGATCGAGGGTACAGCCAATAATGTTAGGGAGAGCAGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCA-ACCACTC-GACAAGAAAAAAAAAAAAAAAAAAAAAAAAAA----\n",
      "||-|.|-|||||.||||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||||||||||||||||||.||||||||..|||||||||||||||||||||||||||.||||||-|||.|||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||.|.||.||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||-||||-||||||||||||||||||||||||||.||||||||.|||||||||||||||||||||||.|||||.|-|||-||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||.||.|||||||||||.||..|.||||||||||||||.|||||||||||.||||||||||||||||||||||||||.||||||||||||||.||.||..|||||||.||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.|||||||||-||-|||||||||||||||||.|||||||||||||||.||||||||||||||||||||||||.||||||||||||||||||.|||||||||||||||||.||.||||||||.||||||||.|||||||||||||||||||||||||||||||||||.||.|||||||||||.|||||--||||--||||||.||||||.||||.|||||||||||||||||.|||||.|||||||||||||||||.|||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||-||-||||||.|.|||||||||.|||||||||||||||||||||||||||||||||||-|||-||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||.||||.|||||||||||||||.|||||||||||||||||||||||||||||.||||||||||.|||||||||||||||.|||||.|||||.|||||||||||||||||||||||.|||||||||||||||||.||||||||||||||-||||-|||||||||||||||||||.|||.||||||||||||||||||||||||||||.|||||.||||||||.|||||.|||||||||.|||||||||||||||||||.||.|||||-||-|||.|||||||||||-|-||||||||||||||.|.||||||||||||||||||||||||||||||||.|||||||||||.|||||.||.|||||||||||||||||||||||.|||.||||.||.||||||||||||||||||.|.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||.||.|.||||||||||||.||||||||.||||||||||||||.|||.|||||||||||||||||||||||||||||||.|||||||||||.||.||||||||-|||-||||||||||||||.||.||||||||-||.|-|||||||||||||.||.|||||.||..|.|||||.|||||||.|||||||||.||.||.|||||.|||||.||.|||||||||||..|||||||||.|||.|||||.|||||.|||||||..||-||-||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||.||||||||.|||||||||||.||..|||||||.|||||||||||.||.||||||||||||||||||||||||||.|||||.||.|||||||||||.||||||||.||.||.||||||.||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||.|||||||||||||||||||.|||||.|||||||.|||||.||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||.|||.||||||||||||||||||.|||||||||||||.||||||||||||||.||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||.||||||||||||||.||.|||||||||||||||||||||||.||||||||.||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||||||...||.||||||||||||||||||||.|||||.||||||||.||||||||.||.||||||||||||||||||||.|||||||||||||||.||.|.|.|||||||||||||||||||||.||||||||||||||||||||||||||.|||||||||||.|||||.|||||||||.|||||||||||||.|||||||||||||||||.||--||-||||||..||||.||||||||||||||.|.|||.|||.|||||||||||||||||||||||||||||.|||||.||||||||||||||||||||.||||||||.||||||..|.||-||||..|-|.|-|||||||.||||||||.|||||||||||-|.-|||||||||..|--|-|--|.-|||.|-|||.|||||-||.||.|||-|-|-||-|||----|.|.---|-|||||.||||.||||...||-|-||.|||||-|.|-|||||||||||-|.||--|||.||||...||||.|||.|..||||||||||||-|||-||||||||||.|.||||||||||||.||||||||.||||||||||||||||||||.||.|||||.||||||||||||||.||||||||||||||||||||||||||||||||.|||||.|||||||||||.||||||||.||||||||.||||||||||||.|.||||||.||||.|||||.|||||-||.-|||||||||.||||||||||.||.|||||.||.|||||||||.|.||||||||||||||.|||||.|||||.||||.||||||||||||||||-||-|||||.|.|||||||||||||||.|||||.||.||-|.||-||||.|||||||||||||||||||||||||||||||||||.|||.|||||||||||||||||||||||||.|||||||.||||||||||||-||-||.|||||||||.|||||||||||||||||||||||||||||||||||.||.||||||||||||||.|||||.||||||||-|.|-|||||.|.|||||||||||||||||||||||||.|.|||||||.||||||||||||||||||.|||.||||||.|||.-|-|-||||||||||||..||.||||||||||||||||.|||||.||.||.||.||||||||.|||||||||||||||||||||||||||||.||||||||||||||.||||||||||||||||||||||..||.|||||||.|||||||||.||||||||||.||||||||||||||||||||||.||.||||.||.|.||||||||||||.||||||||||||||||||||||||||||||||.||||||||.||.||||||||.|||||||||||||||||.||||||||||||||.|||||.|||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||.|||||||||.|||||||||||||||||||||.|||||||||||||||.|||||||||||||||||.||||||||.|||||.||||||||||||||||||||.||.|||||||||||.|||.||||.||||||||.|||||||||||||||||.|||||.||.||.||.|||||||||||.|||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||.||||||||.||||||||||||.||||||||||.||||||||.|||||||||-||-||.|||||.||.||.||-||-||||||.|||||||||||||||||.|||||.|||||||||||.||.||||||||.||||||||||||||.||||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||.||||||||||||.|.|||||||||||||||||||||||||||||.|||||||||||.||||||||||||||.||||||||||||||||||||||||||.||||||||||||||.||||||||.||.|||||.|||||||||||||||..||||||||||||.|||||||||||||||||.|||||||||||||||||||.||||||.|||||||||||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||||||||.|||||||.|||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.|||||||||||.||-||-|||||..||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||.|||||||||||||||||.|||||||||||||||||.|||||||||||.|||||||||||||||.|||||||||||||||||||.|||||||||||.|||.|||||||.||.|||||.||.|||||.||||||||.||||||||.|||||||||||||||.|.|||||.||||||||||||||||||||||||||||||||.|||||||||||||.|||.|||||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||.||.|||||||||||||||||||||||.||||||||||||||||||||||||.|||||||.|||||.|||||||||||||.|||||||||||||||||||.|||||||.||||||||.|||||||||||||||||||||||.|||||.||.||.||.||||||||||||||||||||||||..|||||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||||||||.||.||||||||||||||||||||.||.||.||||||.|||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||.||.||.|||||||||||||||||||||||||||||.|||||||||||...|||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||||||||.||.|||||||||||||||||||||||||.|||||||||.|||||||||||.||.||..|||||||||||||||||||.||.|||||||||||.||||||||-|||||-||||||||||||||||||.|||||||||||||||||.|||||||||||.||||||||||||||||||||||||.|.|||||.||.|||||||||||||||||||||||||||||.|||||.|||||.||||||||||..|||||.|||||||||||||||||.||||||||.|||||||||||||||||.|||||||.||||-||.|-||.||||.|.|||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||.|.||.||.|||||||||||.||||||||||.||.|||||.|||||||||||||||||.||||||||.||||||||||||||||||||||||||||||||.||||||||||||||||||||.-||-||.||||||||.||.||||||.||||||||||.||||||||.||||||||||||||||.|||||||||.|||||.||.||||||||.|||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||.||||||||||||||||||-||-||||||||||||.||.|||||||||||||||||||||||.|||||.||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||.|||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||||.||||||||||||.|||||||||||||||||||||||||.||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||.||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||-||-|||||||||||||||||||||||||||||||||.||||||||-|||-|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||.||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||.|||||||||.|||||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||.||||||||||||||||||||||||||||||||||||||||||||.||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||.||||||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||.|||||.||||||||||||||.||||||||||||||.|||||||..||||||||||||||.||.||.||||||||||||||||||.||.|||||||.||||||||||||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||.||.||.||||||||.||.||.||||||||.||||||||||||||||||||||||||.||.||.|||||||||||||||||.||||||||.||||||||.|||||.|||||.|||||||||||.|||||.||||||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.|||||||||||.|||||||-|-||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||.||.||||||||||||.|.|||||.||||||||||||||.|||||..|..|||.|||.|||||||||||||||||||||||||||||||.||||||.|.||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||.||||||.|.||||||||.|||||.||||||||||||||||||||.|||||||||||||||.|.||.||.||.||.|||||.|||||.||.|||||.|||||.||||||||||||||.|||||||||||||.|||||||||.|||||||||||-|||-||||||||.||||||||.||.||||||||.||||||||.||.||.||||||||||||||||||||||||||||||||.||.||.|||||.||||||||||||||.|||||||||||||||||||||||.||||||||.||||||||||||||.|||||||||||||||||||||||.|||||||||.|.||||||||||||||.||||||||||||||.||.|||||.||||||||.||||||||.|||||||||||||||||||||||.|||||||||||||||||.|||||.|||||.||.||.|||||.|||||||||||.|||||..||||||||||||||||||||||||||||||||||.|||||||||||||||||.||.|||||||||||.|||||||||||||||||.||||||||||||||||||||.||.|||.|||||||||||||||||||||||||||||||.||||||||.|||||.||||||.||||||||||..|||||||.||.||.||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||.||||||||.||||||||||||||||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||.||||||||||||||||||||.|||||.|||||||||||.|||||||.||||.|||||||.|||||||||||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||.||.|||||..|.||.|||||||||||||||||||||||||||||-|||-||||||||||||||.||-|-||||||||||||||||.||||||||.|||||||||||||||||||||||||||||||||||.||.|||||.||.||.||.||.||||||||.|||||.||||||||.||.||||||.|.|||||.||||||||.||||||||||||||.||.|||||||||||||||||.||.|||||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||.|||||||||||.|||||||||||.|||||||||||||||||.|||||||||||||||.||||||||||||||||.||||||||||||||.||.||||||||||||||||||||.||.|||||.|||||.||||||||.||.||.||||||||||||||||||||.||.|||.|||||||||||||||||||||||||.|||||||||||.||.||||||||.|||||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||.|||||.||||||||.||||||||.|||||||||||||||||||||||||||||||.|||.||.|||||||||||||||||.||.|||||||||||.||||||||.||||||||||||||||||||||||.|||||||||||||.||.|||||||||||.|||||.||.|||||.||.|||.|.||||||||||||||.||.||.|||||.|||||||||||||||||.|||||||||||||||||||||||||||.||||||||||||||||||||||.|||||||||||||||||.||..|||||||||||||||||||||||||||||.|.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||||||||.||-||||-||||||||||||||||.|||||||||||.||.|||||.||||||||.||.|||.|||||||||||||.|||||||||||.||||||||||||||||||||||||||.|||.|||||||||||||||||.|.||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||.|||||||||||..||||||||||||||||.||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||||||.|||||||.||||||||||||||.||.|||||||||||||||||.|||||.||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||.||||||||.||||||||.||||||||||||||||||||||||||||||||.|||||.|||||||||||.|||||.|||||.|||||||||||||||||.|||||.||.|||||||||||||||||.||.|||||.||||||||.||.|||||.||.|||||.||.|||||.||.||.|||||.||||.|||.||.||.||.|||||.||.|||||||||||||||||..||||.||.|||||||||||.||.||.||.||||||||||||||||||||.||.||||||||||||||||||||-|||-|||||||||-||-||.|||||.||||||||.|||||.||||||||||||||.||.|||||.||.-||-||||||||.|||||-|||-||.||.||.|||||||||||.|||||.||.|||||||||||.-||-|||||||||||.|||||||||||||||||||||||.||||||-||-||.||.||||||||.||||||.|.||.|||-||-||.|||||.||.|||||.||.|||||.-||-||.||||||||.||||.|||.|||||||||||.|||||||||||||||||.||.||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||.|||||||||||||||||.||.|||||.||||||||||||||||||||||||||.||||||||.|||||.||.|||||||||||||||||||||||.-|||-|||||||||||||.|||||||||||||||||||||.|.|||||||||||.||-||.-|||||.|||||.||.|||||||||||.|||||.||||||||..|.||.||||||--|||-|-||.||.|||||.||||||||||-||-|.|||||.||||||||||||.|.|||||||||||.||.|||||.||||||||||||||||||||.||.|||||||||||.|..||||||||.|||||.|||||.||.||||||||||||||.||.||||||||.||||||||.|||||||||.|||||||.||||||||.|||||.||.|||||||||||||||||.|||||.||.|||||||||||.||.|||||.|||||||||||.||.-||-||.|||||.||||-|.-|.|||||||||||||||||.||||||.|..||||.||..-||-|||||.|.|||||||||||||||||||||||||||||.||.||||||||||||||||||.|||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||.|||||||||||.||.-||||-||||||||||||||||||||||||.||||||||.|||||||||||||||||.||.|||||||||||.||||||||||||||||||||||||||.-||-|||||.|||||||||||||||||..|.||||||||||||||||||||.|||||.|||||.||||||||||||||.||||||||.||||||||||||||||||||||||||.||.||||||||.||.||.|||||.|||||||||||||||||||||||||||||||||||.||.||.||||||||.||..|.||||||||.|||||||||||||||||.||||||||||||||.||||||||||||||.|||||||||||.|||||||.|||||||||||||||.||||||||.|.||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||.|.|||||.||.|||||.-|||-||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.|||||||||||.||.|||||||||||.||.|||||.||.|||||.|||||||||||-|||-|||||||||||||||||.||||||||||||||||||||.||.||.|.|||||||-||-||.|||||-|||-|||||||||||.|||||..|.||.|||||||||||||||||.||.|||||.||.||..||||-||.|-|||||||.||||||...||||||||-|||-|||||||||.||||.|||||||||||||||||||||||||||||||||||.|||||||||||||||||.||..|.|||||||||.||.|.||||||||.||.||.||||||||.||||||||.||||||||||||||||||||.|||||||||||..|.|||||||||||.|||||.||||||||.|||||||||||.||.|||.|-|-|||||||||||||.|||.|||||||.||||||||.|||||.||.||.|||||.||.||||||||.|||||.|||||..||||.|||||.||.||||||||||||.||||.|||||.|||||.|||||.||.|||||.||.|||||.||||||.||||.--|.|||-|-|||-|-|.|||||.|||.||||.|||||.||.|||||.|||.|.||||||||-|-|.||-||.-||.||||||||..||||.|||||.|||.|.|||.||||-||..-|.|||||--|||--||||||||||||||.|.||||||||.||||||||||||||..||...||-|||-|||||||||||||||||||||||.|||||.|||||.||.||||||.||||.||||||||||||||.||.||.||.||||||||||||||||||||.|||||.|||||.|||||.||||||||.|||||||||||||||||.|||||.||||||||.||||||||.||.|||||.||||||||.||||||||.|||.|.||||||||.|||||||||||||||||||||||||||||||.|||..|.||.||||||||||||-||--||-|||||.||||||||.||.||||||||.|||||.|||||.|||.|.||..||||.|||||||||||.||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||||||||.||.|||||||||||.|||||||||||||||||.||||||||.|||||.||||||||.|||||.|||||.||.||||||||||||||.|||||||||||.|||||.||||||||.|||||.|||||.||.|||||||||||.|||||.|||||||||||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||.||||||||.||.||.|||||.|||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||.||||||||-|||-|||||||||||||||||||||||||||||.||||||||||||||||||||.||||||||.|||||.||||||||||||||||||||.|||||||||||||..||||||||||||||||||||.||.||.||||||||.||||||||||||||.|||||||||||.|||||.||.||.|||||||||||||||||||-||-|.|||||||.|||||||||||||||||||||||||||||||||||||||.||||||||.|||||.|||||||||||||||||.|||.|||||||.||||||||||||||.|||||.||.|||||||||||||||||||||||||||||.||||||||.||||||||.||.||.||.|||||.|||||||||-||-||.|||||||||||.||.|||||.|||||||||||||||||.-||-||||||||||||||.|||||||||||||||||||||||.||.||.-|||-|.||.|||.||||.|||||.||||||||||||||.||||||||.||||||||.||.|||||.|||||.||.||.||.|-|-|||||.|.|||.||..|.||.||||||||.||||||||.|||||||||||-|-|||||||||||||.|||||.|||||.||.||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||.|||||||||||||||||.|||||||||||||||||.|||||.|||||||||||.|||||..||||.||.|||||||||||.|||||.||.||.|||||||||||||||||||||||||||.||||.|||||.|||||.||.||.||.||.||||||||-|||-||.||||||||.||||||||||||||-||||-||||.||||||||.||.||.||-|||.-|||||||||||-|||-|.||.||||||||.||||||||.|||||.|||||.||||||||.|||||.||.|||||.|||||||||||||||||||||||.|||||||||||.||.|||||.||.||.|||||.||.||.|||||.-||-||||||||||||||||||-||.-|.||.||||||||.||-|.|-|||||.|||-||-|||||.||.|||||.||.||.||.|||||.||||||||||||||||||||.||||||||||||||.||||||||||||||.||.|||||||||||||||||.||.||.|||||.||.||.|||||||||||||||||.||.||.||||||||||||||||||||||||||.||||||||||||||.||.|||||||||||.||.|||||.||||||||.||.||.|||||.||.||||||||.||||||||.||.|||||.||||||||.||||||||.||||||||.-||-||.|||||.||.||.||.||||||||.|||||||||||.|||||||||||||||||||||||.|||||.|||||.||||||||||||||.|||||.||||||||.|||||.||.||||||-||-||||||||||||||.||.||.|||||||||||||||-||-||..|||||||.||||||||||||||.|||||.||.|||||.|||||||||||.||.||||||||.|||||||||||||||||||||||||||||||||||.|||.|||||||.||||||||.|||||.||||||||||||||.||||||.|.||||||||.||||||||||||.||||||||||.||||||||||||||||||||||||||..||||.||||||||||||||.|||||.||||||||.|||||||||||||||||||||||||||||.||||||||.||.|||||.|||||.||.|||||||||||||||||||||||||||||||||||.|||||||||||.||.||.|||||||||||.|||||.|||||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||||||||.||||||||||||||.||||||||||||||.|||||||||||.|||||||||||||||||.||||||||||||||||||||.||||||||||||||-|||-||.|||||.|||||..|||||||.||.||.||.||||||||.||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.|||||||.||.|||||.||.||||||||.||.||.||||||||||||||||||||||||||||||||.||.||||||||.||||||||||||||||||||||||||.||||||||||||||.||||||||||||||||||||.||.||||||||.||||||||.|||||||||||||||||.|||||||||||.|||||||||||||||||.||||||||.|||||.||.||||||||||||||.||.||||||||.||||||||.||.|||||||||||.|||||.|||||.|||||.|||||||||||.||.|||||||||||.|||||.||.||.||.|||||||||||||||||.|||||||||||.|||||||||||||||||.||.||.|||||||||||-|||.-|.||.|||||||||||||||||.||||||||.||.|||||||||||||||||||||||.|||||||||||.||.||.||||||||||||||||||.|.||||||||.||.|||||.||||||||-|.|-|||||.-||-|||||||||||||||||||||||||||||||||-||-|||||||..|||||.|||.|.||.|||.|.||||||||.|||||||||||.||||||||.||.|||||||||||.||.|||||.|||.|.||.|||||.-||-|||||.||||||||||||||.||.|||||||||||.|||||||||||.||||||||.-||-|||||.||||||||.||.|||||.|||||||||||||||.|.||||||||||||||||||||.||||||||||||||.||||||||||||||||||||.||||||||||||||.|||||||||||..|.||.||.|||||||||||.|||||.|||||.|||||.||||||||.||||||||.|||||.||.||.|||.||||..|.|||.||||.|||||.|||||||||||||||||.||||||||.||.||.||||||||--|||-|.|.-|-|.-|||||.|||.|||||||||.|||.|||||.||.||.||.|||||||||||.||.||.||||||||||||||||||||-|||-||.|||||.||.||||||||.||||||||.||||||.|.|||||.|||||.|||.-||-|.||.||||||||.||.||||||-||-|||||||||||||||||.-||-|||||.||||||||||||||.||.||.|||||.||..|.||.||||||||.|||||.|.|||.||.||.|||||.||.|||.|.||.||||||||||||||||||.||||.|||.|.||.|||||.||.||||||||.|||||||||||||||||.||.||.||||||||||||||||||||.|||||.||||||||||...|.||.|||||.|||||||||||.||||||||||||||||||||.|||||||||||.|||||.||.|||||.|||||||||||||||||.|||||.|||.||.|||||||.||.||.|||||.||||||||||||||||||||.||.||.|||||.||.||.||.||||||||||||.|||||||.-||-||.||.|||||||||||||||||.||.||.||.|||||-||||-||||||-|-|||||.||.||-|-||||.||.||.|||||.|||||.||.||.||||||||.|||||||||||||||||||||||||||.|.|||||.||.||||||||||||-|-||||||.||||||||||||||.|||.|.||||||||.||.||.||.|||.|.||.||.||||||.|.|||||.|||||.|||||.|||||||||||.|||||.-||||-|.|.||.||||||||||||||.|||||||||||.|||||.|||||.||||||||.||||||||||||||.|||||.|||..-|-||||.||||||.|.||.||-|||-|||||||||||||||||.||||||||||||||.|||||.||.|||||||||||.||||||||.|||||||||||||||||||||||.|||||.|||||.|||||||||||.||.||||||||||||||.||.|||||||||..|||-||.|-||||.|||||.||||||||.||.|||||.||||||-||-|||.||||.|||||||||||||||||.|||.|.|||||||||||.||-|||--||-||||||||||||||||||||||.|||||||.|.||.||||||||||||||||||||-|-|.|||||||.|||.||.||||||.|||||||..||||||||||.||-|||-||||||||||||||.||-||.-|||||||||||.||||||-||-|||||||||||.||||||||||||||||||||.||.|||||||||||.||.||||||.|.|||||||||||||||||||||||.||.||.||||||||||||||.||.|||||||||||.|||||.||.|||||.|||.|.|||||.||.|||||.|||.|.|||||||||||.||.||||||||.|||||.|.-|.||-||.||.||||.||||||||.|||||||||||||||||.||.||.||||||-||-||||||||.|||||||||||||||-|||.||-||||||||...|||||.||.|||-||-||.||-|.|-||.||...|||.||||||.|.||.|||||||||||||||||||||.|.||||||||||||||||||||-||-|||||||||.|||||..||.|.||.|||||.||.|||||.|||||||||||.||.|||.|.|||.|.||-|||.||-|||||-|-|||-||-||||||.|||||||-||-||..|.|||||||||||||||||.|||||.|||||..||||.||||||||.|||||.||.||||||||||||||||||||||||.|.||.|||||.||.-||-|||.||.||||.|||||.||.|||.|.|.-|-||||.||||-|.||.||.-||||.|||||||||||||||||.||.|||||.||||||||.||||||||||||||||||||.|||||||||||||||||.||.||||||||.||.-||-|||||||||||.|||||||||||||||||.||.||||||-||-||.||.|||||.||.||.||||||||.||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.|||||.|||||||||||||||||||||||||||||.||||||||.||||||||||||||||||||.||.||||||||.||||||.||.|.|||||||||||.|||||||||||||||||||||||.|.|||.|||||||||||||||||||||||||||||.||.||||||||||||||.||||||||.|||-|||-|.||||||||.|||||||||||||||||||||||||||||.||.||.||||||||||||||.|||||.||.|||||||||-|||-|||||||||||||.||.|||||.||.|||||||||||.|||-||||-|||.|||||.|||||||||||.|||||.||.||||||.||||||||||.||||||||.||||||||||||||||||||||||||.||||||||.||.||||||||||||||||||||||||||.|||||.||.|||||.||.||||||||||||||||||||.||||||||.|||||..||||.||.|||||||||||.|||.|||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||.|||||.||.||.||-|||-||||||||.|||||||||||||||||||||||.||.||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||.|||||||||||.||.|||||||||||||||||.|||||.|||||||||||.|||||||||||||||||.||||||||.||.||||||||||||||||||||.|||||||||||||||||.||||||||..|||||||.||.||.||||||.||||.||||||||||||||||||||||||||||||||||-||||-|||||||||||||-|--|-||-|||.-||||..-||..-|||||||||||-|||-|||..||||||.|.|.|-|..|||.--||-|.||||||||||-||||.-|.|||||||||||||||||||-||-|||-|-|||||---|.||.||.|.|-|-||.|-||||..|-|.|||||-||.||-||-|||.|||||||-||-|||---||--||-|-|||-|..|....|.|||||.|---||.||||||..||||||.|||.|..||.-||||-||.||.||..|.||.||||||.|||||||-|-|.||.||.||.||.||.||||||||.|||||.||.|||-||-||--|-||.||--|-||.||.||-|||-||.|||||.|||||.||.||||||.||||.||.||.|||-|||||..|||||-||.|||||||.|..-||.|-||.|||.||.||.||.|||||-||.|-|-|||-|.|.-||.||.||--|.|..|-||.||||-|-|||---||||||.||||||||.|||||.||..|.-|||||-|.||--|.||.||...||.||..-|-|||.-|-|||||.|.||||||.||.|||||-|||.||||-||-|-|||||..||||.||.||.||.||.|||...||.||-|.|||-||||.-|-||||.-||.-|||-|.||-|.|.||-|-||||.||--||||||||.||.||.|||.|-||.-||.||.||.||||..||-||-|||..|.||..||-||-||||||.|--|---|||-|||.-||..|..|..-|-||.|||||.|||..|.|.||.|||||||-|-|||||.||||||.||||-|.|-||.|||.|..|.|||||||||||.|||||.||||||||.||||||||.||.|||||||||||||||||.||.||.||||||||||||||.|||||-|||-|||||..||||.||.||||||||.|||||||||||..||||.|||||.||..|.|||.|.||||||||||||||||||||.||-|-||||.||.|||...-|||-|||||||.|||||.||.|||.|.||.||.|||||.|||-||-||-||.-|.|||--|.|-|.-||..||||.|||||..||||..|.||.|||||..|.||.|||||.||||||||.||||||||.||.||.|||.|||||||.|.||||.|.||||||||.|.|||.|||||.|||.|||||.|.||.|||..-|.|-|||||||||||..|.||.|||||.||||||||....||||||||.|||||||||||||||||.||||||||||||||.||.||.|||||||||||..||..|||-----|--|-|-||.||-|-|||||||||.-|-|||..||.|.-||.|||-||..||----|-|-|.|||||.|||||.||||||.||||--||.|.|-|.|..|-||.-|-|.-|.||||-||--|--|-|----|.--||--|-|-|--||||-|.---||..|-|-|||-|----|-||-|||--||..-|-----|||||.|-||||-|.||.||.||.-||-||.|||||.||||||.||||||||||.||.|||||||||||-||||--|-||.||.-||.|.|-|.|||||-||.-||.||.|||||||||||.||.||..|.|-|||-|||.|||||-|||-|||||.|||..|||.|.|||||.|.|.||||||||||||||.|.|||--|.|--|||||..||.||||.|.||.||||||||.||.|||||||||||.||||||||||||||.||.||||||||||||||||||||||||||||||||||||||.||||||.-|.-||||.|||||||||||.||.||||||||||||||.||.||.||...|.||||.||||||||.|||||.||.|||.||||||||.|||||.||||.||..|||||||.||||||||.|..|||||||||-|||-||||.||||||||||||||..-||-|.||||||||||||||-||||-|||||||.|||||.||.|||||.||.||.||.-|..-|||-|.-|--|.--||--||-|||..--|||.||.-||.|.||..|.||.||.||||||||.||||||||.||.||.||-|-|.|||||-|-||||.||.|||||.||||||||.|||||||||.|.||||||||||||||.|||.||.|.||.||.||.|||.|.||.||.||.|||||||||||.|||||.||.||||||||.||.|-|-||.||.|||||-|||-|||||.|||||||||||||||||.||||||.||||.||||||||||.|||.||.|||.||||.||||||||.|||||||||||.||||||||.|||||.||.|||||.||.|||||||||||.|||||.||||||||||||||||||||||||||.|||.||||.||.|||||.|||||.|||||||||||.|||||||||||-||-|||.|||||.|||||.||.|||||.|||||.||.||.||||||||.||.||||||||.-||||-||.||||.||||||||||..|.||||||||.||||||||.||.||.||||||||.||.|||||.||||||||||||||||||||.|||-|.-|||||.--|||||.|.-|||||||.-||...||||.||||.||||||||||||||.||||||||.-||-||||||||||||||.||||||||||||||||||||||||||.|||||||||||.||||||||||||||||||||.||.|||||.||.|||||.||.|||||||||||||||||.|||||.|||||.||--|||--|.||.|||||.|||||||||||||||-||-||.|||||.||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||.||.|||||.||.||||||||.|||||||||||.||||||||.||.||.||||||||||||||||||||.||||||||.|||||.|||||.||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||.|||||||||||.|||||||||||.||||||||.|||||||||||||||||||||||.||.|||||.||.|||||.||-|||-||||||||.|||||||||||.||||||||.|||||||||||..|.|||.|.||||||||||||||||||||.||||||||||||||.|||||.||.|||||||||||.||.|||||||||||-|||-|||||.||||||||||||||||||||||||||||||||.|||||||||||-|-||||.||.||.||.|||||.||||||||.|||||||||||||||||||||||.||||||.|||||||||||||||||||||||||.|||||.|||-||-||.|||||.||||||||||||||||||||||||||.||||||||||||||.|||.|||||-||-|||||.||.|||||||||||||||||.|||||.||.||||||||.||||||||||||||||||||||||||||-|||-||||||||||||.|||||.||||||||||||||||||||.||||||||.|||||.||||||||||||||..|||||||.||.||.||.|||||-||.-|||||.|||||.|||||.|||||..|||||||.||.|||||.|||||.||.||||||||||||||||||||||||||.|||||.|||||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||..||||||||||.|||-||.||-||||||||||||||-||-||||||||||.||.|||.|||||||.||||||||||.|||||||||||||||.||||||||||||||||||||.||||||||||||||.||||||||||||||||.||||||||.|||||.||.||.|||||||||||||||||..||||.|||||.||.||||||||.|||||||||||||||||-|-||.|||||.||||||||||.||.|||-||-|||||-|-||||||.||||-|||-|||||.|||||||||.|||||||||.||||-||-|-||-||||||.|.|.||||.|||||.||||||||||||.|.-||-|||-||-||||||||.|||||||||||.||.||.|||||.||||||||||||||.||||||||.|||||||||||||||||.||||||||||||||.||||||||||||||.||||||||||||||.||.|..||||||||.||.||-||-|||.||||||||..|.||||||||.|||||||||||.||.|||||.|||||.|||||.||.||.|||||||||||||||||.|||||.|||||.|||||.||.|||||||||||||||||.|||.|||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||.||||.||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.|||.|..|||||||||.|||||||||||||||||||||||||||-||-|||||||||||||||||||||||.|||.|.||.|||||||||||||||-||.|-||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||-||-||||||||||||||||||||.|||||||||||||||||||||||||||||.||||||||.|||||||||||.||.||||||||.|||||||||||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||.||.||||||||||||||||||.|.||||||||.||.||||||||.|||||||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||.||.||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||.||||||||.|||||||||||||||.||||||||.|.|.||||||||||||.|.||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||-||-|||||||.||||||||||||.|.||||.|.||||||||.|||||||||||||||.|.|.|||||||||.||||||||||||||||||||||||||.||||||||.||.||||||||||||||||||||||||||||||||.||||||||||||.||||||||||||||||||||||||||||||||||||||||.||.|.||||..||.||||||||||||.||||||||||||||.||||||||.|.||||||||||||||||||||||-||-|||||.||..||||.|||||.|||||.||.||||||||.||||.|||||||||||||.||||||||||.|.|||||||.||||.||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||.||||||||||||||||||||.|||||||||||||||||||||||||-||-||||||||||||||||||||||||||||||||||||||||-|.|-|||||.|||||||||||.||.||.|||||||||||||||||||||.||||.||||||||.|||||||||||||||||.|||||.|||||||||||.||-|.-|||||||||.|||||.|||||||||||||||||||||||.|||||||||||.||.|||||||-|-||.|||||.|||||||||||.|||||.||.||.||||||||.|||||||||||||-|-||||||||||||||||||||||||||||||||.||||||||.|||||.||.|||||||-|-||||||||||||.|||||||.||||||.||||||||||.|||||||||||||||||||||.||||||||||||||||||||||||||||||.||.||.|.|||||||||||||||.||||||||||||||||||||||.||.||.||.|.|.||.|||||.||||.|||.||.|||.|.|||||||||||.||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||.||||||..|||||||||||.|||||.||||||||.||||||||||||..|..||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||.||.|||||||||||.|||||.|||||||||||.|||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||.|||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.||.||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||..||||||||||||||||||||||.||.||.||||||||||||||||.||||.|.|||||.|||||||||||||||||.|||||||||||||||||||||||||||||||||||.|||||.||||||||.||||||||||||||.||.|||-||-|||||.||.||.|||||-|||.||-|||||||||||||||||||||||||||||||||||||||||.|||.||||||||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||.||.|||||.||||||||.||||||||||||||||||||||||||||||||.|||.||||.||-||.-||.|||||.|||||||||||||||||||||-||||-|||||||-||-|||||..|||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||-||-||.||.|||||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||..||.||||.|||||.|||||.|||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||||||..||.||||.||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.-|..|...-|||||.||||||||||||||||||||||||||----\n",
      "AT-TAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACG-TCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTT-ATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGATGCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTCGTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTAGGCGACGAGCTTGGCACTGATCCTTAT-GAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTGTTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGGCCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTC--ATGCACTTTGTCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTGCTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCC-CTTAAATTCCATAATCAAGACTATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTT-ATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCACCAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCAGACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACTACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAA-TTTATTGTCCAGCATGTCACAATTCAGAAGTAGGACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCGCACTATTGCCTT-TGGAGGCTGTGTGTTCT-CTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCACGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACAACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGATTATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAA-AAAGGTGCCTGGAATATTGGTGAACAGAAAT-CAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCGTGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCCGCTATAACAATACT-AGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTGGCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTTAAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAATTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGTAAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCCTACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAAGCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAAGGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAAATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACCACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAATTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAGAAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATTTGGTGCCACTTCTGCTGCT-CTTCAAC-CTG-AAGAAGAGCAAGAAGAAGATTGGTTAGA-TGATGATAGTCAACAAACTGTTGGTCAACAAGACGGCAGT-GAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTCAACCTCAATTAG-AGATGGAACTTA-CACCAGTTGTT-CAGA--CTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTA-CTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACAGTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTACTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAA-GTGGGTGGTAGTTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGTGAAGACATTCAACTT-CTTAAGAGTGCTTATGAAAATTTTAATCAGCACGA-AGTTCTACTTGCACCATTATTATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAATGTCTA-CTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGTGAAAAGCAAGTTGAACAAAA-GATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTAAACCTTCAGTTGAACAGAGAAAACAAGATGAT-A-A-GAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAACTCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCAGATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAATGCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAATGGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTATTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGCAGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAATATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAACAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTT-GGCTATGTAACACATGG-CTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTTTCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAGAACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACAACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCACCTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTAAGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACAACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGTAAAACATTTTATGTTTTACCTAATGATGACACTCT-ACGTGTTGAGGCTTTTGAGTACTACCACACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAATGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAAATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTAACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAATGAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGTGGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAATTTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTCACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGTGAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAGACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAGTTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAGTTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAACCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATT-TTTCCCTGACTTAAATGGTGATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAACCTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTGTCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTG-AAGTCAGAGGACGCGCAGGGAATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAATAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTTACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTT-GCTACTCATGGTTTAGCTGCTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTACAACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTATTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAA-AGCATCTATGCCGACTACTATAGCAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAACTGATAAATATTATAATTTGGTTTTTACTATTAAGTGTTTGCCTAGGTTCTTTAATCTACTCAACCGCTGCTTTAGGTGTTTTAATGTCTAATTTAGGCATGCCTTCTTACTGTACTGGTTACAGAGAAGGCTATTTGAACTCTACTAATGTCACTATTGCAACCTACTGTACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTAGACACCTATCCTTCTTTAGAAACTATACAAATTACCATTTCATCTTTTAAATGGGATTTAACTGCTTTTGG-CTTAGTTGCAGAGTGGTTTTTGGCATATATTCTTTTCACTAGGTT-TTTCTATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCGATTTCAGCTATGGTTAGAATGTACATCTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGACGGTTGTAATTCATCAACTTGTATGATGTGTTACAAACGTAATAGAGCAACAAGAGTCGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACACAATTGGAATTGTGTTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGATGAAGTTGCGAGAGACTTGTCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCTTACATCGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAAGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAACCTGAGAGCTAATAACACTAAAGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCAAAATCAGCGTCTGTTTACTACAGTCAGCTTATGTGTCAACCTATACTGTTACTAGATCAGGCATTAGTGTCTGATGTTGGTGATAGTGCGGAAGTTGCAGTTAAAATGTTTGATGCTTACGTTAATACGTTTTCATCAACTTTTAACGTACCAATGGAAAAACTCAAAACACTAGTTGCAACTGCAGAAGCTGAACTTGCAAAGAATGTGTCCTTAGACAATGTCTTATCTACTTTTATTTCAGCAGCTCGGCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACATCAATCTGACATAGAAGTTACTGGCGATAGTTGTAATAACTATATGCTCACCTATAACAAAGTTGAAAACATGACACCCCGTGACCTTGGTGCTTGTATTGACTGTAGTGCGCGTCATATTAATGCGCAGGTAGCAAAAAGTCACAACATTGCTTTGATATGGAACGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGTAGTGCTGCTAAAAAGAATAACTTACCTTTTA-AGTTGACATGTGCAACTACTAGACAAGTTGTTAATGTTGTAACAACAAAGATAGCACTTAAGGGTGGTAAAATTGTTAATAATTGGTTGAAGCAGTTAATTAAAGTTACACTTGTGTTCCTTTTTGTTGCTGCTATTTTCTATTTAATAACACCTGTTCATGTCATGTCTAAACATACTGACTTTTCAAGTGAAATCATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCATCTACAGATACTTGTTTTGCTAACAAACATGCTGATTTTGACACATGGTTTAGCCAGCGTGGTGGTAGTTATACTAATGACAAAGCTTGCCCATTGATTGCTGCAGTCATAACAAGAGAAGTGGGTTTTGTCGTGCCTGGTTTGCCTGGCACGATATTACGCACAACTAATGGTGACTTTTTGCATTT-CTTACCTAGAGTTTTTAGTGCAGTTGGTAACATCTGTTACACACCATCAAAACTTATAGAGTACACTGACTTTGCAACATCAGCTTGTGTTTTGGCTGCTGAATGTACAATTTTTAAAGATGCTTCTGGTAAGCCAGTACCATATTGTTATGATACCAATGTACTAGAAGGTTCTGTTGCTTATGAAAGTTTACGCCCTGACACACGTTATGTGCTCATGGATGGCTCTATTATTCAATTTCCTAACACCTACCTTGAAGGTTCTGTTAGAGTGGTAACAACTTTTGATTCTGAGTACTGTAGGCACGGCACTTGTGAAAGATCAGAAGCTGGTGTTTGTGTATCTACTAGTGGTAGATGGGTACTTAACAATGATTATTACAGATCTTTACCAGGAGTTTTCTGTGGTGTAGATGCTGTAAATTTACTTACTAATATGTTTACACCACTAATTCAACCTATTGGTGCTTTGGACATATCAGCATCTATAGTAGCTGGTGGTATTGTAGCTATCGTAGTAACATGCCTTGCCTACTATTTTATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAATACTTTACTATTCCTTATGTCATTCACTGTACTCTGTTTAACACCAGTTTACTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTTTTAGCACATATTCAGTGGATGGTTATGTTCACACCTTTAGTACCTTTCTGGATAACAATTGCTTATATCATTTGTATTTCCACAAAGCATTTCTATTGGTTCTTTAGTAATTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACTTTTGAAGAAGCTGCGCTGTGCACCTTTTTGTTAAATAAAGAAATGTATCT-AAAGTTGCGTAGTGATGTGCTAT-TACCTCTTACGCAATATAATAGATACTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGAGCAATGGATACAACTAGCTACAGAGAAGCTGCTTGTTGTCATCTCGCAAAGGCTCTCAATGACTTCAGTAACTCAGGTTCTGATGTTCTTTACCAACCACCACAAACCTCTATCACCTCAGCTGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAGGGTTGTATGGTACAAGTAACTTGTGGTACAACTACACTTAACGGTCTTTGGCTTGATGACGTAGTTTACTGTCCAAGACATGTGATCTGCACCTCTGAAGACATGCTTAACCCTAATTATGAAGATTTACTCATTCGTAAGTCTAATCATAATTTCTTGGTACAGGCTGGTAATGTTCAACTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTTAAGCTTAAGGTTGATACAGCCAATCCTAAGACACCTAAGTATAAGTTTGTTCGCATTCAACCAGGACAGACTTTTTCAGTGTTAGCTTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCTATGAGGCCCAATTTCACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAACATAGATTATGACTGTGTCTCTTTTTGTTACATGCACCATATGGAATTACCAACTGGAGTTCATGCTGGCACAGACTTAGAAGGTAACTTTTATGGACCTTTTGTTGACAGGCAAACAGCACAAGCAGCTGGTACGGACACAACTATTACAGTTAATGTTTTAGCTTGGTTGTACGCTGCTGTTATAAATGGAGACAGGTGGTTTCTCAATCGATTTACCACAACTCTTAATGACTTTAACCTTGTGGCTATGAAGTACAATTATGAACCTCTAACACAAGACCATGTTGACATACTAGGACCTCTTTCTGCTCAAACTGGAATTGCCGTTTTAGATATGTGTGCTTCATTAAAAGAATTACTGCAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTTGATGTTGTTAGACAATGCTCAGGTGTTAC-TTTCCAAAGTGCAGTGAAAAGAACAATCAAGGGTACACACCACTGGTTGTTACTCACAATTTTGACTTCACTTTTAGTTTTAGTCCAGAGTACTCAATGGTCTTTGTTCTTTTTTTTGTATGAAAATGCCTTTTTACCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCAATGATGTTTGTCAAACATAAGCATGCATTTCTCTGTTTGTTTTTGTTACCTTCTCTTGCCACTGTAGCTTATTTTAATATGGTCTATATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATGGTTGATACTAGTTTGTCTGGTTTTAAGCTAAAAGACTGTGTTATGTATGCATCAGCTGTAGTGTTACTAATCCTTATGACAGCAAGAACTGTGTATGATGATGGTGCTAGGAGAGTGTGGACACTTATGAATGTCTTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGATCAAGCCATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGCCCTATTTTCTTCATAACTGGTAATACACTTCAGTGTATAATGCTAGTTTATTGTTTCTTAGGCTATTTTTGTACTTGTTACTTTGGCCTCTTTTGTTTACTCAACCGCTACTTTAGACTGACTCTTGGTGTTTATGATTACTTAGTTTCTACACAGGAGTTTAGATATATGAATTCACAGGGACTACTCCCACCCAAGAATAGCATAGATGCCTTCAAACTCAACATTAAATTGTTGGGTGTTGGTGGCAAACCTTGTATCAAAGTAGCCACTGTACAGTCTAAAATGTCAGATGTAAAGTGCACATCAGTAGT-CTTACTCTCAGTT-TTGCAACAACTCAGAGTAGAATCATCATCTAAATTGTGGGCTCAATGTGTCCAG-TTACACAATGACATTCT-CTTAGCTAAAGATACTACTGAAGCCTTTGAAAAAATGGTTTCACTACTT-TCTGTTTTGCTTTCCATGCAGGGTGCTGTAGACATAAACAAGCTT-TGTGAAGAAATGCTGGACAACAGGGCAACC-TTACAAGCTATAGCCTCAGAGTTTAGTTCCCTT-CCATCATATGCAGCTTTTGCTACTGCTCAAGAAGCTTATGAGCAGGCTGTTGCTAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTGAAGAAGTCTTTGAATGTGGCTAAATCTGAATTTGACCGTGATGCAGCCATGCAACGTAAGTTGGAAAAGATGGCTGATCAAGCTATGACCCAAATGTATAAACAGGCTAGATCTGAGGACAAGAGGGCAAAAGTTACTAGTGCTATGCAGACAATGCTTTTCACTATGCTTAGAAAG-TTGGATAATGATGCACTCAACAACATTATCAACAATGCAAGAGATGGTTGTGTTCC-CTTGAACATAATACCTCTTACAACAGCAGCCAAACTAATGGTTGTCATACCAGACTAT--AACACATATAAAAATACGTGTGATGGTA-CAACATTTACTTATGCATCAGCATTGTGGGAAATCCAACAGGTTGTAGATGCAGATAGTAAAATTGTTCAACTTAGTGAAATTAGTATGGACAATTCACCTAATTTAGCATGGCCTCTTATTGTAACAGCTTTAAGGGCCAATTCTGCTGTCAAATTACAGAATAATGAGCTTAGTCCTGTTGCACTACGACAGATGTCTTGTGCTGCCGGTACTACACAAACTGCTTGCACTGATGACAATGCG-TTAGCTTACTACAACACAA-CAAAGGGAGGTAGGTTTGTACTTGCACTGTTATCCGATTTAC-AGGATTTGAAATGGGCTAGATTCCCTAAGAGTGATGGAACTGGTACTATCTATACAGAACTGGAACCACCTTGTAGGTTTGTTACAGACACACCTAAAGGTCCTAAAGTGAAGTATTTATACTTTATTAAAGGA-TTAAACAACCTAAATAGAGGTATGGTACTTGGTAGTTTAGCTGCCACAGTACGTCTACAAGCTGGTAATGCAACAGAAGTGCCTGCCAATTCAACTGTA-TTATCTTTCTGTGCTTTTGCTGTAGATGCTGCTAAAGCTTACAAAGATTATCTAGCTAGTGGGGGACAACCAATCACTAATTGTGTTAAGATGTTGTGTACACACACTGGTACTGGTCAGGCAATAACAGTTACACCGGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCATCGTGTTGTCTGTACTGCCGTTGCCACATAGATCATCCAAATCCTAAAGGATTTTGTGACTTAAAAGGTAAGTATGTACAAATACCTACAACTTGTGCTAATGACCCTGTGGGTTTTACACTTAAAAACACAGTCTGTACCGTCTGCGGTATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGCTTCAGTCAGCTGATGCA-CAATCGTTTTTAAACGGGTTTGCGGTGTAAGTGCAGCCCGTCTTACACCGTGCGGCACAGGCACTAGTACTGATGTCGTATACAGGGCTTTTGACATCTACAATGATAAAGTAGCTGGTTTTGCTAAA-TTCCTAAAAACTAATTGTTGTCGCTTCCAAGAAAAGGACGAAGATGACAATTTAATT-GATTCTTACTTT-GTAGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAA-GGATTGTCCAGCTGTTGCTAAACATGACTTCTTT-AAGTTTAGAATAGACGGTGACATGGTACCACATATATCACGTCAACGTCTTACTAAATACACAATGGCAGACCTCGTCTATGCTTTAAGGCATTTTGATGAAGGTAATTGTGACACATTAAAAGAAATACTTGTCACATACAATTGTTGTGATGATGATTATTTCAATAAAAAGGACTGGTATGATTTTGTAGAAAACCCAGATATAT-TACGCGTATACGCCAACTTAGGTGAACGTGTACGCCAAGCTTTGTTAAAAACAGTACAATTCTGTGATGCCATGCGAAATGCTGGTATTGTTGGTGTACTGACATTAGATAATCAAGATCTCAATGGTAACTGGTATGATTTCGGTGATTTCATACAA--ACCACGCCAGG-TAGTGGAGTTCCTGTTGTAGATTCTTATTATTCATTGTTAATGCCTATAT-TAAC-CTTGACCAGGGCTTTAACTGCAGAGTCACATGTTGACACTGA-CTTAACAAAGCC--TTACATTAAGTGGGATTTGTTAAAATATGACTTCACGGAAGAGAGGTTAAAACTCTTT-GACCGTTATTTTAAATATTGGGATCAGACATACCACCCAAATTGTGTTAACTGTTTGGATGACAGATGCATTCTGCATTGTGCAAACTTTAATGTTTTATTCTCTACAGTGTTCCCACCTACAAGTTTTGGACCACTAGTGAGAAAAATATTTGTTGATGGTGTTCCATTTGTAGTTTCAACTGGATACCACTTCAGAGAGCTAGGTGTTGTACATAATCAGGATGTAAACTTACATAGCTCTAGACTTAGTTTTAAGGAA-TTACTT-GTGTATGCTGCTGACCCTGCTATGCACGCTGCTTCTGGTAATCTATTACTAGATAAACGCACTACGTGCTTTTCAGTAGCTGCACTTACTAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAACAAAGACTTCTATGACTTTGCTGTGTCTAAGGGTTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGTAATGCTGCTATCAGCGATTATGACTACTATCGTTATAATCTACCAACAATGTGTGATATCAGACAACTACTATTTGTAGTTGAAGTTGTTGATAAGTACTTTGATTGTTACGATGGTGGCTGTATTAATGCTAACCAAGTCATCGTCAACAACCTAGACAAATCAGCTGGTTTTCCATTTAATAAATGGGGTAAGGCTAGACTTTATTATGATTCAATGAGTTATGAGGATCAAGATGCACTTTTCGCATATACAAAACGTAATGTCATCCCTACTATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACCGTAGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAAAAATTATTGAAA-TCAATAGCCGCCACTAGAGGAGCTACTGTAGTAATTGGAACAAGCAAATTCTATGGTGGTTGGCACAACATGTTAAAAACTGTTTATAGTGATGTAGAAAACCCTCACCTTATGGGTTGGGATTATCCTAAATGTGATAGAGCCATGCCTAACATGCTTAGAATTATGGCCTCACTTGTTCTTGCTCGCAAACATA-CAACGTGTTGTAGCTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCTCAAGTATTGAGTGAAATGGTCATGTGTGGCGGTTCACTATATGTTAAACCAGGTGGAACCTCATCAGGAGATGCCACAACTGCTTATGCTAATAGTGTTTTTAACATTTGTCAAGCTGTCACGGCCAATGTTAATGCACTT-TTATCTACTGATGGTAACAAAATTGCCGATAAGTATGTCCGCAAT-TTACAACACAGACTTTATGAGTGTCTCTATAGAAATAGAGATGTTGACACAG-ACTTTGTGAATGAGTTTTACGCATATTTGCGTAAACATTTCTCAATGATGATACTCTCTGACGATGCTGTTGTGTGTT-TCAATAGCACTTATGCATCTCAAGGTCTAGTGGCTAGCATAAAGAACTTTAA-GTCAGTTCTTTATTATCAAAACAATGTTTTTATGTCTGAAGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGCTCTCAACATACAATGCTAGTTAAACAGGGTGATGATTATGTGTACCTTCCTTACCCAGATCCATCAAGAATCCTAGGGGCCGGCTGTTTTGTAGATGATATCGTAAAAACAGATGGTACACTTATGATTGAACGGTTCGTGTCTTTAGCTATAGATGCTTACCCACTTACTAAA-CATCCTAATCAGGAGTATGCTGATGTCTTTC-ATTTGTACTTACAATACATAAG-AAAGCTACATGATGAG-TTAACAGGACACATGTTAGACATGTATTCTGTTATGCTTACTAATGATAACACTTCAAGGTATTGGGAACCTGAGTTTTATGAGGCTATGTACACACCGCATACAGTCTTACAGGCTGTTGGGGCTTGTGTTCTT-TGCAATTCACAGACTTCA-TTAAGATGTGGTGCTTGCAT-ACGTAGACCATTC-TTATGTTGTAAATGCTGTTACGACCATGTCATATCAACATCACATAAATTAGTCTTGTCTGTTAATCCGTATGTTTGCAATGCTCCAGGTTGTGATGTCACAGATGTGACTCAACTTTACTTAGGAGGTATGAGCTATTATTGTAAATCACATAAACCACCCATTAGTTTTCCATTGTGTGCTAATGGACAAGTTTTTGGTTTATATAAAAATACATGTGTTGGTAGCGATAATGTTACTGACTTTAATGCAATTGCAACATGTGACTGGACAAATGCTGGTGATTACATT-TTAGCTAACACCTGTACTGAAAGACTCAAGCTTTTTGCAGCAGAAACGCTCAAAGCTACTGAGGAGACATTTAAACTGTCTTATGGTATTGCTACTGTACGTGAAGTGCTGTCTGACAGAGAA-TTACATCTTTCATGGGAAGTTGGTAAACCTAGACCACCACTT-AACCGAAATTATGTCTTTACTGGTTATCGTGTAACTAAAAACAGTAAAGTACAAATAGGAGAGTACACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTTTACCGAGGTACAACAACTTACAAATTAAATGTTGGTGATTATTTTGTGCTGACATCACATACAGTAATGCCATTAAGTGCACCTACACTAGTGCCACAAGAGCACTATGTTAGAATTACTGGCTTATACCCAACACTCAATATCTCAGATGAGTTTTCTAGCAATGTTGCAAATTATCAAAAGGTTGGTATGCAAAAGTATTCTACACTCCAGGGACCACCTGGTACTGGTAAGAGTCATTTTGCTATTGGCCTAGCTCTCTACTACCCTTCTGCTCGCATAGTGTATACAGCTTGCTCTCATGCCGCTGTTGATGCACTATGTGAGAAGGCATTAAAATATTTGCCTATAGATAAATGTAGTAGAATTATACCTGCACGTGCTCGTGTAGAGTGTTTTGATAAATTCAAAGTGAATTCAACATTAGAACAGTATGTCTTT-TGTACTGTAAATGCATTGCCTGAGACGACAGCAGATATAGTTGTCTTTGATGAAATTTCAATGGCCACAAATTATGATTTGAGTGTTGTCAATGCCAGATTACGTGCTAAGCACTATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTGCTGAAATTGTTGACACTGTGAGTGCTTTGGTTTATGATAATAAGCTTAAAGCACATAAAGACAAATCAGCTCAATGCTTTAAAATGTTTTATAAGGGTGTTATCACGCATGATGTTTCATCTGCAATTAACAGGCCACAAATAGGCGTGGTAAGAGAATTCCTTACACGTAACCCTGCTTGGAGAAAAGCTGTCTTTATTTCACCTTATAATTCACAGAATGCTGTAGCCTCAAAGATTTT-GGGACTACCAACTCAAACTGTTGATTCATCACAGGGCTCAGAATATGACTATGTCATATTCACTCAAACCACTGAAACAGCTCACTCTTGTAATGTAAACAGATTTAATGTTGCTATTACCAGAGCAAA-AGTAGGCATACTT-TGCATAATGTCTGATAGAGACCTTTATGACAAG-TTGCAATTTACAAGTCTTGAAATTCCACGTAGGAATGTGGCAACTTTACAAGCTGAAAATGTAACAGGACTCTTTAAAGATTGTAGTAAGGTAATCACTGGG-TTACATCCTACACAGGCACCTACACACCTCAGTGTTGACACTAAATTCAAAACTGAAGGT-TTATGTGTTGACATACCTGGCATACCTAAGGACATGACCTATAGAAGACTCATCTCTATGATGGGTTTTAAAATGAATTATCAAGTTAATGGTTACCCTAACATGTTTATCACCCGCGAAGAAGCTATAAGACATGTACGTGCATGGATTGGCTTCGATGTCGAGGGGTGTCATGCTACTAGAGAAGCTGTTGGTACCAATTTACCTTTACAGCTAGGTTTTTCTACAGGTGTTAACCTAGTTGCTGTACCTACAGGTTATGTTGATACACCTAAT-A-AT-ACAGATTTTTCCAGAGTTAGTGCTAAACCACCGCCTGGAGATCAATTTAAACACCTCATACCACTTATGTACAAAGGACTT-CCTTGGAATGTAGTGCGTATAAAGATTGTACAAATGTTAAGTGACACACTTAAAA-ATCTCTCTGACAGAGTCGTATTTGTC-TTATGGGCACATGGCTTTGAG-TTGACATCTATGAAGTATTTTGTGAAAATAGGACCTGAGCGCACCTGTTGTCTATGTGATAGACGTGCCACATGCTTTTCCACTGCTTCAGACACTTATGCCTGTTGGCATCATTCTATTGGATTTGATTACGTCTATAATCCGTTTATGATTGATGTTCAACAATGGGGTTTTACAGGTAACCTACAAAGCAACCATGATCTGTATTGTCAAGTCCATGGTAATGCACATGTAGCTAGTTGTGATGCAATCATGACTAGGTGTCTAGCTGTCCACGAGTGCTTTGTTAAGCGTGTTGACTGGACTATTGAATATCCTATAATTGGTGATGAACTGAAGATTAATGCGGCTTGTAGAAAGGTTCAACACATGGTTGTTAAAGCTGCATTA-TTAGCAGACAAATTCCCAGTTCTTCACGACATTGGTAACCCTAAAG-CTATTA-AGTGTGTACCTCAAG-CTGATGTAGAATGGAAGTTCTATGATGCACAGCCTTGTAGTGACAAAGCTTATAAAATAGAAGAATTATTCTATTCTTATGCCACACATTC-TGACAAATTCACAGATGGTGTATGCCTATTTTGGAATTGCAATGTCGATAGATATCCTGCTAATTCCATTGTTTGTAGATTTGACACTAGAGTGCTATCTAACCTTAA-CTTGCCTGGTTGTGATGGTGGCAGTTTGTATGTAAATAAACATGCATTCCACACACCAGCTTTTGATAAAAGTGCTTTTGTTA-ATTTAAAACAATTACCATT-TTTCTATTACTCTGACAGTCCATGTGAGTCTCATGGAAAACAAGTAGTGTCAGATATAGATTATGTACCACTAAAGTCTGCTACGTGTATAACACGTTGCAATTTAGGTGGTGCTGTCTGTAGACATCATGCTAATGAGTACAGATTGTATCTCG-ATGCTTATAACATGATGATCTCAGCTGGCTTTAGC-TTGTGGGTTTACAAACAATTTGATACTTATAACCTCTGGAACACTTTTAC-AAGACTT-CAGAGTTTAGAAAATGTGGCTTTTAATGTTGTAAATAAGGGACACTTTGATGGACA-ACAGGGTGAAGTACCAGTTTCTATCATTAATAACACTGTTTACACAAAAGTT-GATGGTGTTGATGTAGA-ATTGTTTGAAAATAAAACAACA-TTACCTGTTAATGTAGCATTTGAGCTTTGGGCTAAGCGCAACATTAAACCAGTACCAGAGGTGAAAATACTCAATAATTTGGGTGTGGACATTGCTGCTAATACTGTGATCTGGGACTACAAAAGAGATGCTCCAGCACATATATCTACTATTGGTGTTTGTTCTATGACTGACATAGCCAAGAAACCAACTGAAAC-GATTTGTGCACCACTCACTGTCTTTTTTGATGGTAGAGTTGATGGTCAAGTAGAC-TTATTTAGAAATGCCCGTAATGGTGTTCTTATTA-CAGAAGGTAGTGTTAAAGGTTTA-CAACCATCTGTA-GGTCCCAAACAAGCTAGTCTTAATGGAGTCACATTAATTGGAGAAGCCGTAAAAACACAGTTCAATTA-TTATAAGAAAGTTGATGGTGTTGTCCAACAATTACCTGAAACTTACTTTACTCAGAGTAGAAATTTACAAGAATTTAAA-CCCAGGAGTCA-AA-TGGAAATTGATTTC-TTAGAATTAGCTATGGATGAATTCATTGAACGGTATAAATTAGAAGGCTATGCCTTCGAACATATCGTTTATGGAGATTTTAGTCATAGTCAGTTAGGTGGT-TTACATCTACTGATTGGACTAGCTAAACGTTTTA-AGGAATCACCTTTTGAAT-TAGAAGATTTTATTCCTATGGACAGTACAGTTAAAAACTATTTCATAACAGATGCGCAAACAGGTTCATCTAAGTGTGTGTGTTCTGTTATTGAT-TTATTACTTGATGATTTTGTTGAAATAATAAAATCCCAAGAT-TTATCTGTAGTTTCTAAGGTTGTCAAAGTGACTATTGACTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGCCATGTAGAAACATTTTACCCAAAATTACAATCTAGTCAAGCGTGGCAACCGGGTGTTGCTATGCCTAATCTTTACAAAATGCAAAGAATGCTATTAGAAAAGTGTGACCTTCAAAATTATGGTGATAGTGCAACATTACCTAAAGGCATAATGATGAATGTCGCAAAATATACTCAACTGTGTCAATATTTAAACACA-TTAACATTAGCTGTACCCTATAATATGAGAGTTATACATTTTGGTGCTGGTTCTGATAAAGGAGTTGCACCAGGTACAGCTGTT-TTAAGACAGTGGTTGCCTACGGGTACGCTGCTTGTCGATTCAGATCTTAA-TGACTTTGTCTCTGATGCAGATTCAACTTTGATTGGTGATTGTGCAACTGTACATACAGCTAATAAATGGGATCTCATTATTAGTGATATGTACGACCCTAAGACTAAAAATGTTACAAAAGAAAATGACTCTAAAGAGGGTTTTTTCACTTACATTTGTGGGTTTATACAACAAAAGCTAGCTCTTGGAGGTTCCGTGGCTATAAAGATAACAGAACATTCTTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCCTTTGTTACTAATGTGAATGCGTCATCATCTGAAGCATTT-TTAATTGGATGTAATTATCTTGGCAAACCACGCGAACAAATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAATACAAATCCAATTCAGTTGTCTTCCTATTCTTTATTTGACATGAGTAAATTTCCCCTTAAATTAAGGGGTACTGCTGTTATGTCTTTAAAAGAAGGTCAAATCAATGATATGATTTTATCTCTTCTTAGTAAAGGTAGACTTATAATTAGAGAAAACAACAGAGTTGTTATTTCTAGTGATGTTCTTGTTAACAACTAAACGAACAATGTTTGTTTTTCTTGTTT-T--A-TT-GCCACTAGTCT-CTAGTCAGTGTGTTAATCTT-ACAACCAGAACTCAATT-ACCCCCTGCAT-ACACTAATTCTTTCACAC-GTGGTGTTTATTACCCTGACA-AAGTTT-TCAGATC---CTCAGTTTTACATTCAACTCAGGACT-TGTTCTTACCTTTCTT-TTCCAATGTTA-CTTGGTTCCATGCTA-TACATGTCTCTGGGACCAATGGT---ACTAAGAGGTTTGATAACCCTGTCCTACCATTT-AATGATGGTGTTTATTTTGCTTCCACTGAGA-AGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACT-TTAGATTCGAAGAC--C-CAGTCCCTACTT-ATTGTTAATAACGCTACTAATGTTGTTATTAAAGTCTGTGAATTTCAATTTT-GTAATGATCCATTT-TTGGGTGTTTATTACCACAAAAACAACAAAAGTTGGATGGAA-AGTGAGTT--CAGAGT-TTATTCT-AGTGCGAATAATTGCACTTTTGAATATGTCTCTCAG-CCTTT-TCTTATGGACCTTGAAGGAAAACAG-GGTAAT-TTCAAAAATCTTAGGGAATTTGTGTTTAAGAATAT-T-GATGGTTATTTTAAAATATATTCTAAGCACACGCCTATTAA-TTTAG-TGCGTGATCTC-CCT-CAGGGTTTTTC-G-GCTTTAGAACCATTGGTAGATTTGCCAAT-AGGTATTAACATCACTAGGTTTCA-AACTTTACTTGCT-TTACATAGAAGTTATTTGACTCCTGGTGATTCTTCTTCAGGTTGGACAGCTGGTGCTGCAGCTTATTAT-GTGGGTTATCTTCAACCTAGG-ACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGATGCTGTAGACTGTGCACTTGACCCTCTCTCAGAAACAAAGTGTACGTTGAAATC-CTTCACTGTAGAAAAAGGAATCTATCAAACTTCTAACTTTAGAGTCCAACCAACAGAATCTATTGTTAGATTTCCTAATATTACAAACT-TGTGCCCTTTTGGTGAAG-TTTTTAACGCCACCAGATTTGCATCTGTTTATGCTTGG-AACAGGAAG-AGAAT--CAGCAACTGTGTTGCTGATTATTCTGTCCTATATAATTCCGCATCATTTTCCACTTTTAAGTGTTATGGAGTGTCTCCTACTAAATTAAATGATCTCTGCTTTACTAATGTCTATGCAGATTCATTTGTAATTAGAGG-TGATGAAGTCAGACAAATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAAATTACCAGATGATTTTACAGGCTGCGTTATAGCTTGGAATTCTAACAATCTTGATTCTAAGGTTGGT-GGTAATTATAAT-TACCTGTATAGATTGTT-TAGGAAGTCTAATCTCAAACCTTTTGAGAGAGATATTTCAACTGAAATCTATCA-GGCCGGTA-GCACACCTTGTAATGGTGTTGAAGGTTTTAATTGTTACTTTCCTTTACAATCATATGGTTTCCAACCCACTAATGGTGTTGGTTACCAACCATACA-GAGTAGTAGTACTT-TCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAA--AGTCTACT-AATTTGGTTAAAAA-CAAATGTGTCAATTTCAACTTCAATGGTTTAACAGG-CACAGGTGTTCTT-ACTGAGTCTAACAAAAAGTTTCTGCCTTTCCAACAATTTGGCAGAGA--CATTGCTGACACTACTGATGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTA-ACCAGGTTGCTGTTCTTTATCAGGATGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGT-TTAATAGGGGCTGAACATGTCAACAACT-CATATGAGTGTGACATACCCA-TTGGTGCAGGTATATGCGCTAGTTATCAGACT-CAGACTAATTCTCCTCGGCGGGCACGTAGTGTAGCTAGTCAATCCATCATTGCCTACACTATGTCACTTGGTGCAGAAAATTC-AGTTGCTTACT-CTAATAACTCTATTGCCATACCCACAAATTTTACTATTAGTGTTACCACAGAAATTCTACCAGTGTCTATGACCAAGACATCAGTAGATTGTACAATGTACATTTGTGGTGATTCAACT-GAATGCAGCAATCTT-TTGTTGCAATATGGCAGTTTTTGTACACAATTAAACCGTGCTTTAACTGGAATAGCTGTTGAACAAGACAAAAACACCCAAGAAGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCT-ACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGTGATTGCCTTGG-TGATATTGCTGCTAGAGACCTCATTTGTGCACAAAAGTTTAACGGCCTTACTGTTTTGCCACCTTTGCTCACAGATGAAATGATTGCT-CA-ATACACTTCTGCACTG-TTAGCGGGTACAATCACTTCTGGTTGGACCTTTGGTGCAGGTGCTGCA-TTACAAATACCATTTGCTATGCAAATGGCTTATAGGTTTAATGGTATTGGAGTTACACAGAATGTTCTCTATGAGAACCAAAAATTGATTGCCAACCAATTTAATAGTGCTATTGGCAAAATTCAAGACTCACTTT--CTTCCACAGCAAGTGCACTTGGAAAACTT-CAAGATGTGGTCAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGCTCCAATTTTGGTGCAATTTCAAGTGTTTTAAATGATATCCTTTCACGTCTTGACAAAGTTGAGGCTGAAGTGCAAATTGATAGGTTGATCACAGGCAGACTTCAAAGTTTGCAGACATATGTGACTCAACAATTAATTAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCTACTAAAATGTCAGAGTGTGTACTTGGACAATCAAAAAGAGTTGATTTTTGTGGAAAGGGCTATCATCTTATGTCCTT-CCCTCAGTCAGCACCTCATGGTGTAGTCTTCTTGCATGTGACTTATGTCCCTGCACAAGAAAAGAACTTCACAACTGCTCCTGCCATTTGTCATGATGGAAAAGCACACTTTCCTCGTGAAGGTGTCTTT-GTTTCAAATGGCACACACTGGTTTGTAACACAAAGGAATTTTTATGAACC-ACAAATCATTACTACAGACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAA-TTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGAT-TTAGGTGACATCTCTGGCATTAATGCTTCAGTTGTAAACATTCAAAAAGAAATTGACCGCCTCAATGAGGTTGCCAAGAA-TTTAAATGAATCTCTCATCGATCTCCAAGAACTTGGAAAGTATGAGCAGTATATAAAATGGCCATGGTACATTTGGCTAGGTTTTATAGCTGG-CTTGATTGCCATAGTAATGGTGACAATTATGCTTTGCTGTATGACCAGTTGCTGTAGTTGTCTCAAGGGCTGTTGTTCTTGTGGATCCTGCTGCAAATTTGATGAAGACGACTCTGAGCCAGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATCTTCACAATTGGAACTGTAACT-TTGAAGCAAGGTGAAATCAA-GGATGCTACTCCTTCAGATTTTGTTCGCGCTACTGCAACGATACCGATACAAGCCTCACTCCCTTTCGGATGGCTTATTGTTGGCGTTGCACTTCTTGCTGTTTTTCAGAGCGCTTCCAAAATCATAACCCTCAAAAAGAGATGGCAACTAGCACTCTCCAAGGGTGTTCACTTTGTTTGCAACTTGCTGT-TGTTGTTTGTAACAGTTTACTCACACCTT-TTGCT-CGTTGCTGCTGGCCTTG-AAGCCCCTTTTCTCTATCTTTATGCTTTAGTCT-ACTT-CTTGCAGAGTATAAACTTTGTAAGAATAATAATGAGGCTT-TGGCTT-TGCTGGAAATGCCGTTCCAAAAACCCATTACTTTATGATGCCAACTATTTTCTTTGCTGGCATACTAATTGTTACGACTATTGTATACCTTACAATAGTGTAACTTCTTCAATTGTCATTACTTCAGGTGATGGCACAAC-AAGTCCTATTTCTGAACATGACTACCAGATTGGTGGTTATACTGAAAAATGGGAATCTGGAGTAAAAGACTGTGTTGTATTACACAGTTACTTCACTTCAGACTATTACCAGCTGTACTCAACTCAATTGAGTACAGACACTGGTGTTGAACATGTTACCTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTCCAAATTCACACAATCGACGGTTCATCCGGAGTTGTTAATCCAGTAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTCGTGGTATTCTTGCTAGTTACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTCTGGTCTAAACGAACTAAATATTATATTAGTTTTTCTGTTTGGAACTTTAATTTTAGCCATGGCAGATTCCAACGGTACTATTACCGTTGAAGAGCTTAAAAAGCTCCTT-GAACAATGGAACCTAGTAATAGGTTTCCTATTCCTTACATGGATTTGTCTTCT-ACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTATATAATTAAGTTAATTTTCCTCTGGCTG-TTATGGCCAGTAACTTTAGCTTGTTTTGTGCTTGCTGCTGTTTACAGAATAAATTGGATCACCGGTGGAATTGCTATCGCAATGGCTTGTCTTGTAGGCTTGATGTGGCTCAGCTACTTCATTGCTTCTTTCAGACTGTTTGCGCGTACGCGTTCCATGTGGTCATTCAATCCAGAAACTAACATTCTTCTCAACGTGCCACTCCATGGCACTATTCTGACCAGACCGCTTCTAGAAAGTGAACTCGTAATCGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCTGGACACCATCTAGGACGCTGTGACATCAAGGACCTGCCTAAAGAAATCACTGTTGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGCAACTATAAATTAAACACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTACTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTACATCATAAACCTCATAATTAAAAATTTATCTAAGTCACTAACTGAGAATAAATATTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACGAACATGAAAATTATTCTT-TTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAGTGTGTTAGAGGTACAACAGTACTTTTAAAAGAACCTTGCTCTTCTGGAACATACGAGGGCAATTCACCATTTCATCCTCTAGCTGATAACAAATTTGCACTGACTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGCGTAAAACACGTCTATCAGTTACGTGCCAGATCAGTTTCACCTAAACTGTTCATCAGACAAGAGGAAGTTCAAGAACTT-TACTCTCCAATTTTTCTTATTGTTGCGGCAATAGTGTTTATAACACTTTGCTTCACACTCAAAAGAAAGACAGAATGATTGAACTTTCATTAATTGACTTCTATTTGTGCTTTTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTTATTATCTTTTGGTTCTCACTTGAACTGCAAGATCATAATGAAACTTGTC-ACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAAT-CATCACAACTGTAGCTGCATTTCACCAAGAATGTAGTTTACAGTCATGTACTCAACATCAACCATATGTAGTTGATGACCCGTGTCCTATTCACTTCTATTCTAA-ATGGTATATTAGAGTAGGAGCTAGAAAATCAGCACCTTTAATTGAATTGTGCGTGGATGAGGCT-GGTTCTAAATCACCCATTCAGTACATCGATATCGGTAATTATACAGTTTCCTGTT-TACCTTTTACAATTAATTGCCAGGAACCTAAATTGGGTAGTCTTGTAGTGCGTTGTTCGT-TCTATGAAGACTTTTTAGAGTATCATGACGTTCGTGTTGTTTTAGATTTCATCTAAACGAACAAACTAAAATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCA-TACAATGTAACACAAGC-TTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTA-CACAGGTGCCATCAAATTGGATGACAAAGATCCA-AATTTCAAAGATCAA-GTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACT-CAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAAACTCATGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATCTCACATAGCAATCTTTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACATTTTCACCGAGGCCACGCGGAGTACGATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAATGACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(alignments[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_alignment(alma, thin=50):\n",
    "    plt.figure(figsize=(20, 5))\n",
    "    nmatch = 0\n",
    "    for i in range(len(alma)):\n",
    "        if alma[i] == '|':\n",
    "            nmatch += 1\n",
    "        if ((i+1) % thin) == 0:\n",
    "            plt.plot((i, i), (0, 1), c='g', alpha=((nmatch - 0.25)/thin))\n",
    "            nmatch = 0\n",
    "    plt.show()\n",
    "    return\n",
    "\n",
    "def plot_alignment2(alma, thin=500):\n",
    "    plt.figure(figsize=(20, 5))\n",
    "    nmatch, x, y = 0, [], []\n",
    "    for i in range(len(alma)):\n",
    "        if alma[i] == '|':\n",
    "            nmatch += 1\n",
    "        if ((i+1) % thin) == 0:\n",
    "            x.append(i)\n",
    "            y.append(nmatch/thin)           \n",
    "            nmatch = 0\n",
    "    plt.plot(x, y)\n",
    "    plt.show()\n",
    "    return  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AT-TAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAA\n",
      "||-|.|-|||||.||||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||\n",
      "ATATTA-GGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAA\n"
     ]
    }
   ],
   "source": [
    "als2 = str(alignments[0])[:len(str(alignments[0])) // 3]\n",
    "alma = str(alignments[0])[len(str(alignments[0])) // 3 : 2 * len(str(alignments[0])) // 3]\n",
    "als1 = str(alignments[0])[2 * len(str(alignments[0])) // 3:]\n",
    "\n",
    "print(als1[:75])\n",
    "print(alma[:75])\n",
    "print(als2[:75])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8812907928304012"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alma.count('|') / len(alma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABIQAAAEyCAYAAACLeQv5AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3X2QXed9H/bvs+/LXQDEAguQ4BsoAjRfSkZWOdS7qHeRykRsp25GmmnsZtyo01iddJJ2qrQdx1X6R51M05nMqHGVsccv01hWXafl1JQlmpIlWxFpQbXsiGQk7AJ8ASgCIHYBEMt9x+kf957Du7uAAJFLAtT5fGZ2ePfc557znOf8nt957o93L0pVVQEAAACgPfoudwcAAAAAeGMpCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsMXK4D79y5s9q7d+/lOjwAAADAT53vfve7L1ZVNXmxdpetILR3794cOHDgch0eAAAA4KdOKeWZS2nnT8YAAAAAWkZBCAAAAKBlFIQAAAAAWkZBCAAAAKBlFIQAAAAAWkZBCAAAAKBlFIQAAAAAWuaiBaFSym+UUo6XUr5/gedLKeWfl1KmSil/VUp52+Z3EwAAAIDNcimfEPrNJPf/mOcfSLK/+/PpJP/itXcLAAAAgNfLwMUaVFX1zVLK3h/T5MEkv11VVZXksVLK1aWUa6uq+tEm9fGK9rlvfC5/NPVHeef178z40HiS5OzS2STJ8MBwHj30aN5z43syOjjavOaPD/1x3n3DuzM8MJwkGRsc6+zjhndm2/C2JMnM/Ey+9dy38uG3fDhLq0tJktGB0Tx6+NG876b3Zevw1qZd7euHv5733fS+Ncf6+tNfz3033ZeVcytJkqH+oXzjmW/kg3s/mNVqtdn26KFH896b3tu8bnRgNF+e+nLeft3bm/NaPrecP3v2z/Lht3w4Y4NjSZJz1bkkyWq1mj88+Ie5d8+9TfvVajX/5rl/k4/e8tGcmDvR7Psbz3wj9910X/N7f19//uzZP8tH3vKRnFk80zn+4Gj+5Ok/yfv3vj+LK4tJkvGh8Xzt8Ndy3977Mrc0lyTZedXO5lh/fOiP854b35O+0qlzbhve1oxr3W50YDQHnj+Qj9zykbz48oud155bzcMHH869192b5XPLSZLBvsGm7/X49/rq9Ffz7hvfncG+wWbbnz77p/nQzR9Kf19/s+3LB7+ct1//yhgmaca6HruVcyv55jPfzAdv/mDT96H+oeb8F1YWkiQjAyPNc48ceiTvueE9TX+TNO3rdgsrCxvib355Pt8+8u3cv+/+Jk7X9L30N8eoz3G9vtLXxFXvsdYff/XcanO95pfnO9dkZFvTbvVcJ/6Wzy3nm898c83YDfUPJUnOLJ7ZcA79pb+ZB3Ucrlar+er0V/OuG97V7Hd8aLxpV8f/QN9Ac/x6W5LmWtdjMtA30IzXN5/5Zt6/9/1N3wb6Bpo+1dewN15riyuLG2J9eGA4Xzv8tbzvpvetiYmkkzfqOVybGJ3Io4cfzXtvfG8zJjPzM01/51fmm3Ot5/XccmdurJxbaeZVnS/OLJ7ptLv5g80x6vhK0szvetv2ke2deX3dvc2YDPYNdvZ7y0eaeT06OLohr40MjDTXpD7XhZWF5lrX17O+Xr3XsD5WX+nLnz7zp/nQWz60pl091vX4L64sNmNS57UkzZjUxz/xcqe/Q/1DzbjWxofGm/3Wc35+eb7Jw3UM9/f1b5ivS6tLzbWr+7R1eGsT//W82j66PUmyvLq84fi9c76Ow+2j2/ONp7+R+/bel+Nzx5u2dbvefFGfax0ng/2D+fLUl/OO697RnH8dL0ny2JHH8sC+B5rxOj53vNlvb16rz6He79LqUtOuvl+dXjzd5IQ6X0+MTiRJ5pbn8u3nvp2P7ftYM4Zbh7duyG9Lq0tNXNXjP9Q/lG8+8818YO8Hmv5sHd6aRw8/mnfd8K6MDnRi6OzS2eb49fn0Xq+vHf5a7rvpvmYeJMnDBx/O269/+5o8sD4P9vf1N+Na32u3jWzbkGvqmL9QvNbbeufV+uufJAeeP5C/futfz/MvPZ+kM1++8cw38oG9H2jOdX5lPn9+9M/z8f0fb/ZX9/fs0tnOdd3/QHP8JJ04uP4da9Yo9XzpVd9Xeufm+fpZX7ve+2U9XuvncJI8fuTxfHz/x3Ns7liSTmx8ZeoreecN72zGf3TglRxax9pQ/1DT9/ra9cZG3d/6uf6+/ibm633Mr8w3ebA21D+Ux48+nvv33Z/Z+dmmXT0nZhc623aP7c4jhx7JO69/Z5NXm/0uzzfx2uvbR76dj93ysRw7e2zDseqYPLN4prmG9Zj0l/4N16Q3J9ZrmPqazy/PN+PVe/71Oqi2e2x3/uCpP8hfu+avNfuox+5d17+rGc89W/bkDw/+Ye7Zc0+Tf3rvUfX1qnNOf+lvjt+75qnnS73tzOKZ5r5Sx+n8ynwTV3UOmVuea+4NtTq+xgbHmn1cM35NkuS5M8/ley98Lw/+zIPNseaW5vK1w1/LO294ZT0+Mz+Tx48+no++5aPN2qSOk7dd87YmTy2tLuXbR769Jk88d+a5JMmxs8ea61rr7+vPd45+Jx/b97Fmrr348ot57Mhj+egtH23WsttHt+exI4+tyWHXbrk2333+u/nZa382T514KknWrNG+9ey38oGbP9DsY3ZhtplD9bnW66v33vTe3Lbztk67+dl8dfqrefv1b18Tp/Varh7/1Wq1uU71mNQWVhaaPFTH8OnF05114y33Z2yos+Y6Pne8uSa9ueaRQ4901mvdfDXUP7ThHlKf19jgWPNc7/uceh+1nVftbOK6fu3iymIeP/p4Htj3QNOuN4bq8RzsG2zuDYP9g2uOX5/r37j1bzTze355vlk31vey4YHhZn1TG+wf7Ix/T27sK33NHO6N3fq9VH2fqPPl8rnlJr/3zo16fVHni9HB0Q1rzr7St2Etf6F1YNJ5P1TPud77QD3WvWuu9WuJ3ntzfazRgdEN74O2Dm9tttV5ql6jz8zPNP09vXC6Oa9Hph/Ju254V3NtklfuL739rOOvvr+fb1t/6d/Qp941ch1D9bj2lb7O+uLmtffXx488ngf2P5DPvuezaZOLFoQuwXVJnuv5/Uh324aCUCnl0+l8iig33njjJhz68nvyxJM5NHsoY0NjaxbHSecN0aFTh7J1ZOuaG+uh2UPZMrylScTbhrdlenY6Y0NjzQ372NyxHJo9lCdOPNEE7/jQeKZnp7N9dHt2jO5Ikrxw9oVmv4dPHc7Vo1evOdb0zHQmRieyvNpJTqODozk0eyi7x3Y3k2i4fzjTs9OZuGqied3Y4FimZ6czOjjanNfi6mKmZ6ez58SeZludbFar1UzPTGdkYKR5brVazdTMVG7cdmOOvnR0TT/rN0ZJZxLX53py/uSac91x1Y4msWwb3pZDs4eyc2xnTi2cSpJct+W6ph/TM9PZNrKtSWwTIxOZmp3K1pGt2TO+Z81+bz9ze46eOdq89uDMwYwNjWVxdbEZk7rv9bZe07PT2TqyNcP9w822qZmp7NmyZ81CeGp2KqODo7l65Oo1r524aqIZu+XV5RyaPZTJscnmtfXxd43tahJ7nVhHBkYyPTOd7SPb1/StHq+63dzy3Ib4O7t0NodmD+XJE082Y5h0YvKa8Wuam9LIwEhzjusN9A00cdV7rPXHXzm30lyv+iayY3RHc171+ddxdc34Nc3513Pj5PzJDefQX/qbedAbh9Oz0xkfGm/iup5X20e3N/E/2D/Y9LPeVo9d75jUN6ezS2dz+NThTByfaMZmsH+w6VN9Dr3xWltYWdgQ6yMDIzk0eyhXj1y9JiaS5NTCqWYO13aP7c7UzFQmRieaMTk2dyyHTnXmSz2uV49cnemZ6Vw7fm2Tf+q4euLEE02+ODl/shnrWh1f9TjsObGn2bbrql2ZmpnKyMBIMyZ1vrjhxRuaeT0+NL4hr9U5ZHxovDnXOk6uGb+muZ71GPZew/pYA30DmZ6dzrUnrl2Ta+r5Um9bWFloxqS3IFTnujpO6v7WebDeXo9hHZv1vKrnS+9Y1/mqd77OL883167u047RHU3817Gze2x309+pmakm39fjenDmYK4Zv6a5hvX137NlT5498+ya67Tjqh1r8kV9/eucNDIwkqmZqWwZ2rLmTVdtamYqUzNTTX+fPfNss9/evFafQ71tcXWxOf+Jkc49Y2ZhpskJdb6uz/X04ulMzU7lphdvWpsHZjvnVfdpYWWhiaveIvjBmYO5dvzapj87r9qZqZmpTF412cTQqYVTmZqZWnNf6y3uTs9OZ+dVO5t5UJ//VUNXrckD6/NgneuuHb+2eeO+Y3THhlxTx/yF4rXetn5e9V7/JDk4czCHZw9nena6uV7TM9PZNbZrTa6dmpnKwZMHm/3V/a3H+uDJg2sKXdMz09kyvGXNGqW+v/Wq7yu9c7POP2vazU5ncmxyzf2y9/x753DSuQ/+8OQP88zpZ5Ik14xfk6nZqYwPjzfjX+fQybHJZjzr++DY0Fhz7Xpjo+5v/dxA30AT8/U+6pxz/YvXN/2p59rPzP5Ms4aqx3VqZqq51nNLnW07RnesWdclnfiq47XX1MxU9m3f15xrHcP7Z/Y3MXly/mQOzhzM1MxUM6/7S/+Ga1KvobaNbGvWMPU1r48/OTa54fx74+rl5ZdzcOZgBvoHmn2MDIzk4MmDmRiZaK7h4upiDs4czPjQeHOuvfmxvl51zqnvw715MEkzX+pt9T1nz4lX1kZzy3NN/q37enrx9IZ7U923bcPbMjU7lRtPvrIe++HJH2ZqZirPnXmu2W+dB3aN7cq2kU7fj509lqmTU9k/sb+J63pebx3amt3ja3Py3qv3Nn2anunMw2dOP5OpmancfPXNTd8G+gY6MXTqZ5q5dvTM0RycOZh9E/ua/+F4zfg1mZqZylu2v2XNa589/Wz2bNmTqZmpJFmzRpuancpNV9/U7OP4y8ebOVSfaz03J8cms2VoS+dc545tWC/U+9s+un3Na+vrVOeV2tzyXJOH6hh+8eUXMz0znae2PdXERH2/2HNiz5pcc2j2ULYOv7Jeq9eSvfO6XufXa7TeXFKvG3rXTXvG9zRxXb+2vl889eJTTbveGKqPX+eQ3WO7Nxy/Ptepmakm5ut5tX10+5pcU7/Pqw33Dzdrnt7rWq9DemO3fi/Vm9eS7tq3m997i/r1/bW30LN+zTnQN7BhLX+hdWDyyn1zYnRizX2gvl69a6563dT7P3zrXFfvt75Ove+D6vv69tHtTZ6qj1W/p33yxJNr3ucdOnUoVw1dtWHNMTE60awbTy2cauKvPv/zbavvP719Wr9G3jaybc3a4PCpw9lxbMeG+ffvXvx3aZs39Eulq6r6QlVV91RVdc/k5OQbeWgAAAAAujajIHQ0Se//Irm+uw0AAACAK9BmFIQeSvLz3X9t7B1JTrfl+4MAAAAA3owu+h1CpZTfTfL+JDtLKUeS/KMkg0lSVdWvJXk4yceTTCV5Ocnffr06CwAAAMBrdyn/ytinLvJ8leSXNq1HAAAAALyu3tAvlQYAAADg8lMQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGiZSyoIlVLuL6X8oJQyVUr57Hmev7GU8vVSyl+UUv6qlPLxze8qAAAAAJvhogWhUkp/ks8neSDJHUk+VUq5Y12z/yHJl6qq+tkkn0zyv212RwEAAADYHJfyCaF7k0xVVXWoqqqlJF9M8uC6NlWSrd3H25I8v3ldBAAAAGAzDVxCm+uSPNfz+5Ekb1/X5leSfLWU8l8mGUvy4U3pHQAAAACbbrO+VPpTSX6zqqrrk3w8ye+UUjbsu5Ty6VLKgVLKgRMnTmzSoQEAAAD4SVxKQehokht6fr++u63XLyb5UpJUVfXtJCNJdq7fUVVVX6iq6p6qqu6ZnJx8dT0GAAAA4DW5lILQd5LsL6XcXEoZSudLox9a1+bZJB9KklLK7ekUhHwECAAAAOAKdNGCUFVVK0k+k+QrSZ5K518Te6KU8rlSyie6zf5Bkr9TSvnLJL+b5D+tqqp6vToNAAAAwKt3KV8qnaqqHk7y8Lptv9zz+Mkk797crgEAAADwetisL5UGAAAA4E1CQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFrmkgpCpZT7Syk/KKVMlVI+e4E2f7OU8mQp5YlSyr/a3G4CAAAAsFkGLtaglNKf5PNJPpLkSJLvlFIeqqrqyZ42+5P8wyTvrqpqtpSy6/XqMAAAAACvzaV8QujeJFNVVR2qqmopyReTPLiuzd9J8vmqqmaTpKqq45vbTQAAAAA2y6UUhK5L8lzP70e623rdmuTWUsq3SimPlVLuP9+OSimfLqUcKKUcOHHixKvrMQAAAACvyWZ9qfRAkv1J3p/kU0n+ZSnl6vWNqqr6QlVV91RVdc/k5OQmHRoAAACAn8SlFISOJrmh5/fru9t6HUnyUFVVy1VVHU7yw3QKRAAAAABcYS6lIPSdJPtLKTeXUoaSfDLJQ+va/N/pfDoopZSd6fwJ2aFN7CcAAAAAm+SiBaGqqlaSfCbJV5I8leRLVVU9UUr5XCnlE91mX0lyspTyZJKvJ/lvqqo6+Xp1GgAAAIBX76L/7HySVFX1cJKH12375Z7HVZK/3/0BAAAA4Aq2WV8qDQAAAMCbhIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMtcUkGolHJ/KeUHpZSpUspnf0y7/6iUUpVS7tm8LgIAAACwmS5aECql9Cf5fJIHktyR5FOllDvO025Lkr+X5PHN7iQAAAAAm+dSPiF0b5KpqqoOVVW1lOSLSR48T7t/nORXkyxsYv8AAAAA2GSXUhC6LslzPb8f6W5rlFLeluSGqqr+cBP7BgAAAMDr4DV/qXQppS/JP0vyDy6h7adLKQdKKQdOnDjxWg8NAAAAwKtwKQWho0lu6Pn9+u622pYk/16SPymlPJ3kHUkeOt8XS1dV9YWqqu6pquqeycnJV99rAAAAAF61SykIfSfJ/lLKzaWUoSSfTPJQ/WRVVaerqtpZVdXeqqr2JnksySeqqjrwuvQYAAAAgNfkogWhqqpWknwmyVeSPJXkS1VVPVFK+Vwp5ROvdwcBAAAA2FwDl9KoqqqHkzy8btsvX6Dt+197twAAAAB4vbzmL5UGAAAA4M1FQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFrmkgpCpZT7Syk/KKVMlVI+e57n/34p5clSyl+VUh4tpdy0+V0FAAAAYDNctCBUSulP8vkkDyS5I8mnSil3rGv2F0nuqarq7iS/n+SfbHZHAQAAANgcl/IJoXuTTFVVdaiqqqUkX0zyYG+Dqqq+XlXVy91fH0ty/eZ2EwAAAIDNcikFoeuSPNfz+5Hutgv5xSRfPt8TpZRPl1IOlFIOnDhx4tJ7CQAAAMCm2dQvlS6l/CdJ7knyT8/3fFVVX6iq6p6qqu6ZnJzczEMDAAAAcIkGLqHN0SQ39Px+fXfbGqWUDyf575PcV1XV4uZ0DwAAAIDNdimfEPpOkv2llJtLKUNJPpnkod4GpZSfTfK/J/lEVVXHN7+bAAAAAGyWixaEqqpaSfKZJF9J8lSSL1VV9UQp5XOllE90m/3TJONJ/s9SyvdKKQ9dYHcAAAAAXGaX8idjqarq4SQPr9v2yz2PP7zJ/QIAAADgdbKpXyoNAAAAwJVPQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZS6pIFRKub+U8oNSylQp5bPneX64lPJ73ecfL6Xs3eyOAgAAALA5LloQKqX0J/l8kgeS3JHkU6WUO9Y1+8Uks1VV7Uvyvyb51c3uKAAAAACb41I+IXRvkqmqqg5VVbWU5ItJHlzX5sEkv9V9/PtJPlRKKZvXTQAAAAA2y8AltLkuyXM9vx9J8vYLtamqaqWUcjrJjiQv9jYqpXw6yaeT5MYbb3yVXb6y3DF5R549/Wzu3nV3xofGkyRnl84mSYYHhnP87PHcvevujA6ONq/50Us/yt277s7wwHCSZGxwLM+ceiZ3774724a3JUl2j+3OS4sv5c7JO7O0upQkGR0YzQtnX8hdu+7K1uGtSZJdY7ua/Z58+eSGYx2fO567dt2VlXMrSZKh/qG8+PKLuXPyzqxWq822H730o9w5eWfzutGB0Tx96uk157V8bjmnF07njsk7MjY4liQ5V51LkqxWq5menV7TfrVazemF07lt523ZMbqj2ffs/Gzu3nV383t/X3/OLJ7JnZN35szimc7xB0dz8uWTuWvXXVlcWUySjA+N59jZY7lz8s7MLc0lSXZetbM51vMvPZ+7dt2VvtKpc24b3panTz2dOyfvbNqNDoxmZn4m12+9PiMDI53XnlvN/on9uXPyziyfW06SDPYNNn2vx7/Xc6efy1277spg32CzbWZ+JrfvvD39ff3NtsOzh3P37lfGpL7+d07e2YzdyrmVnFo4tabvQ/1DOT53PHdM3pGFlYUkafo71D+U5848t6a/9fW/a9ddTbuFlYUN8Te/PJ+5pbncMXlHE6dJcnrxdO6cvDP9pf+VY3TPcb2+0tfEVe+x1h9/9dxqc73ml+c712RkW3Neq+c68bd8bjmz87Od4/e9cvwkObN4ZsM59Jf+Zh7UcbharebImSO5e/fdzX7Hh8abdnX8D/QNNP2styWdOds7JgN9A814nVk4k7t33d30baBvoOlTfQ1747W2uLK4IdaHB4ZzYu7Ehpio+1DP4drE6ESef+n53Dl5ZzMmu8d25+zi2c64rsw353ri5RO5feftmVvuzI2VcyvNvKrzxZnFM3nx5Rdzx+QrH/Ks4ytJM7/rbdtHtnfm9e67mzEZ7BvM7Pzsmnk9Oji6Ia+NDIw016Q+14WVheZa19ezvl6917A+Vl/py8zLM2viZbVazYm5E2vm0OLKYjMmdV5L0uS6+vg7rur0d6h/KC+cfSF3735lrMeHxpvYrOf8/PJ8k4frGO7v698wX5dWl5prV/dp6/DWJv7rebV9dHuSZHl1OUdfOprbd97eHL/u0207b2vicPvo9jz/0vPZv2N/to1sa9rWsdabL+rrX8fJYP9gpman1px/HS91LOyb2NeM17aRbc1+e/NafQ71fpdWl5rxr+9XpxdPNzmhztcToxNJkrnluZxdPJvbdt7WjOHW4a15/qXnc+uOW5tYW1pdauKqHv86D96287amP1uHt2bfxL7s37E/owOdGDq7dDZHXzpQLEw+AAAK20lEQVSa23be1pxP7/V64ewLa+ZBkkzNTG3IA+vzYH9ffzOuu8d2N+O0PtfUMX+heK239c6r9dc/SeaW5nLz9pub/S2sLGRmfqYzX7rnOr/SyeH7d+xv9lf39+zS2by0+FL279jfHD9Jpmen18TB2aWzzf2tV31f6Z2bdf7pVV//3vtlPV7r53CSvLz0cm7dcWsTwxOjE3nm1DNrxn90YLTZbx1rQ/1DOXzq8Jo1T29s1P2tn+vv629ivt7H/Mp8ZudnN8y104unc/P2m3P1yNVNu3pO1Hli99ju7JvYl1t33Nrk1Wa/y/NNvPaaXZjNLRO3NGM91D+UM4tncsvELU1Mnlk8k7mlueyb2Jctw1uaa7j+mqxWq3n29LNr1jD1NZ9fnm/Gq/f8nzn9zJp97B7b3axv6n0M9Q9l/6n9a+bani17sn9i/5r803uPqq9XnXP6S39z/N41Tz1f6m1nFs8095U6TudX5pv8W+eQueW55t5Qq+NrbHAsJ18+mVt33Jprxq9J0plrCysLuWHrDc2xto9sz76JfWvGf+vw1swszOTmq29u1ib9ff25ZeKW7JvY1+SppdWlvHD2hdy07aZmPOvYHB8az5nFM7l1x61N3+r7wN6r9zZzbcvQlpxaOJW3bH9LM9bbR7fn2Nyx3LTtpua1k2OTuXHbjdk93omvJGvWaMfOHluzj90Lu5s5VJ/r6rnO/fK2nbfluq3XJUmuGryqyY29cfr0qafXjP9qtdpcp3pMagsrC00eumrwqiSdvDi3PJfbd96esaHOmmvbyLbmuvbmmudfej537767yVf1+4s7Ju9Ys66rr2v9XO/7nCNnjqyJg51X7Wziun7t4spi5pfn18zr3hiqx3Owb7C5Nwz2d/LU7sXdzbkurixm38S+Zn7PL88368b6XjY8MJxjZ4/lrbvf2hxrsH8wL7784pptfaWvWYf0xm79XqrO63W+XD633OT33rlRry/qfDE6OLphzdlX+jas5S+0Dkw674fqXN57H6ivV++aq1439d7z61xXH2t0YHTD+6Ctw1ubWKvzVL1G3z22u+nv6YXTzXk9f+b5vHX3W5trk3TeS/WuG88unW3ir76/n29bf+nf0KfeNXL9HrEe177Sl1Pzp9bE6/zKfOaX5zfk9TYoVVX9+Aal/FyS+6uq+s+6v/+tJG+vquozPW2+321zpPv7dLfNi+fbZ5Lcc8891YEDBzbhFAAAAABIklLKd6uquudi7S7lT8aOJrmh5/fru9vO26aUMpBkW5KTl9ZVAAAAAN5Il1IQ+k6S/aWUm0spQ0k+meShdW0eSvIL3cc/l+Rr1cU+egQAAADAZXHR7xDqfifQZ5J8JUl/kt+oquqJUsrnkhyoquqhJL+e5HdKKVNJZtIpGgEAAABwBbqUL5VOVVUPJ3l43bZf7nm8kOQ/3tyuAQAAAPB6uJQ/GQMAAADgp4iCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtEypquryHLiUE0meuSwH3zw7k7x4uTsBP4YY5UomPrnSiVGuZOKTK50Y5Ur20x6fN1VVNXmxRpetIPTToJRyoKqqey53P+BCxChXMvHJlU6MciUTn1zpxChXMvHZ4U/GAAAAAFpGQQgAAACgZRSEXpsvXO4OwEWIUa5k4pMrnRjlSiY+udKJUa5k4jO+QwgAAACgdXxCCAAAAKBlFIQAAAAAWkZB6FUqpdxfSvlBKWWqlPLZy90f2qOU8nQp5d+WUr5XSjnQ3TZRSnmklHKw+9/t3e2llPLPu3H6V6WUt/Xs5xe67Q+WUn7hcp0Pb36llN8opRwvpXy/Z9umxWQp5d/vxvxU97XljT1D3swuEJ+/Uko52s2j3yulfLznuX/YjbUflFI+1rP9vPf9UsrNpZTHu9t/r5Qy9MadHW92pZQbSilfL6U8WUp5opTy97rb5VCuCD8mRuVRLrtSykgp5c9LKX/Zjc//sbv9vDFVShnu/j7VfX5vz75+orj9qVFVlZ+f8CdJf5LpJG9JMpTkL5Pccbn75acdP0meTrJz3bZ/kuSz3cefTfKr3ccfT/LlJCXJO5I83t0+keRQ97/bu4+3X+5z8/Pm/EnyviRvS/L9nm2bFpNJ/rzbtnRf+8DlPmc/b56fC8TnryT5r8/T9o7uPX04yc3de33/j7vvJ/lSkk92H/9akv/icp+znzfPT5Jrk7yt+3hLkh9241AO9XNF/PyYGJVH/Vz2n25eG+8+HkzyeDffnTemkvzdJL/WffzJJL/XffwTx+1Py49PCL069yaZqqrqUFVVS0m+mOTBy9wn2u3BJL/VffxbSf6Dnu2/XXU8luTqUsq1ST6W5JGqqmaqqppN8kiS+9/oTvPToaqqbyaZWbd5U2Ky+9zWqqoeqzp37N/u2Rdc1AXi80IeTPLFqqoWq6o6nGQqnXv+ee/73U9afDDJ73df3xvrcFFVVf2oqqr/r/v4pSRPJbkucihXiB8Toxcij/KG6ebCs91fB7s/VS4cU7259feTfKgbgz9R3L7Op/WGUhB6da5L8lzP70fy4xMjbKYqyVdLKd8tpXy6u213VVU/6j5+Icnu7uMLxaoY5vW2WTF5Xffx+u3wWn2m+yc3v1H/OU5+8vjckeRUVVUr67bDT6z7pws/m87/4ZZDueKsi9FEHuUKUErpL6V8L8nxdIrh07lwTDVx2H3+dDox2Nr3TApC8Obznqqq3pbkgSS/VEp5X++T3f8DWF2WnsF5iEmuQP8iyS1J3prkR0n+l8vbHdqulDKe5P9K8l9VVXWm9zk5lCvBeWJUHuWKUFXValVVb01yfTqf6LntMnfpTUVB6NU5muSGnt+v726D111VVUe7/z2e5F+nk/iOdT8Wnu5/j3ebXyhWxTCvt82KyaPdx+u3w6tWVdWx7gLyXJJ/mU4eTX7y+DyZzp/sDKzbDpeslDKYzhvt/6Oqqj/obpZDuWKcL0blUa40VVWdSvL1JO/MhWOqicPu89vSicHWvmdSEHp1vpNkf/fby4fS+UKqhy5zn2iBUspYKWVL/TjJR5N8P534q/9FkV9I8v90Hz+U5Oe7/yrJO5Kc7n4E/StJPlpK2d79iO9Hu9tgs2xKTHafO1NKeUf3b7x/vmdf8KrUb7S7/sN08mjSic9Pdv8VkpuT7E/nC3nPe9/vfnLj60l+rvv63liHi+rmtV9P8lRVVf+s5yk5lCvChWJUHuVKUEqZLKVc3X08muQj6XzP1YViqje3/lySr3Vj8CeK29f/zN44AxdvwnpVVa2UUj6Tzs23P8lvVFX1xGXuFu2wO8m/7tybM5DkX1VV9UellO8k+VIp5ReTPJPkb3bbP5zOv0gyleTlJH87Saqqmiml/ON0klySfK6qqkv90lVYo5Tyu0nen2RnKeVIkn+U5H/O5sXk303ym0lG0/kXcr78Op8SP0UuEJ/vL6W8NZ0/w3k6yX+eJFVVPVFK+VKSJ5OsJPmlqqpWu/u50H3/v03yxVLK/5TkL9J54wSX6t1J/laSf9v9Dowk+e8ih3LluFCMfkoe5QpwbZLfKqX0p/Nhly9VVfX/llKezPlj6teT/E4pZSqdf3Dik8mrjtufCqVTEAMAAACgLfzJGAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDL/P+eOAbB35iIWAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1440x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_alignment(alma[:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'alma' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-2-6b9ec4432eca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mplot_alignment2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'alma' is not defined"
     ]
    }
   ],
   "source": [
    "plot_alignment2(alma[:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC\n",
      "AGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTAGCTGTCGCTCGGCTGCATGCCTAGTGCACCTACGCAGTA\n",
      "11676.0\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__cmp__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_format_generalized',\n",
       " '_format_psl',\n",
       " 'aligned',\n",
       " 'format',\n",
       " 'path',\n",
       " 'query',\n",
       " 'score',\n",
       " 'target']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(alignments[0].query[:75])\n",
    "print(alignments[0].target[:75])\n",
    "print(alignments[0].score)\n",
    "#print(alignments[0].path)\n",
    "#print(alignments[0].aligned)\n",
    "dir(alignments[0])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Final exercise (5/10 points): The misterious patient\n",
    "\n",
    "A few weeks ago, a patient was admitted to the hospital where you work. The patient was coughing, and had difficulty breathing, as well as a fever. Your colleagues Oscar and Maria, from the genomics platform at the hospital, have been able to isolate a viral genome from the patient; this virus is thought to be responsible for the disease. **You are expected to identify what the virus is and, thus, provide a diagnosis**. You are also expected to compare the virus' genome with the genome of similar viruses to try to **understand where the disease came from and how is it different from known diseases**. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **Open and read genome of the virus** using Biopython. The genome is available as a FASTA file from `Files/patient.fasta`). \n",
    "2. Go to the BLAST [web page](https://blast.ncbi.nlm.nih.gov/Blast.cgi) and **BLAST the sequence manually**. BLAST against the standard nucleotide collections using the best algorithm for the alignment of highly similar sequences. After blasting, obtain, show (as a screenshot), and discuss the following visualizations:\n",
    " - A **graphic summary of the alignments**. \n",
    " - A **distance tree** of the results.\n",
    "\n",
    "\n",
    "3. BLAST the sequence again, **using Biopython** this time. Be aware that this can take several minutes to complete (~10 minutes), so be patient. Upon completion of the BLAST search, immediately **save the BLAST results in a file** named `Files/blast_results.xml`.\n",
    "4. Open the results and **display all alignments with percentage identity larger than 99.5%**. Which is your diagnosis for the patient?\n",
    "5. **Display all alignments with percentage identity larger than 85.0% and smaller than 99.5%**. Which animal do you think may have transmitted the disease to humans?\n",
    "6. Choose one of the sequences with percentage similarity between 85% and 90%. Get the whole **GeneBank entry** for this sequence **from Entrez using Biopython**, and display it. List, also, the coding regions of this genome indicating start and end of each coding sequence (CDS).\n",
    "7. Do a **global pairwise alignment** between the genome in question 6 and the patient's viral genome using the **Needleman-Wunsch algorithm in Biopython**. Visualize the aligment using the `plot_alignment()` function defined above, and explain in which region of the original virus (question 6) has the new virus mutated. For extra credit, write a new `plot_alignment_new()` function that helps you address this question  more precisely (e.g. by marking the start and end of CDSs in the genome of question 6)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 1 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Save the graphic summary of the alignments as Files/blast_align.png and, after Ctrl+Enter this cell, the image should show up here!](Files/blast_align.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Save the distance tree of the alignments as Files/blast_tree.png and, after Ctrl+Enter this cell, the image should show up here!](Files/blast_tree.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Discuss the results here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 3 code here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 4 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 4 here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 5 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 5 here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 6 code here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 7 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 7 here*"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
