{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Today's session\n",
    "\n",
    "In today's session, we will get to know some of the genomics tools that were introduced in the theoretical lectures. In particular, we will see how to work with sequences, align them, and so on. To do this, we will be using `Biopyton`, a comprehensive Python module that allows you to do these and many other things. For a complete tour of `Biopython`, take a look at the [complete tutorial](http://biopython.org/DIST/docs/tutorial/Tutorial.html).\n",
    "\n",
    "In the final exercise for this class you will be working on the following case study:\n",
    "\n",
    ">A few weeks ago, a patient was admitted to the hospital where you work. The patient was coughing, and had difficulty breathing, as well as a fever. Your colleagues Oscar and Maria, from the genomics platform at the hospital, have been able to isolate a viral genome from the patient; this virus is thought to be responsible for the disease. You are expected to identify what the virus is and, thus, provide a diagnosis. You are also expected to compare the virus' genome with the genome of similar viruses to try to understand where the disease came from and how is it different from known diseases. \n",
    "\n",
    "# What is Biopython?\n",
    "\n",
    "The Biopython Project is an international association of developers of freely available Python tools for computational molecular biology.\n",
    "\n",
    "![Biopython](Media/biopython_logo.png)\n",
    "\n",
    "\n",
    "The main Biopython releases have lots of functionality, including:\n",
    "\n",
    "* The ability to parse bioinformatics files into Python utilizable data structures, including support for the following formats:\n",
    "    - Blast output – both from standalone and WWW Blast\n",
    "    - Clustalw\n",
    "    - FASTA\n",
    "    - GenBank\n",
    "    - PubMed and Medline\n",
    "    - ExPASy files, like Enzyme and Prosite\n",
    "    - SCOP, including ‘dom’ and ‘lin’ files\n",
    "    - UniGene\n",
    "    - SwissProt \n",
    "* Files in the supported formats can be iterated over record by record or indexed and accessed via a Dictionary interface.\n",
    "* Code to deal with popular on-line bioinformatics destinations such as:\n",
    "    - NCBI – Blast, Entrez and PubMed services\n",
    "    - ExPASy – Swiss-Prot and Prosite entries, as well as Prosite searches \n",
    "* Interfaces to common bioinformatics programs such as:\n",
    "    - Standalone Blast from NCBI\n",
    "    - Clustalw alignment program\n",
    "    - EMBOSS command line tools \n",
    "* A standard sequence class that deals with sequences, ids on sequences, and sequence features.\n",
    "* Tools for performing common operations on sequences, such as translation, transcription and weight calculations.\n",
    "* Code for dealing with alignments, including a standard way to create and deal with substitution matrices.\n",
    "* Code making it easy to split up parallelizable tasks into separate processes.\n",
    "* GUI-based programs to do basic sequence manipulations, translations, BLASTing, etc.\n",
    "* Extensive documentation and help with using the modules, including the tutorial, on-line wiki documentation, the web site, and the mailing list.\n",
    "* Integration with BioSQL, a sequence database schema also supported by the BioPerl and BioJava projects."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Biopython sequences\n",
    "\n",
    "We will start with a quick introduction to the `Biopython` class for dealing with sequences. Most of the time, when we think of sequences what we have in my mind is a string of letters like 'AGTACACTGGT'. In `Biopython` you can create a `Seq` instance from such a sequence as follows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AGTACACTGGT')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "my_seq = Seq(\"AGTACACTGGT\")\n",
    "display(my_seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sequence alphabet\n",
    "\n",
    "The sequence above seems to represent a fragment of DNA (that is, a sequence of bases), but it could also be a protein, since A, G, T, and C could also represent aminoacids. To specify the type of sequence we are dealing with, the `Seq` class has an `alphabet` attribute. Let's see how it works:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AGTACACTGGT', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "IUPACUnambiguousDNA()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from Bio.Alphabet import IUPAC\n",
    "\n",
    "my_seq = Seq(\"AGTACACTGGT\", alphabet=IUPAC.unambiguous_dna)\n",
    "display(my_seq)\n",
    "display(my_seq.alphabet)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transcription and  translation\n",
    "\n",
    "So, *now* we know that we are dealing with a DNA sequence. And, in fact, we can do things with this sequence that we could not do with a sequence of aminoacids. For example, assuming that the `coding_dna` sequence corresponds to the 5'-to-3′ strand of the DNA, we can obtain the corresponding 5'-to-3' complementary strand:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('CTATCGGGCACCCTTTCAGCGGCCCATTACAATGGCCAT', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(coding_dna.reverse_complement())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Or, we can **transcribe** this DNA sequence into an RNA sequence:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "coding_dna = Seq(\"ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\", IUPAC.unambiguous_dna)\n",
    "messenger_rna = coding_dna.transcribe()\n",
    "display(messenger_rna)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Translation is "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('MAIVMGR*KGAR*', HasStopCodon(IUPACProtein(), '*'))"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "messenger_rna.translate()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You should notice in the above protein sequence that, in addition to the end stop character, there is an internal stop as well. This was a deliberate choice of example, as it gives an excuse to talk about some optional arguments, including different translation tables (genetic codes).\n",
    "\n",
    "The translation tables available in Biopython are based on those from [the NCBI](https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi). By default, translation will use the standard genetic code (NCBI table id 1). But suppose we are dealing with a mitochondrial sequence---we need to tell the translation function to use the relevant genetic code instead:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('MAIVMGRWKGAR*', HasStopCodon(IUPACProtein(), '*'))"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "messenger_rna.translate(table=\"Vertebrate Mitochondrial\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Otherwise, sequences are similar to strings\n",
    "\n",
    "Other than the situations described above, most of the time `Seq` instances can be dealt with just like strings. For example, we can extract the elements of a sequence using indices, or slice a sequence as we would slice a string: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'G'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "Seq('GATGGGCC', IUPACUnambiguousDNA())"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of the sequence:       32\n",
      "Number of As in the sequence: 10\n"
     ]
    }
   ],
   "source": [
    "# Seq instances work, mostly, as strings\n",
    "my_seq = Seq(\"GATCGATGGGCCTATATAGGATCGAAAATCGC\", IUPAC.unambiguous_dna)\n",
    "display(my_seq[4])\n",
    "display(my_seq[4:12])\n",
    "print('Length of the sequence:      ', len(my_seq))\n",
    "print('Number of As in the sequence:', my_seq.count('A'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## More on the `Seq` class\n",
    "\n",
    "This is a good opportunity to learn a new Python trick. To get more information about a given class, you can use the `help()` function. For example, to learn more about the `Seq`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class Seq in module Bio.Seq:\n",
      "\n",
      "class Seq(builtins.object)\n",
      " |  Read-only sequence object (essentially a string with an alphabet).\n",
      " |  \n",
      " |  Like normal python strings, our basic sequence object is immutable.\n",
      " |  This prevents you from doing my_seq[5] = \"A\" for example, but does allow\n",
      " |  Seq objects to be used as dictionary keys.\n",
      " |  \n",
      " |  The Seq object provides a number of string like methods (such as count,\n",
      " |  find, split and strip), which are alphabet aware where appropriate.\n",
      " |  \n",
      " |  In addition to the string like sequence, the Seq object has an alphabet\n",
      " |  property. This is an instance of an Alphabet class from Bio.Alphabet,\n",
      " |  for example generic DNA, or IUPAC DNA. This describes the type of molecule\n",
      " |  (e.g. RNA, DNA, protein) and may also indicate the expected symbols\n",
      " |  (letters).\n",
      " |  \n",
      " |  The Seq object also provides some biological methods, such as complement,\n",
      " |  reverse_complement, transcribe, back_transcribe and translate (which are\n",
      " |  not applicable to sequences with a protein alphabet).\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __add__(self, other)\n",
      " |      Add another sequence or string to this sequence.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> Seq(\"MELKI\", generic_protein) + \"LV\"\n",
      " |      Seq('MELKILV', ProteinAlphabet())\n",
      " |      \n",
      " |      When adding two Seq (like) objects, the alphabets are important.\n",
      " |      Consider this example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet.IUPAC import unambiguous_dna, ambiguous_dna\n",
      " |      >>> unamb_dna_seq = Seq(\"ACGT\", unambiguous_dna)\n",
      " |      >>> ambig_dna_seq = Seq(\"ACRGT\", ambiguous_dna)\n",
      " |      >>> unamb_dna_seq\n",
      " |      Seq('ACGT', IUPACUnambiguousDNA())\n",
      " |      >>> ambig_dna_seq\n",
      " |      Seq('ACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      If we add the ambiguous and unambiguous IUPAC DNA alphabets, we get\n",
      " |      the more general ambiguous IUPAC DNA alphabet:\n",
      " |      \n",
      " |      >>> unamb_dna_seq + ambig_dna_seq\n",
      " |      Seq('ACGTACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      However, if the default generic alphabet is included, the result is\n",
      " |      a generic alphabet:\n",
      " |      \n",
      " |      >>> Seq(\"\") + ambig_dna_seq\n",
      " |      Seq('ACRGT')\n",
      " |      \n",
      " |      You can't add RNA and DNA sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"ACGU\", generic_rna)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      \n",
      " |      You can't add nucleotide and protein sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_protein\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"MELKI\", generic_protein)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __contains__(self, char)\n",
      " |      Implement the 'in' keyword, like a python string.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna, generic_protein\n",
      " |      >>> my_dna = Seq(\"ATATGAAATTTGAAAA\", generic_dna)\n",
      " |      >>> \"AAA\" in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\") in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\", generic_dna) in my_dna\n",
      " |      True\n",
      " |      \n",
      " |      Like other Seq methods, this will raise a type error if another Seq\n",
      " |      (or Seq like) object with an incompatible alphabet is used:\n",
      " |      \n",
      " |      >>> Seq(\"AAA\", generic_rna) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      >>> Seq(\"AAA\", generic_protein) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __eq__(self, other)\n",
      " |      Compare the sequence to another sequence or a string (README).\n",
      " |      \n",
      " |      Historically comparing Seq objects has done Python object comparison.\n",
      " |      After considerable discussion (keeping in mind constraints of the\n",
      " |      Python language, hashes and dictionary support), Biopython now uses\n",
      " |      simple string comparison (with a warning about the change).\n",
      " |      \n",
      " |      Note that incompatible alphabets (e.g. DNA to RNA) will trigger a\n",
      " |      warning.\n",
      " |      \n",
      " |      During this transition period, please just do explicit comparisons:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> seq1 = Seq(\"ACGT\")\n",
      " |      >>> seq2 = Seq(\"ACGT\")\n",
      " |      >>> id(seq1) == id(seq2)\n",
      " |      False\n",
      " |      >>> str(seq1) == str(seq2)\n",
      " |      True\n",
      " |      \n",
      " |      The new behaviour is to use string-like equality:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq1 == seq2\n",
      " |      True\n",
      " |      >>> seq1 == \"ACGT\"\n",
      " |      True\n",
      " |      >>> seq1 == Seq(\"ACGT\", generic_dna)\n",
      " |      True\n",
      " |  \n",
      " |  __ge__(self, other)\n",
      " |      Implement the greater-than or equal operand.\n",
      " |  \n",
      " |  __getitem__(self, index)\n",
      " |      Return a subsequence of single letter, use my_seq[index].\n",
      " |      \n",
      " |      >>> my_seq = Seq('ACTCGACGTCG')\n",
      " |      >>> my_seq[5]\n",
      " |      'A'\n",
      " |  \n",
      " |  __gt__(self, other)\n",
      " |      Implement the greater-than operand.\n",
      " |  \n",
      " |  __hash__(self)\n",
      " |      Hash for comparison.\n",
      " |      \n",
      " |      See the __cmp__ documentation - this has changed from past\n",
      " |      versions of Biopython!\n",
      " |  \n",
      " |  __imul__(self, other)\n",
      " |      Multiply Seq in-place.\n",
      " |      \n",
      " |      Note although Seq is immutable, the in-place method is\n",
      " |      included to match the behaviour for regular Python strings.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq = Seq('ATG', generic_dna)\n",
      " |      >>> seq *= 2\n",
      " |      >>> seq\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __init__(self, data, alphabet=Alphabet())\n",
      " |      Create a Seq object.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - seq - Sequence, required (string)\n",
      " |       - alphabet - Optional argument, an Alphabet object from\n",
      " |         Bio.Alphabet\n",
      " |      \n",
      " |      You will typically use Bio.SeqIO to read in sequences from files as\n",
      " |      SeqRecord objects, whose sequence will be exposed as a Seq object via\n",
      " |      the seq property.\n",
      " |      \n",
      " |      However, will often want to create your own Seq objects directly:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF', IUPACProtein())\n",
      " |      >>> print(my_seq)\n",
      " |      MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\n",
      " |      >>> my_seq.alphabet\n",
      " |      IUPACProtein()\n",
      " |  \n",
      " |  __le__(self, other)\n",
      " |      Implement the less-than or equal operand.\n",
      " |  \n",
      " |  __len__(self)\n",
      " |      Return the length of the sequence, use len(my_seq).\n",
      " |  \n",
      " |  __lt__(self, other)\n",
      " |      Implement the less-than operand.\n",
      " |  \n",
      " |  __mul__(self, other)\n",
      " |      Multiply Seq by integer.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> Seq('ATG') * 2\n",
      " |      Seq('ATGATG')\n",
      " |      >>> Seq('ATG', generic_dna) * 2\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __ne__(self, other)\n",
      " |      Implement the not-equal operand.\n",
      " |  \n",
      " |  __radd__(self, other)\n",
      " |      Add a sequence on the left.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> \"LV\" + Seq(\"MELKI\", generic_protein)\n",
      " |      Seq('LVMELKI', ProteinAlphabet())\n",
      " |      \n",
      " |      Adding two Seq (like) objects is handled via the __add__ method.\n",
      " |  \n",
      " |  __repr__(self)\n",
      " |      Return (truncated) representation of the sequence for debugging.\n",
      " |  \n",
      " |  __rmul__(self, other)\n",
      " |      Multiply integer by Seq.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> 2 * Seq('ATG')\n",
      " |      Seq('ATGATG')\n",
      " |      >>> 2 * Seq('ATG', generic_dna)\n",
      " |      Seq('ATGATG', DNAAlphabet())\n",
      " |  \n",
      " |  __str__(self)\n",
      " |      Return the full sequence as a python string, use str(my_seq).\n",
      " |      \n",
      " |      Note that Biopython 1.44 and earlier would give a truncated\n",
      " |      version of repr(my_seq) for str(my_seq).  If you are writing code\n",
      " |      which need to be backwards compatible with really old Biopython,\n",
      " |      you should continue to use my_seq.tostring() as follows::\n",
      " |      \n",
      " |          try:\n",
      " |              # The old way, removed in Biopython 1.73\n",
      " |              as_string = seq_obj.tostring()\n",
      " |          except AttributeError:\n",
      " |              # The new way, needs Biopython 1.45 or later.\n",
      " |              # Don't use this on Biopython 1.44 or older as truncates\n",
      " |              as_string = str(seq_obj)\n",
      " |  \n",
      " |  back_transcribe(self)\n",
      " |      Return the DNA sequence from an RNA sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> messenger_rna = Seq(\"AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG\",\n",
      " |      ...                     IUPAC.unambiguous_rna)\n",
      " |      >>> messenger_rna\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      >>> messenger_rna.back_transcribe()\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      Trying to back-transcribe a protein or DNA sequence raises an\n",
      " |      exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.back_transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be back transcribed!\n",
      " |  \n",
      " |  complement(self)\n",
      " |      Return the complement sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAG\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGCTATC', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-GD\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-GD', DNAAlphabet())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGctaT-CH', DNAAlphabet())\n",
      " |      \n",
      " |      Note in the above example, ambiguous character D denotes\n",
      " |      G, A or T so its complement is H (for C, T or A).\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception.\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  count(self, sub, start=0, end=9223372036854775807)\n",
      " |      Return a non-overlapping count, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name,\n",
      " |      which does a non-overlapping count!\n",
      " |      \n",
      " |      For an overlapping search use the newer count_overlap() method.\n",
      " |      \n",
      " |      Returns an integer, the number of occurrences of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      Optional arguments start and end are interpreted as in slice\n",
      " |      notation.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"AAAATGA\")\n",
      " |      >>> print(my_seq.count(\"A\"))\n",
      " |      5\n",
      " |      >>> print(my_seq.count(\"ATG\"))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(Seq(\"AT\")))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(\"AT\", 2, -1))\n",
      " |      1\n",
      " |      \n",
      " |      HOWEVER, please note because python strings and Seq objects (and\n",
      " |      MutableSeq objects) do a non-overlapping search, this may not give\n",
      " |      the answer you expect:\n",
      " |      \n",
      " |      >>> \"AAAA\".count(\"AA\")\n",
      " |      2\n",
      " |      >>> print(Seq(\"AAAA\").count(\"AA\"))\n",
      " |      2\n",
      " |      \n",
      " |      An overlapping search, as implemented in .count_overlap(),\n",
      " |      would give the answer as three!\n",
      " |  \n",
      " |  count_overlap(self, sub, start=0, end=9223372036854775807)\n",
      " |      Return an overlapping count.\n",
      " |      \n",
      " |      For a non-overlapping search use the count() method.\n",
      " |      \n",
      " |      Returns an integer, the number of occurrences of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      Optional arguments start and end are interpreted as in slice\n",
      " |      notation.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> print(Seq(\"AAAA\").count_overlap(\"AA\"))\n",
      " |      3\n",
      " |      >>> print(Seq(\"ATATATATA\").count_overlap(\"ATA\"))\n",
      " |      4\n",
      " |      >>> print(Seq(\"ATATATATA\").count_overlap(\"ATA\", 3, -1))\n",
      " |      1\n",
      " |      \n",
      " |      Where substrings do not overlap, should behave the same as\n",
      " |      the count() method:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"AAAATGA\")\n",
      " |      >>> print(my_seq.count_overlap(\"A\"))\n",
      " |      5\n",
      " |      >>> my_seq.count_overlap(\"A\") == my_seq.count(\"A\")\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(\"ATG\"))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(\"ATG\") == my_seq.count(\"ATG\")\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(Seq(\"AT\")))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(Seq(\"AT\")) == my_seq.count(Seq(\"AT\"))\n",
      " |      True\n",
      " |      >>> print(my_seq.count_overlap(\"AT\", 2, -1))\n",
      " |      1\n",
      " |      >>> my_seq.count_overlap(\"AT\", 2, -1) == my_seq.count(\"AT\", 2, -1)\n",
      " |      True\n",
      " |      \n",
      " |      HOWEVER, do not use this method for such cases because the\n",
      " |      count() method is much for efficient.\n",
      " |  \n",
      " |  encode(self, encoding='utf-8', errors='strict')\n",
      " |      Return an encoded version of the sequence as a bytes object.\n",
      " |      \n",
      " |      The Seq object aims to match the interfact of a Python string.\n",
      " |      (This means on Python 3 it acts like a unicode string.)\n",
      " |      \n",
      " |      This is essentially to save you doing str(my_seq).encode() when\n",
      " |      you need a bytes string, for example for computing a hash:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> Seq(\"ACGT\").encode(\"ascii\")\n",
      " |      b'ACGT'\n",
      " |  \n",
      " |  endswith(self, suffix, start=0, end=9223372036854775807)\n",
      " |      Return True if the Seq ends with the given suffix, False otherwise.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence ends with the specified suffix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      suffix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.endswith(\"UUG\")\n",
      " |      True\n",
      " |      >>> my_rna.endswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.endswith(\"AUG\", 0, 18)\n",
      " |      True\n",
      " |      >>> my_rna.endswith((\"UCC\", \"UCA\", \"UUG\"))\n",
      " |      True\n",
      " |  \n",
      " |  find(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the first occurrence of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the first typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.find(\"AUG\")\n",
      " |      3\n",
      " |  \n",
      " |  index(self, sub, start=0, end=9223372036854775807)\n",
      " |      Like find() but raise ValueError when the substring is not found.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.find(\"T\")\n",
      " |      -1\n",
      " |      >>> my_rna.index(\"T\")\n",
      " |      Traceback (most recent call last):\n",
      " |                 ...\n",
      " |      ValueError: substring not found...\n",
      " |  \n",
      " |  join(self, other)\n",
      " |      Return a merge of the sequences in other, spaced by the sequence from self.\n",
      " |      \n",
      " |      Accepts all Seq objects and Strings as objects to be concatenated with the spacer\n",
      " |      \n",
      " |      >>> concatenated = Seq('NNNNN').join([Seq(\"AAA\"), Seq(\"TTT\"), Seq(\"PPP\")])\n",
      " |      >>> concatenated\n",
      " |      Seq('AAANNNNNTTTNNNNNPPP')\n",
      " |      \n",
      " |      Throws error if other is not an iterable and if objects inside of the iterable\n",
      " |      are not Seq or String objects\n",
      " |  \n",
      " |  lower(self)\n",
      " |      Return a lower case copy of the sequence.\n",
      " |      \n",
      " |      This will adjust the alphabet if required. Note that the IUPAC\n",
      " |      alphabets are upper case only, and thus a generic alphabet must be\n",
      " |      substituted.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import Gapped, generic_dna\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAG*AAAAAA\",\n",
      " |      ...          Gapped(IUPAC.unambiguous_dna, \"*\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG*AAAAAA', Gapped(IUPACUnambiguousDNA(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('cggtacgcttatgtcacgtag*aaaaaa', Gapped(DNAAlphabet(), '*'))\n",
      " |      \n",
      " |      See also the upper method.\n",
      " |  \n",
      " |  lstrip(self, chars=None)\n",
      " |      Return a new Seq object with leading (left) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.lstrip(\"-\"))\n",
      " |      \n",
      " |      See also the strip and rstrip methods.\n",
      " |  \n",
      " |  reverse_complement(self)\n",
      " |      Return the reverse complement sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAGNR\", IUPAC.ambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAGNR', IUPACAmbiguousDNA())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('YNCTATCGGGGG', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      Note in the above example, since R = G or A, its complement\n",
      " |      is Y (which denotes C or T).\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-G\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-G', DNAAlphabet())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('C-TatcGGGGG', DNAAlphabet())\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.reverse_complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  rfind(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find from right method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the last (right most) occurrence of\n",
      " |      substring argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - sub - a string or another Seq object to look for\n",
      " |       - start - optional integer, slice start\n",
      " |       - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the last typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.rfind(\"AUG\")\n",
      " |      15\n",
      " |  \n",
      " |  rindex(self, sub, start=0, end=9223372036854775807)\n",
      " |      Like rfind() but raise ValueError when the substring is not found.\n",
      " |  \n",
      " |  rsplit(self, sep=None, maxsplit=-1)\n",
      " |      Do a right split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done COUNTING FROM THE RIGHT.\n",
      " |      If maxsplit is omitted, all splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g. print(my_seq.rsplit(\"*\",1))\n",
      " |      \n",
      " |      See also the split method.\n",
      " |  \n",
      " |  rstrip(self, chars=None)\n",
      " |      Return a new Seq object with trailing (right) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. Removing a nucleotide sequence's polyadenylation (poly-A tail):\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAGAAAAAA\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      >>> my_seq.rstrip(\"A\")\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      See also the strip and lstrip methods.\n",
      " |  \n",
      " |  split(self, sep=None, maxsplit=-1)\n",
      " |      Split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done.  If maxsplit is omitted, all\n",
      " |      splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_aa = my_rna.translate()\n",
      " |      >>> my_aa\n",
      " |      Seq('VMAIVMGR*KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> for pep in my_aa.split(\"*\"):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> for pep in my_aa.split(\"*\", 1):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      \n",
      " |      See also the rsplit method:\n",
      " |      \n",
      " |      >>> for pep in my_aa.rsplit(\"*\", 1):\n",
      " |      ...     pep\n",
      " |      Seq('VMAIVMGR*KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |  \n",
      " |  startswith(self, prefix, start=0, end=9223372036854775807)\n",
      " |      Return True if the Seq starts with the given prefix, False otherwise.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence starts with the specified prefix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      prefix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.startswith(\"GUC\")\n",
      " |      True\n",
      " |      >>> my_rna.startswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.startswith(\"AUG\", 3)\n",
      " |      True\n",
      " |      >>> my_rna.startswith((\"UCC\", \"UCA\", \"UCG\"), 1)\n",
      " |      True\n",
      " |  \n",
      " |  strip(self, chars=None)\n",
      " |      Return a new Seq object with leading and trailing ends stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.strip(\"-\"))\n",
      " |      \n",
      " |      See also the lstrip and rstrip methods.\n",
      " |  \n",
      " |  tomutable(self)\n",
      " |      Return the full sequence as a MutableSeq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAAL\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      >>> my_seq.tomutable()\n",
      " |      MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      \n",
      " |      Note that the alphabet is preserved.\n",
      " |  \n",
      " |  transcribe(self)\n",
      " |      Return the RNA sequence from a DNA sequence by creating a new Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> coding_dna = Seq(\"ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\",\n",
      " |      ...                  IUPAC.unambiguous_dna)\n",
      " |      >>> coding_dna\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> coding_dna.transcribe()\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      \n",
      " |      Trying to transcribe a protein or RNA sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be transcribed!\n",
      " |  \n",
      " |  translate(self, table='Standard', stop_symbol='*', to_stop=False, cds=False, gap=None)\n",
      " |      Turn a nucleotide sequence into a protein sequence by creating a new Seq object.\n",
      " |      \n",
      " |      This method will translate DNA or RNA sequences, and those with a\n",
      " |      nucleotide or generic alphabet.  Trying to translate a protein\n",
      " |      sequence raises an exception.\n",
      " |      \n",
      " |      Arguments:\n",
      " |       - table - Which codon table to use?  This can be either a name\n",
      " |         (string), an NCBI identifier (integer), or a CodonTable\n",
      " |         object (useful for non-standard genetic codes).  This\n",
      " |         defaults to the \"Standard\" table.\n",
      " |       - stop_symbol - Single character string, what to use for\n",
      " |         terminators.  This defaults to the asterisk, \"*\".\n",
      " |       - to_stop - Boolean, defaults to False meaning do a full\n",
      " |         translation continuing on past any stop codons (translated as the\n",
      " |         specified stop_symbol).  If True, translation is terminated at\n",
      " |         the first in frame stop codon (and the stop_symbol is not\n",
      " |         appended to the returned protein sequence).\n",
      " |       - cds - Boolean, indicates this is a complete CDS.  If True,\n",
      " |         this checks the sequence starts with a valid alternative start\n",
      " |         codon (which will be translated as methionine, M), that the\n",
      " |         sequence length is a multiple of three, and that there is a\n",
      " |         single in frame stop codon at the end (this will be excluded\n",
      " |         from the protein sequence, regardless of the to_stop option).\n",
      " |         If these tests fail, an exception is raised.\n",
      " |       - gap - Single character string to denote symbol used for gaps.\n",
      " |         It will try to guess the gap character from the alphabet.\n",
      " |      \n",
      " |      e.g. Using the standard table:\n",
      " |      \n",
      " |      >>> coding_dna = Seq(\"GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\")\n",
      " |      >>> coding_dna.translate()\n",
      " |      Seq('VAIVMGR*KGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(stop_symbol=\"@\")\n",
      " |      Seq('VAIVMGR@KGAR@', HasStopCodon(ExtendedIUPACProtein(), '@'))\n",
      " |      >>> coding_dna.translate(to_stop=True)\n",
      " |      Seq('VAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      Now using NCBI table 2, where TGA is not a stop codon:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2)\n",
      " |      Seq('VAIVMGRWKGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(table=2, to_stop=True)\n",
      " |      Seq('VAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      In fact, GTG is an alternative start codon under NCBI table 2, meaning\n",
      " |      this sequence could be a complete CDS:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2, cds=True)\n",
      " |      Seq('MAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      It isn't a valid CDS under NCBI table 1, due to both the start codon\n",
      " |      and also the in frame stop codons:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=1, cds=True)\n",
      " |      Traceback (most recent call last):\n",
      " |          ...\n",
      " |      Bio.Data.CodonTable.TranslationError: First codon 'GTG' is not a start codon\n",
      " |      \n",
      " |      If the sequence has no in-frame stop codon, then the to_stop argument\n",
      " |      has no effect:\n",
      " |      \n",
      " |      >>> coding_dna2 = Seq(\"TTGGCCATTGTAATGGGCCGC\")\n",
      " |      >>> coding_dna2.translate()\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      >>> coding_dna2.translate(to_stop=True)\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      When translating gapped sequences, the gap character is inferred from\n",
      " |      the alphabet:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import Gapped\n",
      " |      >>> coding_dna3 = Seq(\"GTG---GCCATT\", Gapped(IUPAC.unambiguous_dna))\n",
      " |      >>> coding_dna3.translate()\n",
      " |      Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))\n",
      " |      \n",
      " |      It is possible to pass the gap character when the alphabet is missing:\n",
      " |      \n",
      " |      >>> coding_dna4 = Seq(\"GTG---GCCATT\")\n",
      " |      >>> coding_dna4.translate(gap='-')\n",
      " |      Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))\n",
      " |      \n",
      " |      NOTE - Ambiguous codons like \"TAN\" or \"NNN\" could be an amino acid\n",
      " |      or a stop codon.  These are translated as \"X\".  Any invalid codon\n",
      " |      (e.g. \"TA?\" or \"T-A\") will throw a TranslationError.\n",
      " |      \n",
      " |      NOTE - This does NOT behave like the python string's translate\n",
      " |      method.  For that use str(my_seq).translate(...) instead.\n",
      " |  \n",
      " |  ungap(self, gap=None)\n",
      " |      Return a copy of the sequence without the gap character(s).\n",
      " |      \n",
      " |      The gap character can be specified in two ways - either as an explicit\n",
      " |      argument, or via the sequence's alphabet. For example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"-ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('-ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap(\"-\")\n",
      " |      Seq('ATATGAAATTTGAAAA', DNAAlphabet())\n",
      " |      \n",
      " |      If the gap character is not given as an argument, it will be taken from\n",
      " |      the sequence's alphabet (if defined). Notice that the returned\n",
      " |      sequence's alphabet is adjusted since it no longer requires a gapped\n",
      " |      alphabet:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped, HasStopCodon\n",
      " |      >>> my_pro = Seq(\"MVVLE=AD*\", HasStopCodon(Gapped(IUPAC.protein, \"=\")))\n",
      " |      >>> my_pro\n",
      " |      Seq('MVVLE=AD*', HasStopCodon(Gapped(IUPACProtein(), '='), '*'))\n",
      " |      >>> my_pro.ungap()\n",
      " |      Seq('MVVLEAD*', HasStopCodon(IUPACProtein(), '*'))\n",
      " |      \n",
      " |      Or, with a simpler gapped DNA example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped\n",
      " |      >>> my_seq = Seq(\"CGGGTAG=AAAAAA\", Gapped(IUPAC.unambiguous_dna, \"=\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap()\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      As long as it is consistent with the alphabet, although it is\n",
      " |      redundant, you can still supply the gap character as an argument to\n",
      " |      this method:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"=\")\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      However, if the gap character given as the argument disagrees with that\n",
      " |      declared in the alphabet, an exception is raised:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"-\")\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap '-' does not match '=' from alphabet\n",
      " |      \n",
      " |      Finally, if a gap character is not supplied, and the alphabet does not\n",
      " |      define one, an exception is raised:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap character not given and not defined in alphabet\n",
      " |  \n",
      " |  upper(self)\n",
      " |      Return an upper case copy of the sequence.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import HasStopCodon, generic_protein\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"VHLTPeeK*\", HasStopCodon(generic_protein))\n",
      " |      >>> my_seq\n",
      " |      Seq('VHLTPeeK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('vhltpeek*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.upper()\n",
      " |      Seq('VHLTPEEK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      \n",
      " |      This will adjust the alphabet if required. See also the lower method.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reading and writing sequences "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Middle East respiratory syndrome-related coronavirus isolate NL13892, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate NL140455, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/01/human/2020/SWE, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/WH-09/human/2020/CHN, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS0CoV-2/61-TW/human/2020/ NPL, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU01/2020/TWN, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU02/2020/TWN, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL2/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA6/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/Yunnan-01/human/2020/CHN, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/KY/V-029/2020 RNA, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-012/2020 RNA, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-501/2020 RNA, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-521/2020 RNA, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate HZ-1, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WI1/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-MA1/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate SNU01, complete genome\n",
      "Porcine epidemic diarrhea virus strain AH-2018-HF1, complete genome\n",
      "Porcine epidemic diarrhea virus strain Ah2016f2, complete genome\n",
      "Bovine coronavirus GIF-1 RNA, complete genome\n",
      "Bovine coronavirus IWT-9 RNA, complete genome\n",
      "Bovine coronavirus IWT-10 RNA, complete genome\n",
      "Bovine coronavirus IWT-1 RNA, complete genome\n",
      "Bovine coronavirus IWT-2 RNA, complete genome\n",
      "Bovine coronavirus IWT-3 RNA, complete genome\n",
      "Bovine coronavirus IWT-4 RNA, complete genome\n",
      "Bovine coronavirus IWT-5 RNA, complete genome\n",
      "Bovine coronavirus IWT-6 RNA, complete genome\n",
      "Bovine coronavirus IWT-7 RNA, complete genome\n",
      "Bovine coronavirus IWT-8 RNA, complete genome\n",
      "Bovine coronavirus IWT-11 RNA, complete genome\n",
      "Bovine coronavirus IWT-12 RNA, complete genome\n",
      "Bovine coronavirus IWT-13 RNA, complete genome\n",
      "Bovine coronavirus IWT-14 RNA, complete genome\n",
      "Bovine coronavirus IWT-15 RNA, complete genome\n",
      "Bovine coronavirus IWT-18 RNA, complete genome\n",
      "Bovine coronavirus IWT-19 RNA, complete genome\n",
      "Bovine coronavirus IWT-20 RNA, complete genome\n",
      "Bovine coronavirus IWT-21 RNA, complete genome\n",
      "Bovine coronavirus IWT-16 RNA, complete genome\n",
      "Bovine coronavirus IWT-17 RNA, complete genome\n",
      "Bovine coronavirus IWT-22 RNA, complete genome\n",
      "Bovine coronavirus IWT-23 RNA, complete genome\n",
      "Bovine coronavirus IWT-24 RNA, complete genome\n",
      "Bovine coronavirus IWT-25 RNA, complete genome\n",
      "Bovine coronavirus IWT-26 RNA, complete genome\n",
      "Bovine coronavirus IWT-27 RNA, complete genome\n",
      "Bovine coronavirus SHG-1 RNA, complete genome\n",
      "Bovine coronavirus SHG-2 RNA, complete genome\n",
      "Bovine coronavirus SHG-3 RNA, complete genome\n",
      "Bovine coronavirus SHG-4 RNA, complete genome\n",
      "Bovine coronavirus SHG-5 RNA, complete genome\n",
      "Bovine coronavirus SHG-6 RNA, complete genome\n",
      "Bovine coronavirus TCG-4 RNA, complete genome\n",
      "Bovine coronavirus TCG-5 RNA, complete genome\n",
      "Bovine coronavirus TCG-3 RNA, complete genome\n",
      "Bovine coronavirus TCG-2 RNA, complete genome\n",
      "Bovine coronavirus TCG-1 RNA, complete genome\n",
      "Bovine coronavirus TCG-10 RNA, complete genome\n",
      "Bovine coronavirus TCG-11 RNA, complete genome\n",
      "Bovine coronavirus TCG-12 RNA, complete genome\n",
      "Bovine coronavirus TCG-13 RNA, complete genome\n",
      "Bovine coronavirus TCG-14 RNA, complete genome\n",
      "Bovine coronavirus TCG-15 RNA, complete genome\n",
      "Bovine coronavirus TCG-16 RNA, complete genome\n",
      "Bovine coronavirus TCG-6 RNA, complete genome\n",
      "Bovine coronavirus TCG-7 RNA, complete genome\n",
      "Bovine coronavirus TCG-8 RNA, complete genome\n",
      "Bovine coronavirus TCG-18 RNA, complete genome\n",
      "Bovine coronavirus TCG-17 RNA, complete genome\n",
      "Bovine coronavirus TCG-9 RNA, complete genome\n",
      "Bovine coronavirus TCG-19 RNA, complete genome\n",
      "Bovine coronavirus TCG-20 RNA, complete genome\n",
      "Bovine coronavirus TCG-21 RNA, complete genome\n",
      "Bovine coronavirus TCG-22 RNA, complete genome\n",
      "Bovine coronavirus TCG-23 RNA, complete genome\n",
      "Bovine coronavirus TCG-24 RNA, complete genome\n",
      "Bovine coronavirus TCG-25 RNA, complete genome\n",
      "Bovine coronavirus TCG-26 RNA, complete genome\n",
      "Bovine coronavirus TCG-27 RNA, complete genome\n",
      "Bovine coronavirus TCG-28 RNA, complete genome\n",
      "Bovine coronavirus TCG-29 RNA, complete genome\n",
      "Bovine coronavirus TCG-30 RNA, complete genome\n",
      "Bovine coronavirus TCG-31 RNA, complete genome\n",
      "Bovine coronavirus TCG-32 RNA, complete genome\n",
      "Bovine coronavirus TCG-33 RNA, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA3/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA4/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA5/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-A12/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-F6/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-01/2019, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-02/2019, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-03/2019, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-04/2019, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-05/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate Australia/VIC01/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate WIV02, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate WIV04, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate WIV05, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate WIV06, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate WIV07, complete genome\n",
      "Bat coronavirus RaTG13, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU01, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU02, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA1/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA2/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-AZ1/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL1/2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-002a_2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-005b_2020, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1/2020, complete genome\n",
      "Infectious bronchitis virus isolate K046-12, partial genome\n",
      "Infectious bronchitis virus isolate K047-12, partial genome\n",
      "Porcine epidemic diarrhea virus strain FR2019001, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CT P10, complete genome\n",
      "Wuhan seafood market pneumonia virus isolate Wuhan-Hu-1, complete genome\n",
      "Infectious bronchitis virus isolate GA/1359/1994, complete genome\n",
      "Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia00265/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia00958/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cundinamarca03267/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Huila/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cundinamarca03715/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cundinamarca03962/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cundinamarca/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/ValledelCauca05761/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/ValledelCauca05762/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia06285/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia07053/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia08010/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/ValledelCauca/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia08520/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/ValledelCauca08565/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia09796/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia09831/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia10641/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Antioquia14856/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cauca/2015, complete genome\n",
      "Rousettus aegyptiacus bat coronavirus 229E-related isolate 5425, complete genome\n",
      "Miniopterus pusillus bat coronavirus HKU8-related isolate 6610, complete genome\n",
      "Tylonycteris pachypus bat coronavirus HKU4-related isolate GZ131656, complete genome\n",
      "Pipistrellus abramus bat coronavirus HKU5-related isolate BY140568, complete genome\n",
      "Scotophilus kuhlii bat coronavirus 512-related isolate HK140714, complete genome\n",
      "Rhinolophus affinis bat coronavirus HKU2-related isolate 160660, complete genome\n",
      "Hipposideros pomona bat coronavirus HKU10-related isolate 160942, complete genome\n",
      "Miniopterus schreibersii bat coronavirus 1-related isolate 161454, complete genome\n",
      "Hipposideros pomona bat coronavirus CHB25 isolate CHB0025, complete genome\n",
      "Porcine epidemic diarrhea virus strain AJ1102(F12), complete genome\n",
      "Alphacoronavirus sp. isolate WA1087, complete genome\n",
      "Alphacoronavirus sp. isolate WA2028, complete genome\n",
      "Alphacoronavirus sp. isolate WA3301, complete genome\n",
      "Alphacoronavirus sp. isolate WA3607, complete genome\n",
      "Alphacoronavirus sp. isolate WAAlc1, complete genome\n",
      "Infectious bronchitis virus isolate GA/1476/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Jeddah-KSA-173RS1101/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Jeddah-KSA-173RS1570/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-18013832/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Merscov/Egypt/Camel/AHRI-FAO-1/2018, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/GD/GDTS13, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/Can/17-035614, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/Can/17-036989, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/Can/18-048192T, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/Can/18-048430, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/Can/18-049707, complete genome\n",
      "Porcine epidemic diarrhea virus strain HeN170821, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SNJ-P, complete genome\n",
      "Erinaceus hedgehog coronavirus HKU31 strain F6, complete genome\n",
      "Erinaceus hedgehog coronavirus HKU31 strain Rs13, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate llama-passaged-Qatar15, complete genome\n",
      "Porcine epidemic diarrhea virus strain TW/Yunlin550/2018, complete genome\n",
      "Porcine epidemic diarrhea virus isolate G2-HE2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate L6-HB2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate M3-SX2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate N7-GD2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate T10-HB2018, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Ethiopia/CAC1019/2015, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Morocco/CAC2586/2016, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Nigeria/NV1010/2015, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Nigeria/NV1092/2015, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Nigeria/NV1097/2015, complete genome\n",
      "Dromedary camel coronavirus HKU23 isolate DcCoV-HKU23/camel/Nigeria/NV1385/2016, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS05, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS20, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS17, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS25, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS24, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS26, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS22, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS30, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS07, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS21, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS28, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS13, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS11, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS19, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS18, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS29, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS34, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS33, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS27, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS43, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS47, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS50, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS06, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS08, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS16, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS12, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS36, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS39, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS40, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS15, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS44, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS14, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS31, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS32, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS35, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS38, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS37, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS42, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS45, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS41, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS46, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS51, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS53, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS48, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS49, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS52, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS09, complete genome\n",
      "Porcine epidemic diarrhea virus isolate HM2017, complete genome\n",
      "Bat alphacoronavirus strain BtCoV/008_16/M.bra/FIN/2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain SF4017, complete genome\n",
      "Bat coronavirus Vs-CoV-1 genomic RNA, nearly complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC9724/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-19001796/2019, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-19003852/2019, complete genome\n",
      "Bat coronavirus strain BtCoV92, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/YNSL/160501, complete genome\n",
      "Swine enteric alphacoronavirus strain SeACoV-p10, complete genome\n",
      "Human coronavirus NL63 strain HCoV_NL63/Seattle/USA/SC0179/2018, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC0682/2019, complete genome\n",
      "Human coronavirus NL63 strain HCoV_NL63/Seattle/USA/SC0768/2019, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC0810/2019, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC0839/2019, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC0841/2019, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC0865/2019, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC9430/2018, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC9428/2018, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC0776/2019, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ZJ15XS0101-P16, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ZJ15XS0101-P35, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ZJ15XS0101-P120, complete genome\n",
      "Human coronavirus NL63 strain ChinaGD02, complete genome\n",
      "Human coronavirus NL63 strain ChinaGD03, complete genome\n",
      "Human coronavirus NL63 strain ChinaGD05, complete genome\n",
      "Human coronavirus NL63 strain ChinaGD01, complete genome\n",
      "Human coronavirus NL63 strain ChinaGD04, complete genome\n",
      "Porcine epidemic diarrhea virus strain WHLL, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV SH, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV YZ, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1829, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1830, complete genome\n",
      "Porcine epidemic diarrhea virus isolate Aram-feces, complete genome\n",
      "Porcine epidemic diarrhea virus isolate Aram-P5, complete genome\n",
      "Porcine epidemic diarrhea virus isolate Aram-P29-CA, complete genome\n",
      "Porcine epidemic diarrhea virus strain S236, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/78/1989, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/79/1989, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/80/1989, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/162/1997, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/255/1997, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/548/2004, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/195/2012, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/G103/2016, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Poland/G225/2017, complete genome\n",
      "Porcine deltacoronavirus strain CHN/Sichuan/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 013, complete genome\n",
      "Infectious bronchitis virus isolate CK-CH-GX-YL17, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/GD/HY16, complete genome\n",
      "Feline coronavirus isolate XXN, complete genome\n",
      "Human coronavirus OC43 isolate OC43_KLF_01_2018, complete genome\n",
      "Coronavirus BtSk-AlphaCoV/GX2018A, complete genome\n",
      "Coronavirus BtSk-AlphaCoV/GX2018B, complete genome\n",
      "Coronavirus BtSk-AlphaCoV/GX2018C, complete genome\n",
      "Coronavirus BtSk-AlphaCoV/GX2018D, complete genome\n",
      "Coronavirus BtRs-AlphaCoV/YN2018, complete genome\n",
      "Coronavirus BtRl-BetaCoV/SC2018, complete genome\n",
      "Coronavirus BtRs-BetaCoV/YN2018A, complete genome\n",
      "Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "Coronavirus BtRs-BetaCoV/YN2018C, complete genome\n",
      "Coronavirus BtRs-BetaCoV/YN2018D, complete genome\n",
      "Coronavirus BtRt-BetaCoV/GX2018, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus strain GDLX/2019, complete genome\n",
      "Infectious bronchitis virus strain I0221/17, complete genome\n",
      "Infectious bronchitis virus strain I0725/17, complete genome\n",
      "Infectious bronchitis virus strain I0916/16, complete genome\n",
      "Infectious bronchitis virus strain I1209/16, complete genome\n",
      "Pheasant coronavirus strain gammaCoV/ph/China/I0710/17, complete genome\n",
      "Pheasant coronavirus strain gammaCoV/ph/China/I0623/17, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/110615, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/110617, complete genome\n",
      "Infectious bronchitis virus strain I1124/16, complete genome\n",
      "Infectious bronchitis virus strain I0306/17, complete genome\n",
      "Infectious bronchitis virus strain M41, complete genome\n",
      "Infectious bronchitis virus strain H120, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/130906, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/140820, complete genome\n",
      "Infectious bronchitis virus isolate ahysx-1, complete genome\n",
      "Tylonycteris bat coronavirus HKU33 strain GZ151867, complete genome\n",
      "Rhinolophus bat coronavirus HKU32 strain TLC26A, complete genome\n",
      "Rhinolophus bat coronavirus HKU32 strain TLC28A, complete genome\n",
      "Infectious bronchitis virus isolate GA9977/2019, complete genome\n",
      "Porcine epidemic diarrhea virus strain CN/Liaoning25/2018, complete genome\n",
      "Porcine epidemic diarrhea virus strain G2, complete genome\n",
      "Avian coronavirus AvCoV/Gallus gallus//H120 complete genome\n",
      "Infectious bronchitis virus strain E160_YN, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate KNIH/002_05_2015, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S12, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S14, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S97, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S100, complete genome\n",
      "Porcine epidemic diarrhea virus isolate Y178, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S6, complete genome\n",
      "Porcine epidemic diarrhea virus isolate S10, complete genome\n",
      "Porcine deltacoronavirus strain Swine/CHN/SC/2018/1, complete genome\n",
      "UNVERIFIED: Porcine epidemic diarrhea virus isolate JS-2013, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate KSA_1722, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate KSA_1723, complete genome\n",
      "Infectious bronchitis virus strain M41-CK, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/TP-2-2/2018, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/TP-3-1/2018, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/TP-4-3/2018, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/TP-4-4/2018, complete genome\n",
      "Hedgehog coronavirus 1, complete genome\n",
      "Canada goose coronavirus strain Cambridge_Bay_2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH-HB1-2018, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH-HB2-2018, complete genome\n",
      "Porcine epidemic diarrhea virus strain V7-HB2018, complete genome\n",
      "Human coronavirus OC43 isolate MDS6, complete genome\n",
      "Human coronavirus OC43 isolate MDS2, complete genome\n",
      "Human coronavirus OC43 isolate MDS4, complete genome\n",
      "Human coronavirus OC43 isolate MDS11, complete genome\n",
      "Human coronavirus OC43 isolate MDS12, complete genome\n",
      "Human coronavirus OC43 isolate MDS14, complete genome\n",
      "Human coronavirus OC43 isolate MDS16, complete genome\n",
      "Bat coronavirus BtCoV/Rh/YN2012 isolate BtCoV/Rh/YN2012_Rs3376, complete genome\n",
      "Bat coronavirus BtCoV/Rh/YN2012 isolate BtCoV/Rh/YN2012_Rs4125, complete genome\n",
      "Bat coronavirus BtCoV/Rh/YN2012 isolate BtCoV/Rh/YN2012_Rs4259, complete genome\n",
      "Bat coronavirus BtCoV/Rh/YN2012 isolate BtCoV/Rh/YN2012_Ra13591, complete genome\n",
      "UNVERIFIED_ASMBLY: Porcine enteric alphacoronavirus isolate PEAV_GDS04_P12, complete genome\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT039/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT041/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT042/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT045/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT049/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT050/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT055/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/NT058/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN068/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN070/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN071/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN072/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN073/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN074/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN075/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN076/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN079/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN081/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN082/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN085/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN086/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN090/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN092/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN093/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN094/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN095/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN096/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN097/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN098/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN099/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN100/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN102/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN103/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN104/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN106/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN108/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN109/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN110/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN111/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN113/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN114/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN115/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN116/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN117/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN118/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN119/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN120/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN121/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN122/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN123/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN124/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN125/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN128/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN130/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN131/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN132/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN133/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN135/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN137/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN138/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN139/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN140/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN141/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN142/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN143/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN145/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN146/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN147/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN149/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN150/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN151/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN152/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN153/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN154/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN155/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN156/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, partial cds, segment: 3, strain: RVA/Human/JPN/KN157/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN158/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT006/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT011/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT012/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT015/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT020/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT031/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/OT047/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP009/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP011/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP012/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP013/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP015/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP016/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP017/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP019/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP022/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP023/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP025/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP027/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP028/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP030/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP031/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP032/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP034/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/SP035/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM036/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM056/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YR115/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YR116/2013/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YR119/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YR124/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YR125/2013/G1P[8]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN161/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN162/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/KN163/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM079/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM086/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM088/2014/G2P[4]\n",
      "Human rotavirus A VP3 gene for VP3 protein, complete cds, segment: 3, strain: RVA/Human/JPN/YM090/2014/G2P[4]\n",
      "Bacillus mycoides strain TH26 chromosome, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1815, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1816, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1817, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1818, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate camel/MERS/Amibara/118/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate camel/MERS/Amibara/126/2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate LW/L, complete genome\n",
      "Porcine torovirus isolate HB1, complete genome\n",
      "Torovirus sp. isolate Tasmania/Sarcophilus_harrisii/2016/DN128115, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-008D3N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Jeddah-KSA-173RS1101/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Aseer-KSA-173RS1288/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Jeddah-KSA-173RS1512/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Jeddah-KSA-173RS1570/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Jeddah-KSA-182RS2449/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Najran-KSA-182RS2567/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-18012493/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Najran-KSA-183RS279/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Northern-KSA-1847784/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Tabuk-KSA-153/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-18013832/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Qaseem-KSA-18013897/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-18014504/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-18014506/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Albaha-KSA-0800H/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV/KOR/KCDC/001_2018-TSVi, complete genome\n",
      "Human coronavirus OC43 strain ATCC VR-759, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1711 spike protein S1 (S1) gene, partial cds\n",
      "Porcine epidemic diarrhea virus isolate KNU-1801 spike protein (S) gene, complete cds\n",
      "Porcine epidemic diarrhea virus isolate KNU-1802 spike protein (S) gene, complete cds\n",
      "Porcine epidemic diarrhea virus isolate KNU-1803 spike protein (S) gene, complete cds\n",
      "Porcine epidemic diarrhea virus isolate KNU-1804, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1805 spike protein (S) gene, complete cds\n",
      "Porcine epidemic diarrhea virus isolate KNU-1806, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1807, complete genome\n",
      "Bovine coronavirus isolate ICSA21-LBA, complete genome\n",
      "Bovine coronavirus isolate ICSA16-EN, complete genome\n",
      "Bovine coronavirus isolate ICSA16-LBA, complete genome\n",
      "Bovine coronavirus isolate ICSA-pool-EN, complete genome\n",
      "Bovine coronavirus isolate ICSA-pool-LBA, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-002D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-004D1S/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-005D1S/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-006D1S/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-007D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-008D3N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-009D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-012D1S/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-014D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-015D1P/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-031D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-039D3T/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-036D1N/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 011/DAB/C8/F<1, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 011/LOM/C20/F<1, complete genome\n",
      "Infectious bronchitis virus strain HH06, complete genome\n",
      "Porcine enteric alphacoronavirus isolate CHN-GD-2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate FJzz1, complete genome\n",
      "Infectious bronchitis virus isolate IBS037A/2014, complete genome\n",
      "Infectious bronchitis virus isolate IBS130/2015, complete genome\n",
      "Yak coronavirus strain YAK/HY24/CH/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Qatar15, complete genome\n",
      "Human coronavirus HKU1 strain SC2521, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV JS-A, complete genome\n",
      "Porcine epidemic diarrhea virus strain ZJ/ZX2018-C10, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Kenya/C1215/2018, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Kenya/C1272/2018, complete genome\n",
      "Porcine respiratory coronavirus strain PRCV/USA/Minnesota-46140/2016, complete genome\n",
      "Severe acute respiratory syndrome-related coronavirus strain BtKY72, complete genome\n",
      "Human coronavirus OC43 isolate HCoV-OC43/USA/ACRI_0213/2016, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SDSX16, complete genome\n",
      "Infectious bronchitis virus isolate IBV_SES_15AB-01, complete genome\n",
      "Infectious bronchitis virus isolate IBV_SES_15SK-02, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/HBTS/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain JSX2014/ATT, complete genome\n",
      "Porcine epidemic diarrhea virus strain JSX2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCGA/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCZG/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCZY44/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCAZ10/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCZY103/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCLS/2018, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCZJ/2018, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCMY/2018, complete genome\n",
      "Alphacoronavirus Bat-CoV/P.kuhlii/Italy/206645-41/2011, complete genome\n",
      "Alphacoronavirus Bat-CoV/P.kuhlii/Italy/3398-19/2015, complete genome\n",
      "Alphacoronavirus Bat-CoV/P.kuhlii/Italy/206679-3/2010, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0636/16, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0724/17, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0718/17, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0722/17, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0737/17, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0347/11, complete genome\n",
      "Porcine epidemic diarrhea virus strain GDgh, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS-MA, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS-C3, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS-C3-MA, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS-C7, complete genome\n",
      "SARS coronavirus Urbani isolate icSARS-C7-MA, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Jordan-201440011123/2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Jordan-201440011858/2014, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/CHGD/2016, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_23, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_24, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_31, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_32, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_39_c1, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_39_c2, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_45, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_47_c1, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_47_c2, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_5, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_53, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_56, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_61, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_63, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_7, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_76, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_77, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_78, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_84, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16715_86, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16845_24, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16845_47, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16845_53, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16845_64, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_16845_87, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_17819_17, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_17819_22, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_17819_4, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_17819_50, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_20724_95, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_20745_10, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_20745_17, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_20745_6, complete genome\n",
      "Alphacoronavirus sp. strain VZ_AlphaCoV_20745_8, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_16715_52, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_33, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_34_c12, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_34_c13, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_38, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_39, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_20724_43, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_22084_1, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_22084_10, complete genome\n",
      "Betacoronavirus sp. strain VZ_BetaCoV_22084_6, complete genome\n",
      "Infectious bronchitis virus isolate VFAR-047, complete genome\n",
      "Atlantic salmon bafinivirus isolate VT01292015-09, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/BJ9/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/BJ11/2016, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-HN-1601, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Jordan-201440011858/2014, complete genome\n",
      "Human coronavirus HKU1 isolate SI17244, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-023D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-023D3N/2015, complete genome\n",
      "Porcine epidemic diarrhea virus isolate FJ2011, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-023D1N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-023D3N/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain MERS-CoV_England-KSA/1/2018(sputum), complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Amman-Jordan-12641/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Amman-Jordan-13030/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Amman-Jordan-12918/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Amman-Jordan-12716/2015, complete genome\n",
      "Infectious bronchitis virus isolate BP-CaKII, complete genome\n",
      "Infectious bronchitis virus isolate K2, complete genome\n",
      "Feline Alphacoronavirus 1 strain FCoV-SB22, complete genome\n",
      "Infectious bronchitis virus isolate GZ14_F80_vaccine, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Z/1986, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/HB/1988, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota138/2006, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Illinois139/2006, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/NorthCarolina140/2007, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota141/2007, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/NorthCarolina142/2007, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Iowa143/2008, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Tennessee144/2008, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/Mex/145/2008, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Illinois146/2008, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Oklahoma147/2012, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota148/2013, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Illinois149/2013, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota150/2013, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Wisconsin151/2014, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota152/2014, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/Minnesota153/2014, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV/USA/SouthDakota154/2014, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/HD/171018, complete genome\n",
      "Infectious bronchitis virus isolate ArkGA_P1, complete genome\n",
      "Infectious bronchitis virus isolate ArkGA_P20, complete genome\n",
      "Infectious bronchitis virus isolate ArkGA_P40, complete genome\n",
      "Infectious bronchitis virus isolate ArkGA_P60, complete genome\n",
      "Infectious bronchitis virus isolate Ark99_pathogenic_field_virus, complete genome\n",
      "Human coronavirus OC43 strain BJ-112, complete genome\n",
      "Human coronavirus OC43 strain BJ-124, complete genome\n",
      "Human coronavirus OC43 strain BJ-164, complete genome\n",
      "Human coronavirus OC43 strain BJ-165, complete genome\n",
      "Human coronavirus OC43 strain BJ-221, complete genome\n",
      "Human coronavirus OC43 strain CC-23, complete genome\n",
      "Human coronavirus OC43 strain GZYF-26, complete genome\n",
      "Human coronavirus OC43 strain WZ-303, complete genome\n",
      "Human coronavirus OC43 strain WZ-522, complete genome\n",
      "Human coronavirus OC43 strain YC-55, complete genome\n",
      "Human coronavirus OC43 strain YC-67, complete genome\n",
      "Human coronavirus OC43 strain YC-68, complete genome\n",
      "Human coronavirus OC43 strain YC-72, complete genome\n",
      "Human coronavirus OC43 strain YC-207, complete genome\n",
      "Human coronavirus OC43 strain HZ-459, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/GD/QY16, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate SADS-CoV/CN/GDGL/2017, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate SADS-CoV/CN/GDLS/2017, complete genome\n",
      "Infectious bronchitis virus isolate IS-1494, complete genome\n",
      "Porcine epidemic diarrhea virus strain H11-SD2017, complete genome\n",
      "Infectious bronchitis virus isolate gammaCoV/AvCov/Ck/Italy/624I/96, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7178/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7373/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7423/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7680/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-8667/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-8882/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-10308/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9614/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9693/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9689/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9835/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9852/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9522/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-9730/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-5767/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7696/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-10717/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7344/2017, complete genome\n",
      "Avian coronavirus isolate AvCoV/Gallus gallus/Brazil/sample 38/2013 GI-11, complete genome, complete genome\n",
      "Avian coronavirus isolate AvCoV/Gallus gallus/Brazil/sample 22/2013, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate SADS-CoV-CH-FJWT-2018, complete genome\n",
      "Porcine deltacoronavirus strain HNZK-02, complete genome\n",
      "Porcine deltacoronavirus strain HNZK-04, complete genome\n",
      "Porcine deltacoronavirus strain HNZK-06, complete genome\n",
      "Bellinger River virus isolate J248, complete genome\n",
      "Quail deltacoronavirus strain G032/2015, complete genome\n",
      "Bulbul coronavirus HKU11-934, complete genome\n",
      "Betacoronavirus England 1, complete genome\n",
      "Betacoronavirus Erinaceus/VMC/DEU/2012 isolate ErinaceusCoV/2012-174/GER/2012, complete genome\n",
      "Porcine coronavirus HKU15 strain HKU15-155, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7413/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain C3-HB2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain B5-HB2017, complete genome\n",
      "Microviridae sp. isolate ctgh185, complete genome\n",
      "Porcine deltacoronavirus isolate CH/JXJGS02/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 2362, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 2363, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate 2366, complete genome\n",
      "Feline coronavirus isolate Felix, complete genome\n",
      "Avian coronavirus strain D274, complete genome\n",
      "Aeromonas caviae strain R25-6 chromosome, complete genome\n",
      "Campylobacter hyointestinalis subsp. lawsonii strain RM14416 14416_contig00023, whole genome shotgun sequence\n",
      "Human coronavirus OC43 isolate TNP F1778_2, complete genome\n",
      "Human coronavirus OC43 isolate TNP F1790_2, complete genome\n",
      "Human coronavirus OC43 isolate TNP F1832_2, complete genome\n",
      "Human coronavirus OC43 isolate TNP F1834_2, complete genome\n",
      "Human coronavirus OC43 isolate TNP 12636, complete genome\n",
      "Human coronavirus OC43 isolate TNP 12643, complete genome\n",
      "Porcine deltacoronavirus strain CHN/GS/2016/1, complete genome\n",
      "Porcine deltacoronavirus strain CHN/GS/2016/2, complete genome\n",
      "Porcine deltacoronavirus strain CHN/GS/2017/1, complete genome\n",
      "Porcine deltacoronavirus strain CHN/QH/2017/1, complete genome\n",
      "Sparrow deltacoronavirus strain ISU690-4, complete genome\n",
      "Sparrow deltacoronavirus strain ISU690-7, complete genome\n",
      "Sparrow deltacoronavirus strain ISU42824, complete genome\n",
      "Sparrow deltacoronavirus strain ISU73347, complete genome\n",
      "Berne virus isolate P138/72, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS10, complete genome\n",
      "Porcine epidemic diarrhea virus isolate GDS23, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain HCoV-EMC, complete genome\n",
      "Porcine epidemic diarrhea virus strain TKSW M protein gene, complete cds\n",
      "Infectious bronchitis virus isolate YX10 D90 vaccine, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PC273/O, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain HCoV-EMC, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1702, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1703, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1704, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1705, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1706, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1708, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1709, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1710, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_5709_19-May-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_0512_04-Jun-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_3807_11-May-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_0511_01-Jun-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_5402_20-May-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_0522_21-May-2010, complete genome\n",
      "Human coronavirus NL63 isolate Kilifi_HH_3808_24-May-2010, complete genome\n",
      "Massilia oculi strain CCUG 43427 chromosome, complete genome\n",
      "Porcine enteric Alphacoronavirus strain PEAV-GD-CH/2017, complete genome\n",
      "Falcon coronavirus UAE-HKU27 988F genomic RNA, complete genome\n",
      "Houbara coronavirus UAE-HKU28 285F genomic RNA, complete genome\n",
      "Pigeon coronavirus UAE-HKU29 271F genomic RNA, complete genome\n",
      "Quail coronavirus UAE-HKU30 411F genomic RNA, complete genome\n",
      "Quail coronavirus UAE-HKU30 1101F genomic RNA, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-07-P5, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-07-P10, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-07-P20, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-07-P30, complete genome\n",
      "Porcine epidemic diarrhea virus isolate NB1, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-HeB1-2017, complete genome\n",
      "Alphacoronavirus Mink/China/1/2016, complete genome\n",
      "Bovine coronavirus isolate 4-17-03, complete genome\n",
      "Bovine coronavirus isolate 4-17-25, complete genome\n",
      "Bovine coronavirus isolate 4-17-08, complete genome\n",
      "Bovine coronavirus isolate 7-16-23, complete genome\n",
      "Infectious bronchitis virus strain IBV/Chicken/Haryana/53/2013, complete genome\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: AKT/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: GNM-1/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: GNM-2/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: IWT/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: MYZ/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: OKN/JPN/2014\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: YMG/JPN/2014\n",
      "Porcine hemagglutinating encephalomyelitis virus strain CC14, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate NL13845, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate NL140422, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain HCoV-EMC, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/GTO/02/2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/03/2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/19/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/19/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/01/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/02/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/03/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/04/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/05/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/PUE/01/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/MICH/01/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/MICH/19/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/GTO/01/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/VER/01/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/01/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/MICH/02/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/PUE/01/2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/JAL/02/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/MICH/02/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/SON/01/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/VER/01/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/VER/02/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/QRO/01/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/MEX/QRO/02/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Neoromicia/5038, complete genome\n",
      "Porcine deltacoronavirus strain CHN-GD-2016, complete genome\n",
      "Human coronavirus OC43 isolate HCoV-OC43/USA/ACRI_0052/2016, complete genome\n",
      "Bat SARS-like coronavirus isolate bat-SL-CoVZC45, complete genome\n",
      "Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "Rousettus bat coronavirus HKU9 isolate Rousettus spp/Jinghong/2009, complete genome\n",
      "Water deer coronavirus isolate W17-18, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Ethiopia/AAU-EPHI-HKU4412/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Ethiopia/AAU-EPHI-HKU4448/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Ethiopia/AAU-EPHI-HKU4458/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Morocco/CIRAD-HKU213/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Burkina Faso/CIRAD-HKU434/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Burkina Faso/CIRAD-HKU785/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NS004/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Burkina Faso/CIRAD-HKU697/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1405/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1657/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1989/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV2040/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1673/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1712/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV1787/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate MERS-CoV camel/Nigeria/NV2020/2016, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate FarmA, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate FarmB, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate FarmC, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate FarmD, complete genome\n",
      "Swine acute diarrhea syndrome related coronavirus isolate 8462, complete genome\n",
      "Swine acute diarrhea syndrome related coronavirus isolate 8495, complete genome\n",
      "Swine acute diarrhea syndrome related coronavirus isolate 141388, complete genome\n",
      "Swine acute diarrhea syndrome related coronavirus isolate 162140, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 57, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 31, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 207, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 197, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 246, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 211, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 203, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 175, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 182, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 192, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 86, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 75, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 53, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 27, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 188, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 202, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 89, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 44, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 34, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 32, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 250, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 54, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 50, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 189, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 146, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate 82, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate DCD5, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate DCD6, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate DCD7, complete genome\n",
      "Swine acute diarrhea syndrome coronavirus isolate SADS-CoV/CN/GDWT/2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CHN/SH-2012-5/2012, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CHN/SH-2016-4/2016, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0712/11, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0108/17, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSC/99I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/121042, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/OK10240-6/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/OK10240-8/2017, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00901-V215_1978 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate CV777 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate CV777 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate CV777 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate Br1_87 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate CV777 genome assembly, complete genome: monopartite\n",
      "Human coronavirus NL63 isolate CN0601/14, complete genome\n",
      "Porcine epidemic diarrhea virus strain PPC 14, complete genome\n",
      "Porcine epidemic diarrhea virus, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH-HNKF-16, complete genome\n",
      "Sorangium cellulosum strain So ce26 chromosome, complete genome\n",
      "Erwinia pyrifoliae strain EpK1/15 chromosome, complete genome\n",
      "Ball python nidovirus 1 isolate 148, complete genome\n",
      "Hypsugo bat coronavirus HKU25 isolate YD131305, complete genome\n",
      "Hypsugo bat coronavirus HKU25 isolate NL140462, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Bat-CoV/H.savii/Italy/206645-40/2011, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Bat-CoV/P.khulii/Italy/206645-63/2011, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/JXJA/2017, complete genome\n",
      "Feline coronavirus isolate UG-FH8, complete genome\n",
      "Feline coronavirus isolate Cat 1 Karlslunde, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Oman_50_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/Oman_1_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Khobar-KSA-6736/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Aseer-KSA-Rs924/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Hufuf-KSA-11002/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Hufuf-KSA-11442/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Hufuf-KSA-11767/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Hufuf-KSA-11150/2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Hufuf-KSA-11401/2015, complete genome\n",
      "Porcine deltacoronavirus strain SHJS/SL/2016, complete genome\n",
      "Porcine deltacoronavirus strain CH/Hunan/2014, complete genome\n",
      "Porcine deltacoronavirus strain CH/Jiangsu/2014, complete genome\n",
      "Feline coronavirus strain HLJ/HRB/2016/10, complete genome\n",
      "Feline coronavirus strain HLJ/HRB/2016/11, complete genome\n",
      "Feline coronavirus strain HLJ/HRB/2016/13, complete genome\n",
      "Vibrio phage Vp_R1, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_002_2013, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_004_2013, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_009_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_011_2013, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_011_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_015_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_017_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_023_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_025_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_032_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/UAE_X_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_1B-A_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_1H-B_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_1H-D_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_1H-F_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_2B-E_2014, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Camel/UAE_3B-C_2014, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain JL/2008, complete genome\n",
      "Porcine epidemic diarrhea virus strain NW17, complete genome\n",
      "Porcine epidemic diarrhea virus strain NW8, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-7436/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-8677/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-8683/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain Hu/Riyadh-KSA-10024/2017, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-GD16-03, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-GD16-05, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B1_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B2_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B5_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B6_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B7_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B8_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B9_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B10_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B11_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B12_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B13_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B14_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B15_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B16_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B17_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B18_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B19_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B20_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B21_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B22_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B23_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B24_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B25_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B26_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B27_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B28_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B29_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B30_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B31_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B32_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B33_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B34_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B35_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B37_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B38_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B39_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B40_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B41_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B42_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B44_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B45_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B46_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B47_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B48_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B49_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B50_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B52_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B53_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B54_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B55_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B56_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B58_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B59_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B60_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B61_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B62_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B63_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B64_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B65_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B66_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B67_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B68_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B69_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B70_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B71_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B72_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B73_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B74_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B75_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B76_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B77_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B78_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B79_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B80_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B81_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B82_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B83_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B84_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B85_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B86_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B87_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B88_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B90_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B91_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B92_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B93_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B95_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B96_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B97_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B99_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B100_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B101_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B102_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B103_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B104_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B105_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B106_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B107_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B108_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B109_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B110_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_B111_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414485_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414379_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414498_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414377_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414480_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414500_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414486_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_415911_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_415915_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414492_W3_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414485_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_417162_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414481_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414492_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414377_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414500_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_416452_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_414486_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_417163_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_415911_W4_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_416452_W5_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_416452_W6_2015, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus strain camel/UAE_415915_W6_2015, complete genome\n",
      "Human coronavirus OC43 isolate 2002-04, complete genome\n",
      "Human coronavirus OC43 isolate 2007-09, complete genome\n",
      "Bat SARS-like coronavirus isolate As6526, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4081, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4084, complete genome\n",
      "Bat SARS-like coronavirus isolate Rf4092, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4231, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4247, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4255, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs4874, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs7327, complete genome\n",
      "Bat SARS-like coronavirus isolate Rs9401, complete genome\n",
      "Human coronavirus OC43 isolate HCoV-OC43/USA/TCNP_0070/2016, complete genome\n",
      "Human coronavirus OC43 isolate HCoV-OC43/USA/TCNP_00204/2017, complete genome\n",
      "Human coronavirus OC43 isolate HCoV-OC43/USA/TCNP_00212/2017, complete genome\n",
      "Porcine deltacoronavirus strain CHN-LYG-2014, complete genome\n",
      "Murine hepatitis virus strain A59 isolate WT-MHV P250, complete genome\n",
      "Murine hepatitis virus strain A59 isolate MHV-ExoN(-) P250, complete genome\n",
      "Rhinolophus bat coronavirus HKU2 isolate Swine enteric alphacoronavirus CH/GD-01/2017/P2, complete genome\n",
      "Infectious bronchitis virus isolate SZ, complete genome\n",
      "Infectious bronchitis virus isolate SD, complete genome\n",
      "Camel alphacoronavirus Abu Dhabi B38, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/HB/2016, complete genome\n",
      "Cycloclasticus sp. PY97N chromosome, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV-Hjms, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV-LNsy, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-13798/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Qasim-KSA-13922/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-13984/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-14607/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-14675/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-15385/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-14949/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-14670/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-17756/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-17382/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-16849/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Madinah-KSA-390/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Qasim-KSA-893/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-19949/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-20570/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-21155/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-21891/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-24244/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-24241/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-16559/2016, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-K17000887/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-K17000405/2017, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate Hu/Riyadh-KSA-K37029157/2016, complete genome\n",
      "Avian coronavirus strain 1148-A, complete genome\n",
      "Avian coronavirus strain L1148, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-mutant1, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-HG-2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SXYL/2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-mutant2, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-mutant3, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-mutant4, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-mutant5, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-A40, complete genome\n",
      "Porcine epidemic diarrhea virus strain 85-7-C40, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KUPE21, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/Belgorod/dom/2008, complete genome\n",
      "Morelia viridis nidovirus strain S14-1323_MVNV, complete genome\n",
      "Infectious bronchitis virus isolate SCZJ-2, partial genome\n",
      "Ferret coronavirus genomic RNA, complete genome, strain: ferret063\n",
      "Human coronavirus 229E strain 229E/Haiti-1/2016, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-08 spike protein (S) gene, complete cds\n",
      "Porcine deltacoronavirus isolate KNU16-11, complete genome\n",
      "Porcine deltacoronavirus strain GD, complete genome\n",
      "Porcine deltacoronavirus strain SD, complete genome\n",
      "Ferret enteric coronavirus strain FECV1, complete genome\n",
      "Ferret systemic coronavirus strain FSCV6, complete genome\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01013-K01_15-03_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00999-K06_15-07_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00908-K16_14-03_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01018-K01_15-08_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01019-K01_15-09_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01020-K01_15-10_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01060-K07_15-02_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00926-K20_4-01_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01017-K01_15-07_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00855-K14_14-02_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_AUSTRIA_L01065-M10_15-04_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01014-K01_15-04_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00919-K17_14-02_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01012-K01_15-02_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00998-K06_15-03_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00857-K14_14-04_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00927-K20_14-02_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00906-K16_14-01_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00918-K17_14-01_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01011-K01_15-01_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_AUSTRIA_L01063-M10_15-02_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_ROMANIA_L01330-K25_15-02_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_ROMANIA_L01329-K25_15-01_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00932-K22_14-04_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01059-K07_15-01_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00931-K22_14-03_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_AUSTRIA_L01064-M10_15-03_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01015-K01_15-05_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L01420-K06_15-04_2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00798-K11_14-02_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00933-K22_14-05_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00862_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00907-K16_14-02_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV/GER/L01061-K07_15-03/2015 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00799-K11_14-01_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_GER_L00928-K20_14-03_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate PEDV_AUSTRIA_L01062-M10_15-01_2015 genome assembly, complete genome: monopartite\n",
      "Porcine torovirus isolate PToV_GER_L00926-K20_14-01_2014 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus isolate ZJU/G2/2013, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CO/P14/IC, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-SX, complete genome\n",
      "Morelia viridis nidovirus strain S14-1323_MVNV, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/JLDH/2016, complete genome\n",
      "Bat coronavirus strain 16BO133, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain MERS-CoV/KOR/KNIH/001_05_2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/2014/IL/20697 P7, complete genome\n",
      "Infectious bronchitis virus isolate UY/09/CA/01, complete genome\n",
      "Infectious bronchitis virus isolate UY/11/CA/18, complete genome\n",
      "Porcine deltacoronavirus isolate CH/JXJGS01/2016, complete genome\n",
      "Myotis lucifugus coronavirus, complete genome\n",
      "Porcine enteric alphacoronavirus GDS04, complete genome\n",
      "Porcine deltacoronavirus genomic RNA, complete genome, strain: HKD/JPN/2016\n",
      "Porcine epidemic diarrhea virus isolate HUN/5031/2016, complete genome\n",
      "Feline coronavirus strain HLJ/DQ/2016/01, complete genome\n",
      "Coronavirus HKU15 genomic RNA, complete genome, strain: S579N\n",
      "Coronavirus HKU15 genomic RNA, complete genome, strain: S582N\n",
      "Canine coronavirus isolate HLJ-071, complete genome\n",
      "Canine coronavirus isolate HLJ-072, complete genome\n",
      "Canine coronavirus isolate HLJ-073, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1601, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-S DEL2, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-S DEL2/ORF3, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-S DEL5, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-S DEL5/ORF3, complete genome\n",
      "Human coronavirus 229E strain 229E/UF-1/2016, complete genome\n",
      "Coronavirus AcCoV-JC34, complete genome\n",
      "Goat torovirus strain SZ, complete genome\n",
      "Goat torovirus strain SZ, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2481/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2854/2015, complete genome\n",
      "Human coronavirus NL63 strain HCoV_NL63/Seattle/USA/SC2940/2015, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC3112/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC3118/2015, complete genome\n",
      "UNVERIFIED: Porcine epidemic diarrhea virus strain CH/JX/01/P0, complete genome\n",
      "UNVERIFIED: Porcine epidemic diarrhea virus strain CH/JX/01/P5, complete genome\n",
      "UNVERIFIED: Porcine epidemic diarrhea virus strain CH/JX/01/P30, complete genome\n",
      "Coronavirus AcCoV-JC34, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH_hubei_2016, complete genome\n",
      "Porcine epidemic diarrhea virus strain PT-P5, complete genome\n",
      "Porcine epidemic diarrhea virus strain PT-P96, complete genome\n",
      "Infectious bronchitis virus strain SNU-8065, complete genome\n",
      "Infectious bronchitis virus strain QIA-03342, complete genome\n",
      "Infectious bronchitis virus strain QIA-KR/D79/05, complete genome\n",
      "Infectious bronchitis virus strain SNU-9106, complete genome\n",
      "Infectious bronchitis virus strain SNU-11045, complete genome\n",
      "Infectious bronchitis virus strain SNU-10043, complete genome\n",
      "Infectious bronchitis virus strain QIA-Q43, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/EG/CU/1/2014, complete genome\n",
      "Infectious bronchitis virus isolate IBV/Ck/EG/CU/4/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/GX/2015/750A, complete genome\n",
      "UNVERIFIED: SARS-related coronavirus isolate F46, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC-PC177, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A, complete genome\n",
      "Bacillus mycoides strain SB8 NODE_536, whole genome shotgun sequence\n",
      "Bat coronavirus isolate PREDICT/PDF-2180, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/GDZHDM/1401, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-11739/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-11740/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-11898/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-11958/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-12160/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-10208/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jeddah-KSA-161RS1146/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-12832/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-12969/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-13127/2016, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Artawiyah-KSA-13328/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC379/2016, complete genome\n",
      "Human coronavirus NL63 strain N07-262B, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2924/2015, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC2872/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2770/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2730/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2476/2015, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2345/2015, complete genome\n",
      "Bat coronavirus isolate PREDICT/PDF-2180, complete genome\n",
      "Porcine deltacoronavirus isolate KNU16-07, complete genome\n",
      "Avian coronavirus strain BR-I, complete genome\n",
      "Avian coronavirus strain Ma5, complete genome\n",
      "Bat coronavirus isolate Anlong-43, complete genome\n",
      "Bat coronavirus isolate Anlong-57, complete genome\n",
      "Bat coronavirus isolate Anlong-103, complete genome\n",
      "Bat coronavirus isolate Anlong-112, complete genome\n",
      "Bat coronavirus isolate Jiyuan-84, complete genome\n",
      "Bacillus mycoides strain Gnyt1 chromosome, complete genome\n",
      "Human coronavirus NL63 strain N06-1144B, complete genome\n",
      "Human coronavirus NL63 strain N07-185B, complete genome\n",
      "Human coronavirus NL63 strain N07-196B, complete genome\n",
      "Human coronavirus NL63 strain N07-324B_182X, complete genome\n",
      "Human coronavirus NL63 strain N07-468B_176X, complete genome\n",
      "Human coronavirus OC43 strain N07-1541B_433X, complete genome\n",
      "Human coronavirus OC43 strain N07-1689B_116X, complete genome\n",
      "Human coronavirus OC43 strain N08-33B_360X, complete genome\n",
      "Human coronavirus OC43 strain N09-382B, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/101150, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110409, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110739, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/07III, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/140528, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/140511, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/08-9, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/131006, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/05II, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/101223, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111218, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU25049, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU0331, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU0582, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1785, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1209, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU24992, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1655, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1362, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1727, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1765, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain PHEV CoV USA-15TOSU1582, complete genome\n",
      "Bovine coronavirus isolate BCoV_2014_13, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC399/2016, complete genome\n",
      "Human coronavirus NL63 strain N07-6B, complete genome\n",
      "Human coronavirus NL63 strain N07-64B, complete genome\n",
      "Human coronavirus OC43 strain N07-1609B, complete genome\n",
      "Human coronavirus OC43 strain N07-1647B, complete genome\n",
      "Human coronavirus 229E strain N08-434B, complete genome\n",
      "Human coronavirus OC43 strain N09-595B, complete genome\n",
      "Human coronavirus HKU1 genotype B strain N08-87, complete genome\n",
      "Human coronavirus HKU1 strain N09-1663B, complete genome\n",
      "Human coronavirus HKU1 strain N09-1627B, complete genome\n",
      "Human coronavirus HKU1 strain N09-1605B, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC2269/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC2282/2017, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV 1842/2016 ITA, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0114/14, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0111/14, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I0118/14, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/ck/China/I1101/16, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/07V, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/110673, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/110836, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/99I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/120481, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/110123, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/140734, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/130622, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/140530, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/111210, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGX/111119, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/121219, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/130205, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CHSD2014, complete genome\n",
      "Porcine epidemic diarrhea virus isolate XM1-2, complete genome\n",
      "Porcine epidemic diarrhea virus isolate XM2-4, complete genome\n",
      "Canine respiratory coronavirus strain BJ232, complete genome\n",
      "Porcine epidemic diarrhea virus strain ZJ15XS0101 P1, complete genome\n",
      "Porcine deltacoronavirus isolate DH1, complete genome\n",
      "Porcine deltacoronavirus isolate DH2, complete genome\n",
      "Transmissible gastroenteritis virus strain TGEV AHHF, complete genome\n",
      "Porcine deltacoronavirus strain CH-01, complete genome\n",
      "Yellow head virus isolate 20120706, complete genome\n",
      "Human coronavirus OC43 isolate MY-U002/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U208/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U236/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U413/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U464/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U523/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U710/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U732/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U774/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U868/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U945/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U1024/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U1057/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U1140/12, complete genome\n",
      "Human coronavirus OC43 isolate MY-U1758/13, complete genome\n",
      "Human coronavirus OC43 isolate MY-U1975/13, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHN/101211, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/090330, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/101113, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/101237, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/110439, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHN/120338, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/090314, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/111111, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/120552, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/120848, complete genome\n",
      "Porcine epidemic diarrhea virus strain KB2013-4, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/Swine/Vietnam/HaNoi6/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/Swine/Vietnam/Binh21/2015, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/141103, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/140927, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/110943, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/130705, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD08-8, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD08-7, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/131049, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/101111, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/090334, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/09091, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/131102, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/04I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/150430, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/130744, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/090806, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/130822, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHuB/131123, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/101115, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/140924, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGS/131148, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/100512, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110912, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/03I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111037, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111025, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/150701, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/08-1, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/100305, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/100408, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/130325, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSX/130132, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/100412, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/091203, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/091003, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/120742, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/120497, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/140709, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/090921 complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/090605, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/090515, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/98I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSHH/03II, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/140542, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130589, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130628, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/131144, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/07I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/05I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/06I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/090312, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/150619, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/07I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/090510, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/090910, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/100311, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/120605, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/131040, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/08-6, complete genome\n",
      "Infectious bronchitis virus isolate AR251-15, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/SCCD/2014, complete genome\n",
      "Infectious bronchitis virus strain LD3, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/95I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGS/08I, complete genome\n",
      "Infectious bronchitis virus strain LH1, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P140.BI, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/111190, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130569, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130337, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/120402, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P3, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P95-13, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P100-C4, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P100-C6, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P120, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC22A-P160, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/07I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGS/06I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/111168, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJS/101109, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/150311, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LXJ/111265, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LXJ/02I, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/120913, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/130211, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/130611, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/120437, complete genome\n",
      "Lucheng Rn rat coronavirus isolate Lucheng-19, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/05III, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LAH/120907, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/140514, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LAH/120721, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC831/2016, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC622/2016, complete genome\n",
      "Human coronavirus OC43 strain HCoV_OC43/Seattle/USA/SC9741/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC579/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC677/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC1143/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC1212/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC9731/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC1073/2016, complete genome\n",
      "Human coronavirus 229E strain HCoV_229E/Seattle/USA/SC9773/2016, complete genome\n",
      "Porcine epidemic diarrhea virus isolate JSCZ1601, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/GDZH02/1401, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/GDZHDM/1401, complete genome\n",
      "Infectious bronchitis virus isolate 3575/08, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D1164.1/14, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D2597.2/14, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D252/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D374/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D383/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D389/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D998/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D1157/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D1271/15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate D1189.1/15, complete genome\n",
      "Transmissible gastroenteritis virus isolate HE-1, complete genome\n",
      "UNVERIFIED: Transmissible gastroenteritis virus strain CN12 genomic sequence\n",
      "NL63-related bat coronavirus strain BtKYNL63-9a, complete genome\n",
      "Infectious bronchitis virus isolate gammaCoV/Ck/Poland/G052/2016, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/150434-III, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/HNZZ47/2016, complete genome\n",
      "Porcine deltacoronavirus strain CHN/Tianjin/2016, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SLOreBAS-1/2015, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SLOreBAS-2/2015, complete genome\n",
      "NL63-related bat coronavirus strain BtKYNL63-9a, complete genome\n",
      "NL63-related bat coronavirus strain BtKYNL63-9b, complete genome\n",
      "NL63-related bat coronavirus strain BtKYNL63-15, complete genome\n",
      "229E-related bat coronavirus strain BtKY229E-1, complete genome\n",
      "229E-related bat coronavirus strain BtKY229E-8, complete genome\n",
      "Human coronavirus NL63 strain NL63/UF-2/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain BJ-2011-C, complete genome\n",
      "Porcine epidemic diarrhea virus strain ZL29, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jordan-20140010168/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jordan-201440011123/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA_1725, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA_1724, complete genome\n",
      "Porcine epidemic diarrhea virus isolate JSLS-1/2015, complete genome\n",
      "Porcine epidemic diarrhea virus isolate JS-2/2015, complete genome\n",
      "Rousettus bat coronavirus isolate GCCDC1 356, complete genome\n",
      "Human coronavirus OC43 isolate LRTI_238, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/JC50/KSA/2014, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/JC49/KSA/2014, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/JC52/KSA/2014, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/AC04/KSA/2014, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/JC50_HUH7/KSA/2014, complete genome\n",
      "Human coronavirus 229E isolate HCoV-229E/BN1/GER/2015, complete genome\n",
      "Camel alphacoronavirus Camel229E isolate Camel229E-CoV/KCSP1/KEN/2015, complete genome\n",
      "Rousettus bat coronavirus isolate GCCDC1 356, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ZJU/G1/2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Taif/KSA-7032/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Taif/KSA-7032/2014/S530del, complete genome\n",
      "Avian coronavirus isolate 23/2013, complete genome\n",
      "[Clostridium] stercorarium subsp. thermolacticum DSM 2910, complete genome\n",
      "[Clostridium] stercorarium subsp. leptospartum DSM 9219, complete genome\n",
      "Human coronavirus NL63 isolate NL63/UF-2/2015, complete genome\n",
      "Ferret coronavirus genomic RNA, complete genome, strain: FRCoV4370\n",
      "Porcine epidemic diarrhea virus strain USA/IA/2013/19321, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/IA/2013/20849, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/IL/2013/59573-5, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/MO/2014/03293, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/2010/JT-1, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/2014/QL1403, complete genome\n",
      "Brachyspira hyodysenteriae ATCC 27164, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jeddah-KSA-3RS2702/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-16121/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-16120/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-16117/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-16077/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-16098/2015, complete genome\n",
      "Bovine coronavirus strain BCV-AKS-01, complete genome\n",
      "Infectious bronchitis virus strain SAIBK2, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Texas134/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Kansas126/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Oklahoma133/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Iowa127/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/NorthDakota93/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota4/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota129/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota135/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota123/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Missouri130/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Texas128/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Texas132/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Nebraska108/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota131/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota125/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota124/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/USA/Iowa136/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/USA/Minnesota140/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/USA/Nebraska137/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/USA/Nebraska145/2015, complete genome\n",
      "Ferret coronavirus isolate FRCoV-NL-2010, complete genome\n",
      "Betacoronavirus 1 isolate Buffalo coronavirus B1-24F, complete genome\n",
      "Betacoronavirus 1 isolate Buffalo coronavirus B1-28F, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/SCYB/140913, complete genome\n",
      "Ferret coronavirus isolate FRCoV-NL-2010, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SD2014, complete genome\n",
      "Swine enteric coronavirus isolate SeCoV_GER_L00930_2012 genome assembly, complete genome: monopartite\n",
      "Porcine epidemic diarrhea virus strain YC2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/050-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/066-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/077-2-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/080-3-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/162-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/169-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/Seoul/177-3-2015, complete genome\n",
      "Porcine deltacoronavirus strain NH, complete genome\n",
      "Porcine deltacoronavirus strain NH isolate passage 0, complete genome\n",
      "Porcine deltacoronavirus strain NH isolate passage 5, complete genome\n",
      "Porcine deltacoronavirus strain NH isolate passage 10, complete genome\n",
      "Bat coronavirus isolate JTMC15, complete genome\n",
      "Bat coronavirus isolate JPDB144, complete genome\n",
      "Alteromonas stellipolaris strain PQQ-44, complete genome\n",
      "Porcine epidemic diarrhea virus isolate AH2012/12, complete genome\n",
      "Porcine deltacoronavirus isolate TT_1115, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/GD/GZ14, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH-SD01, complete genome\n",
      "Infectious bronchitis virus isolate gammaCoV/Ck/Poland/74/2009, complete genome\n",
      "Collimonas pratensis strain Ter291, complete genome\n",
      "Human coronavirus HKU1 isolate BJ01-p3, complete genome\n",
      "Human coronavirus HKU1 isolate BJ01-p9, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/Swine/Thailand/S5011/2015, complete genome\n",
      "Porcine deltacoronavirus strain PDCoV/Swine/Thailand/S5015L/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS CoV/camel/Egypt/NRCE-NC163/2014, partial genome\n",
      "Alteromonas macleodii strain HOT1A3, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Korea/Seoul/SNU1-035/2015, complete genome\n",
      "Porcine deltacoronavirus strain USA/Arkansas61/2015, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota442/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota214/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Michigan447/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Michigan448/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Indiana453/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Illinois449/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota/2013, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota454/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota455/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Illinois272/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Illinois273/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/NorthCarolina452/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota159/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Nebraska209/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Nebraska210/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Ohio444/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Ohio445/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Minnesota292/2014, complete genome\n",
      "Porcine deltacoronavirus strain USA/Iowa459/2014, complete genome\n",
      "Transmissible gastroenteritis virus strain TH-98, complete genome\n",
      "Infectious bronchitis virus isolate ck/CH/LDL/150434-I, complete genome\n",
      "Infectious bronchitis virus isolate ckCHLDL150434-II, complete genome\n",
      "Human coronavirus OC43 strain HCoV-OC43/UK/London/2011, complete genome\n",
      "Alteromonas stellipolaris strain LMG 21861, complete genome\n",
      "Alteromonas sp. Mac1, complete genome\n",
      "Alteromonas sp. Mac2, complete genome\n",
      "Alteromonas addita strain R10SW13, complete genome\n",
      "Alteromonas macleodii strain D7, complete genome\n",
      "[Clostridium] propionicum DSM 1682, complete genome\n",
      "Infectious bronchitis virus isolate LDT3-A, complete genome\n",
      "Infectious bronchitis virus isolate tl/CH/LDT3/03, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/IL20697/2014 Passage 5, complete genome\n",
      "Feline coronavirus isolate inoculum, complete genome\n",
      "Feline coronavirus isolate Cat1_day7, complete genome\n",
      "Feline coronavirus isolate Cat2_day21_deletion, complete genome\n",
      "Feline coronavirus isolate Cat2_day21_withoutdeletion, complete genome\n",
      "Feline coronavirus isolate Cat3_day9, complete genome\n",
      "Feline coronavirus isolate Cat1_day28_deletion, complete genome\n",
      "Feline coronavirus isolate Cat1_day28_withoutdeletion, complete genome\n",
      "Feline coronavirus isolate Cat2_day84, complete genome\n",
      "Feline coronavirus isolate Cat3_day28_deletion, complete genome\n",
      "Feline coronavirus isolate Cat3_day28_withoutdeletion, complete genome\n",
      "Porcine epidemic diarrhea virus strain COL/Cundinamarca/2014 from Colombia, complete genome\n",
      "Transmissible gastroenteritis virus strain JS2012, complete genome\n",
      "Janthinobacterium sp. B9-8, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/NC/2013/35140, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/IA/2013/49379, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/NC/2013/49469, complete genome\n",
      "Pseudomonas fragi strain P121, complete genome\n",
      "SARS-like coronavirus WIV16, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jeddah-KSA-C20843/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jeddah-KSA-C21271/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Hufuf-KSA-11002/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Hufuf-KSA-9158/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Khobar-KSA-6736/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-3181/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh-KSA-2716/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Kharj-KSA-2599/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Kharj-KSA-2598/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Najran-KSA-C20915/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Jeddah-KSA-C20860/2015, complete genome\n",
      "Bovine torovirus genomic RNA, complete genome, strain: BToV Ishikawa/2010\n",
      "Bovine torovirus genomic RNA, complete genome, strain: BToV Kagoshima/2014\n",
      "Camel alphacoronavirus isolate camel/Riyadh/Ry141/2015, complete genome\n",
      "Swine enteric coronavirus strain Italy/213306/2009, complete genome\n",
      "BtMr-AlphaCoV/SAX2011, complete genome\n",
      "BtRf-AlphaCoV/HuB2013, complete genome\n",
      "BtRf-AlphaCoV/YN2012, complete genome\n",
      "BtNv-AlphaCoV/SC2013, complete genome\n",
      "Campylobacter iguaniorum strain 2463D, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH10123-14, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH8593-14, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH9097-14, complete genome\n",
      "Porcine epidemic diarrhea virus strain Ukraine/Poltava01/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota187/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota211/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Hawaii/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma1/2015 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois176/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota163/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Nebraska287/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain MEX/Mexico329/2014 from Mexico, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas432/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota179/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois197/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois255/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota402/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kentucky248/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota281/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/SouthDakota371/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota338/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota250/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas275/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota276/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri177/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio343/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio295/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois308/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota269/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Nebraska348/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois259/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois261/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/SouthDakota285/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota306/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois307/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Texas424/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kentucky291/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma418/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota212/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota236/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois201/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois262/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas280/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota289/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota159/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Indiana195/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Indiana254/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois260/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Nebraska288/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois333/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa161/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota312/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Michigan189/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa162/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Nebraska266/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma320/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/SouthDakota336/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Colorado420/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota271/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas166/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois194/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota202/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois256/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota290/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas431/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri337/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri164/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Michigan252/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Colorado203/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma471/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota200/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois258/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa303/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota265/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota309/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio300/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain CAN/Quebec334/2014 from Canada, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota379/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma477/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Texas435/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio249/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota408/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota409/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota372/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri373/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma466/2014 from USA, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri270/2014 from USA, complete genome\n",
      "BtMf-AlphaCoV/AH2011, complete genome\n",
      "BtMf-AlphaCoV/JX2012, complete genome\n",
      "BtMf-AlphaCoV/GD2012, complete genome\n",
      "BtMf-AlphaCoV/HuB2013, complete genome\n",
      "BtMf-AlphaCoV/FJ2012, complete genome\n",
      "BtMf-AlphaCoV/HeN2013, complete genome\n",
      "BtMr-AlphaCoV/SAX2011, complete genome\n",
      "BtRf-AlphaCoV/HuB2013, complete genome\n",
      "BtRf-AlphaCoV/YN2012, complete genome\n",
      "BtNv-AlphaCoV/SC2013, complete genome\n",
      "BtMs-AlphaCoV/GS2013, complete genome\n",
      "BtRf-BetaCoV/JL2012, complete genome\n",
      "BtRf-BetaCoV/HeB2013, complete genome\n",
      "BtRf-BetaCoV/SX2013, complete genome\n",
      "BtRs-BetaCoV/HuB2013, complete genome\n",
      "BtRs-BetaCoV/GX2013, complete genome\n",
      "BtRs-BetaCoV/YN2013, complete genome\n",
      "BtPa-BetaCoV/GD2013, complete genome\n",
      "BtTp-BetaCoV/GX2012, complete genome\n",
      "Porcine epidemic diarrhea virus strain SLO/JH-11/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D511-14, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D469-14, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-HN-2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/F13A/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry23N/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry84N/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/401/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D100/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D33(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D34/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D35/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D36/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D38(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D40/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D42/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D43(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D45/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D46(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D47/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D48/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D49/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D50(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D88/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D90/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/D92/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/N51/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/N62(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/N68(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/O23(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/O24/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/O30/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/O47(b)/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/S100/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/S73/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/S93/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/S94/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/S99/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd1(b)/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd4/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd6(b)/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd7/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd85/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd86/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd87/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd90/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd175/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Jeddah/Jd199/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry136/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry137/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry159(b)/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry162/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry173/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry177/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry178/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry179/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry63/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry64/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry79/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Riyadh/Ry86/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T3/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T7/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T16/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T22/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T68/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T89/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T91(b)/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T92/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T98/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T150/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate camel/Taif/T157(b)/2015, complete genome\n",
      "Camel coronavirus HKU23 isolate camel/Riyadh/Ry123/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd29/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd55/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd63/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/O23(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/O47(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/O100/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/N38/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/N60/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/N62(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/N68(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/D33(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/D38(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/D43(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/D46(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/D50(a)/2014, complete genome\n",
      "Camel alphacoronavirus isolate camel/Riyadh/Ry141/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Riyadh/Ry159(a)/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd1(a)/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd6(a)/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd8/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Jeddah/Jd28/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Taif/T91(a)/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Taif/T96/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Taif/T157(a)/2015, complete genome\n",
      "Camel alphacoronavirus isolate camel/Taif/T193/2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain CV777, complete genome\n",
      "Porcine epidemic diarrhea virus strain LNCT2, complete genome\n",
      "Porcine epidemic diarrhea virus strain Italy/7239/2009, complete genome\n",
      "Swine enteric coronavirus strain Italy/213306/2009, complete genome\n",
      "Porcine epidemic diarrhea virus strain HUA-14PED96, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain D2731.3/14, complete genome\n",
      "European turkey coronavirus 080385d, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/HNLH/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/163-2-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/163-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/014-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/014-2-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/035-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/035-2-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/168-1-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KOREA/Seoul/168-2-2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-5, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-3, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-9, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-4, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-6, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-2, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-10, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-8, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-1, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFMC-7, complete genome\n",
      "Alteromonas stellipolaris LMG 21856, complete genome\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: AOM-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: FKO-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: EHM-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IWT-4/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KGS-1/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KGS-2/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: OKN-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KMM-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KMM-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-5/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-1/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-2/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: TTR-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: GNM-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KGW-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: HRS-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KGS-4/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: MIE-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: TTR-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: GNM-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: NIG-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-6/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-7/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: AOM-3/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IWT-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IWT-3/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: OKN-1/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: AOM-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: MYG-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IWT-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: NIG-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-3/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-4/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: IBR-8/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KCH-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: KCH-2/JPN/2014\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: MYZ-1/JPN/2013\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: OKY-1/JPN/2014\n",
      "Porcine epidemic diarrhea virus strain YN1, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN15, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN30, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN60, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN90, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN144, complete genome\n",
      "Porcine epidemic diarrhea virus strain YN200, complete genome\n",
      "Porcine epidemic diarrhea virus strain XY2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain JSHA2013, complete genome\n",
      "Human coronavirus NL63 isolate NL63/UF-1/2015, complete genome\n",
      "Porcine coronavirus HKU15 strain OH11846, complete genome\n",
      "Asaia bogorensis NBRC 16594 DNA, complete genome\n",
      "Infectious bronchitis virus strain B1648, complete genome\n",
      "Infectious bronchitis virus isolate P100, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/HNQX-3/14, complete genome\n",
      "Infectious bronchitis virus isolate B17, complete genome\n",
      "Porcine deltacoronavirus strain CH/SXD1/2015, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/111043, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LJL/130908, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/HNYF/14, complete genome\n",
      "Human coronavirus OC43 isolate 1908A/2010, complete genome\n",
      "Human coronavirus OC43 isolate 1997A/2010, complete genome\n",
      "Human coronavirus OC43 isolate 1926/2006, complete genome\n",
      "Human coronavirus OC43 isolate 39A/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5345/2007, complete genome\n",
      "Human coronavirus OC43 isolate 2151A/2010, complete genome\n",
      "Human coronavirus OC43 isolate 5352/2007, complete genome\n",
      "Human coronavirus OC43 isolate 10285/2010, complete genome\n",
      "Human coronavirus OC43 isolate 3269A/2012, complete genome\n",
      "Human coronavirus OC43 isolate 3184A/2012, complete genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Human coronavirus OC43 isolate 3582/2006, complete genome\n",
      "Human coronavirus OC43 isolate 5472/2007, complete genome\n",
      "Human coronavirus OC43 isolate 12689/2012, complete genome\n",
      "Human coronavirus OC43 isolate 12691/2012, complete genome\n",
      "Human coronavirus OC43 isolate 12694/2012, complete genome\n",
      "Human coronavirus OC43 isolate 229/2005, complete genome\n",
      "Human coronavirus OC43 isolate 5370/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5414/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5442/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5445/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5479/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5484/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5485/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5508/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5517/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5519/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5566/2007, complete genome\n",
      "Human coronavirus OC43 isolate 10108/2010, complete genome\n",
      "Human coronavirus OC43 isolate 5595/2007, complete genome\n",
      "Human coronavirus OC43 isolate 5617/2007, complete genome\n",
      "Human coronavirus OC43 isolate 69A/2007, complete genome\n",
      "Human coronavirus OC43 isolate 8164/2009, complete genome\n",
      "Human coronavirus OC43 isolate 892A/2008, complete genome\n",
      "Human coronavirus OC43 isolate 10290/2010, complete genome\n",
      "Human coronavirus OC43 isolate 10574/2010, complete genome\n",
      "Porcine deltacoronavirus isolate PDCoV/CHJXNI2/2015, complete genome\n",
      "Alphacoronavirus 1 strain 23/03, complete genome\n",
      "Bat SARS-like coronavirus YNLF_31C, complete genome\n",
      "Bat SARS-like coronavirus YNLF_34C, complete genome\n",
      "Streptococcus pyogenes strain NGAS322, complete genome\n",
      "Infectious bronchitis virus strain gammaCoV/Ck/Italy/I2022/13, complete genome\n",
      "Porcine respiratory coronavirus strain OH7269, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/HNAY/2015, complete genome\n",
      "Infectious bronchitis virus isolate Ind-TN92-03, complete genome\n",
      "Bifidobacterium actinocoloniiforme DSM 22766, complete genome\n",
      "Equine coronavirus RNA, complete genome, strain: Tokachi09\n",
      "Equine coronavirus RNA, complete genome, strain: Obihiro12-1\n",
      "Equine coronavirus RNA, complete genome, strain: Obihiro12-2\n",
      "Human coronavirus NL63 isolate HCoV NL63/Haiti-1/2015, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-feces, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-P5, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-141112-P10, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC21A, complete genome\n",
      "Porcine epidemic diarrhea virus strain PC177, complete genome\n",
      "Porcine epidemic diarrhea virus strain GDS01, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LDL/140520, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LHLJ/140756, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LHLJ/141105, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LHLJ/140901, complete genome\n",
      "Bacillus thuringiensis serovar indiana strain HD521, complete genome\n",
      "Porcine deltacoronavirus strain CH/Sichuan/S27/2012, complete genome\n",
      "Porcine epidemic diarrhea virus strain SQ2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain FL2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/THA/CU/17_06_2015, complete genome\n",
      "Porcine epidemic diarrhea virus strain Hawaii/39249/2014, complete genome\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, strain: Tottori2/JPN/2014\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Oman_2285_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Oman_2874_2013, complete genome\n",
      "Porcine epidemic diarrhea virus isolate HLJBY, complete genome\n",
      "Phytobacter ursingii strain CAV1151 chromosome, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV/KOR/KNIH/002_05_2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh_KSA_2959_2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh_KSA_4050_2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh_KSA_2959_2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu/Riyadh_KSA_4050_2015, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGX/130530, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain ChinaGD01, complete genome\n",
      "Streptococcus pyogenes strain NGAS743, complete genome\n",
      "Streptococcus pyogenes strain NGAS596, complete genome\n",
      "Streptococcus pyogenes strain NGAS327, complete genome\n",
      "[Polyangium] brachysporum strain DSM 7029, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh-KSA-2345/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh-KSA-2343/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh-KSA-2466/2015, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Hu/Riyadh-KSA-2049/2015, complete genome\n",
      "Hoeflea sp. IMCC20628, complete genome\n",
      "Porcine epidemic diarrhea virus clone EAS1, complete genome\n",
      "Porcine epidemic diarrhea virus clone EAS2, complete genome\n",
      "Porcine epidemic diarrhea virus clone CBR1, complete genome\n",
      "Porcine epidemic diarrhea virus clone CBR2, complete genome\n",
      "Porcine epidemic diarrhea virus genomic RNA, complete genome, isolate: AVCT12\n",
      "Duck coronavirus isolate DK/GD/27/2014, complete genome\n",
      "Calothrix sp. 336/3, complete genome\n",
      "Porcine epidemic diarrhea virus strain FR/001/2014, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-AH-2004, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-HB-2014, complete genome\n",
      "Porcine deltacoronavirus isolate CHN-JS-2014, complete genome\n",
      "Infectious bronchitis virus isolate ck/ZA/3665/11, complete genome\n",
      "Chinook salmon bafinivirus isolate NIDO, complete genome\n",
      "Porcine epidemic diarrhea virus isolate 15V010/BEL/2015, complete genome\n",
      "Guinea fowl coronavirus GfCoV/FR/2011 complete genome\n",
      "Chinook salmon bafinivirus isolate NIDO, complete genome\n",
      "Porcine deltacoronavirus strain USA/IL/2014/026PDV_P11, complete genome\n",
      "Transmissible gastroenteritis virus isolate SHXB, complete genome\n",
      "Pseudomonas synxantha strain LBUM223 chromosome, complete genome\n",
      "Canine coronavirus strain CB/05, complete genome\n",
      "Rotavirus A strain RVA/Human-wt/BEL/BE5028/2012/G3P[14] segment 3, complete sequence\n",
      "UNVERIFIED: Murine coronavirus strain AM2, complete genome\n",
      "Infectious bronchitis virus isolate VicS-v, complete genome\n",
      "Infectious bronchitis virus isolate VicS-del, complete genome\n",
      "Burkholderia glumae LMG 2196 = ATCC 33617 chromosome I, complete sequence\n",
      "Paenibacillus beijingensis strain DSM 24997, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SC1402, complete genome\n",
      "Geminocystis sp. NIES-3709 DNA, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130927, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/140413, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LBJ/140402, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/121041, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110851, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110857, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111235, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/140532, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/130102, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130575, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130578, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/111246, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LLN/130101, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110410, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu-France (UAE) - FRA1_1627-2013_BAL_Sanger, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu-France - FRA2_130569-2013_IS_HTS, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu-France - FRA2_130569-2013_InSpu_Sanger, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hu-France - FRA2_130569-2013_Isolate_Sanger, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LGD/090907, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-1C, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-7C, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-8C, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-10F, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-14, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-15F, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-CHZ, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-LY, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-LS, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-LYG, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV-WS, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/140906, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/121010, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130630, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/130925, complete genome\n",
      "Feline coronavirus isolate 27C, complete genome\n",
      "Feline coronavirus isolate 28O, complete genome\n",
      "Feline coronavirus isolate 65F, complete genome\n",
      "Feline coronavirus isolate 67F, complete genome\n",
      "Feline coronavirus isolate 80F, complete genome\n",
      "Feline coronavirus isolate 26M, complete genome\n",
      "Bifidobacterium kashiwanohense JCM 15439 = DSM 21854 DNA, complete genome\n",
      "Bacillus amyloliquefaciens subsp. amyloliquefaciens KHG19, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CHM2013, complete genome\n",
      "Equine rotavirus A strain H2 segment 3, complete sequence\n",
      "Equine rotavirus A strain FI23 segment 3, complete sequence\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1164.9/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1164.10/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1164.11/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1164.14/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1339.2/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1243.12/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/UAE/D1209/2014, complete genome\n",
      "Francisella philomiragia strain GA01-2801, complete genome\n",
      "Human coronavirus OC43 isolate 2145A/2010, complete genome\n",
      "Human coronavirus OC43 isolate 5240/2007, complete genome\n",
      "Human coronavirus OC43 isolate 3074A/2012, complete genome\n",
      "Human coronavirus OC43 isolate 3647/2006, complete genome\n",
      "Human coronavirus OC43 isolate 3194A/2012, complete genome\n",
      "Bacillus mycoides strain ATCC 6462, complete genome\n",
      "Rotavirus A clone RVA/Human-wt/USA/2013774166/2013/G12P[8] VP3 (VP3) gene, complete cds\n",
      "Middle East respiratory syndrome coronavirus isolate England/4/2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate England/3/2013, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/JX-2/2013, complete genome\n",
      "Human coronavirus OC43 strain 2058A/10, complete genome\n",
      "Human coronavirus OC43 strain 1783A/10, complete genome\n",
      "Alcanivorax pacificus W11-5, complete genome\n",
      "Betacoronavirus HKU24 strain HKU24-R05005I, complete genome\n",
      "Betacoronavirus HKU24 strain HKU24-R05005I, complete genome\n",
      "Betacoronavirus HKU24 strain HKU24-R05009I, complete genome\n",
      "Betacoronavirus HKU24 strain HKU24-R05010I, complete genome\n",
      "Porcine epidemic diarrhea virus L00721/GER/2014, complete genome\n",
      "Porcine epidemic diarrhea GER/L00719/2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Florida/USA-2_Saudi Arabia_2014, complete genome\n",
      "Rotavirus A isolate RVA/Pig-wt/BEL/12R022/2012/G2P[27] VP3 gene, partial cds\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_8_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_18_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_16_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_30_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi/Gayathi_UAE_2_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_33_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_9_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Abu Dhabi_UAE_26_2014, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/XDC- 2/2013, complete genome\n",
      "Human coronavirus OC43 strain LY341, complete genome\n",
      "Human coronavirus OC43 strain LY342, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C7149/KSA/2014-04-05, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C7569/KSA/2014-04-03, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C7770/KSA/2014-04-07, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C8826/KSA/2014-04-12, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C9055/KSA/2014-04-14, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_C10306/KSA/2014-04-20, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Makkah_C9355/KSA/Makkah/2014-04-15, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_2014KSA_683/KSA/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/JX-1/2013, complete genome\n",
      "Porcine torovirus strain PToV/NPL/2014, complete genome\n",
      "Lucheng Rn rat coronavirus isolate Lucheng-19, complete genome\n",
      "Porcine deltacoronavirus KNU14-04, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/GDZQ/2014, complete genome\n",
      "Yata virus isolate DakArB 2181, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1406-1, complete genome\n",
      "Bat Hp-betacoronavirus/Zhejiang2013, complete genome\n",
      "Bat Hp-betacoronavirus/Zhejiang2013, complete genome\n",
      "Pluralibacter gergoviae strain FB2 chromosome, complete genome\n",
      "Porcine coronavirus HKU15 strain Michigan/8977/2014, complete genome\n",
      "Paenibacillus borealis strain DSM 13188, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC22A-P10, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PE103(PC21A)-P4, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC168-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC170-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC-PC173-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC177-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC180-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC PC182-P2, complete genome\n",
      "Porcine epidemic diarrhea virus strain TC Iowa106 (PV39)-P1, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/KS/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain NPL-PEDv/2013/p10.1, complete genome\n",
      "Porcine epidemic diarrhea virus strain PEDV/USA/Minnesota188/2014, complete genome\n",
      "Ball python nidovirus strain 07-53, complete genome\n",
      "Ball python nidovirus strain 07-53, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ON-018, complete genome\n",
      "Porcine epidemic diarrhea virus strain LZW isolate FGE_20140427, complete genome\n",
      "Porcine epidemic diarrhea virus strain LZW isolate CPGEN_20140427, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KUIDL-PED-2014-007, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KUIDL-PED-2014-002, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KUIDL-PED-2014-001, complete genome\n",
      "Swine deltacoronavirus OhioCVM1/2014, complete genome\n",
      "Deltacoronavirus PDCoV/USA/Illinois133/2014 from USA, complete genome\n",
      "Deltacoronavirus PDCoV/USA/Illinois134/2014 from USA, complete genome\n",
      "Deltacoronavirus PDCoV/USA/Illinois136/2014 from USA, complete genome\n",
      "Deltacoronavirus PDCoV/USA/Ohio137/2014 from USA, complete genome\n",
      "Amycolatopsis japonica strain MG417-CF17, complete genome\n",
      "Porcine epidemic diarrhea virus clone VN/JFP1013_1/2013/Vinh An/Vietnam, complete genome\n",
      "Porcine epidemic diarrhea virus clone VN/VAP1113_1/2013/Vung Tua/Vietnam, complete genome\n",
      "Porcine epidemic diarrhea virus clone VN/KCHY-310113/2013/KhoaiChau/HungYen/Vietnam, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Indiana12.83/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa28/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas29/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Colorado30/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Texas31/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma32/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Indiana34/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma35/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas36/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Oklahoma38/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Texas39/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/NorthCarolina40/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota41/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota42/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa23.57/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas46/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Colorado47/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota54/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Wisconsin55/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Tennesse56/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota58/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota59/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio60/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota62/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois63/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota64/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota65/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/NorthCarolina66/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota67/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio68/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio69/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa70/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota72/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota73/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Wisconsin74/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio75/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota76/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota77/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota78/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota79/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois81/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota82/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota83/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota85/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota86/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois87/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota89/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota90/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/NorthCarolina91/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri92/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri93/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota94/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota95/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa96/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois97/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Illinois98/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota100/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri101/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Missouri102/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa103/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa106/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa107/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Texas128/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio120/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio123/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain MEX/124/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Kansas125/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Ohio126/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota127/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota52/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota61/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota71/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Minnesota84/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain MEX/104/2013, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CH/ZJCX-1/2012, complete genome\n",
      "Porcine epidemic diarrhea virus strain AH-M, complete genome\n",
      "Porcine epidemic diarrhea virus isolate KNU-1305, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate England/2/2013, complete genome\n",
      "Porcine coronavirus HKU15 strain MI6148, complete genome\n",
      "Nitrososphaera viennensis EN76, complete genome\n",
      "Porcine coronavirus HKU15 strain IL2768, complete genome\n",
      "Porcine coronavirus HKU15 strain SD3424, complete genome\n",
      "Porcine coronavirus HKU15 strain KY4813, complete genome\n",
      "Porcine coronavirus HKU15 strain PA3148, complete genome\n",
      "Porcine coronavirus HKU15 strain NE3579, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH15962, complete genome\n",
      "Porcine epidemic diarrhea virus isolate NPL-PEDv/2013, complete genome\n",
      "Porcine epidemic diarrhea virus isolate NPL-PEDv/2013/P10, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Florida/USA-2_Saudi Arabia_2014, complete genome\n",
      "Middle East respiratory syndrome coronavirus strain Indiana/USA-1_Saudi Arabia_2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/YNKM-8/2013, complete genome\n",
      "BtVs-BetaCoV/SC2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Camel/Qatar_2_2014, complete genome\n",
      "Human betacoronavirus 2c Jordan-N3/2012 isolate MG167, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA-CAMEL-505, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA-CAMEL-378, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA-CAMEL-503, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA-CAMEL-363, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KSA-CAMEL-376, complete genome\n",
      "Porcine epidemic diarrhea virus strain K14JB01, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV-Jeddah-Camel-1, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate MERS-CoV-Jeddah-human-1, complete genome\n",
      "Rhinolophus affinis coronavirus isolate LYRa11, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/110931, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/120557, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/110526, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/110825, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/111172, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/111232, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/111268, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/120403, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/120749, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/121024, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/121040, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130573, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130598, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/130642, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/131118, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/131132, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/131142, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/131143, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/090908, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/091205, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/110310, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/111050, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/131216, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHN/090909, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/121059, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110505, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110529, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/110726, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111219, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/111241, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/121228, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LSD/1112150, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFU-HKU 13, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFU-HKU 19Dam, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate KFU-HKU 1, complete genome\n",
      "Porcine epidemic diarrhea virus strain SHQP/YM/2013, complete genome\n",
      "Dromedary camel coronavirus HKU23 strain HKU23-265F, complete genome\n",
      "Dromedary camel coronavirus HKU23 strain HKU23-362F, complete genome\n",
      "Dromedary camel coronavirus HKU23 strain HKU23-368F, complete genome\n",
      "Infectious bronchitis virus isolate IBVUkr27-11, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH1414, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/SD/121220, complete genome\n",
      "Porcine coronavirus HKU15 strain OH1987, complete genome\n",
      "Porcine coronavirus HKU15 strain IN2847, complete genome\n",
      "Porcine epidemic diarrhea virus isolate CHYJ130330, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Jeddah_1_2013, complete genome\n",
      "Porcine deltacoronavirus 8734/USA-IA/2014, complete genome\n",
      "Porcine epidemic diarrhea virus strain OH851, complete genome\n",
      "Mink coronavirus strain WD1127, complete genome\n",
      "Mycobacterium phage Violet, complete genome\n",
      "Deltacoronavirus PDCoV/USA/Illinois121/2014 from USA, complete genome\n",
      "Middle East respiratory syndrome coronavirus, complete genome\n",
      "Middle East respiratory syndrome-related coronavirus isolate NRCE-HKU270, complete genome\n",
      "Infectious bronchitis virus isolate SDZB0808, complete genome\n",
      "Bifidobacterium breve 12L, complete genome\n",
      "Bifidobacterium breve JCM 7019, complete genome\n",
      "Arthrospira sp. str. PCC 8005 chromosome, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_25_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_9_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hafr-Al-Batin_6_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Wadi-Ad-Dawasir_1_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hafr-Al-Batin_2_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_14_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_5_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Taif_1_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_4_2013, complete genome\n",
      "Bdellovibrio bacteriovorus W, complete genome\n",
      "Niabella soli DSM 19437, complete genome\n",
      "Feline infectious peritonitis virus isolate FCoVWSU791146_P1, complete genome\n",
      "Feline infectious peritonitis virus isolate FCoVWSU791146_P8, complete genome\n",
      "Feline infectious peritonitis virus isolate FCoVWSU791146_P50, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Qatar3, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Qatar4, complete genome\n",
      "Murine hepatitis virus strain JHM, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Iowa/16465/2013, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Indiana/17846/2013, complete genome\n",
      "Infectious bronchitis virus isolate ck/CH/IBTZ/2012, complete genome\n",
      "Infectious bronchitis virus isolate ck/CH/IBWF/2007, complete genome\n",
      "Infectious bronchitis virus isolate ck/CH/IBYZ/2011, complete genome\n",
      "Infectious bronchitis virus isolate CK/CH/SD09/005, complete genome\n",
      "Infectious bronchitis virus strain Connecticut vaccine, complete genome\n",
      "Bottlenose dolphin coronavirus HKU22 isolate CF090325, complete genome\n",
      "Bottlenose dolphin coronavirus HKU22 isolate CF090327, complete genome\n",
      "Bottlenose dolphin coronavirus HKU22 isolate CF090331, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate FRA/UAE, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-19338E-IN-homogenate, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-19338E-IN-passage3, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-19338E-IN-passage9, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-22038-IA-homogenate, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-22038-IA-passage3, complete genome\n",
      "Porcine epidemic diarrhea virus isolate ISU13-22038-IA-passage9, complete genome\n",
      "Porcine epidemic diarrhea virus isolate USA/Iowa/18984/2013, complete genome\n",
      "Porcine torovirus strain SH1, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LLN/111169, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LGX/091109, complete genome\n",
      "Porcine torovirus strain SH1, complete genome\n",
      "Bat SARS-like coronavirus RsSHC014, complete genome\n",
      "Bat SARS-like coronavirus Rs3367, complete genome\n",
      "Bat SARS-like coronavirus WIV1, complete genome\n",
      "Infectious bronchitis virus isolate SDIB821/2012, complete genome\n",
      "Infectious bronchitis virus, complete genome\n",
      "Infectious bronchitis virus strain 4/91 vaccine, complete genome\n",
      "Porcine epidemic diarrhea virus isolate MN, complete genome\n",
      "Porcine epidemic diarrhea virus isolate IA1, complete genome\n",
      "Porcine epidemic diarrhea virus isolate IA2, complete genome\n",
      "Betacoronavirus Erinaceus/VMC/DEU/2012 isolate ErinaceusCoV/2012-174/GER/2012, complete genome\n",
      "Betacoronavirus Erinaceus/VMC/DEU/2012 isolate ErinaceusCoV/2012-216/GER/2012, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/851-15/1985, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/901-43/1990, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9612-48/1996, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9612-9/1996, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/901-41/1990, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/901-33/1990, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/912-10/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/007-11/2000, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/982-4/1998, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/991-19/1999, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/925-1/1992, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9712-13/1997, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/8912-37/1989, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9212-33/1992, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/953-23/1995, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/911-11/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/873-16/1987, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9612-29/1996, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/913-29/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9712-31/1997, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/991-5/1999, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/912-11/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/873-19/1987, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/951-18/1995, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/871-25/1987, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/872-5/1987, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/873-6/1987, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/901-54/1990, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/911-66/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/931-85/1993, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/911-58/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/008-5/2000, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/912-36/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/912-6/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/911-38/1991, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/9211-43/1992, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/965-6/1996, complete genome\n",
      "Human coronavirus OC43 strain OC43/human/USA/971-5/1997, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/904-20/1990, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/012-31/2001, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/8712-17/1987, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/911-56/1991, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/891-6/1989, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/903-28/1990, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/838-9/1983, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/901-24/1990, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/0111-25/2001, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/905-25/1990, complete genome\n",
      "Human coronavirus NL63 strain NL63/human/USA/891-4/1989, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-5/2009, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-10/2010, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-11/2009, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-13/2010, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-15/2009, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-12/2010, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_1_2012, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_3_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Bisha_1_2012, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_12_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Hafr-Al-Batin_1_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Buraidah_1_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_19_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_21_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_16_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_15_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_17_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_18_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Riyadh_2_2012, complete genome\n",
      "Feline coronavirus strain Felis catus/NLD/UU88/2010, complete genome\n",
      "Bat coronavirus CDPHE15/USA/2006, complete genome\n",
      "Bat coronavirus CDPHE15/USA/2006, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.5P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c8P10/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.4P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.9P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.4P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.10P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1P20/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c8P1/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c3P10/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c2P10/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.10P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.1P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.8P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.5P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.6P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.1P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.9P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.2P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c13P1/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.7P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c2P1/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c2P20/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c8P20/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c13P10/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c13P20/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.6P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5P20/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.7P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.8P20/2010, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5.3P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c3P1/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1P10/2009, complete genome\n",
      "SARS coronavirus ExoN1 strain SARS/VeroE6_lab/USA/ExoN1_c5P10/2009, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.2P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c1.3P20/2010, complete genome\n",
      "SARS coronavirus wtic-MB strain SARS/VeroE6_lab/USA/WTic_c3P20/2009, complete genome\n",
      "Human coronavirus 229E strain 229E/human/USA/933-50/1993, complete genome\n",
      "Human coronavirus 229E strain 229E/human/USA/932-72/1993, complete genome\n",
      "Human coronavirus 229E strain 229E/human/USA/933-40/1993, complete genome\n",
      "Murine coronavirus strain MHV/BHKR_lab/USA/infA59_H126A/2012, complete genome\n",
      "Murine coronavirus strain MHV/BHKR_lab/USA/icA59_17Cl1/2012, complete genome\n",
      "Murine coronavirus strain MHV/BHKR_lab/USA/icA59_L94P/2012, complete genome\n",
      "Murine coronavirus strain MHV/BHKR_lab/USA/icA59_ns2M/2012, complete genome\n",
      "Human coronavirus HKU1 strain HKU1/human/USA/HKU1-18/2010, complete genome\n",
      "Cycloclasticus zancles 78-ME, complete genome\n",
      "Porcine epidemic diarrhea virus isolate 13-019349, complete genome\n",
      "Coronavirus Neoromicia/PML-PHE1/RSA/2011, complete genome\n",
      "Anaplasma phagocytophilum str. HZ2, complete genome\n",
      "Anaplasma phagocytophilum str. JM, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/GDGZ/2012, complete genome\n",
      "Streptococcus pyogenes HSC5, complete genome\n",
      "Porcine epidemic diarrhea virus strain USA/Colorado/2013, complete genome\n",
      "Bat coronavirus Rp/Shaanxi2011, complete genome\n",
      "Bat coronavirus Cp/Yunnan2011, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_4_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_3_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_2_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus isolate Al-Hasa_1_2013, complete genome\n",
      "Middle East respiratory syndrome coronavirus, complete genome\n",
      "Porcine epidemic diarrhea virus, complete genome\n",
      "Transmissible gastroenteritis virus isolate TGEV-HX, complete genome\n",
      "Strawberry lethal yellows phytoplasma (CPA) str. NZSb11, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/FJZZ-9/2012, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/111054, complete genome\n",
      "Thermoclostridium stercorarium subsp. stercorarium DSM 8532 chromosome, complete genome\n",
      "Human betacoronavirus 2c Jordan-N3/2012, complete genome\n",
      "Porcine epidemic diarrhea virus strain AH2012, complete genome\n",
      "Porcine epidemic diarrhea virus strain JS2008, complete genome\n",
      "Porcine epidemic diarrhea virus strain JS-HZ2012, complete genome\n",
      "Human betacoronavirus 2c England-Qatar/2012, complete genome\n",
      "Rotavirus A strain PA130/2010 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PAI11/1996 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA3/2004 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA150/2006 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA83/2007 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA108/2007 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA17/2008 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA84/2008 VP3 mRNA, complete cds\n",
      "Rotavirus A strain PA133/2011 VP3 mRNA, complete cds\n",
      "Bartonella australis Aust/NH1, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LJL/110302, complete genome\n",
      "Porcine epidemic diarrhea virus isolate JS2008, complete genome\n",
      "Thermoclostridium stercorarium subsp. stercorarium DSM 8532 chromosome, complete genome\n",
      "Rotavirus A strain RVA/Simian-tc/ZAF/SA11-N2/1958/G3P[2] segment 3, complete sequence\n",
      "Rotavirus A strain RVA/Simian-tc/ZAF/SA11-N5/1958/G3P[2] segment 3, complete sequence\n",
      "UNVERIFIED: Infectious bronchitis virus isolate GX-NN09032, complete genome\n",
      "Murine rotavirus strain EB-Po/G16P[16] segment 3, complete sequence\n",
      "Porcine epidemic diarrhea virus strain CH/ZMDZY/11, complete genome\n",
      "Singulisphaera acidiphila DSM 18658, complete genome\n",
      "Middle East respiratory syndrome coronavirus, complete genome\n",
      "Cyanobacterium stanieri PCC 7202, complete genome\n",
      "Cyanobacterium aponinum PCC 10605 chromosome, complete genome\n",
      "Calothrix sp. PCC 6303, complete genome\n",
      "Stanieria cyanosphaera PCC 7437, complete genome\n",
      "Infectious bronchitis virus strain Ck/CH/LGD/120723, complete genome\n",
      "Infectious bronchitis virus strain Ck/CH/LGD/120724, complete genome\n",
      "Betacoronavirus England 1, complete genome\n",
      "Rivularia sp. PCC 7116, complete genome\n",
      "Synechococcus sp. PCC 6312, complete genome\n",
      "Porcine epidemic diarrhea virus strain SD-M, complete genome\n",
      "Canine coronavirus strain 171, complete genome\n",
      "Canine coronavirus strain K378, complete genome\n",
      "Canine coronavirus strain S378, complete genome\n",
      "Infectious bronchitis virus strain YX10, complete genome\n",
      "Porcine epidemic diarrhea virus strain GD-1, complete genome\n",
      "Canine respiratory coronavirus strain K37, complete genome\n",
      "Human coronavirus NL63 isolate NL63/RPTEC/2004, complete genome\n",
      "Porcine epidemic diarrhea virus strain ZJCZ4, complete genome\n",
      "Porcine epidemic diarrhea virus strain LC, complete genome\n",
      "Rousettus bat coronavirus HKU10, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate TT3A, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate SL12A, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate TLC1310A, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate LSH5A, complete genome\n",
      "Rousettus bat coronavirus HKU10 isolate 183A, complete genome\n",
      "Rousettus bat coronavirus HKU10 isolate 175A, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate TLC1343A, complete genome\n",
      "Hipposideros bat coronavirus HKU10 isolate TLC1347A, complete genome\n",
      "Cycloclasticus sp. P1 chromosome, complete genome\n",
      "Alteromonas macleodii str. 'Black Sea 11', complete genome\n",
      "Human betacoronavirus 2c EMC/2012, complete genome\n",
      "Human coronavirus 229E isolate 0349, complete genome\n",
      "Human coronavirus 229E isolate J0304, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10912, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10851, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10895, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10912, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10851, complete genome\n",
      "SARS coronavirus isolate Tor2/FP1-10895, complete genome\n",
      "Porcine epidemic diarrhea virus strain CHGD-01, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/091022, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LZJ/111113, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/97I substrain P5, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/97I substrain P115, complete genome\n",
      "Human coronavirus NL63 isolate CBJ123, complete genome\n",
      "Feline coronavirus strain WSU 79-1683, complete genome\n",
      "Porcine epidemic diarrhea virus strain AJ1102, complete genome\n",
      "Infectious bronchitis virus isolate SNU8067, complete genome\n",
      "Infectious bronchitis virus isolate KM91, complete genome\n",
      "Porcine epidemic diarrhea virus strain GD-A, complete genome\n",
      "Porcine epidemic diarrhea virus strain GD-B, complete genome\n",
      "Human coronavirus NL63 isolate CBJ 037, complete genome\n",
      "Murine coronavirus strain JHM-WU-Dns2, complete genome\n",
      "Murine coronavirus strain JHM.WU, complete genome\n",
      "Streptococcus pyogenes M1 476 DNA, complete genome\n",
      "Rabbit coronavirus HKU14, complete genome\n",
      "SARS coronavirus ExoN1 isolate c5P10, complete genome\n",
      "SARS coronavirus HKU-39849 isolate recSARS-CoV HKU-39849, complete genome\n",
      "Infectious bronchitis virus strain CK/SWE/0658946/10, complete genome\n",
      "Infectious bronchitis virus strain Sczy3, complete genome\n",
      "Porcine epidemic diarrhea virus strain virulent DR13, complete genome\n",
      "Porcine epidemic diarrhea virus strain attenuated DR13, complete genome\n",
      "Rat coronavirus isolate 681, complete genome\n",
      "Rat coronavirus isolate 8190, complete genome\n",
      "Alpaca respiratory coronavirus isolate CA08-1/2008, complete genome\n",
      "Feline coronavirus strain DF-2 R3i, complete genome\n",
      "Feline infectious peritonitis virus strain DF-2, complete genome\n",
      "Canine coronavirus strain 1-71, complete genome\n",
      "Canine coronavirus strain TN-449, complete genome\n",
      "White-eye coronavirus HKU16, complete genome\n",
      "Sparrow coronavirus HKU17, complete genome\n",
      "Magpie-robin coronavirus HKU18, complete genome\n",
      "Night-heron coronavirus HKU19, complete genome\n",
      "Wigeon coronavirus HKU20, complete genome\n",
      "Common-moorhen coronavirus HKU21, complete genome\n",
      "Rabbit coronavirus HKU14 strain HKU14-1, complete genome\n",
      "Rabbit coronavirus HKU14 strain HKU14-3, complete genome\n",
      "Rabbit coronavirus HKU14 strain HKU14-8, complete genome\n",
      "Rabbit coronavirus HKU14 strain HKU14-10, complete genome\n",
      "Murine hepatitis virus strain S/3239-17, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2009/9, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2009/14, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2009/15, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2008/16, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2009/20, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/193, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/232, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/235, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/271, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/347, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/1062, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/1862, complete genome\n",
      "Human coronavirus NL63 strain NL63/DEN/2005/1876, complete genome\n",
      "SARS coronavirus HKU-39849 isolate UOB, complete genome\n",
      "Porcine coronavirus HKU15 strain HKU15-44, complete genome\n",
      "Porcine coronavirus HKU15 strain HKU15-155, complete genome\n",
      "White-eye coronavirus HKU16 strain HKU16-6847, complete genome\n",
      "Sparrow coronavirus HKU17 strain HKU17-6124, complete genome\n",
      "Magpie-robin coronavirus HKU18 strain HKU18-chu3, complete genome\n",
      "Night-heron coronavirus HKU19 strain HKU19-6918, complete genome\n",
      "Wigeon coronavirus HKU20 strain HKU20-9243, complete genome\n",
      "Common-moorhen coronavirus HKU21 strain HKU21-8295, complete genome\n",
      "Porcine epidemic diarrhea virus isolate BJ-2011-1, complete genome\n",
      "Streptococcus pyogenes MGAS1882, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/FJND-3/2011, complete genome\n",
      "Lactococcus lactis subsp. lactis IO-1 DNA, complete genome\n",
      "Flavobacterium columnare ATCC 49512, complete genome\n",
      "Bacillus cereus F837/76 plasmid pF837_55, complete sequence\n",
      "Rotavirus A strain RVA/Horse-wt/ARG/E4040/2008/G14P[12] segment 3, complete sequence\n",
      "Paenibacillus terrae HPL-003, complete genome\n",
      "Canine coronavirus strain A76, complete genome\n",
      "Porcine epidemic diarrhea virus strain CH/S, complete genome\n",
      "Human coronavirus OC43 strain HK04-01, complete genome\n",
      "Human coronavirus OC43 strain HK04-02, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/100902, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LDL/101212, complete genome\n",
      "Oscillibacter valericigenes Sjm18-20 DNA, complete genome\n",
      "Mycobacterium phage Violet, complete genome\n",
      "Duck coronavirus isolate DK/CH/HN/ZZ2004, complete genome\n",
      "Infectious bronchitis virus serotype Cal56b, complete genome\n",
      "Infectious bronchitis virus serotype Delaware 072, complete genome\n",
      "Infectious bronchitis virus serotype FL18288, complete genome\n",
      "Infectious bronchitis virus serotype Gray, complete genome\n",
      "Infectious bronchitis virus serotype H120, complete genome\n",
      "Infectious bronchitis virus serotype Holte, complete genome\n",
      "Infectious bronchitis virus serotype Iowa 97, complete genome\n",
      "Infectious bronchitis virus serotype JMK, complete genome\n",
      "Transmissible gastroenteritis virus isolate AYU, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHLJ/07VII, complete genome\n",
      "Feline coronavirus UU47, complete genome\n",
      "Feline coronavirus UU54, complete genome\n",
      "Infectious bronchitis virus isolate YN, complete genome\n",
      "Pseudothermotoga thermarum DSM 5069 chromosome, complete genome\n",
      "Rotavirus A RVA/Human-tc/USA/DS-1/1976/G2P[4] segment 3, complete sequence\n",
      "Streptococcus parauberis KCTC 11537, complete genome\n",
      "Alteromonas naphthalenivorans strain SN2, complete genome\n",
      "Infectious bronchitis virus strain Mass41 2006, complete genome\n",
      "Infectious bronchitis virus strain Cal 1995, complete genome\n",
      "Infectious bronchitis virus strain Cal557 2003, complete genome\n",
      "Infectious bronchitis virus strain Conn46 1996, complete genome\n",
      "Infectious bronchitis virus strain Conn46 1972, complete genome\n",
      "Infectious bronchitis virus strain Conn46 1983, complete genome\n",
      "Infectious bronchitis virus strain Conn46 1991, complete genome\n",
      "Infectious bronchitis virus strain Mass41 1965, complete genome\n",
      "Infectious bronchitis virus strain Mass41 1972, complete genome\n",
      "Infectious bronchitis virus strain Mass41 1979, complete genome\n",
      "Infectious bronchitis virus strain Mass41 1985, complete genome\n",
      "Verrucosispora maris AB-18-032, complete genome\n",
      "Infectious bronchitis virus isolate GX-YL5, complete genome\n",
      "Infectious bronchitis virus isolate GX-YL9, complete genome\n",
      "Yellow head virus strain YHV1995, complete genome\n",
      "Yellow head virus strain YHV1999, complete genome\n",
      "Porcine epidemic diarrhea virus isolate SM98, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2ym1, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2ym2, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2ym3, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2ym4, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2ym5, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d4ym1, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d4ym2, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d4ym3, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2om1, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2om2, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2om3, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2om4, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d2om5, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d3om1, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d3om2, complete genome\n",
      "SARS coronavirus MA15 isolate d2ym1, complete genome\n",
      "SARS coronavirus MA15 isolate d2om1, complete genome\n",
      "SARS coronavirus MA15 isolate d2om2, complete genome\n",
      "SARS coronavirus MA15 isolate d2om3, complete genome\n",
      "SARS coronavirus MA15 isolate d2om4, complete genome\n",
      "SARS coronavirus MA15 isolate d2om5, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d4ym4, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d4ym5, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d3om3, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d3om4, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate d3om5, complete genome\n",
      "SARS coronavirus MA15 isolate d2ym2, complete genome\n",
      "SARS coronavirus MA15 isolate d2ym3, complete genome\n",
      "SARS coronavirus MA15 isolate d2ym4, complete genome\n",
      "SARS coronavirus MA15 isolate d2ym5, complete genome\n",
      "SARS coronavirus MA15 isolate d4ym1, complete genome\n",
      "SARS coronavirus MA15 isolate d4ym2, complete genome\n",
      "SARS coronavirus MA15 isolate d4ym3, complete genome\n",
      "SARS coronavirus MA15 isolate d4ym4, complete genome\n",
      "SARS coronavirus MA15 isolate d4ym5, complete genome\n",
      "SARS coronavirus MA15 isolate d3om1, complete genome\n",
      "SARS coronavirus MA15 isolate d3om2, complete genome\n",
      "SARS coronavirus MA15 isolate d3om3, complete genome\n",
      "SARS coronavirus MA15 isolate d3om4, complete genome\n",
      "SARS coronavirus MA15 isolate d3om5, complete genome\n",
      "SARS coronavirus wtic-MB isolate c1P1, complete genome\n",
      "SARS coronavirus ExoN1 isolate c5P1, complete genome\n",
      "Methanobacterium lacus strain AL-21, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LHB/100801, complete genome\n",
      "Infectious bronchitis virus strain ck/CH/LNM/091017, complete genome\n",
      "Transmissible gastroenteritis virus strain WH-1, complete genome\n",
      "Leishmania donovani BPK282A1 complete genome, chromosome 19\n",
      "Leishmania major strain Friedlin complete genome, chromosome 19\n",
      "Leishmania infantum JPCM5 genome chromosome 19\n",
      "Infectious bronchitis virus strain Arkansas DPI, complete genome\n",
      "Infectious bronchitis virus strain Arkansas Vaccine, complete genome\n",
      "Infectious bronchitis virus strain Georgia 1998 pass8, complete genome\n",
      "Infectious bronchitis virus strain Georgia 1998 Vaccine, complete genome\n",
      "Infectious bronchitis virus strain Massachusetts, complete genome\n",
      "Infectious bronchitis virus strain Mass41 Vaccine, complete genome\n",
      "Erwinia sp. Ejp617, complete genome\n",
      "Stigmatella aurantiaca DW4/3-1, complete genome\n",
      "Feline coronavirus UU40, complete genome\n",
      "Feline coronavirus UU19, complete genome\n",
      "Feline coronavirus UU20, complete genome\n",
      "Feline coronavirus UU30, complete genome\n",
      "Human coronavirus HKU1 isolate Caen1, complete genome\n",
      "Bat coronavirus HKU9-5-1, complete genome\n",
      "Bat coronavirus HKU9-5-2, complete genome\n",
      "Bat coronavirus HKU9-10-1, complete genome\n",
      "Bat coronavirus HKU9-10-2, complete genome\n",
      "Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "Feline coronavirus UU17, complete genome\n",
      "Feline coronavirus UU18, complete genome\n",
      "Feline coronavirus UU21, complete genome\n",
      "Feline coronavirus UU24, complete genome\n",
      "Feline coronavirus UU31, complete genome\n",
      "Feline coronavirus UU34, complete genome\n",
      "Desulfarculus baarsii DSM 2075, complete genome\n",
      "Caprine rotavirus A strain GO34 VP3 protein (VP3) gene, complete cds\n",
      "Mink coronavirus strain WD1127, complete genome\n",
      "Mink coronavirus strain WD1133, complete genome\n",
      "Murine hepatitis virus strain S, complete genome\n",
      "Infectious bronchitis virus isolate DY07, complete genome\n",
      "Infectious bronchitis virus isolate CQ04-1, complete genome\n",
      "Brachyspira murdochii DSM 12563, complete genome\n",
      "Rotavirus A strain RVA/Cat-wt/ITA/BA222/2005/G3P[9] VP3 gene, complete cds\n",
      "Bat SARS coronavirus HKU3-4, complete genome\n",
      "Bat SARS coronavirus HKU3-5, complete genome\n",
      "Bat SARS coronavirus HKU3-6, complete genome\n",
      "Bat SARS coronavirus HKU3-7, complete genome\n",
      "Bat SARS coronavirus HKU3-8, complete genome\n",
      "Bat SARS coronavirus HKU3-9, complete genome\n",
      "Bat SARS coronavirus HKU3-10, complete genome\n",
      "Bat SARS coronavirus HKU3-11, complete genome\n",
      "Bat SARS coronavirus HKU3-12, complete genome\n",
      "Bat SARS coronavirus HKU3-13, complete genome\n",
      "Murine hepatitis virus RNA, complete genome, strain: MHV-MI\n",
      "Feline coronavirus UU22 isolate TCVSP-ROTTIER-00022, complete genome\n",
      "Feline coronavirus UU23 isolate TCVSP-ROTTIER-00023, complete genome\n",
      "SARS coronavirus HKU-39849 isolate TCVSP-HARROD-00001, complete genome\n",
      "SARS coronavirus HKU-39849 isolate TCVSP-HARROD-00002, complete genome\n",
      "SARS coronavirus HKU-39849 isolate TCVSP-HARROD-00003, complete genome\n",
      "Bacillus pseudofirmus OF4, complete genome\n",
      "Canine coronavirus strain CCoV/NTU336/F/2008, complete genome\n",
      "Erwinia pyrifoliae DSM 12163 complete genome, culture collection DSM:12163\n",
      "Bat SARS CoV Rs672/2006, complete genome\n",
      "SARS coronavirus ExoN1, complete genome\n",
      "SARS coronavirus wtic-MB isolate P1pp1, complete genome\n",
      "SARS coronavirus ExoN1 isolate P1pp1, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp1, complete genome\n",
      "SARS coronavirus ExoN1, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp12, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp14, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp6, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp29, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp21, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp2, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp18, complete genome\n",
      "SARS coronavirus wtic-MB, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp16, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp37, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp8, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate P3pp5, complete genome\n",
      "SARS coronavirus MA15 ExoN1, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp23, complete genome\n",
      "SARS coronavirus MA15 isolate P3pp6, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp13, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp7, complete genome\n",
      "SARS coronavirus MA15 isolate P3pp3, complete genome\n",
      "SARS coronavirus wtic-MB isolate P3pp23, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp60, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate P3pp3, complete genome\n",
      "SARS coronavirus MA15 isolate P3pp4, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate P3pp4, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp46, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp19, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp53, complete genome\n",
      "SARS coronavirus MA15, complete genome\n",
      "SARS coronavirus MA15 isolate P3pp7, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate P3pp6, complete genome\n",
      "SARS coronavirus ExoN1 isolate P3pp34, complete genome\n",
      "SARS coronavirus MA15 isolate P3pp5, complete genome\n",
      "SARS coronavirus MA15 ExoN1 isolate P3pp10, complete genome\n",
      "Infectious bronchitis virus ITA/90254/2005, complete genome\n",
      "Infectious bronchitis virus NGA/A116E7/2006, complete genome\n",
      "SARS coronavirus isolate A001, complete genome\n",
      "Turkey coronavirus strain TCoV/VA-74/03, complete genome\n",
      "Turkey coronavirus strain TCoV/TX-GL/01, complete genome\n",
      "Turkey coronavirus strain TCoV/IN-517/94, complete genome\n",
      "Turkey coronavirus strain TCoV/TX-1038/98, complete genome\n",
      "Human rotavirus A strain GER1H-09 methyltransferase (VP3) gene, complete cds\n",
      "Infectious bronchitis virus isolate ZJ971, complete genome\n",
      "Infectious bronchitis virus isolate SC021202, complete genome\n",
      "Infectious bronchitis virus strain H52, complete genome\n",
      "Cyanothece sp. PCC 8802, complete genome\n",
      "Rat coronavirus Parker, complete genome\n",
      "Feline coronavirus RM, complete genome\n",
      "Feline coronavirus UU11, complete genome\n",
      "Feline coronavirus UU7, complete genome\n",
      "Feline coronavirus UU4, complete genome\n",
      "Feline coronavirus UU8, complete genome\n",
      "Feline coronavirus UU5, complete genome\n",
      "Feline coronavirus UU15, complete genome\n",
      "Feline coronavirus UU16, complete genome\n",
      "Feline coronavirus UU10, complete genome\n",
      "Feline coronavirus UU2, complete genome\n",
      "Feline coronavirus UU3, complete genome\n",
      "Feline coronavirus UU9, complete genome\n",
      "Bovine coronavirus E-DB2-TC, complete genome\n",
      "Bovine coronavirus E-AH187-TC, complete genome\n",
      "Bovine respiratory coronavirus AH187, complete genome\n",
      "Bovine respiratory coronavirus bovine/US/OH-440-TC/1996, complete genome\n",
      "Human enteric coronavirus strain 4408, complete genome\n",
      "Rat coronavirus Parker, complete genome\n",
      "Yellow head virus strain YHV1992, complete genome\n",
      "Feline coronavirus strain FCoV/NTU156/P/2007, complete genome\n",
      "SARS coronavirus P2, complete genome\n",
      "Infectious bronchitis virus strain H120, complete genome\n",
      "Rotavirus A strain 02V0002G3 segment 3, complete sequence\n",
      "Murine hepatitis virus strain A59 B11 variant, complete genome\n",
      "Murine hepatitis virus strain A59 B12 variant, complete genome\n",
      "Infectious bronchitis virus strain H120, complete genome\n",
      "Acidobacterium capsulatum ATCC 51196, complete genome\n",
      "Murine coronavirus RA59/R13, complete genome\n",
      "Murine coronavirus RJHM/A, complete genome\n",
      "Murine coronavirus RA59/SJHM, complete genome\n",
      "Murine coronavirus repA59/RJHM, complete genome\n",
      "Murine coronavirus SA59/RJHM, complete genome\n",
      "Murine coronavirus MHV-1, complete genome\n",
      "Murine coronavirus MHV-3, complete genome\n",
      "Murine coronavirus inf-MHV-A59, complete genome\n",
      "Murine coronavirus MHV-JHM.IA, complete genome\n",
      "Murine coronavirus repJHM/RA59, complete genome\n",
      "Rotavirus A strain RVA/Guanaco-wt/ARG/Chubut/1999/G8P[14] VP3 gene, complete cds\n",
      "Rotavirus A strain RVA/Cow-wt/ARG/B383/1998/G15P[11] VP3 gene, complete cds\n",
      "Rotavirus A strain RVA/Guanaco-wt/ARG/Rio_Negro/1998/G8P[1] VP3 gene, complete cds\n",
      "Brachyspira hyodysenteriae WA1, complete genome\n",
      "Transmissible gastroenteritis virus strain H16, complete genome\n",
      "Lamb rotavirus strain Lamb-NT segment 3, complete sequence\n",
      "Campylobacter lari RM2100 chromosome, complete genome\n",
      "Geobacter daltonii FRC-32, complete genome\n",
      "Rotavirus A strain RVA/Rhesus-tc/USA/PTRV/1990/G8P[1] VP3 gene, complete cds\n",
      "Rotavirus A strain RVA/Antelope-wt/ZAF/RC-18-08/G6P[14] structural protein VP3 gene, complete cds\n",
      "SARS coronavirus ZJ02, complete genome\n",
      "SARS coronavirus BJ182a, complete genome\n",
      "SARS coronavirus BJ182b, complete genome\n",
      "SARS coronavirus BJ182-4, complete genome\n",
      "SARS coronavirus BJ182-8, complete genome\n",
      "SARS coronavirus BJ182-12, complete genome\n",
      "Infectious bronchitis virus strain ArkDPI101, complete genome\n",
      "Infectious bronchitis virus strain ArkDPI11, complete genome\n",
      "Bacillus cereus B4264, complete genome\n",
      "Cyanothece sp. PCC 8801, complete genome\n",
      "Cyanothece sp. PCC 7424, complete genome\n",
      "Waterbuck coronavirus US/OH-WD358-TC/1994, complete genome\n",
      "Waterbuck coronavirus US/OH-WD358-GnC/1994, complete genome\n",
      "Waterbuck coronavirus US/OH-WD358/1994, complete genome\n",
      "White-tailed deer coronavirus US/OH-WD470/1994, complete genome\n",
      "Sambar deer coronavirus US/OH-WD388-TC/1994, complete genome\n",
      "Sambar deer coronavirus US/OH-WD388/1994, complete genome\n",
      "Sambar deer coronavirus US/OH-WD388-TC/1994 calf-passaged, complete genome\n",
      "Human enteric coronavirus 4408, complete genome\n",
      "Thrush coronavirus HKU12-600, complete genome\n",
      "Munia coronavirus HKU13-3514, complete genome\n",
      "Bulbul coronavirus HKU11-934, complete genome\n",
      "Bulbul coronavirus HKU11-796, complete genome\n",
      "Thrush coronavirus HKU12-600, complete genome\n",
      "Munia coronavirus HKU13-3514, complete genome\n",
      "Rotavirus A segment 3, complete genome\n",
      "Streptococcus pyogenes NZ131, complete genome\n",
      "Mycobacterium phage ScottMcG, complete genome\n",
      "Mycobacterium phage ScottMcG, complete genome\n",
      "Yellow head virus, complete genome\n",
      "Turkey coronavirus isolate TCoV-540, complete genome\n",
      "Turkey coronavirus isolate TCoV-ATCC, complete genome\n",
      "Turkey coronavirus, complete genome\n",
      "Turkey coronavirus isolate MG10, complete genome\n",
      "Infectious bronchitis virus strain CK/CH/LSD/05I from China, complete genome\n",
      "Beluga Whale coronavirus SW1, complete genome\n",
      "Infectious bronchitis virus strain A2, complete genome\n",
      "Coronavirus SW1, complete genome\n",
      "Candidatus Phytoplasma australiense complete genome\n",
      "Bat coronavirus 1A, complete genome\n",
      "Bat coronavirus HKU8, complete genome\n",
      "Bat coronavirus 1B strain AFCD307, complete genome\n",
      "Miniopterus bat coronavirus 1, complete genome\n",
      "Bat coronavirus HKU8 strain AFCD77, complete genome\n",
      "Haemophilus somnus 2336, complete genome\n",
      "Gill-associated virus, complete genome\n",
      "Feline coronavirus isolate Black, complete genome\n",
      "Bacillus weihenstephanensis KBAB4 plasmid pBWB401, complete genome\n",
      "Bat coronavirus HKU2, complete genome\n",
      "Equine coronavirus strain NC99, complete genome\n",
      "Desulfococcus oleovorans Hxd3, complete genome\n",
      "Bat coronavirus HKU2 strain HKU2/GD/430/2006, complete genome\n",
      "Bat coronavirus HKU2 strain HKU2/HK/46/2006, complete genome\n",
      "Bat coronavirus HKU2 strain HKU2/HK/298/2006, complete genome\n",
      "Bat coronavirus HKU2 strain HKU2/HK/33/2006, complete genome\n",
      "Salinispora arenicola CNS-205, complete genome\n",
      "Transmissible gastroenteritis virus strain attenuated H, complete genome\n",
      "Bovine coronavirus genomic RNA, complete genome, strain: Kakegawa\n",
      "SARS coronavirus strain CV7, complete genome\n",
      "Scotophilus bat coronavirus 512, complete genome\n",
      "Bovine coronavirus E-AH65, complete genome\n",
      "Bovine coronavirus E-AH65-TC, complete genome\n",
      "Bovine coronavirus R-AH65, complete genome\n",
      "Bovine coronavirus R-AH65-TC, complete genome\n",
      "Bovine coronavirus E-AH187, complete genome\n",
      "Bovine coronavirus R-AH187, complete genome\n",
      "Sable antelope coronavirus US/OH1/2003, complete genome\n",
      "Giraffe coronavirus US/OH3-TC/2006, complete genome\n",
      "Giraffe coronavirus US/OH3/2003, complete genome\n",
      "Calf-giraffe coronavirus US/OH3/2006, complete genome\n",
      "Bat coronavirus HKU4-1, complete genome\n",
      "Bat coronavirus HKU5-1, complete genome\n",
      "Bat coronavirus HKU9-1, complete genome\n",
      "Feline coronavirus strain DF-2, complete genome\n",
      "Bat coronavirus HKU4-1, complete genome\n",
      "Bat coronavirus HKU4-2, complete genome\n",
      "Bat coronavirus HKU4-3, complete genome\n",
      "Bat coronavirus HKU4-4, complete genome\n",
      "Bat coronavirus HKU5-1, complete genome\n",
      "Bat coronavirus HKU5-2, complete genome\n",
      "Bat coronavirus HKU5-3, complete genome\n",
      "Bat coronavirus HKU5-5, complete genome\n",
      "Bat coronavirus HKU9-1, complete genome\n",
      "Bat coronavirus HKU9-2, complete genome\n",
      "Bat coronavirus HKU9-3, complete genome\n",
      "Bat coronavirus HKU9-4, complete genome\n",
      "Porcine epidemic diarrhea virus strain LZC from China, complete genome\n",
      "Polaromonas naphthalenivorans CJ2, complete genome\n",
      "White bream virus, complete genome\n",
      "White bream virus strain DF24/00, complete genome\n",
      "Human coronavirus NL63 isolate Amsterdam 057, complete genome\n",
      "Human coronavirus NL63 isolate Amsterdam 496, complete genome\n",
      "Bovine coronavirus isolate Alpaca, complete genome\n",
      "Feline coronavirus strain FCoV C1Je, complete genome\n",
      "Infectious bronchitis virus strain M41, complete genome\n",
      "Haemophilus somnus 129PT, complete genome\n",
      "Bovine coronavirus DB2, complete genome\n",
      "TGEV Miller M6, complete genome\n",
      "TGEV Miller M60, complete genome\n",
      "PRCV ISU-1, complete genome\n",
      "TGEV Purdue P115, complete genome\n",
      "TGEV virulent Purdue, complete genome\n",
      "Bat coronavirus (BtCoV/133/2005), complete genome\n",
      "Bat coronavirus (BtCoV/273/2005), complete genome\n",
      "Bat coronavirus (BtCoV/279/2005), complete genome\n",
      "Bat SARS coronavirus Rf1, complete genome\n",
      "Bat SARS coronavirus Rm1, complete genome\n",
      "Chelativorans sp. BNC1, complete genome\n",
      "Infectious bronchitis virus isolate TW2575/98, complete genome\n",
      "Human coronavirus HKU1 genotype B, complete genome\n",
      "Human coronavirus HKU1 strain N5P8 genotype A/B recombinant, complete genome\n",
      "Human coronavirus HKU1 strain N19 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N20 genotype C, complete genome\n",
      "Human coronavirus HKU1 strain N21 genotype C, complete genome\n",
      "Human coronavirus HKU1 strain N22 genotype C, complete genome\n",
      "Human coronavirus HKU1 strain N23 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N24 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N25 genotype B, complete genome\n",
      "Human coronavirus HKU1 strain N3 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N6 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N7 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N9 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N10 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N11 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N13 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N14 genotype A, complete genome\n",
      "Human coronavirus HKU1 strain N15 genotype B, complete genome\n",
      "Human coronavirus HKU1 strain N16 genotype C, complete genome\n",
      "Human coronavirus HKU1 strain N17 genotype C, complete genome\n",
      "Human coronavirus HKU1 strain N18 genotype A, complete genome\n",
      "SARS coronavirus GDH-BJH01, complete genome\n",
      "SARS coronavirus BJ162, complete genome\n",
      "SARS coronavirus BJ202, complete genome\n",
      "SARS coronavirus strain MA-15, complete genome\n",
      "Streptococcus pyogenes MGAS10270, complete genome\n",
      "Streptococcus pyogenes MGAS10750, complete genome\n",
      "SARS coronavirus Frankfurt 1 genomic RNA, nearly complete genome, clone: persistent virus #21\n",
      "Transmissible gastroenteritis virus strain SC-Y, complete genome\n",
      "Anaplasma phagocytophilum HZ, complete genome\n",
      "Transmissible gastroenteritis virus strain TS, complete genome\n",
      "Porcine hemagglutinating encephalomyelitis virus strain VW572, complete genome\n",
      "Moorella thermoacetica ATCC 39073, complete genome\n",
      "Avian infectious bronchitis virus isolate SAIBK, complete genome\n",
      "Bat SARS coronavirus Rp3, complete genome\n",
      "Breda virus, complete genome\n",
      "Bat SARS coronavirus HKU3-1, complete genome\n",
      "bat SARS coronavirus HKU3-2, complete genome\n",
      "bat SARS coronavirus HKU3-3, complete genome\n",
      "SARS coronavirus ZJ0301 from China, complete genome\n",
      "Breda virus strain Breda 1, complete genome\n",
      "Rotavirus A strain RVA/Human-wt/BEL/B4106/2000/G3P[14] VP3 (VP3) gene, complete cds\n",
      "SARS coronavirus civet007, complete genome\n",
      "SARS coronavirus civet010, complete genome\n",
      "SARS coronavirus civet020, complete genome\n",
      "Methanosarcina barkeri str. Fusaro, complete genome\n",
      "Dechloromonas aromatica RCB, complete genome\n",
      "SARS coronavirus A022, complete genome\n",
      "SARS coronavirus B039, complete genome\n",
      "Human coronavirus OC43 strain 87309 Belgium 2003, complete genome\n",
      "Human coronavirus OC43 strain 19572 Belgium 2004, complete genome\n",
      "Feline coronavirus strain FIPV 79-1146, complete genome\n",
      "Avian infectious bronchitis virus isolate IBV-EP3, complete genome\n",
      "Avian infectious bronchitis virus isolate IBV-p65, complete genome\n",
      "Feline infectious peritonitis virus, complete genome\n",
      "Synthetic construct murine hepatitis virus, complete genome\n",
      "SARS coronavirus isolate HC/SZ/79/03, complete genome\n",
      "SARS coronavirus isolate HC/SZ/DM1/03, complete genome\n",
      "SARS coronavirus isolate HC/SZ/266/03, complete genome\n",
      "SARS coronavirus isolate HC/GZ/81/03, complete genome\n",
      "SARS coronavirus isolate HC/GZ/32/03, complete genome\n",
      "SARS coronavirus isolate CFB/SZ/94/03, complete genome\n",
      "SARS coronavirus WH20, complete genome\n",
      "SARS coronavirus GZ0401, complete genome\n",
      "SARS coronavirus GZ0402, complete genome\n",
      "SARS coronavirus PC4-13, complete genome\n",
      "SARS coronavirus PC4-136, complete genome\n",
      "SARS coronavirus PC4-227, complete genome\n",
      "Avian infectious bronchitis virus strain Mass 41, complete genome\n",
      "SARS coronavirus HC/SZ/61/03, complete genome\n",
      "Human coronavirus HKU1, complete genome\n",
      "Human coronavirus HKU1 genotype A, complete genome\n",
      "SARS Coronavirus CDC#200301157, complete genome\n",
      "Murine hepatitis virus strain A59, complete genome\n",
      "Human coronavirus OC43 strain ATCC VR-759, complete genome\n",
      "Human coronavirus OC43 serotype OC43-Paris, complete genome\n",
      "Streptococcus pyogenes MGAS10394, complete genome\n",
      "SARS coronavirus TJF, complete genome\n",
      "Avian infectious bronchitis virus partridge/GD/S14/2003, complete genome\n",
      "Avian infectious bronchitis virus isolate Peafowl/GD/KQ6/2003, complete genome\n",
      "SARS coronavirus LLJ-2004, complete genome\n",
      "Photobacterium profundum SS9; segment 12/12\n",
      "Human group 1 coronavirus associated with pneumonia, complete genome\n",
      "SARS coronavirus Sin842, complete genome\n",
      "SARS coronavirus Sin852, complete genome\n",
      "SARS coronavirus Sin3408, complete genome\n",
      "SARS coronavirus Sin3765V, complete genome\n",
      "SARS coronavirus Sin848, complete genome\n",
      "SARS coronavirus Sin849, complete genome\n",
      "SARS coronavirus Sin3725V, complete genome\n",
      "SARS coronavirus SinP1, complete genome\n",
      "SARS coronavirus SinP2, complete genome\n",
      "SARS coronavirus SinP3, complete genome\n",
      "SARS coronavirus SinP4, complete genome\n",
      "SARS coronavirus SinP5, complete genome\n",
      "SARS coronavirus Sin845, complete genome\n",
      "SARS coronavirus Sin846, complete genome\n",
      "SARS coronavirus Sin847, complete genome\n",
      "SARS coronavirus Sin850, complete genome\n",
      "SARS coronavirus Sin3408L, complete genome\n",
      "Human Coronavirus NL63, complete genome\n",
      "Human Coronavirus NL63, complete genome\n",
      "SARS coronavirus GZ02, complete genome\n",
      "Infectious bronchitis virus serotype California 99, complete genome\n",
      "SARS coronavirus GZ-B, complete genome\n",
      "SARS coronavirus GZ-C, complete genome\n",
      "SARS coronavirus HSZ2-A, complete genome\n",
      "SARS coronavirus HSZ-Bb, complete genome\n",
      "SARS coronavirus HSZ-Cb, complete genome\n",
      "SARS coronavirus HZS2-Fb, complete genome\n",
      "SARS coronavirus HZS2-D, complete genome\n",
      "SARS coronavirus HZS2-E, complete genome\n",
      "SARS coronavirus HZS2-Fc, complete genome\n",
      "SARS coronavirus HZS2-C, complete genome\n",
      "SARS coronavirus HGZ8L2, complete genome\n",
      "SARS coronavirus HSZ-Bc, complete genome\n",
      "SARS coronavirus HSZ-Cc, complete genome\n",
      "SARS coronavirus ZS-B, complete genome\n",
      "SARS coronavirus ZS-A, complete genome\n",
      "SARS coronavirus LC1, complete genome\n",
      "SARS coronavirus LC2, complete genome\n",
      "SARS coronavirus LC3, complete genome\n",
      "SARS coronavirus LC4, complete genome\n",
      "SARS coronavirus LC5, complete genome\n",
      "SARS coronavirus ZS-C, complete genome\n",
      "SARS coronavirus NS-1, complete genome\n",
      "SARS coronavirus TW10, complete genome\n",
      "SARS coronavirus TW11, complete genome\n",
      "SARS coronavirus TW2, complete genome\n",
      "SARS coronavirus TW3, complete genome\n",
      "SARS coronavirus TW4, complete genome\n",
      "SARS coronavirus TW5, complete genome\n",
      "SARS coronavirus TW6, complete genome\n",
      "SARS coronavirus TW7, complete genome\n",
      "SARS coronavirus TW8, complete genome\n",
      "SARS coronavirus TW9, complete genome\n",
      "SARS coronavirus ShanghaiQXC1, complete genome\n",
      "SARS coronavirus ShanghaiQXC2, complete genome\n",
      "SARS coronavirus GD69, complete genome\n",
      "SARS coronavirus Sino1-11, complete genome\n",
      "SARS coronavirus Sino3-11, complete genome\n",
      "SARS coronavirus CUHK-AG01, complete genome\n",
      "SARS coronavirus CUHK-AG02, complete genome\n",
      "SARS coronavirus CUHK-AG03, complete genome\n",
      "SARS coronavirus SoD, complete genome\n",
      "SARS coronavirus PUMC01, complete genome\n",
      "SARS coronavirus PUMC02, complete genome\n",
      "SARS coronavirus PUMC03, complete genome\n",
      "Human coronavirus OC43, complete genome\n",
      "SARS coronavirus AS, complete genome\n",
      "SARS coronavirus WHU, complete genome\n",
      "SARS coronavirus SZ3, complete genome\n",
      "SARS coronavirus SZ16, complete genome\n",
      "SARS coronavirus GZ50, complete genome\n",
      "SARS coronavirus TWC2, complete genome\n",
      "SARS coronavirus TWC3, complete genome\n",
      "SARS coronavirus FRA, complete genome\n",
      "SARS coronavirus ZMY 1, complete genome\n",
      "SARS coronavirus TWH genomic RNA, complete genome\n",
      "SARS coronavirus TWJ genomic RNA, complete genome\n",
      "SARS coronavirus TWK genomic RNA, complete genome\n",
      "SARS coronavirus TWS genomic RNA, complete genome\n",
      "SARS coronavirus TWY genomic RNA, complete genome\n",
      "SARS coronavirus Taiwan TC3, complete genome\n",
      "Avian infectious bronchitis virus isolate BJ, complete genome\n",
      "SARS coronavirus Taiwan TC1, complete genome\n",
      "SARS coronavirus Taiwan TC2, complete genome\n",
      "SARS coronavirus HSR 1, complete genome\n",
      "SARS coronavirus TWC, complete genome\n",
      "SARS coronavirus Frankfurt 1, complete genome\n",
      "SARS coronavirus ZJ01, complete genome\n",
      "Oceanobacillus iheyensis HTE831 DNA, complete genome\n",
      "SARS coronavirus TW1, complete genome\n",
      "SARS coronavirus Sin2500, complete genome\n",
      "SARS coronavirus Sin2677, complete genome\n",
      "SARS coronavirus Sin2679, complete genome\n",
      "SARS coronavirus Sin2748, complete genome\n",
      "SARS coronavirus Sin2774, complete genome\n",
      "SARS coronavirus CUHK-Su10, complete genome\n",
      "SARS coronavirus BJ04, complete genome\n",
      "SARS coronavirus BJ02, complete genome\n",
      "SARS coronavirus BJ01, complete genome\n",
      "SARS coronavirus GD01, complete genome\n",
      "SARS coronavirus BJ03, complete genome\n",
      "SARS coronavirus Urbani, complete genome\n",
      "SARS coronavirus HKU-39849, complete genome\n",
      "SARS coronavirus CUHK-W1, complete genome\n",
      "Severe acute respiratory syndrome-related coronavirus isolate Tor2, complete genome\n",
      "SARS coronavirus, complete genome\n",
      "Bovine coronavirus isolate BCoV-LUN, complete genome\n",
      "Bovine coronavirus strain Quebec, complete genome\n",
      "Bacillus halodurans C-125 DNA, complete genome\n",
      "Bovine coronavirus isolate BCoV-ENT, complete genome\n",
      "Bovine coronavirus, complete genome\n",
      "Porcine epidemic diarrhea virus strain CV777, complete genome\n",
      "Porcine epidemic diarrhea virus, complete genome\n",
      "Human coronavirus 229E, complete genome\n",
      "Human coronavirus 229E, complete genome\n",
      "Gill-associated virus from Australia, complete genome\n",
      "Murine hepatitis virus strain Penn 97-1, complete genome\n",
      "Murine hepatitis virus strain ML-10, complete genome\n",
      "Transmissible gastroenteritis virus complete genome, genomic RNA\n",
      "Feline infectious peritonitis virus, complete genome\n",
      "Transmissible gastroenteritis virus complete genome, genomic RNA\n",
      "Murine hepatitis virus strain 2, complete genome\n",
      "Mouse hepatitis virus strain MHV-A59 C12 mutant, complete genome\n",
      "Mouse hepatitis virus strain MHV-A59 C12 mutant, complete genome\n",
      "Bovine coronavirus strain Mebus, complete genome\n",
      "Avian infectious bronchitis virus pol protein, spike protein, small virion-associated protein, membrane protein, and nucleocapsid protein genes, complete cds\n",
      "Avian infectious bronchitis virus, complete genome\n"
     ]
    }
   ],
   "source": [
    "records = list(SeqIO.parse('Data/sequence.gb', 'genbank'))\n",
    "for record in records:\n",
    "    print(record.description)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SeqRecord(seq=Seq('GATTTAAGTGAATAGCCTAGCTATCTCACCCCCTCTCGTTCTCTTGCAGAACTC...AAA', IUPACAmbiguousDNA()), id='MG987420.1', name='MG987420', description='Middle East respiratory syndrome-related coronavirus isolate NL13892, complete genome', dbxrefs=[])"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'accessions': ['MG987420'],\n",
       " 'data_file_division': 'VRL',\n",
       " 'date': '23-FEB-2020',\n",
       " 'gi': '1386872237',\n",
       " 'keywords': [''],\n",
       " 'molecule_type': 'RNA',\n",
       " 'organism': 'Middle East respiratory syndrome-related coronavirus',\n",
       " 'references': [Reference(title='Discovery of novel bat betacoronaviruses in south China', ...),\n",
       "  Reference(title='Direct Submission', ...)],\n",
       " 'sequence_version': 1,\n",
       " 'source': 'Middle East respiratory syndrome-related coronavirus (MERS-CoV)',\n",
       " 'structured_comment': OrderedDict([('Assembly-Data',\n",
       "               OrderedDict([('Assembly Method', 'Lasergene Seqman v. V7.0'),\n",
       "                            ('Assembly Name', 'Assembly 1'),\n",
       "                            ('Sequencing Technology',\n",
       "                             'Illumina; Sanger dideoxy sequencing')]))]),\n",
       " 'taxonomy': ['Viruses',\n",
       "  'ssRNA viruses',\n",
       "  'ssRNA positive-strand viruses, no DNA stage',\n",
       "  'Nidovirales',\n",
       "  'Coronaviridae',\n",
       "  'Coronavirinae',\n",
       "  'Betacoronavirus'],\n",
       " 'topology': 'linear'}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records[0].annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "record_dict = SeqIO.to_dict(SeqIO.parse('Data/sequence.gb', 'genbank'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGATGCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTCGTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTAGGCGACGAGCTTGGCACTGATCCTTATGAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTGTTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGGCCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTCATGCACTTTGTCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTGCTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCCTTAAATTCCATAATCAAGACTATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTTATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCACCAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCAGACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACTACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAATTTATTGTCCAGCATGTCACAATTCAGAAGTAGGACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCGCACTATTGCCTTTGGAGGCTGTGTGTTCTCTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCACGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACAACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGATTATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAAAAAGGTGCCTGGAATATTGGTGAACAGAAATCAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCGTGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCCGCTATAACAATACTAGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTGGCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTTAAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAATTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGTAAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCCTACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAAGCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAAGGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAAATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACCACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAATTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAGAAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATTTGGTGCCACTTCTGCTGCTCTTCAACCTGAAGAAGAGCAAGAAGAAGATTGGTTAGATGATGATAGTCAACAAACTGTTGGTCAACAAGACGGCAGTGAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTCAACCTCAATTAGAGATGGAACTTACACCAGTTGTTCAGACTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACAGTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTACTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAAGTGGGTGGTAGTTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGTGAAGACATTCAACTTCTTAAGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTATTATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAATGTCTACTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGTGAAAAGCAAGTTGAACAAAAGATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTAAACCTTCAGTTGAACAGAGAAAACAAGATGATAAGAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAACTCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCAGATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAATGCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAATGGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTATTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGCAGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAATATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAACAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTTGGCTATGTAACACATGGCTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTTTCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAGAACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACAACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCACCTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTAAGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACAACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGTAAAACATTTTATGTTTTACCTAATGATGACACTCTACGTGTTGAGGCTTTTGAGTACTACCACACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAATGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAAATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTAACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAATGAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGTGGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAATTTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTCACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGTGAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAGACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAGTTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAGTTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAACCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATTTTTCCCTGACTTAAATGGTGATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAACCTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTGTCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTGAAGTCAGAGGACGCGCAGGGAATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAATAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTTACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTTGCTACTCATGGTTTAGCTGCTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTACAACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTATTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAAAGCATCTATGCCGACTACTATAGCAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAACTGATAAATATTATAATTTGGTTTTTACTATTAAGTGTTTGCCTAGGTTCTTTAATCTACTCAACCGCTGCTTTAGGTGTTTTAATGTCTAATTTAGGCATGCCTTCTTACTGTACTGGTTACAGAGAAGGCTATTTGAACTCTACTAATGTCACTATTGCAACCTACTGTACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTAGACACCTATCCTTCTTTAGAAACTATACAAATTACCATTTCATCTTTTAAATGGGATTTAACTGCTTTTGGCTTAGTTGCAGAGTGGTTTTTGGCATATATTCTTTTCACTAGGTTTTTCTATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCGATTTCAGCTATGGTTAGAATGTACATCTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGACGGTTGTAATTCATCAACTTGTATGATGTGTTACAAACGTAATAGAGCAACAAGAGTCGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACACAATTGGAATTGTGTTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGATGAAGTTGCGAGAGACTTGTCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCTTACATCGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAAGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAACCTGAGAGCTAATAACACTAAAGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCAAAATCAGCGTCTGTTTACTACAGTCAGCTTATGTGTCAACCTATACTGTTACTAGATCAGGCATTAGTGTCTGATGTTGGTGATAGTGCGGAAGTTGCAGTTAAAATGTTTGATGCTTACGTTAATACGTTTTCATCAACTTTTAACGTACCAATGGAAAAACTCAAAACACTAGTTGCAACTGCAGAAGCTGAACTTGCAAAGAATGTGTCCTTAGACAATGTCTTATCTACTTTTATTTCAGCAGCTCGGCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACATCAATCTGACATAGAAGTTACTGGCGATAGTTGTAATAACTATATGCTCACCTATAACAAAGTTGAAAACATGACACCCCGTGACCTTGGTGCTTGTATTGACTGTAGTGCGCGTCATATTAATGCGCAGGTAGCAAAAAGTCACAACATTGCTTTGATATGGAACGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGTAGTGCTGCTAAAAAGAATAACTTACCTTTTAAGTTGACATGTGCAACTACTAGACAAGTTGTTAATGTTGTAACAACAAAGATAGCACTTAAGGGTGGTAAAATTGTTAATAATTGGTTGAAGCAGTTAATTAAAGTTACACTTGTGTTCCTTTTTGTTGCTGCTATTTTCTATTTAATAACACCTGTTCATGTCATGTCTAAACATACTGACTTTTCAAGTGAAATCATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCATCTACAGATACTTGTTTTGCTAACAAACATGCTGATTTTGACACATGGTTTAGCCAGCGTGGTGGTAGTTATACTAATGACAAAGCTTGCCCATTGATTGCTGCAGTCATAACAAGAGAAGTGGGTTTTGTCGTGCCTGGTTTGCCTGGCACGATATTACGCACAACTAATGGTGACTTTTTGCATTTCTTACCTAGAGTTTTTAGTGCAGTTGGTAACATCTGTTACACACCATCAAAACTTATAGAGTACACTGACTTTGCAACATCAGCTTGTGTTTTGGCTGCTGAATGTACAATTTTTAAAGATGCTTCTGGTAAGCCAGTACCATATTGTTATGATACCAATGTACTAGAAGGTTCTGTTGCTTATGAAAGTTTACGCCCTGACACACGTTATGTGCTCATGGATGGCTCTATTATTCAATTTCCTAACACCTACCTTGAAGGTTCTGTTAGAGTGGTAACAACTTTTGATTCTGAGTACTGTAGGCACGGCACTTGTGAAAGATCAGAAGCTGGTGTTTGTGTATCTACTAGTGGTAGATGGGTACTTAACAATGATTATTACAGATCTTTACCAGGAGTTTTCTGTGGTGTAGATGCTGTAAATTTACTTACTAATATGTTTACACCACTAATTCAACCTATTGGTGCTTTGGACATATCAGCATCTATAGTAGCTGGTGGTATTGTAGCTATCGTAGTAACATGCCTTGCCTACTATTTTATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAATACTTTACTATTCCTTATGTCATTCACTGTACTCTGTTTAACACCAGTTTACTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTTTTAGCACATATTCAGTGGATGGTTATGTTCACACCTTTAGTACCTTTCTGGATAACAATTGCTTATATCATTTGTATTTCCACAAAGCATTTCTATTGGTTCTTTAGTAATTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACTTTTGAAGAAGCTGCGCTGTGCACCTTTTTGTTAAATAAAGAAATGTATCTAAAGTTGCGTAGTGATGTGCTATTACCTCTTACGCAATATAATAGATACTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGAGCAATGGATACAACTAGCTACAGAGAAGCTGCTTGTTGTCATCTCGCAAAGGCTCTCAATGACTTCAGTAACTCAGGTTCTGATGTTCTTTACCAACCACCACAAACCTCTATCACCTCAGCTGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAGGGTTGTATGGTACAAGTAACTTGTGGTACAACTACACTTAACGGTCTTTGGCTTGATGACGTAGTTTACTGTCCAAGACATGTGATCTGCACCTCTGAAGACATGCTTAACCCTAATTATGAAGATTTACTCATTCGTAAGTCTAATCATAATTTCTTGGTACAGGCTGGTAATGTTCAACTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTTAAGCTTAAGGTTGATACAGCCAATCCTAAGACACCTAAGTATAAGTTTGTTCGCATTCAACCAGGACAGACTTTTTCAGTGTTAGCTTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCTATGAGGCCCAATTTCACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAACATAGATTATGACTGTGTCTCTTTTTGTTACATGCACCATATGGAATTACCAACTGGAGTTCATGCTGGCACAGACTTAGAAGGTAACTTTTATGGACCTTTTGTTGACAGGCAAACAGCACAAGCAGCTGGTACGGACACAACTATTACAGTTAATGTTTTAGCTTGGTTGTACGCTGCTGTTATAAATGGAGACAGGTGGTTTCTCAATCGATTTACCACAACTCTTAATGACTTTAACCTTGTGGCTATGAAGTACAATTATGAACCTCTAACACAAGACCATGTTGACATACTAGGACCTCTTTCTGCTCAAACTGGAATTGCCGTTTTAGATATGTGTGCTTCATTAAAAGAATTACTGCAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTTGATGTTGTTAGACAATGCTCAGGTGTTACTTTCCAAAGTGCAGTGAAAAGAACAATCAAGGGTACACACCACTGGTTGTTACTCACAATTTTGACTTCACTTTTAGTTTTAGTCCAGAGTACTCAATGGTCTTTGTTCTTTTTTTTGTATGAAAATGCCTTTTTACCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCAATGATGTTTGTCAAACATAAGCATGCATTTCTCTGTTTGTTTTTGTTACCTTCTCTTGCCACTGTAGCTTATTTTAATATGGTCTATATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATGGTTGATACTAGTTTGTCTGGTTTTAAGCTAAAAGACTGTGTTATGTATGCATCAGCTGTAGTGTTACTAATCCTTATGACAGCAAGAACTGTGTATGATGATGGTGCTAGGAGAGTGTGGACACTTATGAATGTCTTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGATCAAGCCATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGCCCTATTTTCTTCATAACTGGTAATACACTTCAGTGTATAATGCTAGTTTATTGTTTCTTAGGCTATTTTTGTACTTGTTACTTTGGCCTCTTTTGTTTACTCAACCGCTACTTTAGACTGACTCTTGGTGTTTATGATTACTTAGTTTCTACACAGGAGTTTAGATATATGAATTCACAGGGACTACTCCCACCCAAGAATAGCATAGATGCCTTCAAACTCAACATTAAATTGTTGGGTGTTGGTGGCAAACCTTGTATCAAAGTAGCCACTGTACAGTCTAAAATGTCAGATGTAAAGTGCACATCAGTAGTCTTACTCTCAGTTTTGCAACAACTCAGAGTAGAATCATCATCTAAATTGTGGGCTCAATGTGTCCAGTTACACAATGACATTCTCTTAGCTAAAGATACTACTGAAGCCTTTGAAAAAATGGTTTCACTACTTTCTGTTTTGCTTTCCATGCAGGGTGCTGTAGACATAAACAAGCTTTGTGAAGAAATGCTGGACAACAGGGCAACCTTACAAGCTATAGCCTCAGAGTTTAGTTCCCTTCCATCATATGCAGCTTTTGCTACTGCTCAAGAAGCTTATGAGCAGGCTGTTGCTAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTGAAGAAGTCTTTGAATGTGGCTAAATCTGAATTTGACCGTGATGCAGCCATGCAACGTAAGTTGGAAAAGATGGCTGATCAAGCTATGACCCAAATGTATAAACAGGCTAGATCTGAGGACAAGAGGGCAAAAGTTACTAGTGCTATGCAGACAATGCTTTTCACTATGCTTAGAAAGTTGGATAATGATGCACTCAACAACATTATCAACAATGCAAGAGATGGTTGTGTTCCCTTGAACATAATACCTCTTACAACAGCAGCCAAACTAATGGTTGTCATACCAGACTATAACACATATAAAAATACGTGTGATGGTACAACATTTACTTATGCATCAGCATTGTGGGAAATCCAACAGGTTGTAGATGCAGATAGTAAAATTGTTCAACTTAGTGAAATTAGTATGGACAATTCACCTAATTTAGCATGGCCTCTTATTGTAACAGCTTTAAGGGCCAATTCTGCTGTCAAATTACAGAATAATGAGCTTAGTCCTGTTGCACTACGACAGATGTCTTGTGCTGCCGGTACTACACAAACTGCTTGCACTGATGACAATGCGTTAGCTTACTACAACACAACAAAGGGAGGTAGGTTTGTACTTGCACTGTTATCCGATTTACAGGATTTGAAATGGGCTAGATTCCCTAAGAGTGATGGAACTGGTACTATCTATACAGAACTGGAACCACCTTGTAGGTTTGTTACAGACACACCTAAAGGTCCTAAAGTGAAGTATTTATACTTTATTAAAGGATTAAACAACCTAAATAGAGGTATGGTACTTGGTAGTTTAGCTGCCACAGTACGTCTACAAGCTGGTAATGCAACAGAAGTGCCTGCCAATTCAACTGTATTATCTTTCTGTGCTTTTGCTGTAGATGCTGCTAAAGCTTACAAAGATTATCTAGCTAGTGGGGGACAACCAATCACTAATTGTGTTAAGATGTTGTGTACACACACTGGTACTGGTCAGGCAATAACAGTTACACCGGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCATCGTGTTGTCTGTACTGCCGTTGCCACATAGATCATCCAAATCCTAAAGGATTTTGTGACTTAAAAGGTAAGTATGTACAAATACCTACAACTTGTGCTAATGACCCTGTGGGTTTTACACTTAAAAACACAGTCTGTACCGTCTGCGGTATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGCTTCAGTCAGCTGATGCACAATCGTTTTTAAACGGGTTTGCGGTGTAAGTGCAGCCCGTCTTACACCGTGCGGCACAGGCACTAGTACTGATGTCGTATACAGGGCTTTTGACATCTACAATGATAAAGTAGCTGGTTTTGCTAAATTCCTAAAAACTAATTGTTGTCGCTTCCAAGAAAAGGACGAAGATGACAATTTAATTGATTCTTACTTTGTAGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAAGGATTGTCCAGCTGTTGCTAAACATGACTTCTTTAAGTTTAGAATAGACGGTGACATGGTACCACATATATCACGTCAACGTCTTACTAAATACACAATGGCAGACCTCGTCTATGCTTTAAGGCATTTTGATGAAGGTAATTGTGACACATTAAAAGAAATACTTGTCACATACAATTGTTGTGATGATGATTATTTCAATAAAAAGGACTGGTATGATTTTGTAGAAAACCCAGATATATTACGCGTATACGCCAACTTAGGTGAACGTGTACGCCAAGCTTTGTTAAAAACAGTACAATTCTGTGATGCCATGCGAAATGCTGGTATTGTTGGTGTACTGACATTAGATAATCAAGATCTCAATGGTAACTGGTATGATTTCGGTGATTTCATACAAACCACGCCAGGTAGTGGAGTTCCTGTTGTAGATTCTTATTATTCATTGTTAATGCCTATATTAACCTTGACCAGGGCTTTAACTGCAGAGTCACATGTTGACACTGACTTAACAAAGCCTTACATTAAGTGGGATTTGTTAAAATATGACTTCACGGAAGAGAGGTTAAAACTCTTTGACCGTTATTTTAAATATTGGGATCAGACATACCACCCAAATTGTGTTAACTGTTTGGATGACAGATGCATTCTGCATTGTGCAAACTTTAATGTTTTATTCTCTACAGTGTTCCCACCTACAAGTTTTGGACCACTAGTGAGAAAAATATTTGTTGATGGTGTTCCATTTGTAGTTTCAACTGGATACCACTTCAGAGAGCTAGGTGTTGTACATAATCAGGATGTAAACTTACATAGCTCTAGACTTAGTTTTAAGGAATTACTTGTGTATGCTGCTGACCCTGCTATGCACGCTGCTTCTGGTAATCTATTACTAGATAAACGCACTACGTGCTTTTCAGTAGCTGCACTTACTAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAACAAAGACTTCTATGACTTTGCTGTGTCTAAGGGTTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGTAATGCTGCTATCAGCGATTATGACTACTATCGTTATAATCTACCAACAATGTGTGATATCAGACAACTACTATTTGTAGTTGAAGTTGTTGATAAGTACTTTGATTGTTACGATGGTGGCTGTATTAATGCTAACCAAGTCATCGTCAACAACCTAGACAAATCAGCTGGTTTTCCATTTAATAAATGGGGTAAGGCTAGACTTTATTATGATTCAATGAGTTATGAGGATCAAGATGCACTTTTCGCATATACAAAACGTAATGTCATCCCTACTATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACCGTAGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAAAAATTATTGAAATCAATAGCCGCCACTAGAGGAGCTACTGTAGTAATTGGAACAAGCAAATTCTATGGTGGTTGGCACAACATGTTAAAAACTGTTTATAGTGATGTAGAAAACCCTCACCTTATGGGTTGGGATTATCCTAAATGTGATAGAGCCATGCCTAACATGCTTAGAATTATGGCCTCACTTGTTCTTGCTCGCAAACATACAACGTGTTGTAGCTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCTCAAGTATTGAGTGAAATGGTCATGTGTGGCGGTTCACTATATGTTAAACCAGGTGGAACCTCATCAGGAGATGCCACAACTGCTTATGCTAATAGTGTTTTTAACATTTGTCAAGCTGTCACGGCCAATGTTAATGCACTTTTATCTACTGATGGTAACAAAATTGCCGATAAGTATGTCCGCAATTTACAACACAGACTTTATGAGTGTCTCTATAGAAATAGAGATGTTGACACAGACTTTGTGAATGAGTTTTACGCATATTTGCGTAAACATTTCTCAATGATGATACTCTCTGACGATGCTGTTGTGTGTTTCAATAGCACTTATGCATCTCAAGGTCTAGTGGCTAGCATAAAGAACTTTAAGTCAGTTCTTTATTATCAAAACAATGTTTTTATGTCTGAAGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGCTCTCAACATACAATGCTAGTTAAACAGGGTGATGATTATGTGTACCTTCCTTACCCAGATCCATCAAGAATCCTAGGGGCCGGCTGTTTTGTAGATGATATCGTAAAAACAGATGGTACACTTATGATTGAACGGTTCGTGTCTTTAGCTATAGATGCTTACCCACTTACTAAACATCCTAATCAGGAGTATGCTGATGTCTTTCATTTGTACTTACAATACATAAGAAAGCTACATGATGAGTTAACAGGACACATGTTAGACATGTATTCTGTTATGCTTACTAATGATAACACTTCAAGGTATTGGGAACCTGAGTTTTATGAGGCTATGTACACACCGCATACAGTCTTACAGGCTGTTGGGGCTTGTGTTCTTTGCAATTCACAGACTTCATTAAGATGTGGTGCTTGCATACGTAGACCATTCTTATGTTGTAAATGCTGTTACGACCATGTCATATCAACATCACATAAATTAGTCTTGTCTGTTAATCCGTATGTTTGCAATGCTCCAGGTTGTGATGTCACAGATGTGACTCAACTTTACTTAGGAGGTATGAGCTATTATTGTAAATCACATAAACCACCCATTAGTTTTCCATTGTGTGCTAATGGACAAGTTTTTGGTTTATATAAAAATACATGTGTTGGTAGCGATAATGTTACTGACTTTAATGCAATTGCAACATGTGACTGGACAAATGCTGGTGATTACATTTTAGCTAACACCTGTACTGAAAGACTCAAGCTTTTTGCAGCAGAAACGCTCAAAGCTACTGAGGAGACATTTAAACTGTCTTATGGTATTGCTACTGTACGTGAAGTGCTGTCTGACAGAGAATTACATCTTTCATGGGAAGTTGGTAAACCTAGACCACCACTTAACCGAAATTATGTCTTTACTGGTTATCGTGTAACTAAAAACAGTAAAGTACAAATAGGAGAGTACACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTTTACCGAGGTACAACAACTTACAAATTAAATGTTGGTGATTATTTTGTGCTGACATCACATACAGTAATGCCATTAAGTGCACCTACACTAGTGCCACAAGAGCACTATGTTAGAATTACTGGCTTATACCCAACACTCAATATCTCAGATGAGTTTTCTAGCAATGTTGCAAATTATCAAAAGGTTGGTATGCAAAAGTATTCTACACTCCAGGGACCACCTGGTACTGGTAAGAGTCATTTTGCTATTGGCCTAGCTCTCTACTACCCTTCTGCTCGCATAGTGTATACAGCTTGCTCTCATGCCGCTGTTGATGCACTATGTGAGAAGGCATTAAAATATTTGCCTATAGATAAATGTAGTAGAATTATACCTGCACGTGCTCGTGTAGAGTGTTTTGATAAATTCAAAGTGAATTCAACATTAGAACAGTATGTCTTTTGTACTGTAAATGCATTGCCTGAGACGACAGCAGATATAGTTGTCTTTGATGAAATTTCAATGGCCACAAATTATGATTTGAGTGTTGTCAATGCCAGATTACGTGCTAAGCACTATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTGCTGAAATTGTTGACACTGTGAGTGCTTTGGTTTATGATAATAAGCTTAAAGCACATAAAGACAAATCAGCTCAATGCTTTAAAATGTTTTATAAGGGTGTTATCACGCATGATGTTTCATCTGCAATTAACAGGCCACAAATAGGCGTGGTAAGAGAATTCCTTACACGTAACCCTGCTTGGAGAAAAGCTGTCTTTATTTCACCTTATAATTCACAGAATGCTGTAGCCTCAAAGATTTTGGGACTACCAACTCAAACTGTTGATTCATCACAGGGCTCAGAATATGACTATGTCATATTCACTCAAACCACTGAAACAGCTCACTCTTGTAATGTAAACAGATTTAATGTTGCTATTACCAGAGCAAAAGTAGGCATACTTTGCATAATGTCTGATAGAGACCTTTATGACAAGTTGCAATTTACAAGTCTTGAAATTCCACGTAGGAATGTGGCAACTTTACAAGCTGAAAATGTAACAGGACTCTTTAAAGATTGTAGTAAGGTAATCACTGGGTTACATCCTACACAGGCACCTACACACCTCAGTGTTGACACTAAATTCAAAACTGAAGGTTTATGTGTTGACATACCTGGCATACCTAAGGACATGACCTATAGAAGACTCATCTCTATGATGGGTTTTAAAATGAATTATCAAGTTAATGGTTACCCTAACATGTTTATCACCCGCGAAGAAGCTATAAGACATGTACGTGCATGGATTGGCTTCGATGTCGAGGGGTGTCATGCTACTAGAGAAGCTGTTGGTACCAATTTACCTTTACAGCTAGGTTTTTCTACAGGTGTTAACCTAGTTGCTGTACCTACAGGTTATGTTGATACACCTAATAATACAGATTTTTCCAGAGTTAGTGCTAAACCACCGCCTGGAGATCAATTTAAACACCTCATACCACTTATGTACAAAGGACTTCCTTGGAATGTAGTGCGTATAAAGATTGTACAAATGTTAAGTGACACACTTAAAAATCTCTCTGACAGAGTCGTATTTGTCTTATGGGCACATGGCTTTGAGTTGACATCTATGAAGTATTTTGTGAAAATAGGACCTGAGCGCACCTGTTGTCTATGTGATAGACGTGCCACATGCTTTTCCACTGCTTCAGACACTTATGCCTGTTGGCATCATTCTATTGGATTTGATTACGTCTATAATCCGTTTATGATTGATGTTCAACAATGGGGTTTTACAGGTAACCTACAAAGCAACCATGATCTGTATTGTCAAGTCCATGGTAATGCACATGTAGCTAGTTGTGATGCAATCATGACTAGGTGTCTAGCTGTCCACGAGTGCTTTGTTAAGCGTGTTGACTGGACTATTGAATATCCTATAATTGGTGATGAACTGAAGATTAATGCGGCTTGTAGAAAGGTTCAACACATGGTTGTTAAAGCTGCATTATTAGCAGACAAATTCCCAGTTCTTCACGACATTGGTAACCCTAAAGCTATTAAGTGTGTACCTCAAGCTGATGTAGAATGGAAGTTCTATGATGCACAGCCTTGTAGTGACAAAGCTTATAAAATAGAAGAATTATTCTATTCTTATGCCACACATTCTGACAAATTCACAGATGGTGTATGCCTATTTTGGAATTGCAATGTCGATAGATATCCTGCTAATTCCATTGTTTGTAGATTTGACACTAGAGTGCTATCTAACCTTAACTTGCCTGGTTGTGATGGTGGCAGTTTGTATGTAAATAAACATGCATTCCACACACCAGCTTTTGATAAAAGTGCTTTTGTTAATTTAAAACAATTACCATTTTTCTATTACTCTGACAGTCCATGTGAGTCTCATGGAAAACAAGTAGTGTCAGATATAGATTATGTACCACTAAAGTCTGCTACGTGTATAACACGTTGCAATTTAGGTGGTGCTGTCTGTAGACATCATGCTAATGAGTACAGATTGTATCTCGATGCTTATAACATGATGATCTCAGCTGGCTTTAGCTTGTGGGTTTACAAACAATTTGATACTTATAACCTCTGGAACACTTTTACAAGACTTCAGAGTTTAGAAAATGTGGCTTTTAATGTTGTAAATAAGGGACACTTTGATGGACAACAGGGTGAAGTACCAGTTTCTATCATTAATAACACTGTTTACACAAAAGTTGATGGTGTTGATGTAGAATTGTTTGAAAATAAAACAACATTACCTGTTAATGTAGCATTTGAGCTTTGGGCTAAGCGCAACATTAAACCAGTACCAGAGGTGAAAATACTCAATAATTTGGGTGTGGACATTGCTGCTAATACTGTGATCTGGGACTACAAAAGAGATGCTCCAGCACATATATCTACTATTGGTGTTTGTTCTATGACTGACATAGCCAAGAAACCAACTGAAACGATTTGTGCACCACTCACTGTCTTTTTTGATGGTAGAGTTGATGGTCAAGTAGACTTATTTAGAAATGCCCGTAATGGTGTTCTTATTACAGAAGGTAGTGTTAAAGGTTTACAACCATCTGTAGGTCCCAAACAAGCTAGTCTTAATGGAGTCACATTAATTGGAGAAGCCGTAAAAACACAGTTCAATTATTATAAGAAAGTTGATGGTGTTGTCCAACAATTACCTGAAACTTACTTTACTCAGAGTAGAAATTTACAAGAATTTAAACCCAGGAGTCAAATGGAAATTGATTTCTTAGAATTAGCTATGGATGAATTCATTGAACGGTATAAATTAGAAGGCTATGCCTTCGAACATATCGTTTATGGAGATTTTAGTCATAGTCAGTTAGGTGGTTTACATCTACTGATTGGACTAGCTAAACGTTTTAAGGAATCACCTTTTGAATTAGAAGATTTTATTCCTATGGACAGTACAGTTAAAAACTATTTCATAACAGATGCGCAAACAGGTTCATCTAAGTGTGTGTGTTCTGTTATTGATTTATTACTTGATGATTTTGTTGAAATAATAAAATCCCAAGATTTATCTGTAGTTTCTAAGGTTGTCAAAGTGACTATTGACTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGCCATGTAGAAACATTTTACCCAAAATTACAATCTAGTCAAGCGTGGCAACCGGGTGTTGCTATGCCTAATCTTTACAAAATGCAAAGAATGCTATTAGAAAAGTGTGACCTTCAAAATTATGGTGATAGTGCAACATTACCTAAAGGCATAATGATGAATGTCGCAAAATATACTCAACTGTGTCAATATTTAAACACATTAACATTAGCTGTACCCTATAATATGAGAGTTATACATTTTGGTGCTGGTTCTGATAAAGGAGTTGCACCAGGTACAGCTGTTTTAAGACAGTGGTTGCCTACGGGTACGCTGCTTGTCGATTCAGATCTTAATGACTTTGTCTCTGATGCAGATTCAACTTTGATTGGTGATTGTGCAACTGTACATACAGCTAATAAATGGGATCTCATTATTAGTGATATGTACGACCCTAAGACTAAAAATGTTACAAAAGAAAATGACTCTAAAGAGGGTTTTTTCACTTACATTTGTGGGTTTATACAACAAAAGCTAGCTCTTGGAGGTTCCGTGGCTATAAAGATAACAGAACATTCTTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCCTTTGTTACTAATGTGAATGCGTCATCATCTGAAGCATTTTTAATTGGATGTAATTATCTTGGCAAACCACGCGAACAAATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAATACAAATCCAATTCAGTTGTCTTCCTATTCTTTATTTGACATGAGTAAATTTCCCCTTAAATTAAGGGGTACTGCTGTTATGTCTTTAAAAGAAGGTCAAATCAATGATATGATTTTATCTCTTCTTAGTAAAGGTAGACTTATAATTAGAGAAAACAACAGAGTTGTTATTTCTAGTGATGTTCTTGTTAACAACTAAACGAACAATGTTTGTTTTTCTTGTTTTATTGCCACTAGTCTCTAGTCAGTGTGTTAATCTTACAACCAGAACTCAATTACCCCCTGCATACACTAATTCTTTCACACGTGGTGTTTATTACCCTGACAAAGTTTTCAGATCCTCAGTTTTACATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCAATGGTACTAAGAGGTTTGATAACCCTGTCCTACCATTTAATGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACTTTAGATTCGAAGACCCAGTCCCTACTTATTGTTAATAACGCTACTAATGTTGTTATTAAAGTCTGTGAATTTCAATTTTGTAATGATCCATTTTTGGGTGTTTATTACCACAAAAACAACAAAAGTTGGATGGAAAGTGAGTTCAGAGTTTATTCTAGTGCGAATAATTGCACTTTTGAATATGTCTCTCAGCCTTTTCTTATGGACCTTGAAGGAAAACAGGGTAATTTCAAAAATCTTAGGGAATTTGTGTTTAAGAATATTGATGGTTATTTTAAAATATATTCTAAGCACACGCCTATTAATTTAGTGCGTGATCTCCCTCAGGGTTTTTCGGCTTTAGAACCATTGGTAGATTTGCCAATAGGTATTAACATCACTAGGTTTCAAACTTTACTTGCTTTACATAGAAGTTATTTGACTCCTGGTGATTCTTCTTCAGGTTGGACAGCTGGTGCTGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGATGCTGTAGACTGTGCACTTGACCCTCTCTCAGAAACAAAGTGTACGTTGAAATCCTTCACTGTAGAAAAAGGAATCTATCAAACTTCTAACTTTAGAGTCCAACCAACAGAATCTATTGTTAGATTTCCTAATATTACAAACTTGTGCCCTTTTGGTGAAGTTTTTAACGCCACCAGATTTGCATCTGTTTATGCTTGGAACAGGAAGAGAATCAGCAACTGTGTTGCTGATTATTCTGTCCTATATAATTCCGCATCATTTTCCACTTTTAAGTGTTATGGAGTGTCTCCTACTAAATTAAATGATCTCTGCTTTACTAATGTCTATGCAGATTCATTTGTAATTAGAGGTGATGAAGTCAGACAAATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAAATTACCAGATGATTTTACAGGCTGCGTTATAGCTTGGAATTCTAACAATCTTGATTCTAAGGTTGGTGGTAATTATAATTACCTGTATAGATTGTTTAGGAAGTCTAATCTCAAACCTTTTGAGAGAGATATTTCAACTGAAATCTATCAGGCCGGTAGCACACCTTGTAATGGTGTTGAAGGTTTTAATTGTTACTTTCCTTTACAATCATATGGTTTCCAACCCACTAATGGTGTTGGTTACCAACCATACAGAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTGGTTAAAAACAAATGTGTCAATTTCAACTTCAATGGTTTAACAGGCACAGGTGTTCTTACTGAGTCTAACAAAAAGTTTCTGCCTTTCCAACAATTTGGCAGAGACATTGCTGACACTACTGATGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTAACCAGGTTGCTGTTCTTTATCAGGATGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGTTTAATAGGGGCTGAACATGTCAACAACTCATATGAGTGTGACATACCCATTGGTGCAGGTATATGCGCTAGTTATCAGACTCAGACTAATTCTCCTCGGCGGGCACGTAGTGTAGCTAGTCAATCCATCATTGCCTACACTATGTCACTTGGTGCAGAAAATTCAGTTGCTTACTCTAATAACTCTATTGCCATACCCACAAATTTTACTATTAGTGTTACCACAGAAATTCTACCAGTGTCTATGACCAAGACATCAGTAGATTGTACAATGTACATTTGTGGTGATTCAACTGAATGCAGCAATCTTTTGTTGCAATATGGCAGTTTTTGTACACAATTAAACCGTGCTTTAACTGGAATAGCTGTTGAACAAGACAAAAACACCCAAGAAGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCTACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGTGATTGCCTTGGTGATATTGCTGCTAGAGACCTCATTTGTGCACAAAAGTTTAACGGCCTTACTGTTTTGCCACCTTTGCTCACAGATGAAATGATTGCTCAATACACTTCTGCACTGTTAGCGGGTACAATCACTTCTGGTTGGACCTTTGGTGCAGGTGCTGCATTACAAATACCATTTGCTATGCAAATGGCTTATAGGTTTAATGGTATTGGAGTTACACAGAATGTTCTCTATGAGAACCAAAAATTGATTGCCAACCAATTTAATAGTGCTATTGGCAAAATTCAAGACTCACTTTCTTCCACAGCAAGTGCACTTGGAAAACTTCAAGATGTGGTCAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGCTCCAATTTTGGTGCAATTTCAAGTGTTTTAAATGATATCCTTTCACGTCTTGACAAAGTTGAGGCTGAAGTGCAAATTGATAGGTTGATCACAGGCAGACTTCAAAGTTTGCAGACATATGTGACTCAACAATTAATTAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCTACTAAAATGTCAGAGTGTGTACTTGGACAATCAAAAAGAGTTGATTTTTGTGGAAAGGGCTATCATCTTATGTCCTTCCCTCAGTCAGCACCTCATGGTGTAGTCTTCTTGCATGTGACTTATGTCCCTGCACAAGAAAAGAACTTCACAACTGCTCCTGCCATTTGTCATGATGGAAAAGCACACTTTCCTCGTGAAGGTGTCTTTGTTTCAAATGGCACACACTGGTTTGTAACACAAAGGAATTTTTATGAACCACAAATCATTACTACAGACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAATTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGATTTAGGTGACATCTCTGGCATTAATGCTTCAGTTGTAAACATTCAAAAAGAAATTGACCGCCTCAATGAGGTTGCCAAGAATTTAAATGAATCTCTCATCGATCTCCAAGAACTTGGAAAGTATGAGCAGTATATAAAATGGCCATGGTACATTTGGCTAGGTTTTATAGCTGGCTTGATTGCCATAGTAATGGTGACAATTATGCTTTGCTGTATGACCAGTTGCTGTAGTTGTCTCAAGGGCTGTTGTTCTTGTGGATCCTGCTGCAAATTTGATGAAGACGACTCTGAGCCAGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATCTTCACAATTGGAACTGTAACTTTGAAGCAAGGTGAAATCAAGGATGCTACTCCTTCAGATTTTGTTCGCGCTACTGCAACGATACCGATACAAGCCTCACTCCCTTTCGGATGGCTTATTGTTGGCGTTGCACTTCTTGCTGTTTTTCAGAGCGCTTCCAAAATCATAACCCTCAAAAAGAGATGGCAACTAGCACTCTCCAAGGGTGTTCACTTTGTTTGCAACTTGCTGTTGTTGTTTGTAACAGTTTACTCACACCTTTTGCTCGTTGCTGCTGGCCTTGAAGCCCCTTTTCTCTATCTTTATGCTTTAGTCTACTTCTTGCAGAGTATAAACTTTGTAAGAATAATAATGAGGCTTTGGCTTTGCTGGAAATGCCGTTCCAAAAACCCATTACTTTATGATGCCAACTATTTTCTTTGCTGGCATACTAATTGTTACGACTATTGTATACCTTACAATAGTGTAACTTCTTCAATTGTCATTACTTCAGGTGATGGCACAACAAGTCCTATTTCTGAACATGACTACCAGATTGGTGGTTATACTGAAAAATGGGAATCTGGAGTAAAAGACTGTGTTGTATTACACAGTTACTTCACTTCAGACTATTACCAGCTGTACTCAACTCAATTGAGTACAGACACTGGTGTTGAACATGTTACCTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTCCAAATTCACACAATCGACGGTTCATCCGGAGTTGTTAATCCAGTAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTCGTGGTATTCTTGCTAGTTACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTCTGGTCTAAACGAACTAAATATTATATTAGTTTTTCTGTTTGGAACTTTAATTTTAGCCATGGCAGATTCCAACGGTACTATTACCGTTGAAGAGCTTAAAAAGCTCCTTGAACAATGGAACCTAGTAATAGGTTTCCTATTCCTTACATGGATTTGTCTTCTACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTATATAATTAAGTTAATTTTCCTCTGGCTGTTATGGCCAGTAACTTTAGCTTGTTTTGTGCTTGCTGCTGTTTACAGAATAAATTGGATCACCGGTGGAATTGCTATCGCAATGGCTTGTCTTGTAGGCTTGATGTGGCTCAGCTACTTCATTGCTTCTTTCAGACTGTTTGCGCGTACGCGTTCCATGTGGTCATTCAATCCAGAAACTAACATTCTTCTCAACGTGCCACTCCATGGCACTATTCTGACCAGACCGCTTCTAGAAAGTGAACTCGTAATCGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCTGGACACCATCTAGGACGCTGTGACATCAAGGACCTGCCTAAAGAAATCACTGTTGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGCAACTATAAATTAAACACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTACTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTACATCATAAACCTCATAATTAAAAATTTATCTAAGTCACTAACTGAGAATAAATATTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACGAACATGAAAATTATTCTTTTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAGTGTGTTAGAGGTACAACAGTACTTTTAAAAGAACCTTGCTCTTCTGGAACATACGAGGGCAATTCACCATTTCATCCTCTAGCTGATAACAAATTTGCACTGACTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGCGTAAAACACGTCTATCAGTTACGTGCCAGATCAGTTTCACCTAAACTGTTCATCAGACAAGAGGAAGTTCAAGAACTTTACTCTCCAATTTTTCTTATTGTTGCGGCAATAGTGTTTATAACACTTTGCTTCACACTCAAAAGAAAGACAGAATGATTGAACTTTCATTAATTGACTTCTATTTGTGCTTTTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTTATTATCTTTTGGTTCTCACTTGAACTGCAAGATCATAATGAAACTTGTCACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAATCATCACAACTGTAGCTGCATTTCACCAAGAATGTAGTTTACAGTCATGTACTCAACATCAACCATATGTAGTTGATGACCCGTGTCCTATTCACTTCTATTCTAAATGGTATATTAGAGTAGGAGCTAGAAAATCAGCACCTTTAATTGAATTGTGCGTGGATGAGGCTGGTTCTAAATCACCCATTCAGTACATCGATATCGGTAATTATACAGTTTCCTGTTTACCTTTTACAATTAATTGCCAGGAACCTAAATTGGGTAGTCTTGTAGTGCGTTGTTCGTTCTATGAAGACTTTTTAGAGTATCATGACGTTCGTGTTGTTTTAGATTTCATCTAAACGAACAAACTAAAATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAAACTCATGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATCTCACATAGCAATCTTTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACATTTTCACCGAGGCCACGCGGAGTACGATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAATGACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n"
     ]
    }
   ],
   "source": [
    "print(record_dict['MN908947.3'].seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sjsfnsjns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"roger.guimera@urv.cat\"  # Always tell NCBI who you are\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id='MN908947.3', rettype=\"gb\", retmode=\"text\")\n",
    "patient_rec = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: MN908947.3\n",
      "Name: MN908947\n",
      "Description: Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
      "Number of features: 23\n",
      "/taxonomy=['Viruses', 'Riboviria', 'Nidovirales', 'Cornidovirineae', 'Coronaviridae', 'Orthocoronavirinae', 'Betacoronavirus', 'Sarbecovirus']\n",
      "/topology=linear\n",
      "/keywords=['']\n",
      "/date=11-FEB-2020\n",
      "/sequence_version=3\n",
      "/comment=On Jan 17, 2020 this sequence version replaced MN908947.2.\n",
      "/references=[Reference(title='A new coronavirus associated with human respiratory disease in China', ...), Reference(title='Direct Submission', ...)]\n",
      "/data_file_division=VRL\n",
      "/source=Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n",
      "/organism=Severe acute respiratory syndrome coronavirus 2\n",
      "/structured_comment=OrderedDict([('Assembly-Data', OrderedDict([('Assembly Method', 'Megahit v. V1.1.3'), ('Sequencing Technology', 'Illumina')]))])\n",
      "/accessions=['MN908947']\n",
      "/molecule_type=ss-RNA\n",
      "Seq('ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGT...AAA', IUPACAmbiguousDNA())\n"
     ]
    }
   ],
   "source": [
    "print(patient_rec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGATGCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTCGTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTAGGCGACGAGCTTGGCACTGATCCTTATGAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTGTTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGGCCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTCATGCACTTTGTCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTGCTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCCTTAAATTCCATAATCAAGACTATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTTATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCACCAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCAGACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACTACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAATTTATTGTCCAGCATGTCACAATTCAGAAGTAGGACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCGCACTATTGCCTTTGGAGGCTGTGTGTTCTCTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCACGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACAACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGATTATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAAAAAGGTGCCTGGAATATTGGTGAACAGAAATCAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCGTGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCCGCTATAACAATACTAGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTGGCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTTAAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAATTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGTAAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCCTACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAAGCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAAGGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAAATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACCACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAATTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAGAAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATTTGGTGCCACTTCTGCTGCTCTTCAACCTGAAGAAGAGCAAGAAGAAGATTGGTTAGATGATGATAGTCAACAAACTGTTGGTCAACAAGACGGCAGTGAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTCAACCTCAATTAGAGATGGAACTTACACCAGTTGTTCAGACTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACAGTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTACTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAAGTGGGTGGTAGTTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGTGAAGACATTCAACTTCTTAAGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTATTATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAATGTCTACTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGTGAAAAGCAAGTTGAACAAAAGATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTAAACCTTCAGTTGAACAGAGAAAACAAGATGATAAGAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAACTCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCAGATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAATGCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAATGGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTATTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGCAGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAATATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAACAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTTGGCTATGTAACACATGGCTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTTTCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAGAACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACAACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCACCTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTAAGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACAACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGTAAAACATTTTATGTTTTACCTAATGATGACACTCTACGTGTTGAGGCTTTTGAGTACTACCACACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAATGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAAATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTAACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAATGAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGTGGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAATTTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTCACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGTGAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAGACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAGTTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAGTTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAACCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATTTTTCCCTGACTTAAATGGTGATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAACCTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTGTCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTGAAGTCAGAGGACGCGCAGGGAATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAATAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTTACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTTGCTACTCATGGTTTAGCTGCTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTACAACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTATTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAAAGCATCTATGCCGACTACTATAGCAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAACTGATAAATATTATAATTTGGTTTTTACTATTAAGTGTTTGCCTAGGTTCTTTAATCTACTCAACCGCTGCTTTAGGTGTTTTAATGTCTAATTTAGGCATGCCTTCTTACTGTACTGGTTACAGAGAAGGCTATTTGAACTCTACTAATGTCACTATTGCAACCTACTGTACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTAGACACCTATCCTTCTTTAGAAACTATACAAATTACCATTTCATCTTTTAAATGGGATTTAACTGCTTTTGGCTTAGTTGCAGAGTGGTTTTTGGCATATATTCTTTTCACTAGGTTTTTCTATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCGATTTCAGCTATGGTTAGAATGTACATCTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGACGGTTGTAATTCATCAACTTGTATGATGTGTTACAAACGTAATAGAGCAACAAGAGTCGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACACAATTGGAATTGTGTTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGATGAAGTTGCGAGAGACTTGTCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCTTACATCGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAAGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAACCTGAGAGCTAATAACACTAAAGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCAAAATCAGCGTCTGTTTACTACAGTCAGCTTATGTGTCAACCTATACTGTTACTAGATCAGGCATTAGTGTCTGATGTTGGTGATAGTGCGGAAGTTGCAGTTAAAATGTTTGATGCTTACGTTAATACGTTTTCATCAACTTTTAACGTACCAATGGAAAAACTCAAAACACTAGTTGCAACTGCAGAAGCTGAACTTGCAAAGAATGTGTCCTTAGACAATGTCTTATCTACTTTTATTTCAGCAGCTCGGCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACATCAATCTGACATAGAAGTTACTGGCGATAGTTGTAATAACTATATGCTCACCTATAACAAAGTTGAAAACATGACACCCCGTGACCTTGGTGCTTGTATTGACTGTAGTGCGCGTCATATTAATGCGCAGGTAGCAAAAAGTCACAACATTGCTTTGATATGGAACGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGTAGTGCTGCTAAAAAGAATAACTTACCTTTTAAGTTGACATGTGCAACTACTAGACAAGTTGTTAATGTTGTAACAACAAAGATAGCACTTAAGGGTGGTAAAATTGTTAATAATTGGTTGAAGCAGTTAATTAAAGTTACACTTGTGTTCCTTTTTGTTGCTGCTATTTTCTATTTAATAACACCTGTTCATGTCATGTCTAAACATACTGACTTTTCAAGTGAAATCATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCATCTACAGATACTTGTTTTGCTAACAAACATGCTGATTTTGACACATGGTTTAGCCAGCGTGGTGGTAGTTATACTAATGACAAAGCTTGCCCATTGATTGCTGCAGTCATAACAAGAGAAGTGGGTTTTGTCGTGCCTGGTTTGCCTGGCACGATATTACGCACAACTAATGGTGACTTTTTGCATTTCTTACCTAGAGTTTTTAGTGCAGTTGGTAACATCTGTTACACACCATCAAAACTTATAGAGTACACTGACTTTGCAACATCAGCTTGTGTTTTGGCTGCTGAATGTACAATTTTTAAAGATGCTTCTGGTAAGCCAGTACCATATTGTTATGATACCAATGTACTAGAAGGTTCTGTTGCTTATGAAAGTTTACGCCCTGACACACGTTATGTGCTCATGGATGGCTCTATTATTCAATTTCCTAACACCTACCTTGAAGGTTCTGTTAGAGTGGTAACAACTTTTGATTCTGAGTACTGTAGGCACGGCACTTGTGAAAGATCAGAAGCTGGTGTTTGTGTATCTACTAGTGGTAGATGGGTACTTAACAATGATTATTACAGATCTTTACCAGGAGTTTTCTGTGGTGTAGATGCTGTAAATTTACTTACTAATATGTTTACACCACTAATTCAACCTATTGGTGCTTTGGACATATCAGCATCTATAGTAGCTGGTGGTATTGTAGCTATCGTAGTAACATGCCTTGCCTACTATTTTATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAATACTTTACTATTCCTTATGTCATTCACTGTACTCTGTTTAACACCAGTTTACTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTTTTAGCACATATTCAGTGGATGGTTATGTTCACACCTTTAGTACCTTTCTGGATAACAATTGCTTATATCATTTGTATTTCCACAAAGCATTTCTATTGGTTCTTTAGTAATTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACTTTTGAAGAAGCTGCGCTGTGCACCTTTTTGTTAAATAAAGAAATGTATCTAAAGTTGCGTAGTGATGTGCTATTACCTCTTACGCAATATAATAGATACTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGAGCAATGGATACAACTAGCTACAGAGAAGCTGCTTGTTGTCATCTCGCAAAGGCTCTCAATGACTTCAGTAACTCAGGTTCTGATGTTCTTTACCAACCACCACAAACCTCTATCACCTCAGCTGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAGGGTTGTATGGTACAAGTAACTTGTGGTACAACTACACTTAACGGTCTTTGGCTTGATGACGTAGTTTACTGTCCAAGACATGTGATCTGCACCTCTGAAGACATGCTTAACCCTAATTATGAAGATTTACTCATTCGTAAGTCTAATCATAATTTCTTGGTACAGGCTGGTAATGTTCAACTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTTAAGCTTAAGGTTGATACAGCCAATCCTAAGACACCTAAGTATAAGTTTGTTCGCATTCAACCAGGACAGACTTTTTCAGTGTTAGCTTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCTATGAGGCCCAATTTCACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAACATAGATTATGACTGTGTCTCTTTTTGTTACATGCACCATATGGAATTACCAACTGGAGTTCATGCTGGCACAGACTTAGAAGGTAACTTTTATGGACCTTTTGTTGACAGGCAAACAGCACAAGCAGCTGGTACGGACACAACTATTACAGTTAATGTTTTAGCTTGGTTGTACGCTGCTGTTATAAATGGAGACAGGTGGTTTCTCAATCGATTTACCACAACTCTTAATGACTTTAACCTTGTGGCTATGAAGTACAATTATGAACCTCTAACACAAGACCATGTTGACATACTAGGACCTCTTTCTGCTCAAACTGGAATTGCCGTTTTAGATATGTGTGCTTCATTAAAAGAATTACTGCAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTTGATGTTGTTAGACAATGCTCAGGTGTTACTTTCCAAAGTGCAGTGAAAAGAACAATCAAGGGTACACACCACTGGTTGTTACTCACAATTTTGACTTCACTTTTAGTTTTAGTCCAGAGTACTCAATGGTCTTTGTTCTTTTTTTTGTATGAAAATGCCTTTTTACCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCAATGATGTTTGTCAAACATAAGCATGCATTTCTCTGTTTGTTTTTGTTACCTTCTCTTGCCACTGTAGCTTATTTTAATATGGTCTATATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATGGTTGATACTAGTTTGTCTGGTTTTAAGCTAAAAGACTGTGTTATGTATGCATCAGCTGTAGTGTTACTAATCCTTATGACAGCAAGAACTGTGTATGATGATGGTGCTAGGAGAGTGTGGACACTTATGAATGTCTTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGATCAAGCCATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGCCCTATTTTCTTCATAACTGGTAATACACTTCAGTGTATAATGCTAGTTTATTGTTTCTTAGGCTATTTTTGTACTTGTTACTTTGGCCTCTTTTGTTTACTCAACCGCTACTTTAGACTGACTCTTGGTGTTTATGATTACTTAGTTTCTACACAGGAGTTTAGATATATGAATTCACAGGGACTACTCCCACCCAAGAATAGCATAGATGCCTTCAAACTCAACATTAAATTGTTGGGTGTTGGTGGCAAACCTTGTATCAAAGTAGCCACTGTACAGTCTAAAATGTCAGATGTAAAGTGCACATCAGTAGTCTTACTCTCAGTTTTGCAACAACTCAGAGTAGAATCATCATCTAAATTGTGGGCTCAATGTGTCCAGTTACACAATGACATTCTCTTAGCTAAAGATACTACTGAAGCCTTTGAAAAAATGGTTTCACTACTTTCTGTTTTGCTTTCCATGCAGGGTGCTGTAGACATAAACAAGCTTTGTGAAGAAATGCTGGACAACAGGGCAACCTTACAAGCTATAGCCTCAGAGTTTAGTTCCCTTCCATCATATGCAGCTTTTGCTACTGCTCAAGAAGCTTATGAGCAGGCTGTTGCTAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTGAAGAAGTCTTTGAATGTGGCTAAATCTGAATTTGACCGTGATGCAGCCATGCAACGTAAGTTGGAAAAGATGGCTGATCAAGCTATGACCCAAATGTATAAACAGGCTAGATCTGAGGACAAGAGGGCAAAAGTTACTAGTGCTATGCAGACAATGCTTTTCACTATGCTTAGAAAGTTGGATAATGATGCACTCAACAACATTATCAACAATGCAAGAGATGGTTGTGTTCCCTTGAACATAATACCTCTTACAACAGCAGCCAAACTAATGGTTGTCATACCAGACTATAACACATATAAAAATACGTGTGATGGTACAACATTTACTTATGCATCAGCATTGTGGGAAATCCAACAGGTTGTAGATGCAGATAGTAAAATTGTTCAACTTAGTGAAATTAGTATGGACAATTCACCTAATTTAGCATGGCCTCTTATTGTAACAGCTTTAAGGGCCAATTCTGCTGTCAAATTACAGAATAATGAGCTTAGTCCTGTTGCACTACGACAGATGTCTTGTGCTGCCGGTACTACACAAACTGCTTGCACTGATGACAATGCGTTAGCTTACTACAACACAACAAAGGGAGGTAGGTTTGTACTTGCACTGTTATCCGATTTACAGGATTTGAAATGGGCTAGATTCCCTAAGAGTGATGGAACTGGTACTATCTATACAGAACTGGAACCACCTTGTAGGTTTGTTACAGACACACCTAAAGGTCCTAAAGTGAAGTATTTATACTTTATTAAAGGATTAAACAACCTAAATAGAGGTATGGTACTTGGTAGTTTAGCTGCCACAGTACGTCTACAAGCTGGTAATGCAACAGAAGTGCCTGCCAATTCAACTGTATTATCTTTCTGTGCTTTTGCTGTAGATGCTGCTAAAGCTTACAAAGATTATCTAGCTAGTGGGGGACAACCAATCACTAATTGTGTTAAGATGTTGTGTACACACACTGGTACTGGTCAGGCAATAACAGTTACACCGGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCATCGTGTTGTCTGTACTGCCGTTGCCACATAGATCATCCAAATCCTAAAGGATTTTGTGACTTAAAAGGTAAGTATGTACAAATACCTACAACTTGTGCTAATGACCCTGTGGGTTTTACACTTAAAAACACAGTCTGTACCGTCTGCGGTATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGCTTCAGTCAGCTGATGCACAATCGTTTTTAAACGGGTTTGCGGTGTAAGTGCAGCCCGTCTTACACCGTGCGGCACAGGCACTAGTACTGATGTCGTATACAGGGCTTTTGACATCTACAATGATAAAGTAGCTGGTTTTGCTAAATTCCTAAAAACTAATTGTTGTCGCTTCCAAGAAAAGGACGAAGATGACAATTTAATTGATTCTTACTTTGTAGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAAGGATTGTCCAGCTGTTGCTAAACATGACTTCTTTAAGTTTAGAATAGACGGTGACATGGTACCACATATATCACGTCAACGTCTTACTAAATACACAATGGCAGACCTCGTCTATGCTTTAAGGCATTTTGATGAAGGTAATTGTGACACATTAAAAGAAATACTTGTCACATACAATTGTTGTGATGATGATTATTTCAATAAAAAGGACTGGTATGATTTTGTAGAAAACCCAGATATATTACGCGTATACGCCAACTTAGGTGAACGTGTACGCCAAGCTTTGTTAAAAACAGTACAATTCTGTGATGCCATGCGAAATGCTGGTATTGTTGGTGTACTGACATTAGATAATCAAGATCTCAATGGTAACTGGTATGATTTCGGTGATTTCATACAAACCACGCCAGGTAGTGGAGTTCCTGTTGTAGATTCTTATTATTCATTGTTAATGCCTATATTAACCTTGACCAGGGCTTTAACTGCAGAGTCACATGTTGACACTGACTTAACAAAGCCTTACATTAAGTGGGATTTGTTAAAATATGACTTCACGGAAGAGAGGTTAAAACTCTTTGACCGTTATTTTAAATATTGGGATCAGACATACCACCCAAATTGTGTTAACTGTTTGGATGACAGATGCATTCTGCATTGTGCAAACTTTAATGTTTTATTCTCTACAGTGTTCCCACCTACAAGTTTTGGACCACTAGTGAGAAAAATATTTGTTGATGGTGTTCCATTTGTAGTTTCAACTGGATACCACTTCAGAGAGCTAGGTGTTGTACATAATCAGGATGTAAACTTACATAGCTCTAGACTTAGTTTTAAGGAATTACTTGTGTATGCTGCTGACCCTGCTATGCACGCTGCTTCTGGTAATCTATTACTAGATAAACGCACTACGTGCTTTTCAGTAGCTGCACTTACTAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAACAAAGACTTCTATGACTTTGCTGTGTCTAAGGGTTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGTAATGCTGCTATCAGCGATTATGACTACTATCGTTATAATCTACCAACAATGTGTGATATCAGACAACTACTATTTGTAGTTGAAGTTGTTGATAAGTACTTTGATTGTTACGATGGTGGCTGTATTAATGCTAACCAAGTCATCGTCAACAACCTAGACAAATCAGCTGGTTTTCCATTTAATAAATGGGGTAAGGCTAGACTTTATTATGATTCAATGAGTTATGAGGATCAAGATGCACTTTTCGCATATACAAAACGTAATGTCATCCCTACTATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACCGTAGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAAAAATTATTGAAATCAATAGCCGCCACTAGAGGAGCTACTGTAGTAATTGGAACAAGCAAATTCTATGGTGGTTGGCACAACATGTTAAAAACTGTTTATAGTGATGTAGAAAACCCTCACCTTATGGGTTGGGATTATCCTAAATGTGATAGAGCCATGCCTAACATGCTTAGAATTATGGCCTCACTTGTTCTTGCTCGCAAACATACAACGTGTTGTAGCTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCTCAAGTATTGAGTGAAATGGTCATGTGTGGCGGTTCACTATATGTTAAACCAGGTGGAACCTCATCAGGAGATGCCACAACTGCTTATGCTAATAGTGTTTTTAACATTTGTCAAGCTGTCACGGCCAATGTTAATGCACTTTTATCTACTGATGGTAACAAAATTGCCGATAAGTATGTCCGCAATTTACAACACAGACTTTATGAGTGTCTCTATAGAAATAGAGATGTTGACACAGACTTTGTGAATGAGTTTTACGCATATTTGCGTAAACATTTCTCAATGATGATACTCTCTGACGATGCTGTTGTGTGTTTCAATAGCACTTATGCATCTCAAGGTCTAGTGGCTAGCATAAAGAACTTTAAGTCAGTTCTTTATTATCAAAACAATGTTTTTATGTCTGAAGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGCTCTCAACATACAATGCTAGTTAAACAGGGTGATGATTATGTGTACCTTCCTTACCCAGATCCATCAAGAATCCTAGGGGCCGGCTGTTTTGTAGATGATATCGTAAAAACAGATGGTACACTTATGATTGAACGGTTCGTGTCTTTAGCTATAGATGCTTACCCACTTACTAAACATCCTAATCAGGAGTATGCTGATGTCTTTCATTTGTACTTACAATACATAAGAAAGCTACATGATGAGTTAACAGGACACATGTTAGACATGTATTCTGTTATGCTTACTAATGATAACACTTCAAGGTATTGGGAACCTGAGTTTTATGAGGCTATGTACACACCGCATACAGTCTTACAGGCTGTTGGGGCTTGTGTTCTTTGCAATTCACAGACTTCATTAAGATGTGGTGCTTGCATACGTAGACCATTCTTATGTTGTAAATGCTGTTACGACCATGTCATATCAACATCACATAAATTAGTCTTGTCTGTTAATCCGTATGTTTGCAATGCTCCAGGTTGTGATGTCACAGATGTGACTCAACTTTACTTAGGAGGTATGAGCTATTATTGTAAATCACATAAACCACCCATTAGTTTTCCATTGTGTGCTAATGGACAAGTTTTTGGTTTATATAAAAATACATGTGTTGGTAGCGATAATGTTACTGACTTTAATGCAATTGCAACATGTGACTGGACAAATGCTGGTGATTACATTTTAGCTAACACCTGTACTGAAAGACTCAAGCTTTTTGCAGCAGAAACGCTCAAAGCTACTGAGGAGACATTTAAACTGTCTTATGGTATTGCTACTGTACGTGAAGTGCTGTCTGACAGAGAATTACATCTTTCATGGGAAGTTGGTAAACCTAGACCACCACTTAACCGAAATTATGTCTTTACTGGTTATCGTGTAACTAAAAACAGTAAAGTACAAATAGGAGAGTACACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTTTACCGAGGTACAACAACTTACAAATTAAATGTTGGTGATTATTTTGTGCTGACATCACATACAGTAATGCCATTAAGTGCACCTACACTAGTGCCACAAGAGCACTATGTTAGAATTACTGGCTTATACCCAACACTCAATATCTCAGATGAGTTTTCTAGCAATGTTGCAAATTATCAAAAGGTTGGTATGCAAAAGTATTCTACACTCCAGGGACCACCTGGTACTGGTAAGAGTCATTTTGCTATTGGCCTAGCTCTCTACTACCCTTCTGCTCGCATAGTGTATACAGCTTGCTCTCATGCCGCTGTTGATGCACTATGTGAGAAGGCATTAAAATATTTGCCTATAGATAAATGTAGTAGAATTATACCTGCACGTGCTCGTGTAGAGTGTTTTGATAAATTCAAAGTGAATTCAACATTAGAACAGTATGTCTTTTGTACTGTAAATGCATTGCCTGAGACGACAGCAGATATAGTTGTCTTTGATGAAATTTCAATGGCCACAAATTATGATTTGAGTGTTGTCAATGCCAGATTACGTGCTAAGCACTATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTGCTGAAATTGTTGACACTGTGAGTGCTTTGGTTTATGATAATAAGCTTAAAGCACATAAAGACAAATCAGCTCAATGCTTTAAAATGTTTTATAAGGGTGTTATCACGCATGATGTTTCATCTGCAATTAACAGGCCACAAATAGGCGTGGTAAGAGAATTCCTTACACGTAACCCTGCTTGGAGAAAAGCTGTCTTTATTTCACCTTATAATTCACAGAATGCTGTAGCCTCAAAGATTTTGGGACTACCAACTCAAACTGTTGATTCATCACAGGGCTCAGAATATGACTATGTCATATTCACTCAAACCACTGAAACAGCTCACTCTTGTAATGTAAACAGATTTAATGTTGCTATTACCAGAGCAAAAGTAGGCATACTTTGCATAATGTCTGATAGAGACCTTTATGACAAGTTGCAATTTACAAGTCTTGAAATTCCACGTAGGAATGTGGCAACTTTACAAGCTGAAAATGTAACAGGACTCTTTAAAGATTGTAGTAAGGTAATCACTGGGTTACATCCTACACAGGCACCTACACACCTCAGTGTTGACACTAAATTCAAAACTGAAGGTTTATGTGTTGACATACCTGGCATACCTAAGGACATGACCTATAGAAGACTCATCTCTATGATGGGTTTTAAAATGAATTATCAAGTTAATGGTTACCCTAACATGTTTATCACCCGCGAAGAAGCTATAAGACATGTACGTGCATGGATTGGCTTCGATGTCGAGGGGTGTCATGCTACTAGAGAAGCTGTTGGTACCAATTTACCTTTACAGCTAGGTTTTTCTACAGGTGTTAACCTAGTTGCTGTACCTACAGGTTATGTTGATACACCTAATAATACAGATTTTTCCAGAGTTAGTGCTAAACCACCGCCTGGAGATCAATTTAAACACCTCATACCACTTATGTACAAAGGACTTCCTTGGAATGTAGTGCGTATAAAGATTGTACAAATGTTAAGTGACACACTTAAAAATCTCTCTGACAGAGTCGTATTTGTCTTATGGGCACATGGCTTTGAGTTGACATCTATGAAGTATTTTGTGAAAATAGGACCTGAGCGCACCTGTTGTCTATGTGATAGACGTGCCACATGCTTTTCCACTGCTTCAGACACTTATGCCTGTTGGCATCATTCTATTGGATTTGATTACGTCTATAATCCGTTTATGATTGATGTTCAACAATGGGGTTTTACAGGTAACCTACAAAGCAACCATGATCTGTATTGTCAAGTCCATGGTAATGCACATGTAGCTAGTTGTGATGCAATCATGACTAGGTGTCTAGCTGTCCACGAGTGCTTTGTTAAGCGTGTTGACTGGACTATTGAATATCCTATAATTGGTGATGAACTGAAGATTAATGCGGCTTGTAGAAAGGTTCAACACATGGTTGTTAAAGCTGCATTATTAGCAGACAAATTCCCAGTTCTTCACGACATTGGTAACCCTAAAGCTATTAAGTGTGTACCTCAAGCTGATGTAGAATGGAAGTTCTATGATGCACAGCCTTGTAGTGACAAAGCTTATAAAATAGAAGAATTATTCTATTCTTATGCCACACATTCTGACAAATTCACAGATGGTGTATGCCTATTTTGGAATTGCAATGTCGATAGATATCCTGCTAATTCCATTGTTTGTAGATTTGACACTAGAGTGCTATCTAACCTTAACTTGCCTGGTTGTGATGGTGGCAGTTTGTATGTAAATAAACATGCATTCCACACACCAGCTTTTGATAAAAGTGCTTTTGTTAATTTAAAACAATTACCATTTTTCTATTACTCTGACAGTCCATGTGAGTCTCATGGAAAACAAGTAGTGTCAGATATAGATTATGTACCACTAAAGTCTGCTACGTGTATAACACGTTGCAATTTAGGTGGTGCTGTCTGTAGACATCATGCTAATGAGTACAGATTGTATCTCGATGCTTATAACATGATGATCTCAGCTGGCTTTAGCTTGTGGGTTTACAAACAATTTGATACTTATAACCTCTGGAACACTTTTACAAGACTTCAGAGTTTAGAAAATGTGGCTTTTAATGTTGTAAATAAGGGACACTTTGATGGACAACAGGGTGAAGTACCAGTTTCTATCATTAATAACACTGTTTACACAAAAGTTGATGGTGTTGATGTAGAATTGTTTGAAAATAAAACAACATTACCTGTTAATGTAGCATTTGAGCTTTGGGCTAAGCGCAACATTAAACCAGTACCAGAGGTGAAAATACTCAATAATTTGGGTGTGGACATTGCTGCTAATACTGTGATCTGGGACTACAAAAGAGATGCTCCAGCACATATATCTACTATTGGTGTTTGTTCTATGACTGACATAGCCAAGAAACCAACTGAAACGATTTGTGCACCACTCACTGTCTTTTTTGATGGTAGAGTTGATGGTCAAGTAGACTTATTTAGAAATGCCCGTAATGGTGTTCTTATTACAGAAGGTAGTGTTAAAGGTTTACAACCATCTGTAGGTCCCAAACAAGCTAGTCTTAATGGAGTCACATTAATTGGAGAAGCCGTAAAAACACAGTTCAATTATTATAAGAAAGTTGATGGTGTTGTCCAACAATTACCTGAAACTTACTTTACTCAGAGTAGAAATTTACAAGAATTTAAACCCAGGAGTCAAATGGAAATTGATTTCTTAGAATTAGCTATGGATGAATTCATTGAACGGTATAAATTAGAAGGCTATGCCTTCGAACATATCGTTTATGGAGATTTTAGTCATAGTCAGTTAGGTGGTTTACATCTACTGATTGGACTAGCTAAACGTTTTAAGGAATCACCTTTTGAATTAGAAGATTTTATTCCTATGGACAGTACAGTTAAAAACTATTTCATAACAGATGCGCAAACAGGTTCATCTAAGTGTGTGTGTTCTGTTATTGATTTATTACTTGATGATTTTGTTGAAATAATAAAATCCCAAGATTTATCTGTAGTTTCTAAGGTTGTCAAAGTGACTATTGACTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGCCATGTAGAAACATTTTACCCAAAATTACAATCTAGTCAAGCGTGGCAACCGGGTGTTGCTATGCCTAATCTTTACAAAATGCAAAGAATGCTATTAGAAAAGTGTGACCTTCAAAATTATGGTGATAGTGCAACATTACCTAAAGGCATAATGATGAATGTCGCAAAATATACTCAACTGTGTCAATATTTAAACACATTAACATTAGCTGTACCCTATAATATGAGAGTTATACATTTTGGTGCTGGTTCTGATAAAGGAGTTGCACCAGGTACAGCTGTTTTAAGACAGTGGTTGCCTACGGGTACGCTGCTTGTCGATTCAGATCTTAATGACTTTGTCTCTGATGCAGATTCAACTTTGATTGGTGATTGTGCAACTGTACATACAGCTAATAAATGGGATCTCATTATTAGTGATATGTACGACCCTAAGACTAAAAATGTTACAAAAGAAAATGACTCTAAAGAGGGTTTTTTCACTTACATTTGTGGGTTTATACAACAAAAGCTAGCTCTTGGAGGTTCCGTGGCTATAAAGATAACAGAACATTCTTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCCTTTGTTACTAATGTGAATGCGTCATCATCTGAAGCATTTTTAATTGGATGTAATTATCTTGGCAAACCACGCGAACAAATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAATACAAATCCAATTCAGTTGTCTTCCTATTCTTTATTTGACATGAGTAAATTTCCCCTTAAATTAAGGGGTACTGCTGTTATGTCTTTAAAAGAAGGTCAAATCAATGATATGATTTTATCTCTTCTTAGTAAAGGTAGACTTATAATTAGAGAAAACAACAGAGTTGTTATTTCTAGTGATGTTCTTGTTAACAACTAAACGAACAATGTTTGTTTTTCTTGTTTTATTGCCACTAGTCTCTAGTCAGTGTGTTAATCTTACAACCAGAACTCAATTACCCCCTGCATACACTAATTCTTTCACACGTGGTGTTTATTACCCTGACAAAGTTTTCAGATCCTCAGTTTTACATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCAATGGTACTAAGAGGTTTGATAACCCTGTCCTACCATTTAATGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACTTTAGATTCGAAGACCCAGTCCCTACTTATTGTTAATAACGCTACTAATGTTGTTATTAAAGTCTGTGAATTTCAATTTTGTAATGATCCATTTTTGGGTGTTTATTACCACAAAAACAACAAAAGTTGGATGGAAAGTGAGTTCAGAGTTTATTCTAGTGCGAATAATTGCACTTTTGAATATGTCTCTCAGCCTTTTCTTATGGACCTTGAAGGAAAACAGGGTAATTTCAAAAATCTTAGGGAATTTGTGTTTAAGAATATTGATGGTTATTTTAAAATATATTCTAAGCACACGCCTATTAATTTAGTGCGTGATCTCCCTCAGGGTTTTTCGGCTTTAGAACCATTGGTAGATTTGCCAATAGGTATTAACATCACTAGGTTTCAAACTTTACTTGCTTTACATAGAAGTTATTTGACTCCTGGTGATTCTTCTTCAGGTTGGACAGCTGGTGCTGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGATGCTGTAGACTGTGCACTTGACCCTCTCTCAGAAACAAAGTGTACGTTGAAATCCTTCACTGTAGAAAAAGGAATCTATCAAACTTCTAACTTTAGAGTCCAACCAACAGAATCTATTGTTAGATTTCCTAATATTACAAACTTGTGCCCTTTTGGTGAAGTTTTTAACGCCACCAGATTTGCATCTGTTTATGCTTGGAACAGGAAGAGAATCAGCAACTGTGTTGCTGATTATTCTGTCCTATATAATTCCGCATCATTTTCCACTTTTAAGTGTTATGGAGTGTCTCCTACTAAATTAAATGATCTCTGCTTTACTAATGTCTATGCAGATTCATTTGTAATTAGAGGTGATGAAGTCAGACAAATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAAATTACCAGATGATTTTACAGGCTGCGTTATAGCTTGGAATTCTAACAATCTTGATTCTAAGGTTGGTGGTAATTATAATTACCTGTATAGATTGTTTAGGAAGTCTAATCTCAAACCTTTTGAGAGAGATATTTCAACTGAAATCTATCAGGCCGGTAGCACACCTTGTAATGGTGTTGAAGGTTTTAATTGTTACTTTCCTTTACAATCATATGGTTTCCAACCCACTAATGGTGTTGGTTACCAACCATACAGAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTGGTTAAAAACAAATGTGTCAATTTCAACTTCAATGGTTTAACAGGCACAGGTGTTCTTACTGAGTCTAACAAAAAGTTTCTGCCTTTCCAACAATTTGGCAGAGACATTGCTGACACTACTGATGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTAACCAGGTTGCTGTTCTTTATCAGGATGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGTTTAATAGGGGCTGAACATGTCAACAACTCATATGAGTGTGACATACCCATTGGTGCAGGTATATGCGCTAGTTATCAGACTCAGACTAATTCTCCTCGGCGGGCACGTAGTGTAGCTAGTCAATCCATCATTGCCTACACTATGTCACTTGGTGCAGAAAATTCAGTTGCTTACTCTAATAACTCTATTGCCATACCCACAAATTTTACTATTAGTGTTACCACAGAAATTCTACCAGTGTCTATGACCAAGACATCAGTAGATTGTACAATGTACATTTGTGGTGATTCAACTGAATGCAGCAATCTTTTGTTGCAATATGGCAGTTTTTGTACACAATTAAACCGTGCTTTAACTGGAATAGCTGTTGAACAAGACAAAAACACCCAAGAAGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCTACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGTGATTGCCTTGGTGATATTGCTGCTAGAGACCTCATTTGTGCACAAAAGTTTAACGGCCTTACTGTTTTGCCACCTTTGCTCACAGATGAAATGATTGCTCAATACACTTCTGCACTGTTAGCGGGTACAATCACTTCTGGTTGGACCTTTGGTGCAGGTGCTGCATTACAAATACCATTTGCTATGCAAATGGCTTATAGGTTTAATGGTATTGGAGTTACACAGAATGTTCTCTATGAGAACCAAAAATTGATTGCCAACCAATTTAATAGTGCTATTGGCAAAATTCAAGACTCACTTTCTTCCACAGCAAGTGCACTTGGAAAACTTCAAGATGTGGTCAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGCTCCAATTTTGGTGCAATTTCAAGTGTTTTAAATGATATCCTTTCACGTCTTGACAAAGTTGAGGCTGAAGTGCAAATTGATAGGTTGATCACAGGCAGACTTCAAAGTTTGCAGACATATGTGACTCAACAATTAATTAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCTACTAAAATGTCAGAGTGTGTACTTGGACAATCAAAAAGAGTTGATTTTTGTGGAAAGGGCTATCATCTTATGTCCTTCCCTCAGTCAGCACCTCATGGTGTAGTCTTCTTGCATGTGACTTATGTCCCTGCACAAGAAAAGAACTTCACAACTGCTCCTGCCATTTGTCATGATGGAAAAGCACACTTTCCTCGTGAAGGTGTCTTTGTTTCAAATGGCACACACTGGTTTGTAACACAAAGGAATTTTTATGAACCACAAATCATTACTACAGACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAATTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGATTTAGGTGACATCTCTGGCATTAATGCTTCAGTTGTAAACATTCAAAAAGAAATTGACCGCCTCAATGAGGTTGCCAAGAATTTAAATGAATCTCTCATCGATCTCCAAGAACTTGGAAAGTATGAGCAGTATATAAAATGGCCATGGTACATTTGGCTAGGTTTTATAGCTGGCTTGATTGCCATAGTAATGGTGACAATTATGCTTTGCTGTATGACCAGTTGCTGTAGTTGTCTCAAGGGCTGTTGTTCTTGTGGATCCTGCTGCAAATTTGATGAAGACGACTCTGAGCCAGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATCTTCACAATTGGAACTGTAACTTTGAAGCAAGGTGAAATCAAGGATGCTACTCCTTCAGATTTTGTTCGCGCTACTGCAACGATACCGATACAAGCCTCACTCCCTTTCGGATGGCTTATTGTTGGCGTTGCACTTCTTGCTGTTTTTCAGAGCGCTTCCAAAATCATAACCCTCAAAAAGAGATGGCAACTAGCACTCTCCAAGGGTGTTCACTTTGTTTGCAACTTGCTGTTGTTGTTTGTAACAGTTTACTCACACCTTTTGCTCGTTGCTGCTGGCCTTGAAGCCCCTTTTCTCTATCTTTATGCTTTAGTCTACTTCTTGCAGAGTATAAACTTTGTAAGAATAATAATGAGGCTTTGGCTTTGCTGGAAATGCCGTTCCAAAAACCCATTACTTTATGATGCCAACTATTTTCTTTGCTGGCATACTAATTGTTACGACTATTGTATACCTTACAATAGTGTAACTTCTTCAATTGTCATTACTTCAGGTGATGGCACAACAAGTCCTATTTCTGAACATGACTACCAGATTGGTGGTTATACTGAAAAATGGGAATCTGGAGTAAAAGACTGTGTTGTATTACACAGTTACTTCACTTCAGACTATTACCAGCTGTACTCAACTCAATTGAGTACAGACACTGGTGTTGAACATGTTACCTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTCCAAATTCACACAATCGACGGTTCATCCGGAGTTGTTAATCCAGTAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTCGTGGTATTCTTGCTAGTTACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTCTGGTCTAAACGAACTAAATATTATATTAGTTTTTCTGTTTGGAACTTTAATTTTAGCCATGGCAGATTCCAACGGTACTATTACCGTTGAAGAGCTTAAAAAGCTCCTTGAACAATGGAACCTAGTAATAGGTTTCCTATTCCTTACATGGATTTGTCTTCTACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTATATAATTAAGTTAATTTTCCTCTGGCTGTTATGGCCAGTAACTTTAGCTTGTTTTGTGCTTGCTGCTGTTTACAGAATAAATTGGATCACCGGTGGAATTGCTATCGCAATGGCTTGTCTTGTAGGCTTGATGTGGCTCAGCTACTTCATTGCTTCTTTCAGACTGTTTGCGCGTACGCGTTCCATGTGGTCATTCAATCCAGAAACTAACATTCTTCTCAACGTGCCACTCCATGGCACTATTCTGACCAGACCGCTTCTAGAAAGTGAACTCGTAATCGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCTGGACACCATCTAGGACGCTGTGACATCAAGGACCTGCCTAAAGAAATCACTGTTGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGCAACTATAAATTAAACACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTACTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTACATCATAAACCTCATAATTAAAAATTTATCTAAGTCACTAACTGAGAATAAATATTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACGAACATGAAAATTATTCTTTTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAGTGTGTTAGAGGTACAACAGTACTTTTAAAAGAACCTTGCTCTTCTGGAACATACGAGGGCAATTCACCATTTCATCCTCTAGCTGATAACAAATTTGCACTGACTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGCGTAAAACACGTCTATCAGTTACGTGCCAGATCAGTTTCACCTAAACTGTTCATCAGACAAGAGGAAGTTCAAGAACTTTACTCTCCAATTTTTCTTATTGTTGCGGCAATAGTGTTTATAACACTTTGCTTCACACTCAAAAGAAAGACAGAATGATTGAACTTTCATTAATTGACTTCTATTTGTGCTTTTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTTATTATCTTTTGGTTCTCACTTGAACTGCAAGATCATAATGAAACTTGTCACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAATCATCACAACTGTAGCTGCATTTCACCAAGAATGTAGTTTACAGTCATGTACTCAACATCAACCATATGTAGTTGATGACCCGTGTCCTATTCACTTCTATTCTAAATGGTATATTAGAGTAGGAGCTAGAAAATCAGCACCTTTAATTGAATTGTGCGTGGATGAGGCTGGTTCTAAATCACCCATTCAGTACATCGATATCGGTAATTATACAGTTTCCTGTTTACCTTTTACAATTAATTGCCAGGAACCTAAATTGGGTAGTCTTGTAGTGCGTTGTTCGTTCTATGAAGACTTTTTAGAGTATCATGACGTTCGTGTTGTTTTAGATTTCATCTAAACGAACAAACTAAAATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAAACTCATGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATCTCACATAGCAATCTTTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACATTTTCACCGAGGCCACGCGGAGTACGATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAATGACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n"
     ]
    }
   ],
   "source": [
    "print(patient_rec.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(29903), strand=1), type='source'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(265), strand=1), type=\"5'UTR\"),\n",
       " SeqFeature(FeatureLocation(ExactPosition(265), ExactPosition(21555), strand=1), type='gene'),\n",
       " SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(265), ExactPosition(13468), strand=1), FeatureLocation(ExactPosition(13467), ExactPosition(21555), strand=1)], 'join'), type='CDS', location_operator='join'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(21562), ExactPosition(25384), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(21562), ExactPosition(25384), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(25392), ExactPosition(26220), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(25392), ExactPosition(26220), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26244), ExactPosition(26472), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26244), ExactPosition(26472), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26522), ExactPosition(27191), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(26522), ExactPosition(27191), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27201), ExactPosition(27387), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27201), ExactPosition(27387), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27393), ExactPosition(27759), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27393), ExactPosition(27759), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27893), ExactPosition(28259), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27893), ExactPosition(28259), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(28273), ExactPosition(29533), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(28273), ExactPosition(29533), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29557), ExactPosition(29674), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29557), ExactPosition(29674), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(29674), ExactPosition(29903), strand=1), type=\"3'UTR\")]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_rec.features\n",
    "# CDS=coding sequence"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sgsgsgsg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIWWW"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function qblast in module Bio.Blast.NCBIWWW:\n",
      "\n",
      "qblast(program, database, sequence, url_base='https://blast.ncbi.nlm.nih.gov/Blast.cgi', auto_format=None, composition_based_statistics=None, db_genetic_code=None, endpoints=None, entrez_query='(none)', expect=10.0, filter=None, gapcosts=None, genetic_code=None, hitlist_size=50, i_thresh=None, layout=None, lcase_mask=None, matrix_name=None, nucl_penalty=None, nucl_reward=None, other_advanced=None, perc_ident=None, phi_pattern=None, query_file=None, query_believe_defline=None, query_from=None, query_to=None, searchsp_eff=None, service=None, threshold=None, ungapped_alignment=None, word_size=None, short_query=None, alignments=500, alignment_view=None, descriptions=500, entrez_links_new_window=None, expect_low=None, expect_high=None, format_entrez_query=None, format_object=None, format_type='XML', ncbi_gi=None, results_file=None, show_overview=None, megablast=None, template_type=None, template_length=None)\n",
      "    BLAST search using NCBI's QBLAST server or a cloud service provider.\n",
      "    \n",
      "    Supports all parameters of the old qblast API for Put and Get.\n",
      "    \n",
      "    Please note that NCBI uses the new Common URL API for BLAST searches\n",
      "    on the internet (http://ncbi.github.io/blast-cloud/dev/api.html). Thus,\n",
      "    some of the parameters used by this function are not (or are no longer)\n",
      "    officially supported by NCBI. Although they are still functioning, this\n",
      "    may change in the future.\n",
      "    \n",
      "    The Common URL API (http://ncbi.github.io/blast-cloud/dev/api.html) allows\n",
      "    doing BLAST searches on cloud servers. To use this feature, please set\n",
      "    ``url_base='http://host.my.cloud.service.provider.com/cgi-bin/blast.cgi'``\n",
      "    and ``format_object='Alignment'``. For more details, please see\n",
      "    https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=CloudBlast\n",
      "    \n",
      "    Some useful parameters:\n",
      "    \n",
      "     - program        blastn, blastp, blastx, tblastn, or tblastx (lower case)\n",
      "     - database       Which database to search against (e.g. \"nr\").\n",
      "     - sequence       The sequence to search.\n",
      "     - ncbi_gi        TRUE/FALSE whether to give 'gi' identifier.\n",
      "     - descriptions   Number of descriptions to show.  Def 500.\n",
      "     - alignments     Number of alignments to show.  Def 500.\n",
      "     - expect         An expect value cutoff.  Def 10.0.\n",
      "     - matrix_name    Specify an alt. matrix (PAM30, PAM70, BLOSUM80, BLOSUM45).\n",
      "     - filter         \"none\" turns off filtering.  Default no filtering\n",
      "     - format_type    \"HTML\", \"Text\", \"ASN.1\", or \"XML\".  Def. \"XML\".\n",
      "     - entrez_query   Entrez query to limit Blast search\n",
      "     - hitlist_size   Number of hits to return. Default 50\n",
      "     - megablast      TRUE/FALSE whether to use MEga BLAST algorithm (blastn only)\n",
      "     - short_query    TRUE/FALSE whether to adjust the search parameters for a\n",
      "                      short query sequence. Note that this will override\n",
      "                      manually set parameters like word size and e value. Turns\n",
      "                      off when sequence length is > 30 residues. Default: None.\n",
      "     - service        plain, psi, phi, rpsblast, megablast (lower case)\n",
      "    \n",
      "    This function does no checking of the validity of the parameters\n",
      "    and passes the values to the server as is.  More help is available at:\n",
      "    https://ncbi.github.io/blast-cloud/dev/api.html\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(NCBIWWW.qblast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 92 ms, sys: 44 ms, total: 136 ms\n",
      "Wall time: 10min 14s\n"
     ]
    }
   ],
   "source": [
    "%time result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", record_dict['MN908947.3'].seq)\n",
    "with open(\"Files/my_blast.xml\", \"w\") as out_handle:\n",
    "    out_handle.write(result_handle.read())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIXML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = open(\"Files/my_blast.xml\")\n",
    "blast_records = list(NCBIXML.parse(result_handle))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![UML diagram of the BlastRecord](Media/BlastRecord.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1798174254|ref|NC_045512.2| Wuhan seafood market pneumonia virus isolate Wuhan-Hu-1, complete genome >gi|1798172431|gb|MN908947.3| Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29903\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293633|gb|MT019531.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-03/2019, complete genome\n",
      "length    : 29899\n",
      "e value   : 0.0\n",
      "identities: 29898\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293611|gb|MT019529.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-01/2019, complete genome\n",
      "length    : 29899\n",
      "e value   : 0.0\n",
      "identities: 29896\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633808|gb|MN996528.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV04, complete genome\n",
      "length    : 29891\n",
      "e value   : 0.0\n",
      "identities: 29891\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808633715|gb|MT049951.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/Yunnan-01/human/2020/CHN, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29898\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293644|gb|MT019532.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-04/2019, complete genome\n",
      "length    : 29890\n",
      "e value   : 0.0\n",
      "identities: 29890\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800455117|gb|MN988668.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU01, complete genome >gi|1800455123|gb|MN988669.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV WHU02, complete genome\n",
      "length    : 29881\n",
      "e value   : 0.0\n",
      "identities: 29881\n",
      "TTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860439|gb|MT039890.1| Severe acute respiratory syndrome coronavirus 2 isolate SNU01, complete genome\n",
      "length    : 29903\n",
      "e value   : 0.0\n",
      "identities: 29894\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293655|gb|MT019533.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-05/2020, complete genome\n",
      "length    : 29883\n",
      "e value   : 0.0\n",
      "identities: 29882\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800242661|gb|MN975262.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-005b_2020, complete genome\n",
      "length    : 29891\n",
      "e value   : 0.0\n",
      "identities: 29886\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802471970|gb|MN994468.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA2/2020, complete genome\n",
      "length    : 29883\n",
      "e value   : 0.0\n",
      "identities: 29881\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1806553209|gb|MT027064.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA5/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29880\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1806553187|gb|MT027062.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA3/2020, complete genome >gi|1806553198|gb|MT027063.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA4/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800408777|gb|MN985325.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1/2020, complete genome >gi|1805599854|gb|MT020880.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-A12/2020, complete genome >gi|1805599865|gb|MT020881.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WA1-F6/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805293622|gb|MT019530.1| Severe acute respiratory syndrome coronavirus 2 isolate BetaCoV/Wuhan/IPBCAMS-WH-02/2019, complete genome\n",
      "length    : 29889\n",
      "e value   : 0.0\n",
      "identities: 29883\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1803016604|gb|MT007544.1| Severe acute respiratory syndrome coronavirus 2 isolate Australia/VIC01/2020, complete genome\n",
      "length    : 29893\n",
      "e value   : 0.0\n",
      "identities: 29890\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802498786|gb|MN997409.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-AZ1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29878\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860452|gb|MT039887.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-WI1/2020, complete genome\n",
      "length    : 29879\n",
      "e value   : 0.0\n",
      "identities: 29878\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860463|gb|MT039888.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-MA1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29877\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800489756|gb|MN988713.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294619|gb|MT093571.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/01/human/2020/SWE, complete genome\n",
      "length    : 29886\n",
      "e value   : 0.0\n",
      "identities: 29879\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816337|dbj|LC522974.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-501/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816315|dbj|LC522972.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/KY/V-029/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29874\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153870|emb|LR757996.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29868\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "CAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153869|emb|LR757995.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29872\n",
      "e value   : 0.0\n",
      "identities: 29869\n",
      "TTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816348|dbj|LC522975.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-521/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29873\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807816326|dbj|LC522973.1| Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-012/2020 RNA, complete genome\n",
      "length    : 29878\n",
      "e value   : 0.0\n",
      "identities: 29873\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1809484504|gb|MT066176.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU02/2020/TWN, complete genome\n",
      "length    : 29870\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1809484493|gb|MT066175.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/NTU01/2020/TWN, complete genome\n",
      "length    : 29870\n",
      "e value   : 0.0\n",
      "identities: 29868\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808500211|gb|MT044257.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-IL2/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29875\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802471959|gb|MN994467.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA1/2020, complete genome\n",
      "length    : 29882\n",
      "e value   : 0.0\n",
      "identities: 29875\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294674|gb|MT093631.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/WH-09/human/2020/CHN, complete genome\n",
      "length    : 29911\n",
      "e value   : 0.0\n",
      "identities: 29863\n",
      "TTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1811294674|gb|MT093631.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/WH-09/human/2020/CHN, complete genome\n",
      "length    : 29911\n",
      "e value   : 6.60229e-13\n",
      "identities: 51\n",
      "AGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTAT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTAT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1804153872|emb|LR757998.1| Wuhan seafood market pneumonia virus genome assembly, chromosome: whole_genome\n",
      "length    : 29866\n",
      "e value   : 0.0\n",
      "identities: 29864\n",
      "AACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633830|gb|MN996530.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV06, complete genome\n",
      "length    : 29854\n",
      "e value   : 0.0\n",
      "identities: 29854\n",
      "CCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633841|gb|MN996531.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV07, complete genome\n",
      "length    : 29857\n",
      "e value   : 0.0\n",
      "identities: 29855\n",
      "ACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633819|gb|MN996529.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV05, complete genome\n",
      "length    : 29852\n",
      "e value   : 0.0\n",
      "identities: 29850\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1807860410|gb|MT039873.1| Severe acute respiratory syndrome coronavirus 2 isolate HZ-1, complete genome\n",
      "length    : 29833\n",
      "e value   : 0.0\n",
      "identities: 29833\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1800242639|gb|MN938384.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV_HKU-SZ-002a_2020, complete genome\n",
      "length    : 29838\n",
      "e value   : 0.0\n",
      "identities: 29835\n",
      "CAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1808500222|gb|MT044258.1| Severe acute respiratory syndrome coronavirus 2 isolate 2019-nCoV/USA-CA6/2020, complete genome\n",
      "length    : 29858\n",
      "e value   : 0.0\n",
      "identities: 29858\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633797|gb|MN996527.1| Severe acute respiratory syndrome coronavirus 2 isolate WIV02, complete genome\n",
      "length    : 29825\n",
      "e value   : 0.0\n",
      "identities: 29823\n",
      "AACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCAT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCAT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1810678290|gb|MT072688.1| Severe acute respiratory syndrome coronavirus 2 isolate SARS0CoV-2/61-TW/human/2020/ NPL, complete genome\n",
      "length    : 29811\n",
      "e value   : 0.0\n",
      "identities: 29810\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1802633852|gb|MN996532.1| Bat coronavirus RaTG13, complete genome\n",
      "length    : 29855\n",
      "e value   : 0.0\n",
      "identities: 28717\n",
      "CTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "||| ||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CTTTCCAGGTAACAAACCAACGAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805599841|gb|MT020781.1| Severe acute respiratory syndrome coronavirus 2 isolate nCoV-FIN-29-Jan-2020, partial genome\n",
      "length    : 29847\n",
      "e value   : 0.0\n",
      "identities: 21581\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1805599841|gb|MT020781.1| Severe acute respiratory syndrome coronavirus 2 isolate nCoV-FIN-29-Jan-2020, partial genome\n",
      "length    : 29847\n",
      "e value   : 0.0\n",
      "identities: 8103\n",
      "ATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "ATTCAACTCAGGACTTGTTCTTACCTTTCTTTTCCAATGTTACTTGGTTCCATGCTATACATGTCTCTGGGACCA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125417|gb|MG772933.1| Bat SARS-like coronavirus isolate bat-SL-CoVZC45, complete genome\n",
      "length    : 29802\n",
      "e value   : 0.0\n",
      "identities: 20145\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| |||||||||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125417|gb|MG772933.1| Bat SARS-like coronavirus isolate bat-SL-CoVZC45, complete genome\n",
      "length    : 29802\n",
      "e value   : 0.0\n",
      "identities: 6063\n",
      "TACCAACCATACA-GAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAA...\n",
      "|||||| | |||| | || || ||  | || ||||| |||||| |||||||||| || ||||||||||| |||  ...\n",
      "TACCAAGC-TACAAGGGTTGTTGTTTTGTCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACCAAAACT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541441|gb|KY417146.1| Bat SARS-like coronavirus isolate Rs4231, complete genome\n",
      "length    : 29782\n",
      "e value   : 0.0\n",
      "identities: 21517\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAACGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541441|gb|KY417146.1| Bat SARS-like coronavirus isolate Rs4231, complete genome\n",
      "length    : 29782\n",
      "e value   : 0.0\n",
      "identities: 2393\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 17067\n",
      "ACTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTG...\n",
      "|||| ||||   |||| ||| |  |||||||||||| | ||||||||||| | |||||||||||| |||||||| ...\n",
      "ACTAGTGAACAAAATAATTTCACAGGTTATTTAAAATTAACTGACAATGTTTTCATTAAAAATGCTGACATTGTA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 6048\n",
      "TCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTGGTTAAAAACAAATGT...\n",
      "|| ||||| |||||| |||||||||| || ||||||||||| |||   || ||  |  | || |||||  | || ...\n",
      "TCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACCAAAATTATCCACACAACTAGTAAAAAATCAGTGC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1369125429|gb|MG772934.1| Bat SARS-like coronavirus isolate bat-SL-CoVZXC21, complete genome\n",
      "length    : 29732\n",
      "e value   : 0.0\n",
      "identities: 2983\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| |||||||||||||||||||||||| |||| ||||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 19382\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAATGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 2114\n",
      "TAAACGAACATGAAAATTATTCTTTTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAG...\n",
      "||||||||||||||||||||||| ||| || || ||||  |||| ||  |||| ||||| |||||||| || |||...\n",
      "TAAACGAACATGAAAATTATTCTCTTCCTGACATTGATTGCACTTGCATCTTGCGAGCTATATCACTATCAGGAG...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1693074671|gb|MK211376.1| Coronavirus BtRs-BetaCoV/YN2018B, complete genome\n",
      "length    : 30256\n",
      "e value   : 0.0\n",
      "identities: 2404\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 14872\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| |||||||||||||||||||||||||||   ||||||   ||  ||||| ||| | || ||  |||| |  ...\n",
      "TTTACTGGTTATTTAAAACTTACTGACAATGTTGCCATTAAGTGTGTGGACATCGTGAAGGAGGCGCAAAACGCT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 5645\n",
      "AGAGTAGTAGTACTTTCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAAAGTCTACTAATTTG...\n",
      "|| || || || ||||| ||||| ||| || ||||||| || || |||||||||||||||   || ||     | ...\n",
      "AGGGTTGTTGTGCTTTCATTTGAGCTTTTAAATGCACCTGCTACAGTTTGTGGACCTAAACTATCCACAGGACTT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 0.0\n",
      "identities: 2389\n",
      "AGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "|||||| ||||| |||||| ||  | ||||||||| | |||||||||||||||||||||||||||||||||||||...\n",
      "AGGTTTTTACCTACCCAGGAAA--AGCCAACCAACCT-CGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 1.19056e-66\n",
      "identities: 375\n",
      "TGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGA...\n",
      "||| ||||||||||| ||| ||||  ||  ||  || ||| |  | |  | |||||||||||| || || || ||...\n",
      "TGCTGCTTATTATGTTGGTAATCTAAAATATACCACCTTTATGCTTAGTTTTAATGAAAATGGGACTATCACTGA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|1270541454|gb|KY417147.1| Bat SARS-like coronavirus isolate Rs4237, complete genome\n",
      "length    : 29741\n",
      "e value   : 2.80736e-11\n",
      "identities: 78\n",
      "TTTGATAACCCTGTCCTACCATTTAATGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGC...\n",
      "|||||||||||| |  ||   |||  |||||| |||||||| ||  | ||||| ||||||||  |||| ||||||...\n",
      "TTTGATAACCCTATTTTAGACTTTGGTGATGGCGTTTATTTCGCCGCTACTGAAAAGTCTAATGTAATCAGAGGC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 14838\n",
      "TTTAGTGGTTATTTAAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTA...\n",
      "|||| ||| ||||||||||||||||| |||||   ||||||   ||  ||||||||  | || ||  |||  |  ...\n",
      "TTTACTGGCTATTTAAAACTTACTGATAATGTTGCCATTAAGTGTGTTGACATTGTTAAGGAGGCACAAAGCGCC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 6107\n",
      "TGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGA...\n",
      "||| || |||||||| || |||||| | ||||| || ||| | ||| ||| ||||   ||||| || |||    |...\n",
      "TGCTGCATATTATGTAGGCTATCTTAAGCCTAGAACATTTATGTTACAATTTAATAGCAATGGCACAATTGTTAA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|641457801|gb|KJ473814.1| BtRs-BetaCoV/HuB2013, complete genome\n",
      "length    : 29658\n",
      "e value   : 0.0\n",
      "identities: 2304\n",
      "TCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAA---TT...\n",
      "|||||||||||||||||||||||||||||||||| ||||||||||||| ||||||||  |||||||||||   ||...\n",
      "TCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCGCTCGGCTGCATGCCTAGTGCACCTACGCAGTATAAATATT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 14075\n",
      "CTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTAC...\n",
      "||| || || | |||||| || || || |||||  |  ||   || ||||  || ||   ||| ||| |||| ||...\n",
      "CTTTAAAAAGGGTGCTCCTTACATTGTTGGTGACATCATTAGCGAAGGTGAATTGACCTGTGTTGTTTTACCAAC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 3948\n",
      "GCTGGTGCTGCAGCTTATTATGTGGGTTATCTTCAACCTAGGACTTTTCTATTAAAATATAATGAAAATGGAACC...\n",
      "||||  ||| | ||||| | ||||||  ||||  |||||   ||  | ||  |  | | | | || ||||| |||...\n",
      "GCTGACGCTTCTGCTTACTTTGTGGGCCATCTAAAACCTCTCACCATGCTTGTTGACTTTGACGAGAATGGCACC...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 1359\n",
      "TTCGTGTTGTTTTAGATTTC----ATCTAAACGAACAAACTAAAATGTCTGATAATGGACCC-CAAAATCAGCGA...\n",
      "|| || |||| ||||| ||     ||||||||||||||  ||||||| ||||||||||||   |||| || ||| ...\n",
      "TTTGTTTTGTGTTAGAATTACAAGATCTAAACGAACAA--TAAAATGACTGATAATGGACAATCAAACTC-GCGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 0.0\n",
      "identities: 1409\n",
      "TTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGT...\n",
      "|||||||||||||| ||||||||| ||||||||||  |||||||| ||||||||||  | ||||    |||||||...\n",
      "TTTAAAATCTGTGTAGCTGTCACTTGGCTGCATGCCCAGTGCACTTACGCAGTATATCTTATAAACTTTTACTGT...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 1.11507e-60\n",
      "identities: 335\n",
      "CCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATG...\n",
      "|| |||||| |||| || || ||| |  | ||| | || || |  ||  | ||||||||  |  | |||  | | ...\n",
      "CCTGTTTGTGTTAATGGTCTAATGCTCTTAGAATTGAAGGAAAAGGAGCAATACTGTGCATTGTCTCCTTCTCTA...\n",
      "\n",
      "****Alignment****\n",
      "sequence  : gi|301298998|gb|GU190215.1| Bat coronavirus BM48-31/BGR/2008, complete genome\n",
      "length    : 29276\n",
      "e value   : 6.18363e-07\n",
      "identities: 117\n",
      "TGATGGTGTTTATTTTGCTTCCACTGAGAAGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACTTTAGATTC...\n",
      "||||||||||||||||| |   || ||||| |||||  |  |  |||| |||||||||||| | || |||||   ...\n",
      "TGATGGTGTTTATTTTGGTCTAACCGAGAAATCTAATGTTTTTCGAGGTTGGATTTTTGGTTCGACATTAGACAA...\n"
     ]
    }
   ],
   "source": [
    "E_VALUE_THRESH = 0.01\n",
    "print(len(blast_records[0].alignments))\n",
    "for alignment in blast_records[0].alignments:\n",
    "    for hsp in alignment.hsps:  # HSPs = high-scoring pairs\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            print(\"\\n****Alignment****\")\n",
    "            print(\"sequence  :\", alignment.title)\n",
    "            print(\"length    :\", alignment.length)\n",
    "            print(\"e value   :\", hsp.expect)\n",
    "            print(\"identities:\", hsp.identities)\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.match[0:75] + \"...\")\n",
    "            print(hsp.sbjct[0:75] + \"...\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"roger.guimera@urv.cat\"  # Always tell NCBI who you are\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"MG772933.1\", rettype=\"gb\", retmode=\"text\")\n",
    "record = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('ATATTAGGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGT...AAA', IUPACAmbiguousDNA())"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq1 = record.seq\n",
    "seq2 = patient_rec.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Align"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligner = Align.PairwiseAligner()\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: 0.000000\n",
      "  target_internal_open_gap_score: 0.000000\n",
      "  target_internal_extend_gap_score: 0.000000\n",
      "  target_left_open_gap_score: 0.000000\n",
      "  target_left_extend_gap_score: 0.000000\n",
      "  target_right_open_gap_score: 0.000000\n",
      "  target_right_extend_gap_score: 0.000000\n",
      "  query_internal_open_gap_score: 0.000000\n",
      "  query_internal_extend_gap_score: 0.000000\n",
      "  query_left_open_gap_score: 0.000000\n",
      "  query_left_extend_gap_score: 0.000000\n",
      "  query_right_open_gap_score: 0.000000\n",
      "  query_right_extend_gap_score: 0.000000\n",
      "  mode: global\n",
      "\n",
      "Needleman-Wunsch\n"
     ]
    }
   ],
   "source": [
    "print(aligner)\n",
    "print(aligner.algorithm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rguimera/.local/lib/python3.5/site-packages/Bio/Align/substitution_matrices/__init__.py:21: BiopythonExperimentalWarning: Bio.Align.substitution_matrices is an experimental module which may still undergo significant changes. In particular, the location of this module may change, and the Array class defined in this module may be moved to other existing or new modules in Biopython.\n",
      "  BiopythonExperimentalWarning)\n"
     ]
    }
   ],
   "source": [
    "from Bio.Align import substitution_matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BENNER22 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3B.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.5 -1.2 -0.2 -0.3 -3.1  0.8 -1.6 -0.4 -1.0 -1.7 -0.8  0.0  0.8 -0.9 -1.2  1.3  1.4  0.4 -5.5 -3.5\n",
      "C -1.2 12.6 -3.7 -4.3 -0.1 -1.7 -1.5 -2.4 -3.3 -2.6 -2.5 -1.9 -3.1 -3.3 -1.6  0.3 -1.1 -1.7  0.5  0.6\n",
      "D -0.2 -3.7  4.8  3.9 -5.4  0.7  0.3 -4.0  0.2 -4.9 -3.9  2.4 -1.8  0.6 -1.0  0.1 -0.7 -3.0 -6.4 -3.0\n",
      "E -0.3 -4.3  3.9  4.6 -5.7  0.5 -0.2 -3.6  1.0 -4.4 -3.4  1.2 -1.7  1.7 -0.1 -0.5 -0.9 -2.7 -6.3 -4.0\n",
      "F -3.1 -0.1 -5.4 -5.7  7.7 -5.8  0.3  0.5 -5.1  2.2  0.7 -3.5 -3.4 -3.6 -4.3 -2.2 -2.6 -0.1  0.5  5.9\n",
      "G  0.8 -1.7  0.7  0.5 -5.8  6.2 -2.0 -3.8 -1.0 -4.9 -3.8  0.4 -1.8 -1.4 -0.7  0.6 -0.7 -2.5 -4.5 -4.8\n",
      "H -1.6 -1.5  0.3 -0.2  0.3 -2.0  6.1 -3.2  0.8 -2.1 -2.4  1.4 -0.4  2.4  1.5 -0.5 -1.1 -3.0 -2.7  3.7\n",
      "I -0.4 -2.4 -4.0 -3.6  0.5 -3.8 -3.2  4.2 -3.0  2.7  3.1 -2.7 -2.3 -2.7 -3.2 -1.4  0.3  3.6 -4.4 -2.2\n",
      "K -1.0 -3.3  0.2  1.0 -5.1 -1.0  0.8 -3.0  4.4 -3.3 -2.0  1.0 -1.6  2.2  3.9 -0.4 -0.4 -2.7 -3.7 -3.6\n",
      "L -1.7 -2.6 -4.9 -4.4  2.2 -4.9 -2.1  2.7 -3.3  4.6  3.2 -3.5 -1.3 -2.0 -2.9 -2.1 -1.0  2.0 -1.8 -0.7\n",
      "M -0.8 -2.5 -3.9 -3.4  0.7 -3.8 -2.4  3.1 -2.0  3.2  4.9 -2.6 -2.0 -1.7 -2.1 -1.5  0.1  2.5 -2.8 -1.8\n",
      "N  0.0 -1.9  2.4  1.2 -3.5  0.4  1.4 -2.7  1.0 -3.5 -2.6  3.3 -1.1  0.5  0.4  1.1  0.5 -2.3 -5.2 -1.2\n",
      "P  0.8 -3.1 -1.8 -1.7 -3.4 -1.8 -0.4 -2.3 -1.6 -1.3 -2.0 -1.1  7.0 -0.1 -1.2  1.1  0.4 -1.7 -5.8 -3.5\n",
      "Q -0.9 -3.3  0.6  1.7 -3.6 -1.4  2.4 -2.7  2.2 -2.0 -1.7  0.5 -0.1  4.2  2.2 -0.6 -0.7 -2.4 -3.3 -1.9\n",
      "R -1.2 -1.6 -1.0 -0.1 -4.3 -0.7  1.5 -3.2  3.9 -2.9 -2.1  0.4 -1.2  2.2  5.0 -0.5 -0.7 -2.9 -1.1 -2.7\n",
      "S  1.3  0.3  0.1 -0.5 -2.2  0.6 -0.5 -1.4 -0.4 -2.1 -1.5  1.1  1.1 -0.6 -0.5  2.0  1.5 -0.9 -3.9 -1.9\n",
      "T  1.4 -1.1 -0.7 -0.9 -2.6 -0.7 -1.1  0.3 -0.4 -1.0  0.1  0.5  0.4 -0.7 -0.7  1.5  2.5  0.4 -4.5 -3.0\n",
      "V  0.4 -1.7 -3.0 -2.7 -0.1 -2.5 -3.0  3.6 -2.7  2.0  2.5 -2.3 -1.7 -2.4 -2.9 -0.9  0.4  3.7 -4.5 -2.6\n",
      "W -5.5  0.5 -6.4 -6.3  0.5 -4.5 -2.7 -4.4 -3.7 -1.8 -2.8 -5.2 -5.8 -3.3 -1.1 -3.9 -4.5 -4.5 15.7  1.5\n",
      "Y -3.5  0.6 -3.0 -4.0  5.9 -4.8  3.7 -2.2 -3.6 -0.7 -1.8 -1.2 -3.5 -1.9 -2.7 -1.9 -3.0 -2.6  1.5  9.0\n",
      "\n",
      "BENNER6 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3A.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.5 -1.7 -0.6 -0.7 -3.2  0.8 -2.1  0.1 -1.9 -1.3 -0.2  0.0  1.1 -1.7 -1.7  1.4  1.7  0.7 -4.3 -4.0\n",
      "C -1.7 12.1 -3.7 -4.7 -0.1 -1.3 -1.2 -3.6 -2.8 -3.8 -3.7 -1.6 -2.7 -3.2 -0.4  0.9 -1.5 -3.1  1.6  2.6\n",
      "D -0.6 -3.7  5.2  4.4 -5.7  0.8  0.1 -4.2 -0.2 -5.3 -4.3  2.5 -2.8  0.6 -1.5 -0.4 -1.2 -3.3 -6.3 -2.3\n",
      "E -0.7 -4.7  4.4  5.2 -6.7  0.5 -0.2 -4.1  0.9 -5.0 -4.1  1.1 -2.6  2.1 -0.4 -1.2 -1.6 -3.0 -5.6 -4.1\n",
      "F -3.2 -0.1 -5.7 -6.7  8.3 -5.7  0.1  0.0 -6.3  2.4 -0.1 -3.5 -3.2 -4.4 -4.9 -1.8 -2.4 -0.5 -1.6  5.6\n",
      "G  0.8 -1.3  0.8  0.5 -5.7  5.8 -2.1 -3.4 -1.4 -4.6 -3.7 -0.1 -1.7 -1.6 -0.1  0.8 -0.5 -2.3 -1.7 -4.9\n",
      "H -2.1 -1.2  0.1 -0.2  0.1 -2.1  6.1 -3.7  0.9 -2.2 -3.4  1.4 -0.4  3.2  1.8 -0.9 -1.7 -3.8 -2.8  4.4\n",
      "I  0.1 -3.6 -4.2 -4.1  0.0 -3.4 -3.7  4.4 -3.8  2.4  4.0 -2.5 -2.0 -3.8 -3.8 -1.2  0.7  3.9 -5.0 -3.3\n",
      "K -1.9 -2.8 -0.2  0.9 -6.3 -1.4  0.9 -3.8  5.6 -4.1 -2.9  1.0 -2.3  2.5  4.3 -1.2 -1.1 -3.8 -1.4 -4.0\n",
      "L -1.3 -3.8 -5.3 -5.0  2.4 -4.6 -2.2  2.4 -4.1  4.8 -2.9 -3.4 -0.2 -2.4 -3.2 -1.5 -0.4  1.9 -3.0 -1.6\n",
      "M -0.2 -3.7 -4.3 -4.1 -0.1 -3.7 -3.4  4.0 -2.9 -2.9  4.8 -2.5 -1.8 -3.1 -3.0 -1.3  0.6  3.3 -4.4 -3.6\n",
      "N  0.0 -1.6  2.5  1.1 -3.5 -0.1  1.4 -2.5  1.0 -3.4 -2.5  3.6 -1.1  0.1 -0.1  1.2  0.5 -2.4 -4.4 -0.9\n",
      "P  1.1 -2.7 -2.8 -2.6 -3.2 -1.7 -0.4 -2.0 -2.3 -0.2 -1.8 -1.1  6.5  0.1 -1.3  1.4  0.6 -1.6 -4.8 -3.8\n",
      "Q -1.7 -3.2  0.6  2.1 -4.4 -1.6  3.2 -3.8  2.5 -2.4 -3.1  0.1  0.1  5.3  2.5 -1.4 -1.7 -3.5 -2.6 -1.4\n",
      "R -1.7 -0.4 -1.5 -0.4 -4.9 -0.1  1.8 -3.8  4.3 -3.2 -3.0 -0.1 -1.3  2.5  5.1 -0.9 -1.3 -3.7  2.0 -2.6\n",
      "S  1.4  0.9 -0.4 -1.2 -1.8  0.8 -0.9 -1.2 -1.2 -1.5 -1.3  1.2  1.4 -1.4 -0.9  2.1  1.5 -0.9 -2.9 -1.8\n",
      "T  1.7 -1.5 -1.2 -1.6 -2.4 -0.5 -1.7  0.7 -1.1 -0.4  0.6  0.5  0.6 -1.7 -1.3  1.5  2.4  0.6 -2.6 -3.4\n",
      "V  0.7 -3.1 -3.3 -3.0 -0.5 -2.3 -3.8  3.9 -3.8  1.9  3.3 -2.4 -1.6 -3.5 -3.7 -0.9  0.6  4.0 -4.8 -3.8\n",
      "W -4.3  1.6 -6.3 -5.6 -1.6 -1.7 -2.8 -5.0 -1.4 -3.0 -4.4 -4.4 -4.8 -2.6  2.0 -2.9 -2.6 -4.8 14.7 -0.3\n",
      "Y -4.0  2.6 -2.3 -4.1  5.6 -4.9  4.4 -3.3 -4.0 -1.6 -3.6 -0.9 -3.8 -1.4 -2.6 -1.8 -3.4 -3.8 -0.3  9.5\n",
      "\n",
      "BENNER74 #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences\".\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  Figure 3C.\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.4  0.3 -0.3 -0.1 -2.6  0.6 -1.0 -0.8 -0.4 -1.4 -0.8 -0.2  0.4 -0.3 -0.8  1.1  0.7  0.1 -4.1 -2.6\n",
      "C  0.3 11.8 -3.2 -3.2 -0.7 -2.0 -1.3 -1.2 -2.9 -1.6 -1.2 -1.8 -3.1 -2.6 -2.2  0.1 -0.6 -0.2 -0.9 -0.4\n",
      "D -0.3 -3.2  4.8  2.9 -4.7  0.2  0.4 -3.9  0.4 -4.2 -3.2  2.2 -1.0  0.8 -0.5  0.4 -0.2 -2.9 -5.5 -2.8\n",
      "E -0.1 -3.2  2.9  3.7 -4.3 -0.5  0.2 -2.9  1.2 -3.1 -2.2  1.0 -0.7  1.7  0.3  0.1 -0.2 -2.1 -4.7 -3.0\n",
      "F -2.6 -0.7 -4.7 -4.3  7.2 -5.4  0.0  0.9 -3.6  2.1  1.3 -3.2 -3.8 -2.8 -3.5 -2.6 -2.2  0.1  3.0  5.3\n",
      "G  0.6 -2.0  0.2 -0.5 -5.4  6.6 -1.6 -4.3 -1.1 -4.6 -3.5  0.4 -1.7 -1.1 -1.0  0.4 -1.0 -3.1 -4.1 -4.3\n",
      "H -1.0 -1.3  0.4  0.2  0.0 -1.6  6.1 -2.3  0.6 -1.9 -1.5  1.2 -1.0  1.4  1.0 -0.3 -0.5 -2.1 -1.0  2.5\n",
      "I -0.8 -1.2 -3.9 -2.9  0.9 -4.3 -2.3  4.0 -2.3  2.8  2.6 -2.8 -2.6 -2.0 -2.6 -1.8 -0.3  3.2 -2.3 -1.0\n",
      "K -0.4 -2.9  0.4  1.2 -3.6 -1.1  0.6 -2.3  3.4 -2.4 -1.5  0.9 -0.8  1.7  2.9  0.0  0.1 -1.9 -3.6 -2.4\n",
      "L -1.4 -1.6 -4.2 -3.1  2.1 -4.6 -1.9  2.8 -2.4  4.2  2.9 -3.1 -2.2 -1.7 -2.4 -2.2 -1.1  1.9 -0.9 -0.1\n",
      "M -0.8 -1.2 -3.2 -2.2  1.3 -3.5 -1.5  2.6 -1.5  2.9  4.5 -2.2 -2.4 -1.0 -1.8 -1.4 -0.4  1.8 -1.3 -0.5\n",
      "N -0.2 -1.8  2.2  1.0 -3.2  0.4  1.2 -2.8  0.9 -3.1 -2.2  3.6 -1.0  0.7  0.3  0.9  0.4 -2.2 -4.0 -1.4\n",
      "P  0.4 -3.1 -1.0 -0.7 -3.8 -1.7 -1.0 -2.6 -0.8 -2.2 -2.4 -1.0  7.5 -0.2 -0.1  0.5  0.1 -1.9 -5.2 -3.4\n",
      "Q -0.3 -2.6  0.8  1.7 -2.8 -1.1  1.4 -2.0  1.7 -1.7 -1.0  0.7 -0.2  3.0  1.6  0.1 -0.1 -1.7 -2.8 -1.8\n",
      "R -0.8 -2.2 -0.5  0.3 -3.5 -1.0  1.0 -2.6  2.9 -2.4 -1.8  0.3 -0.1  1.6  4.8 -0.2 -0.3 -2.2 -1.6 -2.0\n",
      "S  1.1  0.1  0.4  0.1 -2.6  0.4 -0.3 -1.8  0.0 -2.2 -1.4  0.9  0.5  0.1 -0.2  2.1  1.4 -1.0 -3.4 -1.9\n",
      "T  0.7 -0.6 -0.2 -0.2 -2.2 -1.0 -0.5 -0.3  0.1 -1.1 -0.4  0.4  0.1 -0.1 -0.3  1.4  2.5  0.2 -3.7 -2.1\n",
      "V  0.1 -0.2 -2.9 -2.1  0.1 -3.1 -2.1  3.2 -1.9  1.9  1.8 -2.2 -1.9 -1.7 -2.2 -1.0  0.2  3.4 -2.9 -1.4\n",
      "W -4.1 -0.9 -5.5 -4.7  3.0 -4.1 -1.0 -2.3 -3.6 -0.9 -1.3 -4.0 -5.2 -2.8 -1.6 -3.4 -3.7 -2.9 14.7  3.6\n",
      "Y -2.6 -0.4 -2.8 -3.0  5.3 -4.3  2.5 -1.0 -2.4 -0.1 -0.5 -1.4 -3.4 -1.8 -2.0 -1.9 -2.1 -1.4  3.6  8.1\n",
      "\n",
      "BLOSUM45 #  Matrix made by matblas from blosum45.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 45\n",
      "#  Entropy =   0.3795, Expected =  -0.2789\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -1.0 -2.0 -1.0 -1.0 -1.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  0.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -5.0\n",
      "R -2.0  7.0  0.0 -1.0 -3.0  1.0  0.0 -2.0  0.0 -3.0 -2.0  3.0 -1.0 -2.0 -2.0 -1.0 -1.0 -2.0 -1.0 -2.0 -1.0  0.0 -1.0 -5.0\n",
      "N -1.0  0.0  6.0  2.0 -2.0  0.0  0.0  0.0  1.0 -2.0 -3.0  0.0 -2.0 -2.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  4.0  0.0 -1.0 -5.0\n",
      "D -2.0 -1.0  2.0  7.0 -3.0  0.0  2.0 -1.0  0.0 -4.0 -3.0  0.0 -3.0 -4.0 -1.0  0.0 -1.0 -4.0 -2.0 -3.0  5.0  1.0 -1.0 -5.0\n",
      "C -1.0 -3.0 -2.0 -3.0 12.0 -3.0 -3.0 -3.0 -3.0 -3.0 -2.0 -3.0 -2.0 -2.0 -4.0 -1.0 -1.0 -5.0 -3.0 -1.0 -2.0 -3.0 -2.0 -5.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  6.0  2.0 -2.0  1.0 -2.0 -2.0  1.0  0.0 -4.0 -1.0  0.0 -1.0 -2.0 -1.0 -3.0  0.0  4.0 -1.0 -5.0\n",
      "E -1.0  0.0  0.0  2.0 -3.0  2.0  6.0 -2.0  0.0 -3.0 -2.0  1.0 -2.0 -3.0  0.0  0.0 -1.0 -3.0 -2.0 -3.0  1.0  4.0 -1.0 -5.0\n",
      "G  0.0 -2.0  0.0 -1.0 -3.0 -2.0 -2.0  7.0 -2.0 -4.0 -3.0 -2.0 -2.0 -3.0 -2.0  0.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -1.0 -5.0\n",
      "H -2.0  0.0  1.0  0.0 -3.0  1.0  0.0 -2.0 10.0 -3.0 -2.0 -1.0  0.0 -2.0 -2.0 -1.0 -2.0 -3.0  2.0 -3.0  0.0  0.0 -1.0 -5.0\n",
      "I -1.0 -3.0 -2.0 -4.0 -3.0 -2.0 -3.0 -4.0 -3.0  5.0  2.0 -3.0  2.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0  3.0 -3.0 -3.0 -1.0 -5.0\n",
      "L -1.0 -2.0 -3.0 -3.0 -2.0 -2.0 -2.0 -3.0 -2.0  2.0  5.0 -3.0  2.0  1.0 -3.0 -3.0 -1.0 -2.0  0.0  1.0 -3.0 -2.0 -1.0 -5.0\n",
      "K -1.0  3.0  0.0  0.0 -3.0  1.0  1.0 -2.0 -1.0 -3.0 -3.0  5.0 -1.0 -3.0 -1.0 -1.0 -1.0 -2.0 -1.0 -2.0  0.0  1.0 -1.0 -5.0\n",
      "M -1.0 -1.0 -2.0 -3.0 -2.0  0.0 -2.0 -2.0  0.0  2.0  2.0 -1.0  6.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0  1.0 -2.0 -1.0 -1.0 -5.0\n",
      "F -2.0 -2.0 -2.0 -4.0 -2.0 -4.0 -3.0 -3.0 -2.0  0.0  1.0 -3.0  0.0  8.0 -3.0 -2.0 -1.0  1.0  3.0  0.0 -3.0 -3.0 -1.0 -5.0\n",
      "P -1.0 -2.0 -2.0 -1.0 -4.0 -1.0  0.0 -2.0 -2.0 -2.0 -3.0 -1.0 -2.0 -3.0  9.0 -1.0 -1.0 -3.0 -3.0 -3.0 -2.0 -1.0 -1.0 -5.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0  0.0  0.0 -1.0 -2.0 -3.0 -1.0 -2.0 -2.0 -1.0  4.0  2.0 -4.0 -2.0 -1.0  0.0  0.0  0.0 -5.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  2.0  5.0 -3.0 -1.0  0.0  0.0 -1.0  0.0 -5.0\n",
      "W -2.0 -2.0 -4.0 -4.0 -5.0 -2.0 -3.0 -2.0 -3.0 -2.0 -2.0 -2.0 -2.0  1.0 -3.0 -4.0 -3.0 15.0  3.0 -3.0 -4.0 -2.0 -2.0 -5.0\n",
      "Y -2.0 -1.0 -2.0 -2.0 -3.0 -1.0 -2.0 -3.0  2.0  0.0  0.0 -1.0  0.0  3.0 -3.0 -2.0 -1.0  3.0  8.0 -1.0 -2.0 -2.0 -1.0 -5.0\n",
      "V  0.0 -2.0 -3.0 -3.0 -1.0 -3.0 -3.0 -3.0 -3.0  3.0  1.0 -2.0  1.0  0.0 -3.0 -1.0  0.0 -3.0 -1.0  5.0 -3.0 -3.0 -1.0 -5.0\n",
      "B -1.0 -1.0  4.0  5.0 -2.0  0.0  1.0 -1.0  0.0 -3.0 -3.0  0.0 -2.0 -3.0 -2.0  0.0  0.0 -4.0 -2.0 -3.0  4.0  2.0 -1.0 -5.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  4.0  4.0 -2.0  0.0 -3.0 -2.0  1.0 -1.0 -3.0 -1.0  0.0 -1.0 -2.0 -2.0 -3.0  2.0  4.0 -1.0 -5.0\n",
      "X  0.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -5.0\n",
      "* -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0  1.0\n",
      "\n",
      "BLOSUM50 #  Matrix made by matblas from blosum50.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 50\n",
      "#  Entropy =   0.4808, Expected =  -0.3573\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -1.0 -2.0 -1.0 -1.0 -1.0  0.0 -2.0 -1.0 -2.0 -1.0 -1.0 -3.0 -1.0  1.0  0.0 -3.0 -2.0  0.0 -2.0 -1.0 -1.0 -5.0\n",
      "R -2.0  7.0 -1.0 -2.0 -4.0  1.0  0.0 -3.0  0.0 -4.0 -3.0  3.0 -2.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -3.0 -1.0  0.0 -1.0 -5.0\n",
      "N -1.0 -1.0  7.0  2.0 -2.0  0.0  0.0  0.0  1.0 -3.0 -4.0  0.0 -2.0 -4.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  4.0  0.0 -1.0 -5.0\n",
      "D -2.0 -2.0  2.0  8.0 -4.0  0.0  2.0 -1.0 -1.0 -4.0 -4.0 -1.0 -4.0 -5.0 -1.0  0.0 -1.0 -5.0 -3.0 -4.0  5.0  1.0 -1.0 -5.0\n",
      "C -1.0 -4.0 -2.0 -4.0 13.0 -3.0 -3.0 -3.0 -3.0 -2.0 -2.0 -3.0 -2.0 -2.0 -4.0 -1.0 -1.0 -5.0 -3.0 -1.0 -3.0 -3.0 -2.0 -5.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  7.0  2.0 -2.0  1.0 -3.0 -2.0  2.0  0.0 -4.0 -1.0  0.0 -1.0 -1.0 -1.0 -3.0  0.0  4.0 -1.0 -5.0\n",
      "E -1.0  0.0  0.0  2.0 -3.0  2.0  6.0 -3.0  0.0 -4.0 -3.0  1.0 -2.0 -3.0 -1.0 -1.0 -1.0 -3.0 -2.0 -3.0  1.0  5.0 -1.0 -5.0\n",
      "G  0.0 -3.0  0.0 -1.0 -3.0 -2.0 -3.0  8.0 -2.0 -4.0 -4.0 -2.0 -3.0 -4.0 -2.0  0.0 -2.0 -3.0 -3.0 -4.0 -1.0 -2.0 -2.0 -5.0\n",
      "H -2.0  0.0  1.0 -1.0 -3.0  1.0  0.0 -2.0 10.0 -4.0 -3.0  0.0 -1.0 -1.0 -2.0 -1.0 -2.0 -3.0  2.0 -4.0  0.0  0.0 -1.0 -5.0\n",
      "I -1.0 -4.0 -3.0 -4.0 -2.0 -3.0 -4.0 -4.0 -4.0  5.0  2.0 -3.0  2.0  0.0 -3.0 -3.0 -1.0 -3.0 -1.0  4.0 -4.0 -3.0 -1.0 -5.0\n",
      "L -2.0 -3.0 -4.0 -4.0 -2.0 -2.0 -3.0 -4.0 -3.0  2.0  5.0 -3.0  3.0  1.0 -4.0 -3.0 -1.0 -2.0 -1.0  1.0 -4.0 -3.0 -1.0 -5.0\n",
      "K -1.0  3.0  0.0 -1.0 -3.0  2.0  1.0 -2.0  0.0 -3.0 -3.0  6.0 -2.0 -4.0 -1.0  0.0 -1.0 -3.0 -2.0 -3.0  0.0  1.0 -1.0 -5.0\n",
      "M -1.0 -2.0 -2.0 -4.0 -2.0  0.0 -2.0 -3.0 -1.0  2.0  3.0 -2.0  7.0  0.0 -3.0 -2.0 -1.0 -1.0  0.0  1.0 -3.0 -1.0 -1.0 -5.0\n",
      "F -3.0 -3.0 -4.0 -5.0 -2.0 -4.0 -3.0 -4.0 -1.0  0.0  1.0 -4.0  0.0  8.0 -4.0 -3.0 -2.0  1.0  4.0 -1.0 -4.0 -4.0 -2.0 -5.0\n",
      "P -1.0 -3.0 -2.0 -1.0 -4.0 -1.0 -1.0 -2.0 -2.0 -3.0 -4.0 -1.0 -3.0 -4.0 10.0 -1.0 -1.0 -4.0 -3.0 -3.0 -2.0 -1.0 -2.0 -5.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0 -1.0  0.0 -1.0 -3.0 -3.0  0.0 -2.0 -3.0 -1.0  5.0  2.0 -4.0 -2.0 -2.0  0.0  0.0 -1.0 -5.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  2.0  5.0 -3.0 -2.0  0.0  0.0 -1.0  0.0 -5.0\n",
      "W -3.0 -3.0 -4.0 -5.0 -5.0 -1.0 -3.0 -3.0 -3.0 -3.0 -2.0 -3.0 -1.0  1.0 -4.0 -4.0 -3.0 15.0  2.0 -3.0 -5.0 -2.0 -3.0 -5.0\n",
      "Y -2.0 -1.0 -2.0 -3.0 -3.0 -1.0 -2.0 -3.0  2.0 -1.0 -1.0 -2.0  0.0  4.0 -3.0 -2.0 -2.0  2.0  8.0 -1.0 -3.0 -2.0 -1.0 -5.0\n",
      "V  0.0 -3.0 -3.0 -4.0 -1.0 -3.0 -3.0 -4.0 -4.0  4.0  1.0 -3.0  1.0 -1.0 -3.0 -2.0  0.0 -3.0 -1.0  5.0 -4.0 -3.0 -1.0 -5.0\n",
      "B -2.0 -1.0  4.0  5.0 -3.0  0.0  1.0 -1.0  0.0 -4.0 -4.0  0.0 -3.0 -4.0 -2.0  0.0  0.0 -5.0 -3.0 -4.0  5.0  2.0 -1.0 -5.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  4.0  5.0 -2.0  0.0 -3.0 -3.0  1.0 -1.0 -4.0 -1.0  0.0 -1.0 -2.0 -2.0 -3.0  2.0  5.0 -1.0 -5.0\n",
      "X -1.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0  0.0 -3.0 -1.0 -1.0 -1.0 -1.0 -1.0 -5.0\n",
      "* -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0 -5.0  1.0\n",
      "\n",
      "BLOSUM62 #  Matrix made by matblas from blosum62.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/2 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 62\n",
      "#  Entropy =   0.6979, Expected =  -0.5209\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  4.0 -1.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  0.0 -3.0 -2.0  0.0 -2.0 -1.0  0.0 -4.0\n",
      "R -1.0  5.0  0.0 -2.0 -3.0  1.0  0.0 -2.0  0.0 -3.0 -2.0  2.0 -1.0 -3.0 -2.0 -1.0 -1.0 -3.0 -2.0 -3.0 -1.0  0.0 -1.0 -4.0\n",
      "N -2.0  0.0  6.0  1.0 -3.0  0.0  0.0  0.0  1.0 -3.0 -3.0  0.0 -2.0 -3.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  3.0  0.0 -1.0 -4.0\n",
      "D -2.0 -2.0  1.0  6.0 -3.0  0.0  2.0 -1.0 -1.0 -3.0 -4.0 -1.0 -3.0 -3.0 -1.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0\n",
      "C  0.0 -3.0 -3.0 -3.0  9.0 -3.0 -4.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -2.0 -3.0 -1.0 -1.0 -2.0 -2.0 -1.0 -3.0 -3.0 -2.0 -4.0\n",
      "Q -1.0  1.0  0.0  0.0 -3.0  5.0  2.0 -2.0  0.0 -3.0 -2.0  1.0  0.0 -3.0 -1.0  0.0 -1.0 -2.0 -1.0 -2.0  0.0  3.0 -1.0 -4.0\n",
      "E -1.0  0.0  0.0  2.0 -4.0  2.0  5.0 -2.0  0.0 -3.0 -3.0  1.0 -2.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0\n",
      "G  0.0 -2.0  0.0 -1.0 -3.0 -2.0 -2.0  6.0 -2.0 -4.0 -4.0 -2.0 -3.0 -3.0 -2.0  0.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -1.0 -4.0\n",
      "H -2.0  0.0  1.0 -1.0 -3.0  0.0  0.0 -2.0  8.0 -3.0 -3.0 -1.0 -2.0 -1.0 -2.0 -1.0 -2.0 -2.0  2.0 -3.0  0.0  0.0 -1.0 -4.0\n",
      "I -1.0 -3.0 -3.0 -3.0 -1.0 -3.0 -3.0 -4.0 -3.0  4.0  2.0 -3.0  1.0  0.0 -3.0 -2.0 -1.0 -3.0 -1.0  3.0 -3.0 -3.0 -1.0 -4.0\n",
      "L -1.0 -2.0 -3.0 -4.0 -1.0 -2.0 -3.0 -4.0 -3.0  2.0  4.0 -2.0  2.0  0.0 -3.0 -2.0 -1.0 -2.0 -1.0  1.0 -4.0 -3.0 -1.0 -4.0\n",
      "K -1.0  2.0  0.0 -1.0 -3.0  1.0  1.0 -2.0 -1.0 -3.0 -2.0  5.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  0.0  1.0 -1.0 -4.0\n",
      "M -1.0 -1.0 -2.0 -3.0 -1.0  0.0 -2.0 -3.0 -2.0  1.0  2.0 -1.0  5.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0  1.0 -3.0 -1.0 -1.0 -4.0\n",
      "F -2.0 -3.0 -3.0 -3.0 -2.0 -3.0 -3.0 -3.0 -1.0  0.0  0.0 -3.0  0.0  6.0 -4.0 -2.0 -2.0  1.0  3.0 -1.0 -3.0 -3.0 -1.0 -4.0\n",
      "P -1.0 -2.0 -2.0 -1.0 -3.0 -1.0 -1.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -4.0  7.0 -1.0 -1.0 -4.0 -3.0 -2.0 -2.0 -1.0 -2.0 -4.0\n",
      "S  1.0 -1.0  1.0  0.0 -1.0  0.0  0.0  0.0 -1.0 -2.0 -2.0  0.0 -1.0 -2.0 -1.0  4.0  1.0 -3.0 -2.0 -2.0  0.0  0.0  0.0 -4.0\n",
      "T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  5.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -4.0\n",
      "W -3.0 -3.0 -4.0 -4.0 -2.0 -2.0 -3.0 -2.0 -2.0 -3.0 -2.0 -3.0 -1.0  1.0 -4.0 -3.0 -2.0 11.0  2.0 -3.0 -4.0 -3.0 -2.0 -4.0\n",
      "Y -2.0 -2.0 -2.0 -3.0 -2.0 -1.0 -2.0 -3.0  2.0 -1.0 -1.0 -2.0 -1.0  3.0 -3.0 -2.0 -2.0  2.0  7.0 -1.0 -3.0 -2.0 -1.0 -4.0\n",
      "V  0.0 -3.0 -3.0 -3.0 -1.0 -2.0 -2.0 -3.0 -3.0  3.0  1.0 -2.0  1.0 -1.0 -2.0 -2.0  0.0 -3.0 -1.0  4.0 -3.0 -2.0 -1.0 -4.0\n",
      "B -2.0 -1.0  3.0  4.0 -3.0  0.0  1.0 -1.0  0.0 -3.0 -4.0  0.0 -3.0 -3.0 -2.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0\n",
      "Z -1.0  0.0  0.0  1.0 -3.0  3.0  4.0 -2.0  0.0 -3.0 -3.0  1.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0\n",
      "X  0.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0  0.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -4.0\n",
      "* -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0  1.0\n",
      "\n",
      "BLOSUM80 #  Matrix made by matblas from blosum80_3.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/3 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 80\n",
      "#  Entropy =   0.9868, Expected =  -0.7442\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  7.0 -3.0 -3.0 -3.0 -1.0 -2.0 -2.0  0.0 -3.0 -3.0 -3.0 -1.0 -2.0 -4.0 -1.0  2.0  0.0 -5.0 -4.0 -1.0 -3.0 -2.0 -1.0 -8.0\n",
      "R -3.0  9.0 -1.0 -3.0 -6.0  1.0 -1.0 -4.0  0.0 -5.0 -4.0  3.0 -3.0 -5.0 -3.0 -2.0 -2.0 -5.0 -4.0 -4.0 -2.0  0.0 -2.0 -8.0\n",
      "N -3.0 -1.0  9.0  2.0 -5.0  0.0 -1.0 -1.0  1.0 -6.0 -6.0  0.0 -4.0 -6.0 -4.0  1.0  0.0 -7.0 -4.0 -5.0  5.0 -1.0 -2.0 -8.0\n",
      "D -3.0 -3.0  2.0 10.0 -7.0 -1.0  2.0 -3.0 -2.0 -7.0 -7.0 -2.0 -6.0 -6.0 -3.0 -1.0 -2.0 -8.0 -6.0 -6.0  6.0  1.0 -3.0 -8.0\n",
      "C -1.0 -6.0 -5.0 -7.0 13.0 -5.0 -7.0 -6.0 -7.0 -2.0 -3.0 -6.0 -3.0 -4.0 -6.0 -2.0 -2.0 -5.0 -5.0 -2.0 -6.0 -7.0 -4.0 -8.0\n",
      "Q -2.0  1.0  0.0 -1.0 -5.0  9.0  3.0 -4.0  1.0 -5.0 -4.0  2.0 -1.0 -5.0 -3.0 -1.0 -1.0 -4.0 -3.0 -4.0 -1.0  5.0 -2.0 -8.0\n",
      "E -2.0 -1.0 -1.0  2.0 -7.0  3.0  8.0 -4.0  0.0 -6.0 -6.0  1.0 -4.0 -6.0 -2.0 -1.0 -2.0 -6.0 -5.0 -4.0  1.0  6.0 -2.0 -8.0\n",
      "G  0.0 -4.0 -1.0 -3.0 -6.0 -4.0 -4.0  9.0 -4.0 -7.0 -7.0 -3.0 -5.0 -6.0 -5.0 -1.0 -3.0 -6.0 -6.0 -6.0 -2.0 -4.0 -3.0 -8.0\n",
      "H -3.0  0.0  1.0 -2.0 -7.0  1.0  0.0 -4.0 12.0 -6.0 -5.0 -1.0 -4.0 -2.0 -4.0 -2.0 -3.0 -4.0  3.0 -5.0 -1.0  0.0 -2.0 -8.0\n",
      "I -3.0 -5.0 -6.0 -7.0 -2.0 -5.0 -6.0 -7.0 -6.0  7.0  2.0 -5.0  2.0 -1.0 -5.0 -4.0 -2.0 -5.0 -3.0  4.0 -6.0 -6.0 -2.0 -8.0\n",
      "L -3.0 -4.0 -6.0 -7.0 -3.0 -4.0 -6.0 -7.0 -5.0  2.0  6.0 -4.0  3.0  0.0 -5.0 -4.0 -3.0 -4.0 -2.0  1.0 -7.0 -5.0 -2.0 -8.0\n",
      "K -1.0  3.0  0.0 -2.0 -6.0  2.0  1.0 -3.0 -1.0 -5.0 -4.0  8.0 -3.0 -5.0 -2.0 -1.0 -1.0 -6.0 -4.0 -4.0 -1.0  1.0 -2.0 -8.0\n",
      "M -2.0 -3.0 -4.0 -6.0 -3.0 -1.0 -4.0 -5.0 -4.0  2.0  3.0 -3.0  9.0  0.0 -4.0 -3.0 -1.0 -3.0 -3.0  1.0 -5.0 -3.0 -2.0 -8.0\n",
      "F -4.0 -5.0 -6.0 -6.0 -4.0 -5.0 -6.0 -6.0 -2.0 -1.0  0.0 -5.0  0.0 10.0 -6.0 -4.0 -4.0  0.0  4.0 -2.0 -6.0 -6.0 -3.0 -8.0\n",
      "P -1.0 -3.0 -4.0 -3.0 -6.0 -3.0 -2.0 -5.0 -4.0 -5.0 -5.0 -2.0 -4.0 -6.0 12.0 -2.0 -3.0 -7.0 -6.0 -4.0 -4.0 -2.0 -3.0 -8.0\n",
      "S  2.0 -2.0  1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -2.0 -4.0 -4.0 -1.0 -3.0 -4.0 -2.0  7.0  2.0 -6.0 -3.0 -3.0  0.0 -1.0 -1.0 -8.0\n",
      "T  0.0 -2.0  0.0 -2.0 -2.0 -1.0 -2.0 -3.0 -3.0 -2.0 -3.0 -1.0 -1.0 -4.0 -3.0  2.0  8.0 -5.0 -3.0  0.0 -1.0 -2.0 -1.0 -8.0\n",
      "W -5.0 -5.0 -7.0 -8.0 -5.0 -4.0 -6.0 -6.0 -4.0 -5.0 -4.0 -6.0 -3.0  0.0 -7.0 -6.0 -5.0 16.0  3.0 -5.0 -8.0 -5.0 -5.0 -8.0\n",
      "Y -4.0 -4.0 -4.0 -6.0 -5.0 -3.0 -5.0 -6.0  3.0 -3.0 -2.0 -4.0 -3.0  4.0 -6.0 -3.0 -3.0  3.0 11.0 -3.0 -5.0 -4.0 -3.0 -8.0\n",
      "V -1.0 -4.0 -5.0 -6.0 -2.0 -4.0 -4.0 -6.0 -5.0  4.0  1.0 -4.0  1.0 -2.0 -4.0 -3.0  0.0 -5.0 -3.0  7.0 -6.0 -4.0 -2.0 -8.0\n",
      "B -3.0 -2.0  5.0  6.0 -6.0 -1.0  1.0 -2.0 -1.0 -6.0 -7.0 -1.0 -5.0 -6.0 -4.0  0.0 -1.0 -8.0 -5.0 -6.0  6.0  0.0 -3.0 -8.0\n",
      "Z -2.0  0.0 -1.0  1.0 -7.0  5.0  6.0 -4.0  0.0 -6.0 -5.0  1.0 -3.0 -6.0 -2.0 -1.0 -2.0 -5.0 -4.0 -4.0  0.0  6.0 -1.0 -8.0\n",
      "X -1.0 -2.0 -2.0 -3.0 -4.0 -2.0 -2.0 -3.0 -2.0 -2.0 -2.0 -2.0 -2.0 -3.0 -3.0 -1.0 -1.0 -5.0 -3.0 -2.0 -3.0 -1.0 -2.0 -8.0\n",
      "* -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0  1.0\n",
      "\n",
      "BLOSUM90 #  Matrix made by matblas from blosum90.iij\n",
      "#  * column uses minimum score\n",
      "#  BLOSUM Clustered Scoring Matrix in 1/2 Bit Units\n",
      "#  Blocks Database = /data/blocks_5.0/blocks.dat\n",
      "#  Cluster Percentage: >= 90\n",
      "#  Entropy =   1.1806, Expected =  -0.8887\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  5.0 -2.0 -2.0 -3.0 -1.0 -1.0 -1.0  0.0 -2.0 -2.0 -2.0 -1.0 -2.0 -3.0 -1.0  1.0  0.0 -4.0 -3.0 -1.0 -2.0 -1.0 -1.0 -6.0\n",
      "R -2.0  6.0 -1.0 -3.0 -5.0  1.0 -1.0 -3.0  0.0 -4.0 -3.0  2.0 -2.0 -4.0 -3.0 -1.0 -2.0 -4.0 -3.0 -3.0 -2.0  0.0 -2.0 -6.0\n",
      "N -2.0 -1.0  7.0  1.0 -4.0  0.0 -1.0 -1.0  0.0 -4.0 -4.0  0.0 -3.0 -4.0 -3.0  0.0  0.0 -5.0 -3.0 -4.0  4.0 -1.0 -2.0 -6.0\n",
      "D -3.0 -3.0  1.0  7.0 -5.0 -1.0  1.0 -2.0 -2.0 -5.0 -5.0 -1.0 -4.0 -5.0 -3.0 -1.0 -2.0 -6.0 -4.0 -5.0  4.0  0.0 -2.0 -6.0\n",
      "C -1.0 -5.0 -4.0 -5.0  9.0 -4.0 -6.0 -4.0 -5.0 -2.0 -2.0 -4.0 -2.0 -3.0 -4.0 -2.0 -2.0 -4.0 -4.0 -2.0 -4.0 -5.0 -3.0 -6.0\n",
      "Q -1.0  1.0  0.0 -1.0 -4.0  7.0  2.0 -3.0  1.0 -4.0 -3.0  1.0  0.0 -4.0 -2.0 -1.0 -1.0 -3.0 -3.0 -3.0 -1.0  4.0 -1.0 -6.0\n",
      "E -1.0 -1.0 -1.0  1.0 -6.0  2.0  6.0 -3.0 -1.0 -4.0 -4.0  0.0 -3.0 -5.0 -2.0 -1.0 -1.0 -5.0 -4.0 -3.0  0.0  4.0 -2.0 -6.0\n",
      "G  0.0 -3.0 -1.0 -2.0 -4.0 -3.0 -3.0  6.0 -3.0 -5.0 -5.0 -2.0 -4.0 -5.0 -3.0 -1.0 -3.0 -4.0 -5.0 -5.0 -2.0 -3.0 -2.0 -6.0\n",
      "H -2.0  0.0  0.0 -2.0 -5.0  1.0 -1.0 -3.0  8.0 -4.0 -4.0 -1.0 -3.0 -2.0 -3.0 -2.0 -2.0 -3.0  1.0 -4.0 -1.0  0.0 -2.0 -6.0\n",
      "I -2.0 -4.0 -4.0 -5.0 -2.0 -4.0 -4.0 -5.0 -4.0  5.0  1.0 -4.0  1.0 -1.0 -4.0 -3.0 -1.0 -4.0 -2.0  3.0 -5.0 -4.0 -2.0 -6.0\n",
      "L -2.0 -3.0 -4.0 -5.0 -2.0 -3.0 -4.0 -5.0 -4.0  1.0  5.0 -3.0  2.0  0.0 -4.0 -3.0 -2.0 -3.0 -2.0  0.0 -5.0 -4.0 -2.0 -6.0\n",
      "K -1.0  2.0  0.0 -1.0 -4.0  1.0  0.0 -2.0 -1.0 -4.0 -3.0  6.0 -2.0 -4.0 -2.0 -1.0 -1.0 -5.0 -3.0 -3.0 -1.0  1.0 -1.0 -6.0\n",
      "M -2.0 -2.0 -3.0 -4.0 -2.0  0.0 -3.0 -4.0 -3.0  1.0  2.0 -2.0  7.0 -1.0 -3.0 -2.0 -1.0 -2.0 -2.0  0.0 -4.0 -2.0 -1.0 -6.0\n",
      "F -3.0 -4.0 -4.0 -5.0 -3.0 -4.0 -5.0 -5.0 -2.0 -1.0  0.0 -4.0 -1.0  7.0 -4.0 -3.0 -3.0  0.0  3.0 -2.0 -4.0 -4.0 -2.0 -6.0\n",
      "P -1.0 -3.0 -3.0 -3.0 -4.0 -2.0 -2.0 -3.0 -3.0 -4.0 -4.0 -2.0 -3.0 -4.0  8.0 -2.0 -2.0 -5.0 -4.0 -3.0 -3.0 -2.0 -2.0 -6.0\n",
      "S  1.0 -1.0  0.0 -1.0 -2.0 -1.0 -1.0 -1.0 -2.0 -3.0 -3.0 -1.0 -2.0 -3.0 -2.0  5.0  1.0 -4.0 -3.0 -2.0  0.0 -1.0 -1.0 -6.0\n",
      "T  0.0 -2.0  0.0 -2.0 -2.0 -1.0 -1.0 -3.0 -2.0 -1.0 -2.0 -1.0 -1.0 -3.0 -2.0  1.0  6.0 -4.0 -2.0 -1.0 -1.0 -1.0 -1.0 -6.0\n",
      "W -4.0 -4.0 -5.0 -6.0 -4.0 -3.0 -5.0 -4.0 -3.0 -4.0 -3.0 -5.0 -2.0  0.0 -5.0 -4.0 -4.0 11.0  2.0 -3.0 -6.0 -4.0 -3.0 -6.0\n",
      "Y -3.0 -3.0 -3.0 -4.0 -4.0 -3.0 -4.0 -5.0  1.0 -2.0 -2.0 -3.0 -2.0  3.0 -4.0 -3.0 -2.0  2.0  8.0 -3.0 -4.0 -3.0 -2.0 -6.0\n",
      "V -1.0 -3.0 -4.0 -5.0 -2.0 -3.0 -3.0 -5.0 -4.0  3.0  0.0 -3.0  0.0 -2.0 -3.0 -2.0 -1.0 -3.0 -3.0  5.0 -4.0 -3.0 -2.0 -6.0\n",
      "B -2.0 -2.0  4.0  4.0 -4.0 -1.0  0.0 -2.0 -1.0 -5.0 -5.0 -1.0 -4.0 -4.0 -3.0  0.0 -1.0 -6.0 -4.0 -4.0  4.0  0.0 -2.0 -6.0\n",
      "Z -1.0  0.0 -1.0  0.0 -5.0  4.0  4.0 -3.0  0.0 -4.0 -4.0  1.0 -2.0 -4.0 -2.0 -1.0 -1.0 -4.0 -3.0 -3.0  0.0  4.0 -1.0 -6.0\n",
      "X -1.0 -2.0 -2.0 -2.0 -3.0 -1.0 -2.0 -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -3.0 -2.0 -2.0 -2.0 -1.0 -2.0 -6.0\n",
      "* -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0 -6.0  1.0\n",
      "\n",
      "DAYHOFF #  M.O. Dayhoff, R.M. Schwartz, and B.C. Orcutt:\n",
      "#  \"A Model of Evolutionary Change in Proteins.\"\n",
      "#  Margaret O. Dayhoff: Atlas of Protein Sequence and Structure,\n",
      "#  Volume 5, Supplement 3, 1978, pages 345-352.\n",
      "#  The National Biomedical Research Foundation, 1979.\n",
      "#  Figure 84, page 352.\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  0.2 -0.2  0.0  0.0 -0.4  0.1 -0.1 -0.1 -0.1 -0.2 -0.1  0.0  0.1  0.0 -0.2  0.1  0.1  0.0 -0.6 -0.3\n",
      "C -0.2  1.2 -0.5 -0.5 -0.4 -0.3 -0.3 -0.2 -0.5 -0.6 -0.5 -0.4 -0.3 -0.5 -0.4  0.0 -0.2 -0.2 -0.8  0.0\n",
      "D  0.0 -0.5  0.4  0.3 -0.6  0.1  0.1 -0.2  0.0 -0.4 -0.3  0.2 -0.1  0.2 -0.1  0.0  0.0 -0.2 -0.7 -0.4\n",
      "E  0.0 -0.5  0.3  0.4 -0.5  0.0  0.1 -0.2  0.0 -0.3 -0.2  0.1 -0.1  0.2 -0.1  0.0  0.0 -0.2 -0.7 -0.4\n",
      "F -0.4 -0.4 -0.6 -0.5  0.9 -0.5 -0.2  0.1 -0.5  0.2  0.0 -0.4 -0.5 -0.5 -0.4 -0.3 -0.3 -0.1  0.0  0.7\n",
      "G  0.1 -0.3  0.1  0.0 -0.5  0.5 -0.2 -0.3 -0.2 -0.4 -0.3  0.0 -0.1 -0.1 -0.3  0.1  0.0 -0.1 -0.7 -0.5\n",
      "H -0.1 -0.3  0.1  0.1 -0.2 -0.2  0.6 -0.2  0.0 -0.2 -0.2  0.2  0.0  0.3  0.2 -0.1 -0.1 -0.2 -0.3  0.0\n",
      "I -0.1 -0.2 -0.2 -0.2  0.1 -0.3 -0.2  0.5 -0.2  0.2  0.2 -0.2 -0.2 -0.2 -0.2 -0.1  0.0  0.4 -0.5 -0.1\n",
      "K -0.1 -0.5  0.0  0.0 -0.5 -0.2  0.0 -0.2  0.5 -0.3  0.0  0.1 -0.1  0.1  0.3  0.0  0.0 -0.2 -0.3 -0.4\n",
      "L -0.2 -0.6 -0.4 -0.3  0.2 -0.4 -0.2  0.2 -0.3  0.6  0.4 -0.3 -0.3 -0.2 -0.3 -0.3 -0.2  0.2 -0.2 -0.1\n",
      "M -0.1 -0.5 -0.3 -0.2  0.0 -0.3 -0.2  0.2  0.0  0.4  0.6 -0.2 -0.2 -0.1  0.0 -0.2 -0.1  0.2 -0.4 -0.2\n",
      "N  0.0 -0.4  0.2  0.1 -0.4  0.0  0.2 -0.2  0.1 -0.3 -0.2  0.2 -0.1  0.1  0.0  0.1  0.0 -0.2 -0.4 -0.2\n",
      "P  0.1 -0.3 -0.1 -0.1 -0.5 -0.1  0.0 -0.2 -0.1 -0.3 -0.2 -0.1  0.6  0.0  0.0  0.1  0.0 -0.1 -0.6 -0.5\n",
      "Q  0.0 -0.5  0.2  0.2 -0.5 -0.1  0.3 -0.2  0.1 -0.2 -0.1  0.1  0.0  0.4  0.1 -0.1 -0.1 -0.2 -0.5 -0.4\n",
      "R -0.2 -0.4 -0.1 -0.1 -0.4 -0.3  0.2 -0.2  0.3 -0.3  0.0  0.0  0.0  0.1  0.6  0.0 -0.1 -0.2  0.2 -0.4\n",
      "S  0.1  0.0  0.0  0.0 -0.3  0.1 -0.1 -0.1  0.0 -0.3 -0.2  0.1  0.1 -0.1  0.0  0.2  0.1 -0.1 -0.2 -0.3\n",
      "T  0.1 -0.2  0.0  0.0 -0.3  0.0 -0.1  0.0  0.0 -0.2 -0.1  0.0  0.0 -0.1 -0.1  0.1  0.3  0.0 -0.5 -0.3\n",
      "V  0.0 -0.2 -0.2 -0.2 -0.1 -0.1 -0.2  0.4 -0.2  0.2  0.2 -0.2 -0.1 -0.2 -0.2 -0.1  0.0  0.4 -0.6 -0.2\n",
      "W -0.6 -0.8 -0.7 -0.7  0.0 -0.7 -0.3 -0.5 -0.3 -0.2 -0.4 -0.4 -0.6 -0.5  0.2 -0.2 -0.5 -0.6  1.7  0.0\n",
      "Y -0.3  0.0 -0.4 -0.4  0.7 -0.5  0.0 -0.1 -0.4 -0.1 -0.2 -0.2 -0.5 -0.4 -0.4 -0.3 -0.3 -0.2  0.0  1.0\n",
      "\n",
      "FENG #  D.F. Feng, M.S. Johnson, R.F. Doolittle:\n",
      "#  \"Aligning amino acid sequences: Comparison of commonly used methods.\"\n",
      "#  Journal of Molecular Evolution 21(2): 112-125 (1985).\n",
      "#  Table 1, upper triangle.\n",
      "#  PMID 6100188\n",
      "    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y\n",
      "A 6.0 2.0 4.0 4.0 2.0 5.0 2.0 2.0 3.0 2.0 2.0 3.0 5.0 3.0 2.0 5.0 5.0 5.0 2.0 2.0\n",
      "C 2.0 6.0 1.0 0.0 3.0 3.0 2.0 2.0 0.0 2.0 2.0 2.0 2.0 1.0 2.0 4.0 2.0 2.0 3.0 3.0\n",
      "D 4.0 1.0 6.0 5.0 1.0 4.0 3.0 1.0 3.0 1.0 0.0 5.0 2.0 4.0 2.0 3.0 2.0 3.0 0.0 2.0\n",
      "E 4.0 0.0 5.0 6.0 0.0 4.0 2.0 1.0 4.0 1.0 1.0 3.0 3.0 4.0 2.0 3.0 3.0 4.0 1.0 1.0\n",
      "F 2.0 3.0 1.0 0.0 6.0 1.0 2.0 4.0 0.0 4.0 2.0 1.0 2.0 1.0 1.0 3.0 1.0 4.0 3.0 5.0\n",
      "G 5.0 3.0 4.0 4.0 1.0 6.0 1.0 2.0 2.0 2.0 1.0 3.0 3.0 2.0 3.0 5.0 2.0 4.0 3.0 2.0\n",
      "H 2.0 2.0 3.0 2.0 2.0 1.0 6.0 1.0 3.0 3.0 1.0 4.0 3.0 4.0 4.0 3.0 2.0 1.0 1.0 3.0\n",
      "I 2.0 2.0 1.0 1.0 4.0 2.0 1.0 6.0 2.0 5.0 4.0 2.0 2.0 1.0 2.0 2.0 3.0 5.0 2.0 3.0\n",
      "K 3.0 0.0 3.0 4.0 0.0 2.0 3.0 2.0 6.0 2.0 2.0 4.0 2.0 4.0 5.0 3.0 4.0 3.0 1.0 1.0\n",
      "L 2.0 2.0 1.0 1.0 4.0 2.0 3.0 5.0 2.0 6.0 5.0 1.0 3.0 2.0 2.0 2.0 2.0 5.0 4.0 3.0\n",
      "M 2.0 2.0 0.0 1.0 2.0 1.0 1.0 4.0 2.0 5.0 6.0 1.0 2.0 2.0 2.0 1.0 3.0 4.0 3.0 2.0\n",
      "N 3.0 2.0 5.0 3.0 1.0 3.0 4.0 2.0 4.0 1.0 1.0 6.0 2.0 3.0 2.0 5.0 4.0 2.0 0.0 3.0\n",
      "P 5.0 2.0 2.0 3.0 2.0 3.0 3.0 2.0 2.0 3.0 2.0 2.0 6.0 3.0 3.0 4.0 4.0 3.0 2.0 2.0\n",
      "Q 3.0 1.0 4.0 4.0 1.0 2.0 4.0 1.0 4.0 2.0 2.0 3.0 3.0 6.0 3.0 3.0 3.0 2.0 1.0 2.0\n",
      "R 2.0 2.0 2.0 2.0 1.0 3.0 4.0 2.0 5.0 2.0 2.0 2.0 3.0 3.0 6.0 3.0 3.0 2.0 2.0 1.0\n",
      "S 5.0 4.0 3.0 3.0 3.0 5.0 3.0 2.0 3.0 2.0 1.0 5.0 4.0 3.0 3.0 6.0 5.0 2.0 2.0 3.0\n",
      "T 5.0 2.0 2.0 3.0 1.0 2.0 2.0 3.0 4.0 2.0 3.0 4.0 4.0 3.0 3.0 5.0 6.0 3.0 1.0 2.0\n",
      "V 5.0 2.0 3.0 4.0 4.0 4.0 1.0 5.0 3.0 5.0 4.0 2.0 3.0 2.0 2.0 2.0 3.0 6.0 3.0 3.0\n",
      "W 2.0 3.0 0.0 1.0 3.0 3.0 1.0 2.0 1.0 4.0 3.0 0.0 2.0 1.0 2.0 2.0 1.0 3.0 6.0 3.0\n",
      "Y 2.0 3.0 2.0 1.0 5.0 2.0 3.0 3.0 1.0 3.0 2.0 3.0 2.0 2.0 1.0 3.0 2.0 3.0 3.0 6.0\n",
      "\n",
      "GENETIC #  S.A. Benner, M.A. Cohen, G.H. Gonnet:\n",
      "#  \"Amino acid substitution during functionally constrained divergent evolution\n",
      "#  of protein sequences.\"\n",
      "#  Figure 5.\n",
      "#  Protein Engineering 7(11): 1323-1332 (1994).\n",
      "#  PMID 7700864\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  4.0 -1.9  1.0  1.3 -2.4  1.2 -2.1 -1.8 -1.9 -2.3 -2.0 -1.7  0.8 -2.1 -1.6  0.1  0.9  1.0 -2.2 -2.4\n",
      "C -1.9  5.5 -1.6 -3.0  1.8  1.0 -1.6 -1.9 -3.2 -1.3 -2.7 -1.5 -1.9 -3.1  0.7  1.5 -1.9 -2.2  4.1  2.6\n",
      "D  1.0 -1.6  4.8  3.8 -1.7  1.1  1.7 -2.1  0.3 -2.4 -2.5  1.7 -2.2  0.3 -2.3 -2.1 -2.1  1.0 -2.9  2.3\n",
      "E  1.3 -3.0  3.8  5.7 -2.9  1.4  0.3 -2.3  2.0 -2.5 -1.8  0.3 -2.1  2.0 -2.0 -2.8 -2.1  1.3 -3.2 -0.9\n",
      "F -2.4  1.8 -1.7 -2.9  4.5 -1.9 -1.1  1.3 -2.8  2.2  0.5 -1.3 -1.8 -2.1 -1.5  0.0 -2.1  1.0  0.0  2.0\n",
      "G  1.2  1.0  1.1  1.4 -1.9  4.2 -2.2 -2.5 -2.2 -2.2 -2.3 -2.6 -1.8 -2.1  0.8 -0.6 -2.1  1.1  1.4 -1.8\n",
      "H -2.1 -1.6  1.7  0.3 -1.1 -2.2  4.7 -1.8  0.6 -0.1 -1.8  1.8  0.7  3.6  3.6 -1.6 -1.8 -2.1 -2.1  2.3\n",
      "I -1.8 -1.9 -2.1 -2.3  1.3 -2.5 -1.8  4.1  0.7  1.2  3.3  0.9 -1.6 -1.9 -1.2 -0.5  0.8  1.0 -2.2 -1.6\n",
      "K -1.9 -3.2  0.3  2.0 -2.8 -2.2  0.6  0.7  5.6 -2.0  1.6  3.5 -1.5  2.2 -0.2 -1.5  1.0 -2.1 -3.0 -0.8\n",
      "L -2.3 -1.3 -2.4 -2.5  2.2 -2.2 -0.1  1.2 -2.0  3.4  1.5 -2.2  0.0  0.1 -0.4 -1.2 -1.9  1.1 -0.3 -1.6\n",
      "M -2.0 -2.7 -2.5 -1.8  0.5 -2.3 -1.8  3.3  1.6  1.5  5.4  0.1 -1.4 -1.2 -0.4 -1.3  0.7  1.0 -2.0 -2.9\n",
      "N -1.7 -1.5  1.7  0.3 -1.3 -2.6  1.8  0.9  3.5 -2.2  0.1  4.7 -1.6  0.4 -1.5 -0.3  0.9 -2.2 -3.0  2.5\n",
      "P  0.8 -1.9 -2.2 -2.1 -1.8 -1.8  0.7 -1.6 -1.5  0.0 -1.4 -1.6  3.8  1.0  0.3  0.4  1.1 -2.1 -1.6 -2.3\n",
      "Q -2.1 -3.1  0.3  2.0 -2.1 -2.1  3.6 -1.9  2.2  0.1 -1.2  0.4  1.0  5.5  0.3 -2.3 -1.7 -2.0 -2.3 -0.8\n",
      "R -1.6  0.7 -2.3 -2.0 -1.5  0.8  3.6 -1.2 -0.2 -0.4 -0.4 -1.5  0.3  0.3  2.9  0.3 -0.6 -2.1  1.8 -1.9\n",
      "S  0.1  1.5 -2.1 -2.8  0.0 -0.6 -1.6 -0.5 -1.5 -1.2 -1.3 -0.3  0.4 -2.3  0.3  2.6  1.0 -2.2  0.8  0.3\n",
      "T  0.9 -1.9 -2.1 -2.1 -2.1 -2.1 -1.8  0.8  1.0 -1.9  0.7  0.9  1.1 -1.7 -0.6  1.0  4.0 -2.2 -2.2 -2.1\n",
      "V  1.0 -2.2  1.0  1.3  1.0  1.1 -2.1  1.0 -2.1  1.1  1.0 -2.2 -2.1 -2.0 -2.1 -2.2 -2.2  4.1 -2.1 -2.2\n",
      "W -2.2  4.1 -2.9 -3.2  0.0  1.4 -2.1 -2.2 -3.0 -0.3 -2.0 -3.0 -1.6 -2.3  1.8  0.8 -2.2 -2.1  7.5 -0.5\n",
      "Y -2.4  2.6  2.3 -0.9  2.0 -1.8  2.3 -1.6 -0.8 -1.6 -2.9  2.5 -2.3 -0.8 -1.9  0.3 -2.1 -2.2 -0.5  6.5\n",
      "\n",
      "GONNET1992 #  Gaston H. Gonnet, Mark A. Cohen, Steven A. Benner:\n",
      "#  \"Exhaustive matching of the entire protein sequence database.\"\n",
      "#  Science 256(5062): 1443-1445 (1992).\n",
      "#  Figure 2.\n",
      "#  PMID 1604319\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.4  0.5 -0.3  0.0 -2.3  0.5 -0.8 -0.8 -0.4 -1.2 -0.7 -0.3  0.3 -0.2 -0.6  1.1  0.6  0.1 -3.6 -2.2\n",
      "C  0.5 11.5 -3.2 -3.0 -0.8 -2.0 -1.3 -1.1 -2.8 -1.5 -0.9 -1.8 -3.1 -2.4 -2.2  0.1 -0.5  0.0 -1.0 -0.5\n",
      "D -0.3 -3.2  4.7  2.7 -4.5  0.1  0.4 -3.8  0.5 -4.0 -3.0  2.2 -0.7  0.9 -0.3  0.5  0.0 -2.9 -5.2 -2.8\n",
      "E  0.0 -3.0  2.7  3.6 -3.9 -0.8  0.4 -2.7  1.2 -2.8 -2.0  0.9 -0.5  1.7  0.4  0.2 -0.1 -1.9 -4.3 -2.7\n",
      "F -2.3 -0.8 -4.5 -3.9  7.0 -5.2 -0.1  1.0 -3.3  2.0  1.6 -3.1 -3.8 -2.6 -3.2 -2.8 -2.2  0.1  3.6  5.1\n",
      "G  0.5 -2.0  0.1 -0.8 -5.2  6.6 -1.4 -4.5 -1.1 -4.4 -3.5  0.4 -1.6 -1.0 -1.0  0.4 -1.1 -3.3 -4.0 -4.0\n",
      "H -0.8 -1.3  0.4  0.4 -0.1 -1.4  6.0 -2.2  0.6 -1.9 -1.3  1.2 -1.1  1.2  0.6 -0.2 -0.3 -2.0 -0.8  2.2\n",
      "I -0.8 -1.1 -3.8 -2.7  1.0 -4.5 -2.2  4.0 -2.1  2.8  2.5 -2.8 -2.6 -1.9 -2.4 -1.8 -0.6  3.1 -1.8 -0.7\n",
      "K -0.4 -2.8  0.5  1.2 -3.3 -1.1  0.6 -2.1  3.2 -2.1 -1.4  0.8 -0.6  1.5  2.7  0.1  0.1 -1.7 -3.5 -2.1\n",
      "L -1.2 -1.5 -4.0 -2.8  2.0 -4.4 -1.9  2.8 -2.1  4.0  2.8 -3.0 -2.3 -1.6 -2.2 -2.1 -1.3  1.8 -0.7  0.0\n",
      "M -0.7 -0.9 -3.0 -2.0  1.6 -3.5 -1.3  2.5 -1.4  2.8  4.3 -2.2 -2.4 -1.0 -1.7 -1.4 -0.6  1.6 -1.0 -0.2\n",
      "N -0.3 -1.8  2.2  0.9 -3.1  0.4  1.2 -2.8  0.8 -3.0 -2.2  3.8 -0.9  0.7  0.3  0.9  0.5 -2.2 -3.6 -1.4\n",
      "P  0.3 -3.1 -0.7 -0.5 -3.8 -1.6 -1.1 -2.6 -0.6 -2.3 -2.4 -0.9  7.6 -0.2 -0.9  0.4  0.1 -1.8 -5.0 -3.1\n",
      "Q -0.2 -2.4  0.9  1.7 -2.6 -1.0  1.2 -1.9  1.5 -1.6 -1.0  0.7 -0.2  2.7  1.5  0.2  0.0 -1.5 -2.7 -1.7\n",
      "R -0.6 -2.2 -0.3  0.4 -3.2 -1.0  0.6 -2.4  2.7 -2.2 -1.7  0.3 -0.9  1.5  4.7 -0.2 -0.2 -2.0 -1.6 -1.8\n",
      "S  1.1  0.1  0.5  0.2 -2.8  0.4 -0.2 -1.8  0.1 -2.1 -1.4  0.9  0.4  0.2 -0.2  2.2  1.5 -1.0 -3.3 -1.9\n",
      "T  0.6 -0.5  0.0 -0.1 -2.2 -1.1 -0.3 -0.6  0.1 -1.3 -0.6  0.5  0.1  0.0 -0.2  1.5  2.5  0.0 -3.5 -1.9\n",
      "V  0.1  0.0 -2.9 -1.9  0.1 -3.3 -2.0  3.1 -1.7  1.8  1.6 -2.2 -1.8 -1.5 -2.0 -1.0  0.0  3.4 -2.6 -1.1\n",
      "W -3.6 -1.0 -5.2 -4.3  3.6 -4.0 -0.8 -1.8 -3.5 -0.7 -1.0 -3.6 -5.0 -2.7 -1.6 -3.3 -3.5 -2.6 14.2  4.1\n",
      "Y -2.2 -0.5 -2.8 -2.7  5.1 -4.0  2.2 -0.7 -2.1  0.0 -0.2 -1.4 -3.1 -1.7 -1.8 -1.9 -1.9 -1.1  4.1  7.8\n",
      "\n",
      "JOHNSON #  Mark S. Johnson and John P. Overington:\n",
      "#  \"A structural basis for sequence comparisons. An evaluation of scoring\n",
      "#  methodologies.\"\n",
      "#  Journal of Molecular Biology 233(4): 716-738 (1993).\n",
      "#  Table 3, upper triangle.\n",
      "#  PMID 8411177\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  0.6 -0.3 -0.2 -0.1 -0.3 -0.1 -0.3 -0.2 -0.1 -0.3 -0.1 -0.1 -0.1 -0.1 -0.2  0.0 -0.1 -0.1 -0.6 -0.4\n",
      "C -0.3  1.6 -1.0 -0.7 -0.4 -0.8 -0.8 -0.8 -0.9 -0.9 -0.4 -0.8 -0.9 -0.7 -0.6 -0.8 -0.6 -0.5 -0.9 -0.8\n",
      "D -0.2 -1.0  0.8  0.2 -0.7 -0.2 -0.1 -0.5 -0.1 -0.8 -0.6  0.3 -0.1 -0.1 -0.3 -0.0 -0.2 -0.5 -0.6 -0.4\n",
      "E -0.1 -0.7  0.2  0.9 -0.6 -0.2 -0.2 -0.5  0.1 -0.6 -0.3 -0.1 -0.1  0.2 -0.0 -0.2 -0.1 -0.4 -0.8 -0.4\n",
      "F -0.3 -0.4 -0.7 -0.6  1.0 -0.9 -0.2  0.1 -0.6  0.2 -0.1 -0.4 -0.5 -0.6 -0.6 -0.5 -0.5 -0.1  0.3  0.3\n",
      "G -0.1 -0.8 -0.2 -0.2 -0.9  0.8 -0.3 -0.6 -0.3 -0.7 -0.5 -0.1 -0.2 -0.3 -0.3 -0.1 -0.4 -0.6 -0.6 -0.5\n",
      "H -0.3 -0.8 -0.1 -0.2 -0.2 -0.3  1.3 -0.5  0.0 -0.4 -0.2  0.2 -0.4  0.1  0.0 -0.3 -0.3 -0.4 -0.4 -0.0\n",
      "I -0.2 -0.8 -0.5 -0.5  0.1 -0.6 -0.5  0.8 -0.5  0.3  0.3 -0.5 -0.6 -0.7 -0.5 -0.5 -0.3  0.4 -0.3 -0.2\n",
      "K -0.1 -0.9 -0.1  0.1 -0.6 -0.3  0.0 -0.5  0.8 -0.3 -0.2  0.0 -0.1  0.1  0.3 -0.1 -0.0 -0.4 -0.5 -0.4\n",
      "L -0.3 -0.9 -0.8 -0.6  0.2 -0.7 -0.4  0.3 -0.3  0.7  0.4 -0.5 -0.3 -0.4 -0.4 -0.5 -0.5  0.2 -0.1 -0.2\n",
      "M -0.1 -0.4 -0.6 -0.3 -0.1 -0.5 -0.2  0.3 -0.2  0.4  1.1 -0.4 -1.0 -0.1 -0.4 -0.5 -0.3  0.1 -0.1 -0.1\n",
      "N -0.1 -0.8  0.3 -0.1 -0.4 -0.1  0.2 -0.5  0.0 -0.5 -0.4  0.8 -0.2 -0.1 -0.1  0.1  0.0 -0.6 -0.6 -0.1\n",
      "P -0.1 -0.9 -0.1 -0.1 -0.5 -0.2 -0.4 -0.6 -0.1 -0.3 -1.0 -0.2  1.0 -0.4 -0.4 -0.1 -0.2 -0.5 -0.7 -0.7\n",
      "Q -0.1 -0.7 -0.1  0.2 -0.6 -0.3  0.1 -0.7  0.1 -0.4 -0.1 -0.1 -0.4  0.9  0.2 -0.1 -0.0 -0.4 -0.8 -0.5\n",
      "R -0.2 -0.6 -0.3 -0.0 -0.6 -0.3  0.0 -0.5  0.3 -0.4 -0.4 -0.1 -0.4  0.2  1.0 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "S  0.0 -0.8 -0.0 -0.2 -0.5 -0.1 -0.3 -0.5 -0.1 -0.5 -0.5  0.1 -0.1 -0.1 -0.1  0.6  0.2 -0.4 -0.6 -0.3\n",
      "T -0.1 -0.6 -0.2 -0.1 -0.5 -0.4 -0.3 -0.3 -0.0 -0.5 -0.3  0.0 -0.2 -0.0 -0.1  0.2  0.7 -0.2 -0.9 -0.3\n",
      "V -0.1 -0.5 -0.5 -0.4 -0.1 -0.6 -0.4  0.4 -0.4  0.2  0.1 -0.6 -0.5 -0.4 -0.5 -0.4 -0.2  0.7 -0.5 -0.2\n",
      "W -0.6 -0.9 -0.6 -0.8  0.3 -0.6 -0.4 -0.3 -0.5 -0.1 -0.1 -0.6 -0.7 -0.8 -0.4 -0.6 -0.9 -0.5  1.5  0.2\n",
      "Y -0.4 -0.8 -0.4 -0.4  0.3 -0.5 -0.0 -0.2 -0.4 -0.2 -0.1 -0.1 -0.7 -0.5 -0.2 -0.3 -0.3 -0.2  0.2  1.1\n",
      "\n",
      "JONES #  David T. Jones, William R. Taylor, Janet M. Thornton:\n",
      "#  \"The rapid generation of mutation data matrices from protein sequences.\"\n",
      "#  Computer Applications in the Biosciences: CABIOS 8(3): 275-282 (1992).\n",
      "#  Table I, lower triangle.\n",
      "#  PMID 1633570\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V\n",
      "A  0.2 -0.1  0.0  0.0 -0.1 -0.1 -0.1  0.1 -0.2  0.0 -0.1 -0.1 -0.1 -0.3  0.1  0.1  0.2 -0.4 -0.3  0.1\n",
      "R -0.1  0.5  0.0 -0.1 -0.1  0.2  0.0  0.0  0.2 -0.3 -0.3  0.4 -0.2 -0.4 -0.1 -0.1 -0.1  0.0 -0.2 -0.3\n",
      "N  0.0  0.0  0.3  0.2 -0.1  0.0  0.1  0.0  0.1 -0.2 -0.3  0.1 -0.2 -0.3 -0.1  0.1  0.1 -0.5 -0.1 -0.2\n",
      "D  0.0 -0.1  0.2  0.5 -0.3  0.1  0.4  0.1  0.0 -0.3 -0.4  0.0 -0.3 -0.5 -0.2  0.0 -0.1 -0.5 -0.2 -0.2\n",
      "C -0.1 -0.1 -0.1 -0.3  1.1 -0.3 -0.4 -0.1  0.0 -0.2 -0.3 -0.3 -0.2  0.0 -0.2  0.1 -0.1  0.1  0.2 -0.2\n",
      "Q -0.1  0.2  0.0  0.1 -0.3  0.5  0.2 -0.1  0.2 -0.3 -0.2  0.2 -0.2 -0.4  0.0 -0.1 -0.1 -0.3 -0.2 -0.3\n",
      "E -0.1  0.0  0.1  0.4 -0.4  0.2  0.5  0.0  0.0 -0.3 -0.4  0.1 -0.3 -0.5 -0.2 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "G  0.1  0.0  0.0  0.1 -0.1 -0.1  0.0  0.5 -0.2 -0.3 -0.4 -0.1 -0.3 -0.5 -0.1  0.1 -0.1 -0.2 -0.4 -0.2\n",
      "H -0.2  0.2  0.1  0.0  0.0  0.2  0.0 -0.2  0.6 -0.3 -0.2  0.1 -0.2  0.0  0.0 -0.1 -0.1 -0.3  0.4 -0.3\n",
      "I  0.0 -0.3 -0.2 -0.3 -0.2 -0.3 -0.3 -0.3 -0.3  0.4  0.2 -0.3  0.3  0.0 -0.2 -0.1  0.1 -0.4 -0.2  0.4\n",
      "L -0.1 -0.3 -0.3 -0.4 -0.3 -0.2 -0.4 -0.4 -0.2  0.2  0.5 -0.3  0.3  0.2  0.0 -0.2 -0.1 -0.2 -0.1  0.2\n",
      "K -0.1  0.4  0.1  0.0 -0.3  0.2  0.1 -0.1  0.1 -0.3 -0.3  0.5 -0.2 -0.5 -0.2 -0.1 -0.1 -0.3 -0.3 -0.3\n",
      "M -0.1 -0.2 -0.2 -0.3 -0.2 -0.2 -0.3 -0.3 -0.2  0.3  0.3 -0.2  0.6  0.0 -0.2 -0.1  0.0 -0.3 -0.2  0.2\n",
      "F -0.3 -0.4 -0.3 -0.5  0.0 -0.4 -0.5 -0.5  0.0  0.0  0.2 -0.5  0.0  0.8 -0.3 -0.2 -0.2 -0.1  0.5  0.0\n",
      "P  0.1 -0.1 -0.1 -0.2 -0.2  0.0 -0.2 -0.1  0.0 -0.2  0.0 -0.2 -0.2 -0.3  0.6  0.1  0.1 -0.4 -0.3 -0.1\n",
      "S  0.1 -0.1  0.1  0.0  0.1 -0.1 -0.1  0.1 -0.1 -0.1 -0.2 -0.1 -0.1 -0.2  0.1  0.2  0.1 -0.3 -0.1 -0.1\n",
      "T  0.2 -0.1  0.1 -0.1 -0.1 -0.1 -0.1 -0.1 -0.1  0.1 -0.1 -0.1  0.0 -0.2  0.1  0.1  0.2 -0.4 -0.3  0.0\n",
      "W -0.4  0.0 -0.5 -0.5  0.1 -0.3 -0.5 -0.2 -0.3 -0.4 -0.2 -0.3 -0.3 -0.1 -0.4 -0.3 -0.4  1.5  0.0 -0.3\n",
      "Y -0.3 -0.2 -0.1 -0.2  0.2 -0.2 -0.4 -0.4  0.4 -0.2 -0.1 -0.3 -0.2  0.5 -0.3 -0.1 -0.3  0.0  0.9 -0.3\n",
      "V  0.1 -0.3 -0.2 -0.2 -0.2 -0.3 -0.2 -0.2 -0.3  0.4  0.2 -0.3  0.2  0.0 -0.1 -0.1  0.0 -0.3 -0.3  0.4\n",
      "\n",
      "LEVIN #  Jonathan M. Levin, Barry Robson, Jean Garnier:\n",
      "#  \"An algorithm for secondary structure determination in proteins based on\n",
      "#  sequence similarity.\"\n",
      "#  FEBS Letters 205(2): 303-308 (1986).\n",
      "#  Figure 1.\n",
      "#  PMID 3743779\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.0  0.0  0.0  1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0  0.0  0.0  1.0  0.0  0.0 -1.0 -1.0\n",
      "C  0.0  2.0  0.0  0.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0\n",
      "D  0.0  0.0  2.0  1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  1.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "E  1.0  0.0  1.0  2.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  1.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "F -1.0 -1.0 -1.0 -1.0  2.0 -1.0 -1.0  1.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  1.0\n",
      "G  0.0  0.0  0.0  0.0 -1.0  2.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "H  0.0  0.0  0.0  0.0 -1.0  0.0  2.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0 -1.0 -1.0  0.0\n",
      "I  0.0  0.0 -1.0 -1.0  1.0 -1.0 -1.0  2.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  1.0  0.0  0.0\n",
      "K  0.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  2.0 -1.0 -1.0  1.0  0.0  0.0  1.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "L  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  2.0  2.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  1.0  0.0  0.0\n",
      "M  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  2.0  2.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  0.0  0.0\n",
      "N  0.0  0.0  1.0  0.0 -1.0  0.0  0.0 -1.0  1.0 -1.0 -1.0  3.0  0.0  1.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "P -1.0  0.0  0.0 -1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  3.0  0.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "Q  0.0  0.0  0.0  1.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  1.0  0.0  2.0  0.0  0.0  0.0 -1.0 -1.0 -1.0\n",
      "R  0.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  1.0 -1.0 -1.0  0.0  0.0  0.0  2.0  0.0  0.0 -1.0  0.0 -1.0\n",
      "S  1.0  0.0  0.0  0.0 -1.0  0.0  0.0 -1.0  0.0 -1.0 -1.0  0.0  0.0  0.0  0.0  2.0  0.0 -1.0 -1.0 -1.0\n",
      "T  0.0  0.0  0.0  0.0 -1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  2.0  0.0 -1.0 -1.0\n",
      "V  0.0  0.0 -1.0 -1.0  0.0 -1.0 -1.0  1.0 -1.0  1.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  2.0  0.0  0.0\n",
      "W -1.0 -1.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0  0.0 -1.0 -1.0  0.0  2.0  0.0\n",
      "Y -1.0 -1.0 -1.0 -1.0  1.0 -1.0  0.0  0.0 -1.0  0.0  0.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0  0.0  0.0  2.0\n",
      "\n",
      "MCLACHLAN #  A.D. McLachlan:\n",
      "#  \"Tests for comparing related amino-acid sequences. Cytochrome c and\n",
      "#  cytochrome c 551.\"\n",
      "#  Journal of Molecular Biology 61(2): 409-424 (1971).\n",
      "#  Figure 1.\n",
      "#  PMID 5167087\n",
      "    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y\n",
      "A 8.0 1.0 3.0 4.0 1.0 3.0 3.0 2.0 3.0 2.0 3.0 3.0 4.0 3.0 2.0 4.0 3.0 3.0 1.0 1.0\n",
      "C 1.0 9.0 1.0 0.0 0.0 1.0 3.0 1.0 0.0 0.0 3.0 1.0 0.0 0.0 1.0 2.0 2.0 1.0 2.0 1.0\n",
      "D 3.0 1.0 8.0 5.0 1.0 3.0 4.0 1.0 3.0 1.0 2.0 5.0 3.0 4.0 1.0 3.0 3.0 1.0 0.0 1.0\n",
      "E 4.0 0.0 5.0 8.0 0.0 3.0 2.0 1.0 4.0 1.0 1.0 4.0 4.0 5.0 3.0 4.0 4.0 2.0 1.0 2.0\n",
      "F 1.0 0.0 1.0 0.0 9.0 0.0 4.0 3.0 0.0 5.0 5.0 0.0 1.0 0.0 1.0 2.0 1.0 3.0 6.0 6.0\n",
      "G 3.0 1.0 3.0 3.0 0.0 8.0 2.0 1.0 3.0 1.0 1.0 3.0 3.0 2.0 3.0 3.0 2.0 2.0 1.0 0.0\n",
      "H 3.0 3.0 4.0 2.0 4.0 2.0 8.0 2.0 4.0 2.0 3.0 4.0 3.0 4.0 5.0 3.0 4.0 2.0 3.0 4.0\n",
      "I 2.0 1.0 1.0 1.0 3.0 1.0 2.0 8.0 1.0 5.0 5.0 1.0 1.0 0.0 1.0 2.0 3.0 5.0 3.0 3.0\n",
      "K 3.0 0.0 3.0 4.0 0.0 3.0 4.0 1.0 8.0 2.0 1.0 4.0 3.0 4.0 5.0 3.0 3.0 2.0 1.0 1.0\n",
      "L 2.0 0.0 1.0 1.0 5.0 1.0 2.0 5.0 2.0 8.0 6.0 1.0 1.0 3.0 2.0 2.0 3.0 5.0 3.0 3.0\n",
      "M 3.0 3.0 2.0 1.0 5.0 1.0 3.0 5.0 1.0 6.0 8.0 2.0 1.0 3.0 1.0 2.0 3.0 4.0 1.0 2.0\n",
      "N 3.0 1.0 5.0 4.0 0.0 3.0 4.0 1.0 4.0 1.0 2.0 8.0 1.0 4.0 3.0 5.0 3.0 1.0 0.0 2.0\n",
      "P 4.0 0.0 3.0 4.0 1.0 3.0 3.0 1.0 3.0 1.0 1.0 1.0 8.0 3.0 3.0 3.0 3.0 2.0 0.0 0.0\n",
      "Q 3.0 0.0 4.0 5.0 0.0 2.0 4.0 0.0 4.0 3.0 3.0 4.0 3.0 8.0 5.0 4.0 3.0 2.0 2.0 1.0\n",
      "R 2.0 1.0 1.0 3.0 1.0 3.0 5.0 1.0 5.0 2.0 1.0 3.0 3.0 5.0 8.0 4.0 3.0 2.0 3.0 2.0\n",
      "S 4.0 2.0 3.0 4.0 2.0 3.0 3.0 2.0 3.0 2.0 2.0 5.0 3.0 4.0 4.0 8.0 5.0 2.0 3.0 3.0\n",
      "T 3.0 2.0 3.0 4.0 1.0 2.0 4.0 3.0 3.0 3.0 3.0 3.0 3.0 3.0 3.0 5.0 8.0 3.0 2.0 1.0\n",
      "V 3.0 1.0 1.0 2.0 3.0 2.0 2.0 5.0 2.0 5.0 4.0 1.0 2.0 2.0 2.0 2.0 3.0 8.0 2.0 3.0\n",
      "W 1.0 2.0 0.0 1.0 6.0 1.0 3.0 3.0 1.0 3.0 1.0 0.0 0.0 2.0 3.0 3.0 2.0 2.0 9.0 6.0\n",
      "Y 1.0 1.0 1.0 2.0 6.0 0.0 4.0 3.0 1.0 3.0 2.0 2.0 0.0 1.0 2.0 3.0 1.0 3.0 6.0 9.0\n",
      "\n",
      "MDM78 #  R.M. Schwartz and M.O. Dayhoff:\n",
      "#  \"Matrices for Detecting Distant Relationships.\"\n",
      "#  Margaret O. Dayhoff: Atlas of Protein Sequence and Structure,\n",
      "#  Volume 5, Supplement 3, 1978, pages 353-358.\n",
      "#  The National Biomedical Research Foundation, 1979.\n",
      "#  Figure 85, page 354.\n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V\n",
      "A  0.2 -0.1  0.0  0.0 -0.2 -0.0  0.0  0.1 -0.1 -0.1 -0.2 -0.1 -0.1 -0.3  0.1  0.1  0.1 -0.6 -0.3  0.0\n",
      "R -0.1  0.6  0.0 -0.1 -0.4  0.1 -0.1 -0.3  0.2 -0.2 -0.3  0.3 -0.0 -0.5 -0.0 -0.0 -0.1  0.2 -0.4 -0.2\n",
      "N  0.0  0.0  0.2  0.2 -0.4  0.1  0.1  0.0  0.2 -0.2 -0.3  0.1 -0.2 -0.3 -0.1  0.1  0.0 -0.4 -0.2 -0.2\n",
      "D  0.0 -0.1  0.2  0.4 -0.5  0.2  0.3  0.1  0.1 -0.2 -0.4  0.0 -0.3 -0.6 -0.1  0.0 -0.0 -0.7 -0.4 -0.2\n",
      "C -0.2 -0.4 -0.4 -0.5  1.2 -0.5 -0.5 -0.3 -0.3 -0.2 -0.6 -0.5 -0.5 -0.4 -0.3  0.0 -0.2 -0.8  0.0 -0.2\n",
      "Q -0.0  0.1  0.1  0.2 -0.5  0.4  0.2 -0.1  0.3 -0.2 -0.2  0.1 -0.1 -0.5  0.0 -0.1 -0.1 -0.5 -0.4 -0.2\n",
      "E  0.0 -0.1  0.1  0.3 -0.5  0.2  0.4  0.0  0.1 -0.2 -0.3 -0.0 -0.2 -0.5 -0.1  0.0 -0.0 -0.7 -0.4 -0.2\n",
      "G  0.1 -0.3  0.0  0.1 -0.3 -0.1  0.0  0.5 -0.2 -0.3 -0.4 -0.2 -0.3 -0.5 -0.1  0.1  0.0 -0.7 -0.5 -0.1\n",
      "H -0.1  0.2  0.2  0.1 -0.3  0.3  0.1 -0.2  0.7 -0.2 -0.2  0.0 -0.2 -0.2 -0.0 -0.1 -0.1 -0.3 -0.0 -0.2\n",
      "I -0.1 -0.2 -0.2 -0.2 -0.2 -0.2 -0.2 -0.3 -0.2  0.5  0.2 -0.2  0.2  0.1 -0.2 -0.1  0.0 -0.5 -0.1  0.4\n",
      "L -0.2 -0.3 -0.3 -0.4 -0.6 -0.2 -0.3 -0.4 -0.2  0.2  0.6 -0.3  0.4  0.2 -0.2 -0.3 -0.2 -0.2 -0.1  0.2\n",
      "K -0.1  0.3  0.1  0.0 -0.5  0.1 -0.0 -0.2  0.0 -0.2 -0.3  0.5  0.0 -0.5 -0.1 -0.0  0.0 -0.3 -0.4 -0.2\n",
      "M -0.1 -0.0 -0.2 -0.3 -0.5 -0.1 -0.2 -0.3 -0.2  0.2  0.4  0.0  0.6  0.0 -0.2 -0.2 -0.1 -0.4 -0.2  0.2\n",
      "F -0.3 -0.5 -0.3 -0.6 -0.4 -0.5 -0.5 -0.5 -0.2  0.1  0.2 -0.5  0.0  0.9 -0.5 -0.3 -0.3  0.0  0.7 -0.1\n",
      "P  0.1 -0.0 -0.1 -0.1 -0.3  0.0 -0.1 -0.1 -0.0 -0.2 -0.2 -0.1 -0.2 -0.5  0.6  0.1  0.0 -0.6 -0.5 -0.1\n",
      "S  0.1 -0.0  0.1  0.0  0.0 -0.1  0.0  0.1 -0.1 -0.1 -0.3 -0.0 -0.2 -0.3  0.1  0.2  0.1 -0.2 -0.3 -0.1\n",
      "T  0.1 -0.1  0.0 -0.0 -0.2 -0.1 -0.0  0.0 -0.1  0.0 -0.2  0.0 -0.1 -0.3  0.0  0.1  0.3 -0.5 -0.3  0.0\n",
      "W -0.6  0.2 -0.4 -0.7 -0.8 -0.5 -0.7 -0.7 -0.3 -0.5 -0.2 -0.3 -0.4  0.0 -0.6 -0.2 -0.5  1.7 -0.0 -0.6\n",
      "Y -0.3 -0.4 -0.2 -0.4  0.0 -0.4 -0.4 -0.5 -0.0 -0.1 -0.1 -0.4 -0.2  0.7 -0.5 -0.3 -0.3 -0.0  1.0 -0.2\n",
      "V  0.0 -0.2 -0.2 -0.2 -0.2 -0.2 -0.2 -0.1 -0.2  0.4  0.2 -0.2  0.2 -0.1 -0.1 -0.1  0.0 -0.6 -0.2  0.4\n",
      "\n",
      "NUC.4.4 #  \n",
      "#  This matrix was created by Todd Lowe   12/10/92\n",
      "#  \n",
      "#  Uses ambiguous nucleotide codes, probabilities rounded to\n",
      "#  nearest integer\n",
      "#  \n",
      "#  Lowest score = -4, Highest score = 5\n",
      "#  \n",
      "     A    T    G    C    S    W    R    Y    K    M    B    V    H    D    N\n",
      "A  5.0 -4.0 -4.0 -4.0 -4.0  1.0  1.0 -4.0 -4.0  1.0 -4.0 -1.0 -1.0 -1.0 -2.0\n",
      "T -4.0  5.0 -4.0 -4.0 -4.0  1.0 -4.0  1.0  1.0 -4.0 -1.0 -4.0 -1.0 -1.0 -2.0\n",
      "G -4.0 -4.0  5.0 -4.0  1.0 -4.0  1.0 -4.0  1.0 -4.0 -1.0 -1.0 -4.0 -1.0 -2.0\n",
      "C -4.0 -4.0 -4.0  5.0  1.0 -4.0 -4.0  1.0 -4.0  1.0 -1.0 -1.0 -1.0 -4.0 -2.0\n",
      "S -4.0 -4.0  1.0  1.0 -1.0 -4.0 -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -3.0 -3.0 -1.0\n",
      "W  1.0  1.0 -4.0 -4.0 -4.0 -1.0 -2.0 -2.0 -2.0 -2.0 -3.0 -3.0 -1.0 -1.0 -1.0\n",
      "R  1.0 -4.0  1.0 -4.0 -2.0 -2.0 -1.0 -4.0 -2.0 -2.0 -3.0 -1.0 -3.0 -1.0 -1.0\n",
      "Y -4.0  1.0 -4.0  1.0 -2.0 -2.0 -4.0 -1.0 -2.0 -2.0 -1.0 -3.0 -1.0 -3.0 -1.0\n",
      "K -4.0  1.0  1.0 -4.0 -2.0 -2.0 -2.0 -2.0 -1.0 -4.0 -1.0 -3.0 -3.0 -1.0 -1.0\n",
      "M  1.0 -4.0 -4.0  1.0 -2.0 -2.0 -2.0 -2.0 -4.0 -1.0 -3.0 -1.0 -1.0 -3.0 -1.0\n",
      "B -4.0 -1.0 -1.0 -1.0 -1.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -2.0 -2.0 -2.0 -1.0\n",
      "V -1.0 -4.0 -1.0 -1.0 -1.0 -3.0 -1.0 -3.0 -3.0 -1.0 -2.0 -1.0 -2.0 -2.0 -1.0\n",
      "H -1.0 -1.0 -4.0 -1.0 -3.0 -1.0 -3.0 -1.0 -3.0 -1.0 -2.0 -2.0 -1.0 -2.0 -1.0\n",
      "D -1.0 -1.0 -1.0 -4.0 -3.0 -1.0 -1.0 -3.0 -1.0 -3.0 -2.0 -2.0 -2.0 -1.0 -1.0\n",
      "N -2.0 -2.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0\n",
      "\n",
      "PAM250 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 250 substitution matrix, scale = ln(2)/3 = 0.231049\n",
      "#  \n",
      "#  Expected score = -0.844, Entropy = 0.354 bits\n",
      "#  \n",
      "#  Lowest score = -8, Highest score = 17\n",
      "#  \n",
      "     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *\n",
      "A  2.0 -2.0  0.0  0.0 -2.0  0.0  0.0  1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -3.0  1.0  1.0  1.0 -6.0 -3.0  0.0  0.0  0.0  0.0 -8.0\n",
      "R -2.0  6.0  0.0 -1.0 -4.0  1.0 -1.0 -3.0  2.0 -2.0 -3.0  3.0  0.0 -4.0  0.0  0.0 -1.0  2.0 -4.0 -2.0 -1.0  0.0 -1.0 -8.0\n",
      "N  0.0  0.0  2.0  2.0 -4.0  1.0  1.0  0.0  2.0 -2.0 -3.0  1.0 -2.0 -3.0  0.0  1.0  0.0 -4.0 -2.0 -2.0  2.0  1.0  0.0 -8.0\n",
      "D  0.0 -1.0  2.0  4.0 -5.0  2.0  3.0  1.0  1.0 -2.0 -4.0  0.0 -3.0 -6.0 -1.0  0.0  0.0 -7.0 -4.0 -2.0  3.0  3.0 -1.0 -8.0\n",
      "C -2.0 -4.0 -4.0 -5.0 12.0 -5.0 -5.0 -3.0 -3.0 -2.0 -6.0 -5.0 -5.0 -4.0 -3.0  0.0 -2.0 -8.0  0.0 -2.0 -4.0 -5.0 -3.0 -8.0\n",
      "Q  0.0  1.0  1.0  2.0 -5.0  4.0  2.0 -1.0  3.0 -2.0 -2.0  1.0 -1.0 -5.0  0.0 -1.0 -1.0 -5.0 -4.0 -2.0  1.0  3.0 -1.0 -8.0\n",
      "E  0.0 -1.0  1.0  3.0 -5.0  2.0  4.0  0.0  1.0 -2.0 -3.0  0.0 -2.0 -5.0 -1.0  0.0  0.0 -7.0 -4.0 -2.0  3.0  3.0 -1.0 -8.0\n",
      "G  1.0 -3.0  0.0  1.0 -3.0 -1.0  0.0  5.0 -2.0 -3.0 -4.0 -2.0 -3.0 -5.0  0.0  1.0  0.0 -7.0 -5.0 -1.0  0.0  0.0 -1.0 -8.0\n",
      "H -1.0  2.0  2.0  1.0 -3.0  3.0  1.0 -2.0  6.0 -2.0 -2.0  0.0 -2.0 -2.0  0.0 -1.0 -1.0 -3.0  0.0 -2.0  1.0  2.0 -1.0 -8.0\n",
      "I -1.0 -2.0 -2.0 -2.0 -2.0 -2.0 -2.0 -3.0 -2.0  5.0  2.0 -2.0  2.0  1.0 -2.0 -1.0  0.0 -5.0 -1.0  4.0 -2.0 -2.0 -1.0 -8.0\n",
      "L -2.0 -3.0 -3.0 -4.0 -6.0 -2.0 -3.0 -4.0 -2.0  2.0  6.0 -3.0  4.0  2.0 -3.0 -3.0 -2.0 -2.0 -1.0  2.0 -3.0 -3.0 -1.0 -8.0\n",
      "K -1.0  3.0  1.0  0.0 -5.0  1.0  0.0 -2.0  0.0 -2.0 -3.0  5.0  0.0 -5.0 -1.0  0.0  0.0 -3.0 -4.0 -2.0  1.0  0.0 -1.0 -8.0\n",
      "M -1.0  0.0 -2.0 -3.0 -5.0 -1.0 -2.0 -3.0 -2.0  2.0  4.0  0.0  6.0  0.0 -2.0 -2.0 -1.0 -4.0 -2.0  2.0 -2.0 -2.0 -1.0 -8.0\n",
      "F -3.0 -4.0 -3.0 -6.0 -4.0 -5.0 -5.0 -5.0 -2.0  1.0  2.0 -5.0  0.0  9.0 -5.0 -3.0 -3.0  0.0  7.0 -1.0 -4.0 -5.0 -2.0 -8.0\n",
      "P  1.0  0.0  0.0 -1.0 -3.0  0.0 -1.0  0.0  0.0 -2.0 -3.0 -1.0 -2.0 -5.0  6.0  1.0  0.0 -6.0 -5.0 -1.0 -1.0  0.0 -1.0 -8.0\n",
      "S  1.0  0.0  1.0  0.0  0.0 -1.0  0.0  1.0 -1.0 -1.0 -3.0  0.0 -2.0 -3.0  1.0  2.0  1.0 -2.0 -3.0 -1.0  0.0  0.0  0.0 -8.0\n",
      "T  1.0 -1.0  0.0  0.0 -2.0 -1.0  0.0  0.0 -1.0  0.0 -2.0  0.0 -1.0 -3.0  0.0  1.0  3.0 -5.0 -3.0  0.0  0.0 -1.0  0.0 -8.0\n",
      "W -6.0  2.0 -4.0 -7.0 -8.0 -5.0 -7.0 -7.0 -3.0 -5.0 -2.0 -3.0 -4.0  0.0 -6.0 -2.0 -5.0 17.0  0.0 -6.0 -5.0 -6.0 -4.0 -8.0\n",
      "Y -3.0 -4.0 -2.0 -4.0  0.0 -4.0 -4.0 -5.0  0.0 -1.0 -1.0 -4.0 -2.0  7.0 -5.0 -3.0 -3.0  0.0 10.0 -2.0 -3.0 -4.0 -2.0 -8.0\n",
      "V  0.0 -2.0 -2.0 -2.0 -2.0 -2.0 -2.0 -1.0 -2.0  4.0  2.0 -2.0  2.0 -1.0 -1.0 -1.0  0.0 -6.0 -2.0  4.0 -2.0 -2.0 -1.0 -8.0\n",
      "B  0.0 -1.0  2.0  3.0 -4.0  1.0  3.0  0.0  1.0 -2.0 -3.0  1.0 -2.0 -4.0 -1.0  0.0  0.0 -5.0 -3.0 -2.0  3.0  2.0 -1.0 -8.0\n",
      "Z  0.0  0.0  1.0  3.0 -5.0  3.0  3.0  0.0  2.0 -2.0 -3.0  0.0 -2.0 -5.0  0.0  0.0 -1.0 -6.0 -4.0 -2.0  2.0  3.0 -1.0 -8.0\n",
      "X  0.0 -1.0  0.0 -1.0 -3.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  0.0  0.0 -4.0 -2.0 -1.0 -1.0 -1.0 -1.0 -8.0\n",
      "* -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0 -8.0  1.0\n",
      "\n",
      "PAM30 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 30 substitution matrix, scale = ln(2)/2 = 0.346574\n",
      "#  \n",
      "#  Expected score = -5.06, Entropy = 2.57 bits\n",
      "#  \n",
      "#  Lowest score = -17, Highest score = 13\n",
      "#  \n",
      "      A     R     N     D     C     Q     E     G     H     I     L     K     M     F     P     S     T     W     Y     V     B     Z     X     *\n",
      "A   6.0  -7.0  -4.0  -3.0  -6.0  -4.0  -2.0  -2.0  -7.0  -5.0  -6.0  -7.0  -5.0  -8.0  -2.0   0.0  -1.0 -13.0  -8.0  -2.0  -3.0  -3.0  -3.0 -17.0\n",
      "R  -7.0   8.0  -6.0 -10.0  -8.0  -2.0  -9.0  -9.0  -2.0  -5.0  -8.0   0.0  -4.0  -9.0  -4.0  -3.0  -6.0  -2.0 -10.0  -8.0  -7.0  -4.0  -6.0 -17.0\n",
      "N  -4.0  -6.0   8.0   2.0 -11.0  -3.0  -2.0  -3.0   0.0  -5.0  -7.0  -1.0  -9.0  -9.0  -6.0   0.0  -2.0  -8.0  -4.0  -8.0   6.0  -3.0  -3.0 -17.0\n",
      "D  -3.0 -10.0   2.0   8.0 -14.0  -2.0   2.0  -3.0  -4.0  -7.0 -12.0  -4.0 -11.0 -15.0  -8.0  -4.0  -5.0 -15.0 -11.0  -8.0   6.0   1.0  -5.0 -17.0\n",
      "C  -6.0  -8.0 -11.0 -14.0  10.0 -14.0 -14.0  -9.0  -7.0  -6.0 -15.0 -14.0 -13.0 -13.0  -8.0  -3.0  -8.0 -15.0  -4.0  -6.0 -12.0 -14.0  -9.0 -17.0\n",
      "Q  -4.0  -2.0  -3.0  -2.0 -14.0   8.0   1.0  -7.0   1.0  -8.0  -5.0  -3.0  -4.0 -13.0  -3.0  -5.0  -5.0 -13.0 -12.0  -7.0  -3.0   6.0  -5.0 -17.0\n",
      "E  -2.0  -9.0  -2.0   2.0 -14.0   1.0   8.0  -4.0  -5.0  -5.0  -9.0  -4.0  -7.0 -14.0  -5.0  -4.0  -6.0 -17.0  -8.0  -6.0   1.0   6.0  -5.0 -17.0\n",
      "G  -2.0  -9.0  -3.0  -3.0  -9.0  -7.0  -4.0   6.0  -9.0 -11.0 -10.0  -7.0  -8.0  -9.0  -6.0  -2.0  -6.0 -15.0 -14.0  -5.0  -3.0  -5.0  -5.0 -17.0\n",
      "H  -7.0  -2.0   0.0  -4.0  -7.0   1.0  -5.0  -9.0   9.0  -9.0  -6.0  -6.0 -10.0  -6.0  -4.0  -6.0  -7.0  -7.0  -3.0  -6.0  -1.0  -1.0  -5.0 -17.0\n",
      "I  -5.0  -5.0  -5.0  -7.0  -6.0  -8.0  -5.0 -11.0  -9.0   8.0  -1.0  -6.0  -1.0  -2.0  -8.0  -7.0  -2.0 -14.0  -6.0   2.0  -6.0  -6.0  -5.0 -17.0\n",
      "L  -6.0  -8.0  -7.0 -12.0 -15.0  -5.0  -9.0 -10.0  -6.0  -1.0   7.0  -8.0   1.0  -3.0  -7.0  -8.0  -7.0  -6.0  -7.0  -2.0  -9.0  -7.0  -6.0 -17.0\n",
      "K  -7.0   0.0  -1.0  -4.0 -14.0  -3.0  -4.0  -7.0  -6.0  -6.0  -8.0   7.0  -2.0 -14.0  -6.0  -4.0  -3.0 -12.0  -9.0  -9.0  -2.0  -4.0  -5.0 -17.0\n",
      "M  -5.0  -4.0  -9.0 -11.0 -13.0  -4.0  -7.0  -8.0 -10.0  -1.0   1.0  -2.0  11.0  -4.0  -8.0  -5.0  -4.0 -13.0 -11.0  -1.0 -10.0  -5.0  -5.0 -17.0\n",
      "F  -8.0  -9.0  -9.0 -15.0 -13.0 -13.0 -14.0  -9.0  -6.0  -2.0  -3.0 -14.0  -4.0   9.0 -10.0  -6.0  -9.0  -4.0   2.0  -8.0 -10.0 -13.0  -8.0 -17.0\n",
      "P  -2.0  -4.0  -6.0  -8.0  -8.0  -3.0  -5.0  -6.0  -4.0  -8.0  -7.0  -6.0  -8.0 -10.0   8.0  -2.0  -4.0 -14.0 -13.0  -6.0  -7.0  -4.0  -5.0 -17.0\n",
      "S   0.0  -3.0   0.0  -4.0  -3.0  -5.0  -4.0  -2.0  -6.0  -7.0  -8.0  -4.0  -5.0  -6.0  -2.0   6.0   0.0  -5.0  -7.0  -6.0  -1.0  -5.0  -3.0 -17.0\n",
      "T  -1.0  -6.0  -2.0  -5.0  -8.0  -5.0  -6.0  -6.0  -7.0  -2.0  -7.0  -3.0  -4.0  -9.0  -4.0   0.0   7.0 -13.0  -6.0  -3.0  -3.0  -6.0  -4.0 -17.0\n",
      "W -13.0  -2.0  -8.0 -15.0 -15.0 -13.0 -17.0 -15.0  -7.0 -14.0  -6.0 -12.0 -13.0  -4.0 -14.0  -5.0 -13.0  13.0  -5.0 -15.0 -10.0 -14.0 -11.0 -17.0\n",
      "Y  -8.0 -10.0  -4.0 -11.0  -4.0 -12.0  -8.0 -14.0  -3.0  -6.0  -7.0  -9.0 -11.0   2.0 -13.0  -7.0  -6.0  -5.0  10.0  -7.0  -6.0  -9.0  -7.0 -17.0\n",
      "V  -2.0  -8.0  -8.0  -8.0  -6.0  -7.0  -6.0  -5.0  -6.0   2.0  -2.0  -9.0  -1.0  -8.0  -6.0  -6.0  -3.0 -15.0  -7.0   7.0  -8.0  -6.0  -5.0 -17.0\n",
      "B  -3.0  -7.0   6.0   6.0 -12.0  -3.0   1.0  -3.0  -1.0  -6.0  -9.0  -2.0 -10.0 -10.0  -7.0  -1.0  -3.0 -10.0  -6.0  -8.0   6.0   0.0  -5.0 -17.0\n",
      "Z  -3.0  -4.0  -3.0   1.0 -14.0   6.0   6.0  -5.0  -1.0  -6.0  -7.0  -4.0  -5.0 -13.0  -4.0  -5.0  -6.0 -14.0  -9.0  -6.0   0.0   6.0  -5.0 -17.0\n",
      "X  -3.0  -6.0  -3.0  -5.0  -9.0  -5.0  -5.0  -5.0  -5.0  -5.0  -6.0  -5.0  -5.0  -8.0  -5.0  -3.0  -4.0 -11.0  -7.0  -5.0  -5.0  -5.0  -5.0 -17.0\n",
      "* -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0 -17.0   1.0\n",
      "\n",
      "PAM70 #  \n",
      "#  This matrix was produced by \"pam\" Version 1.0.6 [28-Jul-93]\n",
      "#  \n",
      "#  PAM 70 substitution matrix, scale = ln(2)/2 = 0.346574\n",
      "#  \n",
      "#  Expected score = -2.77, Entropy = 1.60 bits\n",
      "#  \n",
      "#  Lowest score = -11, Highest score = 13\n",
      "#  \n",
      "      A     R     N     D     C     Q     E     G     H     I     L     K     M     F     P     S     T     W     Y     V     B     Z     X     *\n",
      "A   5.0  -4.0  -2.0  -1.0  -4.0  -2.0  -1.0   0.0  -4.0  -2.0  -4.0  -4.0  -3.0  -6.0   0.0   1.0   1.0  -9.0  -5.0  -1.0  -1.0  -1.0  -2.0 -11.0\n",
      "R  -4.0   8.0  -3.0  -6.0  -5.0   0.0  -5.0  -6.0   0.0  -3.0  -6.0   2.0  -2.0  -7.0  -2.0  -1.0  -4.0   0.0  -7.0  -5.0  -4.0  -2.0  -3.0 -11.0\n",
      "N  -2.0  -3.0   6.0   3.0  -7.0  -1.0   0.0  -1.0   1.0  -3.0  -5.0   0.0  -5.0  -6.0  -3.0   1.0   0.0  -6.0  -3.0  -5.0   5.0  -1.0  -2.0 -11.0\n",
      "D  -1.0  -6.0   3.0   6.0  -9.0   0.0   3.0  -1.0  -1.0  -5.0  -8.0  -2.0  -7.0 -10.0  -4.0  -1.0  -2.0 -10.0  -7.0  -5.0   5.0   2.0  -3.0 -11.0\n",
      "C  -4.0  -5.0  -7.0  -9.0   9.0  -9.0  -9.0  -6.0  -5.0  -4.0 -10.0  -9.0  -9.0  -8.0  -5.0  -1.0  -5.0 -11.0  -2.0  -4.0  -8.0  -9.0  -6.0 -11.0\n",
      "Q  -2.0   0.0  -1.0   0.0  -9.0   7.0   2.0  -4.0   2.0  -5.0  -3.0  -1.0  -2.0  -9.0  -1.0  -3.0  -3.0  -8.0  -8.0  -4.0  -1.0   5.0  -2.0 -11.0\n",
      "E  -1.0  -5.0   0.0   3.0  -9.0   2.0   6.0  -2.0  -2.0  -4.0  -6.0  -2.0  -4.0  -9.0  -3.0  -2.0  -3.0 -11.0  -6.0  -4.0   2.0   5.0  -3.0 -11.0\n",
      "G   0.0  -6.0  -1.0  -1.0  -6.0  -4.0  -2.0   6.0  -6.0  -6.0  -7.0  -5.0  -6.0  -7.0  -3.0   0.0  -3.0 -10.0  -9.0  -3.0  -1.0  -3.0  -3.0 -11.0\n",
      "H  -4.0   0.0   1.0  -1.0  -5.0   2.0  -2.0  -6.0   8.0  -6.0  -4.0  -3.0  -6.0  -4.0  -2.0  -3.0  -4.0  -5.0  -1.0  -4.0   0.0   1.0  -3.0 -11.0\n",
      "I  -2.0  -3.0  -3.0  -5.0  -4.0  -5.0  -4.0  -6.0  -6.0   7.0   1.0  -4.0   1.0   0.0  -5.0  -4.0  -1.0  -9.0  -4.0   3.0  -4.0  -4.0  -3.0 -11.0\n",
      "L  -4.0  -6.0  -5.0  -8.0 -10.0  -3.0  -6.0  -7.0  -4.0   1.0   6.0  -5.0   2.0  -1.0  -5.0  -6.0  -4.0  -4.0  -4.0   0.0  -6.0  -4.0  -4.0 -11.0\n",
      "K  -4.0   2.0   0.0  -2.0  -9.0  -1.0  -2.0  -5.0  -3.0  -4.0  -5.0   6.0   0.0  -9.0  -4.0  -2.0  -1.0  -7.0  -7.0  -6.0  -1.0  -2.0  -3.0 -11.0\n",
      "M  -3.0  -2.0  -5.0  -7.0  -9.0  -2.0  -4.0  -6.0  -6.0   1.0   2.0   0.0  10.0  -2.0  -5.0  -3.0  -2.0  -8.0  -7.0   0.0  -6.0  -3.0  -3.0 -11.0\n",
      "F  -6.0  -7.0  -6.0 -10.0  -8.0  -9.0  -9.0  -7.0  -4.0   0.0  -1.0  -9.0  -2.0   8.0  -7.0  -4.0  -6.0  -2.0   4.0  -5.0  -7.0  -9.0  -5.0 -11.0\n",
      "P   0.0  -2.0  -3.0  -4.0  -5.0  -1.0  -3.0  -3.0  -2.0  -5.0  -5.0  -4.0  -5.0  -7.0   7.0   0.0  -2.0  -9.0  -9.0  -3.0  -4.0  -2.0  -3.0 -11.0\n",
      "S   1.0  -1.0   1.0  -1.0  -1.0  -3.0  -2.0   0.0  -3.0  -4.0  -6.0  -2.0  -3.0  -4.0   0.0   5.0   2.0  -3.0  -5.0  -3.0   0.0  -2.0  -1.0 -11.0\n",
      "T   1.0  -4.0   0.0  -2.0  -5.0  -3.0  -3.0  -3.0  -4.0  -1.0  -4.0  -1.0  -2.0  -6.0  -2.0   2.0   6.0  -8.0  -4.0  -1.0  -1.0  -3.0  -2.0 -11.0\n",
      "W  -9.0   0.0  -6.0 -10.0 -11.0  -8.0 -11.0 -10.0  -5.0  -9.0  -4.0  -7.0  -8.0  -2.0  -9.0  -3.0  -8.0  13.0  -3.0 -10.0  -7.0 -10.0  -7.0 -11.0\n",
      "Y  -5.0  -7.0  -3.0  -7.0  -2.0  -8.0  -6.0  -9.0  -1.0  -4.0  -4.0  -7.0  -7.0   4.0  -9.0  -5.0  -4.0  -3.0   9.0  -5.0  -4.0  -7.0  -5.0 -11.0\n",
      "V  -1.0  -5.0  -5.0  -5.0  -4.0  -4.0  -4.0  -3.0  -4.0   3.0   0.0  -6.0   0.0  -5.0  -3.0  -3.0  -1.0 -10.0  -5.0   6.0  -5.0  -4.0  -2.0 -11.0\n",
      "B  -1.0  -4.0   5.0   5.0  -8.0  -1.0   2.0  -1.0   0.0  -4.0  -6.0  -1.0  -6.0  -7.0  -4.0   0.0  -1.0  -7.0  -4.0  -5.0   5.0   1.0  -2.0 -11.0\n",
      "Z  -1.0  -2.0  -1.0   2.0  -9.0   5.0   5.0  -3.0   1.0  -4.0  -4.0  -2.0  -3.0  -9.0  -2.0  -2.0  -3.0 -10.0  -7.0  -4.0   1.0   5.0  -3.0 -11.0\n",
      "X  -2.0  -3.0  -2.0  -3.0  -6.0  -2.0  -3.0  -3.0  -3.0  -3.0  -4.0  -3.0  -3.0  -5.0  -3.0  -1.0  -2.0  -7.0  -5.0  -2.0  -2.0  -3.0  -3.0 -11.0\n",
      "* -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0 -11.0   1.0\n",
      "\n",
      "RAO #  J.K. Mohana Rao:\n",
      "#  \"New scoring matrix for amino acid residue exchanges based on residue\n",
      "#  characteristic physical parameters.\"\n",
      "#  International Journal of Peptide and Protein Research: 29(2): 276-281 (1987).\n",
      "#  Figure 1, lower triangle.\n",
      "#  PMID 3570667\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A 16.0 11.0  9.0 10.0 10.0  8.0 11.0  9.0 10.0 11.0 11.0  9.0  6.0 11.0  8.0 10.0 10.0  9.0 11.0  9.0\n",
      "C 11.0 16.0  8.0  9.0 10.0  8.0 10.0  8.0  9.0 11.0 10.0  9.0  7.0 10.0  8.0 10.0 10.0  8.0 11.0 10.0\n",
      "D  9.0  8.0 16.0 11.0  4.0  9.0  9.0  3.0 11.0  6.0  5.0 11.0  8.0 11.0 10.0 10.0  9.0  3.0  6.0  7.0\n",
      "E 10.0  9.0 11.0 16.0  6.0  6.0 11.0  4.0 11.0  7.0  8.0 10.0  5.0 11.0  9.0  9.0  8.0  4.0  7.0  6.0\n",
      "F 10.0 10.0  4.0  6.0 16.0  7.0  9.0 12.0  6.0 11.0 10.0  6.0  4.0  7.0  5.0  8.0 10.0 11.0 11.0 10.0\n",
      "G  8.0  8.0  9.0  6.0  7.0 16.0  7.0  6.0  7.0  6.0  4.0 10.0 11.0  8.0  7.0 11.0 10.0  6.0  8.0 10.0\n",
      "H 11.0 10.0  9.0 11.0  9.0  7.0 16.0  8.0 11.0 10.0 10.0 10.0  5.0 11.0 10.0 10.0 10.0  9.0 10.0  9.0\n",
      "I  9.0  8.0  3.0  4.0 12.0  6.0  8.0 16.0  4.0 10.0  9.0  5.0  3.0  6.0  4.0  8.0 10.0 12.0 11.0 10.0\n",
      "K 10.0  9.0 11.0 11.0  6.0  7.0 11.0  4.0 16.0  7.0  8.0 11.0  6.0 12.0 11.0 10.0  9.0  5.0  7.0  7.0\n",
      "L 11.0 11.0  6.0  7.0 11.0  6.0 10.0 10.0  7.0 16.0 11.0  7.0  4.0  9.0  6.0  8.0  9.0 10.0 11.0  9.0\n",
      "M 11.0 10.0  5.0  8.0 10.0  4.0 10.0  9.0  8.0 11.0 16.0  6.0  2.0  9.0  6.0  7.0  8.0  9.0 10.0  8.0\n",
      "N  9.0  9.0 11.0 10.0  6.0 10.0 10.0  5.0 11.0  7.0  6.0 16.0  9.0 11.0 10.0 11.0 10.0  5.0  8.0  8.0\n",
      "P  6.0  7.0  8.0  5.0  4.0 11.0  5.0  3.0  6.0  4.0  2.0  9.0 16.0  7.0  6.0 10.0  8.0  3.0  6.0  8.0\n",
      "Q 11.0 10.0 11.0 11.0  7.0  8.0 11.0  6.0 12.0  9.0  9.0 11.0  7.0 16.0 10.0 10.0 10.0  6.0  9.0  8.0\n",
      "R  8.0  8.0 10.0  9.0  5.0  7.0 10.0  4.0 11.0  6.0  6.0 10.0  6.0 10.0 16.0  9.0  9.0  5.0  7.0  7.0\n",
      "S 10.0 10.0 10.0  9.0  8.0 11.0 10.0  8.0 10.0  8.0  7.0 11.0 10.0 10.0  9.0 16.0 11.0  8.0 10.0 11.0\n",
      "T 10.0 10.0  9.0  8.0 10.0 10.0 10.0 10.0  9.0  9.0  8.0 10.0  8.0 10.0  9.0 11.0 16.0 10.0 11.0 11.0\n",
      "V  9.0  8.0  3.0  4.0 11.0  6.0  9.0 12.0  5.0 10.0  9.0  5.0  3.0  6.0  5.0  8.0 10.0 16.0 11.0 10.0\n",
      "W 11.0 11.0  6.0  7.0 11.0  8.0 10.0 11.0  7.0 11.0 10.0  8.0  6.0  9.0  7.0 10.0 11.0 11.0 16.0 11.0\n",
      "Y  9.0 10.0  7.0  6.0 10.0 10.0  9.0 10.0  7.0  9.0  8.0  8.0  8.0  8.0  7.0 11.0 11.0 10.0 11.0 16.0\n",
      "\n",
      "RISLER #  J.L. Risler, M.O. Delorme, H. Delacroix, A. Henaut:\n",
      "#  \"Amino acid substitutions in structurally related proteins. A pattern\n",
      "#  recognition approach. Determination of a new and efficient scoring matrix.\"\n",
      "#  Journal of Molecular Biology 204(4): 1019-1029 (1988).\n",
      "#  Figure 5.\n",
      "#  PMID 3221397\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  2.2 -1.5  0.2  1.7  0.6  0.6 -0.6  1.7  1.4  1.3  1.0  1.3 -0.2  1.8  1.5  2.0  1.9  2.0 -0.9  0.2\n",
      "C -1.5  2.2 -1.7 -1.5 -1.6 -1.7 -1.8 -1.6 -1.6 -1.5 -1.6 -1.6 -1.8 -1.4 -1.5 -1.3 -1.4 -1.4 -1.8 -1.1\n",
      "D  0.2 -1.7  2.2  1.0 -0.3 -0.4 -1.3  0.0  0.1 -0.2 -0.5  0.8 -1.2  0.6 -0.1  0.7  0.0  0.0 -1.4 -0.4\n",
      "E  1.7 -1.5  1.0  2.2  0.6  0.3 -0.6  1.5  1.4  0.9  0.6  1.4 -0.1  2.1  1.9  1.8  1.6  1.6 -1.0  0.2\n",
      "F  0.6 -1.6 -0.3  0.6  2.2 -0.4 -1.1  1.0  0.1  1.0 -0.2  0.4 -1.1  0.7  0.4  0.5  0.3  0.8 -0.9  2.0\n",
      "G  0.6 -1.7 -0.4  0.3 -0.4  2.2 -1.2  0.0 -0.1 -0.2 -0.4  0.2 -1.2  0.2  0.1  0.7  0.2  0.1 -1.3 -0.2\n",
      "H -0.6 -1.8 -1.3 -0.6 -1.1 -1.2  2.2 -0.8 -1.0 -0.9 -1.2 -0.3 -1.6 -0.5 -0.4 -0.4 -0.9 -0.7 -1.7 -0.8\n",
      "I  1.7 -1.6  0.0  1.5  1.0  0.0 -0.8  2.2  1.0  2.1  0.9  0.9 -0.6  1.4  1.4  1.6  1.6  2.2 -0.7  0.4\n",
      "K  1.4 -1.6  0.1  1.4  0.1 -0.1 -1.0  1.0  2.2  0.7  0.4  1.0 -0.7  1.7  2.1  1.4  1.2  1.2 -1.1  0.5\n",
      "L  1.3 -1.5 -0.2  0.9  1.0 -0.2 -0.9  2.1  0.7  2.2  1.8  0.8 -0.8  1.1  1.2  1.3  1.2  2.0 -0.8  0.5\n",
      "M  1.0 -1.6 -0.5  0.6 -0.2 -0.4 -1.2  0.9  0.4  1.8  2.2  0.0 -1.2  1.2  1.1  0.6  0.8  0.8 -1.3 -0.2\n",
      "N  1.3 -1.6  0.8  1.4  0.4  0.2 -0.3  0.9  1.0  0.8  0.0  2.2 -1.0  1.6  1.2  1.9  1.1  1.1 -1.1 -0.1\n",
      "P -0.2 -1.8 -1.2 -0.1 -1.1 -1.2 -1.6 -0.6 -0.7 -0.8 -1.2 -1.0  2.2 -0.6 -0.3 -0.3 -0.5 -0.6 -1.6 -1.2\n",
      "Q  1.8 -1.4  0.6  2.1  0.7  0.2 -0.5  1.4  1.7  1.1  1.2  1.6 -0.6  2.2  2.0  1.8  1.7  1.5 -1.0  0.5\n",
      "R  1.5 -1.5 -0.1  1.9  0.4  0.1 -0.4  1.4  2.1  1.2  1.1  1.2 -0.3  2.0  2.2  2.0  1.9  1.5 -0.8  0.8\n",
      "S  2.0 -1.3  0.7  1.8  0.5  0.7 -0.4  1.6  1.4  1.3  0.6  1.9 -0.3  1.8  2.0  2.2  2.1  1.8 -0.8  0.4\n",
      "T  1.9 -1.4  0.0  1.6  0.3  0.2 -0.9  1.6  1.2  1.2  0.8  1.1 -0.5  1.7  1.9  2.1  2.2  1.6 -1.0  0.3\n",
      "V  2.0 -1.4  0.0  1.6  0.8  0.1 -0.7  2.2  1.2  2.0  0.8  1.1 -0.6  1.5  1.5  1.8  1.6  2.2 -0.7  0.3\n",
      "W -0.9 -1.8 -1.4 -1.0 -0.9 -1.3 -1.7 -0.7 -1.1 -0.8 -1.3 -1.1 -1.6 -1.0 -0.8 -0.8 -1.0 -0.7  2.2 -0.6\n",
      "Y  0.2 -1.1 -0.4  0.2  2.0 -0.2 -0.8  0.4  0.5  0.5 -0.2 -0.1 -1.2  0.5  0.8  0.4  0.3  0.3 -0.6  2.2\n",
      "\n",
      "SCHNEIDER #  Adrian Schneider, Gina M. Cannarozzi, and Gaston H. Gonnet:\n",
      "#  \"Empirical codon substitution matrix.\"\n",
      "#  BMC Bioinformatics 6:134 (2005).\n",
      "#  Additional File 3.\n",
      "#  PMID 15927081\n",
      "      AAA   AAC   AAG   AAT   ACA   ACC   ACG   ACT   AGA   AGC   AGG   AGT   ATA   ATC   ATG   ATT   CAA   CAC   CAG   CAT   CCA   CCC   CCG   CCT   CGA   CGC   CGG   CGT   CTA   CTC   CTG   CTT   GAA   GAC   GAG   GAT   GCA   GCC   GCG   GCT   GGA   GGC   GGG   GGT   GTA   GTC   GTG   GTT   TAA   TAC   TAG   TAT   TCA   TCC   TCG   TCT   TGA   TGC   TGG   TGT   TTA   TTC   TTG   TTT\n",
      "AAA  11.6  -2.7   9.7  -1.7  -2.7  -6.4  -3.9  -5.6   5.1  -5.0   3.6  -4.2  -6.3 -13.0  -7.1 -11.5   0.4  -6.0  -1.9  -5.3  -8.5 -11.2  -8.9 -10.8   2.1   0.0   1.4   0.2 -10.2 -13.5 -13.0 -12.5  -2.6  -8.5  -5.0  -8.1  -6.3  -9.9  -7.5  -9.0  -7.1 -10.2  -8.2  -9.2  -8.2 -12.5 -11.1 -11.4 -50.0 -14.8 -50.0 -13.8  -7.3 -10.1  -8.4  -9.1 -50.0 -13.0 -13.5 -12.4 -10.7 -18.1 -11.8 -17.2\n",
      "AAC  -2.7  13.0  -3.3  10.9  -3.5  -0.4  -3.3  -1.8  -5.4   4.6  -5.5   3.0 -10.2  -7.9  -9.9  -9.6  -5.0   0.5  -5.5  -1.0 -10.3  -8.1  -9.4  -9.6  -8.1  -5.0  -7.3  -6.3 -13.4 -11.3 -14.4 -12.9  -6.3   0.8  -6.4  -1.1  -7.4  -5.0  -6.2  -6.5  -5.6  -1.6  -4.7  -3.0 -10.8  -8.7 -11.9 -10.0 -50.0  -6.2 -50.0  -7.5  -6.3  -4.3  -6.2  -5.4 -50.0  -7.0 -16.3  -8.2 -13.2 -12.3 -13.1 -13.3\n",
      "AAG   9.7  -3.3  11.6  -2.8  -4.5  -6.7  -3.1  -6.9   3.3  -5.5   4.8  -5.1  -8.9 -13.2  -5.7 -12.6  -1.5  -6.1  -0.6  -6.1 -10.0 -11.8  -8.6 -11.9   1.2   0.5   2.2   0.1 -11.8 -14.0 -11.9 -13.4  -4.9  -9.1  -3.4  -8.9  -8.0 -10.0  -7.2 -10.0  -9.1 -10.2  -7.1  -9.9 -10.1 -13.0 -10.6 -12.7 -50.0 -14.9 -50.0 -14.4  -8.9 -10.8  -8.9 -10.4 -50.0 -13.1 -11.8 -13.0 -12.4 -19.4 -11.5 -17.8\n",
      "AAT  -1.7  10.9  -2.8  12.9  -2.7  -2.2  -2.8   0.2  -4.8   2.9  -5.2   5.2  -9.1  -9.5  -9.0  -7.0  -4.0  -1.0  -5.0   1.0  -9.0  -8.8  -8.9  -7.7  -7.0  -6.8  -7.1  -4.3 -12.4 -12.9 -13.9 -10.0  -4.9  -0.7  -5.6   1.4  -6.0  -6.5  -6.2  -4.8  -5.1  -3.2  -5.0  -1.2  -9.8  -9.9 -11.3  -8.2 -50.0  -7.6 -50.0  -5.1  -5.5  -5.8  -5.9  -4.2 -50.0  -8.3 -15.5  -5.9 -11.6 -14.2 -12.0 -11.6\n",
      "ACA  -2.7  -3.5  -4.5  -2.7  11.7   9.0  10.6   9.6  -3.2  -0.7  -5.1  -0.4   0.7  -4.6  -0.4  -3.6  -4.8  -8.9  -6.5  -7.8  -1.4  -4.1  -2.9  -3.3  -8.0  -9.1  -8.0  -8.2  -6.0  -9.3  -7.7  -8.5  -6.4 -10.1  -7.5  -8.6   2.8  -0.7   0.9   0.1  -5.9  -7.3  -6.5  -6.5  -0.4  -3.9  -2.6  -3.1 -50.0 -14.6 -50.0 -12.6   2.9   0.1   1.8   1.0 -50.0  -9.2 -13.6  -7.7  -4.8 -12.2  -6.2 -11.0\n",
      "ACC  -6.4  -0.4  -6.7  -2.2   9.0  12.3   9.8   9.6  -7.1   2.4  -7.2   0.3  -2.9  -1.2  -3.0  -3.3  -7.6  -6.7  -8.1  -7.9  -4.9  -1.9  -4.5  -3.8  -9.5  -7.8  -9.3  -8.9  -9.4  -7.1  -9.0  -9.0  -9.4  -7.5  -9.3  -9.2  -0.7   2.6   0.2  -0.1  -8.0  -4.6  -7.7  -6.1  -3.6  -1.0  -4.1  -3.1 -50.0 -11.6 -50.0 -11.6   0.2   2.3   0.4   0.5 -50.0  -6.5 -15.3  -7.8  -8.4  -9.5  -8.3 -11.1\n",
      "ACG  -3.9  -3.3  -3.1  -2.8  10.6   9.8  12.2   9.8  -4.9   0.1  -3.6  -0.2  -1.7  -4.1   0.9  -3.5  -5.9  -8.9  -5.6  -8.5  -3.9  -3.9  -2.2  -4.5  -7.3  -7.5  -6.4  -8.5  -7.1  -8.2  -6.7  -7.8  -7.9  -9.1  -7.1  -9.4   0.9  -0.2   2.5  -0.3  -6.9  -6.5  -5.5  -6.8  -1.7  -3.4  -1.7  -2.8 -50.0 -13.0 -50.0 -11.5   1.5   0.7   2.7   0.6 -50.0  -8.9 -12.1  -7.4  -6.1 -11.5  -5.4 -10.4\n",
      "ACT  -5.6  -1.8  -6.9   0.2   9.6   9.6   9.8  11.6  -6.6   0.9  -7.2   2.5  -2.3  -3.4  -2.3  -0.6  -6.5  -8.2  -8.0  -6.3  -3.7  -3.5  -3.9  -1.6  -9.9  -8.9  -9.6  -7.0  -8.6  -8.9  -8.8  -6.8  -8.6  -8.2  -8.8  -6.8   0.3   0.1   0.2   2.4  -7.4  -6.2  -6.9  -4.4  -2.9  -2.9  -3.5  -0.9 -50.0 -12.2 -50.0 -10.1   1.2   0.6   1.2   2.4 -50.0  -7.6 -16.1  -5.8  -7.2 -10.8  -7.4  -9.0\n",
      "AGA   5.1  -5.4   3.3  -4.8  -3.2  -7.1  -4.9  -6.6  13.3  -2.8  11.2  -1.9  -5.7 -12.5  -7.2 -11.6  -0.7  -4.8  -3.1  -4.3 -10.0 -11.6  -8.8 -11.7  10.5   7.7   9.1   8.5  -9.8 -12.7 -11.7 -11.8  -6.3 -11.4  -8.8 -11.0  -7.5 -10.3  -8.3  -9.9  -1.9  -6.7  -4.1  -6.4  -7.9 -12.5 -11.4 -11.7 -50.0 -14.0 -50.0 -13.2  -8.5 -10.4  -9.1  -9.6 -50.0  -9.8  -7.7  -8.8 -10.3 -17.9 -11.8 -16.0\n",
      "AGC  -5.0   4.6  -5.5   2.9  -0.7   2.4   0.1   0.9  -2.8  12.8  -2.6  11.0  -8.4  -6.3  -8.3  -7.7  -6.1  -2.9  -6.5  -4.2  -8.5  -5.9  -7.5  -7.7  -5.5  -2.4  -5.5  -4.1 -12.9 -10.3 -12.7 -11.3  -7.4  -3.1  -7.4  -4.9  -4.4  -1.8  -3.5  -3.5  -1.6   3.0  -0.8   0.9  -8.4  -6.3  -9.6  -7.3 -50.0  -9.0 -50.0  -9.7  -2.3  -0.2  -1.2  -1.5 -50.0  -0.7 -13.3  -2.0 -11.8 -11.9 -12.3 -12.7\n",
      "AGG   3.6  -5.5   4.8  -5.2  -5.1  -7.2  -3.6  -7.2  11.2  -2.6  13.4  -2.1  -7.7 -12.5  -5.3 -11.9  -2.2  -4.4  -1.9  -4.8 -10.2 -11.5  -8.1 -12.0   9.3   8.2  10.0   8.0 -10.8 -11.9 -10.7 -12.4  -8.2 -11.8  -6.9 -11.5  -8.4  -9.9  -7.1 -10.6  -4.7  -6.8  -1.2  -7.0  -9.8 -11.6  -9.6 -12.1 -50.0 -14.1 -50.0 -12.7 -10.0 -10.8  -8.8 -10.5 -50.0  -9.2  -4.2  -9.3 -11.0 -18.2 -11.1 -16.1\n",
      "AGT  -4.2   3.0  -5.1   5.2  -0.4   0.3  -0.2   2.5  -1.9  11.0  -2.1  13.2  -7.6  -8.5  -8.0  -5.3  -5.6  -4.5  -6.8  -2.4  -7.8  -7.9  -7.9  -6.2  -5.7  -5.1  -5.4  -2.1 -12.4 -12.3 -13.1 -10.1  -6.6  -4.4  -6.9  -2.6  -3.6  -3.8  -3.7  -2.0  -1.4   0.8  -0.9   3.0  -8.2  -8.1  -9.2  -5.8 -50.0 -10.1 -50.0  -7.4  -1.7  -2.0  -1.5  -0.5 -50.0  -2.1 -12.6  -0.4 -11.6 -13.9 -11.4 -11.1\n",
      "ATA  -6.3 -10.2  -8.9  -9.1   0.7  -2.9  -1.7  -2.3  -5.7  -8.4  -7.7  -7.6  13.2   9.6   3.5   9.7  -8.7 -12.4 -10.7 -10.7  -7.8 -10.2  -8.8  -9.6  -9.5 -11.9 -10.9 -10.5   2.3  -0.4  -0.2  -0.3  -9.7 -15.5 -11.4 -13.7  -3.0  -6.4  -4.1  -5.4  -9.6 -11.9 -10.5 -11.8   6.2   3.3   3.7   3.6 -50.0 -13.6 -50.0 -11.9  -5.6  -8.8  -7.2  -8.9 -50.0 -12.4 -14.1 -11.6   2.8  -6.4   0.5  -5.2\n",
      "ATC -13.0  -7.9 -13.2  -9.5  -4.6  -1.2  -4.1  -3.4 -12.5  -6.3 -12.5  -8.5   9.6  12.7   0.2  10.5 -12.4 -11.6 -13.2 -12.0 -12.4 -10.1 -11.6 -12.0 -15.1 -13.1 -14.3 -13.4  -1.4   1.4  -1.6  -0.5 -14.9 -14.3 -15.2 -16.8  -7.7  -4.7  -6.3  -6.6 -14.6 -10.8 -12.8 -13.0   2.3   6.0   2.5   3.6 -50.0 -11.2 -50.0 -11.9 -10.5  -9.2 -10.5 -10.5 -50.0 -10.3 -16.2 -11.3  -1.7  -3.7  -2.5  -5.4\n",
      "ATG  -7.1  -9.9  -5.7  -9.0  -0.4  -3.0   0.9  -2.3  -7.2  -8.3  -5.3  -8.0   3.5   0.2  14.3   1.0  -7.3 -11.2  -6.9  -9.7  -8.5 -10.2  -7.7  -9.8 -10.0 -10.4  -8.4 -10.2   1.0  -0.6   1.7  -0.5 -11.1 -14.8  -9.7 -14.5  -3.7  -5.8  -3.1  -5.6 -10.4 -11.5  -8.2 -11.1   0.6  -1.6   1.6  -1.1 -50.0 -12.4 -50.0 -11.5  -5.5  -8.0  -4.9  -7.2 -50.0 -12.4 -10.3 -11.6   0.7  -6.8   2.4  -6.0\n",
      "ATT -11.5  -9.6 -12.6  -7.0  -3.6  -3.3  -3.5  -0.6 -11.6  -7.7 -11.9  -5.3   9.7  10.5   1.0  12.6 -12.0 -12.2 -12.5  -9.7 -11.1 -11.2 -11.9  -9.4 -12.9 -14.3 -14.0 -11.9  -1.0  -0.4  -1.6   1.2 -13.9 -15.4 -13.7 -12.6  -6.2  -6.7  -6.3  -4.4 -13.0 -12.2 -13.0 -10.4   2.9   3.8   2.5   5.8 -50.0 -12.5 -50.0 -10.2  -9.6  -9.7  -9.6  -8.3 -50.0 -11.6 -15.4  -9.4  -1.1  -5.4  -1.6  -3.3\n",
      "CAA   0.4  -5.0  -1.5  -4.0  -4.8  -7.6  -5.9  -6.5  -0.7  -6.1  -2.2  -5.6  -8.7 -12.4  -7.3 -12.0  12.8   2.3  10.2   3.0   0.0  -3.4  -0.7  -3.2   2.5  -0.8   0.9   0.2  -3.0  -7.2  -5.7  -6.1  -0.2  -6.6  -1.8  -6.0  -5.3  -8.2  -5.7  -7.6  -6.7  -9.6  -7.5  -9.3  -7.1 -10.5  -9.6  -9.4 -50.0  -8.1 -50.0  -7.2  -4.3  -6.8  -5.7  -6.3 -50.0  -9.5  -9.6  -8.9  -6.2 -12.8  -6.8 -11.8\n",
      "CAC  -6.0   0.5  -6.1  -1.0  -8.9  -6.7  -8.9  -8.2  -4.8  -2.9  -4.4  -4.5 -12.4 -11.6 -11.2 -12.2   2.3  14.6   1.9  12.9  -5.5  -2.6  -4.1  -4.7  -1.2   3.0  -1.1   1.6  -8.0  -4.7  -8.7  -6.0  -8.2  -4.7  -7.8  -6.7 -10.0  -8.3  -8.9  -9.9 -10.8  -7.1 -10.0  -9.2 -11.6 -10.1 -12.3 -11.9 -50.0   2.2 -50.0   0.9  -7.9  -5.9  -7.4  -6.9 -50.0  -4.6 -11.6  -5.6  -9.7  -5.9  -9.9  -7.3\n",
      "CAG  -1.9  -5.5  -0.6  -5.0  -6.5  -8.1  -5.6  -8.0  -3.1  -6.5  -1.9  -6.8 -10.7 -13.2  -6.9 -12.5  10.2   1.9  11.9   2.1  -2.3  -4.1  -0.5  -4.6  -0.1  -0.9   2.4  -0.8  -5.1  -6.9  -4.8  -6.7  -2.3  -7.1  -0.8  -7.2  -7.2  -8.0  -5.6  -8.5  -9.4  -9.7  -7.2  -9.8  -9.2 -11.1  -9.5 -10.7 -50.0  -8.7 -50.0  -8.6  -6.1  -7.5  -5.5  -7.7 -50.0 -10.4  -7.7 -10.0  -7.3 -13.6  -6.4 -13.3\n",
      "CAT  -5.3  -1.0  -6.1   1.0  -7.8  -7.9  -8.5  -6.3  -4.3  -4.2  -4.8  -2.4 -10.7 -12.0  -9.7  -9.7   3.0  12.9   2.1  14.7  -4.9  -3.8  -4.0  -2.4  -1.0   0.8  -1.3   3.4  -7.6  -5.7  -8.1  -3.5  -6.8  -5.8  -7.2  -4.2  -8.9 -10.2  -8.4  -8.2  -9.9  -8.4  -9.7  -6.4 -11.1 -11.1 -11.8 -10.0 -50.0   0.4 -50.0   2.6  -7.2  -7.0  -7.4  -5.1 -50.0  -5.7  -9.8  -3.2  -8.4  -7.4  -8.9  -5.9\n",
      "CCA  -8.5 -10.3 -10.0  -9.0  -1.4  -4.9  -3.9  -3.7 -10.0  -8.5 -10.2  -7.8  -7.8 -12.4  -8.5 -11.1   0.0  -5.5  -2.3  -4.9  12.6  10.0  11.1  10.5  -6.0  -8.3  -6.9  -7.5  -2.2  -7.4  -5.3  -5.9  -8.9 -12.5  -9.4 -11.6  -0.7  -3.9  -2.3  -3.0  -9.1  -9.8  -9.2  -9.5  -5.6  -9.0  -8.0  -8.5 -50.0 -15.3 -50.0 -14.1   2.3  -1.4   0.3  -0.5 -50.0 -13.6 -14.3 -11.5  -5.3 -13.7  -6.1 -12.7\n",
      "CCC -11.2  -8.1 -11.8  -8.8  -4.1  -1.9  -3.9  -3.5 -11.6  -5.9 -11.5  -7.9 -10.2 -10.1 -10.2 -11.2  -3.4  -2.6  -4.1  -3.8  10.0  13.1  10.7  10.6  -8.1  -5.2  -8.0  -6.7  -6.1  -3.7  -7.5  -5.6 -11.1 -10.3 -10.6 -11.9  -3.2  -1.0  -2.2  -2.8 -10.5  -8.1  -9.4  -9.5  -8.3  -6.9  -9.4  -8.8 -50.0 -11.3 -50.0 -12.8  -0.6   2.2  -0.3   0.1 -50.0 -10.1 -17.5 -11.1  -8.6  -9.9  -8.6 -11.9\n",
      "CCG  -8.9  -9.4  -8.6  -8.9  -2.9  -4.5  -2.2  -3.9  -8.8  -7.5  -8.1  -7.9  -8.8 -11.6  -7.7 -11.9  -0.7  -4.1  -0.5  -4.0  11.1  10.7  13.2  10.4  -5.7  -5.8  -3.5  -6.1  -3.6  -6.5  -3.3  -5.5  -9.4 -11.0  -8.7 -11.2  -1.8  -2.7   0.7  -3.0  -9.1  -8.2  -7.3  -9.4  -7.2  -8.2  -7.2  -8.2 -50.0 -13.2 -50.0 -13.3   0.5  -0.9   1.8  -0.6 -50.0 -11.4 -11.1 -10.5  -6.1 -12.9  -5.0 -11.9\n",
      "CCT -10.8  -9.6 -11.9  -7.7  -3.3  -3.8  -4.5  -1.6 -11.7  -7.7 -12.0  -6.2  -9.6 -12.0  -9.8  -9.4  -3.2  -4.7  -4.6  -2.4  10.5  10.6  10.4  12.6  -8.2  -7.9  -9.0  -4.7  -6.1  -6.0  -7.5  -2.9 -10.9 -12.2 -11.2 -10.9  -2.4  -2.9  -2.7  -0.9 -10.4  -9.7 -10.2  -8.0  -8.4  -9.1  -9.3  -6.5 -50.0 -13.3 -50.0 -11.0  -0.0  -0.4  -0.4   2.3 -50.0 -10.9 -17.2  -8.3  -8.3 -12.7  -7.8  -9.3\n",
      "CGA   2.1  -8.1   1.2  -7.0  -8.0  -9.5  -7.3  -9.9  10.5  -5.5   9.3  -5.7  -9.5 -15.1 -10.0 -12.9   2.5  -1.2  -0.1  -1.0  -6.0  -8.1  -5.7  -8.2  13.8  11.3  11.8  12.1  -6.0  -9.1  -9.0  -9.5  -8.9 -13.4 -10.0 -13.1  -9.9 -11.7  -9.8 -12.8  -5.9  -9.4  -6.5  -8.1 -10.7 -14.0 -13.3 -13.9 -50.0 -11.9 -50.0  -9.7  -8.9 -10.8  -9.0 -11.1 -50.0  -6.5  -5.3  -6.3 -10.5 -16.2 -10.4 -15.2\n",
      "CGC   0.0  -5.0   0.5  -6.8  -9.1  -7.8  -7.5  -8.9   7.7  -2.4   8.2  -5.1 -11.9 -13.1 -10.4 -14.3  -0.8   3.0  -0.9   0.8  -8.3  -5.2  -5.8  -7.9  11.3  15.0  11.2  12.8  -9.7  -5.3  -9.2  -7.8 -11.7 -10.6 -10.1 -12.2 -11.7  -8.8  -8.9 -11.8  -9.3  -5.0  -8.0  -8.2 -13.6 -11.4 -12.8 -13.0 -50.0  -7.2 -50.0  -8.4 -11.0  -7.7  -8.7 -10.2 -50.0  -1.5  -7.1  -4.9 -11.2 -11.6 -11.6 -15.0\n",
      "CGG   1.4  -7.3   2.2  -7.1  -8.0  -9.3  -6.4  -9.6   9.1  -5.5  10.0  -5.4 -10.9 -14.3  -8.4 -14.0   0.9  -1.1   2.4  -1.3  -6.9  -8.0  -3.5  -9.0  11.8  11.2  13.4  11.4  -7.3  -8.8  -6.5  -8.5 -10.0 -12.5  -7.8 -12.5  -9.6 -11.0  -7.1 -10.7  -7.9  -8.0  -4.3  -9.4 -11.4 -12.3 -10.9 -12.9 -50.0 -11.8 -50.0 -11.2  -9.4  -9.8  -7.3 -10.5 -50.0  -6.2  -2.2  -6.7  -9.8 -15.2  -8.5 -14.9\n",
      "CGT   0.2  -6.3   0.1  -4.3  -8.2  -8.9  -8.5  -7.0   8.5  -4.1   8.0  -2.1 -10.5 -13.4 -10.2 -11.9   0.2   1.6  -0.8   3.4  -7.5  -6.7  -6.1  -4.7  12.1  12.8  11.4  14.7  -8.7  -7.4  -9.4  -5.3  -9.9 -11.8 -10.5 -10.3  -9.9 -10.1  -9.3  -9.3  -8.1  -7.2  -8.0  -5.2 -12.0 -12.1 -12.6 -10.9 -50.0  -7.9 -50.0  -5.3  -9.3  -8.8  -8.9  -7.0 -50.0  -3.6  -7.1  -1.2 -10.2 -12.7  -9.9 -11.0\n",
      "CTA -10.2 -13.4 -11.8 -12.4  -6.0  -9.4  -7.1  -8.6  -9.8 -12.9 -10.8 -12.4   2.3  -1.4   1.0  -1.0  -3.0  -8.0  -5.1  -7.6  -2.2  -6.1  -3.6  -6.1  -6.0  -9.7  -7.3  -8.7  11.2   7.9   8.9   8.1 -12.1 -17.4 -13.2 -16.8  -6.3  -9.5  -7.5  -8.8 -12.7 -15.2 -13.1 -14.2   0.0  -3.1  -1.8  -3.1 -50.0 -10.7 -50.0  -9.3  -4.6  -8.9  -5.9  -8.3 -50.0 -12.8  -9.6 -10.6   9.5  -3.2   8.2  -2.8\n",
      "CTC -13.5 -11.3 -14.0 -12.9  -9.3  -7.1  -8.2  -8.9 -12.7 -10.3 -11.9 -12.3  -0.4   1.4  -0.6  -0.4  -7.2  -4.7  -6.9  -5.7  -7.4  -3.7  -6.5  -6.0  -9.1  -5.3  -8.8  -7.4   7.9  11.9   7.8   9.3 -15.2 -15.1 -14.6 -17.2  -9.7  -7.3  -8.2  -9.2 -15.0 -12.8 -14.1 -14.9  -2.8   0.2  -2.9  -2.3 -50.0  -7.4 -50.0  -8.7  -9.4  -7.3  -9.4  -9.4 -50.0  -8.0 -11.9  -9.7   6.5   0.5   6.4  -1.8\n",
      "CTG -13.0 -14.4 -11.9 -13.9  -7.7  -9.0  -6.7  -8.8 -11.7 -12.7 -10.7 -13.1  -0.2  -1.6   1.7  -1.6  -5.7  -8.7  -4.8  -8.1  -5.3  -7.5  -3.3  -7.5  -9.0  -9.2  -6.5  -9.4   8.9   7.8  10.1   7.8 -14.3 -17.7 -13.0 -17.8  -8.0  -9.4  -6.2  -9.2 -15.1 -14.9 -12.2 -14.9  -1.8  -3.2  -0.8  -3.2 -50.0 -10.8 -50.0  -9.9  -7.3  -9.6  -6.4  -9.4 -50.0 -12.3  -8.7 -11.3   7.6  -3.3   8.7  -2.9\n",
      "CTT -12.5 -12.9 -13.4 -10.0  -8.5  -9.0  -7.8  -6.8 -11.8 -11.3 -12.4 -10.1  -0.3  -0.5  -0.5   1.2  -6.1  -6.0  -6.7  -3.5  -5.9  -5.6  -5.5  -2.9  -9.5  -7.8  -8.5  -5.3   8.1   9.3   7.8  11.8 -14.4 -15.7 -14.5 -14.6  -8.4  -8.7  -7.9  -7.1 -14.3 -13.8 -13.8 -12.3  -2.4  -2.2  -2.9  -0.2 -50.0  -8.9 -50.0  -7.1  -8.1  -9.1  -8.7  -6.6 -50.0  -9.7 -11.4  -8.0   6.8  -1.8   6.9   0.4\n",
      "GAA  -2.6  -6.3  -4.9  -4.9  -6.4  -9.4  -7.9  -8.6  -6.3  -7.4  -8.2  -6.6  -9.7 -14.9 -11.1 -13.9  -0.2  -8.2  -2.3  -6.8  -8.9 -11.1  -9.4 -10.9  -8.9 -11.7 -10.0  -9.9 -12.1 -15.2 -14.3 -14.4  11.1   2.9   9.2   3.5  -3.0  -7.0  -4.3  -6.2  -2.2  -6.6  -3.7  -5.7  -6.0 -10.4  -8.6  -9.5 -50.0 -15.5 -50.0 -13.9  -7.8 -10.5  -8.7  -9.6 -50.0 -16.5 -17.1 -14.8 -12.1 -18.9 -12.4 -17.4\n",
      "GAC  -8.5   0.8  -9.1  -0.7 -10.1  -7.5  -9.1  -8.2 -11.4  -3.1 -11.8  -4.4 -15.5 -14.3 -14.8 -15.4  -6.6  -4.7  -7.1  -5.8 -12.5 -10.3 -11.0 -12.2 -13.4 -10.6 -12.5 -11.8 -17.4 -15.1 -17.7 -15.7   2.9  12.5   3.1  10.3  -7.7  -5.2  -6.2  -7.3  -5.4  -1.7  -5.4  -3.7 -11.6  -9.1 -13.0 -10.8 -50.0 -10.3 -50.0 -11.2  -9.8  -8.3  -9.1  -9.7 -50.0 -12.7 -20.6 -13.1 -16.0 -16.6 -17.0 -18.4\n",
      "GAG  -5.0  -6.4  -3.4  -5.6  -7.5  -9.3  -7.1  -8.8  -8.8  -7.4  -6.9  -6.9 -11.4 -15.2  -9.7 -13.7  -1.8  -7.8  -0.8  -7.2  -9.4 -10.6  -8.7 -11.2 -10.0 -10.1  -7.8 -10.5 -13.2 -14.6 -13.0 -14.5   9.2   3.1  10.8   3.0  -4.5  -6.7  -2.7  -6.7  -5.0  -6.8  -2.5  -6.7  -7.5 -10.4  -7.4 -10.2 -50.0 -15.5 -50.0 -14.2  -8.9 -10.2  -8.5 -10.0 -50.0 -15.6 -15.3 -15.5 -13.0 -18.8 -12.6 -18.3\n",
      "GAT  -8.1  -1.1  -8.9   1.4  -8.6  -9.2  -9.4  -6.8 -11.0  -4.9 -11.5  -2.6 -13.7 -16.8 -14.5 -12.6  -6.0  -6.7  -7.2  -4.2 -11.6 -11.9 -11.2 -10.9 -13.1 -12.2 -12.5 -10.3 -16.8 -17.2 -17.8 -14.6   3.5  10.3   3.0  12.4  -6.9  -7.5  -6.8  -4.9  -5.4  -4.2  -5.4  -1.3 -10.8 -11.5 -12.4  -8.2 -50.0 -12.5 -50.0  -8.9  -9.2 -10.0  -9.4  -7.9 -50.0 -14.3 -19.9 -11.4 -16.2 -19.2 -15.8 -15.9\n",
      "GCA  -6.3  -7.4  -8.0  -6.0   2.8  -0.7   0.9   0.3  -7.5  -4.4  -8.4  -3.6  -3.0  -7.7  -3.7  -6.2  -5.3 -10.0  -7.2  -8.9  -0.7  -3.2  -1.8  -2.4  -9.9 -11.7  -9.6  -9.9  -6.3  -9.7  -8.0  -8.4  -3.0  -7.7  -4.5  -6.9  11.3   8.2   9.4   9.1  -1.0  -3.3  -1.8  -2.4   1.5  -2.1  -0.4  -1.1 -50.0 -14.5 -50.0 -12.9   2.8  -0.0   1.4   0.7 -50.0  -8.7 -14.1  -7.5  -5.2 -12.5  -6.5 -10.7\n",
      "GCC  -9.9  -5.0 -10.0  -6.5  -0.7   2.6  -0.2   0.1 -10.3  -1.8  -9.9  -3.8  -6.4  -4.7  -5.8  -6.7  -8.2  -8.3  -8.0 -10.2  -3.9  -1.0  -2.7  -2.9 -11.7  -8.8 -11.0 -10.1  -9.5  -7.3  -9.4  -8.7  -7.0  -5.2  -6.7  -7.5   8.2  11.6   8.8   9.1  -3.7  -0.9  -3.1  -3.0  -2.0   1.2  -2.3  -1.3 -50.0 -12.1 -50.0 -12.6   0.1   2.5   0.7   0.6 -50.0  -6.0 -14.8  -7.5  -8.6  -9.4  -8.5 -11.2\n",
      "GCG  -7.5  -6.2  -7.2  -6.2   0.9   0.2   2.5   0.2  -8.3  -3.5  -7.1  -3.7  -4.1  -6.3  -3.1  -6.3  -5.7  -8.9  -5.6  -8.4  -2.3  -2.2   0.7  -2.7  -9.8  -8.9  -7.1  -9.3  -7.5  -8.2  -6.2  -7.9  -4.3  -6.2  -2.7  -6.8   9.4   8.8  12.1   8.8  -2.1  -2.0   0.1  -2.4   0.0  -1.1   1.3  -0.8 -50.0 -11.9 -50.0 -12.6   1.2   0.7   3.5   0.8 -50.0  -7.1 -12.4  -7.1  -7.1 -10.6  -5.0 -10.6\n",
      "GCT  -9.0  -6.5 -10.0  -4.8   0.1  -0.1  -0.3   2.4  -9.9  -3.5 -10.6  -2.0  -5.4  -6.6  -5.6  -4.4  -7.6  -9.9  -8.5  -8.2  -3.0  -2.8  -3.0  -0.9 -12.8 -11.8 -10.7  -9.3  -8.8  -9.2  -9.2  -7.1  -6.2  -7.3  -6.7  -4.9   9.1   9.1   8.8  11.2  -3.1  -3.0  -3.3  -1.0  -1.1  -1.4  -2.0   1.2 -50.0 -12.8 -50.0 -10.9   0.8   0.5   0.7   2.3 -50.0  -7.5 -16.5  -5.9  -7.8 -11.1  -7.8  -9.0\n",
      "GGA  -7.1  -5.6  -9.1  -5.1  -5.9  -8.0  -6.9  -7.4  -1.9  -1.6  -4.7  -1.4  -9.6 -14.6 -10.4 -13.0  -6.7 -10.8  -9.4  -9.9  -9.1 -10.5  -9.1 -10.4  -5.9  -9.3  -7.9  -8.1 -12.7 -15.0 -15.1 -14.3  -2.2  -5.4  -5.0  -5.4  -1.0  -3.7  -2.1  -3.1  12.8   9.6  11.1  10.1  -4.8  -9.4  -8.1  -8.6 -50.0 -18.3 -50.0 -15.4  -5.9  -7.5  -6.6  -7.1 -50.0  -9.5 -11.7  -8.4 -12.5 -17.3 -13.2 -15.4\n",
      "GGC -10.2  -1.6 -10.2  -3.2  -7.3  -4.6  -6.5  -6.2  -6.7   3.0  -6.8   0.8 -11.9 -10.8 -11.5 -12.2  -9.6  -7.1  -9.7  -8.4  -9.8  -8.1  -8.2  -9.7  -9.4  -5.0  -8.0  -7.2 -15.2 -12.8 -14.9 -13.8  -6.6  -1.7  -6.8  -4.2  -3.3  -0.9  -2.0  -3.0   9.6  12.8   9.7  10.5  -8.5  -5.6  -9.3  -7.8 -50.0 -12.7 -50.0 -13.8  -6.8  -5.1  -5.3  -6.5 -50.0  -4.1 -12.8  -5.8 -14.4 -14.1 -13.6 -14.8\n",
      "GGG  -8.2  -4.7  -7.1  -5.0  -6.5  -7.7  -5.5  -6.9  -4.1  -0.8  -1.2  -0.9 -10.5 -12.8  -8.2 -13.0  -7.5 -10.0  -7.2  -9.7  -9.2  -9.4  -7.3 -10.2  -6.5  -8.0  -4.3  -8.0 -13.1 -14.1 -12.2 -13.8  -3.7  -5.4  -2.5  -5.4  -1.8  -3.1   0.1  -3.3  11.1   9.7  12.9   9.8  -5.5  -8.1  -5.4  -7.8 -50.0 -16.4 -50.0 -14.8  -5.9  -7.1  -4.7  -7.2 -50.0  -8.4  -6.4  -8.1 -12.1 -16.2 -10.3 -15.8\n",
      "GGT  -9.2  -3.0  -9.9  -1.2  -6.5  -6.1  -6.8  -4.4  -6.4   0.9  -7.0   3.0 -11.8 -13.0 -11.1 -10.4  -9.3  -9.2  -9.8  -6.4  -9.5  -9.5  -9.4  -8.0  -8.1  -8.2  -9.4  -5.2 -14.2 -14.9 -14.9 -12.3  -5.7  -3.7  -6.7  -1.3  -2.4  -3.0  -2.4  -1.0  10.1  10.5   9.8  13.1  -7.2  -8.0  -8.4  -5.0 -50.0 -14.7 -50.0 -11.4  -6.3  -6.7  -5.8  -5.0 -50.0  -6.1 -13.4  -3.6 -13.0 -16.0 -13.7 -13.3\n",
      "GTA  -8.2 -10.8 -10.1  -9.8  -0.4  -3.6  -1.7  -2.9  -7.9  -8.4  -9.8  -8.2   6.2   2.3   0.6   2.9  -7.1 -11.6  -9.2 -11.1  -5.6  -8.3  -7.2  -8.4 -10.7 -13.6 -11.4 -12.0   0.0  -2.8  -1.8  -2.4  -6.0 -11.6  -7.5 -10.8   1.5  -2.0   0.0  -1.1  -4.8  -8.5  -5.5  -7.2  11.9   8.6  10.0   9.0 -50.0 -14.2 -50.0 -12.8  -4.1  -7.4  -5.7  -6.5 -50.0 -11.0 -14.3  -9.9   1.4  -7.9  -0.7  -6.8\n",
      "GTC -12.5  -8.7 -13.0  -9.9  -3.9  -1.0  -3.4  -2.9 -12.5  -6.3 -11.6  -8.1   3.3   6.0  -1.6   3.8 -10.5 -10.1 -11.1 -11.1  -9.0  -6.9  -8.2  -9.1 -14.0 -11.4 -12.3 -12.1  -3.1   0.2  -3.2  -2.2 -10.4  -9.1 -10.4 -11.5  -2.1   1.2  -1.1  -1.4  -9.4  -5.6  -8.1  -8.0   8.6  12.4   8.6   9.5 -50.0 -10.4 -50.0 -11.6  -7.2  -5.7  -7.5  -7.4 -50.0  -8.6 -15.7  -9.7  -3.0  -3.7  -3.2  -5.9\n",
      "GTG -11.1 -11.9 -10.6 -11.3  -2.6  -4.1  -1.7  -3.5 -11.4  -9.6  -9.6  -9.2   3.7   2.5   1.6   2.5  -9.6 -12.3  -9.5 -11.8  -8.0  -9.4  -7.2  -9.3 -13.3 -12.8 -10.9 -12.6  -1.8  -2.9  -0.8  -2.9  -8.6 -13.0  -7.4 -12.4  -0.4  -2.3   1.3  -2.0  -8.1  -9.3  -5.4  -8.4  10.0   8.6  11.4   8.9 -50.0 -13.7 -50.0 -13.2  -6.1  -8.1  -5.5  -7.5 -50.0 -11.3 -12.7 -10.1  -1.6  -7.8  -0.3  -7.2\n",
      "GTT -11.4 -10.0 -12.7  -8.2  -3.1  -3.1  -2.8  -0.9 -11.7  -7.3 -12.1  -5.8   3.6   3.6  -1.1   5.8  -9.4 -11.9 -10.7 -10.0  -8.5  -8.8  -8.2  -6.5 -13.9 -13.0 -12.9 -10.9  -3.1  -2.3  -3.2  -0.2  -9.5 -10.8 -10.2  -8.2  -1.1  -1.3  -0.8   1.2  -8.6  -7.8  -7.8  -5.0   9.0   9.5   8.9  12.0 -50.0 -13.0 -50.0 -10.3  -6.2  -7.0  -7.1  -5.4 -50.0  -9.3 -14.0  -8.2  -2.4  -6.3  -2.4  -3.9\n",
      "TAA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0  30.6 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAC -14.8  -6.2 -14.9  -7.6 -14.6 -11.6 -13.0 -12.2 -14.0  -9.0 -14.1 -10.1 -13.6 -11.2 -12.4 -12.5  -8.1   2.2  -8.7   0.4 -15.3 -11.3 -13.2 -13.3 -11.9  -7.2 -11.8  -7.9 -10.7  -7.4 -10.8  -8.9 -15.5 -10.3 -15.5 -12.5 -14.5 -12.1 -11.9 -12.8 -18.3 -12.7 -16.4 -14.7 -14.2 -10.4 -13.7 -13.0 -50.0  15.1 -50.0  13.3  -9.9  -5.8  -8.9  -7.6 -50.0  -1.5  -7.8  -3.0  -8.1   3.6  -9.4   2.0\n",
      "TAG -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  30.6 -50.0  35.2 -50.0 -50.0 -50.0 -50.0 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAT -13.8  -7.5 -14.4  -5.1 -12.6 -11.6 -11.5 -10.1 -13.2  -9.7 -12.7  -7.4 -11.9 -11.9 -11.5 -10.2  -7.2   0.9  -8.6   2.6 -14.1 -12.8 -13.3 -11.0  -9.7  -8.4 -11.2  -5.3  -9.3  -8.7  -9.9  -7.1 -13.9 -11.2 -14.2  -8.9 -12.9 -12.6 -12.6 -10.9 -15.4 -13.8 -14.8 -11.4 -12.8 -11.6 -13.2 -10.3 -50.0  13.3 -50.0  15.2  -8.6  -7.0  -8.4  -4.8 -50.0  -3.0  -7.1  -0.1  -7.3   2.3  -8.7   3.9\n",
      "TCA  -7.3  -6.3  -8.9  -5.5   2.9   0.2   1.5   1.2  -8.5  -2.3 -10.0  -1.7  -5.6 -10.5  -5.5  -9.6  -4.3  -7.9  -6.1  -7.2   2.3  -0.6   0.5  -0.0  -8.9 -11.0  -9.4  -9.3  -4.6  -9.4  -7.3  -8.1  -7.8  -9.8  -8.9  -9.2   2.8   0.1   1.2   0.8  -5.9  -6.8  -5.9  -6.3  -4.1  -7.2  -6.1  -6.2 -50.0  -9.9 -50.0  -8.6  12.5   9.4  11.0   9.8 -50.0  -4.8  -8.7  -3.5  -0.0  -8.6  -2.1  -7.0\n",
      "TCC -10.1  -4.3 -10.8  -5.8   0.1   2.3   0.7   0.6 -10.4  -0.2 -10.8  -2.0  -8.8  -9.2  -8.0  -9.7  -6.8  -5.9  -7.5  -7.0  -1.4   2.2  -0.9  -0.4 -10.8  -7.7  -9.8  -8.8  -8.9  -7.3  -9.6  -9.1 -10.5  -8.3 -10.2 -10.0  -0.0   2.5   0.7   0.5  -7.5  -5.1  -7.1  -6.7  -7.4  -5.7  -8.1  -7.0 -50.0  -5.8 -50.0  -7.0   9.4  12.7  10.1  10.0 -50.0  -1.5 -11.3  -3.3  -5.4  -4.2  -5.6  -6.8\n",
      "TCG  -8.4  -6.2  -8.9  -5.9   1.8   0.4   2.7   1.2  -9.1  -1.2  -8.8  -1.5  -7.2 -10.5  -4.9  -9.6  -5.7  -7.4  -5.5  -7.4   0.3  -0.3   1.8  -0.4  -9.0  -8.7  -7.3  -8.9  -5.9  -9.4  -6.4  -8.7  -8.7  -9.1  -8.5  -9.4   1.4   0.7   3.5   0.7  -6.6  -5.3  -4.7  -5.8  -5.7  -7.5  -5.5  -7.1 -50.0  -8.9 -50.0  -8.4  11.0  10.1  13.2  10.2 -50.0  -4.2  -6.1  -3.6  -3.5  -8.4  -1.2  -6.8\n",
      "TCT  -9.1  -5.4 -10.4  -4.2   1.0   0.5   0.6   2.4  -9.6  -1.5 -10.5  -0.5  -8.9 -10.5  -7.2  -8.3  -6.3  -6.9  -7.7  -5.1  -0.5   0.1  -0.6   2.3 -11.1 -10.2 -10.5  -7.0  -8.3  -9.4  -9.4  -6.6  -9.6  -9.7 -10.0  -7.9   0.7   0.6   0.8   2.3  -7.1  -6.5  -7.2  -5.0  -6.5  -7.4  -7.5  -5.4 -50.0  -7.6 -50.0  -4.8   9.8  10.0  10.2  12.1 -50.0  -3.1 -11.4  -0.7  -4.6  -6.6  -4.9  -3.9\n",
      "TGA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TGC -13.0  -7.0 -13.1  -8.3  -9.2  -6.5  -8.9  -7.6  -9.8  -0.7  -9.2  -2.1 -12.4 -10.3 -12.4 -11.6  -9.5  -4.6 -10.4  -5.7 -13.6 -10.1 -11.4 -10.9  -6.5  -1.5  -6.2  -3.6 -12.8  -8.0 -12.3  -9.7 -16.5 -12.7 -15.6 -14.3  -8.7  -6.0  -7.1  -7.5  -9.5  -4.1  -8.4  -6.1 -11.0  -8.6 -11.3  -9.3 -50.0  -1.5 -50.0  -3.0  -4.8  -1.5  -4.2  -3.1 -50.0  16.4  -5.1  14.2 -10.1  -4.3 -10.0  -5.4\n",
      "TGG -13.5 -16.3 -11.8 -15.5 -13.6 -15.3 -12.1 -16.1  -7.7 -13.3  -4.2 -12.6 -14.1 -16.2 -10.3 -15.4  -9.6 -11.6  -7.7  -9.8 -14.3 -17.5 -11.1 -17.2  -5.3  -7.1  -2.2  -7.1  -9.6 -11.9  -8.7 -11.4 -17.1 -20.6 -15.3 -19.9 -14.1 -14.8 -12.4 -16.5 -11.7 -12.8  -6.4 -13.4 -14.3 -15.7 -12.7 -14.0 -50.0  -7.8 -50.0  -7.1  -8.7 -11.3  -6.1 -11.4 -50.0  -5.1  18.6  -4.8  -8.6  -8.1  -4.5  -7.3\n",
      "TGT -12.4  -8.2 -13.0  -5.9  -7.7  -7.8  -7.4  -5.8  -8.8  -2.0  -9.3  -0.4 -11.6 -11.3 -11.6  -9.4  -8.9  -5.6 -10.0  -3.2 -11.5 -11.1 -10.5  -8.3  -6.3  -4.9  -6.7  -1.2 -10.6  -9.7 -11.3  -8.0 -14.8 -13.1 -15.5 -11.4  -7.5  -7.5  -7.1  -5.9  -8.4  -5.8  -8.1  -3.6  -9.9  -9.7 -10.1  -8.2 -50.0  -3.0 -50.0  -0.1  -3.5  -3.3  -3.6  -0.7 -50.0  14.2  -4.8  16.4  -8.9  -5.8  -8.9  -3.4\n",
      "TTA -10.7 -13.2 -12.4 -11.6  -4.8  -8.4  -6.1  -7.2 -10.3 -11.8 -11.0 -11.6   2.8  -1.7   0.7  -1.1  -6.2  -9.7  -7.3  -8.4  -5.3  -8.6  -6.1  -8.3 -10.5 -11.2  -9.8 -10.2   9.5   6.5   7.6   6.8 -12.1 -16.0 -13.0 -16.2  -5.2  -8.6  -7.1  -7.8 -12.5 -14.4 -12.1 -13.0   1.4  -3.0  -1.6  -2.4 -50.0  -8.1 -50.0  -7.3  -0.0  -5.4  -3.5  -4.6 -50.0 -10.1  -8.6  -8.9  13.2  -0.9   9.7  -0.3\n",
      "TTC -18.1 -12.3 -19.4 -14.2 -12.2  -9.5 -11.5 -10.8 -17.9 -11.9 -18.2 -13.9  -6.4  -3.7  -6.8  -5.4 -12.8  -5.9 -13.6  -7.4 -13.7  -9.9 -12.9 -12.7 -16.2 -11.6 -15.2 -12.7  -3.2   0.5  -3.3  -1.8 -18.9 -16.6 -18.8 -19.2 -12.5  -9.4 -10.6 -11.1 -17.3 -14.1 -16.2 -16.0  -7.9  -3.7  -7.8  -6.3 -50.0   3.6 -50.0   2.3  -8.6  -4.2  -8.4  -6.6 -50.0  -4.3  -8.1  -5.8  -0.9  14.2  -1.8  11.6\n",
      "TTG -11.8 -13.1 -11.5 -12.0  -6.2  -8.3  -5.4  -7.4 -11.8 -12.3 -11.1 -11.4   0.5  -2.5   2.4  -1.6  -6.8  -9.9  -6.4  -8.9  -6.1  -8.6  -5.0  -7.8 -10.4 -11.6  -8.5  -9.9   8.2   6.4   8.7   6.9 -12.4 -17.0 -12.6 -15.8  -6.5  -8.5  -5.0  -7.8 -13.2 -13.6 -10.3 -13.7  -0.7  -3.2  -0.3  -2.4 -50.0  -9.4 -50.0  -8.7  -2.1  -5.6  -1.2  -4.9 -50.0 -10.0  -4.5  -8.9   9.7  -1.8  11.3  -0.9\n",
      "TTT -17.2 -13.3 -17.8 -11.6 -11.0 -11.1 -10.4  -9.0 -16.0 -12.7 -16.1 -11.1  -5.2  -5.4  -6.0  -3.3 -11.8  -7.3 -13.3  -5.9 -12.7 -11.9 -11.9  -9.3 -15.2 -15.0 -14.9 -11.0  -2.8  -1.8  -2.9   0.4 -17.4 -18.4 -18.3 -15.9 -10.7 -11.2 -10.6  -9.0 -15.4 -14.8 -15.8 -13.3  -6.8  -5.9  -7.2  -3.9 -50.0   2.0 -50.0   3.9  -7.0  -6.8  -6.8  -3.9 -50.0  -5.4  -7.3  -3.4  -0.3  11.6  -0.9  14.1\n",
      "\n",
      "STR #  Steven Henikoff and Jorja G. Henikoff:\n",
      "#  \"Performance evaluation of amino acid substitution matrices.\"\n",
      "#  Proteins: Structure, Function, and Genetics: 17(1): 49-61 (1993).\n",
      "#  Figure 1, lower triangle.\n",
      "#  PMID 8234244\n",
      "     A    C    D    E    F    G    H    I    K    L    M    N    P    Q    R    S    T    V    W    Y\n",
      "A  4.0 -2.0 -1.0  0.0 -3.0  0.0 -2.0 -2.0 -1.0 -2.0  0.0 -1.0 -1.0  0.0 -1.0  0.0 -1.0  0.0 -3.0 -3.0\n",
      "C -2.0 11.0 -7.0 -3.0 -2.0 -6.0 -6.0 -4.0 -4.0 -6.0 -5.0 -6.0 -8.0 -3.0 -2.0 -4.0 -5.0 -4.0 -6.0 -6.0\n",
      "D -1.0 -7.0  6.0  2.0 -5.0 -1.0  0.0 -3.0 -1.0 -6.0 -4.0  2.0 -1.0  0.0 -2.0  0.0 -1.0 -4.0 -6.0 -3.0\n",
      "E  0.0 -3.0  2.0  5.0 -4.0 -2.0 -2.0 -3.0  1.0 -4.0 -2.0  0.0 -1.0  2.0  0.0 -1.0  0.0 -2.0 -6.0 -2.0\n",
      "F -3.0 -2.0 -5.0 -4.0  7.0 -6.0 -2.0  1.0 -3.0  2.0  0.0 -3.0 -5.0 -4.0 -4.0 -3.0 -3.0 -1.0  2.0  3.0\n",
      "G  0.0 -6.0 -1.0 -2.0 -6.0  5.0 -3.0 -5.0 -3.0 -5.0 -4.0 -1.0 -2.0 -2.0 -2.0 -1.0 -3.0 -4.0 -4.0 -3.0\n",
      "H -2.0 -6.0  0.0 -2.0 -2.0 -3.0  8.0 -5.0  0.0 -3.0 -2.0  2.0 -3.0  0.0  0.0 -2.0 -2.0 -2.0 -3.0  0.0\n",
      "I -2.0 -4.0 -3.0 -3.0  1.0 -5.0 -5.0  6.0 -3.0  2.0  1.0 -3.0 -4.0 -5.0 -3.0 -3.0 -2.0  2.0 -2.0 -1.0\n",
      "K -1.0 -4.0 -1.0  1.0 -3.0 -3.0  0.0 -3.0  5.0 -2.0 -1.0  0.0 -1.0  1.0  2.0 -1.0  0.0 -3.0 -3.0 -2.0\n",
      "L -2.0 -6.0 -6.0 -4.0  2.0 -5.0 -3.0  2.0 -2.0  5.0  3.0 -3.0 -3.0 -3.0 -3.0 -4.0 -3.0  1.0 -1.0 -2.0\n",
      "M  0.0 -5.0 -4.0 -2.0  0.0 -4.0 -2.0  1.0 -1.0  3.0  8.0 -2.0 -6.0  1.0 -4.0 -4.0 -2.0  0.0 -2.0 -1.0\n",
      "N -1.0 -6.0  2.0  0.0 -3.0 -1.0  2.0 -3.0  0.0 -3.0 -2.0  5.0 -2.0  0.0 -1.0  0.0  0.0 -4.0 -5.0 -1.0\n",
      "P -1.0 -8.0 -1.0 -1.0 -5.0 -2.0 -3.0 -4.0 -1.0 -3.0 -6.0 -2.0  7.0 -2.0 -2.0 -1.0 -1.0 -4.0 -4.0 -6.0\n",
      "Q  0.0 -3.0  0.0  2.0 -4.0 -2.0  0.0 -5.0  1.0 -3.0  1.0  0.0 -2.0  6.0  1.0 -1.0  0.0 -2.0 -5.0 -3.0\n",
      "R -1.0 -2.0 -2.0  0.0 -4.0 -2.0  0.0 -3.0  2.0 -3.0 -4.0 -1.0 -2.0  1.0  7.0  0.0 -1.0 -3.0 -2.0 -1.0\n",
      "S  0.0 -4.0  0.0 -1.0 -3.0 -1.0 -2.0 -3.0 -1.0 -4.0 -4.0  0.0 -1.0 -1.0  0.0  4.0  1.0 -3.0 -5.0 -2.0\n",
      "T -1.0 -5.0 -1.0  0.0 -3.0 -3.0 -2.0 -2.0  0.0 -3.0 -2.0  0.0 -1.0  0.0 -1.0  1.0  5.0 -1.0 -5.0 -2.0\n",
      "V  0.0 -4.0 -4.0 -2.0 -1.0 -4.0 -2.0  2.0 -3.0  1.0  0.0 -4.0 -4.0 -2.0 -3.0 -3.0 -1.0  5.0 -4.0 -1.0\n",
      "W -3.0 -6.0 -6.0 -6.0  2.0 -4.0 -3.0 -2.0 -3.0 -1.0 -2.0 -5.0 -4.0 -5.0 -2.0 -5.0 -5.0 -4.0 10.0  2.0\n",
      "Y -3.0 -6.0 -3.0 -2.0  3.0 -3.0  0.0 -1.0 -2.0 -2.0 -1.0 -1.0 -6.0 -3.0 -1.0 -2.0 -2.0 -1.0  2.0  7.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for m in substitution_matrices.load():\n",
    "    print(m, substitution_matrices.load(m))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#  Adrian Schneider, Gina M. Cannarozzi, and Gaston H. Gonnet:\n",
      "#  \"Empirical codon substitution matrix.\"\n",
      "#  BMC Bioinformatics 6:134 (2005).\n",
      "#  Additional File 3.\n",
      "#  PMID 15927081\n",
      "      AAA   AAC   AAG   AAT   ACA   ACC   ACG   ACT   AGA   AGC   AGG   AGT   ATA   ATC   ATG   ATT   CAA   CAC   CAG   CAT   CCA   CCC   CCG   CCT   CGA   CGC   CGG   CGT   CTA   CTC   CTG   CTT   GAA   GAC   GAG   GAT   GCA   GCC   GCG   GCT   GGA   GGC   GGG   GGT   GTA   GTC   GTG   GTT   TAA   TAC   TAG   TAT   TCA   TCC   TCG   TCT   TGA   TGC   TGG   TGT   TTA   TTC   TTG   TTT\n",
      "AAA  11.6  -2.7   9.7  -1.7  -2.7  -6.4  -3.9  -5.6   5.1  -5.0   3.6  -4.2  -6.3 -13.0  -7.1 -11.5   0.4  -6.0  -1.9  -5.3  -8.5 -11.2  -8.9 -10.8   2.1   0.0   1.4   0.2 -10.2 -13.5 -13.0 -12.5  -2.6  -8.5  -5.0  -8.1  -6.3  -9.9  -7.5  -9.0  -7.1 -10.2  -8.2  -9.2  -8.2 -12.5 -11.1 -11.4 -50.0 -14.8 -50.0 -13.8  -7.3 -10.1  -8.4  -9.1 -50.0 -13.0 -13.5 -12.4 -10.7 -18.1 -11.8 -17.2\n",
      "AAC  -2.7  13.0  -3.3  10.9  -3.5  -0.4  -3.3  -1.8  -5.4   4.6  -5.5   3.0 -10.2  -7.9  -9.9  -9.6  -5.0   0.5  -5.5  -1.0 -10.3  -8.1  -9.4  -9.6  -8.1  -5.0  -7.3  -6.3 -13.4 -11.3 -14.4 -12.9  -6.3   0.8  -6.4  -1.1  -7.4  -5.0  -6.2  -6.5  -5.6  -1.6  -4.7  -3.0 -10.8  -8.7 -11.9 -10.0 -50.0  -6.2 -50.0  -7.5  -6.3  -4.3  -6.2  -5.4 -50.0  -7.0 -16.3  -8.2 -13.2 -12.3 -13.1 -13.3\n",
      "AAG   9.7  -3.3  11.6  -2.8  -4.5  -6.7  -3.1  -6.9   3.3  -5.5   4.8  -5.1  -8.9 -13.2  -5.7 -12.6  -1.5  -6.1  -0.6  -6.1 -10.0 -11.8  -8.6 -11.9   1.2   0.5   2.2   0.1 -11.8 -14.0 -11.9 -13.4  -4.9  -9.1  -3.4  -8.9  -8.0 -10.0  -7.2 -10.0  -9.1 -10.2  -7.1  -9.9 -10.1 -13.0 -10.6 -12.7 -50.0 -14.9 -50.0 -14.4  -8.9 -10.8  -8.9 -10.4 -50.0 -13.1 -11.8 -13.0 -12.4 -19.4 -11.5 -17.8\n",
      "AAT  -1.7  10.9  -2.8  12.9  -2.7  -2.2  -2.8   0.2  -4.8   2.9  -5.2   5.2  -9.1  -9.5  -9.0  -7.0  -4.0  -1.0  -5.0   1.0  -9.0  -8.8  -8.9  -7.7  -7.0  -6.8  -7.1  -4.3 -12.4 -12.9 -13.9 -10.0  -4.9  -0.7  -5.6   1.4  -6.0  -6.5  -6.2  -4.8  -5.1  -3.2  -5.0  -1.2  -9.8  -9.9 -11.3  -8.2 -50.0  -7.6 -50.0  -5.1  -5.5  -5.8  -5.9  -4.2 -50.0  -8.3 -15.5  -5.9 -11.6 -14.2 -12.0 -11.6\n",
      "ACA  -2.7  -3.5  -4.5  -2.7  11.7   9.0  10.6   9.6  -3.2  -0.7  -5.1  -0.4   0.7  -4.6  -0.4  -3.6  -4.8  -8.9  -6.5  -7.8  -1.4  -4.1  -2.9  -3.3  -8.0  -9.1  -8.0  -8.2  -6.0  -9.3  -7.7  -8.5  -6.4 -10.1  -7.5  -8.6   2.8  -0.7   0.9   0.1  -5.9  -7.3  -6.5  -6.5  -0.4  -3.9  -2.6  -3.1 -50.0 -14.6 -50.0 -12.6   2.9   0.1   1.8   1.0 -50.0  -9.2 -13.6  -7.7  -4.8 -12.2  -6.2 -11.0\n",
      "ACC  -6.4  -0.4  -6.7  -2.2   9.0  12.3   9.8   9.6  -7.1   2.4  -7.2   0.3  -2.9  -1.2  -3.0  -3.3  -7.6  -6.7  -8.1  -7.9  -4.9  -1.9  -4.5  -3.8  -9.5  -7.8  -9.3  -8.9  -9.4  -7.1  -9.0  -9.0  -9.4  -7.5  -9.3  -9.2  -0.7   2.6   0.2  -0.1  -8.0  -4.6  -7.7  -6.1  -3.6  -1.0  -4.1  -3.1 -50.0 -11.6 -50.0 -11.6   0.2   2.3   0.4   0.5 -50.0  -6.5 -15.3  -7.8  -8.4  -9.5  -8.3 -11.1\n",
      "ACG  -3.9  -3.3  -3.1  -2.8  10.6   9.8  12.2   9.8  -4.9   0.1  -3.6  -0.2  -1.7  -4.1   0.9  -3.5  -5.9  -8.9  -5.6  -8.5  -3.9  -3.9  -2.2  -4.5  -7.3  -7.5  -6.4  -8.5  -7.1  -8.2  -6.7  -7.8  -7.9  -9.1  -7.1  -9.4   0.9  -0.2   2.5  -0.3  -6.9  -6.5  -5.5  -6.8  -1.7  -3.4  -1.7  -2.8 -50.0 -13.0 -50.0 -11.5   1.5   0.7   2.7   0.6 -50.0  -8.9 -12.1  -7.4  -6.1 -11.5  -5.4 -10.4\n",
      "ACT  -5.6  -1.8  -6.9   0.2   9.6   9.6   9.8  11.6  -6.6   0.9  -7.2   2.5  -2.3  -3.4  -2.3  -0.6  -6.5  -8.2  -8.0  -6.3  -3.7  -3.5  -3.9  -1.6  -9.9  -8.9  -9.6  -7.0  -8.6  -8.9  -8.8  -6.8  -8.6  -8.2  -8.8  -6.8   0.3   0.1   0.2   2.4  -7.4  -6.2  -6.9  -4.4  -2.9  -2.9  -3.5  -0.9 -50.0 -12.2 -50.0 -10.1   1.2   0.6   1.2   2.4 -50.0  -7.6 -16.1  -5.8  -7.2 -10.8  -7.4  -9.0\n",
      "AGA   5.1  -5.4   3.3  -4.8  -3.2  -7.1  -4.9  -6.6  13.3  -2.8  11.2  -1.9  -5.7 -12.5  -7.2 -11.6  -0.7  -4.8  -3.1  -4.3 -10.0 -11.6  -8.8 -11.7  10.5   7.7   9.1   8.5  -9.8 -12.7 -11.7 -11.8  -6.3 -11.4  -8.8 -11.0  -7.5 -10.3  -8.3  -9.9  -1.9  -6.7  -4.1  -6.4  -7.9 -12.5 -11.4 -11.7 -50.0 -14.0 -50.0 -13.2  -8.5 -10.4  -9.1  -9.6 -50.0  -9.8  -7.7  -8.8 -10.3 -17.9 -11.8 -16.0\n",
      "AGC  -5.0   4.6  -5.5   2.9  -0.7   2.4   0.1   0.9  -2.8  12.8  -2.6  11.0  -8.4  -6.3  -8.3  -7.7  -6.1  -2.9  -6.5  -4.2  -8.5  -5.9  -7.5  -7.7  -5.5  -2.4  -5.5  -4.1 -12.9 -10.3 -12.7 -11.3  -7.4  -3.1  -7.4  -4.9  -4.4  -1.8  -3.5  -3.5  -1.6   3.0  -0.8   0.9  -8.4  -6.3  -9.6  -7.3 -50.0  -9.0 -50.0  -9.7  -2.3  -0.2  -1.2  -1.5 -50.0  -0.7 -13.3  -2.0 -11.8 -11.9 -12.3 -12.7\n",
      "AGG   3.6  -5.5   4.8  -5.2  -5.1  -7.2  -3.6  -7.2  11.2  -2.6  13.4  -2.1  -7.7 -12.5  -5.3 -11.9  -2.2  -4.4  -1.9  -4.8 -10.2 -11.5  -8.1 -12.0   9.3   8.2  10.0   8.0 -10.8 -11.9 -10.7 -12.4  -8.2 -11.8  -6.9 -11.5  -8.4  -9.9  -7.1 -10.6  -4.7  -6.8  -1.2  -7.0  -9.8 -11.6  -9.6 -12.1 -50.0 -14.1 -50.0 -12.7 -10.0 -10.8  -8.8 -10.5 -50.0  -9.2  -4.2  -9.3 -11.0 -18.2 -11.1 -16.1\n",
      "AGT  -4.2   3.0  -5.1   5.2  -0.4   0.3  -0.2   2.5  -1.9  11.0  -2.1  13.2  -7.6  -8.5  -8.0  -5.3  -5.6  -4.5  -6.8  -2.4  -7.8  -7.9  -7.9  -6.2  -5.7  -5.1  -5.4  -2.1 -12.4 -12.3 -13.1 -10.1  -6.6  -4.4  -6.9  -2.6  -3.6  -3.8  -3.7  -2.0  -1.4   0.8  -0.9   3.0  -8.2  -8.1  -9.2  -5.8 -50.0 -10.1 -50.0  -7.4  -1.7  -2.0  -1.5  -0.5 -50.0  -2.1 -12.6  -0.4 -11.6 -13.9 -11.4 -11.1\n",
      "ATA  -6.3 -10.2  -8.9  -9.1   0.7  -2.9  -1.7  -2.3  -5.7  -8.4  -7.7  -7.6  13.2   9.6   3.5   9.7  -8.7 -12.4 -10.7 -10.7  -7.8 -10.2  -8.8  -9.6  -9.5 -11.9 -10.9 -10.5   2.3  -0.4  -0.2  -0.3  -9.7 -15.5 -11.4 -13.7  -3.0  -6.4  -4.1  -5.4  -9.6 -11.9 -10.5 -11.8   6.2   3.3   3.7   3.6 -50.0 -13.6 -50.0 -11.9  -5.6  -8.8  -7.2  -8.9 -50.0 -12.4 -14.1 -11.6   2.8  -6.4   0.5  -5.2\n",
      "ATC -13.0  -7.9 -13.2  -9.5  -4.6  -1.2  -4.1  -3.4 -12.5  -6.3 -12.5  -8.5   9.6  12.7   0.2  10.5 -12.4 -11.6 -13.2 -12.0 -12.4 -10.1 -11.6 -12.0 -15.1 -13.1 -14.3 -13.4  -1.4   1.4  -1.6  -0.5 -14.9 -14.3 -15.2 -16.8  -7.7  -4.7  -6.3  -6.6 -14.6 -10.8 -12.8 -13.0   2.3   6.0   2.5   3.6 -50.0 -11.2 -50.0 -11.9 -10.5  -9.2 -10.5 -10.5 -50.0 -10.3 -16.2 -11.3  -1.7  -3.7  -2.5  -5.4\n",
      "ATG  -7.1  -9.9  -5.7  -9.0  -0.4  -3.0   0.9  -2.3  -7.2  -8.3  -5.3  -8.0   3.5   0.2  14.3   1.0  -7.3 -11.2  -6.9  -9.7  -8.5 -10.2  -7.7  -9.8 -10.0 -10.4  -8.4 -10.2   1.0  -0.6   1.7  -0.5 -11.1 -14.8  -9.7 -14.5  -3.7  -5.8  -3.1  -5.6 -10.4 -11.5  -8.2 -11.1   0.6  -1.6   1.6  -1.1 -50.0 -12.4 -50.0 -11.5  -5.5  -8.0  -4.9  -7.2 -50.0 -12.4 -10.3 -11.6   0.7  -6.8   2.4  -6.0\n",
      "ATT -11.5  -9.6 -12.6  -7.0  -3.6  -3.3  -3.5  -0.6 -11.6  -7.7 -11.9  -5.3   9.7  10.5   1.0  12.6 -12.0 -12.2 -12.5  -9.7 -11.1 -11.2 -11.9  -9.4 -12.9 -14.3 -14.0 -11.9  -1.0  -0.4  -1.6   1.2 -13.9 -15.4 -13.7 -12.6  -6.2  -6.7  -6.3  -4.4 -13.0 -12.2 -13.0 -10.4   2.9   3.8   2.5   5.8 -50.0 -12.5 -50.0 -10.2  -9.6  -9.7  -9.6  -8.3 -50.0 -11.6 -15.4  -9.4  -1.1  -5.4  -1.6  -3.3\n",
      "CAA   0.4  -5.0  -1.5  -4.0  -4.8  -7.6  -5.9  -6.5  -0.7  -6.1  -2.2  -5.6  -8.7 -12.4  -7.3 -12.0  12.8   2.3  10.2   3.0   0.0  -3.4  -0.7  -3.2   2.5  -0.8   0.9   0.2  -3.0  -7.2  -5.7  -6.1  -0.2  -6.6  -1.8  -6.0  -5.3  -8.2  -5.7  -7.6  -6.7  -9.6  -7.5  -9.3  -7.1 -10.5  -9.6  -9.4 -50.0  -8.1 -50.0  -7.2  -4.3  -6.8  -5.7  -6.3 -50.0  -9.5  -9.6  -8.9  -6.2 -12.8  -6.8 -11.8\n",
      "CAC  -6.0   0.5  -6.1  -1.0  -8.9  -6.7  -8.9  -8.2  -4.8  -2.9  -4.4  -4.5 -12.4 -11.6 -11.2 -12.2   2.3  14.6   1.9  12.9  -5.5  -2.6  -4.1  -4.7  -1.2   3.0  -1.1   1.6  -8.0  -4.7  -8.7  -6.0  -8.2  -4.7  -7.8  -6.7 -10.0  -8.3  -8.9  -9.9 -10.8  -7.1 -10.0  -9.2 -11.6 -10.1 -12.3 -11.9 -50.0   2.2 -50.0   0.9  -7.9  -5.9  -7.4  -6.9 -50.0  -4.6 -11.6  -5.6  -9.7  -5.9  -9.9  -7.3\n",
      "CAG  -1.9  -5.5  -0.6  -5.0  -6.5  -8.1  -5.6  -8.0  -3.1  -6.5  -1.9  -6.8 -10.7 -13.2  -6.9 -12.5  10.2   1.9  11.9   2.1  -2.3  -4.1  -0.5  -4.6  -0.1  -0.9   2.4  -0.8  -5.1  -6.9  -4.8  -6.7  -2.3  -7.1  -0.8  -7.2  -7.2  -8.0  -5.6  -8.5  -9.4  -9.7  -7.2  -9.8  -9.2 -11.1  -9.5 -10.7 -50.0  -8.7 -50.0  -8.6  -6.1  -7.5  -5.5  -7.7 -50.0 -10.4  -7.7 -10.0  -7.3 -13.6  -6.4 -13.3\n",
      "CAT  -5.3  -1.0  -6.1   1.0  -7.8  -7.9  -8.5  -6.3  -4.3  -4.2  -4.8  -2.4 -10.7 -12.0  -9.7  -9.7   3.0  12.9   2.1  14.7  -4.9  -3.8  -4.0  -2.4  -1.0   0.8  -1.3   3.4  -7.6  -5.7  -8.1  -3.5  -6.8  -5.8  -7.2  -4.2  -8.9 -10.2  -8.4  -8.2  -9.9  -8.4  -9.7  -6.4 -11.1 -11.1 -11.8 -10.0 -50.0   0.4 -50.0   2.6  -7.2  -7.0  -7.4  -5.1 -50.0  -5.7  -9.8  -3.2  -8.4  -7.4  -8.9  -5.9\n",
      "CCA  -8.5 -10.3 -10.0  -9.0  -1.4  -4.9  -3.9  -3.7 -10.0  -8.5 -10.2  -7.8  -7.8 -12.4  -8.5 -11.1   0.0  -5.5  -2.3  -4.9  12.6  10.0  11.1  10.5  -6.0  -8.3  -6.9  -7.5  -2.2  -7.4  -5.3  -5.9  -8.9 -12.5  -9.4 -11.6  -0.7  -3.9  -2.3  -3.0  -9.1  -9.8  -9.2  -9.5  -5.6  -9.0  -8.0  -8.5 -50.0 -15.3 -50.0 -14.1   2.3  -1.4   0.3  -0.5 -50.0 -13.6 -14.3 -11.5  -5.3 -13.7  -6.1 -12.7\n",
      "CCC -11.2  -8.1 -11.8  -8.8  -4.1  -1.9  -3.9  -3.5 -11.6  -5.9 -11.5  -7.9 -10.2 -10.1 -10.2 -11.2  -3.4  -2.6  -4.1  -3.8  10.0  13.1  10.7  10.6  -8.1  -5.2  -8.0  -6.7  -6.1  -3.7  -7.5  -5.6 -11.1 -10.3 -10.6 -11.9  -3.2  -1.0  -2.2  -2.8 -10.5  -8.1  -9.4  -9.5  -8.3  -6.9  -9.4  -8.8 -50.0 -11.3 -50.0 -12.8  -0.6   2.2  -0.3   0.1 -50.0 -10.1 -17.5 -11.1  -8.6  -9.9  -8.6 -11.9\n",
      "CCG  -8.9  -9.4  -8.6  -8.9  -2.9  -4.5  -2.2  -3.9  -8.8  -7.5  -8.1  -7.9  -8.8 -11.6  -7.7 -11.9  -0.7  -4.1  -0.5  -4.0  11.1  10.7  13.2  10.4  -5.7  -5.8  -3.5  -6.1  -3.6  -6.5  -3.3  -5.5  -9.4 -11.0  -8.7 -11.2  -1.8  -2.7   0.7  -3.0  -9.1  -8.2  -7.3  -9.4  -7.2  -8.2  -7.2  -8.2 -50.0 -13.2 -50.0 -13.3   0.5  -0.9   1.8  -0.6 -50.0 -11.4 -11.1 -10.5  -6.1 -12.9  -5.0 -11.9\n",
      "CCT -10.8  -9.6 -11.9  -7.7  -3.3  -3.8  -4.5  -1.6 -11.7  -7.7 -12.0  -6.2  -9.6 -12.0  -9.8  -9.4  -3.2  -4.7  -4.6  -2.4  10.5  10.6  10.4  12.6  -8.2  -7.9  -9.0  -4.7  -6.1  -6.0  -7.5  -2.9 -10.9 -12.2 -11.2 -10.9  -2.4  -2.9  -2.7  -0.9 -10.4  -9.7 -10.2  -8.0  -8.4  -9.1  -9.3  -6.5 -50.0 -13.3 -50.0 -11.0  -0.0  -0.4  -0.4   2.3 -50.0 -10.9 -17.2  -8.3  -8.3 -12.7  -7.8  -9.3\n",
      "CGA   2.1  -8.1   1.2  -7.0  -8.0  -9.5  -7.3  -9.9  10.5  -5.5   9.3  -5.7  -9.5 -15.1 -10.0 -12.9   2.5  -1.2  -0.1  -1.0  -6.0  -8.1  -5.7  -8.2  13.8  11.3  11.8  12.1  -6.0  -9.1  -9.0  -9.5  -8.9 -13.4 -10.0 -13.1  -9.9 -11.7  -9.8 -12.8  -5.9  -9.4  -6.5  -8.1 -10.7 -14.0 -13.3 -13.9 -50.0 -11.9 -50.0  -9.7  -8.9 -10.8  -9.0 -11.1 -50.0  -6.5  -5.3  -6.3 -10.5 -16.2 -10.4 -15.2\n",
      "CGC   0.0  -5.0   0.5  -6.8  -9.1  -7.8  -7.5  -8.9   7.7  -2.4   8.2  -5.1 -11.9 -13.1 -10.4 -14.3  -0.8   3.0  -0.9   0.8  -8.3  -5.2  -5.8  -7.9  11.3  15.0  11.2  12.8  -9.7  -5.3  -9.2  -7.8 -11.7 -10.6 -10.1 -12.2 -11.7  -8.8  -8.9 -11.8  -9.3  -5.0  -8.0  -8.2 -13.6 -11.4 -12.8 -13.0 -50.0  -7.2 -50.0  -8.4 -11.0  -7.7  -8.7 -10.2 -50.0  -1.5  -7.1  -4.9 -11.2 -11.6 -11.6 -15.0\n",
      "CGG   1.4  -7.3   2.2  -7.1  -8.0  -9.3  -6.4  -9.6   9.1  -5.5  10.0  -5.4 -10.9 -14.3  -8.4 -14.0   0.9  -1.1   2.4  -1.3  -6.9  -8.0  -3.5  -9.0  11.8  11.2  13.4  11.4  -7.3  -8.8  -6.5  -8.5 -10.0 -12.5  -7.8 -12.5  -9.6 -11.0  -7.1 -10.7  -7.9  -8.0  -4.3  -9.4 -11.4 -12.3 -10.9 -12.9 -50.0 -11.8 -50.0 -11.2  -9.4  -9.8  -7.3 -10.5 -50.0  -6.2  -2.2  -6.7  -9.8 -15.2  -8.5 -14.9\n",
      "CGT   0.2  -6.3   0.1  -4.3  -8.2  -8.9  -8.5  -7.0   8.5  -4.1   8.0  -2.1 -10.5 -13.4 -10.2 -11.9   0.2   1.6  -0.8   3.4  -7.5  -6.7  -6.1  -4.7  12.1  12.8  11.4  14.7  -8.7  -7.4  -9.4  -5.3  -9.9 -11.8 -10.5 -10.3  -9.9 -10.1  -9.3  -9.3  -8.1  -7.2  -8.0  -5.2 -12.0 -12.1 -12.6 -10.9 -50.0  -7.9 -50.0  -5.3  -9.3  -8.8  -8.9  -7.0 -50.0  -3.6  -7.1  -1.2 -10.2 -12.7  -9.9 -11.0\n",
      "CTA -10.2 -13.4 -11.8 -12.4  -6.0  -9.4  -7.1  -8.6  -9.8 -12.9 -10.8 -12.4   2.3  -1.4   1.0  -1.0  -3.0  -8.0  -5.1  -7.6  -2.2  -6.1  -3.6  -6.1  -6.0  -9.7  -7.3  -8.7  11.2   7.9   8.9   8.1 -12.1 -17.4 -13.2 -16.8  -6.3  -9.5  -7.5  -8.8 -12.7 -15.2 -13.1 -14.2   0.0  -3.1  -1.8  -3.1 -50.0 -10.7 -50.0  -9.3  -4.6  -8.9  -5.9  -8.3 -50.0 -12.8  -9.6 -10.6   9.5  -3.2   8.2  -2.8\n",
      "CTC -13.5 -11.3 -14.0 -12.9  -9.3  -7.1  -8.2  -8.9 -12.7 -10.3 -11.9 -12.3  -0.4   1.4  -0.6  -0.4  -7.2  -4.7  -6.9  -5.7  -7.4  -3.7  -6.5  -6.0  -9.1  -5.3  -8.8  -7.4   7.9  11.9   7.8   9.3 -15.2 -15.1 -14.6 -17.2  -9.7  -7.3  -8.2  -9.2 -15.0 -12.8 -14.1 -14.9  -2.8   0.2  -2.9  -2.3 -50.0  -7.4 -50.0  -8.7  -9.4  -7.3  -9.4  -9.4 -50.0  -8.0 -11.9  -9.7   6.5   0.5   6.4  -1.8\n",
      "CTG -13.0 -14.4 -11.9 -13.9  -7.7  -9.0  -6.7  -8.8 -11.7 -12.7 -10.7 -13.1  -0.2  -1.6   1.7  -1.6  -5.7  -8.7  -4.8  -8.1  -5.3  -7.5  -3.3  -7.5  -9.0  -9.2  -6.5  -9.4   8.9   7.8  10.1   7.8 -14.3 -17.7 -13.0 -17.8  -8.0  -9.4  -6.2  -9.2 -15.1 -14.9 -12.2 -14.9  -1.8  -3.2  -0.8  -3.2 -50.0 -10.8 -50.0  -9.9  -7.3  -9.6  -6.4  -9.4 -50.0 -12.3  -8.7 -11.3   7.6  -3.3   8.7  -2.9\n",
      "CTT -12.5 -12.9 -13.4 -10.0  -8.5  -9.0  -7.8  -6.8 -11.8 -11.3 -12.4 -10.1  -0.3  -0.5  -0.5   1.2  -6.1  -6.0  -6.7  -3.5  -5.9  -5.6  -5.5  -2.9  -9.5  -7.8  -8.5  -5.3   8.1   9.3   7.8  11.8 -14.4 -15.7 -14.5 -14.6  -8.4  -8.7  -7.9  -7.1 -14.3 -13.8 -13.8 -12.3  -2.4  -2.2  -2.9  -0.2 -50.0  -8.9 -50.0  -7.1  -8.1  -9.1  -8.7  -6.6 -50.0  -9.7 -11.4  -8.0   6.8  -1.8   6.9   0.4\n",
      "GAA  -2.6  -6.3  -4.9  -4.9  -6.4  -9.4  -7.9  -8.6  -6.3  -7.4  -8.2  -6.6  -9.7 -14.9 -11.1 -13.9  -0.2  -8.2  -2.3  -6.8  -8.9 -11.1  -9.4 -10.9  -8.9 -11.7 -10.0  -9.9 -12.1 -15.2 -14.3 -14.4  11.1   2.9   9.2   3.5  -3.0  -7.0  -4.3  -6.2  -2.2  -6.6  -3.7  -5.7  -6.0 -10.4  -8.6  -9.5 -50.0 -15.5 -50.0 -13.9  -7.8 -10.5  -8.7  -9.6 -50.0 -16.5 -17.1 -14.8 -12.1 -18.9 -12.4 -17.4\n",
      "GAC  -8.5   0.8  -9.1  -0.7 -10.1  -7.5  -9.1  -8.2 -11.4  -3.1 -11.8  -4.4 -15.5 -14.3 -14.8 -15.4  -6.6  -4.7  -7.1  -5.8 -12.5 -10.3 -11.0 -12.2 -13.4 -10.6 -12.5 -11.8 -17.4 -15.1 -17.7 -15.7   2.9  12.5   3.1  10.3  -7.7  -5.2  -6.2  -7.3  -5.4  -1.7  -5.4  -3.7 -11.6  -9.1 -13.0 -10.8 -50.0 -10.3 -50.0 -11.2  -9.8  -8.3  -9.1  -9.7 -50.0 -12.7 -20.6 -13.1 -16.0 -16.6 -17.0 -18.4\n",
      "GAG  -5.0  -6.4  -3.4  -5.6  -7.5  -9.3  -7.1  -8.8  -8.8  -7.4  -6.9  -6.9 -11.4 -15.2  -9.7 -13.7  -1.8  -7.8  -0.8  -7.2  -9.4 -10.6  -8.7 -11.2 -10.0 -10.1  -7.8 -10.5 -13.2 -14.6 -13.0 -14.5   9.2   3.1  10.8   3.0  -4.5  -6.7  -2.7  -6.7  -5.0  -6.8  -2.5  -6.7  -7.5 -10.4  -7.4 -10.2 -50.0 -15.5 -50.0 -14.2  -8.9 -10.2  -8.5 -10.0 -50.0 -15.6 -15.3 -15.5 -13.0 -18.8 -12.6 -18.3\n",
      "GAT  -8.1  -1.1  -8.9   1.4  -8.6  -9.2  -9.4  -6.8 -11.0  -4.9 -11.5  -2.6 -13.7 -16.8 -14.5 -12.6  -6.0  -6.7  -7.2  -4.2 -11.6 -11.9 -11.2 -10.9 -13.1 -12.2 -12.5 -10.3 -16.8 -17.2 -17.8 -14.6   3.5  10.3   3.0  12.4  -6.9  -7.5  -6.8  -4.9  -5.4  -4.2  -5.4  -1.3 -10.8 -11.5 -12.4  -8.2 -50.0 -12.5 -50.0  -8.9  -9.2 -10.0  -9.4  -7.9 -50.0 -14.3 -19.9 -11.4 -16.2 -19.2 -15.8 -15.9\n",
      "GCA  -6.3  -7.4  -8.0  -6.0   2.8  -0.7   0.9   0.3  -7.5  -4.4  -8.4  -3.6  -3.0  -7.7  -3.7  -6.2  -5.3 -10.0  -7.2  -8.9  -0.7  -3.2  -1.8  -2.4  -9.9 -11.7  -9.6  -9.9  -6.3  -9.7  -8.0  -8.4  -3.0  -7.7  -4.5  -6.9  11.3   8.2   9.4   9.1  -1.0  -3.3  -1.8  -2.4   1.5  -2.1  -0.4  -1.1 -50.0 -14.5 -50.0 -12.9   2.8  -0.0   1.4   0.7 -50.0  -8.7 -14.1  -7.5  -5.2 -12.5  -6.5 -10.7\n",
      "GCC  -9.9  -5.0 -10.0  -6.5  -0.7   2.6  -0.2   0.1 -10.3  -1.8  -9.9  -3.8  -6.4  -4.7  -5.8  -6.7  -8.2  -8.3  -8.0 -10.2  -3.9  -1.0  -2.7  -2.9 -11.7  -8.8 -11.0 -10.1  -9.5  -7.3  -9.4  -8.7  -7.0  -5.2  -6.7  -7.5   8.2  11.6   8.8   9.1  -3.7  -0.9  -3.1  -3.0  -2.0   1.2  -2.3  -1.3 -50.0 -12.1 -50.0 -12.6   0.1   2.5   0.7   0.6 -50.0  -6.0 -14.8  -7.5  -8.6  -9.4  -8.5 -11.2\n",
      "GCG  -7.5  -6.2  -7.2  -6.2   0.9   0.2   2.5   0.2  -8.3  -3.5  -7.1  -3.7  -4.1  -6.3  -3.1  -6.3  -5.7  -8.9  -5.6  -8.4  -2.3  -2.2   0.7  -2.7  -9.8  -8.9  -7.1  -9.3  -7.5  -8.2  -6.2  -7.9  -4.3  -6.2  -2.7  -6.8   9.4   8.8  12.1   8.8  -2.1  -2.0   0.1  -2.4   0.0  -1.1   1.3  -0.8 -50.0 -11.9 -50.0 -12.6   1.2   0.7   3.5   0.8 -50.0  -7.1 -12.4  -7.1  -7.1 -10.6  -5.0 -10.6\n",
      "GCT  -9.0  -6.5 -10.0  -4.8   0.1  -0.1  -0.3   2.4  -9.9  -3.5 -10.6  -2.0  -5.4  -6.6  -5.6  -4.4  -7.6  -9.9  -8.5  -8.2  -3.0  -2.8  -3.0  -0.9 -12.8 -11.8 -10.7  -9.3  -8.8  -9.2  -9.2  -7.1  -6.2  -7.3  -6.7  -4.9   9.1   9.1   8.8  11.2  -3.1  -3.0  -3.3  -1.0  -1.1  -1.4  -2.0   1.2 -50.0 -12.8 -50.0 -10.9   0.8   0.5   0.7   2.3 -50.0  -7.5 -16.5  -5.9  -7.8 -11.1  -7.8  -9.0\n",
      "GGA  -7.1  -5.6  -9.1  -5.1  -5.9  -8.0  -6.9  -7.4  -1.9  -1.6  -4.7  -1.4  -9.6 -14.6 -10.4 -13.0  -6.7 -10.8  -9.4  -9.9  -9.1 -10.5  -9.1 -10.4  -5.9  -9.3  -7.9  -8.1 -12.7 -15.0 -15.1 -14.3  -2.2  -5.4  -5.0  -5.4  -1.0  -3.7  -2.1  -3.1  12.8   9.6  11.1  10.1  -4.8  -9.4  -8.1  -8.6 -50.0 -18.3 -50.0 -15.4  -5.9  -7.5  -6.6  -7.1 -50.0  -9.5 -11.7  -8.4 -12.5 -17.3 -13.2 -15.4\n",
      "GGC -10.2  -1.6 -10.2  -3.2  -7.3  -4.6  -6.5  -6.2  -6.7   3.0  -6.8   0.8 -11.9 -10.8 -11.5 -12.2  -9.6  -7.1  -9.7  -8.4  -9.8  -8.1  -8.2  -9.7  -9.4  -5.0  -8.0  -7.2 -15.2 -12.8 -14.9 -13.8  -6.6  -1.7  -6.8  -4.2  -3.3  -0.9  -2.0  -3.0   9.6  12.8   9.7  10.5  -8.5  -5.6  -9.3  -7.8 -50.0 -12.7 -50.0 -13.8  -6.8  -5.1  -5.3  -6.5 -50.0  -4.1 -12.8  -5.8 -14.4 -14.1 -13.6 -14.8\n",
      "GGG  -8.2  -4.7  -7.1  -5.0  -6.5  -7.7  -5.5  -6.9  -4.1  -0.8  -1.2  -0.9 -10.5 -12.8  -8.2 -13.0  -7.5 -10.0  -7.2  -9.7  -9.2  -9.4  -7.3 -10.2  -6.5  -8.0  -4.3  -8.0 -13.1 -14.1 -12.2 -13.8  -3.7  -5.4  -2.5  -5.4  -1.8  -3.1   0.1  -3.3  11.1   9.7  12.9   9.8  -5.5  -8.1  -5.4  -7.8 -50.0 -16.4 -50.0 -14.8  -5.9  -7.1  -4.7  -7.2 -50.0  -8.4  -6.4  -8.1 -12.1 -16.2 -10.3 -15.8\n",
      "GGT  -9.2  -3.0  -9.9  -1.2  -6.5  -6.1  -6.8  -4.4  -6.4   0.9  -7.0   3.0 -11.8 -13.0 -11.1 -10.4  -9.3  -9.2  -9.8  -6.4  -9.5  -9.5  -9.4  -8.0  -8.1  -8.2  -9.4  -5.2 -14.2 -14.9 -14.9 -12.3  -5.7  -3.7  -6.7  -1.3  -2.4  -3.0  -2.4  -1.0  10.1  10.5   9.8  13.1  -7.2  -8.0  -8.4  -5.0 -50.0 -14.7 -50.0 -11.4  -6.3  -6.7  -5.8  -5.0 -50.0  -6.1 -13.4  -3.6 -13.0 -16.0 -13.7 -13.3\n",
      "GTA  -8.2 -10.8 -10.1  -9.8  -0.4  -3.6  -1.7  -2.9  -7.9  -8.4  -9.8  -8.2   6.2   2.3   0.6   2.9  -7.1 -11.6  -9.2 -11.1  -5.6  -8.3  -7.2  -8.4 -10.7 -13.6 -11.4 -12.0   0.0  -2.8  -1.8  -2.4  -6.0 -11.6  -7.5 -10.8   1.5  -2.0   0.0  -1.1  -4.8  -8.5  -5.5  -7.2  11.9   8.6  10.0   9.0 -50.0 -14.2 -50.0 -12.8  -4.1  -7.4  -5.7  -6.5 -50.0 -11.0 -14.3  -9.9   1.4  -7.9  -0.7  -6.8\n",
      "GTC -12.5  -8.7 -13.0  -9.9  -3.9  -1.0  -3.4  -2.9 -12.5  -6.3 -11.6  -8.1   3.3   6.0  -1.6   3.8 -10.5 -10.1 -11.1 -11.1  -9.0  -6.9  -8.2  -9.1 -14.0 -11.4 -12.3 -12.1  -3.1   0.2  -3.2  -2.2 -10.4  -9.1 -10.4 -11.5  -2.1   1.2  -1.1  -1.4  -9.4  -5.6  -8.1  -8.0   8.6  12.4   8.6   9.5 -50.0 -10.4 -50.0 -11.6  -7.2  -5.7  -7.5  -7.4 -50.0  -8.6 -15.7  -9.7  -3.0  -3.7  -3.2  -5.9\n",
      "GTG -11.1 -11.9 -10.6 -11.3  -2.6  -4.1  -1.7  -3.5 -11.4  -9.6  -9.6  -9.2   3.7   2.5   1.6   2.5  -9.6 -12.3  -9.5 -11.8  -8.0  -9.4  -7.2  -9.3 -13.3 -12.8 -10.9 -12.6  -1.8  -2.9  -0.8  -2.9  -8.6 -13.0  -7.4 -12.4  -0.4  -2.3   1.3  -2.0  -8.1  -9.3  -5.4  -8.4  10.0   8.6  11.4   8.9 -50.0 -13.7 -50.0 -13.2  -6.1  -8.1  -5.5  -7.5 -50.0 -11.3 -12.7 -10.1  -1.6  -7.8  -0.3  -7.2\n",
      "GTT -11.4 -10.0 -12.7  -8.2  -3.1  -3.1  -2.8  -0.9 -11.7  -7.3 -12.1  -5.8   3.6   3.6  -1.1   5.8  -9.4 -11.9 -10.7 -10.0  -8.5  -8.8  -8.2  -6.5 -13.9 -13.0 -12.9 -10.9  -3.1  -2.3  -3.2  -0.2  -9.5 -10.8 -10.2  -8.2  -1.1  -1.3  -0.8   1.2  -8.6  -7.8  -7.8  -5.0   9.0   9.5   8.9  12.0 -50.0 -13.0 -50.0 -10.3  -6.2  -7.0  -7.1  -5.4 -50.0  -9.3 -14.0  -8.2  -2.4  -6.3  -2.4  -3.9\n",
      "TAA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0  30.6 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAC -14.8  -6.2 -14.9  -7.6 -14.6 -11.6 -13.0 -12.2 -14.0  -9.0 -14.1 -10.1 -13.6 -11.2 -12.4 -12.5  -8.1   2.2  -8.7   0.4 -15.3 -11.3 -13.2 -13.3 -11.9  -7.2 -11.8  -7.9 -10.7  -7.4 -10.8  -8.9 -15.5 -10.3 -15.5 -12.5 -14.5 -12.1 -11.9 -12.8 -18.3 -12.7 -16.4 -14.7 -14.2 -10.4 -13.7 -13.0 -50.0  15.1 -50.0  13.3  -9.9  -5.8  -8.9  -7.6 -50.0  -1.5  -7.8  -3.0  -8.1   3.6  -9.4   2.0\n",
      "TAG -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  30.6 -50.0  35.2 -50.0 -50.0 -50.0 -50.0 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TAT -13.8  -7.5 -14.4  -5.1 -12.6 -11.6 -11.5 -10.1 -13.2  -9.7 -12.7  -7.4 -11.9 -11.9 -11.5 -10.2  -7.2   0.9  -8.6   2.6 -14.1 -12.8 -13.3 -11.0  -9.7  -8.4 -11.2  -5.3  -9.3  -8.7  -9.9  -7.1 -13.9 -11.2 -14.2  -8.9 -12.9 -12.6 -12.6 -10.9 -15.4 -13.8 -14.8 -11.4 -12.8 -11.6 -13.2 -10.3 -50.0  13.3 -50.0  15.2  -8.6  -7.0  -8.4  -4.8 -50.0  -3.0  -7.1  -0.1  -7.3   2.3  -8.7   3.9\n",
      "TCA  -7.3  -6.3  -8.9  -5.5   2.9   0.2   1.5   1.2  -8.5  -2.3 -10.0  -1.7  -5.6 -10.5  -5.5  -9.6  -4.3  -7.9  -6.1  -7.2   2.3  -0.6   0.5  -0.0  -8.9 -11.0  -9.4  -9.3  -4.6  -9.4  -7.3  -8.1  -7.8  -9.8  -8.9  -9.2   2.8   0.1   1.2   0.8  -5.9  -6.8  -5.9  -6.3  -4.1  -7.2  -6.1  -6.2 -50.0  -9.9 -50.0  -8.6  12.5   9.4  11.0   9.8 -50.0  -4.8  -8.7  -3.5  -0.0  -8.6  -2.1  -7.0\n",
      "TCC -10.1  -4.3 -10.8  -5.8   0.1   2.3   0.7   0.6 -10.4  -0.2 -10.8  -2.0  -8.8  -9.2  -8.0  -9.7  -6.8  -5.9  -7.5  -7.0  -1.4   2.2  -0.9  -0.4 -10.8  -7.7  -9.8  -8.8  -8.9  -7.3  -9.6  -9.1 -10.5  -8.3 -10.2 -10.0  -0.0   2.5   0.7   0.5  -7.5  -5.1  -7.1  -6.7  -7.4  -5.7  -8.1  -7.0 -50.0  -5.8 -50.0  -7.0   9.4  12.7  10.1  10.0 -50.0  -1.5 -11.3  -3.3  -5.4  -4.2  -5.6  -6.8\n",
      "TCG  -8.4  -6.2  -8.9  -5.9   1.8   0.4   2.7   1.2  -9.1  -1.2  -8.8  -1.5  -7.2 -10.5  -4.9  -9.6  -5.7  -7.4  -5.5  -7.4   0.3  -0.3   1.8  -0.4  -9.0  -8.7  -7.3  -8.9  -5.9  -9.4  -6.4  -8.7  -8.7  -9.1  -8.5  -9.4   1.4   0.7   3.5   0.7  -6.6  -5.3  -4.7  -5.8  -5.7  -7.5  -5.5  -7.1 -50.0  -8.9 -50.0  -8.4  11.0  10.1  13.2  10.2 -50.0  -4.2  -6.1  -3.6  -3.5  -8.4  -1.2  -6.8\n",
      "TCT  -9.1  -5.4 -10.4  -4.2   1.0   0.5   0.6   2.4  -9.6  -1.5 -10.5  -0.5  -8.9 -10.5  -7.2  -8.3  -6.3  -6.9  -7.7  -5.1  -0.5   0.1  -0.6   2.3 -11.1 -10.2 -10.5  -7.0  -8.3  -9.4  -9.4  -6.6  -9.6  -9.7 -10.0  -7.9   0.7   0.6   0.8   2.3  -7.1  -6.5  -7.2  -5.0  -6.5  -7.4  -7.5  -5.4 -50.0  -7.6 -50.0  -4.8   9.8  10.0  10.2  12.1 -50.0  -3.1 -11.4  -0.7  -4.6  -6.6  -4.9  -3.9\n",
      "TGA -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0  29.2 -50.0  28.5 -50.0 -50.0 -50.0 -50.0 -50.0  33.3 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0 -50.0\n",
      "TGC -13.0  -7.0 -13.1  -8.3  -9.2  -6.5  -8.9  -7.6  -9.8  -0.7  -9.2  -2.1 -12.4 -10.3 -12.4 -11.6  -9.5  -4.6 -10.4  -5.7 -13.6 -10.1 -11.4 -10.9  -6.5  -1.5  -6.2  -3.6 -12.8  -8.0 -12.3  -9.7 -16.5 -12.7 -15.6 -14.3  -8.7  -6.0  -7.1  -7.5  -9.5  -4.1  -8.4  -6.1 -11.0  -8.6 -11.3  -9.3 -50.0  -1.5 -50.0  -3.0  -4.8  -1.5  -4.2  -3.1 -50.0  16.4  -5.1  14.2 -10.1  -4.3 -10.0  -5.4\n",
      "TGG -13.5 -16.3 -11.8 -15.5 -13.6 -15.3 -12.1 -16.1  -7.7 -13.3  -4.2 -12.6 -14.1 -16.2 -10.3 -15.4  -9.6 -11.6  -7.7  -9.8 -14.3 -17.5 -11.1 -17.2  -5.3  -7.1  -2.2  -7.1  -9.6 -11.9  -8.7 -11.4 -17.1 -20.6 -15.3 -19.9 -14.1 -14.8 -12.4 -16.5 -11.7 -12.8  -6.4 -13.4 -14.3 -15.7 -12.7 -14.0 -50.0  -7.8 -50.0  -7.1  -8.7 -11.3  -6.1 -11.4 -50.0  -5.1  18.6  -4.8  -8.6  -8.1  -4.5  -7.3\n",
      "TGT -12.4  -8.2 -13.0  -5.9  -7.7  -7.8  -7.4  -5.8  -8.8  -2.0  -9.3  -0.4 -11.6 -11.3 -11.6  -9.4  -8.9  -5.6 -10.0  -3.2 -11.5 -11.1 -10.5  -8.3  -6.3  -4.9  -6.7  -1.2 -10.6  -9.7 -11.3  -8.0 -14.8 -13.1 -15.5 -11.4  -7.5  -7.5  -7.1  -5.9  -8.4  -5.8  -8.1  -3.6  -9.9  -9.7 -10.1  -8.2 -50.0  -3.0 -50.0  -0.1  -3.5  -3.3  -3.6  -0.7 -50.0  14.2  -4.8  16.4  -8.9  -5.8  -8.9  -3.4\n",
      "TTA -10.7 -13.2 -12.4 -11.6  -4.8  -8.4  -6.1  -7.2 -10.3 -11.8 -11.0 -11.6   2.8  -1.7   0.7  -1.1  -6.2  -9.7  -7.3  -8.4  -5.3  -8.6  -6.1  -8.3 -10.5 -11.2  -9.8 -10.2   9.5   6.5   7.6   6.8 -12.1 -16.0 -13.0 -16.2  -5.2  -8.6  -7.1  -7.8 -12.5 -14.4 -12.1 -13.0   1.4  -3.0  -1.6  -2.4 -50.0  -8.1 -50.0  -7.3  -0.0  -5.4  -3.5  -4.6 -50.0 -10.1  -8.6  -8.9  13.2  -0.9   9.7  -0.3\n",
      "TTC -18.1 -12.3 -19.4 -14.2 -12.2  -9.5 -11.5 -10.8 -17.9 -11.9 -18.2 -13.9  -6.4  -3.7  -6.8  -5.4 -12.8  -5.9 -13.6  -7.4 -13.7  -9.9 -12.9 -12.7 -16.2 -11.6 -15.2 -12.7  -3.2   0.5  -3.3  -1.8 -18.9 -16.6 -18.8 -19.2 -12.5  -9.4 -10.6 -11.1 -17.3 -14.1 -16.2 -16.0  -7.9  -3.7  -7.8  -6.3 -50.0   3.6 -50.0   2.3  -8.6  -4.2  -8.4  -6.6 -50.0  -4.3  -8.1  -5.8  -0.9  14.2  -1.8  11.6\n",
      "TTG -11.8 -13.1 -11.5 -12.0  -6.2  -8.3  -5.4  -7.4 -11.8 -12.3 -11.1 -11.4   0.5  -2.5   2.4  -1.6  -6.8  -9.9  -6.4  -8.9  -6.1  -8.6  -5.0  -7.8 -10.4 -11.6  -8.5  -9.9   8.2   6.4   8.7   6.9 -12.4 -17.0 -12.6 -15.8  -6.5  -8.5  -5.0  -7.8 -13.2 -13.6 -10.3 -13.7  -0.7  -3.2  -0.3  -2.4 -50.0  -9.4 -50.0  -8.7  -2.1  -5.6  -1.2  -4.9 -50.0 -10.0  -4.5  -8.9   9.7  -1.8  11.3  -0.9\n",
      "TTT -17.2 -13.3 -17.8 -11.6 -11.0 -11.1 -10.4  -9.0 -16.0 -12.7 -16.1 -11.1  -5.2  -5.4  -6.0  -3.3 -11.8  -7.3 -13.3  -5.9 -12.7 -11.9 -11.9  -9.3 -15.2 -15.0 -14.9 -11.0  -2.8  -1.8  -2.9   0.4 -17.4 -18.4 -18.3 -15.9 -10.7 -11.2 -10.6  -9.0 -15.4 -14.8 -15.8 -13.3  -6.8  -5.9  -7.2  -3.9 -50.0   2.0 -50.0   3.9  -7.0  -6.8  -6.8  -3.9 -50.0  -5.4  -7.3  -3.4  -0.3  11.6  -0.9  14.1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "matrix = substitution_matrices.load(\"SCHNEIDER\")\n",
    "print(matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligner = Align.PairwiseAligner(matrix=matrix)\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: 0.000000\n",
      "  target_internal_open_gap_score: 0.000000\n",
      "  target_internal_extend_gap_score: 0.000000\n",
      "  target_left_open_gap_score: 0.000000\n",
      "  target_left_extend_gap_score: 0.000000\n",
      "  target_right_open_gap_score: 0.000000\n",
      "  target_right_extend_gap_score: 0.000000\n",
      "  query_internal_open_gap_score: 0.000000\n",
      "  query_internal_extend_gap_score: 0.000000\n",
      "  query_left_open_gap_score: 0.000000\n",
      "  query_left_extend_gap_score: 0.000000\n",
      "  query_right_open_gap_score: 0.000000\n",
      "  query_right_extend_gap_score: 0.000000\n",
      "  mode: global\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(aligner)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "OverflowError",
     "evalue": "number of optimal alignments is larger than 9223372036854775807",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOverflowError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-e0f92bf7d593>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malignments\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/Bio/Align/__init__.py\u001b[0m in \u001b[0;36m__len__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1345\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1346\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1347\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpaths\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1348\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1349\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOverflowError\u001b[0m: number of optimal alignments is larger than 9223372036854775807"
     ]
    }
   ],
   "source": [
    "print(len(alignments))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: -2.000000\n",
      "  target_internal_open_gap_score: -2.500000\n",
      "  target_internal_extend_gap_score: -2.500000\n",
      "  target_left_open_gap_score: -2.500000\n",
      "  target_left_extend_gap_score: -2.500000\n",
      "  target_right_open_gap_score: -2.500000\n",
      "  target_right_extend_gap_score: -2.500000\n",
      "  query_internal_open_gap_score: -2.500000\n",
      "  query_internal_extend_gap_score: -2.500000\n",
      "  query_left_open_gap_score: -2.500000\n",
      "  query_left_extend_gap_score: -2.500000\n",
      "  query_right_open_gap_score: -2.500000\n",
      "  query_right_extend_gap_score: -2.500000\n",
      "  mode: global\n",
      "\n"
     ]
    }
   ],
   "source": [
    "aligner = Align.PairwiseAligner()\n",
    "aligner.match_score = 1.0\n",
    "aligner.mismatch_score = -2.0\n",
    "aligner.gap_score = -2.5\n",
    "print(aligner)\n",
    "alignments = aligner.align(seq1, seq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19106.5\n"
     ]
    }
   ],
   "source": [
    "print(alignments[0].score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATATTA-GGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGACTGTCACTTAGCTGCATGCTTAGTGCACTCACGCAGTTTAATTA-TAATTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGTTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCATACCTTGGTTTCGTCCGGGTGTGACCGAGAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGGTCCA-CTCAGTTTGCCTGTTTTACAGGTTCGTGACGTGCTTGTACGTGGCTTTGGAGACTCCGTAGAGGAAG-CTTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGTGTTTTACCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCTGATGCCCGAACTGCACCTCACGGCCATGTTATGGTCGAATTAGTAGCAGAACTCGATGGCATTCAGTATGGTCGTAGTGGTGAGACACTTGGTGTTCTTGTCCCTCATGTAGGAGAGGTACCAGTTGCTTACCGTAAAGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCTTTTGACTTAGGCGACGAGCTTGGTACTGATCCT-ATTGAAGATTTTCAAGAAAATTGGAACACTAAACATGGCAGTGGTGTTACCCGTGAACTCAAGCGTGAGCTTAACGGAGGTGCATACACTCGCTATGTAGACAACAACTTTTGTGGCCCAGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTGGCTCGTGCTGGTAAGGCTTCTTATGC--TTTGTCTGAACAATTGGATTTTATTGACACTAAGAGAGGTGTGTACTGCTGCCGTGAACACGAGCATGAAATTGCTTGGTACACGGAACGCTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAGTTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCTCT-AAATTCAACAATCAAGACCATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGG-TTTCATGGGTAGAATTCGATCTGTCTATCCTGTTGCTTCACCAAATGAATGCAACCAAATGTGCCTCTCAATTCTCATGAAGTGTGACCATTGTGGTGAAACTTCATGGCAGACGGGTGATTTTGTTAGAGCCACTTGCGAATTCTGTGGTACTGAAAATTTGACTAAAGAAGGTGCCACAACTTGTGGTTACTTACCTCAAAATGCTGTTGT-AAAACTTTATTGTCCAGCATGTCATAATCCAGAAGTAGGACCTGAGCATAGTCTTGCTGAATATCATAATGAGTCTGGTTTGAAAACCGTTCTTCGTAAGGGTGGTCGTACCATTGC-TTATGGGGGCTGTGTGTT-TGCTTATGTTGGTTGCTACAACAAGTGTGCCTATTGGGTTCCACGTGCTAGTGCTAACATAGGCTGTAATCACACAGGTGTTGTTGGAGAAGGTTCTGAAAGTCTAAACGACAACCTTCTTGAAATATTGCAAAAGGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATTGCCATTATTTTGGCATCTTTTTCTGCCTCTATAAGTGCTTTTGTAGAAACTGTAAAAGGTTTGGATTACAAAACATTCAAACAAATTGTTGAATCCTGTGGTAACTTTAAAGTTACGAAGGGAAAAGC-AAAGAAAGGTGCCTGGAACATAGGTGAACA-AAGTTCAATACTGAGTCCGCTCTATGCGTTCCCTTCAGATGCTGCTCATGTTGTACGGTCTATCTTCTCACGCACACTAGAAACTGCTCACCATTCTGTGCATGTCTTACAAAAGGCTGCTATAATTAT-CTTAGATGGAATTTCACAGTATTCATTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGTTACTAACAATCTAGTTGTAATGGCTTACATTACGGGTGGTGTTGTACAAATGACTTCACAGTGGCTAACAAATATCTTTGGCACTGTTTATGAAAAACTTAAACCGGTTCTTGATTGGCTCGAAGAGAAATTCAAAGAAGGTATAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACTTGTGCTTGTGAAATAGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTCGTAAATAAATTTTTGGCTTTGTGTGCTGACTCCATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACACACTCAAAGGGATTGTACAGAAAGTGTGTTAGATCCAGAGAAGAAACTGGCTTACTCTTGCCTCTGAAAGCTCCAAAAGAAATTATTTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGTTTTACAACCATTAGAACAACCTACTAATGAGGCTGTTGAAGCTCCATTGATTGGTACACCAGTCTGTATTAACGGGCTCATGTTGCTCGAAATTAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAATAATACCTTCACACTTAAAGGTGGTGCACCAACAAAAGTCACTTTTGGTGATGACACTGTGATTGAAGTGCAGGGTTACAAGAGTGTAAACATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTAATTACACAGTTGAACTCGGTACAGAGGTAAACGAGTTCGCTTGTGTTGTTGCTGATGCTGTCATAAAAACTTTACAACCAGTATCTGAACTAATCATACCACTGGGCATTGATTTAGACGAGTGGAGTATGGCTACATACTACTTGTTTGATGAGTCCGGTGAATTTAAATTGTCTTCACATATGTACTGTTCTTTCTACCCTCCTGA--AG-ATGAAGGGGAAGATGATTGTGAAGAAGGACAGTGTGAACCATCAACTCAATATGAGTATGGTACTGAGGATGACTACCAAGGTAAACCTTTGGAGTTTGGTGCTACTTCTTTTTCTTCTTCTTCACAGGAAGAAGAACAAGAAGAGGATTGGTTAGAATC-TGATAGTCAGGA--C-G--GC-CAAGA-GACTGCAGTTGAAGAAAAT-A-A-AA-TAC----CGAG---T-GTTGAAGTTCCACCTGTTTT-GCAGGTGGAA-TCAACACCAGTTGTTACTGAAACTAGTGAACAAAATAATTTCACAGGTTATTTAAAA-TTAACTGACAATGTCTTCATTAAAAATGCTGACATTGTAGAAGAAGCTAAAAAGGTAAAGCCTACAGTAGTTGTTAATGCAGCTAATGTTTACCTTAAACATGGAGGAGGTGTTGCTGGAGCTTTAAATAAGGCAACTAACAACGCCATGCAGGTTGAATCTGATAAGTACATAACTACCAATGGGCCACT-AATTGTGGGTGGTGGTTGTGTTTTGAGTGGACATAACCTTGCTAAAAATTGTCTTCATGTTGTTGGCCCTAATGTCAACAGAGGTGAAGACATTCAA-TTGCTTAAAAATGCTTATGAAAATTTCAATCAACATGAGATTT-TACTCGCACCATTATTATCAGCTGGTATTTTTGGTGCTGATCCTGTACATTCTTTAAGAGTTTGTGTAGAAACTGTTCACACAAATGTCTATCT-AGTTGTCTTTGACAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTAGAGATGAAGAGTGAAAAACAAGTAGAACAAAAAGTT-GCTGAAAATCCTAAAGAGGAAGTTAAGCCATTTTTTACTGAAAATAAACCTTCAGTTGAACAAAGACAACAAGCTGAAGAGAAGAAAATCAAAGCCAGTATTGAAGAAGTTACAACTACTCTAGAGGAGACCAAGTTCCTTACAGAAAACTTGTTACTTTATATTGACATCAATGGCAATCTTCACCCAGATTCTGCCACTCTTGTTAAAGATATTGACACCACTTTCTTGAAGAAAGATGTTCCATATATAGTGGGTGATGTTATTAAAGAAGGCGCTTTAACTGCTGTAGTTATACCTACTAAAAAGGCTGGTGGCACTACCGAAATGCTTGCTAAAGCTTTAAGAAAAGTGCCAACAGATAATTATATAACCACCTACCCTGGTCAGGGTTTAAATGGTTACACTGTAGAAGAAGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCTTTTTACATTTTACCATCTATTATCTCTAATGCGAAGCAAGAAATTCTAGGAACTGTTTCTTGGAACTTGCGAGAGATGCTCGCACATGCAGAAGAAACACGTAAGTTAATGCCTGTTTGTATGGAGACTAAAGCTATAGTTTCAACTATACAACGTAAGTACAAAGGCATTAAAATACAGGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACTAGTAAAACTACTGTAGCATCACTTATTAACACACTTAACAATCTAAATGAGACTCTTGTCACAATGCCA-TTAGGATATGTGACGCACGGTCT-AAATTTAGAAGAAGCTGCTCGGTACATGAGGTCTCTCAAAGTACCGGCTACAGTCTCTGTTTCTTCACCAGATGCTGTTACAGCATATAATGGTTATCTTACTTCTTCTTCAAAAACACCTGAAGAACACTTTATTGAAACCGTTTCACTTGCTGGTTCCTATAAAGATTGGTCTTATTCTGGACAGTCTACACAACTAGGCATAGAATTTCTTAAGAGAGGTGATAAGAGTGTATATTACACCAGTAATCCCACTACATTTCACCTAGATGGTGAAACTATCACCTTTGATAATCTTAAGACACTTCTCTCTTTGAGAGAAGTGAGGAATATTAAAGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGATATGTCTATGACATATGGACAACAGTTCGGTCCAATTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCACAATTCACATGAAGGTAAAACATTTTATGTTTTGCCTAATGATGATAC-CTTACGTGCAGAGGCTTTTGAGTACTACCATACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCTTTAAACCACACTAAAAAGTGGAAATACCCACAAGTAAATGGTTTAACTTCTATAAAATGGGCAGATAACAATTGTTATCTTGCTACTGCATTGTTAACAATCCAACAAATAGAGTTGAAATTTAATCCACCAGCTTTACAAGACGCCTATTATAGGGCAAGAGCTGGTGAGGCTGCTAATTTTTGTGCACTTATCCTGGCCTATTGTAATAAGACAGTAGGTGAGTTAGGTGATGTCAGAGAAACAATGAATTATTTGTTTCAACATGCCAATTTAGATTCTTGTAAAAGAGTCTTGAATGTGGTGTGTAAAACTTGTGGACAACAGCAAACAACTCTCAAGGGTGTAGAAGCTGTTATGTATATGGGCACACTTTCTTATGAACAACTTAAGAAGGGTGTGCAGATACCTTGTATGTGTGGTAAACAAGCTACACAATATCTGGTACAACAAGAGTCACCTTTTGTTATGATGTCTGCACCGCCCGCCCAATATGAACTTAAGCATGGTACATTTGTTTGTGCTAGTGAGTATACTGGTAATTACCAGTGTGGTCACTACAAACATATAACTTCTAAAGAAACCTTGTATTGCATAGATGGCGCTTTACTTACAAAGTCCTCTGAGTATAAAGGTTCTATTACAGATGTTTTCTATAAAGAAAACAGTTATACAACAACCATAAAACCAGTTACATACAAGTTGGATGGTGTTGTTTGTACAGAAATTGATCCTAAGTTGGATGGTTATTATAAGAAAGATAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCTTACCCGAATGCAAGCTTTGACAATTTTAAGTTTGTATGTGATAATACCAAATTTGCCGATGATTTAAATCAATTGTCTGGTTATAAGAAACCTGCCTCGAGAGAGCTTAAGGTTACATTCTTTCC-TGACTTAAATGGTGATGTAGTGGCTATTGATTATAAGCACTACACACCTTCTTTTAAGAAAGGAGCTAAATTGCTGCATAAGCCAATTGTTTGGCATGTTAACAATGCAACTAACAAAGCAACGTACAAACCAAATATTTGGTGCATACGTTGTCTTTGGAGTACAAAACCGGTTGAAACATCAAATTCTTTTGATGCACTGGAATT-AGGGGACACACAGGGAATGGATAATCTTGCCTGTGAAGTTCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACATTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCGGCAAGTGATGGTCTAAAAATTACAAAAGAGGTTGGTCATACAGACCTAATGGCTGCTTATGTTGACAATTCAAGTCTTACTATTAAGAAACCTAATGAATTATCCAGAGTATTAGGTTTGAAAACT-TTAGCCACTCATGGCTTGGCTGCTATTAATAGTGTTCCTTGGGACACTATAGCTAATTATGTTAAGCCTTTCCTTAATAAGGTTGTTAGCACAACTACTAACATAGTTACACGGTGTCTAAACCGTGTTTGTACTAATTATATGCCTTATTTGTTTACTTTATTGCTACAATTGTGTACTTTTACTAAAAGTACAAATTCTAGAAT-AAGAGCATCTATGCCAACCACTATAGCAAAGAATACTGTTAAAAGTGTTGGTAAATTTTGTATAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAATTGATAAATATTGTAATTTGGTTTTTATTATTAAGTGTTTGCCTAGGTTCTTTAATCTATTCAACTGCTGCTTTAGGTGTCTTAATGTCTAATTTAGGCATGCCTTCTTATTGTACTGTTTACAGAGAAGGTTATTTGAACTCTACTAATGTCACTACTGCAACCTACTGCACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTGGATACTTACCCATCCTTAGAAACTATACAAATTACCATTTCGTCTTTTAAATGGGATTTAACTGCTTTTGGTCT-AGTTGCAGAGTGGTTTTTGGCATATATTCTTTTTACTAGGTTCTTT-TATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCAATTTCAGCTATGGTTAGAATGTATATTTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGATGGTTGTACTTCATCAACTTGTATGATGTGTTATAAACGTAATAGAGCAACAAGAGTTGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACATAACTGGAATTGTATTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGACGAAGTTGCTAGAGACTTATCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCCTATATTGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAGGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAATCTGAGAGCTAATAACACTAAGGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCTAAATCAGCGTCTGTTTATTATAGTCAGCTTATGTGTCAACCTATACTGTTACTGGACCAGGCATTAGTGTCTGATGTTGGTGACAGTGCAGAAGTTGCAGTTAAGATGTTTGATGCTTATGTTAATATATTTTCATCAACTTTCAATGTTCCAATGGAAAAACTCAAAGCATTAGTTGCGACTGCAGAAGCTGAACTTGCAAAGAATGTGTCTTTAGACAACGTCTTATCTACTTTTATTTCAGCAGCTCGTCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACACCAATCTGACATAGAAGTTACAGGTGACAGTTGTAACAATTACATGCTCACTTATAACAAAGTTGAAAACATGACACCTCGGGATCTTGGTGCTTGTATTGATTGTAGTGCACGTCATATCAATGCACAGGTGGCAAAAAGTCATAACATAGCTTTGATTTGGAATGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGCAGTGCTGCTAAGAAGAATAACTTGCCTTTTAGA-TTGACATGTGCAACCACTAGACAAGTTGTTAATGTTGTTACAACAAAAATAGCACTTAAGGGTGGTAAAATTGTTAACAACTGGTTGAAGCAGCTGATTAAGGTTACACTTGTGTTTCTTTTCATCACTGTTATCTTCTATTTAATAACACCTGTTCATGTCATGTTTAAACACAATGACTTTTCAAGTGAAATTATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCGTCAACAGATACTTGTTTTGCTAACAAACATGCTGACTTTGACTCTTGGTTTAGTCAGCGCGGTGGTAGTTATACTAATGATAAAGCTTGCCCATTGGTAGCAGCTGTTATCACAAGGGAAGTTGGCTTTGTTGTGCCCGGTTTGCCTGGCACAATATTACGCACAATTAATGGTGATTTTTTGCATTTTCTT-CCTAGAGTGTTTAGTGCGGTAGGTAACATTTGTTACACTCCTTCTAAACTTATAGAGTACACTGACTTTGCAACATCGGCATGCGTTTTAGCTGCTGAATGTACCATTTTTAAAGATGCTTCTGGTAAACCAGTACCTTATTGTTATGATACTAATGTACTAGAAGGTTCTGTTGCGTATGAAAGTCTCCGCCCTGACACACGCTATGTGCTCATGGACGGTTCTATAATTCAATTCCCTAACACTTACCTTGAAGGTTCTGTTAGAGTAGTAACAACTTTTGATTCAGAGTATTGTAGACATGGTACTTGCGAAAGATCAGAGGCTGGCATTTGTGTATCTACTAGTGGTAGATGGGTACTTAATAATGATTATTACAGATCCTTGCCAGGAGTTTTTTGTGGTGTAGATGCTGTGAATTTACTTACTAATATGTTCACGCCATTAATTCAACCTATTGGTGCTTTGGACATATCTGCATCTATTGTAGCAGGTGGTGTTGTAGCTATTATAGTAACTTGTCTAGCCTACTACTTCATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAACACTCTACTATTCTTTATGTCATTCACTGTACTCTGTTTAACACCAGTCTATTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTCTTAGCACATATCCAATGGATGGTTATGTTCACACCCTTAGTGCCTTTCTGGATGACAATTGTTTATGTCATTTGCATTTCCACAAAGCATTTTTATTGGTTCTTTAGTAACTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACATTTGAGGAGGCTGCATTATGTACCTTTTTGTTAAATAAAGAAATGTATCTGAAA-TTGCGTAGTGATGTACT-TCTACCTCTTACGCAATACAATAGATATTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGGGCCATGGACACTACCAGTTATAGAGAAGCAGCTTGCTGTCATCTGGCTAAGGCTATAAATGATTTCAGTAATTCAGGTTCTGATGTCCTCTACCAACCACCACAAACTTCAATCACATCAGCGGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAAGGTTGCATGGTACAAGTTACTTGTGGTACCACTACACTTAATGGTCTTTGGCTTGATGATGTAGTTTACTGTCCACGACATGTGATCTGCACTTCTGAAGACATGCTCAATCCTAATTATGAAGATTTACTTATACGTAAATCTAACCATAATTTTTTAGTTCAGGCTGGTAATGTTCAACTTAGAGTTGTTGGACATTCTATGCAAAATTGTGTTCTTAAGCTTAAAGTAGATACAGCTAATCCTAAGACACCTAAGTATAAGTTTGTGCGCATTCAACCCGGACAGACTTTTTCAGTATTAGCCTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCCATGAGACCTAATTTTACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAATATAGACTATGACTGTGTCTCTTTTTGTTATATGCATCATATGGAGTTACCAACGGGAGTTCATGCTGGCACAGACTTAGAAGGTACCTTCTACGGACCTTTTGTTGACAGACAGACAGCACAAGCGGCTGGTACTGACACAACTATTACAGTTAATGTTCTAGCTTGGTTGTATGCAGCTGTTATAAACGGAGATAGATGGTTCCTTAATAGGTTTACCACAACTCTAAACGATTTTAATCTTGTGGCTATGAAGTATAATTATGAACCTCTAACACAAGACCATCTTGACATACTAGGACCTCTTTCAGCTCAAACTGGAATTGCAGTCCTAGATATGTGTGCTTCATTAAAAGAATTATTACAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTCGATGTTGTTAGACAATGTTCAGGTGTCACCTTTC-AAAGTGCAGTGAAAAGGACAATCAAGGGCACGCACCATTGGTTGTTGCTTACAGTTTTGACTTCACTCTTAGTTTTAGTTCAGAGTACTCAATGGTCTTTGTTCTTCTTTGTGTATGAAAATGCCTTTATGCCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCTATGATGTTTGTCAAACATAAGCATGCATTCCTCTGTTTGTTCCTGTTACCTTCTCTTGCTACTGTAGCTTATTTTAATATGGTCTACATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATAGTTGATACTAGTTTGTCTGGTTTCAAGCTAAAGGACTGTGTTATGTATGCATCAGCTGTAGTGTTATTAATCCTCATGACAGCAAGAACCGTATATGATGATGGTGCTAGAAGAGTTTGGACACTTATGAATGTCCTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGACCAAGCTATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGTCCTATCTTCTTTATAACTGGCAATACACTCCAGTGTATAATGCTAGTTTATTGTTTCTTAGGTTATTTCTGTACTTGTTATTTTGGTCTCTTCTGTTTACTCAACCGCTATTTTAGGCTTACTCTTGGTGTTTATGACTATTTAGTCTCTACACAAGAATTTAGGTACATGAACTCTCAGGGGCTCCTGCCACCTAAGAGTAGTATTGACGCTTTCAAGCTTAACATTAAATTGTTGGGCATTGGAGGTAAACCTTGTATTAAGGTTGCTACTGTACAGTCTAAAATGTCTGACGTAAAGTGCACATCAGTAGTGCTT-CTCTCAGTTCTT-CAGCAACTTAGAGTAGAGTCATCTTCTAAATTGTGGGCACAGTGTGTACAACTT-CACAATGATATTCTTCTT-GCCAAGGACACTACTGAAGCTTTTGAGAAGATGGTTTCACTT-TTGTCTGTTTTGCTGTCCATGCAGGGTGCTGTAGACATTAACAAG-TTGTGCGAGGAAATGCTCGACAACCGTGCTACCCTT-CAGGCTATTGCTTCAGAATTCAGTTCT-TTACCTTCATATGCTGCTTATGCCACTGCTCAAGAGGCTTATGAGCAGGCTGTAGCAAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTAAAGAAATCTTTGAATGTGGCTAAATCTGAGTTTGACCGTGATGCTGCCATGCAACGTAAGTTGGAAAAGATGGCGGATCAGGCTATGACCCAAATGTACAAGCAGGCAAGATCTGAGGACAAGAGGGCAAAAGTAACTAGTGCAATGCAAACTATGCTTTTCACTATGCTTAGAAAACTTG-ATAATGATGCACTTAACAACATTATCAACAATGCACGTGATGGTTGTGTACCACTC-AACATCATACCACTCACAACAGCAGCAAAACTCATGGTTGTTGTCCCTGACTATGGAAC-C-TACAAGAATACTTGTGATGGTAACA-CTTTTACATATGCATCAGCACTCTGGGAAATCCAGCAAGTTGTTGATGCAGATAGTAAAATTGTGCAGCTTAGTGAAATCAACATGGACAACTCACCAAATTTGGCTTGGCCTCTTATTGTTACTGCTTTAAGAGCCAATTCAGCTGTCAAACTACAGAACAATGAGCTGAGTCCAGTAGCACTACGACAGATGTCCTGTGCGGCTGGTACTACACAGACAGCTTGTACTGATGACAACGCACTT-GCCTACTATAACA-ATTCTAAGGGAGGTAGGTTTGTGCTTGCATTACTATCAGACC-ACCAGGATCTCAAATGGGCTAGATTCCCTAAGAGTGATGGTACAGGTACTATCTATACAGAATTGGAACCACCTTGCAGGTTTGTTACAGACACACCTAAAGGACCTAAAGTGAAGTATTTGTACTTTATTAAGGGTCTTAA-CAACCTAAATAGAGGTATGGTACTGGGTAGTTTGGCTGCCACAGTACGTCTTCAGGCTGGTAATGCGACAGAAGTGCCTGCCAATTCAACTGTGCTT-TCTTTTTGTGCTTTTGCTGTAGACCCAGCTAAAGCTTACAAAGATTACCTAGCAAGTGGTGGACAACCAATCACGAATTGTGTGAAGATGTTGTGTACACACACTGGTACAGGACAGGCAATCACGGTAACACCAGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCTTCATGCTGTCTGTATTGTAGATGCCACATTGATCATCCAAATCCTAAGGGATTTTGTGACTTGAAAGGTAAGTATGTTCAAATACCTACCACTTGTGTTAATGACCCTGTGGGGTTTACACTCAGAAACACAGTCTGTACCGTCTGCGGAATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGATGCAGTCTGCGGATGCGTCAA-CGTTTTTAAACGGGTTTGCGGTGTAAGTGCGGCCCGTCTTACACCGTGCGGCACAGGCACTAGCACTGATGTCGTTTATAGGGCTTTTGATATTTACAACGAGAAAGTTGCTGGTTTTGC-AAAGTTCCTAAAAACTAATTGCTGTCGCTTCCAAGAAAAGGATGAGGAAGGCAATTTA-TTAGACTCTTA-TTTCGTAGTTAAGAGGCACACAATGTCCAACTACCAACATGAAGAGACTATTTACAACTTGGTTAAAGGGT-GTCCAGCCGTTGCTGTTCATGACTT-TTTCAAGTTTAGAGTAGATGGTGACATGGTACCACATATATCACGTCAACGTCTAACTAAATACACAATGGCTGATTTAGTCTATGCTCTACGTCATTTTGACGAGGGCAATTGTGATACATTAAAGGAAATACTTGTCACATACAAATGTTGTGATGACAACTATTTCAATAAGAAGGATTGGTATGACTTTGTAGAAAATCCTGATGT-TCTACGCGTATACGCGAACCTAGGTGAGCGTGTACGTCAAGCCTTATTGAAAACTGTGCAATTCTGCGATGCTATGCGCGATGCGGGTATCGTAGGTGTACTGACACTAGACAATCAGGATCTGAATGGGAATTGGTACGACTTCGGAGATTTCGTACAGGTAGCAC-C-AGGCT-GCGGAGTCCCTATTGTGGATTCATACTATTCTTTGCTGATGCCTAT-TCTGACACTC-ACAAGGGCTTTGGCTGCTGAGTCCCATATGGACGCTGATCTCG-CGAAGCCACTTA--TTAAGTGGGATTTGCTGAAATATGATTTCACGGAAGAGAGACTATGTCT-TTTCGACCGTTATTTTAAATATTGGGACCAGACGTACCATCCTAATTGTATTAATTGTTTGGATGACAGGTGTATCCTTCATTGTGCAAACTTTAATGTGTTATTTTCTACTGTGTTTCCACCTACGAGTTTTGGACCACTAGTAAGAAAGATATTTGTAGATGGTGTACCTTTTGTTGTTTCAACGGGATACCATTTCCGTGAGCTAGGGGTTGTACATAATCAGGATGTAAACTTACATAACTCGCGTCTCAGTTTTAAGGAACTT--TTAGTGTACGCTGCTGATCCAGCTATGCATGCTGCCTCTGGCAATTTGTTGTTAGACAAACGCACTACATGCTTTTCAGTAGCTGCACTAACAAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAATAAAGACTTTTATGACTTTGCTGTGTCTAAAGGCTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGCAATGCTGCTATCAGTGATTATGACTATTACCGTTATAATCTGCCAACAATGTGTGATATTAGACAACTCCTATTCGTAGTTGAGGTTGTCGATAAATATTTTGATTGTTACGACGGTGGCTGTATCAATGCCAACCAAGTTATCGTTAACAATCTGGACAAATCAGCCGGTTTCCCATTTAATAAGTGGGGTAAGGCTAGACTTTATTATGACTCAATGAGTTATGAGGATCAAGATGCACTGTTCGCATACACTAAGCGTAACGTCATCCCTACAATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACTGTTGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAGAAATTATT-AAAGTCAATAGCCGCCACTAGAGGAGCTACTGTGGTAATTGGAACAAGCAAATTTTATGGTGGCTGGCATAACATGTTAAAAACTGTTTACAGTGATGTAGAAAGTCCTCACCTTATGGGTTGGGACTACCCAAAATGTGACAGAGCCATGCCTAATATGCTTAGAATCATGGCTTCCCTCGTTCTTGCTCGCAAACATAGCA-CTTGTTGTAACTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCACAAGTATTAAGTGAGATGGTCATGTGTGGCGGCTCATTATATGTGAAACCAGGTGGAACATCATCCGGTGATGCCACAACTGCTTATGCTAATAGTGTGTTTAACATCTGTCAAGCAGTAACAGCTAATGTAAATGCACTTCTT-TCAACTGATGGTAATAAGATTGCTGATAAGTATGTCCGCAACCTT-CAACACAGACTTTACGAGTGTCTCTATAGAAATAGAGACGTAGAT-CAGGAATTCGTGGATGAATTTTATGCATATTTGCGTAAGCATTTCTCCATGATGATTCTTTCTGATGATGCCGTCGTATGCTAT-AATAGTAATTACGCGGCACAGGGTCTAGTAGCTAGCATTAAGAACTTTAAAG-CAGTTCTTTATTACCAAAATAATGTATTCATGTCTGAGGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGTTCTCAGCATACCATGCTAGTTAAACAGGGAGATGATTATGTGTACCTGCCTTATCCAGATCCATCCAGAATATTAGGCGCAGGCTGTTTTGTCGATGACATTGTCAAAACAGATGGTACACTTATGATTGAAAGGTTTGTGTCATTAGCAATTGACGCCTATCCACTTAC-AAAGCACCCTAATCAAGAGTATGCTGATGT-TTTCCATTTATACTTACAGTATATTAGGAAAT-TACATGATGAGCTTA-CTGGTCACATGTTGGACATGTACTCTGTAATGCTAACTAATGACAACACCTCGAGGTACTGGGAACCTGAGTTTTATGAGGCAATGTACACACCACACACAGTTTTGCAAGCTGTAGGTGCGTGTGTG-TTATGCAATTCACAGACTTCACTTC-GTTGCGGTGCTTGTATTAGG-AGACCTTTCCTT-TGTTGCAAGTGCTGCTATGATCACGTCATCTCAACATCACATAAATTAGTGTTGTCTGTTAATCCCTATGTTTGCAATGCACCCGGTTGTGATGTCACAGACGTAACACAACTCTATTTGGGAGGTATGAGCTATTACTGCAAGTCACATAAACCACCCATTAGTTTTCCGTTGTGTGCTAATGGTCAGGTTTTTGGTTTGTACAAAAACACATGTGTGGGCAGTGATAACGTAACTGACTTCAATGCAATAGCGACATGCGACTGGACTAATGCTGGCGATTACATACTT-GCCAACACTTGCACAGAGAGACTCAAACTTTTTGCAGCGGAAACGCTCAAAGCTACTGAGGAAACATTCAAACTATCTTATGGTATTGCCACTGTTCGTGAAGTACTGTCAGATAGAGAACTT-CATCTTTCATGGGAGGTAGGAAAACCTAGACCACCA-TTGAATAGAAATTACGTCTTTACTGGTTACCGTGTGACCAAAAATAGTAAAGTACAGATTGGAGAGTATACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTGTACAGAGGTACTACAACTTATAAATTGAATGTTGGTGATTACTTTGTGTTAACATCACACACAGTAATGCCACTAAGTGCACCAACACTAGTGCCACAAGAGCACTATGTGCGAATAACTGGCTTATACCCTACACTTAATATCTCTGATGAGTTTTCTAGCAATGTTGCAAATTACCAAAAGGTCGGCATGCAGAAGTACTCCACACTCCAGGGACCACCTGGTACTGGTAAGAGTCACTTTGCTATTGGACTTGCCCTCTACTACCCATCTGCCCGCATAGTGTATACAGCTTGCTCTCATGCTGCTGTTGATGCGCTATGTGAGAAGGCATTAAAATACTTGCCTATAGATAAGTGTAGTAGAATTATTCCTGCACGTGCGCGTGTAGAGTGTTTTGACAAATTCAAAGTGAATTCAACCTTAGAACAGTATGT-TTTCTGCACTGTCAATGCGCTGCCTGAAACTACTGCTGATATAGTGGTCTTTGATGAAATTTCAATGGCCACTAATTATGATTTGAGTGTTGTCAATGCCAGACTACGTGCAAAACACTACGTTTACATTGGTGATCCGGCTCAATTACCTGCACCACGCACATTGCTAACAAAAGGCACACTTGAACCAGAATATTTCAATTCAGTGTGCAGACTTATGAAAACAATAGGTCCAGACATGTTCCTTGGGACTTGTCGTCGTTGTCCCGCTGAAATTGTTGACACAGTGAGTGCTTTAGTTTATGATAATAAGCTAAAAGCACACAAAGAGAAGTCAGCTCAATGCTTCAAGATGTTTTACAAGGGTGTGATTACGCATGATGTGTCATCCGCAATCAACAGACCACAAATAGGTGTAGTAAGAGAATTTCTTACGCGCAATCCAGCTTGGAGAAAAGCTGTTTTTATTTCACCATATAATTCACAGAATGCAGTGGCATCAAAGATTTTAGGGT-TGCCTACTCAAACTGTTGATTCTTCACAGGGTTCTGAATATGACTATGTCATATTCACACAAACCACTGAGACTGCACACTCTTGTAATGTAAACCGCTTTAATGTGGCCATTACAAGAGCAAAGATT-GGCATT-TTGTGCATAATGTCTGATAGAGACCTTTATGACAAGCTT-CAATTTATGAGTCTAGAAGTACCGCGTCGAAATGTGGCTACTTTACAAGCAGAAAATGTGACTGGACTCTTTAAGGACTGTAGCAAGATCATTACTGGTCTT-CATCCAACACAGGCACCTACGCATCTCAGTGTTGATACTAAATTCAAGACTGAAGGACTT-TGTGTCGACATACCAGGAATACCAAAGGACATGACCTATCGTAGACTCATCTCTATGATGGGCTTTAAAATGAATTACCAAGTTAATGGTTACCCTAATATGTTTATCACCCGTGAAGAAGCTATTCGTCACGTTCGTGCATGGATAGGCTTTGATGTTGAGGGTTGTCATGCGACTAGAGATGCTGTAGGAACAAATCTACCACTCCAGTTAGGGTTTTCAACAGGTGTTAACCTAGTGGCTGTACCAACTGGCTATGTTGA--CAC-TGAGCACAGCACAGAATTTACCAGAGTTAATGCAAAACCTCCTCCAGGTGATCAATTTAAGCATCTTATACCACTTATGTACAAAGG-CTTGCCCTGGAACGTGGTGCGTATTAAGATTGTTCAAATGCTCAGTGATACACTGAAAGGAT-TATCAGACAGAGTTGTGTTTGTCCTT-TGGGCACATGGCTTTGAACTT-ACATCGATGAAGTATTTTGTTAAGATCGGACCAGAAAGAACGTGTTGTCTGTGTGACAAACGCGCGACTTGCTTCTCTACTTCATCTGACACTTATGCCTGTTGGAATCACTCTGTGGGCTTTGACTATGTCTATAACCCGTTTATGATTGATGTCCAGCAGTGGGGTTTTACAGGTAACCTTCAAAGTAACCATGATCAACACTGCCAAGTGCATGGTAATGCCCATGTAGCTAGTTGTGATGCTATCATGACTAGATGTCTTGCAGTCCATGAGTGCTTTGTTAAGCGCGTTGATTGGTCTGTTGAATACCCGATTATTGGAGATGAACTGAAGATTAATGCCGCATGCAGAAAAGTACAGCATATGGTTGTTAAATCTGCATTGCTT-GCTGATAAATTCCCAGTTCTTCATGATATAGGAAACCC-AAAGGCTATTAGA-TGTGTGCCGCA-GTCTGAAGTGGACTGGAAATTCTACGACGCTCAGCCTTGCAGTGACAAAGCTTATAAAATAGAAGAACTCTTCTACTCATATGCCACACAT-CATGACAAGTTCACAGATGGTGTTTGCTTGTTTTGGAACTGTAACGTTGATCGTTACCCGGCTAATGCTATTGTGTGTAGGTTTGATACTAGAGTGCTTTCTAAT-TTAAACCTACCAGGTTGTGATGGTGGTAGTTTGTATGTTAATAAGCATGCGTTCCACACTCCAGCTTTTGATAAGAGTGCATTTAC-ACATTTGAAACAACTGCCTTTCTTT-TATTACTCTGACAGTCCGTGTGAGTCTCATGGTAAACAGGTTGTGTCAGATATTGATTATGTCCCACTAAAGTCTGCTACGTGTATTACACGATGCAACTTAGGTGGTGCCGTTTGTAGACATCATGCAAACGAGTACAGACAGTA-CTTGGATGCATATAATATGATGATTTCTGCTGGATTTAGCCTT-TGGATTTATAAACAATTTGATACTTACAACTTGTGGAACACTTTCACCAAG--TTGCAGAGTTTAGAAAATGTGGCTTATAATGTTATCAACAAGGGACACTTTGATGGACAGA-ATGGTGAAGCACCTGTGTCTATCGTTAATAATGCTGTTTACACTAA-GTTAGATGGTGTTGATGTGGAGATC-TTTGAAAATAAGACAACACTT-CCTGTTAATGTTGCATTTGAGCTTTGGGCTAAACGTAACATTAAACCGGTGCCAGAGATTAAAATACTCAATAATTTGGGTGTTGATATCGCTGCTAATACTGTTATATGGGACTACAAGAGAGAAGCGCCAGCGCATGTTTCTACAATAGGTGTCTGTACAATGACTGACATTGCAAAGAAACCTACTGAGAGTGCTT-GTTCATCACTTACTGTCTTATTTGATGGTAGAGTTGAGGGACAGGTAGACCTT-TTTAGAAACGCCCGTAATGGTGTT-TTAATAACAGAAGGTTCAGTTAAGGGCTTAACA-CCTTC-GAAAGGACCTGCACAGGCTAGTGTCAACGGAGTCACATTAATTGGAGAATCAGTAAAAACACAGTTCAATTACTT-TAAGAAAGTGGATGGCATTATTCAGCAATTGCCAGAAACCTACTTTACTCAAAGCAGAGACTTAGAGGA-TTTCAAGCCCAG-A-TCACAAATGGAAACTGATTTCCTT-GAGCTCGCTATGGATGAATTCATAGAACGATATAAGCTAGATGGCTATGCTTTCGAGCACATCGTTTATGGAGATTTTAGTCATGGACAATTAGGCGGACTT-CATTTATTGATAGGACTGGCCAAAAGGTC-ACAGGACTCAC-TGTTAAAGCTAGAGGATTTTATTCCTATGGATAGCACAGTGAAAAACTACTTCATAACAGATGCGCAAACGGGTTCATCTAAGTGTGTATGCTCTGTTATCGACCTT-TTACTTGATGACTTTGTTGAAATAATAAAGTCACAAGATCTT-TCAGTGGTTTCAAAAGTAGTCAAAGTTACGATTGATTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGGCATGTGGAAACTTTTTACCCAAAATTACAATCTAGTCAAGCATGGCAACCAGGTGTTGCTATGCCTAATCTCTATAAAATGCAGAGAATGTTACTGGAAAAGTGTGATCTTCAAAATTATGGTGATAGTGCTATATTGCCTAAAGGCATAATGATGAATGTCGCAAAGTACACTCAACTGTGTCAGTATTTAAATACACTTA-CTTTAGCTGTGCCCTATAATATGAGAGTTATACATTTTGGCGCGGGCTCTGATAAAGGAGTAGCACCTGGCACAGCTGTTCTTA-GACAGTGGTTGCCAACTGGTACACTACTTGTCGATTCTGAT-TTAAATGATTTTGTTTCTGATGCAGACTCAACATTAATTGGTCATTGTGCAACCGTACATACGGCTAATAAATGGGATCTCATTATTAGCGATATGTATGATCCTAAGACTAAAAATGTTACAAAAGAGAATGATTCCAAAGAAGGATTTTTCACTTACATTTGTGGATTTATACAGCAAAAATTAGCCCTCGGAGGTTCCGTAGCTGTAAAGATAACAGAGCATTCCTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCTTTTGTTACTAATGTTAATGCTTCATCTTCAGAGGC-TTTCTTAATTGGTTGTAATTATCTTGGCAAACCACGTGAGCAGATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAACACAAATCCAATACAATTGTCTTCCTACTCATTATTTGACATGAGTAAGTTTCCTCTTAAATTAAGAGGTACTGCTGTTATGTCATTAAAAGATGGACAAATCAATGATATGATTTTGTCTCTTCTTAGTAAAGGCAGACTTATTGTTAGAGAGAATAATAGAGTTATTATCTCTAGTGATGTTCTTGTTAACAACTAAACGAACA-TGTT-GTTTTTCTTGTTTCTTCAGTTCGCCT-TAGTAAACTCC-CAGTGTGTTAA-CTTGACAGGCAGAACCCCACTCAATCCCA--ATTATACTAATTCTT-CACAAAGAGGTGTTTATTACCCTGACACAA-TTTAT-AGATCAGACACACTTGTGC-T-CAGC-CAGGGTTATTTTCTT-CCATT-TTATTCTAATGTTAGCT-GGT---AT--TACT-CAT-TAACAACCAACAATGCTGCCACAAAGAGGACTGATAATCCTATATTAG-ATTTCAAGGACGGCATATACTTTGCTGCCACTGA-ACACTCAAATATTATCAGGGGCTGGATCTTTGGAACAACTCTT-GA--C-AACACTTCTCAATCTCT-CTTGATAGTTAACAACGCAACGAATGTTATTATCAAGGTTTGT-AATTTTGATTTTTGTTATGATCCCTACCTTAG-TGGTTACTATCATAACAACAA-AACA-T-GGA-GCATCAGAGAATTTGCTGTCTATTCTTCTTA-TGC---TAATTGTACTTTTGAGTATGTTTCGAAATCCTTTATGTT--GAACATTTCTGGTAATG-GTGGTC-TGTTCAACACTCTTAGAGAGTTTGT-TTTCAGAA-ATGTCGATGGGCATTTCAAGATTTACTCAAAGTTTACACC-AGTAAATTTAAAT-CGTGG-CTTGCCTACTGG-TCTCTCAGTGCTTCAG--CCATTGGTTGAATTACCAGTTAGC-ATAAATATTACTAAATT-CAGAACACTCCTCACTATT-CATAGAGG--A---GAC-CCTA-TGCCTAATAA-C-GGCTGGACTGCTTTTTCAGCTGCTTATT-TCGTGGGCTATCTTAAACC-ACGTACCTTTATGCTGAAATATAATGAGAATGGCACCATTACTGATGCTGTTGATTGTGCACTTGACCCTCTTTCGGAGACAAAGTGTACGTTAAAATCTCTT-ACTGTCCAAAAGGGCATCTATCAGACTTCTAACTTCCGAGTGCAACCCACTCAGTCTGTAGTTAGATTTCCTAATATTACCAA-TGTGTGTCCATTTCAC-AAGGTTTTTAATGCCACGAGGTTTCCTTCCGTCTATGCGTGGGAA-AG-AACTAAAATTTCTG-AT-TGCATTGCAGATTACACTGTTTTCTACAATTCAACTTCTTTTTCTACTTTTAAATGTTATGGTGTTTCACCTTCTAAATTGATTGATTTGTGCTTTACGAGTGTGTATGCTGATACATTTCTCATAAGATTCTCA-GAAGTCAGACAGGTGGCACCAGGACAAACTGGTGTCATTGCTGACTATAATTATAAATTACCTGATGATTTTACAGGTTGTGTCATAGCTTGGAACACTGCCAA-----A--C-A-GGATG-TAGGTAATTATT-TCTACAGGTCTC-ATCGTTCTACCAA----A-T-TGAAACCATTTGAAAGAGATCTTTC--CTCAGA-CGAGAATGGT-G-TCCGTACAC-TT--A--G-T----AC--TT--A-T-G--ACTT-CA---ACCCT-A-ATG-T----A-CC-ACT--TGAA-T-----ACCAAGC-TACAAGGGTTGTTGTT-TTGTCATTTGAGCTTCTAAATGCACCAGCTACAGTTTGTGGACC-AAAACTA-TCCACACAACTAG-TAAAAAATCAG-TGCGTTAATTTCAACTTTAACGGACTCA-AGGGCACTGGTGT-CTTGACTGATTCTTCCAAGAGGTTTCAGTCATTCCAACAATTTGGTAAAGATGCGT--CTGACTTTATTGATTCAGTACGTGATCCTCAAACACTTGAGATACTTGACATTACACCTTGCTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAACACTTCTTTAG-AGGTGGCTGTTCTTTACCAAGATGTTAACTGCACTGATGTACCAACTACTATACATGCAGACCAACTAACACCTGCTTGGCGTATTTATGCTACTGGCACTAATGTGTTTCAAACTCAAGCAGGCTGTCTTA-TAGGAGCTGAACATGTCAATG-CTTCTTATGAGTGTGACAT-CCCAATTGGTGCTGGTATTTGTGCTAGCTACCATACGGCTT-CTA-TA-T--TA--CG--CA-GTACA--AGCCAGA-AAGCTATTGTGGCTTATACTATGTCCCTTGGTGCTGAGAACTCTA-TCGCTTA-TGCTAACAATTCTATAGCCATACCTACAAATTTTTCAATTAGTGTTACCACTGAAGTTATGCCTGTATCAATGGCTAAAACTTCTGTAGATTGTACTATGTATATCTGTGGTGACTCTA-TAGAGTGTAGCAA-CTTGTTGTTACAATATGGCAGTTTTTGCACACAACTAAATCGTGCTTTAAGTGGGATTGCTATTGAGCAAGACAAGAACACCCAAGAGGTTTTTGCTCAAGTTAAGCAAATCTATAAAACACCACCTATTAAGGATTTTGGTGGTTTTAATTTTTCACAGATACTACCTGACCCATCTAAACCCAGCAAGAGGTCGTTTATTGAAGA-CTTACTCTTCAATAAAGTCACTCTTGCTGATGCCGGTTTTATCAAACAGTACGGTGATTGT-TTGGGTGGTATTTCTGCTAGAGATTTGATTTGTGCTCAAAAGTTCAATGGACTTACTGTCTTACCACCATTGCTCACAGATGAAATGATCGCTGCTTATACAG--CTGCATTAATTAGCGGC-ACTGCCACTGCTGGATGGACCTTTGGTGCTGGTGCTGCTCTT-CAAATACCATTTGCCATGCAAATGGCTTATAGGTTTAATGGAATTGGAGTTACTCAGAATGTTCTCTATGAGAATCAGAAATTAATAGCCAATCAGTTTAATAGTGCTATTGGAAAAATCCAAGAGTC--TTTGACATCTACAGCTAGTGCACTTGGAAAA-TTGCAGGATGTTGTTAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGTTCCAATTTTGGTGCAATTTCAAGCGTGTTGAATGACATTCTTTCACGCCTTGACAAAGTCGAGGCTGAGGTTCAGATTGATAGGTTGATCACAGGTAGACTTCAGAGTTTACAGACGTATGTGACTCAACAATTAATCAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCGACTAAAATGTCCGAGTGTGTACTAGGACAATCTAAAAGAGTTGATTTTTGTGGAAAAGGTTATCACCTAATGTCTTTTCCC-CAGTCAGCGCCTCATGGTGTTGTCTTCTTACATGTGACTTACATTCCTTCGCAAGAAAAGAACTTCACAACAGCTCCTGCCATTTGCCATGAAGGTAAAGCACACTTCCCACGTGAAGGTGT-TTTCGTTTCGAATGGCACACACTGGTTTGTAACACAAAGGAACTTTTATGAACCTA-AAATTATAACCACTGACAATACATTTGTCTCTGGTAACTGTGATGTTGTAATTGGAATTATCAACAACACAGTTTATGATCCTTTACAACCAGAACTT-GATTCATTTAAGGAGGAGTTAGATAAATATTTTAAAAATCATACATCACCTGATATTGATCTT-GGTGATATTTCTGGCATTAATGCTTCTGTTGTCAATATTCAAAAGGAAATTGACCGCCTCAATGAGGTTGCCA-GAAATTTAAATGAATCACTCATTGATCTCCAAGAACTTGGAAAATATGAGCAATATATCAAATGGCCATGGTATGTTTGGCTCGGCTTCATTGCTGGACTC-ATTGCTATAGTCATGGTTACAATCCTGCTTTGTTGCATGACAAGTTGTTGCAGTTGTCTCAAGGGCTGTTGTTCTTGCGGATCTTGCTGTAAATTTGATGAAGACGACTCTGAGCCTGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATTTTCACTCTTGGAACTGTTACTCTTAAA-CAAGGTGAAATCAAAGG-TGCTACTCCTACAAATTCTGTTCGCACTACTGCAACAATACCGATACAAGCCACACTCCCTTTCGGATGGCTTGTTGTTGGCGTTGCAATTCTTGCTGTTTTTCAAAGCGCTTCAAAAATAATTACACTCAAAAAGAGATGGCAGTTAGCCCTCTCTAAAGGTGTTCATTTTGTTTGCAACTTGCT-TCTGCTGTTTTTAACAGTTTATTCTCAC-TTGTTGCTTC-TTGCTGGTGGC-TTGGAAGCCACTTTTCTCTTTCTTTATGCATTAG-CTTA-TTGCTTGCAAACTGTAAATTTTGTGAGAATAATAATGCGA-TTCTGG-TTGTGCTGGAAGTGCCGTTCCAAGAATCCTTTACTCTATGATGCCAACTACTTTCTTTGTTGGCATACTAATTGTTATGACTATTGTATACCATACAATAGTGTAACCTCTTCAATTGTCATCACATGTGGTGATGGTACTACGAA-TCCCATTTCTGAGGACGACTACCAAATTGGTGGTTACACGGAAAAGTGGGAGTCTGGTGTTAAGGACTGTGTTGTATTACATAGTTATTTCACCTCAGATTACTACCAGCTGTACTCAACACAAGTGAGTACAGACACTGGTGTTGAACATGTTACTTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTTCAAATTCACACAATCGACGGTACATCTGGAGTTGTTAATCCAGCAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTTGTGGTATTCTTGCTAGTCACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTTTGGTCTAAACGAACTAAATATTATATTAGTCTTTCTGTTTGGAACTTTAATTTTAGCCATGTCAGGTGACAACGGTACCATTACCGTTGAAGAGCTTAAAAAGCTC-TTAGAACAATGGAACCTAGTAATAGGATTCTTGTTTCTTACATGGATTTGT-TTGTTACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTACATAATTAAGTTAATTTTCCTCTGGCTGCTT-TGGCCAGTAACTTTAGCTTGCTTTGTGCTTGCTGCTGTTTACAGAATAAACTGGATCACTGGTGGAATTGCCATTGCAATGGCCTGTCTTGTAGGCTTGATGTGGCTTAGCTACTTCATTGCTTCTTTCAGGCTGTTTGCTCGTACGCGTTCCATGTGGTCATTTAACCCAGAAACTAACATTCTTTTGAACGTGCCTCTTCATGGCACAATTCTGACCAGGCCGCTTCTAGAGAGTGAACTCGTAATTGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCAGGACACCATCTGGGACGCTGTGACATCAAGGACCTGCCCAAAGAAATCACTGTAGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGTAATTACAAATTAAATACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTATTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTATATCATAAATCTCATAATTAAAAATCTATCTAAGCCTCCAACTGAGAATAACTGTTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACTAACATGAAAATTATT-TTCTTCTTGGTACTGATAACACTTGTTACTGGCGAGCTTTACCACTACCAAGAGTGTATAAAAGGTACAACTGTACTTTTAAAAGAACCTTGCTCTTCAGGAACATATGAAGGCAATTCACCATTTCATCCTCTAGCTGATAATAAATTTGCACTGGCTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGTGTTAGACACACCTTTCAGTTACGTGCGAGATCAGTTTCACCCAAACTGTTTACCAGACAAGAGGAAGTTCAAGAA-TTATACTCACCTGTTTTCCTTATCGTTGCAGCTATAGTGTTCATAATACTTTGCTTCACATTCAAAAGAAAAATAGAATGAGTGAATTTTCATTAATTGACTTCTATTTGTGCTTCTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTCATTATCTTTTGGTTCTCACTAGAACTGCAAGATCATAATGAAACTT-TCCACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAATTCTT-ACAACAGTAGCTGCATTCCATCAGGAATGTAGTTTACAGTCATGTGCTCAGCATCAACCCTATGTAGTTGATGACCCTTGTCCAATTCACTTCTACTC-ACGATGGTATATCAGAGTGGGAGCTAGAAAATCAGCACCTTTGATTGAATTGTGTGTTGATGAGG-TAGGCTCTAAGTCACCCATTCAATACATTGACATTGGTAATTACACAGTTTCCTGTTCT-CCTTTTACAATTAATTGCCAGGAACCTAAATTAGGTAGTCTCGTAGTACGGTGTTCGTAT-TATGAAGACTTTCTAGAGTACCATGACATTCGTGTTGTCTTAGATTTCATCTAAACGAACTAACTAAAATGTCTGATAATGGACCCCAAAACCAACGTAGTGCACCCCGCATTACATTTGGTGGACCCTCAGATTCAAGTGACAATAGCAAAAACGGAGAGCGCAATGGTGCACGACCTAAACAACGTCGACCCCAAGGCTTACCCAATAATACTGCATCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAAACCTTACGTTCCCTCGAGGGCAAGGTGTTCCAATCAACACCAATAGCTCTAAAGATGACCAAATTGGCTACTACCGTAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGAGCTCAGCCCCAGATGGTATTTTTACTATCTAGGAACTGGACCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGAAGGCATCATATGGGTTGCAACTGAGGGAGCCTTAAACACACCGAAAGACCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCTTCTTCACGCTCCTCATCACGTAGTCGCAACAGTTCAAGAAACTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGACACTGCTCTTGCTTTGCTGCTGCTAGATAGGTTGAACCAGCTTGAGAACAAAGTATCTGGCAAAGGCCAACAACAACAGGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCATCTAAAAAGCCTCGCCAAAAACGTACTGCTACAAAA-CAGTACAACGTCACTCAAGCATTTGGG-AGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAAGAATTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCTCCAAGTGCCTCTGCATTCTTTGGAATGTCACGCATTGGCATGGAAGTCACACCTTCGGGAACATGGCTGACTTATCAT-GGAGCCATTAAATTGGATGACAAAGATCCACAATT-CAAAGAT-AACGTCATACTGCTGAATAAGCACATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAAAAGGCTGATGAA-CTTCAGGCTTTACCGCAGAGACAGAAGAAACAACAAACTGTGACCCTTCTTCCTGCTGCAGATTTGGATGAATTCTCCAAACAGTTGCAACAATCCATGAGTGGTACTGATTCAACCCAGGCTTAAACTCGTGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATTTCACATAGCAATCTTTAATCAATGTGTAACATTGGGGAGGACTTGAAAGAGCCACCACGTTTTCACCGAGGCCACGCGGAGTACGATCGAGGGTACAGCCAATAATGTTAGGGAGAGCAGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCA-ACCACTC-GACAAGAAAAAAAAAAAAAAAAAAAAAAAAAA----\n",
      "||-|.|-|||||.||||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||||||||||||||||||.||||||||..|||||||||||||||||||||||||||.||||||-|||.|||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||.|.||.||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||-||||-||||||||||||||||||||||||||.||||||||.|||||||||||||||||||||||.|||||.|-|||-||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||.||.|||||||||||.||..|.||||||||||||||.|||||||||||.||||||||||||||||||||||||||.||||||||||||||.||.||..|||||||.||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.|||||||||-||-|||||||||||||||||.|||||||||||||||.||||||||||||||||||||||||.||||||||||||||||||.|||||||||||||||||.||.||||||||.||||||||.|||||||||||||||||||||||||||||||||||.||.|||||||||||.|||||--||||--||||||.||||||.||||.|||||||||||||||||.|||||.|||||||||||||||||.|||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||-||-||||||.|.|||||||||.|||||||||||||||||||||||||||||||||||-|||-||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||.||||.|||||||||||||||.|||||||||||||||||||||||||||||.||||||||||.|||||||||||||||.|||||.|||||.|||||||||||||||||||||||.|||||||||||||||||.||||||||||||||-||||-|||||||||||||||||||.|||.||||||||||||||||||||||||||||.|||||.||||||||.|||||.|||||||||.|||||||||||||||||||.||.|||||-||-|||.|||||||||||-|-||||||||||||||.|.||||||||||||||||||||||||||||||||.|||||||||||.|||||.||.|||||||||||||||||||||||.|||.||||.||.||||||||||||||||||.|.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||.||.|.||||||||||||.||||||||.||||||||||||||.|||.|||||||||||||||||||||||||||||||.|||||||||||.||.||||||||-|||-||||||||||||||.||.||||||||-||.|-|||||||||||||.||.|||||.||..|.|||||.|||||||.|||||||||.||.||.|||||.|||||.||.|||||||||||..|||||||||.|||.|||||.|||||.|||||||..||-||-||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||.||||||||.|||||||||||.||..|||||||.|||||||||||.||.||||||||||||||||||||||||||.|||||.||.|||||||||||.||||||||.||.||.||||||.||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||.|||||||||||||||||||.|||||.|||||||.|||||.||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||.|||.||||||||||||||||||.|||||||||||||.||||||||||||||.||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||.||||||||||||||.||.|||||||||||||||||||||||.||||||||.||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||||||...||.||||||||||||||||||||.|||||.||||||||.||||||||.||.||||||||||||||||||||.|||||||||||||||.||.|.|.|||||||||||||||||||||.||||||||||||||||||||||||||.|||||||||||.|||||.|||||||||.|||||||||||||.|||||||||||||||||.||--||-||||||..||||.||||||||||||||.|.|||.|||.|||||||||||||||||||||||||||||.|||||.||||||||||||||||||||.||||||||.||||||..|.||-||||..|-|.|-|||||||.||||||||.|||||||||||-|.-|||||||||..|--|-|--|.-|||.|-|||.|||||-||.||.|||-|-|-||-|||----|.|.---|-|||||.||||.||||...||-|-||.|||||-|.|-|||||||||||-|.||--|||.||||...||||.|||.|..||||||||||||-|||-||||||||||.|.||||||||||||.||||||||.||||||||||||||||||||.||.|||||.||||||||||||||.||||||||||||||||||||||||||||||||.|||||.|||||||||||.||||||||.||||||||.||||||||||||.|.||||||.||||.|||||.|||||-||.-|||||||||.||||||||||.||.|||||.||.|||||||||.|.||||||||||||||.|||||.|||||.||||.||||||||||||||||-||-|||||.|.|||||||||||||||.|||||.||.||-|.||-||||.|||||||||||||||||||||||||||||||||||.|||.|||||||||||||||||||||||||.|||||||.||||||||||||-||-||.|||||||||.|||||||||||||||||||||||||||||||||||.||.||||||||||||||.|||||.||||||||-|.|-|||||.|.|||||||||||||||||||||||||.|.|||||||.||||||||||||||||||.|||.||||||.|||.-|-|-||||||||||||..||.||||||||||||||||.|||||.||.||.||.||||||||.|||||||||||||||||||||||||||||.||||||||||||||.||||||||||||||||||||||..||.|||||||.|||||||||.||||||||||.||||||||||||||||||||||.||.||||.||.|.||||||||||||.||||||||||||||||||||||||||||||||.||||||||.||.||||||||.|||||||||||||||||.||||||||||||||.|||||.|||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||.|||||||||.|||||||||||||||||||||.|||||||||||||||.|||||||||||||||||.||||||||.|||||.||||||||||||||||||||.||.|||||||||||.|||.||||.||||||||.|||||||||||||||||.|||||.||.||.||.|||||||||||.|||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||.||||||||.||||||||||||.||||||||||.||||||||.|||||||||-||-||.|||||.||.||.||-||-||||||.|||||||||||||||||.|||||.|||||||||||.||.||||||||.||||||||||||||.||||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||.||||||||||||.|.|||||||||||||||||||||||||||||.|||||||||||.||||||||||||||.||||||||||||||||||||||||||.||||||||||||||.||||||||.||.|||||.|||||||||||||||..||||||||||||.|||||||||||||||||.|||||||||||||||||||.||||||.|||||||||||||||||||||||||||||||||||||||||||||||.|||||.||||||||||||||||||||.|||||||.|||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.|||||||||||.||-||-|||||..||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||.|||||||||||||||||.|||||||||||||||||.|||||||||||.|||||||||||||||.|||||||||||||||||||.|||||||||||.|||.|||||||.||.|||||.||.|||||.||||||||.||||||||.|||||||||||||||.|.|||||.||||||||||||||||||||||||||||||||.|||||||||||||.|||.|||||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||.||.|||||||||||||||||||||||.||||||||||||||||||||||||.|||||||.|||||.|||||||||||||.|||||||||||||||||||.|||||||.||||||||.|||||||||||||||||||||||.|||||.||.||.||.||||||||||||||||||||||||..|||||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||||||||.||.||||||||||||||||||||.||.||.||||||.|||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||.||.||.|||||||||||||||||||||||||||||.|||||||||||...|||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||||||||.||.|||||||||||||||||||||||||.|||||||||.|||||||||||.||.||..|||||||||||||||||||.||.|||||||||||.||||||||-|||||-||||||||||||||||||.|||||||||||||||||.|||||||||||.||||||||||||||||||||||||.|.|||||.||.|||||||||||||||||||||||||||||.|||||.|||||.||||||||||..|||||.|||||||||||||||||.||||||||.|||||||||||||||||.|||||||.||||-||.|-||.||||.|.|||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||.|.||.||.|||||||||||.||||||||||.||.|||||.|||||||||||||||||.||||||||.||||||||||||||||||||||||||||||||.||||||||||||||||||||.-||-||.||||||||.||.||||||.||||||||||.||||||||.||||||||||||||||.|||||||||.|||||.||.||||||||.|||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||.||||||||||||||||||-||-||||||||||||.||.|||||||||||||||||||||||.|||||.||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||.|||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||||.||||||||||||.|||||||||||||||||||||||||.||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||.||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||-||-|||||||||||||||||||||||||||||||||.||||||||-|||-|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||.||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||.|||||||||.|||||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||.||||||||||||||||||||||||||||||||||||||||||||.||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||.||||||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||.|||||.||||||||||||||.||||||||||||||.|||||||..||||||||||||||.||.||.||||||||||||||||||.||.|||||||.||||||||||||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||.||.||.||||||||.||.||.||||||||.||||||||||||||||||||||||||.||.||.|||||||||||||||||.||||||||.||||||||.|||||.|||||.|||||||||||.|||||.||||||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.|||||||||||.|||||||-|-||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||.||.||||||||||||.|.|||||.||||||||||||||.|||||..|..|||.|||.|||||||||||||||||||||||||||||||.||||||.|.||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.||.||||||||||||||||||||||||||||||||.||||||.|.||||||||.|||||.||||||||||||||||||||.|||||||||||||||.|.||.||.||.||.|||||.|||||.||.|||||.|||||.||||||||||||||.|||||||||||||.|||||||||.|||||||||||-|||-||||||||.||||||||.||.||||||||.||||||||.||.||.||||||||||||||||||||||||||||||||.||.||.|||||.||||||||||||||.|||||||||||||||||||||||.||||||||.||||||||||||||.|||||||||||||||||||||||.|||||||||.|.||||||||||||||.||||||||||||||.||.|||||.||||||||.||||||||.|||||||||||||||||||||||.|||||||||||||||||.|||||.|||||.||.||.|||||.|||||||||||.|||||..||||||||||||||||||||||||||||||||||.|||||||||||||||||.||.|||||||||||.|||||||||||||||||.||||||||||||||||||||.||.|||.|||||||||||||||||||||||||||||||.||||||||.|||||.||||||.||||||||||..|||||||.||.||.||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||.||||||||.||||||||||||||||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||.||||||||||||||||||||.|||||.|||||||||||.|||||||.||||.|||||||.|||||||||||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||.||.|||||..|.||.|||||||||||||||||||||||||||||-|||-||||||||||||||.||-|-||||||||||||||||.||||||||.|||||||||||||||||||||||||||||||||||.||.|||||.||.||.||.||.||||||||.|||||.||||||||.||.||||||.|.|||||.||||||||.||||||||||||||.||.|||||||||||||||||.||.|||||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||.|||||||||||.|||||||||||.|||||||||||||||||.|||||||||||||||.||||||||||||||||.||||||||||||||.||.||||||||||||||||||||.||.|||||.|||||.||||||||.||.||.||||||||||||||||||||.||.|||.|||||||||||||||||||||||||.|||||||||||.||.||||||||.|||||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||.|||||.||||||||.||||||||.|||||||||||||||||||||||||||||||.|||.||.|||||||||||||||||.||.|||||||||||.||||||||.||||||||||||||||||||||||.|||||||||||||.||.|||||||||||.|||||.||.|||||.||.|||.|.||||||||||||||.||.||.|||||.|||||||||||||||||.|||||||||||||||||||||||||||.||||||||||||||||||||||.|||||||||||||||||.||..|||||||||||||||||||||||||||||.|.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||||||||.||-||||-||||||||||||||||.|||||||||||.||.|||||.||||||||.||.|||.|||||||||||||.|||||||||||.||||||||||||||||||||||||||.|||.|||||||||||||||||.|.||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||.|||||||||||..||||||||||||||||.||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||||||||||||.|||||||.||||||||||||||.||.|||||||||||||||||.|||||.||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||.||||||||.||||||||.||||||||||||||||||||||||||||||||.|||||.|||||||||||.|||||.|||||.|||||||||||||||||.|||||.||.|||||||||||||||||.||.|||||.||||||||.||.|||||.||.|||||.||.|||||.||.||.|||||.||||.|||.||.||.||.|||||.||.|||||||||||||||||..||||.||.|||||||||||.||.||.||.||||||||||||||||||||.||.||||||||||||||||||||-|||-|||||||||-||-||.|||||.||||||||.|||||.||||||||||||||.||.|||||.||.-||-||||||||.|||||-|||-||.||.||.|||||||||||.|||||.||.|||||||||||.-||-|||||||||||.|||||||||||||||||||||||.||||||-||-||.||.||||||||.||||||.|.||.|||-||-||.|||||.||.|||||.||.|||||.-||-||.||||||||.||||.|||.|||||||||||.|||||||||||||||||.||.||||||||||||||||||||||||||||||||.|||||.|||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||.|||||.|||||||||||||||||.||.|||||.||||||||||||||||||||||||||.||||||||.|||||.||.|||||||||||||||||||||||.-|||-|||||||||||||.|||||||||||||||||||||.|.|||||||||||.||-||.-|||||.|||||.||.|||||||||||.|||||.||||||||..|.||.||||||--|||-|-||.||.|||||.||||||||||-||-|.|||||.||||||||||||.|.|||||||||||.||.|||||.||||||||||||||||||||.||.|||||||||||.|..||||||||.|||||.|||||.||.||||||||||||||.||.||||||||.||||||||.|||||||||.|||||||.||||||||.|||||.||.|||||||||||||||||.|||||.||.|||||||||||.||.|||||.|||||||||||.||.-||-||.|||||.||||-|.-|.|||||||||||||||||.||||||.|..||||.||..-||-|||||.|.|||||||||||||||||||||||||||||.||.||||||||||||||||||.|||||||||||||.||||||||||||||||||||||||||.|||||||||||||||||.|||||||||||.||.-||||-||||||||||||||||||||||||.||||||||.|||||||||||||||||.||.|||||||||||.||||||||||||||||||||||||||.-||-|||||.|||||||||||||||||..|.||||||||||||||||||||.|||||.|||||.||||||||||||||.||||||||.||||||||||||||||||||||||||.||.||||||||.||.||.|||||.|||||||||||||||||||||||||||||||||||.||.||.||||||||.||..|.||||||||.|||||||||||||||||.||||||||||||||.||||||||||||||.|||||||||||.|||||||.|||||||||||||||.||||||||.|.||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||.|.|||||.||.|||||.-|||-||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.|||||||||||.||.|||||||||||.||.|||||.||.|||||.|||||||||||-|||-|||||||||||||||||.||||||||||||||||||||.||.||.|.|||||||-||-||.|||||-|||-|||||||||||.|||||..|.||.|||||||||||||||||.||.|||||.||.||..||||-||.|-|||||||.||||||...||||||||-|||-|||||||||.||||.|||||||||||||||||||||||||||||||||||.|||||||||||||||||.||..|.|||||||||.||.|.||||||||.||.||.||||||||.||||||||.||||||||||||||||||||.|||||||||||..|.|||||||||||.|||||.||||||||.|||||||||||.||.|||.|-|-|||||||||||||.|||.|||||||.||||||||.|||||.||.||.|||||.||.||||||||.|||||.|||||..||||.|||||.||.||||||||||||.||||.|||||.|||||.|||||.||.|||||.||.|||||.||||||.||||.--|.|||-|-|||-|-|.|||||.|||.||||.|||||.||.|||||.|||.|.||||||||-|-|.||-||.-||.||||||||..||||.|||||.|||.|.|||.||||-||..-|.|||||--|||--||||||||||||||.|.||||||||.||||||||||||||..||...||-|||-|||||||||||||||||||||||.|||||.|||||.||.||||||.||||.||||||||||||||.||.||.||.||||||||||||||||||||.|||||.|||||.|||||.||||||||.|||||||||||||||||.|||||.||||||||.||||||||.||.|||||.||||||||.||||||||.|||.|.||||||||.|||||||||||||||||||||||||||||||.|||..|.||.||||||||||||-||--||-|||||.||||||||.||.||||||||.|||||.|||||.|||.|.||..||||.|||||||||||.||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||.||||||||.||||||||||||||||||||.||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||||||||.||.|||||||||||.|||||||||||||||||.||||||||.|||||.||||||||.|||||.|||||.||.||||||||||||||.|||||||||||.|||||.||||||||.|||||.|||||.||.|||||||||||.|||||.|||||||||||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||.||||||||.||.||.|||||.|||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||.||.|||||||||||||||||||||||||||||||||||||||||||||||.||||||||-|||-|||||||||||||||||||||||||||||.||||||||||||||||||||.||||||||.|||||.||||||||||||||||||||.|||||||||||||..||||||||||||||||||||.||.||.||||||||.||||||||||||||.|||||||||||.|||||.||.||.|||||||||||||||||||-||-|.|||||||.|||||||||||||||||||||||||||||||||||||||.||||||||.|||||.|||||||||||||||||.|||.|||||||.||||||||||||||.|||||.||.|||||||||||||||||||||||||||||.||||||||.||||||||.||.||.||.|||||.|||||||||-||-||.|||||||||||.||.|||||.|||||||||||||||||.-||-||||||||||||||.|||||||||||||||||||||||.||.||.-|||-|.||.|||.||||.|||||.||||||||||||||.||||||||.||||||||.||.|||||.|||||.||.||.||.|-|-|||||.|.|||.||..|.||.||||||||.||||||||.|||||||||||-|-|||||||||||||.|||||.|||||.||.||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.|||||.|||||||||||||||||.|||||||||||||||||.|||||.|||||||||||.|||||..||||.||.|||||||||||.|||||.||.||.|||||||||||||||||||||||||||.||||.|||||.|||||.||.||.||.||.||||||||-|||-||.||||||||.||||||||||||||-||||-||||.||||||||.||.||.||-|||.-|||||||||||-|||-|.||.||||||||.||||||||.|||||.|||||.||||||||.|||||.||.|||||.|||||||||||||||||||||||.|||||||||||.||.|||||.||.||.|||||.||.||.|||||.-||-||||||||||||||||||-||.-|.||.||||||||.||-|.|-|||||.|||-||-|||||.||.|||||.||.||.||.|||||.||||||||||||||||||||.||||||||||||||.||||||||||||||.||.|||||||||||||||||.||.||.|||||.||.||.|||||||||||||||||.||.||.||||||||||||||||||||||||||.||||||||||||||.||.|||||||||||.||.|||||.||||||||.||.||.|||||.||.||||||||.||||||||.||.|||||.||||||||.||||||||.||||||||.-||-||.|||||.||.||.||.||||||||.|||||||||||.|||||||||||||||||||||||.|||||.|||||.||||||||||||||.|||||.||||||||.|||||.||.||||||-||-||||||||||||||.||.||.|||||||||||||||-||-||..|||||||.||||||||||||||.|||||.||.|||||.|||||||||||.||.||||||||.|||||||||||||||||||||||||||||||||||.|||.|||||||.||||||||.|||||.||||||||||||||.||||||.|.||||||||.||||||||||||.||||||||||.||||||||||||||||||||||||||..||||.||||||||||||||.|||||.||||||||.|||||||||||||||||||||||||||||.||||||||.||.|||||.|||||.||.|||||||||||||||||||||||||||||||||||.|||||||||||.||.||.|||||||||||.|||||.|||||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||||||||.||||||||||||||.||||||||||||||.|||||||||||.|||||||||||||||||.||||||||||||||||||||.||||||||||||||-|||-||.|||||.|||||..|||||||.||.||.||.||||||||.||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.|||||||.||.|||||.||.||||||||.||.||.||||||||||||||||||||||||||||||||.||.||||||||.||||||||||||||||||||||||||.||||||||||||||.||||||||||||||||||||.||.||||||||.||||||||.|||||||||||||||||.|||||||||||.|||||||||||||||||.||||||||.|||||.||.||||||||||||||.||.||||||||.||||||||.||.|||||||||||.|||||.|||||.|||||.|||||||||||.||.|||||||||||.|||||.||.||.||.|||||||||||||||||.|||||||||||.|||||||||||||||||.||.||.|||||||||||-|||.-|.||.|||||||||||||||||.||||||||.||.|||||||||||||||||||||||.|||||||||||.||.||.||||||||||||||||||.|.||||||||.||.|||||.||||||||-|.|-|||||.-||-|||||||||||||||||||||||||||||||||-||-|||||||..|||||.|||.|.||.|||.|.||||||||.|||||||||||.||||||||.||.|||||||||||.||.|||||.|||.|.||.|||||.-||-|||||.||||||||||||||.||.|||||||||||.|||||||||||.||||||||.-||-|||||.||||||||.||.|||||.|||||||||||||||.|.||||||||||||||||||||.||||||||||||||.||||||||||||||||||||.||||||||||||||.|||||||||||..|.||.||.|||||||||||.|||||.|||||.|||||.||||||||.||||||||.|||||.||.||.|||.||||..|.|||.||||.|||||.|||||||||||||||||.||||||||.||.||.||||||||--|||-|.|.-|-|.-|||||.|||.|||||||||.|||.|||||.||.||.||.|||||||||||.||.||.||||||||||||||||||||-|||-||.|||||.||.||||||||.||||||||.||||||.|.|||||.|||||.|||.-||-|.||.||||||||.||.||||||-||-|||||||||||||||||.-||-|||||.||||||||||||||.||.||.|||||.||..|.||.||||||||.|||||.|.|||.||.||.|||||.||.|||.|.||.||||||||||||||||||.||||.|||.|.||.|||||.||.||||||||.|||||||||||||||||.||.||.||||||||||||||||||||.|||||.||||||||||...|.||.|||||.|||||||||||.||||||||||||||||||||.|||||||||||.|||||.||.|||||.|||||||||||||||||.|||||.|||.||.|||||||.||.||.|||||.||||||||||||||||||||.||.||.|||||.||.||.||.||||||||||||.|||||||.-||-||.||.|||||||||||||||||.||.||.||.|||||-||||-||||||-|-|||||.||.||-|-||||.||.||.|||||.|||||.||.||.||||||||.|||||||||||||||||||||||||||.|.|||||.||.||||||||||||-|-||||||.||||||||||||||.|||.|.||||||||.||.||.||.|||.|.||.||.||||||.|.|||||.|||||.|||||.|||||||||||.|||||.-||||-|.|.||.||||||||||||||.|||||||||||.|||||.|||||.||||||||.||||||||||||||.|||||.|||..-|-||||.||||||.|.||.||-|||-|||||||||||||||||.||||||||||||||.|||||.||.|||||||||||.||||||||.|||||||||||||||||||||||.|||||.|||||.|||||||||||.||.||||||||||||||.||.|||||||||..|||-||.|-||||.|||||.||||||||.||.|||||.||||||-||-|||.||||.|||||||||||||||||.|||.|.|||||||||||.||-|||--||-||||||||||||||||||||||.|||||||.|.||.||||||||||||||||||||-|-|.|||||||.|||.||.||||||.|||||||..||||||||||.||-|||-||||||||||||||.||-||.-|||||||||||.||||||-||-|||||||||||.||||||||||||||||||||.||.|||||||||||.||.||||||.|.|||||||||||||||||||||||.||.||.||||||||||||||.||.|||||||||||.|||||.||.|||||.|||.|.|||||.||.|||||.|||.|.|||||||||||.||.||||||||.|||||.|.-|.||-||.||.||||.||||||||.|||||||||||||||||.||.||.||||||-||-||||||||.|||||||||||||||-|||.||-||||||||...|||||.||.|||-||-||.||-|.|-||.||...|||.||||||.|.||.|||||||||||||||||||||.|.||||||||||||||||||||-||-|||||||||.|||||..||.|.||.|||||.||.|||||.|||||||||||.||.|||.|.|||.|.||-|||.||-|||||-|-|||-||-||||||.|||||||-||-||..|.|||||||||||||||||.|||||.|||||..||||.||||||||.|||||.||.||||||||||||||||||||||||.|.||.|||||.||.-||-|||.||.||||.|||||.||.|||.|.|.-|-||||.||||-|.||.||.-||||.|||||||||||||||||.||.|||||.||||||||.||||||||||||||||||||.|||||||||||||||||.||.||||||||.||.-||-|||||||||||.|||||||||||||||||.||.||||||-||-||.||.|||||.||.||.||||||||.||.|||||.||||||||||||||||||||||||||||||||||||||.|||||.|||||.|||||||||||||||||||||||||||||.||||||||.||||||||||||||||||||.||.||||||||.||||||.||.|.|||||||||||.|||||||||||||||||||||||.|.|||.|||||||||||||||||||||||||||||.||.||||||||||||||.||||||||.|||-|||-|.||||||||.|||||||||||||||||||||||||||||.||.||.||||||||||||||.|||||.||.|||||||||-|||-|||||||||||||.||.|||||.||.|||||||||||.|||-||||-|||.|||||.|||||||||||.|||||.||.||||||.||||||||||.||||||||.||||||||||||||||||||||||||.||||||||.||.||||||||||||||||||||||||||.|||||.||.|||||.||.||||||||||||||||||||.||||||||.|||||..||||.||.|||||||||||.|||.|||||||||||||.|||||.||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||.|||||.|||||.||.||.||-|||-||||||||.|||||||||||||||||||||||.||.||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||.||.|||||||||||.||.|||||||||||||||||.|||||.|||||||||||.|||||||||||||||||.||||||||.||.||||||||||||||||||||.|||||||||||||||||.||||||||..|||||||.||.||.||||||.||||.||||||||||||||||||||||||||||||||||-||||-|||||||||||||-|--|-||-|||.-||||..-||..-|||||||||||-|||-|||..||||||.|.|.|-|..|||.--||-|.||||||||||-||||.-|.|||||||||||||||||||-||-|||-|-|||||---|.||.||.|.|-|-||.|-||||..|-|.|||||-||.||-||-|||.|||||||-||-|||---||--||-|-|||-|..|....|.|||||.|---||.||||||..||||||.|||.|..||.-||||-||.||.||..|.||.||||||.|||||||-|-|.||.||.||.||.||.||||||||.|||||.||.|||-||-||--|-||.||--|-||.||.||-|||-||.|||||.|||||.||.||||||.||||.||.||.|||-|||||..|||||-||.|||||||.|..-||.|-||.|||.||.||.||.|||||-||.|-|-|||-|.|.-||.||.||--|.|..|-||.||||-|-|||---||||||.||||||||.|||||.||..|.-|||||-|.||--|.||.||...||.||..-|-|||.-|-|||||.|.||||||.||.|||||-|||.||||-||-|-|||||..||||.||.||.||.||.|||...||.||-|.|||-||||.-|-||||.-||.-|||-|.||-|.|.||-|-||||.||--||||||||.||.||.|||.|-||.-||.||.||.||||..||-||-|||..|.||..||-||-||||||.|--|---|||-|||.-||..|..|..-|-||.|||||.|||..|.|.||.|||||||-|-|||||.||||||.||||-|.|-||.|||.|..|.|||||||||||.|||||.||||||||.||||||||.||.|||||||||||||||||.||.||.||||||||||||||.|||||-|||-|||||..||||.||.||||||||.|||||||||||..||||.|||||.||..|.|||.|.||||||||||||||||||||.||-|-||||.||.|||...-|||-|||||||.|||||.||.|||.|.||.||.|||||.|||-||-||-||.-|.|||--|.|-|.-||..||||.|||||..||||..|.||.|||||..|.||.|||||.||||||||.||||||||.||.||.|||.|||||||.|.||||.|.||||||||.|.|||.|||||.|||.|||||.|.||.|||..-|.|-|||||||||||..|.||.|||||.||||||||....||||||||.|||||||||||||||||.||||||||||||||.||.||.|||||||||||..||..|||-----|--|-|-||.||-|-|||||||||.-|-|||..||.|.-||.|||-||..||----|-|-|.|||||.|||||.||||||.||||--||.|.|-|.|..|-||.-|-|.-|.||||-||--|--|-|----|.--||--|-|-|--||||-|.---||..|-|-|||-|----|-||-|||--||..-|-----|||||.|-||||-|.||.||.||.-||-||.|||||.||||||.||||||||||.||.|||||||||||-||||--|-||.||.-||.|.|-|.|||||-||.-||.||.|||||||||||.||.||..|.|-|||-|||.|||||-|||-|||||.|||..|||.|.|||||.|.|.||||||||||||||.|.|||--|.|--|||||..||.||||.|.||.||||||||.||.|||||||||||.||||||||||||||.||.||||||||||||||||||||||||||||||||||||||.||||||.-|.-||||.|||||||||||.||.||||||||||||||.||.||.||...|.||||.||||||||.|||||.||.|||.||||||||.|||||.||||.||..|||||||.||||||||.|..|||||||||-|||-||||.||||||||||||||..-||-|.||||||||||||||-||||-|||||||.|||||.||.|||||.||.||.||.-|..-|||-|.-|--|.--||--||-|||..--|||.||.-||.|.||..|.||.||.||||||||.||||||||.||.||.||-|-|.|||||-|-||||.||.|||||.||||||||.|||||||||.|.||||||||||||||.|||.||.|.||.||.||.|||.|.||.||.||.|||||||||||.|||||.||.||||||||.||.|-|-||.||.|||||-|||-|||||.|||||||||||||||||.||||||.||||.||||||||||.|||.||.|||.||||.||||||||.|||||||||||.||||||||.|||||.||.|||||.||.|||||||||||.|||||.||||||||||||||||||||||||||.|||.||||.||.|||||.|||||.|||||||||||.|||||||||||-||-|||.|||||.|||||.||.|||||.|||||.||.||.||||||||.||.||||||||.-||||-||.||||.||||||||||..|.||||||||.||||||||.||.||.||||||||.||.|||||.||||||||||||||||||||.|||-|.-|||||.--|||||.|.-|||||||.-||...||||.||||.||||||||||||||.||||||||.-||-||||||||||||||.||||||||||||||||||||||||||.|||||||||||.||||||||||||||||||||.||.|||||.||.|||||.||.|||||||||||||||||.|||||.|||||.||--|||--|.||.|||||.|||||||||||||||-||-||.|||||.||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||.||.||.|||||.||.||||||||.|||||||||||.||||||||.||.||.||||||||||||||||||||.||||||||.|||||.|||||.||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||.|||||||||||.|||||||||||.||||||||.|||||||||||||||||||||||.||.|||||.||.|||||.||-|||-||||||||.|||||||||||.||||||||.|||||||||||..|.|||.|.||||||||||||||||||||.||||||||||||||.|||||.||.|||||||||||.||.|||||||||||-|||-|||||.||||||||||||||||||||||||||||||||.|||||||||||-|-||||.||.||.||.|||||.||||||||.|||||||||||||||||||||||.||||||.|||||||||||||||||||||||||.|||||.|||-||-||.|||||.||||||||||||||||||||||||||.||||||||||||||.|||.|||||-||-|||||.||.|||||||||||||||||.|||||.||.||||||||.||||||||||||||||||||||||||||-|||-||||||||||||.|||||.||||||||||||||||||||.||||||||.|||||.||||||||||||||..|||||||.||.||.||.|||||-||.-|||||.|||||.|||||.|||||..|||||||.||.|||||.|||||.||.||||||||||||||||||||||||||.|||||.|||||.||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||..||||||||||.|||-||.||-||||||||||||||-||-||||||||||.||.|||.|||||||.||||||||||.|||||||||||||||.||||||||||||||||||||.||||||||||||||.||||||||||||||||.||||||||.|||||.||.||.|||||||||||||||||..||||.|||||.||.||||||||.|||||||||||||||||-|-||.|||||.||||||||||.||.|||-||-|||||-|-||||||.||||-|||-|||||.|||||||||.|||||||||.||||-||-|-||-||||||.|.|.||||.|||||.||||||||||||.|.-||-|||-||-||||||||.|||||||||||.||.||.|||||.||||||||||||||.||||||||.|||||||||||||||||.||||||||||||||.||||||||||||||.||||||||||||||.||.|..||||||||.||.||-||-|||.||||||||..|.||||||||.|||||||||||.||.|||||.|||||.|||||.||.||.|||||||||||||||||.|||||.|||||.|||||.||.|||||||||||||||||.|||.|||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||.||||.||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||.|||.|..|||||||||.|||||||||||||||||||||||||||-||-|||||||||||||||||||||||.|||.|.||.|||||||||||||||-||.|-||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||-||-||||||||||||||||||||.|||||||||||||||||||||||||||||.||||||||.|||||||||||.||.||||||||.|||||||||||||||||||||||.|||||||||||||||||||||||.||||||||.|||||||||||||||||||||||.||.||||||||||||||||||.|.||||||||.||.||||||||.|||||||||||.|||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||||.||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||.||.||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||.||||||||.|||||||||||||||.||||||||.|.|.||||||||||||.|.||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||-||-|||||||.||||||||||||.|.||||.|.||||||||.|||||||||||||||.|.|.|||||||||.||||||||||||||||||||||||||.||||||||.||.||||||||||||||||||||||||||||||||.||||||||||||.||||||||||||||||||||||||||||||||||||||||.||.|.||||..||.||||||||||||.||||||||||||||.||||||||.|.||||||||||||||||||||||-||-|||||.||..||||.|||||.|||||.||.||||||||.||||.|||||||||||||.||||||||||.|.|||||||.||||.||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||.||||||||||||||||||||.|||||||||||||||||||||||||-||-||||||||||||||||||||||||||||||||||||||||-|.|-|||||.|||||||||||.||.||.|||||||||||||||||||||.||||.||||||||.|||||||||||||||||.|||||.|||||||||||.||-|.-|||||||||.|||||.|||||||||||||||||||||||.|||||||||||.||.|||||||-|-||.|||||.|||||||||||.|||||.||.||.||||||||.|||||||||||||-|-||||||||||||||||||||||||||||||||.||||||||.|||||.||.|||||||-|-||||||||||||.|||||||.||||||.||||||||||.|||||||||||||||||||||.||||||||||||||||||||||||||||||.||.||.|.|||||||||||||||.||||||||||||||||||||||.||.||.||.|.|.||.|||||.||||.|||.||.|||.|.|||||||||||.||||||||.|||||||||||||||||.||||||||||||||||||||||||||||||||||||.||||||..|||||||||||.|||||.||||||||.||||||||||||..|..||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||.|||||.||.|||||||||||.|||||.|||||||||||.|||||||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||.|||||.|||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||.||.||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||..||||||||||||||||||||||.||.||.||||||||||||||||.||||.|.|||||.|||||||||||||||||.|||||||||||||||||||||||||||||||||||.|||||.||||||||.||||||||||||||.||.|||-||-|||||.||.||.|||||-|||.||-|||||||||||||||||||||||||||||||||||||||||.|||.||||||||||||||||||||||||||||||||||||||||||||||||||||.||.||.||.||.||.|||||.||||||||.||||||||||||||||||||||||||||||||.|||.||||.||-||.-||.|||||.|||||||||||||||||||||-||||-|||||||-||-|||||..|||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||.||||||||||||-||-||.||.|||||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||||.|||||||||||.|||||||||||||||||..||.||||.|||||.|||||.|||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||.|||||||||||.||||||||||||||||||||||||.||||||||||||||||||||||||||||||||.||||||..||.||||.||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.-|..|...-|||||.||||||||||||||||||||||||||----\n",
      "AT-TAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACG-TCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTT-ATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGGCTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGATGCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTCGTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCTTCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTAGGCGACGAGCTTGGCACTGATCCTTAT-GAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTGTTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGGCCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTC--ATGCACTTTGTCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTGCTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAAATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCC-CTTAAATTCCATAATCAAGACTATTCAACCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTT-ATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCACCAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCAGACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACTACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAA-TTTATTGTCCAGCATGTCACAATTCAGAAGTAGGACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCGCACTATTGCCTT-TGGAGGCTGTGTGTTCT-CTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCACGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACAACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGAGATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGATTATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAA-AAAGGTGCCTGGAATATTGGTGAACAGAAAT-CAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCGTGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCCGCTATAACAATACT-AGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTGATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTGGCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTTAAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAATTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGTAAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTAGGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCCTACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTTAACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAAGCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGTACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAAGGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTTGATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAAATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACCACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAATTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAGAAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATTTGGTGCCACTTCTGCTGCT-CTTCAAC-CTG-AAGAAGAGCAAGAAGAAGATTGGTTAGA-TGATGATAGTCAACAAACTGTTGGTCAACAAGACGGCAGT-GAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTCAACCTCAATTAG-AGATGGAACTTA-CACCAGTTGTT-CAGA--CTATTGAAGTGAATAGTTTTAGTGGTTATTTAAAACTTA-CTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACAGTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTACTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAA-GTGGGTGGTAGTTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGTGAAGACATTCAACTT-CTTAAGAGTGCTTATGAAAATTTTAATCAGCACGA-AGTTCTACTTGCACCATTATTATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAATGTCTA-CTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGTGAAAAGCAAGTTGAACAAAA-GATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTAAACCTTCAGTTGAACAGAGAAAACAAGATGAT-A-A-GAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAACTCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCAGATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTGATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAATGCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAATGGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTATTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGCAGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAATATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAACAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTT-GGCTATGTAACACATGG-CTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTTTCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAGAACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACAACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCACCTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTAAGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACAACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGTAAAACATTTTATGTTTTACCTAATGATGACACTCT-ACGTGTTGAGGCTTTTGAGTACTACCACACAACTGATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAATGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAAATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTAACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAATGAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGTGGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAATTTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTCACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGTGAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAGACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAGTTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAGTTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAACCATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAACCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATT-TTTCCCTGACTTAAATGGTGATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAACCTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTGTCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTG-AAGTCAGAGGACGCGCAGGGAATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGAAAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAATAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTTACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTT-GCTACTCATGGTTTAGCTGCTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTACAACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTATTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAA-AGCATCTATGCCGACTACTATAGCAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAATTTTTCTAAACTGATAAATATTATAATTTGGTTTTTACTATTAAGTGTTTGCCTAGGTTCTTTAATCTACTCAACCGCTGCTTTAGGTGTTTTAATGTCTAATTTAGGCATGCCTTCTTACTGTACTGGTTACAGAGAAGGCTATTTGAACTCTACTAATGTCACTATTGCAACCTACTGTACTGGTTCTATACCTTGTAGTGTTTGTCTTAGTGGTTTAGATTCTTTAGACACCTATCCTTCTTTAGAAACTATACAAATTACCATTTCATCTTTTAAATGGGATTTAACTGCTTTTGG-CTTAGTTGCAGAGTGGTTTTTGGCATATATTCTTTTCACTAGGTT-TTTCTATGTACTTGGATTGGCTGCAATCATGCAATTGTTTTTCAGCTATTTTGCAGTACATTTTATTAGTAATTCTTGGCTTATGTGGTTAATAATTAATCTTGTACAAATGGCCCCGATTTCAGCTATGGTTAGAATGTACATCTTCTTTGCATCATTTTATTATGTATGGAAAAGTTATGTGCATGTTGTAGACGGTTGTAATTCATCAACTTGTATGATGTGTTACAAACGTAATAGAGCAACAAGAGTCGAATGTACAACTATTGTTAATGGTGTTAGAAGGTCCTTTTATGTCTATGCTAATGGAGGTAAAGGCTTTTGCAAACTACACAATTGGAATTGTGTTAATTGTGATACATTCTGTGCTGGTAGTACATTTATTAGTGATGAAGTTGCGAGAGACTTGTCACTACAGTTTAAAAGACCAATAAATCCTACTGACCAGTCTTCTTACATCGTTGATAGTGTTACAGTGAAGAATGGTTCCATCCATCTTTACTTTGATAAAGCTGGTCAAAAGACTTATGAAAGACATTCTCTCTCTCATTTTGTTAACTTAGACAACCTGAGAGCTAATAACACTAAAGGTTCATTGCCTATTAATGTTATAGTTTTTGATGGTAAATCAAAATGTGAAGAATCATCTGCAAAATCAGCGTCTGTTTACTACAGTCAGCTTATGTGTCAACCTATACTGTTACTAGATCAGGCATTAGTGTCTGATGTTGGTGATAGTGCGGAAGTTGCAGTTAAAATGTTTGATGCTTACGTTAATACGTTTTCATCAACTTTTAACGTACCAATGGAAAAACTCAAAACACTAGTTGCAACTGCAGAAGCTGAACTTGCAAAGAATGTGTCCTTAGACAATGTCTTATCTACTTTTATTTCAGCAGCTCGGCAAGGGTTTGTTGATTCAGATGTAGAAACTAAAGATGTTGTTGAATGTCTTAAATTGTCACATCAATCTGACATAGAAGTTACTGGCGATAGTTGTAATAACTATATGCTCACCTATAACAAAGTTGAAAACATGACACCCCGTGACCTTGGTGCTTGTATTGACTGTAGTGCGCGTCATATTAATGCGCAGGTAGCAAAAAGTCACAACATTGCTTTGATATGGAACGTTAAAGATTTCATGTCATTGTCTGAACAACTACGAAAACAAATACGTAGTGCTGCTAAAAAGAATAACTTACCTTTTA-AGTTGACATGTGCAACTACTAGACAAGTTGTTAATGTTGTAACAACAAAGATAGCACTTAAGGGTGGTAAAATTGTTAATAATTGGTTGAAGCAGTTAATTAAAGTTACACTTGTGTTCCTTTTTGTTGCTGCTATTTTCTATTTAATAACACCTGTTCATGTCATGTCTAAACATACTGACTTTTCAAGTGAAATCATAGGATACAAGGCTATTGATGGTGGTGTCACTCGTGACATAGCATCTACAGATACTTGTTTTGCTAACAAACATGCTGATTTTGACACATGGTTTAGCCAGCGTGGTGGTAGTTATACTAATGACAAAGCTTGCCCATTGATTGCTGCAGTCATAACAAGAGAAGTGGGTTTTGTCGTGCCTGGTTTGCCTGGCACGATATTACGCACAACTAATGGTGACTTTTTGCATTT-CTTACCTAGAGTTTTTAGTGCAGTTGGTAACATCTGTTACACACCATCAAAACTTATAGAGTACACTGACTTTGCAACATCAGCTTGTGTTTTGGCTGCTGAATGTACAATTTTTAAAGATGCTTCTGGTAAGCCAGTACCATATTGTTATGATACCAATGTACTAGAAGGTTCTGTTGCTTATGAAAGTTTACGCCCTGACACACGTTATGTGCTCATGGATGGCTCTATTATTCAATTTCCTAACACCTACCTTGAAGGTTCTGTTAGAGTGGTAACAACTTTTGATTCTGAGTACTGTAGGCACGGCACTTGTGAAAGATCAGAAGCTGGTGTTTGTGTATCTACTAGTGGTAGATGGGTACTTAACAATGATTATTACAGATCTTTACCAGGAGTTTTCTGTGGTGTAGATGCTGTAAATTTACTTACTAATATGTTTACACCACTAATTCAACCTATTGGTGCTTTGGACATATCAGCATCTATAGTAGCTGGTGGTATTGTAGCTATCGTAGTAACATGCCTTGCCTACTATTTTATGAGGTTTAGAAGAGCTTTTGGTGAATACAGTCATGTAGTTGCCTTTAATACTTTACTATTCCTTATGTCATTCACTGTACTCTGTTTAACACCAGTTTACTCATTCTTACCTGGTGTTTATTCTGTTATTTACTTGTACTTGACATTTTATCTTACTAATGATGTTTCTTTTTTAGCACATATTCAGTGGATGGTTATGTTCACACCTTTAGTACCTTTCTGGATAACAATTGCTTATATCATTTGTATTTCCACAAAGCATTTCTATTGGTTCTTTAGTAATTACCTAAAGAGACGTGTAGTCTTTAATGGTGTTTCCTTTAGTACTTTTGAAGAAGCTGCGCTGTGCACCTTTTTGTTAAATAAAGAAATGTATCT-AAAGTTGCGTAGTGATGTGCTAT-TACCTCTTACGCAATATAATAGATACTTAGCTCTTTATAATAAGTACAAGTATTTTAGTGGAGCAATGGATACAACTAGCTACAGAGAAGCTGCTTGTTGTCATCTCGCAAAGGCTCTCAATGACTTCAGTAACTCAGGTTCTGATGTTCTTTACCAACCACCACAAACCTCTATCACCTCAGCTGTTTTGCAGAGTGGTTTTAGAAAAATGGCATTCCCATCTGGTAAAGTTGAGGGTTGTATGGTACAAGTAACTTGTGGTACAACTACACTTAACGGTCTTTGGCTTGATGACGTAGTTTACTGTCCAAGACATGTGATCTGCACCTCTGAAGACATGCTTAACCCTAATTATGAAGATTTACTCATTCGTAAGTCTAATCATAATTTCTTGGTACAGGCTGGTAATGTTCAACTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTTAAGCTTAAGGTTGATACAGCCAATCCTAAGACACCTAAGTATAAGTTTGTTCGCATTCAACCAGGACAGACTTTTTCAGTGTTAGCTTGTTACAATGGTTCACCATCTGGTGTTTACCAATGTGCTATGAGGCCCAATTTCACTATTAAGGGTTCATTCCTTAATGGTTCATGTGGTAGTGTTGGTTTTAACATAGATTATGACTGTGTCTCTTTTTGTTACATGCACCATATGGAATTACCAACTGGAGTTCATGCTGGCACAGACTTAGAAGGTAACTTTTATGGACCTTTTGTTGACAGGCAAACAGCACAAGCAGCTGGTACGGACACAACTATTACAGTTAATGTTTTAGCTTGGTTGTACGCTGCTGTTATAAATGGAGACAGGTGGTTTCTCAATCGATTTACCACAACTCTTAATGACTTTAACCTTGTGGCTATGAAGTACAATTATGAACCTCTAACACAAGACCATGTTGACATACTAGGACCTCTTTCTGCTCAAACTGGAATTGCCGTTTTAGATATGTGTGCTTCATTAAAAGAATTACTGCAAAATGGTATGAATGGACGTACCATATTGGGTAGTGCTTTATTAGAAGATGAATTTACACCTTTTGATGTTGTTAGACAATGCTCAGGTGTTAC-TTTCCAAAGTGCAGTGAAAAGAACAATCAAGGGTACACACCACTGGTTGTTACTCACAATTTTGACTTCACTTTTAGTTTTAGTCCAGAGTACTCAATGGTCTTTGTTCTTTTTTTTGTATGAAAATGCCTTTTTACCTTTTGCTATGGGTATTATTGCTATGTCTGCTTTTGCAATGATGTTTGTCAAACATAAGCATGCATTTCTCTGTTTGTTTTTGTTACCTTCTCTTGCCACTGTAGCTTATTTTAATATGGTCTATATGCCTGCTAGTTGGGTGATGCGTATTATGACATGGTTGGATATGGTTGATACTAGTTTGTCTGGTTTTAAGCTAAAAGACTGTGTTATGTATGCATCAGCTGTAGTGTTACTAATCCTTATGACAGCAAGAACTGTGTATGATGATGGTGCTAGGAGAGTGTGGACACTTATGAATGTCTTGACACTCGTTTATAAAGTTTATTATGGTAATGCTTTAGATCAAGCCATTTCCATGTGGGCTCTTATAATCTCTGTTACTTCTAACTACTCAGGTGTAGTTACAACTGTCATGTTTTTGGCCAGAGGTATTGTTTTTATGTGTGTTGAGTATTGCCCTATTTTCTTCATAACTGGTAATACACTTCAGTGTATAATGCTAGTTTATTGTTTCTTAGGCTATTTTTGTACTTGTTACTTTGGCCTCTTTTGTTTACTCAACCGCTACTTTAGACTGACTCTTGGTGTTTATGATTACTTAGTTTCTACACAGGAGTTTAGATATATGAATTCACAGGGACTACTCCCACCCAAGAATAGCATAGATGCCTTCAAACTCAACATTAAATTGTTGGGTGTTGGTGGCAAACCTTGTATCAAAGTAGCCACTGTACAGTCTAAAATGTCAGATGTAAAGTGCACATCAGTAGT-CTTACTCTCAGTT-TTGCAACAACTCAGAGTAGAATCATCATCTAAATTGTGGGCTCAATGTGTCCAG-TTACACAATGACATTCT-CTTAGCTAAAGATACTACTGAAGCCTTTGAAAAAATGGTTTCACTACTT-TCTGTTTTGCTTTCCATGCAGGGTGCTGTAGACATAAACAAGCTT-TGTGAAGAAATGCTGGACAACAGGGCAACC-TTACAAGCTATAGCCTCAGAGTTTAGTTCCCTT-CCATCATATGCAGCTTTTGCTACTGCTCAAGAAGCTTATGAGCAGGCTGTTGCTAATGGTGATTCTGAAGTTGTTCTTAAAAAGTTGAAGAAGTCTTTGAATGTGGCTAAATCTGAATTTGACCGTGATGCAGCCATGCAACGTAAGTTGGAAAAGATGGCTGATCAAGCTATGACCCAAATGTATAAACAGGCTAGATCTGAGGACAAGAGGGCAAAAGTTACTAGTGCTATGCAGACAATGCTTTTCACTATGCTTAGAAAG-TTGGATAATGATGCACTCAACAACATTATCAACAATGCAAGAGATGGTTGTGTTCC-CTTGAACATAATACCTCTTACAACAGCAGCCAAACTAATGGTTGTCATACCAGACTAT--AACACATATAAAAATACGTGTGATGGTA-CAACATTTACTTATGCATCAGCATTGTGGGAAATCCAACAGGTTGTAGATGCAGATAGTAAAATTGTTCAACTTAGTGAAATTAGTATGGACAATTCACCTAATTTAGCATGGCCTCTTATTGTAACAGCTTTAAGGGCCAATTCTGCTGTCAAATTACAGAATAATGAGCTTAGTCCTGTTGCACTACGACAGATGTCTTGTGCTGCCGGTACTACACAAACTGCTTGCACTGATGACAATGCG-TTAGCTTACTACAACACAA-CAAAGGGAGGTAGGTTTGTACTTGCACTGTTATCCGATTTAC-AGGATTTGAAATGGGCTAGATTCCCTAAGAGTGATGGAACTGGTACTATCTATACAGAACTGGAACCACCTTGTAGGTTTGTTACAGACACACCTAAAGGTCCTAAAGTGAAGTATTTATACTTTATTAAAGGA-TTAAACAACCTAAATAGAGGTATGGTACTTGGTAGTTTAGCTGCCACAGTACGTCTACAAGCTGGTAATGCAACAGAAGTGCCTGCCAATTCAACTGTA-TTATCTTTCTGTGCTTTTGCTGTAGATGCTGCTAAAGCTTACAAAGATTATCTAGCTAGTGGGGGACAACCAATCACTAATTGTGTTAAGATGTTGTGTACACACACTGGTACTGGTCAGGCAATAACAGTTACACCGGAAGCCAATATGGATCAAGAATCCTTTGGTGGTGCATCGTGTTGTCTGTACTGCCGTTGCCACATAGATCATCCAAATCCTAAAGGATTTTGTGACTTAAAAGGTAAGTATGTACAAATACCTACAACTTGTGCTAATGACCCTGTGGGTTTTACACTTAAAAACACAGTCTGTACCGTCTGCGGTATGTGGAAAGGTTATGGCTGTAGTTGTGATCAACTCCGCGAACCCATGCTTCAGTCAGCTGATGCA-CAATCGTTTTTAAACGGGTTTGCGGTGTAAGTGCAGCCCGTCTTACACCGTGCGGCACAGGCACTAGTACTGATGTCGTATACAGGGCTTTTGACATCTACAATGATAAAGTAGCTGGTTTTGCTAAA-TTCCTAAAAACTAATTGTTGTCGCTTCCAAGAAAAGGACGAAGATGACAATTTAATT-GATTCTTACTTT-GTAGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAA-GGATTGTCCAGCTGTTGCTAAACATGACTTCTTT-AAGTTTAGAATAGACGGTGACATGGTACCACATATATCACGTCAACGTCTTACTAAATACACAATGGCAGACCTCGTCTATGCTTTAAGGCATTTTGATGAAGGTAATTGTGACACATTAAAAGAAATACTTGTCACATACAATTGTTGTGATGATGATTATTTCAATAAAAAGGACTGGTATGATTTTGTAGAAAACCCAGATATAT-TACGCGTATACGCCAACTTAGGTGAACGTGTACGCCAAGCTTTGTTAAAAACAGTACAATTCTGTGATGCCATGCGAAATGCTGGTATTGTTGGTGTACTGACATTAGATAATCAAGATCTCAATGGTAACTGGTATGATTTCGGTGATTTCATACAA--ACCACGCCAGG-TAGTGGAGTTCCTGTTGTAGATTCTTATTATTCATTGTTAATGCCTATAT-TAAC-CTTGACCAGGGCTTTAACTGCAGAGTCACATGTTGACACTGA-CTTAACAAAGCC--TTACATTAAGTGGGATTTGTTAAAATATGACTTCACGGAAGAGAGGTTAAAACTCTTT-GACCGTTATTTTAAATATTGGGATCAGACATACCACCCAAATTGTGTTAACTGTTTGGATGACAGATGCATTCTGCATTGTGCAAACTTTAATGTTTTATTCTCTACAGTGTTCCCACCTACAAGTTTTGGACCACTAGTGAGAAAAATATTTGTTGATGGTGTTCCATTTGTAGTTTCAACTGGATACCACTTCAGAGAGCTAGGTGTTGTACATAATCAGGATGTAAACTTACATAGCTCTAGACTTAGTTTTAAGGAA-TTACTT-GTGTATGCTGCTGACCCTGCTATGCACGCTGCTTCTGGTAATCTATTACTAGATAAACGCACTACGTGCTTTTCAGTAGCTGCACTTACTAACAATGTTGCTTTTCAAACTGTCAAACCCGGTAATTTTAACAAAGACTTCTATGACTTTGCTGTGTCTAAGGGTTTCTTTAAGGAAGGAAGTTCTGTTGAATTAAAACACTTCTTCTTTGCTCAGGATGGTAATGCTGCTATCAGCGATTATGACTACTATCGTTATAATCTACCAACAATGTGTGATATCAGACAACTACTATTTGTAGTTGAAGTTGTTGATAAGTACTTTGATTGTTACGATGGTGGCTGTATTAATGCTAACCAAGTCATCGTCAACAACCTAGACAAATCAGCTGGTTTTCCATTTAATAAATGGGGTAAGGCTAGACTTTATTATGATTCAATGAGTTATGAGGATCAAGATGCACTTTTCGCATATACAAAACGTAATGTCATCCCTACTATAACTCAAATGAATCTTAAGTATGCCATTAGTGCAAAGAATAGAGCTCGCACCGTAGCTGGTGTCTCTATCTGTAGTACTATGACCAATAGACAGTTTCATCAAAAATTATTGAAA-TCAATAGCCGCCACTAGAGGAGCTACTGTAGTAATTGGAACAAGCAAATTCTATGGTGGTTGGCACAACATGTTAAAAACTGTTTATAGTGATGTAGAAAACCCTCACCTTATGGGTTGGGATTATCCTAAATGTGATAGAGCCATGCCTAACATGCTTAGAATTATGGCCTCACTTGTTCTTGCTCGCAAACATA-CAACGTGTTGTAGCTTGTCACACCGTTTCTATAGATTAGCTAATGAGTGTGCTCAAGTATTGAGTGAAATGGTCATGTGTGGCGGTTCACTATATGTTAAACCAGGTGGAACCTCATCAGGAGATGCCACAACTGCTTATGCTAATAGTGTTTTTAACATTTGTCAAGCTGTCACGGCCAATGTTAATGCACTT-TTATCTACTGATGGTAACAAAATTGCCGATAAGTATGTCCGCAAT-TTACAACACAGACTTTATGAGTGTCTCTATAGAAATAGAGATGTTGACACAG-ACTTTGTGAATGAGTTTTACGCATATTTGCGTAAACATTTCTCAATGATGATACTCTCTGACGATGCTGTTGTGTGTT-TCAATAGCACTTATGCATCTCAAGGTCTAGTGGCTAGCATAAAGAACTTTAA-GTCAGTTCTTTATTATCAAAACAATGTTTTTATGTCTGAAGCAAAATGTTGGACTGAGACTGACCTTACTAAAGGACCTCATGAATTTTGCTCTCAACATACAATGCTAGTTAAACAGGGTGATGATTATGTGTACCTTCCTTACCCAGATCCATCAAGAATCCTAGGGGCCGGCTGTTTTGTAGATGATATCGTAAAAACAGATGGTACACTTATGATTGAACGGTTCGTGTCTTTAGCTATAGATGCTTACCCACTTACTAAA-CATCCTAATCAGGAGTATGCTGATGTCTTTC-ATTTGTACTTACAATACATAAG-AAAGCTACATGATGAG-TTAACAGGACACATGTTAGACATGTATTCTGTTATGCTTACTAATGATAACACTTCAAGGTATTGGGAACCTGAGTTTTATGAGGCTATGTACACACCGCATACAGTCTTACAGGCTGTTGGGGCTTGTGTTCTT-TGCAATTCACAGACTTCA-TTAAGATGTGGTGCTTGCAT-ACGTAGACCATTC-TTATGTTGTAAATGCTGTTACGACCATGTCATATCAACATCACATAAATTAGTCTTGTCTGTTAATCCGTATGTTTGCAATGCTCCAGGTTGTGATGTCACAGATGTGACTCAACTTTACTTAGGAGGTATGAGCTATTATTGTAAATCACATAAACCACCCATTAGTTTTCCATTGTGTGCTAATGGACAAGTTTTTGGTTTATATAAAAATACATGTGTTGGTAGCGATAATGTTACTGACTTTAATGCAATTGCAACATGTGACTGGACAAATGCTGGTGATTACATT-TTAGCTAACACCTGTACTGAAAGACTCAAGCTTTTTGCAGCAGAAACGCTCAAAGCTACTGAGGAGACATTTAAACTGTCTTATGGTATTGCTACTGTACGTGAAGTGCTGTCTGACAGAGAA-TTACATCTTTCATGGGAAGTTGGTAAACCTAGACCACCACTT-AACCGAAATTATGTCTTTACTGGTTATCGTGTAACTAAAAACAGTAAAGTACAAATAGGAGAGTACACCTTTGAAAAAGGTGACTATGGTGATGCTGTTGTTTACCGAGGTACAACAACTTACAAATTAAATGTTGGTGATTATTTTGTGCTGACATCACATACAGTAATGCCATTAAGTGCACCTACACTAGTGCCACAAGAGCACTATGTTAGAATTACTGGCTTATACCCAACACTCAATATCTCAGATGAGTTTTCTAGCAATGTTGCAAATTATCAAAAGGTTGGTATGCAAAAGTATTCTACACTCCAGGGACCACCTGGTACTGGTAAGAGTCATTTTGCTATTGGCCTAGCTCTCTACTACCCTTCTGCTCGCATAGTGTATACAGCTTGCTCTCATGCCGCTGTTGATGCACTATGTGAGAAGGCATTAAAATATTTGCCTATAGATAAATGTAGTAGAATTATACCTGCACGTGCTCGTGTAGAGTGTTTTGATAAATTCAAAGTGAATTCAACATTAGAACAGTATGTCTTT-TGTACTGTAAATGCATTGCCTGAGACGACAGCAGATATAGTTGTCTTTGATGAAATTTCAATGGCCACAAATTATGATTTGAGTGTTGTCAATGCCAGATTACGTGCTAAGCACTATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTGCTGAAATTGTTGACACTGTGAGTGCTTTGGTTTATGATAATAAGCTTAAAGCACATAAAGACAAATCAGCTCAATGCTTTAAAATGTTTTATAAGGGTGTTATCACGCATGATGTTTCATCTGCAATTAACAGGCCACAAATAGGCGTGGTAAGAGAATTCCTTACACGTAACCCTGCTTGGAGAAAAGCTGTCTTTATTTCACCTTATAATTCACAGAATGCTGTAGCCTCAAAGATTTT-GGGACTACCAACTCAAACTGTTGATTCATCACAGGGCTCAGAATATGACTATGTCATATTCACTCAAACCACTGAAACAGCTCACTCTTGTAATGTAAACAGATTTAATGTTGCTATTACCAGAGCAAA-AGTAGGCATACTT-TGCATAATGTCTGATAGAGACCTTTATGACAAG-TTGCAATTTACAAGTCTTGAAATTCCACGTAGGAATGTGGCAACTTTACAAGCTGAAAATGTAACAGGACTCTTTAAAGATTGTAGTAAGGTAATCACTGGG-TTACATCCTACACAGGCACCTACACACCTCAGTGTTGACACTAAATTCAAAACTGAAGGT-TTATGTGTTGACATACCTGGCATACCTAAGGACATGACCTATAGAAGACTCATCTCTATGATGGGTTTTAAAATGAATTATCAAGTTAATGGTTACCCTAACATGTTTATCACCCGCGAAGAAGCTATAAGACATGTACGTGCATGGATTGGCTTCGATGTCGAGGGGTGTCATGCTACTAGAGAAGCTGTTGGTACCAATTTACCTTTACAGCTAGGTTTTTCTACAGGTGTTAACCTAGTTGCTGTACCTACAGGTTATGTTGATACACCTAAT-A-AT-ACAGATTTTTCCAGAGTTAGTGCTAAACCACCGCCTGGAGATCAATTTAAACACCTCATACCACTTATGTACAAAGGACTT-CCTTGGAATGTAGTGCGTATAAAGATTGTACAAATGTTAAGTGACACACTTAAAA-ATCTCTCTGACAGAGTCGTATTTGTC-TTATGGGCACATGGCTTTGAG-TTGACATCTATGAAGTATTTTGTGAAAATAGGACCTGAGCGCACCTGTTGTCTATGTGATAGACGTGCCACATGCTTTTCCACTGCTTCAGACACTTATGCCTGTTGGCATCATTCTATTGGATTTGATTACGTCTATAATCCGTTTATGATTGATGTTCAACAATGGGGTTTTACAGGTAACCTACAAAGCAACCATGATCTGTATTGTCAAGTCCATGGTAATGCACATGTAGCTAGTTGTGATGCAATCATGACTAGGTGTCTAGCTGTCCACGAGTGCTTTGTTAAGCGTGTTGACTGGACTATTGAATATCCTATAATTGGTGATGAACTGAAGATTAATGCGGCTTGTAGAAAGGTTCAACACATGGTTGTTAAAGCTGCATTA-TTAGCAGACAAATTCCCAGTTCTTCACGACATTGGTAACCCTAAAG-CTATTA-AGTGTGTACCTCAAG-CTGATGTAGAATGGAAGTTCTATGATGCACAGCCTTGTAGTGACAAAGCTTATAAAATAGAAGAATTATTCTATTCTTATGCCACACATTC-TGACAAATTCACAGATGGTGTATGCCTATTTTGGAATTGCAATGTCGATAGATATCCTGCTAATTCCATTGTTTGTAGATTTGACACTAGAGTGCTATCTAACCTTAA-CTTGCCTGGTTGTGATGGTGGCAGTTTGTATGTAAATAAACATGCATTCCACACACCAGCTTTTGATAAAAGTGCTTTTGTTA-ATTTAAAACAATTACCATT-TTTCTATTACTCTGACAGTCCATGTGAGTCTCATGGAAAACAAGTAGTGTCAGATATAGATTATGTACCACTAAAGTCTGCTACGTGTATAACACGTTGCAATTTAGGTGGTGCTGTCTGTAGACATCATGCTAATGAGTACAGATTGTATCTCG-ATGCTTATAACATGATGATCTCAGCTGGCTTTAGC-TTGTGGGTTTACAAACAATTTGATACTTATAACCTCTGGAACACTTTTAC-AAGACTT-CAGAGTTTAGAAAATGTGGCTTTTAATGTTGTAAATAAGGGACACTTTGATGGACA-ACAGGGTGAAGTACCAGTTTCTATCATTAATAACACTGTTTACACAAAAGTT-GATGGTGTTGATGTAGA-ATTGTTTGAAAATAAAACAACA-TTACCTGTTAATGTAGCATTTGAGCTTTGGGCTAAGCGCAACATTAAACCAGTACCAGAGGTGAAAATACTCAATAATTTGGGTGTGGACATTGCTGCTAATACTGTGATCTGGGACTACAAAAGAGATGCTCCAGCACATATATCTACTATTGGTGTTTGTTCTATGACTGACATAGCCAAGAAACCAACTGAAAC-GATTTGTGCACCACTCACTGTCTTTTTTGATGGTAGAGTTGATGGTCAAGTAGAC-TTATTTAGAAATGCCCGTAATGGTGTTCTTATTA-CAGAAGGTAGTGTTAAAGGTTTA-CAACCATCTGTA-GGTCCCAAACAAGCTAGTCTTAATGGAGTCACATTAATTGGAGAAGCCGTAAAAACACAGTTCAATTA-TTATAAGAAAGTTGATGGTGTTGTCCAACAATTACCTGAAACTTACTTTACTCAGAGTAGAAATTTACAAGAATTTAAA-CCCAGGAGTCA-AA-TGGAAATTGATTTC-TTAGAATTAGCTATGGATGAATTCATTGAACGGTATAAATTAGAAGGCTATGCCTTCGAACATATCGTTTATGGAGATTTTAGTCATAGTCAGTTAGGTGGT-TTACATCTACTGATTGGACTAGCTAAACGTTTTA-AGGAATCACCTTTTGAAT-TAGAAGATTTTATTCCTATGGACAGTACAGTTAAAAACTATTTCATAACAGATGCGCAAACAGGTTCATCTAAGTGTGTGTGTTCTGTTATTGAT-TTATTACTTGATGATTTTGTTGAAATAATAAAATCCCAAGAT-TTATCTGTAGTTTCTAAGGTTGTCAAAGTGACTATTGACTATACAGAAATTTCATTTATGCTTTGGTGTAAAGATGGCCATGTAGAAACATTTTACCCAAAATTACAATCTAGTCAAGCGTGGCAACCGGGTGTTGCTATGCCTAATCTTTACAAAATGCAAAGAATGCTATTAGAAAAGTGTGACCTTCAAAATTATGGTGATAGTGCAACATTACCTAAAGGCATAATGATGAATGTCGCAAAATATACTCAACTGTGTCAATATTTAAACACA-TTAACATTAGCTGTACCCTATAATATGAGAGTTATACATTTTGGTGCTGGTTCTGATAAAGGAGTTGCACCAGGTACAGCTGTT-TTAAGACAGTGGTTGCCTACGGGTACGCTGCTTGTCGATTCAGATCTTAA-TGACTTTGTCTCTGATGCAGATTCAACTTTGATTGGTGATTGTGCAACTGTACATACAGCTAATAAATGGGATCTCATTATTAGTGATATGTACGACCCTAAGACTAAAAATGTTACAAAAGAAAATGACTCTAAAGAGGGTTTTTTCACTTACATTTGTGGGTTTATACAACAAAAGCTAGCTCTTGGAGGTTCCGTGGCTATAAAGATAACAGAACATTCTTGGAATGCTGATCTTTATAAGCTCATGGGACACTTCGCATGGTGGACAGCCTTTGTTACTAATGTGAATGCGTCATCATCTGAAGCATTT-TTAATTGGATGTAATTATCTTGGCAAACCACGCGAACAAATAGATGGTTATGTCATGCATGCAAATTACATATTTTGGAGGAATACAAATCCAATTCAGTTGTCTTCCTATTCTTTATTTGACATGAGTAAATTTCCCCTTAAATTAAGGGGTACTGCTGTTATGTCTTTAAAAGAAGGTCAAATCAATGATATGATTTTATCTCTTCTTAGTAAAGGTAGACTTATAATTAGAGAAAACAACAGAGTTGTTATTTCTAGTGATGTTCTTGTTAACAACTAAACGAACAATGTTTGTTTTTCTTGTTT-T--A-TT-GCCACTAGTCT-CTAGTCAGTGTGTTAATCTT-ACAACCAGAACTCAATT-ACCCCCTGCAT-ACACTAATTCTTTCACAC-GTGGTGTTTATTACCCTGACA-AAGTTT-TCAGATC---CTCAGTTTTACATTCAACTCAGGACT-TGTTCTTACCTTTCTT-TTCCAATGTTA-CTTGGTTCCATGCTA-TACATGTCTCTGGGACCAATGGT---ACTAAGAGGTTTGATAACCCTGTCCTACCATTT-AATGATGGTGTTTATTTTGCTTCCACTGAGA-AGTCTAACATAATAAGAGGCTGGATTTTTGGTACTACT-TTAGATTCGAAGAC--C-CAGTCCCTACTT-ATTGTTAATAACGCTACTAATGTTGTTATTAAAGTCTGTGAATTTCAATTTT-GTAATGATCCATTT-TTGGGTGTTTATTACCACAAAAACAACAAAAGTTGGATGGAA-AGTGAGTT--CAGAGT-TTATTCT-AGTGCGAATAATTGCACTTTTGAATATGTCTCTCAG-CCTTT-TCTTATGGACCTTGAAGGAAAACAG-GGTAAT-TTCAAAAATCTTAGGGAATTTGTGTTTAAGAATAT-T-GATGGTTATTTTAAAATATATTCTAAGCACACGCCTATTAA-TTTAG-TGCGTGATCTC-CCT-CAGGGTTTTTC-G-GCTTTAGAACCATTGGTAGATTTGCCAAT-AGGTATTAACATCACTAGGTTTCA-AACTTTACTTGCT-TTACATAGAAGTTATTTGACTCCTGGTGATTCTTCTTCAGGTTGGACAGCTGGTGCTGCAGCTTATTAT-GTGGGTTATCTTCAACCTAGG-ACTTTTCTATTAAAATATAATGAAAATGGAACCATTACAGATGCTGTAGACTGTGCACTTGACCCTCTCTCAGAAACAAAGTGTACGTTGAAATC-CTTCACTGTAGAAAAAGGAATCTATCAAACTTCTAACTTTAGAGTCCAACCAACAGAATCTATTGTTAGATTTCCTAATATTACAAACT-TGTGCCCTTTTGGTGAAG-TTTTTAACGCCACCAGATTTGCATCTGTTTATGCTTGG-AACAGGAAG-AGAAT--CAGCAACTGTGTTGCTGATTATTCTGTCCTATATAATTCCGCATCATTTTCCACTTTTAAGTGTTATGGAGTGTCTCCTACTAAATTAAATGATCTCTGCTTTACTAATGTCTATGCAGATTCATTTGTAATTAGAGG-TGATGAAGTCAGACAAATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAAATTACCAGATGATTTTACAGGCTGCGTTATAGCTTGGAATTCTAACAATCTTGATTCTAAGGTTGGT-GGTAATTATAAT-TACCTGTATAGATTGTT-TAGGAAGTCTAATCTCAAACCTTTTGAGAGAGATATTTCAACTGAAATCTATCA-GGCCGGTA-GCACACCTTGTAATGGTGTTGAAGGTTTTAATTGTTACTTTCCTTTACAATCATATGGTTTCCAACCCACTAATGGTGTTGGTTACCAACCATACA-GAGTAGTAGTACTT-TCTTTTGAACTTCTACATGCACCAGCAACTGTTTGTGGACCTAAAA--AGTCTACT-AATTTGGTTAAAAA-CAAATGTGTCAATTTCAACTTCAATGGTTTAACAGG-CACAGGTGTTCTT-ACTGAGTCTAACAAAAAGTTTCTGCCTTTCCAACAATTTGGCAGAGA--CATTGCTGACACTACTGATGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTA-ACCAGGTTGCTGTTCTTTATCAGGATGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGT-TTAATAGGGGCTGAACATGTCAACAACT-CATATGAGTGTGACATACCCA-TTGGTGCAGGTATATGCGCTAGTTATCAGACT-CAGACTAATTCTCCTCGGCGGGCACGTAGTGTAGCTAGTCAATCCATCATTGCCTACACTATGTCACTTGGTGCAGAAAATTC-AGTTGCTTACT-CTAATAACTCTATTGCCATACCCACAAATTTTACTATTAGTGTTACCACAGAAATTCTACCAGTGTCTATGACCAAGACATCAGTAGATTGTACAATGTACATTTGTGGTGATTCAACT-GAATGCAGCAATCTT-TTGTTGCAATATGGCAGTTTTTGTACACAATTAAACCGTGCTTTAACTGGAATAGCTGTTGAACAAGACAAAAACACCCAAGAAGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCT-ACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGTGATTGCCTTGG-TGATATTGCTGCTAGAGACCTCATTTGTGCACAAAAGTTTAACGGCCTTACTGTTTTGCCACCTTTGCTCACAGATGAAATGATTGCT-CA-ATACACTTCTGCACTG-TTAGCGGGTACAATCACTTCTGGTTGGACCTTTGGTGCAGGTGCTGCA-TTACAAATACCATTTGCTATGCAAATGGCTTATAGGTTTAATGGTATTGGAGTTACACAGAATGTTCTCTATGAGAACCAAAAATTGATTGCCAACCAATTTAATAGTGCTATTGGCAAAATTCAAGACTCACTTT--CTTCCACAGCAAGTGCACTTGGAAAACTT-CAAGATGTGGTCAACCAAAATGCACAAGCTTTAAACACGCTTGTTAAACAACTTAGCTCCAATTTTGGTGCAATTTCAAGTGTTTTAAATGATATCCTTTCACGTCTTGACAAAGTTGAGGCTGAAGTGCAAATTGATAGGTTGATCACAGGCAGACTTCAAAGTTTGCAGACATATGTGACTCAACAATTAATTAGAGCTGCAGAAATCAGAGCTTCTGCTAATCTTGCTGCTACTAAAATGTCAGAGTGTGTACTTGGACAATCAAAAAGAGTTGATTTTTGTGGAAAGGGCTATCATCTTATGTCCTT-CCCTCAGTCAGCACCTCATGGTGTAGTCTTCTTGCATGTGACTTATGTCCCTGCACAAGAAAAGAACTTCACAACTGCTCCTGCCATTTGTCATGATGGAAAAGCACACTTTCCTCGTGAAGGTGTCTTT-GTTTCAAATGGCACACACTGGTTTGTAACACAAAGGAATTTTTATGAACC-ACAAATCATTACTACAGACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAA-TTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGAT-TTAGGTGACATCTCTGGCATTAATGCTTCAGTTGTAAACATTCAAAAAGAAATTGACCGCCTCAATGAGGTTGCCAAGAA-TTTAAATGAATCTCTCATCGATCTCCAAGAACTTGGAAAGTATGAGCAGTATATAAAATGGCCATGGTACATTTGGCTAGGTTTTATAGCTGG-CTTGATTGCCATAGTAATGGTGACAATTATGCTTTGCTGTATGACCAGTTGCTGTAGTTGTCTCAAGGGCTGTTGTTCTTGTGGATCCTGCTGCAAATTTGATGAAGACGACTCTGAGCCAGTGCTCAAAGGAGTCAAATTACATTACACATAAACGAACTTATGGATTTGTTTATGAGAATCTTCACAATTGGAACTGTAACT-TTGAAGCAAGGTGAAATCAA-GGATGCTACTCCTTCAGATTTTGTTCGCGCTACTGCAACGATACCGATACAAGCCTCACTCCCTTTCGGATGGCTTATTGTTGGCGTTGCACTTCTTGCTGTTTTTCAGAGCGCTTCCAAAATCATAACCCTCAAAAAGAGATGGCAACTAGCACTCTCCAAGGGTGTTCACTTTGTTTGCAACTTGCTGT-TGTTGTTTGTAACAGTTTACTCACACCTT-TTGCT-CGTTGCTGCTGGCCTTG-AAGCCCCTTTTCTCTATCTTTATGCTTTAGTCT-ACTT-CTTGCAGAGTATAAACTTTGTAAGAATAATAATGAGGCTT-TGGCTT-TGCTGGAAATGCCGTTCCAAAAACCCATTACTTTATGATGCCAACTATTTTCTTTGCTGGCATACTAATTGTTACGACTATTGTATACCTTACAATAGTGTAACTTCTTCAATTGTCATTACTTCAGGTGATGGCACAAC-AAGTCCTATTTCTGAACATGACTACCAGATTGGTGGTTATACTGAAAAATGGGAATCTGGAGTAAAAGACTGTGTTGTATTACACAGTTACTTCACTTCAGACTATTACCAGCTGTACTCAACTCAATTGAGTACAGACACTGGTGTTGAACATGTTACCTTCTTCATCTACAATAAAATTGTTGATGAGCCTGAAGAACATGTCCAAATTCACACAATCGACGGTTCATCCGGAGTTGTTAATCCAGTAATGGAACCAATTTATGATGAACCGACGACGACTACTAGCGTGCCTTTGTAAGCACAAGCTGATGAGTACGAACTTATGTACTCATTCGTTTCGGAAGAGACAGGTACGTTAATAGTTAATAGCGTACTTCTTTTTCTTGCTTTCGTGGTATTCTTGCTAGTTACACTAGCCATCCTTACTGCGCTTCGATTGTGTGCGTACTGCTGCAATATTGTTAACGTGAGTCTTGTAAAACCTTCTTTTTACGTTTACTCTCGTGTTAAAAATCTGAATTCTTCTAGAGTTCCTGATCTTCTGGTCTAAACGAACTAAATATTATATTAGTTTTTCTGTTTGGAACTTTAATTTTAGCCATGGCAGATTCCAACGGTACTATTACCGTTGAAGAGCTTAAAAAGCTCCTT-GAACAATGGAACCTAGTAATAGGTTTCCTATTCCTTACATGGATTTGTCTTCT-ACAATTTGCCTATGCCAACAGGAATAGGTTTTTGTATATAATTAAGTTAATTTTCCTCTGGCTG-TTATGGCCAGTAACTTTAGCTTGTTTTGTGCTTGCTGCTGTTTACAGAATAAATTGGATCACCGGTGGAATTGCTATCGCAATGGCTTGTCTTGTAGGCTTGATGTGGCTCAGCTACTTCATTGCTTCTTTCAGACTGTTTGCGCGTACGCGTTCCATGTGGTCATTCAATCCAGAAACTAACATTCTTCTCAACGTGCCACTCCATGGCACTATTCTGACCAGACCGCTTCTAGAAAGTGAACTCGTAATCGGAGCTGTGATCCTTCGTGGACATCTTCGTATTGCTGGACACCATCTAGGACGCTGTGACATCAAGGACCTGCCTAAAGAAATCACTGTTGCTACATCACGAACGCTTTCTTATTACAAATTGGGAGCTTCGCAGCGTGTAGCAGGTGACTCAGGTTTTGCTGCATACAGTCGCTACAGGATTGGCAACTATAAATTAAACACAGACCATTCCAGTAGCAGTGACAATATTGCTTTGCTTGTACAGTAAGTGACAACAGATGTTTCATCTCGTTGACTTTCAGGTTACTATAGCAGAGATATTACTAATTATTATGAGGACTTTTAAAGTTTCCATTTGGAATCTTGATTACATCATAAACCTCATAATTAAAAATTTATCTAAGTCACTAACTGAGAATAAATATTCTCAATTAGATGAAGAGCAACCAATGGAGATTGATTAAACGAACATGAAAATTATTCTT-TTCTTGGCACTGATAACACTCGCTACTTGTGAGCTTTATCACTACCAAGAGTGTGTTAGAGGTACAACAGTACTTTTAAAAGAACCTTGCTCTTCTGGAACATACGAGGGCAATTCACCATTTCATCCTCTAGCTGATAACAAATTTGCACTGACTTGCTTTAGCACTCAATTTGCTTTTGCTTGTCCTGACGGCGTAAAACACGTCTATCAGTTACGTGCCAGATCAGTTTCACCTAAACTGTTCATCAGACAAGAGGAAGTTCAAGAACTT-TACTCTCCAATTTTTCTTATTGTTGCGGCAATAGTGTTTATAACACTTTGCTTCACACTCAAAAGAAAGACAGAATGATTGAACTTTCATTAATTGACTTCTATTTGTGCTTTTTAGCCTTTCTGCTATTCCTTGTTTTAATTATGCTTATTATCTTTTGGTTCTCACTTGAACTGCAAGATCATAATGAAACTTGTC-ACGCCTAAACGAACATGAAATTTCTTGTTTTCTTAGGAAT-CATCACAACTGTAGCTGCATTTCACCAAGAATGTAGTTTACAGTCATGTACTCAACATCAACCATATGTAGTTGATGACCCGTGTCCTATTCACTTCTATTCTAA-ATGGTATATTAGAGTAGGAGCTAGAAAATCAGCACCTTTAATTGAATTGTGCGTGGATGAGGCT-GGTTCTAAATCACCCATTCAGTACATCGATATCGGTAATTATACAGTTTCCTGTT-TACCTTTTACAATTAATTGCCAGGAACCTAAATTGGGTAGTCTTGTAGTGCGTTGTTCGT-TCTATGAAGACTTTTTAGAGTATCATGACGTTCGTGTTGTTTTAGATTTCATCTAAACGAACAAACTAAAATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCA-TACAATGTAACACAAGC-TTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTA-CACAGGTGCCATCAAATTGGATGACAAAGATCCA-AATTTCAAAGATCAA-GTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACT-CAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAAACTCATGCAGACCACACAAGGCAGATGGGCTATATAAACGTTTTCGCTTTTCCGTTTACGATATATAGTCTACTCTTGTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTTAACTTTAATCTCACATAGCAATCTTTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACATTTTCACCGAGGCCACGCGGAGTACGATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGAAGAGCCCTAATGTGTAAAATTAATTTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAATGACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(alignments[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_alignment(alma, thin=50):\n",
    "    plt.figure(figsize=(20, 5))\n",
    "    nmatch = 0\n",
    "    for i in range(len(alma)):\n",
    "        if alma[i] == '|':\n",
    "            nmatch += 1\n",
    "        if ((i+1) % thin) == 0:\n",
    "            plt.plot((i, i), (0, 1), c='g', alpha=((nmatch - 0.25)/thin))\n",
    "            nmatch = 0\n",
    "    plt.show()\n",
    "    return\n",
    "\n",
    "def plot_alignment2(alma, thin=500):\n",
    "    plt.figure(figsize=(20, 5))\n",
    "    nmatch, x, y = 0, [], []\n",
    "    for i in range(len(alma)):\n",
    "        if alma[i] == '|':\n",
    "            nmatch += 1\n",
    "        if ((i+1) % thin) == 0:\n",
    "            x.append(i)\n",
    "            y.append(nmatch/thin)           \n",
    "            nmatch = 0\n",
    "    plt.plot(x, y)\n",
    "    plt.show()\n",
    "    return  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AT-TAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAA\n",
      "||-|.|-|||||.||||||||||||||||||||||||.||||.||||||||||||||||||||||||||||||||\n",
      "ATATTA-GGTTTTTACCTTCCCAGGTAACAAACCAACTAACTCTCGATCTCTTGTAGATCTGTTCTCTAAACGAA\n"
     ]
    }
   ],
   "source": [
    "als2 = str(alignments[0])[:len(str(alignments[0])) // 3]\n",
    "alma = str(alignments[0])[len(str(alignments[0])) // 3 : 2 * len(str(alignments[0])) // 3]\n",
    "als1 = str(alignments[0])[2 * len(str(alignments[0])) // 3:]\n",
    "\n",
    "print(als1[:75])\n",
    "print(alma[:75])\n",
    "print(als2[:75])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8812907928304012"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alma.count('|') / len(alma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABIQAAAEyCAYAAACLeQv5AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3X2QXed9H/bvs+/LXQDEAguQ4BsoAjRfSkZWOdS7qHeRykRsp25GmmnsZtyo01iddJJ2qrQdx1X6R51M05nMqHGVsccv01hWXafl1JQlmpIlWxFpQbXsiGQk7AJ8ASgCIHYBEMt9x+kf957Du7uAAJFLAtT5fGZ2ePfc557znOf8nt957o93L0pVVQEAAACgPfoudwcAAAAAeGMpCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsMXK4D79y5s9q7d+/lOjwAAADAT53vfve7L1ZVNXmxdpetILR3794cOHDgch0eAAAA4KdOKeWZS2nnT8YAAAAAWkZBCAAAAKBlFIQAAAAAWkZBCAAAAKBlFIQAAAAAWkZBCAAAAKBlFIQAAAAAWuaiBaFSym+UUo6XUr5/gedLKeWfl1KmSil/VUp52+Z3EwAAAIDNcimfEPrNJPf/mOcfSLK/+/PpJP/itXcLAAAAgNfLwMUaVFX1zVLK3h/T5MEkv11VVZXksVLK1aWUa6uq+tEm9fGK9rlvfC5/NPVHeef178z40HiS5OzS2STJ8MBwHj30aN5z43syOjjavOaPD/1x3n3DuzM8MJwkGRsc6+zjhndm2/C2JMnM/Ey+9dy38uG3fDhLq0tJktGB0Tx6+NG876b3Zevw1qZd7euHv5733fS+Ncf6+tNfz3033ZeVcytJkqH+oXzjmW/kg3s/mNVqtdn26KFH896b3tu8bnRgNF+e+nLeft3bm/NaPrecP3v2z/Lht3w4Y4NjSZJz1bkkyWq1mj88+Ie5d8+9TfvVajX/5rl/k4/e8tGcmDvR7Psbz3wj9910X/N7f19//uzZP8tH3vKRnFk80zn+4Gj+5Ok/yfv3vj+LK4tJkvGh8Xzt8Ndy3977Mrc0lyTZedXO5lh/fOiP854b35O+0qlzbhve1oxr3W50YDQHnj+Qj9zykbz48oud155bzcMHH869192b5XPLSZLBvsGm7/X49/rq9Ffz7hvfncG+wWbbnz77p/nQzR9Kf19/s+3LB7+ct1//yhgmaca6HruVcyv55jPfzAdv/mDT96H+oeb8F1YWkiQjAyPNc48ceiTvueE9TX+TNO3rdgsrCxvib355Pt8+8u3cv+/+Jk7X9L30N8eoz3G9vtLXxFXvsdYff/XcanO95pfnO9dkZFvTbvVcJ/6Wzy3nm898c83YDfUPJUnOLJ7ZcA79pb+ZB3Ucrlar+er0V/OuG97V7Hd8aLxpV8f/QN9Ac/x6W5LmWtdjMtA30IzXN5/5Zt6/9/1N3wb6Bpo+1dewN15riyuLG2J9eGA4Xzv8tbzvpvetiYmkkzfqOVybGJ3Io4cfzXtvfG8zJjPzM01/51fmm3Ot5/XccmdurJxbaeZVnS/OLJ7ptLv5g80x6vhK0szvetv2ke2deX3dvc2YDPYNdvZ7y0eaeT06OLohr40MjDTXpD7XhZWF5lrX17O+Xr3XsD5WX+nLnz7zp/nQWz60pl091vX4L64sNmNS57UkzZjUxz/xcqe/Q/1DzbjWxofGm/3Wc35+eb7Jw3UM9/f1b5ivS6tLzbWr+7R1eGsT//W82j66PUmyvLq84fi9c76Ow+2j2/ONp7+R+/bel+Nzx5u2dbvefFGfax0ng/2D+fLUl/OO697RnH8dL0ny2JHH8sC+B5rxOj53vNlvb16rz6He79LqUtOuvl+dXjzd5IQ6X0+MTiRJ5pbn8u3nvp2P7ftYM4Zbh7duyG9Lq0tNXNXjP9Q/lG8+8818YO8Hmv5sHd6aRw8/mnfd8K6MDnRi6OzS2eb49fn0Xq+vHf5a7rvpvmYeJMnDBx/O269/+5o8sD4P9vf1N+Na32u3jWzbkGvqmL9QvNbbeufV+uufJAeeP5C/futfz/MvPZ+kM1++8cw38oG9H2jOdX5lPn9+9M/z8f0fb/ZX9/fs0tnOdd3/QHP8JJ04uP4da9Yo9XzpVd9Xeufm+fpZX7ve+2U9XuvncJI8fuTxfHz/x3Ns7liSTmx8ZeoreecN72zGf3TglRxax9pQ/1DT9/ra9cZG3d/6uf6+/ibm633Mr8w3ebA21D+Ux48+nvv33Z/Z+dmmXT0nZhc623aP7c4jhx7JO69/Z5NXm/0uzzfx2uvbR76dj93ysRw7e2zDseqYPLN4prmG9Zj0l/4N16Q3J9ZrmPqazy/PN+PVe/71Oqi2e2x3/uCpP8hfu+avNfuox+5d17+rGc89W/bkDw/+Ye7Zc0+Tf3rvUfX1qnNOf+lvjt+75qnnS73tzOKZ5r5Sx+n8ynwTV3UOmVuea+4NtTq+xgbHmn1cM35NkuS5M8/ley98Lw/+zIPNseaW5vK1w1/LO294ZT0+Mz+Tx48+no++5aPN2qSOk7dd87YmTy2tLuXbR769Jk88d+a5JMmxs8ea61rr7+vPd45+Jx/b97Fmrr348ot57Mhj+egtH23WsttHt+exI4+tyWHXbrk2333+u/nZa382T514KknWrNG+9ey38oGbP9DsY3ZhtplD9bnW66v33vTe3Lbztk67+dl8dfqrefv1b18Tp/Varh7/1Wq1uU71mNQWVhaaPFTH8OnF05114y33Z2yos+Y6Pne8uSa9ueaRQ4901mvdfDXUP7ThHlKf19jgWPNc7/uceh+1nVftbOK6fu3iymIeP/p4Htj3QNOuN4bq8RzsG2zuDYP9g2uOX5/r37j1bzTze355vlk31vey4YHhZn1TG+wf7Ix/T27sK33NHO6N3fq9VH2fqPPl8rnlJr/3zo16fVHni9HB0Q1rzr7St2Etf6F1YNJ5P1TPud77QD3WvWuu9WuJ3ntzfazRgdEN74O2Dm9tttV5ql6jz8zPNP09vXC6Oa9Hph/Ju254V3NtklfuL739rOOvvr+fb1t/6d/Qp941ch1D9bj2lb7O+uLmtffXx488ngf2P5DPvuezaZOLFoQuwXVJnuv5/Uh324aCUCnl0+l8iig33njjJhz68nvyxJM5NHsoY0NjaxbHSecN0aFTh7J1ZOuaG+uh2UPZMrylScTbhrdlenY6Y0NjzQ372NyxHJo9lCdOPNEE7/jQeKZnp7N9dHt2jO5Ikrxw9oVmv4dPHc7Vo1evOdb0zHQmRieyvNpJTqODozk0eyi7x3Y3k2i4fzjTs9OZuGqied3Y4FimZ6czOjjanNfi6mKmZ6ez58SeZludbFar1UzPTGdkYKR5brVazdTMVG7cdmOOvnR0TT/rN0ZJZxLX53py/uSac91x1Y4msWwb3pZDs4eyc2xnTi2cSpJct+W6ph/TM9PZNrKtSWwTIxOZmp3K1pGt2TO+Z81+bz9ze46eOdq89uDMwYwNjWVxdbEZk7rv9bZe07PT2TqyNcP9w822qZmp7NmyZ81CeGp2KqODo7l65Oo1r524aqIZu+XV5RyaPZTJscnmtfXxd43tahJ7nVhHBkYyPTOd7SPb1/StHq+63dzy3Ib4O7t0NodmD+XJE082Y5h0YvKa8Wuam9LIwEhzjusN9A00cdV7rPXHXzm30lyv+iayY3RHc171+ddxdc34Nc3513Pj5PzJDefQX/qbedAbh9Oz0xkfGm/iup5X20e3N/E/2D/Y9LPeVo9d75jUN6ezS2dz+NThTByfaMZmsH+w6VN9Dr3xWltYWdgQ6yMDIzk0eyhXj1y9JiaS5NTCqWYO13aP7c7UzFQmRieaMTk2dyyHTnXmSz2uV49cnemZ6Vw7fm2Tf+q4euLEE02+ODl/shnrWh1f9TjsObGn2bbrql2ZmpnKyMBIMyZ1vrjhxRuaeT0+NL4hr9U5ZHxovDnXOk6uGb+muZ71GPZew/pYA30DmZ6dzrUnrl2Ta+r5Um9bWFloxqS3IFTnujpO6v7WebDeXo9hHZv1vKrnS+9Y1/mqd77OL883167u047RHU3817Gze2x309+pmakm39fjenDmYK4Zv6a5hvX137NlT5498+ya67Tjqh1r8kV9/eucNDIwkqmZqWwZ2rLmTVdtamYqUzNTTX+fPfNss9/evFafQ71tcXWxOf+Jkc49Y2ZhpskJdb6uz/X04ulMzU7lphdvWpsHZjvnVfdpYWWhiaveIvjBmYO5dvzapj87r9qZqZmpTF412cTQqYVTmZqZWnNf6y3uTs9OZ+dVO5t5UJ//VUNXrckD6/NgneuuHb+2eeO+Y3THhlxTx/yF4rXetn5e9V7/JDk4czCHZw9nena6uV7TM9PZNbZrTa6dmpnKwZMHm/3V/a3H+uDJg2sKXdMz09kyvGXNGqW+v/Wq7yu9c7POP2vazU5ncmxyzf2y9/x753DSuQ/+8OQP88zpZ5Ik14xfk6nZqYwPjzfjX+fQybHJZjzr++DY0Fhz7Xpjo+5v/dxA30AT8/U+6pxz/YvXN/2p59rPzP5Ms4aqx3VqZqq51nNLnW07RnesWdclnfiq47XX1MxU9m3f15xrHcP7Z/Y3MXly/mQOzhzM1MxUM6/7S/+Ga1KvobaNbGvWMPU1r48/OTa54fx74+rl5ZdzcOZgBvoHmn2MDIzk4MmDmRiZaK7h4upiDs4czPjQeHOuvfmxvl51zqnvw715MEkzX+pt9T1nz4lX1kZzy3NN/q37enrx9IZ7U923bcPbMjU7lRtPvrIe++HJH2ZqZirPnXmu2W+dB3aN7cq2kU7fj509lqmTU9k/sb+J63pebx3amt3ja3Py3qv3Nn2anunMw2dOP5OpmancfPXNTd8G+gY6MXTqZ5q5dvTM0RycOZh9E/ua/+F4zfg1mZqZylu2v2XNa589/Wz2bNmTqZmpJFmzRpuancpNV9/U7OP4y8ebOVSfaz03J8cms2VoS+dc545tWC/U+9s+un3Na+vrVOeV2tzyXJOH6hh+8eUXMz0znae2PdXERH2/2HNiz5pcc2j2ULYOv7Jeq9eSvfO6XufXa7TeXFKvG3rXTXvG9zRxXb+2vl889eJTTbveGKqPX+eQ3WO7Nxy/Ptepmakm5ut5tX10+5pcU7/Pqw33Dzdrnt7rWq9DemO3fi/Vm9eS7tq3m997i/r1/bW30LN+zTnQN7BhLX+hdWDyyn1zYnRizX2gvl69a6563dT7P3zrXFfvt75Ove+D6vv69tHtTZ6qj1W/p33yxJNr3ucdOnUoVw1dtWHNMTE60awbTy2cauKvPv/zbavvP719Wr9G3jaybc3a4PCpw9lxbMeG+ffvXvx3aZs39Eulq6r6QlVV91RVdc/k5OQbeWgAAAAAujajIHQ0Se//Irm+uw0AAACAK9BmFIQeSvLz3X9t7B1JTrfl+4MAAAAA3owu+h1CpZTfTfL+JDtLKUeS/KMkg0lSVdWvJXk4yceTTCV5Ocnffr06CwAAAMBrdyn/ytinLvJ8leSXNq1HAAAAALyu3tAvlQYAAADg8lMQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGgZBSEAAACAllEQAgAAAGiZSyoIlVLuL6X8oJQyVUr57Hmev7GU8vVSyl+UUv6qlPLxze8qAAAAAJvhogWhUkp/ks8neSDJHUk+VUq5Y12z/yHJl6qq+tkkn0zyv212RwEAAADYHJfyCaF7k0xVVXWoqqqlJF9M8uC6NlWSrd3H25I8v3ldBAAAAGAzDVxCm+uSPNfz+5Ekb1/X5leSfLWU8l8mGUvy4U3pHQAAAACbbrO+VPpTSX6zqqrrk3w8ye+UUjbsu5Ty6VLKgVLKgRMnTmzSoQEAAAD4SVxKQehokht6fr++u63XLyb5UpJUVfXtJCNJdq7fUVVVX6iq6p6qqu6ZnJx8dT0GAAAA4DW5lILQd5LsL6XcXEoZSudLox9a1+bZJB9KklLK7ekUhHwECAAAAOAKdNGCUFVVK0k+k+QrSZ5K518Te6KU8rlSyie6zf5Bkr9TSvnLJL+b5D+tqqp6vToNAAAAwKt3KV8qnaqqHk7y8Lptv9zz+Mkk797crgEAAADwetisL5UGAAAA4E1CQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFrmkgpCpZT7Syk/KKVMlVI+e4E2f7OU8mQp5YlSyr/a3G4CAAAAsFkGLtaglNKf5PNJPpLkSJLvlFIeqqrqyZ42+5P8wyTvrqpqtpSy6/XqMAAAAACvzaV8QujeJFNVVR2qqmopyReTPLiuzd9J8vmqqmaTpKqq45vbTQAAAAA2y6UUhK5L8lzP70e623rdmuTWUsq3SimPlVLuP9+OSimfLqUcKKUcOHHixKvrMQAAAACvyWZ9qfRAkv1J3p/kU0n+ZSnl6vWNqqr6QlVV91RVdc/k5OQmHRoAAACAn8SlFISOJrmh5/fru9t6HUnyUFVVy1VVHU7yw3QKRAAAAABcYS6lIPSdJPtLKTeXUoaSfDLJQ+va/N/pfDoopZSd6fwJ2aFN7CcAAAAAm+SiBaGqqlaSfCbJV5I8leRLVVU9UUr5XCnlE91mX0lyspTyZJKvJ/lvqqo6+Xp1GgAAAIBX76L/7HySVFX1cJKH12375Z7HVZK/3/0BAAAA4Aq2WV8qDQAAAMCbhIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMsoCAEAAAC0jIIQAAAAQMtcUkGolHJ/KeUHpZSpUspnf0y7/6iUUpVS7tm8LgIAAACwmS5aECql9Cf5fJIHktyR5FOllDvO025Lkr+X5PHN7iQAAAAAm+dSPiF0b5KpqqoOVVW1lOSLSR48T7t/nORXkyxsYv8AAAAA2GSXUhC6LslzPb8f6W5rlFLeluSGqqr+cBP7BgAAAMDr4DV/qXQppS/JP0vyDy6h7adLKQdKKQdOnDjxWg8NAAAAwKtwKQWho0lu6Pn9+u622pYk/16SPymlPJ3kHUkeOt8XS1dV9YWqqu6pquqeycnJV99rAAAAAF61SykIfSfJ/lLKzaWUoSSfTPJQ/WRVVaerqtpZVdXeqqr2JnksySeqqjrwuvQYAAAAgNfkogWhqqpWknwmyVeSPJXkS1VVPVFK+Vwp5ROvdwcBAAAA2FwDl9KoqqqHkzy8btsvX6Dt+197twAAAAB4vbzmL5UGAAAA4M1FQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFrmkgpCpZT7Syk/KKVMlVI+e57n/34p5clSyl+VUh4tpdy0+V0FAAAAYDNctCBUSulP8vkkDyS5I8mnSil3rGv2F0nuqarq7iS/n+SfbHZHAQAAANgcl/IJoXuTTFVVdaiqqqUkX0zyYG+Dqqq+XlXVy91fH0ty/eZ2EwAAAIDNcikFoeuSPNfz+5Hutgv5xSRfPt8TpZRPl1IOlFIOnDhx4tJ7CQAAAMCm2dQvlS6l/CdJ7knyT8/3fFVVX6iq6p6qqu6ZnJzczEMDAAAAcIkGLqHN0SQ39Px+fXfbGqWUDyf575PcV1XV4uZ0DwAAAIDNdimfEPpOkv2llJtLKUNJPpnkod4GpZSfTfK/J/lEVVXHN7+bAAAAAGyWixaEqqpaSfKZJF9J8lSSL1VV9UQp5XOllE90m/3TJONJ/s9SyvdKKQ9dYHcAAAAAXGaX8idjqarq4SQPr9v2yz2PP7zJ/QIAAADgdbKpXyoNAAAAwJVPQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZRSEAAAAAFpGQQgAAACgZS6pIFRKub+U8oNSylQp5bPneX64lPJ73ecfL6Xs3eyOAgAAALA5LloQKqX0J/l8kgeS3JHkU6WUO9Y1+8Uks1VV7Uvyvyb51c3uKAAAAACb41I+IXRvkqmqqg5VVbWU5ItJHlzX5sEkv9V9/PtJPlRKKZvXTQAAAAA2y8AltLkuyXM9vx9J8vYLtamqaqWUcjrJjiQv9jYqpXw6yaeT5MYbb3yVXb6y3DF5R549/Wzu3nV3xofGkyRnl84mSYYHhnP87PHcvevujA6ONq/50Us/yt277s7wwHCSZGxwLM+ceiZ3774724a3JUl2j+3OS4sv5c7JO7O0upQkGR0YzQtnX8hdu+7K1uGtSZJdY7ua/Z58+eSGYx2fO567dt2VlXMrSZKh/qG8+PKLuXPyzqxWq822H730o9w5eWfzutGB0Tx96uk157V8bjmnF07njsk7MjY4liQ5V51LkqxWq5menV7TfrVazemF07lt523ZMbqj2ffs/Gzu3nV383t/X3/OLJ7JnZN35szimc7xB0dz8uWTuWvXXVlcWUySjA+N59jZY7lz8s7MLc0lSXZetbM51vMvPZ+7dt2VvtKpc24b3panTz2dOyfvbNqNDoxmZn4m12+9PiMDI53XnlvN/on9uXPyziyfW06SDPYNNn2vx7/Xc6efy1277spg32CzbWZ+JrfvvD39ff3NtsOzh3P37lfGpL7+d07e2YzdyrmVnFo4tabvQ/1DOT53PHdM3pGFlYUkafo71D+U5848t6a/9fW/a9ddTbuFlYUN8Te/PJ+5pbncMXlHE6dJcnrxdO6cvDP9pf+VY3TPcb2+0tfEVe+x1h9/9dxqc73ml+c712RkW3Neq+c68bd8bjmz87Od4/e9cvwkObN4ZsM59Jf+Zh7UcbharebImSO5e/fdzX7Hh8abdnX8D/QNNP2styWdOds7JgN9A814nVk4k7t33d30baBvoOlTfQ1747W2uLK4IdaHB4ZzYu7Ehpio+1DP4drE6ESef+n53Dl5ZzMmu8d25+zi2c64rsw353ri5RO5feftmVvuzI2VcyvNvKrzxZnFM3nx5Rdzx+QrH/Ks4ytJM7/rbdtHtnfm9e67mzEZ7BvM7Pzsmnk9Oji6Ia+NDIw016Q+14WVheZa19ezvl6917A+Vl/py8zLM2viZbVazYm5E2vm0OLKYjMmdV5L0uS6+vg7rur0d6h/KC+cfSF3735lrMeHxpvYrOf8/PJ8k4frGO7v698wX5dWl5prV/dp6/DWJv7rebV9dHuSZHl1OUdfOprbd97eHL/u0207b2vicPvo9jz/0vPZv2N/to1sa9rWsdabL+rrX8fJYP9gpman1px/HS91LOyb2NeM17aRbc1+e/NafQ71fpdWl5rxr+9XpxdPNzmhztcToxNJkrnluZxdPJvbdt7WjOHW4a15/qXnc+uOW5tYW1pdauKqHv86D96287amP1uHt2bfxL7s37E/owOdGDq7dDZHXzpQLEw+AAAK20lEQVSa23be1pxP7/V64ewLa+ZBkkzNTG3IA+vzYH9ffzOuu8d2N+O0PtfUMX+heK239c6r9dc/SeaW5nLz9pub/S2sLGRmfqYzX7rnOr/SyeH7d+xv9lf39+zS2by0+FL279jfHD9Jpmen18TB2aWzzf2tV31f6Z2bdf7pVV//3vtlPV7r53CSvLz0cm7dcWsTwxOjE3nm1DNrxn90YLTZbx1rQ/1DOXzq8Jo1T29s1P2tn+vv629ivt7H/Mp8ZudnN8y104unc/P2m3P1yNVNu3pO1Hli99ju7JvYl1t33Nrk1Wa/y/NNvPaaXZjNLRO3NGM91D+UM4tncsvELU1Mnlk8k7mlueyb2Jctw1uaa7j+mqxWq3n29LNr1jD1NZ9fnm/Gq/f8nzn9zJp97B7b3axv6n0M9Q9l/6n9a+bani17sn9i/5r803uPqq9XnXP6S39z/N41Tz1f6m1nFs8095U6TudX5pv8W+eQueW55t5Qq+NrbHAsJ18+mVt33Jprxq9J0plrCysLuWHrDc2xto9sz76JfWvGf+vw1swszOTmq29u1ib9ff25ZeKW7JvY1+SppdWlvHD2hdy07aZmPOvYHB8az5nFM7l1x61N3+r7wN6r9zZzbcvQlpxaOJW3bH9LM9bbR7fn2Nyx3LTtpua1k2OTuXHbjdk93omvJGvWaMfOHluzj90Lu5s5VJ/r6rnO/fK2nbfluq3XJUmuGryqyY29cfr0qafXjP9qtdpcp3pMagsrC00eumrwqiSdvDi3PJfbd96esaHOmmvbyLbmuvbmmudfej537767yVf1+4s7Ju9Ys66rr2v9XO/7nCNnjqyJg51X7Wziun7t4spi5pfn18zr3hiqx3Owb7C5Nwz2d/LU7sXdzbkurixm38S+Zn7PL88368b6XjY8MJxjZ4/lrbvf2hxrsH8wL7784pptfaWvWYf0xm79XqrO63W+XD633OT33rlRry/qfDE6OLphzdlX+jas5S+0Dkw674fqXN57H6ivV++aq1439d7z61xXH2t0YHTD+6Ctw1ubWKvzVL1G3z22u+nv6YXTzXk9f+b5vHX3W5trk3TeS/WuG88unW3ir76/n29bf+nf0KfeNXL9HrEe177Sl1Pzp9bE6/zKfOaX5zfk9TYoVVX9+Aal/FyS+6uq+s+6v/+tJG+vquozPW2+321zpPv7dLfNi+fbZ5Lcc8891YEDBzbhFAAAAABIklLKd6uquudi7S7lT8aOJrmh5/fru9vO26aUMpBkW5KTl9ZVAAAAAN5Il1IQ+k6S/aWUm0spQ0k+meShdW0eSvIL3cc/l+Rr1cU+egQAAADAZXHR7xDqfifQZ5J8JUl/kt+oquqJUsrnkhyoquqhJL+e5HdKKVNJZtIpGgEAAABwBbqUL5VOVVUPJ3l43bZf7nm8kOQ/3tyuAQAAAPB6uJQ/GQMAAADgp4iCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtEypquryHLiUE0meuSwH3zw7k7x4uTsBP4YY5UomPrnSiVGuZOKTK50Y5Ur20x6fN1VVNXmxRpetIPTToJRyoKqqey53P+BCxChXMvHJlU6MciUTn1zpxChXMvHZ4U/GAAAAAFpGQQgAAACgZRSEXpsvXO4OwEWIUa5k4pMrnRjlSiY+udKJUa5k4jO+QwgAAACgdXxCCAAAAKBlFIQAAAAAWkZB6FUqpdxfSvlBKWWqlPLZy90f2qOU8nQp5d+WUr5XSjnQ3TZRSnmklHKw+9/t3e2llPLPu3H6V6WUt/Xs5xe67Q+WUn7hcp0Pb36llN8opRwvpXy/Z9umxWQp5d/vxvxU97XljT1D3swuEJ+/Uko52s2j3yulfLznuX/YjbUflFI+1rP9vPf9UsrNpZTHu9t/r5Qy9MadHW92pZQbSilfL6U8WUp5opTy97rb5VCuCD8mRuVRLrtSykgp5c9LKX/Zjc//sbv9vDFVShnu/j7VfX5vz75+orj9qVFVlZ+f8CdJf5LpJG9JMpTkL5Pccbn75acdP0meTrJz3bZ/kuSz3cefTfKr3ccfT/LlJCXJO5I83t0+keRQ97/bu4+3X+5z8/Pm/EnyviRvS/L9nm2bFpNJ/rzbtnRf+8DlPmc/b56fC8TnryT5r8/T9o7uPX04yc3de33/j7vvJ/lSkk92H/9akv/icp+znzfPT5Jrk7yt+3hLkh9241AO9XNF/PyYGJVH/Vz2n25eG+8+HkzyeDffnTemkvzdJL/WffzJJL/XffwTx+1Py49PCL069yaZqqrqUFVVS0m+mOTBy9wn2u3BJL/VffxbSf6Dnu2/XXU8luTqUsq1ST6W5JGqqmaqqppN8kiS+9/oTvPToaqqbyaZWbd5U2Ky+9zWqqoeqzp37N/u2Rdc1AXi80IeTPLFqqoWq6o6nGQqnXv+ee/73U9afDDJ73df3xvrcFFVVf2oqqr/r/v4pSRPJbkucihXiB8Toxcij/KG6ebCs91fB7s/VS4cU7259feTfKgbgz9R3L7Op/WGUhB6da5L8lzP70fy4xMjbKYqyVdLKd8tpXy6u213VVU/6j5+Icnu7uMLxaoY5vW2WTF5Xffx+u3wWn2m+yc3v1H/OU5+8vjckeRUVVUr67bDT6z7pws/m87/4ZZDueKsi9FEHuUKUErpL6V8L8nxdIrh07lwTDVx2H3+dDox2Nr3TApC8Obznqqq3pbkgSS/VEp5X++T3f8DWF2WnsF5iEmuQP8iyS1J3prkR0n+l8vbHdqulDKe5P9K8l9VVXWm9zk5lCvBeWJUHuWKUFXValVVb01yfTqf6LntMnfpTUVB6NU5muSGnt+v726D111VVUe7/z2e5F+nk/iOdT8Wnu5/j3ebXyhWxTCvt82KyaPdx+u3w6tWVdWx7gLyXJJ/mU4eTX7y+DyZzp/sDKzbDpeslDKYzhvt/6Oqqj/obpZDuWKcL0blUa40VVWdSvL1JO/MhWOqicPu89vSicHWvmdSEHp1vpNkf/fby4fS+UKqhy5zn2iBUspYKWVL/TjJR5N8P534q/9FkV9I8v90Hz+U5Oe7/yrJO5Kc7n4E/StJPlpK2d79iO9Hu9tgs2xKTHafO1NKeUf3b7x/vmdf8KrUb7S7/sN08mjSic9Pdv8VkpuT7E/nC3nPe9/vfnLj60l+rvv63liHi+rmtV9P8lRVVf+s5yk5lCvChWJUHuVKUEqZLKVc3X08muQj6XzP1YViqje3/lySr3Vj8CeK29f/zN44AxdvwnpVVa2UUj6Tzs23P8lvVFX1xGXuFu2wO8m/7tybM5DkX1VV9UellO8k+VIp5ReTPJPkb3bbP5zOv0gyleTlJH87Saqqmiml/ON0klySfK6qqkv90lVYo5Tyu0nen2RnKeVIkn+U5H/O5sXk303ym0lG0/kXcr78Op8SP0UuEJ/vL6W8NZ0/w3k6yX+eJFVVPVFK+VKSJ5OsJPmlqqpWu/u50H3/v03yxVLK/5TkL9J54wSX6t1J/laSf9v9Dowk+e8ih3LluFCMfkoe5QpwbZLfKqX0p/Nhly9VVfX/llKezPlj6teT/E4pZSqdf3Dik8mrjtufCqVTEAMAAACgLfzJGAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDLKAgBAAAAtIyCEAAAAEDL/P+eOAbB35iIWAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1440x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_alignment(alma[:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'alma' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-2-6b9ec4432eca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mplot_alignment2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'alma' is not defined"
     ]
    }
   ],
   "source": [
    "plot_alignment2(alma[:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAAC\n",
      "AGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTAGCTGTCGCTCGGCTGCATGCCTAGTGCACCTACGCAGTA\n",
      "11676.0\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__cmp__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_format_generalized',\n",
       " '_format_psl',\n",
       " 'aligned',\n",
       " 'format',\n",
       " 'path',\n",
       " 'query',\n",
       " 'score',\n",
       " 'target']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(alignments[0].query[:75])\n",
    "print(alignments[0].target[:75])\n",
    "print(alignments[0].score)\n",
    "#print(alignments[0].path)\n",
    "#print(alignments[0].aligned)\n",
    "dir(alignments[0])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Final exercise (5/10 points): The misterious patient\n",
    "\n",
    "A few weeks ago, a patient was admitted to the hospital where you work. The patient was coughing, and had difficulty breathing, as well as a fever. Your colleagues Oscar and Maria, from the genomics platform at the hospital, have been able to isolate a viral genome from the patient; this virus is thought to be responsible for the disease. **You are expected to identify what the virus is and, thus, provide a diagnosis**. You are also expected to compare the virus' genome with the genome of similar viruses to try to **understand where the disease came from and how is it different from known diseases**. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **Open and read genome of the virus** using Biopython. The genome is available as a FASTA file from `Files/patient.fasta`). \n",
    "2. Go to the BLAST [web page](https://blast.ncbi.nlm.nih.gov/Blast.cgi) and **BLAST the sequence manually**. BLAST against the standard nucleotide collections using the best algorithm for the alignment of highly similar sequences. After blasting, obtain, show (as a screenshot), and discuss the following visualizations:\n",
    " - A **graphic summary of the alignments**. \n",
    " - A **distance tree** of the results.\n",
    "\n",
    "\n",
    "3. BLAST the sequence again, **using Biopython** this time. Be aware that this can take several minutes to complete (~10 minutes), so be patient. Upon completion of the BLAST search, immediately **save the BLAST results in a file** named `Files/blast_results.xml`.\n",
    "4. Open the results and **display all alignments with percentage identity larger than 99.5%**. Which is your diagnosis for the patient?\n",
    "5. **Display all alignments with percentage identity larger than 85.0% and smaller than 99.5%**. Which animal do you think may have transmitted the disease to humans?\n",
    "6. Choose one of the sequences with percentage similarity between 85% and 90%. Get the whole **GeneBank entry** for this sequence **from Entrez using Biopython**, and display it. List, also, the coding regions of this genome indicating start and end of each coding sequence (CDS).\n",
    "7. Do a **global pairwise alignment** between the genome in question 6 and the patient's viral genome using the **Needleman-Wunsch algorithm in Biopython**. Visualize the aligment using the `plot_alignment()` function defined above, and explain in which region of the original virus (question 6) has the new virus mutated. For extra credit, write a new `plot_alignment_new()` function that helps you address this question  more precisely (e.g. by marking the start and end of CDSs in the genome of question 6)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 1 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Save the graphic summary of the alignments as Files/blast_align.png and, after Ctrl+Enter this cell, the image should show up here!](Files/blast_align.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Save the distance tree of the alignments as Files/blast_tree.png and, after Ctrl+Enter this cell, the image should show up here!](Files/blast_tree.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Discuss the results here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 3 code here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 4 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 4 here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 5 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 5 here*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 6 code here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Question 7 code here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Answer to question 7 here*"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
